# Comparing `tmp/scikit-bio-0.5.8.tar.gz` & `tmp/scikit-bio-0.5.9.tar.gz`

## filetype from file(1)

```diff
@@ -1 +1 @@
-gzip compressed data, was "scikit-bio-0.5.8.tar", last modified: Fri Dec  9 21:40:48 2022, max compression
+gzip compressed data, was "scikit-bio-0.5.9.tar", last modified: Wed Aug  2 22:34:45 2023, max compression
```

## Comparing `scikit-bio-0.5.8.tar` & `scikit-bio-0.5.9.tar`

### file list

```diff
@@ -1,750 +1,750 @@
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.884321 scikit-bio-0.5.8/
--rw-r--r--   0 runner    (1001) docker     (123)    81366 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/CHANGELOG.md
--rw-r--r--   0 runner    (1001) docker     (123)     1515 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/LICENSE.txt
--rw-r--r--   0 runner    (1001) docker     (123)      335 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/MANIFEST.in
--rw-r--r--   0 runner    (1001) docker     (123)     1329 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/Makefile
--rw-r--r--   0 runner    (1001) docker     (123)     9502 2022-12-09 21:40:48.884321 scikit-bio-0.5.8/PKG-INFO
--rw-r--r--   0 runner    (1001) docker     (123)     7390 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/README.rst
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.664321 scikit-bio-0.5.8/assets/
--rw-r--r--   0 runner    (1001) docker     (123)  2052678 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/assets/.no.gif
--rw-r--r--   0 runner    (1001) docker     (123)    15003 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/assets/horizontal_powered_by.png
--rw-r--r--   0 runner    (1001) docker     (123)    14388 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/assets/horizontal_powered_by.svg
--rw-r--r--   0 runner    (1001) docker     (123)     6461 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/assets/logo-inverted.svg
--rw-r--r--   0 runner    (1001) docker     (123)    12397 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/assets/logo.png
--rw-r--r--   0 runner    (1001) docker     (123)     6461 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/assets/logo.svg
--rw-r--r--   0 runner    (1001) docker     (123)    63684 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/assets/logo_and_powered.pdf
--rw-r--r--   0 runner    (1001) docker     (123)    15865 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/assets/vertical_powered_by.png
--rw-r--r--   0 runner    (1001) docker     (123)    13932 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/assets/vertical_powered_by.svg
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.664321 scikit-bio-0.5.8/ci/
--rw-r--r--   0 runner    (1001) docker     (123)      175 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/ci/aarch64.conda_requirements.txt
--rw-r--r--   0 runner    (1001) docker     (123)       20 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/ci/aarch64.requirements.txt
--rw-r--r--   0 runner    (1001) docker     (123)      128 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/ci/conda_host_env.yml
--rw-r--r--   0 runner    (1001) docker     (123)      195 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/ci/conda_requirements.txt
--rw-r--r--   0 runner    (1001) docker     (123)       30 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/ci/requirements.doc.txt
--rw-r--r--   0 runner    (1001) docker     (123)       44 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/ci/requirements.lint.txt
--rw-r--r--   0 runner    (1001) docker     (123)       26 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/ci/requirements.test.txt
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.664321 scikit-bio-0.5.8/doc/
--rw-r--r--   0 runner    (1001) docker     (123)     7105 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/Makefile
--rw-r--r--   0 runner    (1001) docker     (123)     8419 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/README.md
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.668321 scikit-bio-0.5.8/doc/source/
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.672321 scikit-bio-0.5.8/doc/source/_static/
--rw-r--r--   0 runner    (1001) docker     (123)     2803 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/_static/copybutton.js
--rw-r--r--   0 runner    (1001) docker     (123)     2424 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/_static/style.css
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.672321 scikit-bio-0.5.8/doc/source/_templates/
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.672321 scikit-bio-0.5.8/doc/source/_templates/autosummary/
--rw-r--r--   0 runner    (1001) docker     (123)      109 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/_templates/autosummary/attribute.rst
--rw-r--r--   0 runner    (1001) docker     (123)     1582 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/_templates/autosummary/class.rst
--rw-r--r--   0 runner    (1001) docker     (123)      106 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/_templates/autosummary/method.rst
--rw-r--r--   0 runner    (1001) docker     (123)      444 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/_templates/autosummary/module.rst
--rw-r--r--   0 runner    (1001) docker     (123)     1290 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/_templates/layout.html
--rw-r--r--   0 runner    (1001) docker     (123)       32 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/alignment.rst
--rw-r--r--   0 runner    (1001) docker     (123)    16848 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/conf.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.672321 scikit-bio-0.5.8/doc/source/development/
--rw-r--r--   0 runner    (1001) docker     (123)    24418 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/development/coding_guidelines.rst
--rw-r--r--   0 runner    (1001) docker     (123)     1870 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/development/new_module.rst
--rw-r--r--   0 runner    (1001) docker     (123)       32 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/diversity.rst
--rw-r--r--   0 runner    (1001) docker     (123)      781 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/index.rst
--rw-r--r--   0 runner    (1001) docker     (123)       25 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/io.rst
--rw-r--r--   0 runner    (1001) docker     (123)       31 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/metadata.rst
--rw-r--r--   0 runner    (1001) docker     (123)       31 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/sequence.rst
--rw-r--r--   0 runner    (1001) docker     (123)       28 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/stats.rst
--rw-r--r--   0 runner    (1001) docker     (123)       27 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/tree.rst
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.672321 scikit-bio-0.5.8/doc/source/user/
--rw-r--r--   0 runner    (1001) docker     (123)     3167 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/user/api_stability.rst
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.672321 scikit-bio-0.5.8/doc/source/user/assets/
--rw-r--r--   0 runner    (1001) docker     (123)    40262 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/user/assets/api-lifecycle.png
--rw-r--r--   0 runner    (1001) docker     (123)       27 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/util.rst
--rw-r--r--   0 runner    (1001) docker     (123)       31 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/doc/source/workflow.rst
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.676321 scikit-bio-0.5.8/licenses/
--rw-r--r--   0 runner    (1001) docker     (123)     1135 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/bx_python.txt
--rw-r--r--   0 runner    (1001) docker     (123)     4951 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/fastq-example-files-readme.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1582 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/flask.txt
--rw-r--r--   0 runner    (1001) docker     (123)     3415 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/ipython.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1574 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/nb-slideshow-template.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1498 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/qiita.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1712 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/scikit-learn.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1580 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/scipy.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1077 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/simde.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1060 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/sphinx-bootstrap-theme.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1557 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/ssw.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1491 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/licenses/verman.txt
--rw-r--r--   0 runner    (1001) docker     (123)      465 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/pyproject.toml
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.680321 scikit-bio-0.5.8/scikit_bio.egg-info/
--rw-r--r--   0 runner    (1001) docker     (123)     9502 2022-12-09 21:40:48.000000 scikit-bio-0.5.8/scikit_bio.egg-info/PKG-INFO
--rw-r--r--   0 runner    (1001) docker     (123)    31861 2022-12-09 21:40:48.000000 scikit-bio-0.5.8/scikit_bio.egg-info/SOURCES.txt
--rw-r--r--   0 runner    (1001) docker     (123)        1 2022-12-09 21:40:48.000000 scikit-bio-0.5.8/scikit_bio.egg-info/dependency_links.txt
--rw-r--r--   0 runner    (1001) docker     (123)      173 2022-12-09 21:40:48.000000 scikit-bio-0.5.8/scikit_bio.egg-info/requires.txt
--rw-r--r--   0 runner    (1001) docker     (123)        6 2022-12-09 21:40:48.000000 scikit-bio-0.5.8/scikit_bio.egg-info/top_level.txt
--rw-r--r--   0 runner    (1001) docker     (123)       38 2022-12-09 21:40:48.884321 scikit-bio-0.5.8/setup.cfg
--rw-r--r--   0 runner    (1001) docker     (123)     5917 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/setup.py
--rw-r--r--   0 runner    (1001) docker     (123)   783002 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/simde-sse2.h
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.680321 scikit-bio-0.5.8/skbio/
--rw-r--r--   0 runner    (1001) docker     (123)     2328 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     1687 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/_base.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.684321 scikit-bio-0.5.8/skbio/alignment/
--rw-r--r--   0 runner    (1001) docker     (123)     6612 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     8579 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/_indexing.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.688321 scikit-bio-0.5.8/skbio/alignment/_lib/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/_lib/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)    32006 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/_lib/ssw.c
--rw-r--r--   0 runner    (1001) docker     (123)     7156 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/_lib/ssw.h
--rw-r--r--   0 runner    (1001) docker     (123)    44831 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/_pairwise.py
--rw-r--r--   0 runner    (1001) docker     (123)     2605 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/_repr.py
--rw-r--r--   0 runner    (1001) docker     (123)   670691 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/_ssw_wrapper.c
--rw-r--r--   0 runner    (1001) docker     (123)    25443 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/_ssw_wrapper.pyx
--rw-r--r--   0 runner    (1001) docker     (123)    81313 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/_tabular_msa.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.688321 scikit-bio-0.5.8/skbio/alignment/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/tests/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)    35055 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/tests/test_pairwise.py
--rw-r--r--   0 runner    (1001) docker     (123)    32597 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/tests/test_ssw.py
--rw-r--r--   0 runner    (1001) docker     (123)   148548 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/alignment/tests/test_tabular_msa.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.692321 scikit-bio-0.5.8/skbio/diversity/
--rw-r--r--   0 runner    (1001) docker     (123)    17124 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)    10602 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/_block.py
--rw-r--r--   0 runner    (1001) docker     (123)    17038 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/_driver.py
--rw-r--r--   0 runner    (1001) docker     (123)   347189 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/_phylogenetic.c
--rw-r--r--   0 runner    (1001) docker     (123)     7018 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/_phylogenetic.pyx
--rw-r--r--   0 runner    (1001) docker     (123)     5686 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/_util.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.692321 scikit-bio-0.5.8/skbio/diversity/alpha/
--rw-r--r--   0 runner    (1001) docker     (123)     2159 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     4650 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/_ace.py
--rw-r--r--   0 runner    (1001) docker     (123)    25177 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/_base.py
--rw-r--r--   0 runner    (1001) docker     (123)     6442 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/_chao1.py
--rw-r--r--   0 runner    (1001) docker     (123)     5557 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/_faith_pd.py
--rw-r--r--   0 runner    (1001) docker     (123)     3669 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/_gini.py
--rw-r--r--   0 runner    (1001) docker     (123)    17448 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/_lladser.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.696321 scikit-bio-0.5.8/skbio/diversity/alpha/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/__init__.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.644321 scikit-bio-0.5.8/skbio/diversity/alpha/tests/data/
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.696321 scikit-bio-0.5.8/skbio/diversity/alpha/tests/data/qiime-191-tt/
--rw-r--r--   0 runner    (1001) docker     (123)      161 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/data/qiime-191-tt/README.md
--rw-r--r--   0 runner    (1001) docker     (123)      102 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/data/qiime-191-tt/faith-pd.txt
--rw-r--r--   0 runner    (1001) docker     (123)      291 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/data/qiime-191-tt/otu-table.tsv
--rw-r--r--   0 runner    (1001) docker     (123)      266 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/data/qiime-191-tt/tree.nwk
--rw-r--r--   0 runner    (1001) docker     (123)     1497 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_ace.py
--rw-r--r--   0 runner    (1001) docker     (123)    10327 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_base.py
--rw-r--r--   0 runner    (1001) docker     (123)     2865 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_chao1.py
--rw-r--r--   0 runner    (1001) docker     (123)     8646 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_faith_pd.py
--rw-r--r--   0 runner    (1001) docker     (123)     2172 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_gini.py
--rw-r--r--   0 runner    (1001) docker     (123)     8393 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_lladser.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.696321 scikit-bio-0.5.8/skbio/diversity/beta/
--rw-r--r--   0 runner    (1001) docker     (123)     1017 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/beta/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)    23491 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/beta/_unifrac.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.696321 scikit-bio-0.5.8/skbio/diversity/beta/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/beta/tests/__init__.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.648321 scikit-bio-0.5.8/skbio/diversity/beta/tests/data/
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.700321 scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/
--rw-r--r--   0 runner    (1001) docker     (123)     1400 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/README.md
--rw-r--r--   0 runner    (1001) docker     (123)      291 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/otu-table.tsv
--rw-r--r--   0 runner    (1001) docker     (123)      266 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/tree.nwk
--rw-r--r--   0 runner    (1001) docker     (123)      950 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/unweighted_unifrac_dm.txt
--rw-r--r--   0 runner    (1001) docker     (123)      982 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/weighted_normalized_unifrac_dm.txt
--rw-r--r--   0 runner    (1001) docker     (123)      950 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/weighted_unifrac_dm.txt
--rw-r--r--   0 runner    (1001) docker     (123)    29880 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/beta/tests/test_unifrac.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.700321 scikit-bio-0.5.8/skbio/diversity/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/tests/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     8358 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/tests/test_block.py
--rw-r--r--   0 runner    (1001) docker     (123)    37085 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/tests/test_driver.py
--rw-r--r--   0 runner    (1001) docker     (123)    10821 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/diversity/tests/test_util.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.704321 scikit-bio-0.5.8/skbio/io/
--rw-r--r--   0 runner    (1001) docker     (123)     8181 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     2744 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/_exception.py
--rw-r--r--   0 runner    (1001) docker     (123)     2615 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/_fileobject.py
--rw-r--r--   0 runner    (1001) docker     (123)     6993 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/_iosources.py
--rw-r--r--   0 runner    (1001) docker     (123)      602 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/_warning.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.712321 scikit-bio-0.5.8/skbio/io/format/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     7483 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/_base.py
--rw-r--r--   0 runner    (1001) docker     (123)     2442 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/_blast.py
--rw-r--r--   0 runner    (1001) docker     (123)    10702 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/_sequence_feature_vocabulary.py
--rw-r--r--   0 runner    (1001) docker     (123)     5795 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/binary_dm.py
--rw-r--r--   0 runner    (1001) docker     (123)    12815 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/blast6.py
--rw-r--r--   0 runner    (1001) docker     (123)    15434 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/blast7.py
--rw-r--r--   0 runner    (1001) docker     (123)    12164 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/clustal.py
--rw-r--r--   0 runner    (1001) docker     (123)    53958 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/embl.py
--rw-r--r--   0 runner    (1001) docker     (123)     1159 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/emptyfile.py
--rw-r--r--   0 runner    (1001) docker     (123)    35631 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/fasta.py
--rw-r--r--   0 runner    (1001) docker     (123)    22117 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/fastq.py
--rw-r--r--   0 runner    (1001) docker     (123)    25100 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/genbank.py
--rw-r--r--   0 runner    (1001) docker     (123)    18541 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/gff3.py
--rw-r--r--   0 runner    (1001) docker     (123)     8088 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/lsmat.py
--rw-r--r--   0 runner    (1001) docker     (123)    20494 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/newick.py
--rw-r--r--   0 runner    (1001) docker     (123)    14421 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/ordination.py
--rw-r--r--   0 runner    (1001) docker     (123)    14164 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/phylip.py
--rw-r--r--   0 runner    (1001) docker     (123)    10094 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/qseq.py
--rw-r--r--   0 runner    (1001) docker     (123)    29559 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/stockholm.py
--rw-r--r--   0 runner    (1001) docker     (123)    13183 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/taxdump.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.720321 scikit-bio-0.5.8/skbio/io/format/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/__init__.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.848321 scikit-bio-0.5.8/skbio/io/format/tests/data/
--rw-r--r--   0 runner    (1001) docker     (123)        9 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast6_custom_minimal
--rw-r--r--   0 runner    (1001) docker     (123)       53 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast6_custom_mixed_nans
--rw-r--r--   0 runner    (1001) docker     (123)      113 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast6_custom_multi_line
--rw-r--r--   0 runner    (1001) docker     (123)       31 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast6_custom_single_line
--rw-r--r--   0 runner    (1001) docker     (123)      143 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast6_default_multi_line
--rw-r--r--   0 runner    (1001) docker     (123)       47 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast6_default_single_line
--rw-r--r--   0 runner    (1001) docker     (123)      141 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast6_invalid_column_types
--rw-r--r--   0 runner    (1001) docker     (123)       36 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast6_invalid_number_of_columns
--rw-r--r--   0 runner    (1001) docker     (123)       95 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast6_invalid_type_in_column
--rw-r--r--   0 runner    (1001) docker     (123)       93 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_custom_minimal
--rw-r--r--   0 runner    (1001) docker     (123)      396 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_custom_mixed_nans
--rw-r--r--   0 runner    (1001) docker     (123)      410 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_custom_multi_line
--rw-r--r--   0 runner    (1001) docker     (123)      377 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_custom_single_line
--rw-r--r--   0 runner    (1001) docker     (123)      424 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_default_multi_line
--rw-r--r--   0 runner    (1001) docker     (123)      324 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_default_single_line
--rw-r--r--   0 runner    (1001) docker     (123)      415 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_invalid_differing_fields
--rw-r--r--   0 runner    (1001) docker     (123)       57 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_invalid_for_sniffer
--rw-r--r--   0 runner    (1001) docker     (123)       53 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_invalid_for_sniffer_2
--rw-r--r--   0 runner    (1001) docker     (123)      144 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_invalid_gibberish
--rw-r--r--   0 runner    (1001) docker     (123)      408 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_invalid_no_data
--rw-r--r--   0 runner    (1001) docker     (123)      410 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_invalid_too_many_columns
--rw-r--r--   0 runner    (1001) docker     (123)      179 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/blast7_invalid_unrecognized_field
--rw-r--r--   0 runner    (1001) docker     (123)     1105 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/embl_constructed
--rw-r--r--   0 runner    (1001) docker     (123)     3923 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/embl_feature_level_record
--rw-r--r--   0 runner    (1001) docker     (123)     3156 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/embl_feature_level_record_no_FT
--rw-r--r--   0 runner    (1001) docker     (123)     6749 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/embl_multi_records
--rw-r--r--   0 runner    (1001) docker     (123)     4136 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/embl_single_record
--rw-r--r--   0 runner    (1001) docker     (123)      272 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/embl_single_record_lower
--rw-r--r--   0 runner    (1001) docker     (123)     4022 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/embl_single_record_simple
--rw-r--r--   0 runner    (1001) docker     (123)      344 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/embl_single_record_upper
--rw-r--r--   0 runner    (1001) docker     (123)     2333 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/embl_uniprot_record
--rw-r--r--   0 runner    (1001) docker     (123)      274 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/embl_w_beginning_whitespace
--rw-r--r--   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/empty
--rw-r--r--   0 runner    (1001) docker     (123)      608 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_diff_ids.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      669 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_double_qual.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      669 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_double_seq.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      609 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_long_qual.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      482 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_no_qual.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      608 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_del.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      608 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_escape.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      610 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_null.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      609 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_space.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      609 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_tab.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      608 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_unit_sep.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      608 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_vtab.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      607 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_short_qual.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      628 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_spaces.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      629 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_tabs.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      548 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_at_plus.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      582 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_at_qual.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      522 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_at_seq.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      562 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_in_plus.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      607 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_in_qual.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      535 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_in_seq.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      513 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_in_title.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      174 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_10_seqs
--rw-r--r--   0 runner    (1001) docker     (123)       63 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_3_seqs_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       66 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_3_seqs_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)      150 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_5_blanks_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      205 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_5_ws_lines_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      151 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_6_blanks_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      217 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_6_ws_lines_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      157 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_blank_lines_between_records
--rw-r--r--   0 runner    (1001) docker     (123)      152 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_blanks_end_of_file
--rw-r--r--   0 runner    (1001) docker     (123)       75 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_description_newline_replacement_empty_str
--rw-r--r--   0 runner    (1001) docker     (123)      109 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_description_newline_replacement_multi_char
--rw-r--r--   0 runner    (1001) docker     (123)       87 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_description_newline_replacement_none
--rw-r--r--   0 runner    (1001) docker     (123)       17 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_id_whitespace_replacement_empty_str
--rw-r--r--   0 runner    (1001) docker     (123)       53 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_id_whitespace_replacement_multi_char
--rw-r--r--   0 runner    (1001) docker     (123)       29 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_id_whitespace_replacement_none
--rw-r--r--   0 runner    (1001) docker     (123)      186 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_after_10_seqs
--rw-r--r--   0 runner    (1001) docker     (123)       52 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_blank_line_after_header
--rw-r--r--   0 runner    (1001) docker     (123)       64 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_blank_line_within_sequence
--rw-r--r--   0 runner    (1001) docker     (123)       46 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_blank_sequence
--rw-r--r--   0 runner    (1001) docker     (123)       40 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_legacy_format
--rw-r--r--   0 runner    (1001) docker     (123)       16 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_missing_header
--rw-r--r--   0 runner    (1001) docker     (123)       46 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_missing_seq_data_first
--rw-r--r--   0 runner    (1001) docker     (123)       47 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_missing_seq_data_last
--rw-r--r--   0 runner    (1001) docker     (123)       45 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_missing_seq_data_middle
--rw-r--r--   0 runner    (1001) docker     (123)       63 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_whitespace_line_after_header
--rw-r--r--   0 runner    (1001) docker     (123)       69 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_whitespace_only_line_within_sequence
--rw-r--r--   0 runner    (1001) docker     (123)       57 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_invalid_whitespace_only_sequence
--rw-r--r--   0 runner    (1001) docker     (123)       32 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_max_width_1
--rw-r--r--   0 runner    (1001) docker     (123)      150 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_max_width_5
--rw-r--r--   0 runner    (1001) docker     (123)       51 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_mixed_qual_scores
--rw-r--r--   0 runner    (1001) docker     (123)      145 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_multi_seq
--rw-r--r--   0 runner    (1001) docker     (123)      114 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_multi_seq_roundtrip
--rw-r--r--   0 runner    (1001) docker     (123)       80 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_prot_seqs_odd_labels
--rw-r--r--   0 runner    (1001) docker     (123)       18 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_single_bio_seq_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       21 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_single_bio_seq_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       18 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_single_dna_seq_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       21 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_single_dna_seq_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       17 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_single_prot_seq_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       19 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_single_prot_seq_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       18 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_single_rna_seq_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       21 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_single_rna_seq_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       23 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_single_seq
--rw-r--r--   0 runner    (1001) docker     (123)       72 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_tabular_msa_different_type
--rw-r--r--   0 runner    (1001) docker     (123)      277 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_ws_lines_between_records
--rw-r--r--   0 runner    (1001) docker     (123)      233 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fasta_ws_lines_end_of_file
--rw-r--r--   0 runner    (1001) docker     (123)       92 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_5_blanks_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      147 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_5_ws_lines_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)       90 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_blank_lines
--rw-r--r--   0 runner    (1001) docker     (123)       88 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_blank_after_header
--rw-r--r--   0 runner    (1001) docker     (123)       88 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_blank_after_plus
--rw-r--r--   0 runner    (1001) docker     (123)       88 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_blank_after_seq
--rw-r--r--   0 runner    (1001) docker     (123)       89 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_blank_in_seq_at_symbol
--rw-r--r--   0 runner    (1001) docker     (123)       89 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_blank_within_qual
--rw-r--r--   0 runner    (1001) docker     (123)       89 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_blank_within_seq
--rw-r--r--   0 runner    (1001) docker     (123)       23 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_missing_header
--rw-r--r--   0 runner    (1001) docker     (123)      103 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_missing_seq_data
--rw-r--r--   0 runner    (1001) docker     (123)       99 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_ws_line_after_header
--rw-r--r--   0 runner    (1001) docker     (123)       99 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_ws_line_after_plus
--rw-r--r--   0 runner    (1001) docker     (123)       99 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_ws_line_after_seq
--rw-r--r--   0 runner    (1001) docker     (123)      100 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_ws_line_within_qual
--rw-r--r--   0 runner    (1001) docker     (123)      100 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_invalid_ws_line_within_seq
--rw-r--r--   0 runner    (1001) docker     (123)       95 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_multi_blank_between_records
--rw-r--r--   0 runner    (1001) docker     (123)       94 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_multi_blank_end_of_file
--rw-r--r--   0 runner    (1001) docker     (123)       93 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_multi_blank_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)       87 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_multi_seq_sanger
--rw-r--r--   0 runner    (1001) docker     (123)      331 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_multi_whitespace_stripping
--rw-r--r--   0 runner    (1001) docker     (123)      159 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_multi_ws_lines_between_records
--rw-r--r--   0 runner    (1001) docker     (123)      183 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_multi_ws_lines_end_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      159 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_multi_ws_lines_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)       33 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_single_seq_illumina1.3
--rw-r--r--   0 runner    (1001) docker     (123)      118 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_single_seq_illumina1.8
--rw-r--r--   0 runner    (1001) docker     (123)      123 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_whitespace_only_lines
--rw-r--r--   0 runner    (1001) docker     (123)      885 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_wrapping_as_illumina_no_description
--rw-r--r--   0 runner    (1001) docker     (123)      885 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_wrapping_as_sanger_no_description
--rw-r--r--   0 runner    (1001) docker     (123)      898 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_wrapping_original_sanger_no_description
--rw-r--r--   0 runner    (1001) docker     (123)       94 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_writer_illumina1.3_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       94 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_writer_sanger_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       94 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_writer_sanger_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       85 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/genbank_5_blanks_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)       86 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/genbank_6_blanks_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)       71 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/genbank_missing_locus_name
--rw-r--r--   0 runner    (1001) docker     (123)     1598 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/genbank_multi_records
--rw-r--r--   0 runner    (1001) docker     (123)     1086 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/genbank_single_record
--rw-r--r--   0 runner    (1001) docker     (123)       90 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/genbank_single_record_lower
--rw-r--r--   0 runner    (1001) docker     (123)       90 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/genbank_single_record_upper
--rw-r--r--   0 runner    (1001) docker     (123)       82 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/genbank_w_beginning_whitespace
--rw-r--r--   0 runner    (1001) docker     (123)      206 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/gff3_bad_missing_directive
--rw-r--r--   0 runner    (1001) docker     (123)      459 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/gff3_bad_wrong_columns
--rw-r--r--   0 runner    (1001) docker     (123)      106 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/gff3_dna
--rw-r--r--   0 runner    (1001) docker     (123)      881 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/gff3_multi_record
--rw-r--r--   0 runner    (1001) docker     (123)      505 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/gff3_single_record
--rw-r--r--   0 runner    (1001) docker     (123)      426 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/illumina_full_range_as_illumina.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      426 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/illumina_full_range_as_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      426 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/illumina_full_range_original_illumina.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      505 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/legacy9_and_blast7_default
--rw-r--r--   0 runner    (1001) docker     (123)      404 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/legacy9_invalid_differing_fields
--rw-r--r--   0 runner    (1001) docker     (123)      262 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/legacy9_invalid_too_many_columns
--rw-r--r--   0 runner    (1001) docker     (123)      296 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/legacy9_mixed_nans
--rw-r--r--   0 runner    (1001) docker     (123)      345 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/legacy9_multi_line
--rw-r--r--   0 runner    (1001) docker     (123)      251 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/legacy9_single_line
--rw-r--r--   0 runner    (1001) docker     (123)     8388 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/longreads_as_illumina.fastq
--rw-r--r--   0 runner    (1001) docker     (123)     8388 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/longreads_as_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)     9466 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/longreads_original_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      767 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/misc_dna_as_illumina.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      767 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/misc_dna_as_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      767 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/misc_dna_original_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      774 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/misc_rna_as_illumina.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      774 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/misc_rna_as_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      774 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/misc_rna_original_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      351 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_L&L_CA_data_scores
--rw-r--r--   0 runner    (1001) docker     (123)     1584 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_PCoA_sample_data_3_scores
--rw-r--r--   0 runner    (1001) docker     (123)     4569 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error1
--rw-r--r--   0 runner    (1001) docker     (123)      306 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error10
--rw-r--r--   0 runner    (1001) docker     (123)     4578 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error11
--rw-r--r--   0 runner    (1001) docker     (123)     1569 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error12
--rw-r--r--   0 runner    (1001) docker     (123)     1569 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error13
--rw-r--r--   0 runner    (1001) docker     (123)     1584 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error14
--rw-r--r--   0 runner    (1001) docker     (123)     1543 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error15
--rw-r--r--   0 runner    (1001) docker     (123)     4579 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error16
--rw-r--r--   0 runner    (1001) docker     (123)     4531 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error17
--rw-r--r--   0 runner    (1001) docker     (123)     1570 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error18
--rw-r--r--   0 runner    (1001) docker     (123)      304 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error19
--rw-r--r--   0 runner    (1001) docker     (123)     4555 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error2
--rw-r--r--   0 runner    (1001) docker     (123)      306 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error20
--rw-r--r--   0 runner    (1001) docker     (123)       10 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error21
--rw-r--r--   0 runner    (1001) docker     (123)       18 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error22
--rw-r--r--   0 runner    (1001) docker     (123)       42 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error23
--rw-r--r--   0 runner    (1001) docker     (123)       75 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error24
--rw-r--r--   0 runner    (1001) docker     (123)     4567 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error3
--rw-r--r--   0 runner    (1001) docker     (123)     4569 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error4
--rw-r--r--   0 runner    (1001) docker     (123)     4569 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error5
--rw-r--r--   0 runner    (1001) docker     (123)     4558 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error6
--rw-r--r--   0 runner    (1001) docker     (123)       65 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error7
--rw-r--r--   0 runner    (1001) docker     (123)     1570 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error8
--rw-r--r--   0 runner    (1001) docker     (123)      304 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error9
--rw-r--r--   0 runner    (1001) docker     (123)     3396 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_example2_scores
--rw-r--r--   0 runner    (1001) docker     (123)     4579 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_example3_scores
--rw-r--r--   0 runner    (1001) docker     (123)     2293 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_site
--rw-r--r--   0 runner    (1001) docker     (123)     2291 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_site_constraints
--rw-r--r--   0 runner    (1001) docker     (123)     2065 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_species
--rw-r--r--   0 runner    (1001) docker     (123)     2062 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_PCoA_site
--rw-r--r--   0 runner    (1001) docker     (123)     1786 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_site
--rw-r--r--   0 runner    (1001) docker     (123)     1787 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_site_constraints
--rw-r--r--   0 runner    (1001) docker     (123)     1070 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_species
--rw-r--r--   0 runner    (1001) docker     (123)       80 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_dna_3_seqs
--rw-r--r--   0 runner    (1001) docker     (123)       70 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_empty_line_after_header
--rw-r--r--   0 runner    (1001) docker     (123)       70 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_empty_line_before_header
--rw-r--r--   0 runner    (1001) docker     (123)       70 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_empty_line_between_seqs
--rw-r--r--   0 runner    (1001) docker     (123)       80 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_header_too_long
--rw-r--r--   0 runner    (1001) docker     (123)       68 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_header_too_short
--rw-r--r--   0 runner    (1001) docker     (123)       64 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_no_header
--rw-r--r--   0 runner    (1001) docker     (123)       71 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_seq_too_long
--rw-r--r--   0 runner    (1001) docker     (123)       67 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_seq_too_short
--rw-r--r--   0 runner    (1001) docker     (123)       37 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_too_few_seqs
--rw-r--r--   0 runner    (1001) docker     (123)       69 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_too_many_seqs
--rw-r--r--   0 runner    (1001) docker     (123)       15 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_zero_seq_len
--rw-r--r--   0 runner    (1001) docker     (123)        4 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_invalid_zero_seqs
--rw-r--r--   0 runner    (1001) docker     (123)       42 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_single_seq_long
--rw-r--r--   0 runner    (1001) docker     (123)       16 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_single_seq_short
--rw-r--r--   0 runner    (1001) docker     (123)       69 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_two_chunks
--rw-r--r--   0 runner    (1001) docker     (123)      106 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_variable_length_ids
--rw-r--r--   0 runner    (1001) docker     (123)       81 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_varied_whitespace_in_seqs
--rw-r--r--   0 runner    (1001) docker     (123)      106 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_whitespace_in_header_1
--rw-r--r--   0 runner    (1001) docker     (123)       86 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_whitespace_in_header_2
--rw-r--r--   0 runner    (1001) docker     (123)      102 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/phylip_whitespace_in_header_3
--rw-r--r--   0 runner    (1001) docker     (123)      217 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qseq_invalid_filter
--rw-r--r--   0 runner    (1001) docker     (123)      217 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qseq_invalid_lane
--rw-r--r--   0 runner    (1001) docker     (123)      217 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qseq_invalid_read
--rw-r--r--   0 runner    (1001) docker     (123)      216 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qseq_invalid_tile
--rw-r--r--   0 runner    (1001) docker     (123)      234 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qseq_invalid_x
--rw-r--r--   0 runner    (1001) docker     (123)      251 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qseq_invalid_y
--rw-r--r--   0 runner    (1001) docker     (123)      226 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qseq_multi_seq_illumina1.3
--rw-r--r--   0 runner    (1001) docker     (123)      217 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qseq_single_seq_sanger
--rw-r--r--   0 runner    (1001) docker     (123)       69 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_2_seqs_defaults
--rw-r--r--   0 runner    (1001) docker     (123)      100 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_3_seqs_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       89 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_3_seqs_defaults_desc_mismatch
--rw-r--r--   0 runner    (1001) docker     (123)      124 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_3_seqs_defaults_extra
--rw-r--r--   0 runner    (1001) docker     (123)       89 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_3_seqs_defaults_id_mismatch
--rw-r--r--   0 runner    (1001) docker     (123)       97 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_3_seqs_defaults_length_mismatch
--rw-r--r--   0 runner    (1001) docker     (123)      105 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_3_seqs_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)      218 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_5_blanks_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      273 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_5_ws_lines_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      219 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_6_blanks_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      285 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_6_ws_lines_start_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      223 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_blank_lines_between_records
--rw-r--r--   0 runner    (1001) docker     (123)      220 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_blanks_end_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      104 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_description_newline_replacement_empty_str
--rw-r--r--   0 runner    (1001) docker     (123)      138 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_description_newline_replacement_multi_char
--rw-r--r--   0 runner    (1001) docker     (123)      116 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_description_newline_replacement_none
--rw-r--r--   0 runner    (1001) docker     (123)       22 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_id_whitespace_replacement_empty_str
--rw-r--r--   0 runner    (1001) docker     (123)       58 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_id_whitespace_replacement_multi_char
--rw-r--r--   0 runner    (1001) docker     (123)       34 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_id_whitespace_replacement_none
--rw-r--r--   0 runner    (1001) docker     (123)      101 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_blank_line_after_header
--rw-r--r--   0 runner    (1001) docker     (123)      102 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_blank_line_within_seq
--rw-r--r--   0 runner    (1001) docker     (123)       84 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_blank_sequence
--rw-r--r--   0 runner    (1001) docker     (123)       47 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_legacy_format
--rw-r--r--   0 runner    (1001) docker     (123)       19 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_missing_header
--rw-r--r--   0 runner    (1001) docker     (123)       92 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_missing_qual_scores_first
--rw-r--r--   0 runner    (1001) docker     (123)       85 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_missing_qual_scores_last
--rw-r--r--   0 runner    (1001) docker     (123)       83 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_missing_qual_scores_middle
--rw-r--r--   0 runner    (1001) docker     (123)      102 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_qual_scores_float
--rw-r--r--   0 runner    (1001) docker     (123)      100 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_qual_scores_negative
--rw-r--r--   0 runner    (1001) docker     (123)      100 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_qual_scores_over_255
--rw-r--r--   0 runner    (1001) docker     (123)      101 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_qual_scores_string
--rw-r--r--   0 runner    (1001) docker     (123)      113 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_whitespace_line_in_seq
--rw-r--r--   0 runner    (1001) docker     (123)       95 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_whitespace_only_sequence
--rw-r--r--   0 runner    (1001) docker     (123)      112 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_invalid_ws_line_after_header
--rw-r--r--   0 runner    (1001) docker     (123)       40 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_max_width_1
--rw-r--r--   0 runner    (1001) docker     (123)      213 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_max_width_5
--rw-r--r--   0 runner    (1001) docker     (123)      213 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_multi_seq
--rw-r--r--   0 runner    (1001) docker     (123)      166 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_multi_seq_roundtrip
--rw-r--r--   0 runner    (1001) docker     (123)      131 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_prot_seqs_odd_labels
--rw-r--r--   0 runner    (1001) docker     (123)       21 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_single_bio_seq_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       21 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_single_dna_seq_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       23 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_single_nuc_seq_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       22 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_single_prot_seq_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       21 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_single_rna_seq_non_defaults
--rw-r--r--   0 runner    (1001) docker     (123)       40 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_single_seq
--rw-r--r--   0 runner    (1001) docker     (123)       85 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_tabular_msa_different_type
--rw-r--r--   0 runner    (1001) docker     (123)      333 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_ws_lines_between_records
--rw-r--r--   0 runner    (1001) docker     (123)      309 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/qual_ws_lines_end_of_file
--rw-r--r--   0 runner    (1001) docker     (123)      550 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/sanger_full_range_as_illumina.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      550 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/sanger_full_range_as_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      550 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/sanger_full_range_original_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      450 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/solexa_full_range_original_solexa.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      404 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_all_data_types
--rw-r--r--   0 runner    (1001) docker     (123)       47 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_blank_lines
--rw-r--r--   0 runner    (1001) docker     (123)       82 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_data_only
--rw-r--r--   0 runner    (1001) docker     (123)      142 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_different_padding
--rw-r--r--   0 runner    (1001) docker     (123)       70 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_differing_gc_data_length
--rw-r--r--   0 runner    (1001) docker     (123)       62 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_differing_gr_data_length
--rw-r--r--   0 runner    (1001) docker     (123)       70 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_differing_seq_lengths
--rw-r--r--   0 runner    (1001) docker     (123)      180 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_duplicate_gc
--rw-r--r--   0 runner    (1001) docker     (123)      195 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_duplicate_gr
--rw-r--r--   0 runner    (1001) docker     (123)      159 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_duplicate_sequence_names
--rw-r--r--   0 runner    (1001) docker     (123)       73 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_duplicate_tree_ids
--rw-r--r--   0 runner    (1001) docker     (123)      652 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_extensive
--rw-r--r--   0 runner    (1001) docker     (123)      660 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_extensive_mixed
--rw-r--r--   0 runner    (1001) docker     (123)       45 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_invalid_data_type
--rw-r--r--   0 runner    (1001) docker     (123)      183 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_invalid_nonexistent_gr
--rw-r--r--   0 runner    (1001) docker     (123)      110 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_invalid_nonexistent_gs
--rw-r--r--   0 runner    (1001) docker     (123)       24 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_malformed_data_line
--rw-r--r--   0 runner    (1001) docker     (123)       61 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_malformed_gc_line
--rw-r--r--   0 runner    (1001) docker     (123)       27 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_malformed_gf_line
--rw-r--r--   0 runner    (1001) docker     (123)       54 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_malformed_gr_line
--rw-r--r--   0 runner    (1001) docker     (123)       43 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_malformed_gs_line
--rw-r--r--   0 runner    (1001) docker     (123)       70 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_metadata_only
--rw-r--r--   0 runner    (1001) docker     (123)       48 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_minimal
--rw-r--r--   0 runner    (1001) docker     (123)      121 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_missing_footer
--rw-r--r--   0 runner    (1001) docker     (123)      108 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_missing_header
--rw-r--r--   0 runner    (1001) docker     (123)       66 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_missing_reference_items
--rw-r--r--   0 runner    (1001) docker     (123)       45 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_missing_rn_tag
--rw-r--r--   0 runner    (1001) docker     (123)       45 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_multi_line_tree_no_id
--rw-r--r--   0 runner    (1001) docker     (123)       59 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_multi_line_tree_with_id
--rw-r--r--   0 runner    (1001) docker     (123)      486 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_multiple_msa
--rw-r--r--   0 runner    (1001) docker     (123)       99 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_multiple_multi_line_trees
--rw-r--r--   0 runner    (1001) docker     (123)      319 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_multiple_references
--rw-r--r--   0 runner    (1001) docker     (123)      100 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_multiple_trees
--rw-r--r--   0 runner    (1001) docker     (123)       19 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_no_data
--rw-r--r--   0 runner    (1001) docker     (123)      110 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_nonstring_labels
--rw-r--r--   0 runner    (1001) docker     (123)      243 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_rna
--rw-r--r--   0 runner    (1001) docker     (123)      156 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_runon_gf_no_whitespace
--rw-r--r--   0 runner    (1001) docker     (123)      157 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_runon_gf_with_whitespace
--rw-r--r--   0 runner    (1001) docker     (123)       87 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_runon_gs_no_whitespace
--rw-r--r--   0 runner    (1001) docker     (123)       88 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_runon_gs_with_whitespace
--rw-r--r--   0 runner    (1001) docker     (123)      150 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_runon_references
--rw-r--r--   0 runner    (1001) docker     (123)      150 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_runon_references_mixed
--rw-r--r--   0 runner    (1001) docker     (123)      119 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_single_reference
--rw-r--r--   0 runner    (1001) docker     (123)       46 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_single_tree_with_id
--rw-r--r--   0 runner    (1001) docker     (123)       32 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_single_tree_without_id
--rw-r--r--   0 runner    (1001) docker     (123)      246 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_two_of_each_metadata
--rw-r--r--   0 runner    (1001) docker     (123)       74 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_whitespace_only_lines
--rw-r--r--   0 runner    (1001) docker     (123)     1159 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/taxdump_names.dmp
--rw-r--r--   0 runner    (1001) docker     (123)     1249 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/taxdump_nodes.dmp
--rw-r--r--   0 runner    (1001) docker     (123)      160 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/tsv_10_fields
--rw-r--r--   0 runner    (1001) docker     (123)      133 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/tsv_8_fields
--rw-r--r--   0 runner    (1001) docker     (123)       47 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/whitespace_only
--rw-r--r--   0 runner    (1001) docker     (123)      963 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/wrapping_as_illumina.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      963 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/wrapping_as_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)      975 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/data/wrapping_original_sanger.fastq
--rw-r--r--   0 runner    (1001) docker     (123)    13438 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_base.py
--rw-r--r--   0 runner    (1001) docker     (123)     4775 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_binary_dm.py
--rw-r--r--   0 runner    (1001) docker     (123)     6179 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_blast6.py
--rw-r--r--   0 runner    (1001) docker     (123)     9968 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_blast7.py
--rw-r--r--   0 runner    (1001) docker     (123)    11715 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_clustal.py
--rw-r--r--   0 runner    (1001) docker     (123)    36301 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_embl.py
--rw-r--r--   0 runner    (1001) docker     (123)     1097 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_emptyfile.py
--rw-r--r--   0 runner    (1001) docker     (123)    47662 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_fasta.py
--rw-r--r--   0 runner    (1001) docker     (123)    28808 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_fastq.py
--rw-r--r--   0 runner    (1001) docker     (123)    15587 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_genbank.py
--rw-r--r--   0 runner    (1001) docker     (123)    13406 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_gff3.py
--rw-r--r--   0 runner    (1001) docker     (123)     9911 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_lsmat.py
--rw-r--r--   0 runner    (1001) docker     (123)    12594 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_newick.py
--rw-r--r--   0 runner    (1001) docker     (123)    11723 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_ordination.py
--rw-r--r--   0 runner    (1001) docker     (123)     9882 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_phylip.py
--rw-r--r--   0 runner    (1001) docker     (123)    13189 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_qseq.py
--rw-r--r--   0 runner    (1001) docker     (123)     3795 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_sequence_feature_vocabulary.py
--rw-r--r--   0 runner    (1001) docker     (123)    41921 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_stockholm.py
--rw-r--r--   0 runner    (1001) docker     (123)     7847 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/format/tests/test_taxdump.py
--rw-r--r--   0 runner    (1001) docker     (123)    43169 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/registry.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.848321 scikit-bio-0.5.8/skbio/io/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/__init__.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.848321 scikit-bio-0.5.8/skbio/io/tests/data/
--rw-r--r--   0 runner    (1001) docker     (123)        5 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/data/big5_file
--rw-r--r--   0 runner    (1001) docker     (123)       46 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/data/big5_file.bz2
--rw-r--r--   0 runner    (1001) docker     (123)       35 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/data/big5_file.gz
--rw-r--r--   0 runner    (1001) docker     (123)       53 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/data/example_file
--rw-r--r--   0 runner    (1001) docker     (123)       84 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/data/example_file.bz2
--rw-r--r--   0 runner    (1001) docker     (123)       83 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/data/example_file.gz
--rw-r--r--   0 runner    (1001) docker     (123)       10 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/data/real_file
--rw-r--r--   0 runner    (1001) docker     (123)       23 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/data/real_file_2
--rw-r--r--   0 runner    (1001) docker     (123)     1484 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/test_iosources.py
--rw-r--r--   0 runner    (1001) docker     (123)    63448 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/test_registry.py
--rw-r--r--   0 runner    (1001) docker     (123)    23129 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/tests/test_util.py
--rw-r--r--   0 runner    (1001) docker     (123)    11105 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/io/util.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.852321 scikit-bio-0.5.8/skbio/metadata/
--rw-r--r--   0 runner    (1001) docker     (123)      720 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)   601338 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/_intersection.c
--rw-r--r--   0 runner    (1001) docker     (123)    18666 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/_intersection.pyx
--rw-r--r--   0 runner    (1001) docker     (123)    36174 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/_interval.py
--rw-r--r--   0 runner    (1001) docker     (123)    18539 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/_mixin.py
--rw-r--r--   0 runner    (1001) docker     (123)     6674 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/_repr.py
--rw-r--r--   0 runner    (1001) docker     (123)    53266 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/_testing.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.852321 scikit-bio-0.5.8/skbio/metadata/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/tests/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     8188 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/tests/test_intersection.py
--rw-r--r--   0 runner    (1001) docker     (123)    32405 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/tests/test_interval.py
--rw-r--r--   0 runner    (1001) docker     (123)     4518 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/metadata/tests/test_mixin.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.852321 scikit-bio-0.5.8/skbio/sequence/
--rw-r--r--   0 runner    (1001) docker     (123)     7671 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)    12736 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/_dna.py
--rw-r--r--   0 runner    (1001) docker     (123)    31064 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/_genetic_code.py
--rw-r--r--   0 runner    (1001) docker     (123)    24487 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/_grammared_sequence.py
--rw-r--r--   0 runner    (1001) docker     (123)    11243 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/_nucleotide_mixin.py
--rw-r--r--   0 runner    (1001) docker     (123)     7000 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/_protein.py
--rw-r--r--   0 runner    (1001) docker     (123)     4193 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/_repr.py
--rw-r--r--   0 runner    (1001) docker     (123)    13231 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/_rna.py
--rw-r--r--   0 runner    (1001) docker     (123)    73555 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/_sequence.py
--rw-r--r--   0 runner    (1001) docker     (123)     5218 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/distance.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.856321 scikit-bio-0.5.8/skbio/sequence/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/tests/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     7215 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/tests/test_distance.py
--rw-r--r--   0 runner    (1001) docker     (123)     1987 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/tests/test_dna.py
--rw-r--r--   0 runner    (1001) docker     (123)    19666 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/tests/test_genetic_code.py
--rw-r--r--   0 runner    (1001) docker     (123)    27075 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/tests/test_grammared_sequence.py
--rw-r--r--   0 runner    (1001) docker     (123)    21247 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/tests/test_nucleotide_sequences.py
--rw-r--r--   0 runner    (1001) docker     (123)     4890 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/tests/test_protein.py
--rw-r--r--   0 runner    (1001) docker     (123)     2083 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/tests/test_rna.py
--rw-r--r--   0 runner    (1001) docker     (123)   108852 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/sequence/tests/test_sequence.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.856321 scikit-bio-0.5.8/skbio/stats/
--rw-r--r--   0 runner    (1001) docker     (123)      919 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)   283902 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/__subsample.c
--rw-r--r--   0 runner    (1001) docker     (123)     1051 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/__subsample.pyx
--rw-r--r--   0 runner    (1001) docker     (123)      869 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/_misc.py
--rw-r--r--   0 runner    (1001) docker     (123)     7864 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/_subsample.py
--rw-r--r--   0 runner    (1001) docker     (123)    43874 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/composition.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.860321 scikit-bio-0.5.8/skbio/stats/distance/
--rw-r--r--   0 runner    (1001) docker     (123)     5626 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     8598 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/_anosim.py
--rw-r--r--   0 runner    (1001) docker     (123)    49177 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/_base.py
--rw-r--r--   0 runner    (1001) docker     (123)     9530 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/_bioenv.py
--rw-r--r--   0 runner    (1001) docker     (123)  1171270 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/_cutils.c
--rw-r--r--   0 runner    (1001) docker     (123)     8991 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/_cutils.pyx
--rw-r--r--   0 runner    (1001) docker     (123)    26065 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/_mantel.py
--rw-r--r--   0 runner    (1001) docker     (123)     5311 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/_permanova.py
--rw-r--r--   0 runner    (1001) docker     (123)    12524 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/_permdisp.py
--rw-r--r--   0 runner    (1001) docker     (123)     5752 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/_utils.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.860321 scikit-bio-0.5.8/skbio/stats/distance/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/__init__.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.864321 scikit-bio-0.5.8/skbio/stats/distance/tests/data/
--rw-r--r--   0 runner    (1001) docker     (123)     1108 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/bioenv_df_vegan.txt
--rw-r--r--   0 runner    (1001) docker     (123)    10023 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/bioenv_dm_vegan.txt
--rw-r--r--   0 runner    (1001) docker     (123)      228 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/bioenv_exp_results_vegan.txt
--rw-r--r--   0 runner    (1001) docker     (123)      644 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/df.txt
--rw-r--r--   0 runner    (1001) docker     (123)      696 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/df_extra_column.txt
--rw-r--r--   0 runner    (1001) docker     (123)      607 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/dm.txt
--rw-r--r--   0 runner    (1001) docker     (123)      607 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/dm2.txt
--rw-r--r--   0 runner    (1001) docker     (123)      607 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/dm3.txt
--rw-r--r--   0 runner    (1001) docker     (123)      607 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/dm4.txt
--rw-r--r--   0 runner    (1001) docker     (123)      607 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/dm_reordered.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1173 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/exp_results.txt
--rw-r--r--   0 runner    (1001) docker     (123)     1173 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/exp_results_different_column_order.txt
--rw-r--r--   0 runner    (1001) docker     (123)       32 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/exp_results_single_column.txt
--rw-r--r--   0 runner    (1001) docker     (123)     9376 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/mantel_env_dm_vegan.txt
--rw-r--r--   0 runner    (1001) docker     (123)     9934 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/mantel_veg_dm_vegan.txt
--rw-r--r--   0 runner    (1001) docker     (123)    20633 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/moving_pictures_dm.tsv
--rw-r--r--   0 runner    (1001) docker     (123)     3670 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/moving_pictures_mf.tsv
--rw-r--r--   0 runner    (1001) docker     (123)      378 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/pwmantel_exp_results_all_dms.txt
--rw-r--r--   0 runner    (1001) docker     (123)      103 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/pwmantel_exp_results_dm_dm2.txt
--rw-r--r--   0 runner    (1001) docker     (123)      159 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/pwmantel_exp_results_duplicate_dms.txt
--rw-r--r--   0 runner    (1001) docker     (123)      214 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/pwmantel_exp_results_minimal.txt
--rw-r--r--   0 runner    (1001) docker     (123)      232 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/pwmantel_exp_results_minimal_with_labels.txt
--rw-r--r--   0 runner    (1001) docker     (123)       95 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/pwmantel_exp_results_na_p_value.txt
--rw-r--r--   0 runner    (1001) docker     (123)      215 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/data/pwmantel_exp_results_reordered_distance_matrices.txt
--rw-r--r--   0 runner    (1001) docker     (123)     4840 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/test_anosim.py
--rw-r--r--   0 runner    (1001) docker     (123)    45090 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/test_base.py
--rw-r--r--   0 runner    (1001) docker     (123)     9918 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/test_bioenv.py
--rw-r--r--   0 runner    (1001) docker     (123)    30434 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/test_mantel.py
--rw-r--r--   0 runner    (1001) docker     (123)     4860 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/test_permanova.py
--rw-r--r--   0 runner    (1001) docker     (123)    11549 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/distance/tests/test_permdisp.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.864321 scikit-bio-0.5.8/skbio/stats/evolve/
--rw-r--r--   0 runner    (1001) docker     (123)      929 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/evolve/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)    11070 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/evolve/_hommola.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.864321 scikit-bio-0.5.8/skbio/stats/evolve/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/evolve/tests/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     7993 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/evolve/tests/test_hommola.py
--rw-r--r--   0 runner    (1001) docker     (123)    32216 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/gradient.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.868321 scikit-bio-0.5.8/skbio/stats/ordination/
--rw-r--r--   0 runner    (1001) docker     (123)     4143 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     8399 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/_canonical_correspondence_analysis.py
--rw-r--r--   0 runner    (1001) docker     (123)     7748 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/_correspondence_analysis.py
--rw-r--r--   0 runner    (1001) docker     (123)  1031727 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/_cutils.c
--rw-r--r--   0 runner    (1001) docker     (123)     5186 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/_cutils.pyx
--rw-r--r--   0 runner    (1001) docker     (123)    16917 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/_ordination_results.py
--rw-r--r--   0 runner    (1001) docker     (123)    17246 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/_principal_coordinate_analysis.py
--rw-r--r--   0 runner    (1001) docker     (123)     9382 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/_redundancy_analysis.py
--rw-r--r--   0 runner    (1001) docker     (123)    10184 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/_utils.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.868321 scikit-bio-0.5.8/skbio/stats/ordination/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/__init__.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.876321 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/
--rw-r--r--   0 runner    (1001) docker     (123)       24 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/L&L_CA_data
--rw-r--r--   0 runner    (1001) docker     (123)     1149 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_biplot_descriptors
--rw-r--r--   0 runner    (1001) docker     (123)     1577 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_biplot_projected_descriptors
--rw-r--r--   0 runner    (1001) docker     (123)     1099 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data
--rw-r--r--   0 runner    (1001) docker     (123)     2647 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data_12dim
--rw-r--r--   0 runner    (1001) docker     (123)      900 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data_2
--rw-r--r--   0 runner    (1001) docker     (123)     1228 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data_3
--rw-r--r--   0 runner    (1001) docker     (123)      830 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data_3_eigh_ref_3dim
--rw-r--r--   0 runner    (1001) docker     (123)      821 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data_3_fsvd_ref_3dim
--rw-r--r--   0 runner    (1001) docker     (123)     1992 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_skbio
--rw-r--r--   0 runner    (1001) docker     (123)     1000 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_X
--rw-r--r--   0 runner    (1001) docker     (123)      128 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_Y
--rw-r--r--   0 runner    (1001) docker     (123)      274 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_biplot_scaling1
--rw-r--r--   0 runner    (1001) docker     (123)      275 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_biplot_scaling2
--rw-r--r--   0 runner    (1001) docker     (123)      834 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_sample_constraints_scaling1
--rw-r--r--   0 runner    (1001) docker     (123)      829 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_sample_constraints_scaling2
--rw-r--r--   0 runner    (1001) docker     (123)      787 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_site_scaling1_from_vegan
--rw-r--r--   0 runner    (1001) docker     (123)      742 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_site_scaling2_from_vegan
--rw-r--r--   0 runner    (1001) docker     (123)      424 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_species_scaling1_from_vegan
--rw-r--r--   0 runner    (1001) docker     (123)      459 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_species_scaling2_from_vegan
--rw-r--r--   0 runner    (1001) docker     (123)     1000 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_X
--rw-r--r--   0 runner    (1001) docker     (123)      196 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_Y
--rw-r--r--   0 runner    (1001) docker     (123)       86 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_biplot_scaling1
--rw-r--r--   0 runner    (1001) docker     (123)       86 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_biplot_scaling2
--rw-r--r--   0 runner    (1001) docker     (123)      864 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_sample_constraints_scaling1
--rw-r--r--   0 runner    (1001) docker     (123)      867 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_sample_constraints_scaling2
--rw-r--r--   0 runner    (1001) docker     (123)      984 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_site_scaling1_from_vegan
--rw-r--r--   0 runner    (1001) docker     (123)      907 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_site_scaling2_from_vegan
--rw-r--r--   0 runner    (1001) docker     (123)      820 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_species_scaling1_from_vegan
--rw-r--r--   0 runner    (1001) docker     (123)      881 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_species_scaling2_from_vegan
--rw-r--r--   0 runner    (1001) docker     (123)     2062 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/exp_PCoAEigenResults_site
--rw-r--r--   0 runner    (1001) docker     (123)     4900 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/exp_PCoAzeros_site
--rw-r--r--   0 runner    (1001) docker     (123)     1735 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_biplot_from_vegan.csv
--rw-r--r--   0 runner    (1001) docker     (123)     2796 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_constraints_from_vegan.csv
--rw-r--r--   0 runner    (1001) docker     (123)      552 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_eigvals_from_vegan.csv
--rw-r--r--   0 runner    (1001) docker     (123)      342 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_propexpl_from_vegan.csv
--rw-r--r--   0 runner    (1001) docker     (123)     2798 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_sites_from_vegan.csv
--rw-r--r--   0 runner    (1001) docker     (123)     5511 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_species_from_vegan.csv
--rw-r--r--   0 runner    (1001) docker     (123)     1770 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/varechem.csv
--rw-r--r--   0 runner    (1001) docker     (123)     4475 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/data/varespec.csv
--rw-r--r--   0 runner    (1001) docker     (123)     6642 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/test_canonical_correspondence_analysis.py
--rw-r--r--   0 runner    (1001) docker     (123)     8193 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/test_correspondence_analysis.py
--rw-r--r--   0 runner    (1001) docker     (123)    12284 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/test_ordination_results.py
--rw-r--r--   0 runner    (1001) docker     (123)    14004 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/test_principal_coordinate_analysis.py
--rw-r--r--   0 runner    (1001) docker     (123)    10422 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/test_redundancy_analysis.py
--rw-r--r--   0 runner    (1001) docker     (123)     4845 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/ordination/tests/test_util.py
--rw-r--r--   0 runner    (1001) docker     (123)    48004 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/power.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.876321 scikit-bio-0.5.8/skbio/stats/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/__init__.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.880321 scikit-bio-0.5.8/skbio/stats/tests/data/
--rw-r--r--   0 runner    (1001) docker     (123)      297 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/cr_data_out
--rw-r--r--   0 runner    (1001) docker     (123)      210 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/cr_data_raw
--rw-r--r--   0 runner    (1001) docker     (123)       90 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/cr_no_data_out
--rw-r--r--   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/cr_no_data_raw
--rw-r--r--   0 runner    (1001) docker     (123)      160 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/gr_w_msg_out
--rw-r--r--   0 runner    (1001) docker     (123)      126 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/gr_w_msg_raw
--rw-r--r--   0 runner    (1001) docker     (123)       97 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/gr_wo_msg_out
--rw-r--r--   0 runner    (1001) docker     (123)       67 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/gr_wo_msg_raw
--rw-r--r--   0 runner    (1001) docker     (123)      448 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/vr_out
--rw-r--r--   0 runner    (1001) docker     (123)      271 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/vr_raw
--rw-r--r--   0 runner    (1001) docker     (123)      517 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/vr_real_out
--rw-r--r--   0 runner    (1001) docker     (123)      205 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/data/vr_real_raw
--rw-r--r--   0 runner    (1001) docker     (123)    52089 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/test_composition.py
--rw-r--r--   0 runner    (1001) docker     (123)    55090 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/test_gradient.py
--rw-r--r--   0 runner    (1001) docker     (123)     1784 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/test_misc.py
--rw-r--r--   0 runner    (1001) docker     (123)    21831 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/test_power.py
--rw-r--r--   0 runner    (1001) docker     (123)     9052 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/stats/tests/test_subsample.py
--rw-r--r--   0 runner    (1001) docker     (123)      437 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/test.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.880321 scikit-bio-0.5.8/skbio/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tests/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     1553 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tests/test_base.py
--rw-r--r--   0 runner    (1001) docker     (123)    13167 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tests/test_workflow.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.880321 scikit-bio-0.5.8/skbio/tree/
--rw-r--r--   0 runner    (1001) docker     (123)     5810 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tree/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)      748 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tree/_exception.py
--rw-r--r--   0 runner    (1001) docker     (123)    10672 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tree/_majority_rule.py
--rw-r--r--   0 runner    (1001) docker     (123)    10825 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tree/_nj.py
--rw-r--r--   0 runner    (1001) docker     (123)   103662 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tree/_tree.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.880321 scikit-bio-0.5.8/skbio/tree/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tree/tests/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)     7851 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tree/tests/test_majority_rule.py
--rw-r--r--   0 runner    (1001) docker     (123)     8368 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tree/tests/test_nj.py
--rw-r--r--   0 runner    (1001) docker     (123)    64583 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/tree/tests/test_tree.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.880321 scikit-bio-0.5.8/skbio/util/
--rw-r--r--   0 runner    (1001) docker     (123)     1869 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/util/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)    12180 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/util/_decorator.py
--rw-r--r--   0 runner    (1001) docker     (123)      639 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/util/_exception.py
--rw-r--r--   0 runner    (1001) docker     (123)     5575 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/util/_misc.py
--rw-r--r--   0 runner    (1001) docker     (123)    13763 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/util/_testing.py
--rw-r--r--   0 runner    (1001) docker     (123)     1509 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/util/_warning.py
-drwxr-xr-x   0 runner    (1001) docker     (123)        0 2022-12-09 21:40:48.880321 scikit-bio-0.5.8/skbio/util/tests/
--rw-r--r--   0 runner    (1001) docker     (123)      353 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/util/tests/__init__.py
--rw-r--r--   0 runner    (1001) docker     (123)    11871 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/util/tests/test_decorator.py
--rw-r--r--   0 runner    (1001) docker     (123)     8027 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/util/tests/test_misc.py
--rw-r--r--   0 runner    (1001) docker     (123)    10360 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/util/tests/test_testing.py
--rw-r--r--   0 runner    (1001) docker     (123)    18819 2022-12-09 21:40:39.000000 scikit-bio-0.5.8/skbio/workflow.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.817829 scikit-bio-0.5.9/
+-rw-r--r--   0 runner    (1001) docker     (123)    83758 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/CHANGELOG.md
+-rw-r--r--   0 runner    (1001) docker     (123)     1515 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/LICENSE.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      335 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/MANIFEST.in
+-rw-r--r--   0 runner    (1001) docker     (123)     1329 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/Makefile
+-rw-r--r--   0 runner    (1001) docker     (123)     9505 2023-08-02 22:34:45.817829 scikit-bio-0.5.9/PKG-INFO
+-rw-r--r--   0 runner    (1001) docker     (123)     7393 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/README.rst
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.729823 scikit-bio-0.5.9/assets/
+-rw-r--r--   0 runner    (1001) docker     (123)  2052678 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/assets/.no.gif
+-rw-r--r--   0 runner    (1001) docker     (123)    15003 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/assets/horizontal_powered_by.png
+-rw-r--r--   0 runner    (1001) docker     (123)    14388 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/assets/horizontal_powered_by.svg
+-rw-r--r--   0 runner    (1001) docker     (123)     6461 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/assets/logo-inverted.svg
+-rw-r--r--   0 runner    (1001) docker     (123)    12397 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/assets/logo.png
+-rw-r--r--   0 runner    (1001) docker     (123)     6461 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/assets/logo.svg
+-rw-r--r--   0 runner    (1001) docker     (123)    63684 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/assets/logo_and_powered.pdf
+-rw-r--r--   0 runner    (1001) docker     (123)    15865 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/assets/vertical_powered_by.png
+-rw-r--r--   0 runner    (1001) docker     (123)    13932 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/assets/vertical_powered_by.svg
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.733823 scikit-bio-0.5.9/ci/
+-rw-r--r--   0 runner    (1001) docker     (123)      153 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/ci/aarch64.conda_requirements.txt
+-rw-r--r--   0 runner    (1001) docker     (123)       20 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/ci/aarch64.requirements.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      128 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/ci/conda_host_env.yml
+-rw-r--r--   0 runner    (1001) docker     (123)      173 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/ci/conda_requirements.txt
+-rw-r--r--   0 runner    (1001) docker     (123)       30 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/ci/requirements.doc.txt
+-rw-r--r--   0 runner    (1001) docker     (123)       44 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/ci/requirements.lint.txt
+-rw-r--r--   0 runner    (1001) docker     (123)       26 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/ci/requirements.test.txt
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.733823 scikit-bio-0.5.9/doc/
+-rw-r--r--   0 runner    (1001) docker     (123)     7105 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/Makefile
+-rw-r--r--   0 runner    (1001) docker     (123)     8419 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/README.md
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.733823 scikit-bio-0.5.9/doc/source/
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.733823 scikit-bio-0.5.9/doc/source/_static/
+-rw-r--r--   0 runner    (1001) docker     (123)     2803 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/_static/copybutton.js
+-rw-r--r--   0 runner    (1001) docker     (123)     2424 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/_static/style.css
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.733823 scikit-bio-0.5.9/doc/source/_templates/
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.733823 scikit-bio-0.5.9/doc/source/_templates/autosummary/
+-rw-r--r--   0 runner    (1001) docker     (123)      109 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/_templates/autosummary/attribute.rst
+-rw-r--r--   0 runner    (1001) docker     (123)     1582 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/_templates/autosummary/class.rst
+-rw-r--r--   0 runner    (1001) docker     (123)      106 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/_templates/autosummary/method.rst
+-rw-r--r--   0 runner    (1001) docker     (123)      444 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/_templates/autosummary/module.rst
+-rw-r--r--   0 runner    (1001) docker     (123)     1290 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/_templates/layout.html
+-rw-r--r--   0 runner    (1001) docker     (123)       32 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/alignment.rst
+-rw-r--r--   0 runner    (1001) docker     (123)    16928 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/conf.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.733823 scikit-bio-0.5.9/doc/source/development/
+-rw-r--r--   0 runner    (1001) docker     (123)    24418 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/development/coding_guidelines.rst
+-rw-r--r--   0 runner    (1001) docker     (123)     1870 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/development/new_module.rst
+-rw-r--r--   0 runner    (1001) docker     (123)       32 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/diversity.rst
+-rw-r--r--   0 runner    (1001) docker     (123)      781 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/index.rst
+-rw-r--r--   0 runner    (1001) docker     (123)       25 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/io.rst
+-rw-r--r--   0 runner    (1001) docker     (123)       31 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/metadata.rst
+-rw-r--r--   0 runner    (1001) docker     (123)       31 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/sequence.rst
+-rw-r--r--   0 runner    (1001) docker     (123)       28 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/stats.rst
+-rw-r--r--   0 runner    (1001) docker     (123)       27 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/tree.rst
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.733823 scikit-bio-0.5.9/doc/source/user/
+-rw-r--r--   0 runner    (1001) docker     (123)     3167 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/user/api_stability.rst
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.733823 scikit-bio-0.5.9/doc/source/user/assets/
+-rw-r--r--   0 runner    (1001) docker     (123)    40262 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/user/assets/api-lifecycle.png
+-rw-r--r--   0 runner    (1001) docker     (123)       27 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/util.rst
+-rw-r--r--   0 runner    (1001) docker     (123)       31 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/doc/source/workflow.rst
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.737824 scikit-bio-0.5.9/licenses/
+-rw-r--r--   0 runner    (1001) docker     (123)     1135 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/bx_python.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     4951 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/fastq-example-files-readme.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1582 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/flask.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     3415 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/ipython.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1574 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/nb-slideshow-template.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1498 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/qiita.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1712 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/scikit-learn.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1580 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/scipy.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1077 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/simde.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1060 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/sphinx-bootstrap-theme.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1557 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/ssw.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1491 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/licenses/verman.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      563 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/pyproject.toml
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.737824 scikit-bio-0.5.9/scikit_bio.egg-info/
+-rw-r--r--   0 runner    (1001) docker     (123)     9505 2023-08-02 22:34:45.000000 scikit-bio-0.5.9/scikit_bio.egg-info/PKG-INFO
+-rw-r--r--   0 runner    (1001) docker     (123)    31861 2023-08-02 22:34:45.000000 scikit-bio-0.5.9/scikit_bio.egg-info/SOURCES.txt
+-rw-r--r--   0 runner    (1001) docker     (123)        1 2023-08-02 22:34:45.000000 scikit-bio-0.5.9/scikit_bio.egg-info/dependency_links.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      153 2023-08-02 22:34:45.000000 scikit-bio-0.5.9/scikit_bio.egg-info/requires.txt
+-rw-r--r--   0 runner    (1001) docker     (123)        6 2023-08-02 22:34:45.000000 scikit-bio-0.5.9/scikit_bio.egg-info/top_level.txt
+-rw-r--r--   0 runner    (1001) docker     (123)       38 2023-08-02 22:34:45.817829 scikit-bio-0.5.9/setup.cfg
+-rw-r--r--   0 runner    (1001) docker     (123)     5843 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/setup.py
+-rw-r--r--   0 runner    (1001) docker     (123)   812925 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/simde-sse2.h
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.737824 scikit-bio-0.5.9/skbio/
+-rw-r--r--   0 runner    (1001) docker     (123)     2329 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     1687 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/_base.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.737824 scikit-bio-0.5.9/skbio/alignment/
+-rw-r--r--   0 runner    (1001) docker     (123)     6612 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8579 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/_indexing.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.737824 scikit-bio-0.5.9/skbio/alignment/_lib/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/_lib/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)    32006 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/_lib/ssw.c
+-rw-r--r--   0 runner    (1001) docker     (123)     7156 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/_lib/ssw.h
+-rw-r--r--   0 runner    (1001) docker     (123)    44831 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/_pairwise.py
+-rw-r--r--   0 runner    (1001) docker     (123)     2605 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/_repr.py
+-rw-r--r--   0 runner    (1001) docker     (123)   670691 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/_ssw_wrapper.c
+-rw-r--r--   0 runner    (1001) docker     (123)    25444 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/_ssw_wrapper.pyx
+-rw-r--r--   0 runner    (1001) docker     (123)    81262 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/_tabular_msa.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.741824 scikit-bio-0.5.9/skbio/alignment/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/tests/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)    35055 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/tests/test_pairwise.py
+-rw-r--r--   0 runner    (1001) docker     (123)    32597 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/tests/test_ssw.py
+-rw-r--r--   0 runner    (1001) docker     (123)   146556 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/alignment/tests/test_tabular_msa.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.741824 scikit-bio-0.5.9/skbio/diversity/
+-rw-r--r--   0 runner    (1001) docker     (123)    17143 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)    10602 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/_block.py
+-rw-r--r--   0 runner    (1001) docker     (123)    17038 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/_driver.py
+-rw-r--r--   0 runner    (1001) docker     (123)   347189 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/_phylogenetic.c
+-rw-r--r--   0 runner    (1001) docker     (123)     7018 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/_phylogenetic.pyx
+-rw-r--r--   0 runner    (1001) docker     (123)     5686 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/_util.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.741824 scikit-bio-0.5.9/skbio/diversity/alpha/
+-rw-r--r--   0 runner    (1001) docker     (123)     2159 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     4650 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/_ace.py
+-rw-r--r--   0 runner    (1001) docker     (123)    25177 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/_base.py
+-rw-r--r--   0 runner    (1001) docker     (123)     6442 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/_chao1.py
+-rw-r--r--   0 runner    (1001) docker     (123)     5557 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/_faith_pd.py
+-rw-r--r--   0 runner    (1001) docker     (123)     3679 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/_gini.py
+-rw-r--r--   0 runner    (1001) docker     (123)    17448 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/_lladser.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.741824 scikit-bio-0.5.9/skbio/diversity/alpha/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/__init__.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.725823 scikit-bio-0.5.9/skbio/diversity/alpha/tests/data/
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.741824 scikit-bio-0.5.9/skbio/diversity/alpha/tests/data/qiime-191-tt/
+-rw-r--r--   0 runner    (1001) docker     (123)      161 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/data/qiime-191-tt/README.md
+-rw-r--r--   0 runner    (1001) docker     (123)      102 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/data/qiime-191-tt/faith-pd.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      291 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/data/qiime-191-tt/otu-table.tsv
+-rw-r--r--   0 runner    (1001) docker     (123)      266 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/data/qiime-191-tt/tree.nwk
+-rw-r--r--   0 runner    (1001) docker     (123)     1497 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_ace.py
+-rw-r--r--   0 runner    (1001) docker     (123)    10327 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_base.py
+-rw-r--r--   0 runner    (1001) docker     (123)     2865 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_chao1.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8646 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_faith_pd.py
+-rw-r--r--   0 runner    (1001) docker     (123)     2314 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_gini.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8393 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_lladser.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.741824 scikit-bio-0.5.9/skbio/diversity/beta/
+-rw-r--r--   0 runner    (1001) docker     (123)     1017 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/beta/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)    23491 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/beta/_unifrac.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.741824 scikit-bio-0.5.9/skbio/diversity/beta/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/beta/tests/__init__.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.725823 scikit-bio-0.5.9/skbio/diversity/beta/tests/data/
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.745824 scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/
+-rw-r--r--   0 runner    (1001) docker     (123)     1400 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/README.md
+-rw-r--r--   0 runner    (1001) docker     (123)      291 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/otu-table.tsv
+-rw-r--r--   0 runner    (1001) docker     (123)      266 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/tree.nwk
+-rw-r--r--   0 runner    (1001) docker     (123)      950 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/unweighted_unifrac_dm.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      982 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/weighted_normalized_unifrac_dm.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      950 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/weighted_unifrac_dm.txt
+-rw-r--r--   0 runner    (1001) docker     (123)    29880 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/beta/tests/test_unifrac.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.745824 scikit-bio-0.5.9/skbio/diversity/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/tests/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8358 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/tests/test_block.py
+-rw-r--r--   0 runner    (1001) docker     (123)    37085 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/tests/test_driver.py
+-rw-r--r--   0 runner    (1001) docker     (123)    10821 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/diversity/tests/test_util.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.745824 scikit-bio-0.5.9/skbio/io/
+-rw-r--r--   0 runner    (1001) docker     (123)     8181 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     2744 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/_exception.py
+-rw-r--r--   0 runner    (1001) docker     (123)     2615 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/_fileobject.py
+-rw-r--r--   0 runner    (1001) docker     (123)     6801 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/_iosources.py
+-rw-r--r--   0 runner    (1001) docker     (123)      602 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/_warning.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.749824 scikit-bio-0.5.9/skbio/io/format/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     7483 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/_base.py
+-rw-r--r--   0 runner    (1001) docker     (123)     2442 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/_blast.py
+-rw-r--r--   0 runner    (1001) docker     (123)    10702 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/_sequence_feature_vocabulary.py
+-rw-r--r--   0 runner    (1001) docker     (123)     5795 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/binary_dm.py
+-rw-r--r--   0 runner    (1001) docker     (123)    12815 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/blast6.py
+-rw-r--r--   0 runner    (1001) docker     (123)    15144 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/blast7.py
+-rw-r--r--   0 runner    (1001) docker     (123)    12164 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/clustal.py
+-rw-r--r--   0 runner    (1001) docker     (123)    53958 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/embl.py
+-rw-r--r--   0 runner    (1001) docker     (123)     1159 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/emptyfile.py
+-rw-r--r--   0 runner    (1001) docker     (123)    35631 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/fasta.py
+-rw-r--r--   0 runner    (1001) docker     (123)    22117 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/fastq.py
+-rw-r--r--   0 runner    (1001) docker     (123)    25100 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/genbank.py
+-rw-r--r--   0 runner    (1001) docker     (123)    18541 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/gff3.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8088 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/lsmat.py
+-rw-r--r--   0 runner    (1001) docker     (123)    20494 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/newick.py
+-rw-r--r--   0 runner    (1001) docker     (123)    14421 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/ordination.py
+-rw-r--r--   0 runner    (1001) docker     (123)    14164 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/phylip.py
+-rw-r--r--   0 runner    (1001) docker     (123)    10094 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/qseq.py
+-rw-r--r--   0 runner    (1001) docker     (123)    29559 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/stockholm.py
+-rw-r--r--   0 runner    (1001) docker     (123)    13183 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/taxdump.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.749824 scikit-bio-0.5.9/skbio/io/format/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/__init__.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.789827 scikit-bio-0.5.9/skbio/io/format/tests/data/
+-rw-r--r--   0 runner    (1001) docker     (123)        9 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast6_custom_minimal
+-rw-r--r--   0 runner    (1001) docker     (123)       53 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast6_custom_mixed_nans
+-rw-r--r--   0 runner    (1001) docker     (123)      113 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast6_custom_multi_line
+-rw-r--r--   0 runner    (1001) docker     (123)       31 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast6_custom_single_line
+-rw-r--r--   0 runner    (1001) docker     (123)      143 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast6_default_multi_line
+-rw-r--r--   0 runner    (1001) docker     (123)       47 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast6_default_single_line
+-rw-r--r--   0 runner    (1001) docker     (123)      141 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast6_invalid_column_types
+-rw-r--r--   0 runner    (1001) docker     (123)       36 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast6_invalid_number_of_columns
+-rw-r--r--   0 runner    (1001) docker     (123)       95 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast6_invalid_type_in_column
+-rw-r--r--   0 runner    (1001) docker     (123)       93 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_custom_minimal
+-rw-r--r--   0 runner    (1001) docker     (123)      396 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_custom_mixed_nans
+-rw-r--r--   0 runner    (1001) docker     (123)      410 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_custom_multi_line
+-rw-r--r--   0 runner    (1001) docker     (123)      377 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_custom_single_line
+-rw-r--r--   0 runner    (1001) docker     (123)      424 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_default_multi_line
+-rw-r--r--   0 runner    (1001) docker     (123)      324 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_default_single_line
+-rw-r--r--   0 runner    (1001) docker     (123)      415 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_invalid_differing_fields
+-rw-r--r--   0 runner    (1001) docker     (123)       57 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_invalid_for_sniffer
+-rw-r--r--   0 runner    (1001) docker     (123)       53 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_invalid_for_sniffer_2
+-rw-r--r--   0 runner    (1001) docker     (123)      144 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_invalid_gibberish
+-rw-r--r--   0 runner    (1001) docker     (123)      408 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_invalid_no_data
+-rw-r--r--   0 runner    (1001) docker     (123)      410 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_invalid_too_many_columns
+-rw-r--r--   0 runner    (1001) docker     (123)      179 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/blast7_invalid_unrecognized_field
+-rw-r--r--   0 runner    (1001) docker     (123)     1105 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/embl_constructed
+-rw-r--r--   0 runner    (1001) docker     (123)     3923 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/embl_feature_level_record
+-rw-r--r--   0 runner    (1001) docker     (123)     3156 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/embl_feature_level_record_no_FT
+-rw-r--r--   0 runner    (1001) docker     (123)     6749 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/embl_multi_records
+-rw-r--r--   0 runner    (1001) docker     (123)     4136 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/embl_single_record
+-rw-r--r--   0 runner    (1001) docker     (123)      272 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/embl_single_record_lower
+-rw-r--r--   0 runner    (1001) docker     (123)     4022 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/embl_single_record_simple
+-rw-r--r--   0 runner    (1001) docker     (123)      344 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/embl_single_record_upper
+-rw-r--r--   0 runner    (1001) docker     (123)     2333 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/embl_uniprot_record
+-rw-r--r--   0 runner    (1001) docker     (123)      274 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/embl_w_beginning_whitespace
+-rw-r--r--   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/empty
+-rw-r--r--   0 runner    (1001) docker     (123)      608 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_diff_ids.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      669 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_double_qual.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      669 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_double_seq.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      609 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_long_qual.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      482 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_no_qual.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      608 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_del.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      608 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_escape.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      610 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_null.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      609 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_space.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      609 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_tab.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      608 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_unit_sep.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      608 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_vtab.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      607 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_short_qual.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      628 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_spaces.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      629 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_tabs.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      548 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_at_plus.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      582 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_at_qual.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      522 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_at_seq.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      562 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_in_plus.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      607 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_in_qual.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      535 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_in_seq.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      513 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_in_title.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      174 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_10_seqs
+-rw-r--r--   0 runner    (1001) docker     (123)       63 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_3_seqs_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       66 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_3_seqs_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)      150 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_5_blanks_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      205 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_5_ws_lines_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      151 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_6_blanks_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      217 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_6_ws_lines_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      157 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_blank_lines_between_records
+-rw-r--r--   0 runner    (1001) docker     (123)      152 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_blanks_end_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)       75 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_description_newline_replacement_empty_str
+-rw-r--r--   0 runner    (1001) docker     (123)      109 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_description_newline_replacement_multi_char
+-rw-r--r--   0 runner    (1001) docker     (123)       87 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_description_newline_replacement_none
+-rw-r--r--   0 runner    (1001) docker     (123)       17 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_id_whitespace_replacement_empty_str
+-rw-r--r--   0 runner    (1001) docker     (123)       53 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_id_whitespace_replacement_multi_char
+-rw-r--r--   0 runner    (1001) docker     (123)       29 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_id_whitespace_replacement_none
+-rw-r--r--   0 runner    (1001) docker     (123)      186 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_after_10_seqs
+-rw-r--r--   0 runner    (1001) docker     (123)       52 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_blank_line_after_header
+-rw-r--r--   0 runner    (1001) docker     (123)       64 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_blank_line_within_sequence
+-rw-r--r--   0 runner    (1001) docker     (123)       46 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_blank_sequence
+-rw-r--r--   0 runner    (1001) docker     (123)       40 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_legacy_format
+-rw-r--r--   0 runner    (1001) docker     (123)       16 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_missing_header
+-rw-r--r--   0 runner    (1001) docker     (123)       46 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_missing_seq_data_first
+-rw-r--r--   0 runner    (1001) docker     (123)       47 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_missing_seq_data_last
+-rw-r--r--   0 runner    (1001) docker     (123)       45 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_missing_seq_data_middle
+-rw-r--r--   0 runner    (1001) docker     (123)       63 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_whitespace_line_after_header
+-rw-r--r--   0 runner    (1001) docker     (123)       69 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_whitespace_only_line_within_sequence
+-rw-r--r--   0 runner    (1001) docker     (123)       57 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_invalid_whitespace_only_sequence
+-rw-r--r--   0 runner    (1001) docker     (123)       32 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_max_width_1
+-rw-r--r--   0 runner    (1001) docker     (123)      150 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_max_width_5
+-rw-r--r--   0 runner    (1001) docker     (123)       51 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_mixed_qual_scores
+-rw-r--r--   0 runner    (1001) docker     (123)      145 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_multi_seq
+-rw-r--r--   0 runner    (1001) docker     (123)      114 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_multi_seq_roundtrip
+-rw-r--r--   0 runner    (1001) docker     (123)       80 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_prot_seqs_odd_labels
+-rw-r--r--   0 runner    (1001) docker     (123)       18 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_single_bio_seq_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       21 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_single_bio_seq_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       18 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_single_dna_seq_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       21 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_single_dna_seq_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       17 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_single_prot_seq_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       19 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_single_prot_seq_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       18 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_single_rna_seq_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       21 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_single_rna_seq_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       23 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_single_seq
+-rw-r--r--   0 runner    (1001) docker     (123)       72 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_tabular_msa_different_type
+-rw-r--r--   0 runner    (1001) docker     (123)      277 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_ws_lines_between_records
+-rw-r--r--   0 runner    (1001) docker     (123)      233 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fasta_ws_lines_end_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)       92 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_5_blanks_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      147 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_5_ws_lines_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)       90 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_blank_lines
+-rw-r--r--   0 runner    (1001) docker     (123)       88 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_blank_after_header
+-rw-r--r--   0 runner    (1001) docker     (123)       88 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_blank_after_plus
+-rw-r--r--   0 runner    (1001) docker     (123)       88 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_blank_after_seq
+-rw-r--r--   0 runner    (1001) docker     (123)       89 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_blank_in_seq_at_symbol
+-rw-r--r--   0 runner    (1001) docker     (123)       89 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_blank_within_qual
+-rw-r--r--   0 runner    (1001) docker     (123)       89 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_blank_within_seq
+-rw-r--r--   0 runner    (1001) docker     (123)       23 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_missing_header
+-rw-r--r--   0 runner    (1001) docker     (123)      103 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_missing_seq_data
+-rw-r--r--   0 runner    (1001) docker     (123)       99 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_ws_line_after_header
+-rw-r--r--   0 runner    (1001) docker     (123)       99 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_ws_line_after_plus
+-rw-r--r--   0 runner    (1001) docker     (123)       99 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_ws_line_after_seq
+-rw-r--r--   0 runner    (1001) docker     (123)      100 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_ws_line_within_qual
+-rw-r--r--   0 runner    (1001) docker     (123)      100 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_invalid_ws_line_within_seq
+-rw-r--r--   0 runner    (1001) docker     (123)       95 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_multi_blank_between_records
+-rw-r--r--   0 runner    (1001) docker     (123)       94 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_multi_blank_end_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)       93 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_multi_blank_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)       87 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_multi_seq_sanger
+-rw-r--r--   0 runner    (1001) docker     (123)      331 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_multi_whitespace_stripping
+-rw-r--r--   0 runner    (1001) docker     (123)      159 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_multi_ws_lines_between_records
+-rw-r--r--   0 runner    (1001) docker     (123)      183 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_multi_ws_lines_end_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      159 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_multi_ws_lines_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)       33 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_single_seq_illumina1.3
+-rw-r--r--   0 runner    (1001) docker     (123)      118 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_single_seq_illumina1.8
+-rw-r--r--   0 runner    (1001) docker     (123)      123 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_whitespace_only_lines
+-rw-r--r--   0 runner    (1001) docker     (123)      885 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_wrapping_as_illumina_no_description
+-rw-r--r--   0 runner    (1001) docker     (123)      885 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_wrapping_as_sanger_no_description
+-rw-r--r--   0 runner    (1001) docker     (123)      898 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_wrapping_original_sanger_no_description
+-rw-r--r--   0 runner    (1001) docker     (123)       94 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_writer_illumina1.3_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       94 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_writer_sanger_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       94 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_writer_sanger_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       85 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/genbank_5_blanks_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)       86 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/genbank_6_blanks_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)       71 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/genbank_missing_locus_name
+-rw-r--r--   0 runner    (1001) docker     (123)     1598 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/genbank_multi_records
+-rw-r--r--   0 runner    (1001) docker     (123)     1086 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/genbank_single_record
+-rw-r--r--   0 runner    (1001) docker     (123)       90 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/genbank_single_record_lower
+-rw-r--r--   0 runner    (1001) docker     (123)       90 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/genbank_single_record_upper
+-rw-r--r--   0 runner    (1001) docker     (123)       82 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/genbank_w_beginning_whitespace
+-rw-r--r--   0 runner    (1001) docker     (123)      206 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/gff3_bad_missing_directive
+-rw-r--r--   0 runner    (1001) docker     (123)      459 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/gff3_bad_wrong_columns
+-rw-r--r--   0 runner    (1001) docker     (123)      106 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/gff3_dna
+-rw-r--r--   0 runner    (1001) docker     (123)      881 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/gff3_multi_record
+-rw-r--r--   0 runner    (1001) docker     (123)      505 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/gff3_single_record
+-rw-r--r--   0 runner    (1001) docker     (123)      426 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/illumina_full_range_as_illumina.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      426 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/illumina_full_range_as_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      426 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/illumina_full_range_original_illumina.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      505 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/legacy9_and_blast7_default
+-rw-r--r--   0 runner    (1001) docker     (123)      404 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/legacy9_invalid_differing_fields
+-rw-r--r--   0 runner    (1001) docker     (123)      262 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/legacy9_invalid_too_many_columns
+-rw-r--r--   0 runner    (1001) docker     (123)      296 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/legacy9_mixed_nans
+-rw-r--r--   0 runner    (1001) docker     (123)      345 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/legacy9_multi_line
+-rw-r--r--   0 runner    (1001) docker     (123)      251 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/legacy9_single_line
+-rw-r--r--   0 runner    (1001) docker     (123)     8388 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/longreads_as_illumina.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)     8388 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/longreads_as_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)     9466 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/longreads_original_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      767 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/misc_dna_as_illumina.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      767 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/misc_dna_as_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      767 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/misc_dna_original_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      774 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/misc_rna_as_illumina.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      774 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/misc_rna_as_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      774 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/misc_rna_original_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      351 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_L&L_CA_data_scores
+-rw-r--r--   0 runner    (1001) docker     (123)     1584 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_PCoA_sample_data_3_scores
+-rw-r--r--   0 runner    (1001) docker     (123)     4569 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error1
+-rw-r--r--   0 runner    (1001) docker     (123)      306 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error10
+-rw-r--r--   0 runner    (1001) docker     (123)     4578 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error11
+-rw-r--r--   0 runner    (1001) docker     (123)     1569 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error12
+-rw-r--r--   0 runner    (1001) docker     (123)     1569 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error13
+-rw-r--r--   0 runner    (1001) docker     (123)     1584 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error14
+-rw-r--r--   0 runner    (1001) docker     (123)     1543 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error15
+-rw-r--r--   0 runner    (1001) docker     (123)     4579 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error16
+-rw-r--r--   0 runner    (1001) docker     (123)     4531 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error17
+-rw-r--r--   0 runner    (1001) docker     (123)     1570 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error18
+-rw-r--r--   0 runner    (1001) docker     (123)      304 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error19
+-rw-r--r--   0 runner    (1001) docker     (123)     4555 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error2
+-rw-r--r--   0 runner    (1001) docker     (123)      306 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error20
+-rw-r--r--   0 runner    (1001) docker     (123)       10 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error21
+-rw-r--r--   0 runner    (1001) docker     (123)       18 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error22
+-rw-r--r--   0 runner    (1001) docker     (123)       42 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error23
+-rw-r--r--   0 runner    (1001) docker     (123)       75 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error24
+-rw-r--r--   0 runner    (1001) docker     (123)     4567 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error3
+-rw-r--r--   0 runner    (1001) docker     (123)     4569 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error4
+-rw-r--r--   0 runner    (1001) docker     (123)     4569 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error5
+-rw-r--r--   0 runner    (1001) docker     (123)     4558 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error6
+-rw-r--r--   0 runner    (1001) docker     (123)       65 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error7
+-rw-r--r--   0 runner    (1001) docker     (123)     1570 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error8
+-rw-r--r--   0 runner    (1001) docker     (123)      304 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error9
+-rw-r--r--   0 runner    (1001) docker     (123)     3396 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_example2_scores
+-rw-r--r--   0 runner    (1001) docker     (123)     4579 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_example3_scores
+-rw-r--r--   0 runner    (1001) docker     (123)     2293 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_site
+-rw-r--r--   0 runner    (1001) docker     (123)     2291 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_site_constraints
+-rw-r--r--   0 runner    (1001) docker     (123)     2065 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_species
+-rw-r--r--   0 runner    (1001) docker     (123)     2062 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_PCoA_site
+-rw-r--r--   0 runner    (1001) docker     (123)     1786 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_site
+-rw-r--r--   0 runner    (1001) docker     (123)     1787 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_site_constraints
+-rw-r--r--   0 runner    (1001) docker     (123)     1070 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_species
+-rw-r--r--   0 runner    (1001) docker     (123)       80 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_dna_3_seqs
+-rw-r--r--   0 runner    (1001) docker     (123)       70 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_empty_line_after_header
+-rw-r--r--   0 runner    (1001) docker     (123)       70 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_empty_line_before_header
+-rw-r--r--   0 runner    (1001) docker     (123)       70 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_empty_line_between_seqs
+-rw-r--r--   0 runner    (1001) docker     (123)       80 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_header_too_long
+-rw-r--r--   0 runner    (1001) docker     (123)       68 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_header_too_short
+-rw-r--r--   0 runner    (1001) docker     (123)       64 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_no_header
+-rw-r--r--   0 runner    (1001) docker     (123)       71 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_seq_too_long
+-rw-r--r--   0 runner    (1001) docker     (123)       67 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_seq_too_short
+-rw-r--r--   0 runner    (1001) docker     (123)       37 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_too_few_seqs
+-rw-r--r--   0 runner    (1001) docker     (123)       69 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_too_many_seqs
+-rw-r--r--   0 runner    (1001) docker     (123)       15 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_zero_seq_len
+-rw-r--r--   0 runner    (1001) docker     (123)        4 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_invalid_zero_seqs
+-rw-r--r--   0 runner    (1001) docker     (123)       42 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_single_seq_long
+-rw-r--r--   0 runner    (1001) docker     (123)       16 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_single_seq_short
+-rw-r--r--   0 runner    (1001) docker     (123)       69 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_two_chunks
+-rw-r--r--   0 runner    (1001) docker     (123)      106 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_variable_length_ids
+-rw-r--r--   0 runner    (1001) docker     (123)       81 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_varied_whitespace_in_seqs
+-rw-r--r--   0 runner    (1001) docker     (123)      106 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_whitespace_in_header_1
+-rw-r--r--   0 runner    (1001) docker     (123)       86 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_whitespace_in_header_2
+-rw-r--r--   0 runner    (1001) docker     (123)      102 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/phylip_whitespace_in_header_3
+-rw-r--r--   0 runner    (1001) docker     (123)      217 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qseq_invalid_filter
+-rw-r--r--   0 runner    (1001) docker     (123)      217 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qseq_invalid_lane
+-rw-r--r--   0 runner    (1001) docker     (123)      217 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qseq_invalid_read
+-rw-r--r--   0 runner    (1001) docker     (123)      216 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qseq_invalid_tile
+-rw-r--r--   0 runner    (1001) docker     (123)      234 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qseq_invalid_x
+-rw-r--r--   0 runner    (1001) docker     (123)      251 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qseq_invalid_y
+-rw-r--r--   0 runner    (1001) docker     (123)      226 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qseq_multi_seq_illumina1.3
+-rw-r--r--   0 runner    (1001) docker     (123)      217 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qseq_single_seq_sanger
+-rw-r--r--   0 runner    (1001) docker     (123)       69 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_2_seqs_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)      100 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_3_seqs_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       89 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_3_seqs_defaults_desc_mismatch
+-rw-r--r--   0 runner    (1001) docker     (123)      124 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_3_seqs_defaults_extra
+-rw-r--r--   0 runner    (1001) docker     (123)       89 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_3_seqs_defaults_id_mismatch
+-rw-r--r--   0 runner    (1001) docker     (123)       97 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_3_seqs_defaults_length_mismatch
+-rw-r--r--   0 runner    (1001) docker     (123)      105 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_3_seqs_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)      218 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_5_blanks_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      273 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_5_ws_lines_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      219 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_6_blanks_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      285 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_6_ws_lines_start_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      223 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_blank_lines_between_records
+-rw-r--r--   0 runner    (1001) docker     (123)      220 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_blanks_end_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      104 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_description_newline_replacement_empty_str
+-rw-r--r--   0 runner    (1001) docker     (123)      138 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_description_newline_replacement_multi_char
+-rw-r--r--   0 runner    (1001) docker     (123)      116 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_description_newline_replacement_none
+-rw-r--r--   0 runner    (1001) docker     (123)       22 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_id_whitespace_replacement_empty_str
+-rw-r--r--   0 runner    (1001) docker     (123)       58 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_id_whitespace_replacement_multi_char
+-rw-r--r--   0 runner    (1001) docker     (123)       34 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_id_whitespace_replacement_none
+-rw-r--r--   0 runner    (1001) docker     (123)      101 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_blank_line_after_header
+-rw-r--r--   0 runner    (1001) docker     (123)      102 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_blank_line_within_seq
+-rw-r--r--   0 runner    (1001) docker     (123)       84 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_blank_sequence
+-rw-r--r--   0 runner    (1001) docker     (123)       47 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_legacy_format
+-rw-r--r--   0 runner    (1001) docker     (123)       19 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_missing_header
+-rw-r--r--   0 runner    (1001) docker     (123)       92 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_missing_qual_scores_first
+-rw-r--r--   0 runner    (1001) docker     (123)       85 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_missing_qual_scores_last
+-rw-r--r--   0 runner    (1001) docker     (123)       83 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_missing_qual_scores_middle
+-rw-r--r--   0 runner    (1001) docker     (123)      102 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_qual_scores_float
+-rw-r--r--   0 runner    (1001) docker     (123)      100 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_qual_scores_negative
+-rw-r--r--   0 runner    (1001) docker     (123)      100 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_qual_scores_over_255
+-rw-r--r--   0 runner    (1001) docker     (123)      101 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_qual_scores_string
+-rw-r--r--   0 runner    (1001) docker     (123)      113 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_whitespace_line_in_seq
+-rw-r--r--   0 runner    (1001) docker     (123)       95 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_whitespace_only_sequence
+-rw-r--r--   0 runner    (1001) docker     (123)      112 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_invalid_ws_line_after_header
+-rw-r--r--   0 runner    (1001) docker     (123)       40 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_max_width_1
+-rw-r--r--   0 runner    (1001) docker     (123)      213 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_max_width_5
+-rw-r--r--   0 runner    (1001) docker     (123)      213 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_multi_seq
+-rw-r--r--   0 runner    (1001) docker     (123)      166 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_multi_seq_roundtrip
+-rw-r--r--   0 runner    (1001) docker     (123)      131 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_prot_seqs_odd_labels
+-rw-r--r--   0 runner    (1001) docker     (123)       21 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_single_bio_seq_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       21 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_single_dna_seq_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       23 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_single_nuc_seq_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       22 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_single_prot_seq_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       21 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_single_rna_seq_non_defaults
+-rw-r--r--   0 runner    (1001) docker     (123)       40 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_single_seq
+-rw-r--r--   0 runner    (1001) docker     (123)       85 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_tabular_msa_different_type
+-rw-r--r--   0 runner    (1001) docker     (123)      333 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_ws_lines_between_records
+-rw-r--r--   0 runner    (1001) docker     (123)      309 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/qual_ws_lines_end_of_file
+-rw-r--r--   0 runner    (1001) docker     (123)      550 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/sanger_full_range_as_illumina.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      550 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/sanger_full_range_as_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      550 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/sanger_full_range_original_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      450 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/solexa_full_range_original_solexa.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      404 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_all_data_types
+-rw-r--r--   0 runner    (1001) docker     (123)       47 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_blank_lines
+-rw-r--r--   0 runner    (1001) docker     (123)       82 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_data_only
+-rw-r--r--   0 runner    (1001) docker     (123)      142 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_different_padding
+-rw-r--r--   0 runner    (1001) docker     (123)       70 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_differing_gc_data_length
+-rw-r--r--   0 runner    (1001) docker     (123)       62 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_differing_gr_data_length
+-rw-r--r--   0 runner    (1001) docker     (123)       70 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_differing_seq_lengths
+-rw-r--r--   0 runner    (1001) docker     (123)      180 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_duplicate_gc
+-rw-r--r--   0 runner    (1001) docker     (123)      195 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_duplicate_gr
+-rw-r--r--   0 runner    (1001) docker     (123)      159 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_duplicate_sequence_names
+-rw-r--r--   0 runner    (1001) docker     (123)       73 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_duplicate_tree_ids
+-rw-r--r--   0 runner    (1001) docker     (123)      652 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_extensive
+-rw-r--r--   0 runner    (1001) docker     (123)      660 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_extensive_mixed
+-rw-r--r--   0 runner    (1001) docker     (123)       45 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_invalid_data_type
+-rw-r--r--   0 runner    (1001) docker     (123)      183 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_invalid_nonexistent_gr
+-rw-r--r--   0 runner    (1001) docker     (123)      110 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_invalid_nonexistent_gs
+-rw-r--r--   0 runner    (1001) docker     (123)       24 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_malformed_data_line
+-rw-r--r--   0 runner    (1001) docker     (123)       61 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_malformed_gc_line
+-rw-r--r--   0 runner    (1001) docker     (123)       27 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_malformed_gf_line
+-rw-r--r--   0 runner    (1001) docker     (123)       54 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_malformed_gr_line
+-rw-r--r--   0 runner    (1001) docker     (123)       43 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_malformed_gs_line
+-rw-r--r--   0 runner    (1001) docker     (123)       70 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_metadata_only
+-rw-r--r--   0 runner    (1001) docker     (123)       48 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_minimal
+-rw-r--r--   0 runner    (1001) docker     (123)      121 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_missing_footer
+-rw-r--r--   0 runner    (1001) docker     (123)      108 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_missing_header
+-rw-r--r--   0 runner    (1001) docker     (123)       66 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_missing_reference_items
+-rw-r--r--   0 runner    (1001) docker     (123)       45 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_missing_rn_tag
+-rw-r--r--   0 runner    (1001) docker     (123)       45 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_multi_line_tree_no_id
+-rw-r--r--   0 runner    (1001) docker     (123)       59 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_multi_line_tree_with_id
+-rw-r--r--   0 runner    (1001) docker     (123)      486 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_multiple_msa
+-rw-r--r--   0 runner    (1001) docker     (123)       99 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_multiple_multi_line_trees
+-rw-r--r--   0 runner    (1001) docker     (123)      319 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_multiple_references
+-rw-r--r--   0 runner    (1001) docker     (123)      100 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_multiple_trees
+-rw-r--r--   0 runner    (1001) docker     (123)       19 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_no_data
+-rw-r--r--   0 runner    (1001) docker     (123)      110 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_nonstring_labels
+-rw-r--r--   0 runner    (1001) docker     (123)      243 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_rna
+-rw-r--r--   0 runner    (1001) docker     (123)      156 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_runon_gf_no_whitespace
+-rw-r--r--   0 runner    (1001) docker     (123)      157 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_runon_gf_with_whitespace
+-rw-r--r--   0 runner    (1001) docker     (123)       87 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_runon_gs_no_whitespace
+-rw-r--r--   0 runner    (1001) docker     (123)       88 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_runon_gs_with_whitespace
+-rw-r--r--   0 runner    (1001) docker     (123)      150 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_runon_references
+-rw-r--r--   0 runner    (1001) docker     (123)      150 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_runon_references_mixed
+-rw-r--r--   0 runner    (1001) docker     (123)      119 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_single_reference
+-rw-r--r--   0 runner    (1001) docker     (123)       46 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_single_tree_with_id
+-rw-r--r--   0 runner    (1001) docker     (123)       32 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_single_tree_without_id
+-rw-r--r--   0 runner    (1001) docker     (123)      246 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_two_of_each_metadata
+-rw-r--r--   0 runner    (1001) docker     (123)       74 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_whitespace_only_lines
+-rw-r--r--   0 runner    (1001) docker     (123)     1159 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/taxdump_names.dmp
+-rw-r--r--   0 runner    (1001) docker     (123)     1249 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/taxdump_nodes.dmp
+-rw-r--r--   0 runner    (1001) docker     (123)      160 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/tsv_10_fields
+-rw-r--r--   0 runner    (1001) docker     (123)      133 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/tsv_8_fields
+-rw-r--r--   0 runner    (1001) docker     (123)       47 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/whitespace_only
+-rw-r--r--   0 runner    (1001) docker     (123)      963 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/wrapping_as_illumina.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      963 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/wrapping_as_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)      975 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/data/wrapping_original_sanger.fastq
+-rw-r--r--   0 runner    (1001) docker     (123)    13438 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_base.py
+-rw-r--r--   0 runner    (1001) docker     (123)     4775 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_binary_dm.py
+-rw-r--r--   0 runner    (1001) docker     (123)     6179 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_blast6.py
+-rw-r--r--   0 runner    (1001) docker     (123)     9968 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_blast7.py
+-rw-r--r--   0 runner    (1001) docker     (123)    11715 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_clustal.py
+-rw-r--r--   0 runner    (1001) docker     (123)    36301 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_embl.py
+-rw-r--r--   0 runner    (1001) docker     (123)     1097 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_emptyfile.py
+-rw-r--r--   0 runner    (1001) docker     (123)    47662 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_fasta.py
+-rw-r--r--   0 runner    (1001) docker     (123)    28808 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_fastq.py
+-rw-r--r--   0 runner    (1001) docker     (123)    15587 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_genbank.py
+-rw-r--r--   0 runner    (1001) docker     (123)    13406 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_gff3.py
+-rw-r--r--   0 runner    (1001) docker     (123)     9911 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_lsmat.py
+-rw-r--r--   0 runner    (1001) docker     (123)    12594 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_newick.py
+-rw-r--r--   0 runner    (1001) docker     (123)    11723 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_ordination.py
+-rw-r--r--   0 runner    (1001) docker     (123)     9882 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_phylip.py
+-rw-r--r--   0 runner    (1001) docker     (123)    13189 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_qseq.py
+-rw-r--r--   0 runner    (1001) docker     (123)     3795 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_sequence_feature_vocabulary.py
+-rw-r--r--   0 runner    (1001) docker     (123)    41921 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_stockholm.py
+-rw-r--r--   0 runner    (1001) docker     (123)     7847 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/format/tests/test_taxdump.py
+-rw-r--r--   0 runner    (1001) docker     (123)    43169 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/registry.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.789827 scikit-bio-0.5.9/skbio/io/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/__init__.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.789827 scikit-bio-0.5.9/skbio/io/tests/data/
+-rw-r--r--   0 runner    (1001) docker     (123)        5 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/data/big5_file
+-rw-r--r--   0 runner    (1001) docker     (123)       46 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/data/big5_file.bz2
+-rw-r--r--   0 runner    (1001) docker     (123)       35 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/data/big5_file.gz
+-rw-r--r--   0 runner    (1001) docker     (123)       53 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/data/example_file
+-rw-r--r--   0 runner    (1001) docker     (123)       84 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/data/example_file.bz2
+-rw-r--r--   0 runner    (1001) docker     (123)       83 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/data/example_file.gz
+-rw-r--r--   0 runner    (1001) docker     (123)       10 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/data/real_file
+-rw-r--r--   0 runner    (1001) docker     (123)       23 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/data/real_file_2
+-rw-r--r--   0 runner    (1001) docker     (123)     1484 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/test_iosources.py
+-rw-r--r--   0 runner    (1001) docker     (123)    63448 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/test_registry.py
+-rw-r--r--   0 runner    (1001) docker     (123)    23129 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/tests/test_util.py
+-rw-r--r--   0 runner    (1001) docker     (123)    11105 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/io/util.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.793827 scikit-bio-0.5.9/skbio/metadata/
+-rw-r--r--   0 runner    (1001) docker     (123)      720 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)   601338 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/_intersection.c
+-rw-r--r--   0 runner    (1001) docker     (123)    18666 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/_intersection.pyx
+-rw-r--r--   0 runner    (1001) docker     (123)    36249 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/_interval.py
+-rw-r--r--   0 runner    (1001) docker     (123)    18539 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/_mixin.py
+-rw-r--r--   0 runner    (1001) docker     (123)     6674 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/_repr.py
+-rw-r--r--   0 runner    (1001) docker     (123)    53258 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/_testing.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.793827 scikit-bio-0.5.9/skbio/metadata/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/tests/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8188 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/tests/test_intersection.py
+-rw-r--r--   0 runner    (1001) docker     (123)    32405 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/tests/test_interval.py
+-rw-r--r--   0 runner    (1001) docker     (123)     4518 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/metadata/tests/test_mixin.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.793827 scikit-bio-0.5.9/skbio/sequence/
+-rw-r--r--   0 runner    (1001) docker     (123)     7671 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)    12736 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/_dna.py
+-rw-r--r--   0 runner    (1001) docker     (123)    31064 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/_genetic_code.py
+-rw-r--r--   0 runner    (1001) docker     (123)    24487 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/_grammared_sequence.py
+-rw-r--r--   0 runner    (1001) docker     (123)    11243 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/_nucleotide_mixin.py
+-rw-r--r--   0 runner    (1001) docker     (123)     7000 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/_protein.py
+-rw-r--r--   0 runner    (1001) docker     (123)     4193 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/_repr.py
+-rw-r--r--   0 runner    (1001) docker     (123)    13231 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/_rna.py
+-rw-r--r--   0 runner    (1001) docker     (123)    73563 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/_sequence.py
+-rw-r--r--   0 runner    (1001) docker     (123)     5218 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/distance.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.793827 scikit-bio-0.5.9/skbio/sequence/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/tests/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     7215 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/tests/test_distance.py
+-rw-r--r--   0 runner    (1001) docker     (123)     1987 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/tests/test_dna.py
+-rw-r--r--   0 runner    (1001) docker     (123)    19666 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/tests/test_genetic_code.py
+-rw-r--r--   0 runner    (1001) docker     (123)    27075 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/tests/test_grammared_sequence.py
+-rw-r--r--   0 runner    (1001) docker     (123)    21247 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/tests/test_nucleotide_sequences.py
+-rw-r--r--   0 runner    (1001) docker     (123)     4890 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/tests/test_protein.py
+-rw-r--r--   0 runner    (1001) docker     (123)     2083 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/tests/test_rna.py
+-rw-r--r--   0 runner    (1001) docker     (123)   108459 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/sequence/tests/test_sequence.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.797827 scikit-bio-0.5.9/skbio/stats/
+-rw-r--r--   0 runner    (1001) docker     (123)      919 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)   283902 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/__subsample.c
+-rw-r--r--   0 runner    (1001) docker     (123)     1051 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/__subsample.pyx
+-rw-r--r--   0 runner    (1001) docker     (123)      869 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/_misc.py
+-rw-r--r--   0 runner    (1001) docker     (123)     7864 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/_subsample.py
+-rw-r--r--   0 runner    (1001) docker     (123)    55166 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/composition.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.797827 scikit-bio-0.5.9/skbio/stats/distance/
+-rw-r--r--   0 runner    (1001) docker     (123)     5626 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8598 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/_anosim.py
+-rw-r--r--   0 runner    (1001) docker     (123)    49177 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/_base.py
+-rw-r--r--   0 runner    (1001) docker     (123)     9530 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/_bioenv.py
+-rw-r--r--   0 runner    (1001) docker     (123)  1171270 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/_cutils.c
+-rw-r--r--   0 runner    (1001) docker     (123)     8992 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/_cutils.pyx
+-rw-r--r--   0 runner    (1001) docker     (123)    26065 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/_mantel.py
+-rw-r--r--   0 runner    (1001) docker     (123)     5311 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/_permanova.py
+-rw-r--r--   0 runner    (1001) docker     (123)    12524 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/_permdisp.py
+-rw-r--r--   0 runner    (1001) docker     (123)     5752 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/_utils.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.801828 scikit-bio-0.5.9/skbio/stats/distance/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/__init__.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.801828 scikit-bio-0.5.9/skbio/stats/distance/tests/data/
+-rw-r--r--   0 runner    (1001) docker     (123)     1108 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/bioenv_df_vegan.txt
+-rw-r--r--   0 runner    (1001) docker     (123)    10023 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/bioenv_dm_vegan.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      228 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/bioenv_exp_results_vegan.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      644 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/df.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      696 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/df_extra_column.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      607 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/dm.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      607 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/dm2.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      607 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/dm3.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      607 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/dm4.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      607 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/dm_reordered.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1173 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/exp_results.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     1173 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/exp_results_different_column_order.txt
+-rw-r--r--   0 runner    (1001) docker     (123)       32 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/exp_results_single_column.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     9376 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/mantel_env_dm_vegan.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     9934 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/mantel_veg_dm_vegan.txt
+-rw-r--r--   0 runner    (1001) docker     (123)    20633 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/moving_pictures_dm.tsv
+-rw-r--r--   0 runner    (1001) docker     (123)     3670 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/moving_pictures_mf.tsv
+-rw-r--r--   0 runner    (1001) docker     (123)      378 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/pwmantel_exp_results_all_dms.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      103 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/pwmantel_exp_results_dm_dm2.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      159 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/pwmantel_exp_results_duplicate_dms.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      214 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/pwmantel_exp_results_minimal.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      232 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/pwmantel_exp_results_minimal_with_labels.txt
+-rw-r--r--   0 runner    (1001) docker     (123)       95 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/pwmantel_exp_results_na_p_value.txt
+-rw-r--r--   0 runner    (1001) docker     (123)      215 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/data/pwmantel_exp_results_reordered_distance_matrices.txt
+-rw-r--r--   0 runner    (1001) docker     (123)     4840 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/test_anosim.py
+-rw-r--r--   0 runner    (1001) docker     (123)    45090 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/test_base.py
+-rw-r--r--   0 runner    (1001) docker     (123)     9918 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/test_bioenv.py
+-rw-r--r--   0 runner    (1001) docker     (123)    30434 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/test_mantel.py
+-rw-r--r--   0 runner    (1001) docker     (123)     4860 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/test_permanova.py
+-rw-r--r--   0 runner    (1001) docker     (123)    11549 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/distance/tests/test_permdisp.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.801828 scikit-bio-0.5.9/skbio/stats/evolve/
+-rw-r--r--   0 runner    (1001) docker     (123)      929 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/evolve/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)    11070 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/evolve/_hommola.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.801828 scikit-bio-0.5.9/skbio/stats/evolve/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/evolve/tests/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     7993 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/evolve/tests/test_hommola.py
+-rw-r--r--   0 runner    (1001) docker     (123)    32310 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/gradient.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.805828 scikit-bio-0.5.9/skbio/stats/ordination/
+-rw-r--r--   0 runner    (1001) docker     (123)     4143 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8399 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/_canonical_correspondence_analysis.py
+-rw-r--r--   0 runner    (1001) docker     (123)     7748 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/_correspondence_analysis.py
+-rw-r--r--   0 runner    (1001) docker     (123)  1031727 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/_cutils.c
+-rw-r--r--   0 runner    (1001) docker     (123)     5186 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/_cutils.pyx
+-rw-r--r--   0 runner    (1001) docker     (123)    16917 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/_ordination_results.py
+-rw-r--r--   0 runner    (1001) docker     (123)    17253 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/_principal_coordinate_analysis.py
+-rw-r--r--   0 runner    (1001) docker     (123)     9382 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/_redundancy_analysis.py
+-rw-r--r--   0 runner    (1001) docker     (123)    10184 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/_utils.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.805828 scikit-bio-0.5.9/skbio/stats/ordination/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/__init__.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.809828 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/
+-rw-r--r--   0 runner    (1001) docker     (123)       24 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/L&L_CA_data
+-rw-r--r--   0 runner    (1001) docker     (123)     1149 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_biplot_descriptors
+-rw-r--r--   0 runner    (1001) docker     (123)     1577 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_biplot_projected_descriptors
+-rw-r--r--   0 runner    (1001) docker     (123)     1099 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data
+-rw-r--r--   0 runner    (1001) docker     (123)     2647 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data_12dim
+-rw-r--r--   0 runner    (1001) docker     (123)      900 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data_2
+-rw-r--r--   0 runner    (1001) docker     (123)     1228 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data_3
+-rw-r--r--   0 runner    (1001) docker     (123)      830 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data_3_eigh_ref_3dim
+-rw-r--r--   0 runner    (1001) docker     (123)      821 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data_3_fsvd_ref_3dim
+-rw-r--r--   0 runner    (1001) docker     (123)     1992 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_skbio
+-rw-r--r--   0 runner    (1001) docker     (123)     1000 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_X
+-rw-r--r--   0 runner    (1001) docker     (123)      128 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_Y
+-rw-r--r--   0 runner    (1001) docker     (123)      274 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_biplot_scaling1
+-rw-r--r--   0 runner    (1001) docker     (123)      275 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_biplot_scaling2
+-rw-r--r--   0 runner    (1001) docker     (123)      834 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_sample_constraints_scaling1
+-rw-r--r--   0 runner    (1001) docker     (123)      829 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_sample_constraints_scaling2
+-rw-r--r--   0 runner    (1001) docker     (123)      787 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_site_scaling1_from_vegan
+-rw-r--r--   0 runner    (1001) docker     (123)      742 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_site_scaling2_from_vegan
+-rw-r--r--   0 runner    (1001) docker     (123)      424 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_species_scaling1_from_vegan
+-rw-r--r--   0 runner    (1001) docker     (123)      459 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_species_scaling2_from_vegan
+-rw-r--r--   0 runner    (1001) docker     (123)     1000 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_X
+-rw-r--r--   0 runner    (1001) docker     (123)      196 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_Y
+-rw-r--r--   0 runner    (1001) docker     (123)       86 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_biplot_scaling1
+-rw-r--r--   0 runner    (1001) docker     (123)       86 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_biplot_scaling2
+-rw-r--r--   0 runner    (1001) docker     (123)      864 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_sample_constraints_scaling1
+-rw-r--r--   0 runner    (1001) docker     (123)      867 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_sample_constraints_scaling2
+-rw-r--r--   0 runner    (1001) docker     (123)      984 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_site_scaling1_from_vegan
+-rw-r--r--   0 runner    (1001) docker     (123)      907 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_site_scaling2_from_vegan
+-rw-r--r--   0 runner    (1001) docker     (123)      820 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_species_scaling1_from_vegan
+-rw-r--r--   0 runner    (1001) docker     (123)      881 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_species_scaling2_from_vegan
+-rw-r--r--   0 runner    (1001) docker     (123)     2062 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/exp_PCoAEigenResults_site
+-rw-r--r--   0 runner    (1001) docker     (123)     4900 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/exp_PCoAzeros_site
+-rw-r--r--   0 runner    (1001) docker     (123)     1735 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_biplot_from_vegan.csv
+-rw-r--r--   0 runner    (1001) docker     (123)     2796 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_constraints_from_vegan.csv
+-rw-r--r--   0 runner    (1001) docker     (123)      552 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_eigvals_from_vegan.csv
+-rw-r--r--   0 runner    (1001) docker     (123)      342 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_propexpl_from_vegan.csv
+-rw-r--r--   0 runner    (1001) docker     (123)     2798 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_sites_from_vegan.csv
+-rw-r--r--   0 runner    (1001) docker     (123)     5511 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_species_from_vegan.csv
+-rw-r--r--   0 runner    (1001) docker     (123)     1770 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/varechem.csv
+-rw-r--r--   0 runner    (1001) docker     (123)     4475 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/data/varespec.csv
+-rw-r--r--   0 runner    (1001) docker     (123)     6642 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/test_canonical_correspondence_analysis.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8193 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/test_correspondence_analysis.py
+-rw-r--r--   0 runner    (1001) docker     (123)    12284 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/test_ordination_results.py
+-rw-r--r--   0 runner    (1001) docker     (123)    14004 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/test_principal_coordinate_analysis.py
+-rw-r--r--   0 runner    (1001) docker     (123)    10422 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/test_redundancy_analysis.py
+-rw-r--r--   0 runner    (1001) docker     (123)     4845 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/ordination/tests/test_util.py
+-rw-r--r--   0 runner    (1001) docker     (123)    48004 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/power.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.809828 scikit-bio-0.5.9/skbio/stats/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/__init__.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.813828 scikit-bio-0.5.9/skbio/stats/tests/data/
+-rw-r--r--   0 runner    (1001) docker     (123)      297 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/cr_data_out
+-rw-r--r--   0 runner    (1001) docker     (123)      210 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/cr_data_raw
+-rw-r--r--   0 runner    (1001) docker     (123)       90 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/cr_no_data_out
+-rw-r--r--   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/cr_no_data_raw
+-rw-r--r--   0 runner    (1001) docker     (123)      160 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/gr_w_msg_out
+-rw-r--r--   0 runner    (1001) docker     (123)      126 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/gr_w_msg_raw
+-rw-r--r--   0 runner    (1001) docker     (123)       97 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/gr_wo_msg_out
+-rw-r--r--   0 runner    (1001) docker     (123)       67 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/gr_wo_msg_raw
+-rw-r--r--   0 runner    (1001) docker     (123)      448 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/vr_out
+-rw-r--r--   0 runner    (1001) docker     (123)      271 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/vr_raw
+-rw-r--r--   0 runner    (1001) docker     (123)      517 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/vr_real_out
+-rw-r--r--   0 runner    (1001) docker     (123)      205 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/data/vr_real_raw
+-rw-r--r--   0 runner    (1001) docker     (123)    56315 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/test_composition.py
+-rw-r--r--   0 runner    (1001) docker     (123)    55450 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/test_gradient.py
+-rw-r--r--   0 runner    (1001) docker     (123)     1784 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/test_misc.py
+-rw-r--r--   0 runner    (1001) docker     (123)    21831 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/test_power.py
+-rw-r--r--   0 runner    (1001) docker     (123)     9052 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/stats/tests/test_subsample.py
+-rw-r--r--   0 runner    (1001) docker     (123)      437 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/test.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.813828 scikit-bio-0.5.9/skbio/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tests/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     1553 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tests/test_base.py
+-rw-r--r--   0 runner    (1001) docker     (123)    13167 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tests/test_workflow.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.813828 scikit-bio-0.5.9/skbio/tree/
+-rw-r--r--   0 runner    (1001) docker     (123)     5810 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tree/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)      748 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tree/_exception.py
+-rw-r--r--   0 runner    (1001) docker     (123)    10672 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tree/_majority_rule.py
+-rw-r--r--   0 runner    (1001) docker     (123)    10825 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tree/_nj.py
+-rw-r--r--   0 runner    (1001) docker     (123)   103674 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tree/_tree.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.813828 scikit-bio-0.5.9/skbio/tree/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tree/tests/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)     7851 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tree/tests/test_majority_rule.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8368 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tree/tests/test_nj.py
+-rw-r--r--   0 runner    (1001) docker     (123)    64865 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/tree/tests/test_tree.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.813828 scikit-bio-0.5.9/skbio/util/
+-rw-r--r--   0 runner    (1001) docker     (123)     1869 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/util/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)    12180 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/util/_decorator.py
+-rw-r--r--   0 runner    (1001) docker     (123)      639 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/util/_exception.py
+-rw-r--r--   0 runner    (1001) docker     (123)     5575 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/util/_misc.py
+-rw-r--r--   0 runner    (1001) docker     (123)    13763 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/util/_testing.py
+-rw-r--r--   0 runner    (1001) docker     (123)     1509 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/util/_warning.py
+drwxr-xr-x   0 runner    (1001) docker     (123)        0 2023-08-02 22:34:45.813828 scikit-bio-0.5.9/skbio/util/tests/
+-rw-r--r--   0 runner    (1001) docker     (123)      353 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/util/tests/__init__.py
+-rw-r--r--   0 runner    (1001) docker     (123)    11871 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/util/tests/test_decorator.py
+-rw-r--r--   0 runner    (1001) docker     (123)     8000 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/util/tests/test_misc.py
+-rw-r--r--   0 runner    (1001) docker     (123)    10360 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/util/tests/test_testing.py
+-rw-r--r--   0 runner    (1001) docker     (123)    18819 2023-08-02 22:34:37.000000 scikit-bio-0.5.9/skbio/workflow.py
```

### Comparing `scikit-bio-0.5.8/CHANGELOG.md` & `scikit-bio-0.5.9/CHANGELOG.md`

 * *Files 2% similar despite different names*

```diff
@@ -1,5086 +1,5235 @@
 00000000: 2320 7363 696b 6974 2d62 696f 2063 6861  # scikit-bio cha
 00000010: 6e67 656c 6f67 0a0a 2323 2056 6572 7369  ngelog..## Versi
-00000020: 6f6e 2030 2e35 2e38 0a0a 2323 2320 4665  on 0.5.8..### Fe
-00000030: 6174 7572 6573 0a0a 2a20 4164 6465 6420  atures..* Added 
-00000040: 4e43 4249 2074 6178 6f6e 6f6d 7920 6461  NCBI taxonomy da
-00000050: 7461 6261 7365 2064 756d 7020 666f 726d  tabase dump form
-00000060: 6174 2028 6074 6178 6475 6d70 6029 2028  at (`taxdump`) (
-00000070: 5b23 3138 3130 5d28 6874 7470 733a 2f2f  [#1810](https://
-00000080: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-00000090: 7265 2f73 6369 6b69 742d 6269 6f2f 7075  re/scikit-bio/pu
-000000a0: 6c6c 2f31 3831 3029 292e 0a2a 2041 6464  ll/1810))..* Add
-000000b0: 6564 2060 5472 6565 4e6f 6465 2e66 726f  ed `TreeNode.fro
-000000c0: 6d5f 7461 7864 756d 7060 2066 6f72 2063  m_taxdump` for c
-000000d0: 6f6e 7665 7274 696e 6720 7461 7864 756d  onverting taxdum
-000000e0: 7020 696e 746f 2061 2074 7265 6520 285b  p into a tree ([
-000000f0: 2331 3831 305d 2868 7474 7073 3a2f 2f67  #1810](https://g
-00000100: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
-00000110: 652f 7363 696b 6974 2d62 696f 2f70 756c  e/scikit-bio/pul
-00000120: 6c2f 3138 3130 2929 2e0a 2a20 7363 696b  l/1810))..* scik
-00000130: 6974 2d6c 6561 726e 2068 6173 2062 6565  it-learn has bee
-00000140: 6e20 7265 6d6f 7665 6420 6173 2061 2064  n removed as a d
-00000150: 6570 656e 6465 6e63 792e 2054 6869 7320  ependency. This 
-00000160: 7761 7320 6120 6661 6972 6c79 2068 6561  was a fairly hea
-00000170: 7679 2d77 6569 6768 7420 6465 7065 6e64  vy-weight depend
-00000180: 656e 6379 2074 6861 7420 7761 7320 7072  ency that was pr
-00000190: 6f76 6964 696e 6720 6d69 6e6f 7220 6675  oviding minor fu
-000001a0: 6e63 7469 6f6e 616c 6974 7920 746f 2073  nctionality to s
-000001b0: 6369 6b69 742d 6269 6f2e 2054 6865 2063  cikit-bio. The c
-000001c0: 7269 7469 6361 6c20 636f 6d70 6f6e 656e  ritical componen
-000001d0: 7473 2068 6176 6520 6265 656e 2069 6d70  ts have been imp
-000001e0: 6c65 6d65 6e74 6564 2069 6e20 7363 696b  lemented in scik
-000001f0: 6974 2d62 696f 2064 6972 6563 746c 792c  it-bio directly,
-00000200: 2061 6e64 2074 6865 206e 6f6e 2d63 7269   and the non-cri
-00000210: 7469 6369 616c 2063 6f6d 706f 6e65 6e74  ticial component
-00000220: 7320 6172 6520 6c69 7374 6564 2075 6e64  s are listed und
-00000230: 6572 2022 4261 636b 7761 7264 2d69 6e63  er "Backward-inc
-00000240: 6f6d 7061 7469 626c 6520 6368 616e 6765  ompatible change
-00000250: 7320 5b65 7870 6572 696d 656e 7461 6c5d  s [experimental]
-00000260: 222e 0a2a 2050 7974 686f 6e20 332e 3131  "..* Python 3.11
-00000270: 2069 7320 6e6f 7720 7375 7070 6f72 7465   is now supporte
-00000280: 642e 0a0a 2323 2320 4261 636b 7761 7264  d...### Backward
-00000290: 2d69 6e63 6f6d 7061 7469 626c 6520 6368  -incompatible ch
-000002a0: 616e 6765 7320 5b65 7870 6572 696d 656e  anges [experimen
-000002b0: 7461 6c5d 0a2a 2057 6974 6820 7468 6520  tal].* With the 
-000002c0: 7265 6d6f 7661 6c20 6f66 2074 6865 2073  removal of the s
-000002d0: 6369 6b69 742d 6c65 6172 6e20 6465 7065  cikit-learn depe
-000002e0: 6e64 656e 6379 2c20 7468 7265 6520 6265  ndency, three be
-000002f0: 7461 2064 6976 6572 7369 7479 206d 6574  ta diversity met
-00000300: 7269 6320 6e61 6d65 7320 6361 6e20 6e6f  ric names can no
-00000310: 206c 6f6e 6765 7220 6265 2073 7065 6369   longer be speci
-00000320: 6669 6564 2e20 5468 6573 6520 6172 6520  fied. These are 
-00000330: 6077 6d69 6e6b 6f77 736b 6960 2c20 606e  `wminkowski`, `n
-00000340: 616e 5f65 7563 6c69 6465 616e 602c 2061  an_euclidean`, a
-00000350: 6e64 2060 6861 7665 7273 696e 6560 2e20  nd `haversine`. 
-00000360: 4f6e 2074 6573 7469 6e67 2c20 6077 6d69  On testing, `wmi
-00000370: 6e6b 6f77 736b 6960 2061 6e64 2060 6861  nkowski` and `ha
-00000380: 7665 7273 696e 6560 2064 6964 206e 6f74  versine` did not
-00000390: 2077 6f72 6b20 7468 726f 7567 6820 6073   work through `s
-000003a0: 6b62 696f 2e64 6976 6572 7369 7479 2e62  kbio.diversity.b
-000003b0: 6574 615f 6469 7665 7273 6974 7960 2028  eta_diversity` (
-000003c0: 6f72 2060 736b 6c65 6172 6e2e 6d65 7472  or `sklearn.metr
-000003d0: 6963 732e 7061 6972 7769 7365 5f64 6973  ics.pairwise_dis
-000003e0: 7461 6e63 6573 6029 2e20 5468 6520 666f  tances`). The fo
-000003f0: 726d 6572 2077 6173 2064 6570 7265 6361  rmer was depreca
-00000400: 7465 6420 696e 2066 6176 6f72 206f 6620  ted in favor of 
-00000410: 6361 6c6c 696e 6720 606d 696e 6b6f 7773  calling `minkows
-00000420: 6b69 6020 7769 7468 2061 2076 6563 746f  ki` with a vecto
-00000430: 7220 6f66 2077 6569 6768 7473 2070 726f  r of weights pro
-00000440: 7669 6465 6420 6173 206b 7761 7267 2060  vided as kwarg `
-00000450: 7760 2028 6578 616d 706c 6520 6265 6c6f  w` (example belo
-00000460: 7729 2c20 616e 6420 7468 6520 6c61 7474  w), and the latt
-00000470: 6572 2064 6f65 7320 6e6f 7420 776f 726b  er does not work
-00000480: 2077 6974 6820 6461 7461 206f 6620 7468   with data of th
-00000490: 6973 2073 6861 7065 2e20 606e 616e 5f65  is shape. `nan_e
-000004a0: 7563 6c69 6465 616e 6020 6361 6e20 7374  uclidean` can st
-000004b0: 696c 6c20 6265 2061 6363 6573 7365 6420  ill be accessed 
-000004c0: 6672 6f6e 2073 6369 6b69 742d 6c65 6172  fron scikit-lear
-000004d0: 6e20 6469 7265 6374 6c79 2069 6620 6e65  n directly if ne
-000004e0: 6564 6564 2c20 6966 2061 2075 7365 7220  eded, if a user 
-000004f0: 696e 7374 616c 6c73 2073 6369 6b69 742d  installs scikit-
-00000500: 6c65 6172 6e20 696e 2074 6865 6972 2065  learn in their e
-00000510: 6e76 6972 6f6e 6d65 6e74 2028 6578 616d  nvironment (exam
-00000520: 706c 6520 6265 6c6f 7729 2e0a 0a20 2020  ple below)...   
-00000530: 2060 6060 0a20 2020 2063 6f75 6e74 7320   ```.    counts 
-00000540: 3d20 5b5b 3233 2c20 3634 2c20 3134 2c20  = [[23, 64, 14, 
-00000550: 302c 2030 2c20 332c 2031 5d2c 0a20 2020  0, 0, 3, 1],.   
-00000560: 2020 2020 2020 2020 205b 302c 2033 2c20           [0, 3, 
-00000570: 3335 2c20 3432 2c20 302c 2031 322c 2031  35, 42, 0, 12, 1
-00000580: 5d2c 0a20 2020 2020 2020 2020 2020 205b  ],.            [
-00000590: 302c 2035 2c20 352c 2030 2c20 3430 2c20  0, 5, 5, 0, 40, 
-000005a0: 3430 2c20 305d 2c0a 2020 2020 2020 2020  40, 0],.        
-000005b0: 2020 2020 5b34 342c 2033 352c 2039 2c20      [44, 35, 9, 
-000005c0: 302c 2031 2c20 302c 2030 5d2c 0a20 2020  0, 1, 0, 0],.   
-000005d0: 2020 2020 2020 2020 205b 302c 2032 2c20           [0, 2, 
-000005e0: 382c 2030 2c20 3335 2c20 3435 2c20 315d  8, 0, 35, 45, 1]
-000005f0: 2c0a 2020 2020 2020 2020 2020 2020 5b30  ,.            [0
-00000600: 2c20 302c 2032 352c 2033 352c 2030 2c20  , 0, 25, 35, 0, 
-00000610: 3139 2c20 305d 2c0a 2020 2020 2020 2020  19, 0],.        
-00000620: 2020 2020 5b38 382c 2033 312c 2030 2c20      [88, 31, 0, 
-00000630: 352c 2035 2c20 352c 2035 5d2c 0a20 2020  5, 5, 5, 5],.   
-00000640: 2020 2020 2020 2020 205b 3434 2c20 3339           [44, 39
-00000650: 2c20 302c 2030 2c20 302c 2030 2c20 305d  , 0, 0, 0, 0, 0]
-00000660: 5d0a 0a20 2020 2023 206e 6577 206d 6563  ]..    # new mec
-00000670: 6861 6e69 736d 206f 6620 6163 6365 7373  hanism of access
-00000680: 696e 6720 776d 696e 6b6f 7773 6b69 0a20  ing wminkowski. 
-00000690: 2020 2066 726f 6d20 736b 6269 6f2e 6469     from skbio.di
-000006a0: 7665 7273 6974 7920 696d 706f 7274 2062  versity import b
-000006b0: 6574 615f 6469 7665 7273 6974 790a 2020  eta_diversity.  
-000006c0: 2020 6265 7461 5f64 6976 6572 7369 7479    beta_diversity
-000006d0: 2822 6d69 6e6b 6f77 736b 6922 2c20 636f  ("minkowski", co
-000006e0: 756e 7473 2c20 773d 5b31 2c31 2c31 2c31  unts, w=[1,1,1,1
-000006f0: 2c31 2c31 2c32 5d29 0a0a 2020 2020 2320  ,1,1,2])..    # 
-00000700: 6163 6365 7373 696e 6720 6e61 6e5f 6575  accessing nan_eu
-00000710: 636c 6964 6561 6e20 7468 726f 7567 6820  clidean through 
-00000720: 7363 696b 6974 2d6c 6561 726e 2064 6972  scikit-learn dir
-00000730: 6563 746c 790a 2020 2020 696d 706f 7274  ectly.    import
-00000740: 2073 6b62 696f 0a20 2020 2066 726f 6d20   skbio.    from 
-00000750: 736b 6c65 6172 6e2e 6d65 7472 6963 7320  sklearn.metrics 
-00000760: 696d 706f 7274 2070 6169 7277 6973 655f  import pairwise_
-00000770: 6469 7374 616e 6365 730a 2020 2020 736b  distances.    sk
-00000780: 6c65 6172 6e5f 646d 203d 2070 6169 7277  learn_dm = pairw
-00000790: 6973 655f 6469 7374 616e 6365 7328 636f  ise_distances(co
-000007a0: 756e 7473 2c20 6d65 7472 6963 3d22 6e61  unts, metric="na
-000007b0: 6e5f 6575 636c 6964 6561 6e22 290a 2020  n_euclidean").  
-000007c0: 2020 736b 6269 6f5f 646d 203d 2073 6b62    skbio_dm = skb
-000007d0: 696f 2e44 6973 7461 6e63 654d 6174 7269  io.DistanceMatri
-000007e0: 7828 736b 6c65 6172 6e5f 646d 290a 2020  x(sklearn_dm).  
-000007f0: 2020 6060 600a 0a23 2323 2044 6570 7265    ```..### Depre
-00000800: 6361 7465 6420 6675 6e63 7469 6f6e 616c  cated functional
-00000810: 6974 7920 5b65 7870 6572 696d 656e 7461  ity [experimenta
-00000820: 6c5d 0a2a 2060 736b 6269 6f2e 616c 6967  l].* `skbio.alig
-00000830: 6e6d 656e 742e 6c6f 6361 6c5f 7061 6972  nment.local_pair
-00000840: 7769 7365 5f61 6c69 676e 5f73 7377 6020  wise_align_ssw` 
-00000850: 6861 7320 6265 656e 2064 6570 7265 6361  has been depreca
-00000860: 7465 6420 285b 2331 3831 345d 2868 7474  ted ([#1814](htt
-00000870: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00000880: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-00000890: 696f 2f69 7373 7565 732f 3138 3134 2929  io/issues/1814))
-000008a0: 2061 6e64 2077 696c 6c20 6265 2072 656d   and will be rem
-000008b0: 6f76 6564 206f 7220 7265 706c 6163 6564  oved or replaced
-000008c0: 2069 6e20 7363 696b 6974 2d62 696f 2030   in scikit-bio 0
-000008d0: 2e36 2e30 2e0a 0a23 2320 5665 7273 696f  .6.0...## Versio
-000008e0: 6e20 302e 352e 370a 0a23 2323 2046 6561  n 0.5.7..### Fea
-000008f0: 7475 7265 730a 0a2a 2049 6e74 726f 6475  tures..* Introdu
-00000900: 6365 2073 7570 706f 7274 2066 6f72 2050  ce support for P
-00000910: 7974 686f 6e20 332e 3130 2028 5b23 3138  ython 3.10 ([#18
-00000920: 3031 5d28 6874 7470 733a 2f2f 6769 7468  01](https://gith
-00000930: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
-00000940: 6369 6b69 742d 6269 6f2f 7075 6c6c 2f31  cikit-bio/pull/1
-00000950: 3830 3129 292e 0a2a 2054 656e 7461 7469  801))..* Tentati
-00000960: 7665 2073 7570 706f 7274 2066 6f72 2041  ve support for A
-00000970: 7070 6c65 204d 3120 285b 2331 3730 395d  pple M1 ([#1709]
-00000980: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-00000990: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
-000009a0: 6974 2d62 696f 2f70 756c 6c2f 3137 3039  it-bio/pull/1709
-000009b0: 2929 2e0a 2a20 4164 6465 6420 7375 7070  ))..* Added supp
-000009c0: 6f72 7420 666f 7220 7265 6164 696e 6720  ort for reading 
-000009d0: 616e 6420 7772 6974 696e 6720 6120 6269  and writing a bi
-000009e0: 6e61 7279 2064 6973 7461 6e63 6520 6d61  nary distance ma
-000009f0: 7472 6978 206f 626a 6563 7420 666f 726d  trix object form
-00000a00: 6174 2e20 285b 2331 3731 365d 2868 7474  at. ([#1716](htt
-00000a10: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00000a20: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-00000a30: 696f 2f70 756c 6c2f 3137 3136 2929 0a2a  io/pull/1716)).*
-00000a40: 2041 6464 6564 2073 7570 706f 7274 2066   Added support f
-00000a50: 6f72 2060 6e70 2e66 6c6f 6174 3332 6020  or `np.float32` 
-00000a60: 7769 7468 2060 4469 7373 696d 696c 6172  with `Dissimilar
-00000a70: 6974 794d 6174 7269 7860 206f 626a 6563  ityMatrix` objec
-00000a80: 7473 2e0a 2a20 4164 6465 6420 7375 7070  ts..* Added supp
-00000a90: 6f72 7420 666f 7220 6d65 7468 6f64 2061  ort for method a
-00000aa0: 6e64 206e 756d 6265 725f 6f66 5f64 696d  nd number_of_dim
-00000ab0: 656e 7369 6f6e 7320 746f 2070 6572 6d64  ensions to permd
-00000ac0: 6973 7020 7265 6475 6369 6e67 2074 6865  isp reducing the
-00000ad0: 2072 756e 7469 6d65 2062 7920 3130 3078   runtime by 100x
-00000ae0: 2061 7420 3430 3030 2073 616d 706c 6573   at 4000 samples
-00000af0: 2c20 5b69 7373 7565 2023 3137 3639 5d28  , [issue #1769](
-00000b00: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-00000b10: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
-00000b20: 742d 6269 6f2f 7075 6c6c 2f31 3736 3929  t-bio/pull/1769)
-00000b30: 2e0a 2a20 4f72 6469 6e61 7469 6f6e 5265  ..* OrdinationRe
-00000b40: 7375 6c74 7320 6f62 6a65 6374 2069 7320  sults object is 
-00000b50: 6e6f 7720 6163 6365 7074 6564 2061 7320  now accepted as 
-00000b60: 696e 7075 7420 666f 7220 7065 726d 6469  input for permdi
-00000b70: 7370 2e0a 0a23 2323 2050 6572 666f 726d  sp...### Perform
-00000b80: 616e 6365 2065 6e68 616e 6365 6d65 6e74  ance enhancement
-00000b90: 730a 0a2a 2041 766f 6964 2061 6e20 696d  s..* Avoid an im
-00000ba0: 706c 6963 6974 2064 6174 6120 636f 7079  plicit data copy
-00000bb0: 206f 6e20 636f 6e73 7472 7563 7469 6f6e   on construction
-00000bc0: 206f 6620 6044 6973 7369 6d69 6c61 7269   of `Dissimilari
-00000bd0: 7479 4d61 7472 6978 6020 6f62 6a65 6374  tyMatrix` object
-00000be0: 732e 0a2a 2041 766f 6964 2076 616c 6964  s..* Avoid valid
-00000bf0: 6174 696f 6e20 6f6e 2063 6f70 7920 6f66  ation on copy of
-00000c00: 2060 4469 7373 696d 696c 6172 6974 794d   `DissimilarityM
-00000c10: 6174 7269 7860 2061 6e64 2060 4469 7374  atrix` and `Dist
-00000c20: 616e 6365 4d61 7472 6978 6020 6f62 6a65  anceMatrix` obje
-00000c30: 6374 732c 2073 6565 205b 5052 2023 3137  cts, see [PR #17
-00000c40: 3437 5d28 6874 7470 733a 2f2f 6769 7468  47](https://gith
-00000c50: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
-00000c60: 6369 6b69 742d 6269 6f2f 7075 6c6c 2f31  cikit-bio/pull/1
-00000c70: 3734 3729 0a2a 2055 7365 2061 6e20 6f70  747).* Use an op
-00000c80: 7469 6d69 7a65 6420 7665 7273 696f 6e20  timized version 
-00000c90: 6f66 2073 796d 6d65 7472 7920 6368 6563  of symmetry chec
-00000ca0: 6b20 696e 2044 6973 7461 6e63 654d 6174  k in DistanceMat
-00000cb0: 7269 782c 2073 6565 205b 5052 2023 3137  rix, see [PR #17
-00000cc0: 3437 5d28 6874 7470 733a 2f2f 6769 7468  47](https://gith
-00000cd0: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
-00000ce0: 6369 6b69 742d 6269 6f2f 7075 6c6c 2f31  cikit-bio/pull/1
-00000cf0: 3734 3729 0a2a 2041 766f 6964 2070 6572  747).* Avoid per
-00000d00: 666f 726d 696e 6720 6669 6c74 6572 696e  forming filterin
-00000d10: 6720 7768 656e 2069 6473 2061 7265 2069  g when ids are i
-00000d20: 6465 6e74 6963 616c 2c20 7365 6520 5b50  dentical, see [P
-00000d30: 5220 2331 3735 325d 2868 7474 7073 3a2f  R #1752](https:/
-00000d40: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
-00000d50: 6f72 652f 7363 696b 6974 2d62 696f 2f70  ore/scikit-bio/p
-00000d60: 756c 6c2f 3137 3532 290a 2a20 6365 6e74  ull/1752).* cent
-00000d70: 6572 5f64 6973 7461 6e63 655f 6d61 7472  er_distance_matr
-00000d80: 6978 2068 6173 2062 6565 6e20 7265 2d69  ix has been re-i
-00000d90: 6d70 6c65 6d65 6e74 6564 2069 6e20 6379  mplemented in cy
-00000da0: 7468 6f6e 2066 6f72 2062 6f74 6820 7370  thon for both sp
-00000db0: 6565 6420 616e 6420 6d65 6d6f 7279 2075  eed and memory u
-00000dc0: 7365 2e20 496e 6469 7265 6374 6c79 2073  se. Indirectly s
-00000dd0: 7065 6564 7320 7570 2070 636f 6120 5b50  peeds up pcoa [P
-00000de0: 5220 2331 3734 395d 2868 7474 7073 3a2f  R #1749](https:/
-00000df0: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
-00000e00: 6f72 652f 7363 696b 6974 2d62 696f 2f70  ore/scikit-bio/p
-00000e10: 756c 6c2f 3137 3439 290a 2a20 5573 6520  ull/1749).* Use 
-00000e20: 6120 6d65 6d6f 7279 2d6f 7074 696d 697a  a memory-optimiz
-00000e30: 6564 2076 6572 7369 6f6e 206f 6620 7065  ed version of pe
-00000e40: 726d 7574 6520 696e 2044 6973 7461 6e63  rmute in Distanc
-00000e50: 654d 6174 7269 782c 2073 6565 205b 5052  eMatrix, see [PR
-00000e60: 2023 3137 3536 5d28 6874 7470 733a 2f2f   #1756](https://
-00000e70: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-00000e80: 7265 2f73 6369 6b69 742d 6269 6f2f 7075  re/scikit-bio/pu
-00000e90: 6c6c 2f31 3735 3629 2e0a 2a20 5265 6661  ll/1756)..* Refa
-00000ea0: 6374 6f72 2070 6561 7273 6f6e 2061 6e64  ctor pearson and
-00000eb0: 2073 7065 6172 6d61 6e20 736b 6269 6f2e   spearman skbio.
-00000ec0: 7374 6174 732e 6469 7374 616e 6365 2e6d  stats.distance.m
-00000ed0: 616e 7465 6c20 696d 706c 656d 656e 7461  antel implementa
-00000ee0: 7469 6f6e 7320 746f 2064 7261 7374 6963  tions to drastic
-00000ef0: 616c 6c79 2069 6d70 726f 7665 206d 656d  ally improve mem
-00000f00: 6f72 7920 6c6f 6361 6c69 7479 2e20 416c  ory locality. Al
-00000f10: 736f 2063 6163 6865 2069 6e74 6572 6d65  so cache interme
-00000f20: 6469 6174 6520 7265 7375 6c74 7320 7468  diate results th
-00000f30: 6174 2061 7265 2069 6e76 6172 6961 6e74  at are invariant
-00000f40: 2061 6372 6f73 7320 7065 726d 7574 6174   across permutat
-00000f50: 696f 6e73 2c20 7365 6520 5b50 5220 2331  ions, see [PR #1
-00000f60: 3735 365d 2868 7474 7073 3a2f 2f67 6974  756](https://git
-00000f70: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
-00000f80: 7363 696b 6974 2d62 696f 2f70 756c 6c2f  scikit-bio/pull/
-00000f90: 3137 3536 292e 0a2a 2052 6566 6163 746f  1756)..* Refacto
-00000fa0: 7220 7065 726d 616e 6f76 6120 746f 2072  r permanova to r
-00000fb0: 656d 6f76 6520 696e 7465 726d 6564 6961  emove intermedia
-00000fc0: 7465 2062 7566 6665 7273 2061 6e64 2063  te buffers and c
-00000fd0: 7974 686f 6e69 7a65 2074 6865 2069 6e74  ythonize the int
-00000fe0: 6572 6e61 6c73 2c20 7365 6520 5b50 5220  ernals, see [PR 
-00000ff0: 2331 3736 385d 2868 7474 7073 3a2f 2f67  #1768](https://g
-00001000: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
-00001010: 652f 7363 696b 6974 2d62 696f 2f70 756c  e/scikit-bio/pul
-00001020: 6c2f 3137 3638 292e 0a0a 2323 2320 4275  l/1768)...### Bu
-00001030: 6720 6669 7865 730a 0a2a 2046 6978 2077  g fixes..* Fix w
-00001040: 696e 646f 7773 2061 6e64 2033 3262 6974  indows and 32bit
-00001050: 2069 6e63 6f6d 7061 7469 6269 6c69 7479   incompatibility
-00001060: 2069 6e20 6075 6e77 6569 6768 7465 645f   in `unweighted_
-00001070: 756e 6966 7261 6360 2e0a 0a23 2323 204d  unifrac`...### M
-00001080: 6973 6365 6c6c 616e 656f 7573 0a0a 2a20  iscellaneous..* 
-00001090: 5079 7468 6f6e 2033 2e36 2068 6173 2062  Python 3.6 has b
-000010a0: 6565 6e20 7265 6d6f 7665 6420 6672 6f6d  een removed from
-000010b0: 206f 7572 2074 6573 7469 6e67 206d 6174   our testing mat
-000010c0: 7269 782e 0a2a 2053 7065 6369 6679 2062  rix..* Specify b
-000010d0: 7569 6c64 2064 6570 656e 6465 6e63 6965  uild dependencie
-000010e0: 7320 696e 2070 7970 726f 6a65 6374 2e74  s in pyproject.t
-000010f0: 6f6d 6c2e 2054 6869 7320 616c 6c6f 7773  oml. This allows
-00001100: 2074 6865 2070 6163 6b61 6765 2074 6f20   the package to 
-00001110: 6265 2069 6e73 7461 6c6c 6564 2077 6974  be installed wit
-00001120: 686f 7574 2068 6176 696e 6720 746f 2066  hout having to f
-00001130: 6972 7374 206d 616e 7561 6c6c 7920 696e  irst manually in
-00001140: 7374 616c 6c20 6e75 6d70 792e 0a2a 2055  stall numpy..* U
-00001150: 7064 6174 6520 6864 6d65 6469 616e 7320  pdate hdmedians 
-00001160: 7061 636b 6167 6520 746f 2061 2076 6572  package to a ver
-00001170: 7369 6f6e 2077 6869 6368 2064 6f65 736e  sion which doesn
-00001180: 2774 2072 6571 7569 7265 2061 6e20 696e  't require an in
-00001190: 6974 6961 6c20 6d61 6e75 616c 206e 756d  itial manual num
-000011a0: 7079 2069 6e73 7461 6c6c 2e0a 2a20 4e6f  py install..* No
-000011b0: 7720 6275 696c 6461 626c 6520 6f6e 206e  w buildable on n
-000011c0: 6f6e 2d78 3836 2070 6c61 7466 6f72 6d73  on-x86 platforms
-000011d0: 2064 7565 2074 6f20 7573 6520 6f66 2074   due to use of t
-000011e0: 6865 205b 5349 4d44 2045 7665 7279 7768  he [SIMD Everywh
-000011f0: 6572 655d 2868 7474 7073 3a2f 2f67 6974  ere](https://git
-00001200: 6875 622e 636f 6d2f 7369 6d64 2d65 7665  hub.com/simd-eve
-00001210: 7279 7768 6572 652f 7369 6d64 6529 206c  rywhere/simde) l
-00001220: 6962 7261 7279 2e0a 2a20 5265 6765 6e65  ibrary..* Regene
-00001230: 7261 7465 2043 7974 686f 6e20 7772 6170  rate Cython wrap
-00001240: 7065 7220 6279 2064 6566 6175 6c74 2074  per by default t
-00001250: 6f20 6176 6f69 6420 696e 636f 6d70 6174  o avoid incompat
-00001260: 6962 696c 6974 6965 7320 7769 7468 2069  ibilities with i
-00001270: 6e73 7461 6c6c 6564 2043 5079 7468 6f6e  nstalled CPython
-00001280: 2e0a 2a20 5570 6461 7465 2064 6f63 756d  ..* Update docum
-00001290: 656e 7461 7469 6f6e 2066 6f72 2074 6865  entation for the
-000012a0: 2060 736b 6269 6f2e 7374 6174 732e 636f   `skbio.stats.co
-000012b0: 6d70 6f73 6974 696f 6e2e 616e 636f 6d60  mposition.ancom`
-000012c0: 2066 756e 6374 696f 6e2e 2028 5b23 3137   function. ([#17
-000012d0: 3431 5d28 6874 7470 733a 2f2f 6769 7468  41](https://gith
-000012e0: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
-000012f0: 6369 6b69 742d 6269 6f2f 7075 6c6c 2f31  cikit-bio/pull/1
-00001300: 3734 3129 290a 0a23 2320 5665 7273 696f  741))..## Versio
-00001310: 6e20 302e 352e 360a 0a23 2323 2046 6561  n 0.5.6..### Fea
-00001320: 7475 7265 730a 0a2a 2041 6464 6564 206f  tures..* Added o
-00001330: 7074 696f 6e20 746f 2072 6574 7572 6e20  ption to return 
-00001340: 6120 6361 7074 7572 6520 6772 6f75 7020  a capture group 
-00001350: 636f 6d70 696c 6564 2072 6567 6578 2070  compiled regex p
-00001360: 6174 7465 726e 2074 6f20 616e 7920 636c  attern to any cl
-00001370: 6173 7320 696e 6865 7269 7469 6e67 2060  ass inheriting `
-00001380: 6047 7261 6d6d 6172 6564 5365 7175 656e  `GrammaredSequen
-00001390: 6365 6060 2074 6872 6f75 6768 2074 6865  ce`` through the
-000013a0: 2060 6074 6f5f 7265 6765 7860 6020 6d65   ``to_regex`` me
-000013b0: 7468 6f64 2e20 285b 2331 3433 315d 2868  thod. ([#1431](h
-000013c0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-000013d0: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
-000013e0: 2d62 696f 2f69 7373 7565 732f 3134 3331  -bio/issues/1431
-000013f0: 2929 0a0a 2a20 4164 6465 6420 6044 6973  ))..* Added `Dis
-00001400: 7369 6d69 6c61 7269 7479 2e77 6974 6869  similarity.withi
-00001410: 6e60 2061 6e64 2060 2e62 6574 7765 656e  n` and `.between
-00001420: 6020 746f 206f 6274 6169 6e20 7468 6520  ` to obtain the 
-00001430: 7265 7370 6563 7469 7665 2064 6973 7461  respective dista
-00001440: 6e63 6573 2061 6e64 2065 7870 7265 7373  nces and express
-00001450: 2074 6865 6d20 6173 2061 2060 4461 7461   them as a `Data
-00001460: 4672 616d 6560 2e20 285b 2331 3636 325d  Frame`. ([#1662]
-00001470: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-00001480: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
-00001490: 6974 2d62 696f 2f70 756c 6c2f 3136 3632  it-bio/pull/1662
-000014a0: 2929 0a0a 2a20 4164 6465 6420 4b65 6e64  ))..* Added Kend
-000014b0: 616c 6c20 5461 7520 6173 2070 6f73 7369  all Tau as possi
-000014c0: 626c 6520 636f 7272 656c 6174 696f 6e20  ble correlation 
-000014d0: 6d65 7468 6f64 2069 6e20 7468 6520 6073  method in the `s
-000014e0: 6b62 696f 2e73 7461 7473 2e64 6973 7461  kbio.stats.dista
-000014f0: 6e63 652e 6d61 6e74 656c 6020 6675 6e63  nce.mantel` func
-00001500: 7469 6f6e 2028 5b23 3136 3735 5d28 6874  tion ([#1675](ht
-00001510: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-00001520: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
-00001530: 6269 6f2f 6973 7375 6573 2f31 3637 3529  bio/issues/1675)
-00001540: 292e 0a0a 2a20 4164 6465 6420 7375 7070  )...* Added supp
-00001550: 6f72 7420 666f 7220 4955 5041 4320 616d  ort for IUPAC am
-00001560: 696e 6f20 6163 6964 2063 6f64 6573 2055  ino acid codes U
-00001570: 2028 7365 6c65 6e6f 6379 7374 6569 6e65   (selenocysteine
-00001580: 292c 204f 2028 7079 7272 6f6c 7973 696e  ), O (pyrrolysin
-00001590: 6529 2c20 616e 6420 4a20 286c 6575 6369  e), and J (leuci
-000015a0: 6e65 206f 7220 6973 6f6c 6575 6369 6e65  ne or isoleucine
-000015b0: 292e 2028 5b23 3135 3736 5d28 6874 7470  ). ([#1576](http
-000015c0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-000015d0: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-000015e0: 6f2f 6973 7375 6573 2f31 3537 3629 0a0a  o/issues/1576)..
-000015f0: 2323 2320 4261 636b 7761 7264 2d69 6e63  ### Backward-inc
-00001600: 6f6d 7061 7469 626c 6520 6368 616e 6765  ompatible change
-00001610: 7320 5b73 7461 626c 655d 0a0a 2323 2320  s [stable]..### 
-00001620: 4261 636b 7761 7264 2d69 6e63 6f6d 7061  Backward-incompa
-00001630: 7469 626c 6520 6368 616e 6765 7320 5b65  tible changes [e
-00001640: 7870 6572 696d 656e 7461 6c5d 0a0a 2a20  xperimental]..* 
-00001650: 4368 616e 6765 6420 6073 6b62 696f 2e74  Changed `skbio.t
-00001660: 7265 652e 5472 6565 4e6f 6465 2e73 7570  ree.TreeNode.sup
-00001670: 706f 7274 6020 6672 6f6d 2061 206d 6574  port` from a met
-00001680: 686f 6420 746f 2061 2070 726f 7065 7274  hod to a propert
-00001690: 792e 0a2a 2041 6464 6564 2060 6173 7369  y..* Added `assi
-000016a0: 676e 5f73 7570 706f 7274 7360 206d 6574  gn_supports` met
-000016b0: 686f 6420 746f 2060 736b 6269 6f2e 7472  hod to `skbio.tr
-000016c0: 6565 2e54 7265 654e 6f64 6560 2074 6f20  ee.TreeNode` to 
-000016d0: 6578 7472 6163 7420 6272 616e 6368 2073  extract branch s
-000016e0: 7570 706f 7274 2076 616c 7565 7320 6672  upport values fr
-000016f0: 6f6d 206e 6f64 6520 6c61 6265 6c73 2e0a  om node labels..
-00001700: 2a20 4d6f 6469 6669 6564 2074 6865 2077  * Modified the w
-00001710: 6179 2061 206e 6f64 6527 7320 6c61 6265  ay a node's labe
-00001720: 6c20 6973 2070 7269 6e74 6564 3a20 6073  l is printed: `s
-00001730: 7570 706f 7274 3a6e 616d 6560 2069 6620  upport:name` if 
-00001740: 626f 7468 2065 7869 7374 2c20 6f72 2060  both exist, or `
-00001750: 7375 7070 6f72 7460 206f 7220 606e 616d  support` or `nam
-00001760: 6560 2069 6620 6569 7468 6572 2065 7869  e` if either exi
-00001770: 7374 732e 0a0a 2323 2320 5065 7266 6f72  sts...### Perfor
-00001780: 6d61 6e63 6520 656e 6861 6e63 656d 656e  mance enhancemen
-00001790: 7473 0a0a 2323 2320 4275 6720 6669 7865  ts..### Bug fixe
-000017a0: 730a 2a20 5265 7175 6972 6520 6053 7068  s.* Require `Sph
-000017b0: 696e 7820 3c3d 2033 2e30 602e 204e 6577  inx <= 3.0`. New
-000017c0: 6572 2053 7068 696e 7820 7665 7273 696f  er Sphinx versio
-000017d0: 6e73 2063 6175 7365 6420 6275 696c 6420  ns caused build 
-000017e0: 6572 726f 7273 2e20 5b23 3137 3139 5d28  errors. [#1719](
-000017f0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-00001800: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
-00001810: 742d 6269 6f2f 7075 6c6c 2f31 3731 3929  t-bio/pull/1719)
-00001820: 0a0a 2a20 2a20 6073 6b62 696f 2e73 7461  ..* * `skbio.sta
-00001830: 7473 2e6f 7264 696e 6174 696f 6e60 2074  ts.ordination` t
-00001840: 6573 7473 2068 6176 6520 6265 656e 2072  ests have been r
-00001850: 656c 6178 6564 2e20 285b 2331 3731 335d  elaxed. ([#1713]
-00001860: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-00001870: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
-00001880: 6974 2d62 696f 2f69 7373 7565 732f 3137  it-bio/issues/17
-00001890: 3133 2929 0a0a 2a20 4669 7865 7320 6275  13))..* Fixes bu
-000018a0: 696c 6420 6572 726f 7273 2066 6f72 206e  ild errors for n
-000018b0: 6577 6572 2076 6572 7369 6f6e 7320 6f66  ewer versions of
-000018c0: 204e 756d 5079 2c20 5061 6e64 6173 2c20   NumPy, Pandas, 
-000018d0: 616e 6420 5363 6950 792e 0a0a 2a20 436f  and SciPy...* Co
-000018e0: 7272 6563 7465 6420 6120 6372 6974 6963  rrected a critic
-000018f0: 6961 6c20 6275 6720 696e 2060 736b 6269  ial bug in `skbi
-00001900: 6f2e 616c 6967 6e6d 656e 742e 5374 7269  o.alignment.Stri
-00001910: 7065 6453 6d69 7468 5761 7465 726d 616e  pedSmithWaterman
-00001920: 602f 6073 6b62 696f 2e61 6c69 676e 6d65  `/`skbio.alignme
-00001930: 6e74 2e6c 6f63 616c 5f70 6169 7277 6973  nt.local_pairwis
-00001940: 655f 616c 6967 6e5f 7373 7760 2077 6869  e_align_ssw` whi
-00001950: 6368 2077 6f75 6c64 2063 6175 7365 2074  ch would cause t
-00001960: 6865 2066 6f72 6d61 7474 696e 6720 6f66  he formatting of
-00001970: 2074 6865 2061 6c69 676e 6564 2073 6571   the aligned seq
-00001980: 7565 6e63 6573 2074 6f20 6d69 7370 6c61  uences to mispla
-00001990: 6365 2067 6170 2063 6861 7261 6374 6572  ce gap character
-000019a0: 7320 6279 2074 6865 206e 756d 6265 7220  s by the number 
-000019b0: 6f66 2067 6170 2063 6861 7261 6374 6572  of gap character
-000019c0: 7320 7072 6573 656e 7420 696e 2074 6865  s present in the
-000019d0: 206f 7070 6f73 696e 6720 616c 6967 6e65   opposing aligne
-000019e0: 6420 7365 7175 656e 6365 2075 7020 746f  d sequence up to
-000019f0: 2074 6861 7420 706f 696e 742e 2054 6869   that point. Thi
-00001a00: 7320 7761 7320 6361 7573 6564 2062 7920  s was caused by 
-00001a10: 6120 6661 756c 7479 2069 6d70 6c65 6d65  a faulty impleme
-00001a20: 6e74 6174 696f 6e20 6f66 2043 4947 4152  ntation of CIGAR
-00001a30: 2073 7472 696e 6720 7061 7273 696e 672c   string parsing,
-00001a40: 2073 6565 205b 2331 3637 395d 2868 7474   see [#1679](htt
-00001a50: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00001a60: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-00001a70: 696f 2f70 756c 6c2f 3136 3739 2920 666f  io/pull/1679) fo
-00001a80: 7220 6675 6c6c 2064 6574 6169 6c73 2e0a  r full details..
-00001a90: 0a2a 2046 6978 6573 2062 7569 6c64 2065  .* Fixes build e
-00001aa0: 7272 6f72 7320 666f 7220 6e65 7765 7220  rrors for newer 
-00001ab0: 7665 7273 696f 6e73 206f 6620 4e75 6d50  versions of NumP
-00001ac0: 792c 2050 616e 6461 732c 2061 6e64 2053  y, Pandas, and S
-00001ad0: 6369 5079 2e0a 0a2a 2043 6f72 7265 6374  ciPy...* Correct
-00001ae0: 6564 2061 2063 7269 7469 6369 616c 2062  ed a criticial b
-00001af0: 7567 2069 6e20 6073 6b62 696f 2e61 6c69  ug in `skbio.ali
-00001b00: 676e 6d65 6e74 2e53 7472 6970 6564 536d  gnment.StripedSm
-00001b10: 6974 6857 6174 6572 6d61 6e60 2f60 736b  ithWaterman`/`sk
-00001b20: 6269 6f2e 616c 6967 6e6d 656e 742e 6c6f  bio.alignment.lo
-00001b30: 6361 6c5f 7061 6972 7769 7365 5f61 6c69  cal_pairwise_ali
-00001b40: 676e 5f73 7377 6020 7768 6963 6820 776f  gn_ssw` which wo
-00001b50: 756c 6420 6361 7573 6520 7468 6520 666f  uld cause the fo
-00001b60: 726d 6174 7469 6e67 206f 6620 7468 6520  rmatting of the 
-00001b70: 616c 6967 6e65 6420 7365 7175 656e 6365  aligned sequence
-00001b80: 7320 746f 206d 6973 706c 6163 6520 6761  s to misplace ga
-00001b90: 7020 6368 6172 6163 7465 7273 2062 7920  p characters by 
-00001ba0: 7468 6520 6e75 6d62 6572 206f 6620 6761  the number of ga
-00001bb0: 7020 6368 6172 6163 7465 7273 2070 7265  p characters pre
-00001bc0: 7365 6e74 2069 6e20 7468 6520 6f70 706f  sent in the oppo
-00001bd0: 7369 6e67 2061 6c69 676e 6564 2073 6571  sing aligned seq
-00001be0: 7565 6e63 6520 7570 2074 6f20 7468 6174  uence up to that
-00001bf0: 2070 6f69 6e74 2e20 5468 6973 2077 6173   point. This was
-00001c00: 2063 6175 7365 6420 6279 2061 2066 6175   caused by a fau
-00001c10: 6c74 7920 696d 706c 656d 656e 7461 7469  lty implementati
-00001c20: 6f6e 206f 6620 4349 4741 5220 7374 7269  on of CIGAR stri
-00001c30: 6e67 2070 6172 7369 6e67 2c20 7365 6520  ng parsing, see 
-00001c40: 5b23 3136 3739 5d28 6874 7470 733a 2f2f  [#1679](https://
-00001c50: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-00001c60: 7265 2f73 6369 6b69 742d 6269 6f2f 7075  re/scikit-bio/pu
-00001c70: 6c6c 2f31 3637 3929 2066 6f72 2066 756c  ll/1679) for ful
-00001c80: 6c20 6465 7461 696c 732e 0a0a 2323 2320  l details...### 
-00001c90: 4465 7072 6563 6174 6564 2066 756e 6374  Deprecated funct
-00001ca0: 696f 6e61 6c69 7479 205b 7374 6162 6c65  ionality [stable
-00001cb0: 5d0a 0a23 2323 2044 6570 7265 6361 7465  ]..### Deprecate
-00001cc0: 6420 6675 6e63 7469 6f6e 616c 6974 7920  d functionality 
-00001cd0: 5b65 7870 6572 696d 656e 7461 6c5d 0a0a  [experimental]..
-00001ce0: 2323 2320 4d69 7363 656c 6c61 6e65 6f75  ### Miscellaneou
-00001cf0: 730a 0a2a 2060 736b 6269 6f2e 6469 7665  s..* `skbio.dive
-00001d00: 7273 6974 792e 6265 7461 5f64 6976 6572  rsity.beta_diver
-00001d10: 7369 7479 6020 6e6f 7720 6163 6365 7074  sity` now accept
-00001d20: 7320 6120 7061 6e64 6173 2044 6174 6146  s a pandas DataF
-00001d30: 7261 6d65 2061 7320 696e 7075 742e 0a0a  rame as input...
-00001d40: 2a20 4176 6f69 6420 7061 6e64 6173 2031  * Avoid pandas 1
-00001d50: 2e30 2e30 2069 6d70 6f72 7420 7761 726e  .0.0 import warn
-00001d60: 696e 6720 285b 2331 3638 385d 2868 7474  ing ([#1688](htt
-00001d70: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00001d80: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-00001d90: 696f 2f69 7373 7565 732f 3136 3838 2929  io/issues/1688))
-00001da0: 0a0a 2a20 4164 6465 6420 7375 7070 6f72  ..* Added suppor
-00001db0: 7420 666f 7220 5079 7468 6f6e 2033 2e38  t for Python 3.8
-00001dc0: 2061 6e64 2064 726f 7070 6564 2073 7570   and dropped sup
-00001dd0: 706f 7274 2066 6f72 2050 7974 686f 6e20  port for Python 
-00001de0: 332e 352e 0a0a 2a20 5468 6973 2076 6572  3.5...* This ver
-00001df0: 7369 6f6e 206e 6f77 2064 6570 656e 6473  sion now depends
-00001e00: 206f 6e20 6073 6369 7079 203e 3d20 312e   on `scipy >= 1.
-00001e10: 3360 2061 6e64 2060 7061 6e64 6173 203e  3` and `pandas >
-00001e20: 3d20 312e 3060 2e0a 0a23 2320 5665 7273  = 1.0`...## Vers
-00001e30: 696f 6e20 302e 352e 3520 2832 3031 382d  ion 0.5.5 (2018-
-00001e40: 3132 2d31 3029 0a0a 2323 2320 4665 6174  12-10)..### Feat
-00001e50: 7572 6573 0a0a 2a20 6073 6b62 696f 2e73  ures..* `skbio.s
-00001e60: 7461 7473 2e63 6f6d 706f 7369 7469 6f6e  tats.composition
-00001e70: 6020 6e6f 7720 6861 7320 6d65 7468 6f64  ` now has method
-00001e80: 7320 746f 2063 6f6d 7075 7465 2061 6464  s to compute add
-00001e90: 6974 6976 6520 6c6f 672d 7261 7469 6f20  itive log-ratio 
-00001ea0: 7472 616e 7366 6f72 6d61 7469 6f6e 2061  transformation a
-00001eb0: 6e64 2069 6e76 6572 7365 2061 6464 6974  nd inverse addit
-00001ec0: 6976 6520 6c6f 672d 7261 7469 6f20 7472  ive log-ratio tr
-00001ed0: 616e 7366 6f72 6d61 7469 6f6e 2028 6061  ansformation (`a
-00001ee0: 6c72 602c 2060 616c 725f 696e 7660 2920  lr`, `alr_inv`) 
-00001ef0: 6173 2077 656c 6c20 6173 2061 206d 6574  as well as a met
-00001f00: 686f 6420 746f 2062 7569 6c64 2061 2062  hod to build a b
-00001f10: 6173 6973 2066 726f 6d20 6120 7365 7175  asis from a sequ
-00001f20: 656e 7469 616c 2062 696e 6172 7920 7061  ential binary pa
-00001f30: 7274 6974 696f 6e20 2860 7362 705f 6261  rtition (`sbp_ba
-00001f40: 7369 7360 292e 0a0a 2323 2320 4261 636b  sis`)...### Back
+00000020: 6f6e 2030 2e35 2e39 0a0a 2323 2320 4665  on 0.5.9..### Fe
+00000030: 6174 7572 6573 0a0a 2a20 4164 6469 6e67  atures..* Adding
+00000040: 2056 6172 6961 6e63 6520 6c6f 6720 7261   Variance log ra
+00000050: 7469 6f20 6573 7469 6d61 746f 7273 2069  tio estimators i
+00000060: 6e20 6073 6b62 696f 2e73 7461 7473 2e63  n `skbio.stats.c
+00000070: 6f6d 706f 7369 7469 6f6e 2e76 6c72 6020  omposition.vlr` 
+00000080: 616e 6420 6073 6b62 696f 2e73 7461 7473  and `skbio.stats
+00000090: 2e63 6f6d 706f 7369 7469 6f6e 2e70 6169  .composition.pai
+000000a0: 7277 6973 655f 766c 7260 2028 5b23 3138  rwise_vlr` ([#18
+000000b0: 3033 5d28 6874 7470 733a 2f2f 6769 7468  03](https://gith
+000000c0: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+000000d0: 6369 6b69 742d 6269 6f2f 7075 6c6c 2f31  cikit-bio/pull/1
+000000e0: 3830 3329 290a 2a20 4164 6465 6420 6073  803)).* Added `s
+000000f0: 6b62 696f 2e73 7461 7473 2e63 6f6d 706f  kbio.stats.compo
+00000100: 7369 7469 6f6e 2e74 7265 655f 6261 7369  sition.tree_basi
+00000110: 7360 2074 6f20 636f 6e73 7472 7563 7420  s` to construct 
+00000120: 494c 5220 6261 7365 7320 6672 6f6d 2060  ILR bases from `
+00000130: 5472 6565 4e6f 6465 6020 6f62 6a65 6374  TreeNode` object
+00000140: 732e 2028 5b23 3138 3632 5d28 6874 7470  s. ([#1862](http
+00000150: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+00000160: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+00000170: 6f2f 7075 6c6c 2f31 3836 3229 290a 2a20  o/pull/1862)).* 
+00000180: 6049 6e74 6572 7661 6c4d 6574 6164 6174  `IntervalMetadat
+00000190: 612e 7175 6572 7960 206e 6f77 2064 6566  a.query` now def
+000001a0: 6175 6c74 7320 746f 206f 6274 6169 6e69  aults to obtaini
+000001b0: 6e67 2061 6c6c 2072 6573 756c 7473 2c20  ng all results, 
+000001c0: 7365 6520 285b 2331 3831 375d 2868 7474  see ([#1817](htt
+000001d0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+000001e0: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
+000001f0: 696f 2f69 7373 7565 732f 3138 3137 292e  io/issues/1817).
+00000200: 0a0a 2323 2320 4261 636b 7761 7264 2d69  ..### Backward-i
+00000210: 6e63 6f6d 7061 7469 626c 6520 6368 616e  ncompatible chan
+00000220: 6765 7320 5b65 7870 6572 696d 656e 7461  ges [experimenta
+00000230: 6c5d 0a2a 2057 6974 6820 7468 6520 696e  l].* With the in
+00000240: 7472 6f64 7563 7469 6f6e 206f 6620 7468  troduction of th
+00000250: 6520 6074 7265 655f 6261 7369 7360 206f  e `tree_basis` o
+00000260: 626a 6563 742c 2074 6865 2049 4c52 2062  bject, the ILR b
+00000270: 6173 6573 2061 7265 206e 6f77 2072 6570  ases are now rep
+00000280: 7265 7365 6e74 6564 2069 6e20 6c6f 672d  resented in log-
+00000290: 6f64 6473 2063 6f6f 7264 696e 6174 6573  odds coordinates
+000002a0: 2072 6174 6865 7220 7468 616e 2069 6e20   rather than in 
+000002b0: 7072 6f62 6162 696c 6974 6965 7320 746f  probabilities to
+000002c0: 206d 696e 696d 697a 6520 6973 7375 6573   minimize issues
+000002d0: 2077 6974 6820 6e75 6d65 7269 6361 6c20   with numerical 
+000002e0: 7374 6162 696c 6974 792e 2046 7572 7468  stability. Furth
+000002f0: 6572 6d6f 7265 2c20 7468 6520 6069 6c72  ermore, the `ilr
+00000300: 6020 616e 6420 6069 6c72 5f69 6e76 6020  ` and `ilr_inv` 
+00000310: 6675 6e63 7469 6f6e 7320 6e6f 7720 7461  functions now ta
+00000320: 6b65 7320 7468 6520 6062 6173 6973 6020  kes the `basis` 
+00000330: 696e 7075 7420 7061 7261 6d65 7465 7220  input parameter 
+00000340: 696e 2074 6572 6d73 206f 6620 6c6f 672d  in terms of log-
+00000350: 6f64 6473 2063 6f6f 7264 696e 6174 6573  odds coordinates
+00000360: 2e20 5468 6973 2061 6666 6563 7473 2074  . This affects t
+00000370: 6865 2060 736b 6269 6f2e 7374 6174 732e  he `skbio.stats.
+00000380: 636f 6d70 6f73 6974 696f 6e2e 7362 705f  composition.sbp_
+00000390: 6261 7369 7360 2061 7320 7765 6c6c 2e20  basis` as well. 
+000003a0: 285b 2331 3836 325d 2868 7474 7073 3a2f  ([#1862](https:/
+000003b0: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
+000003c0: 6f72 652f 7363 696b 6974 2d62 696f 2f70  ore/scikit-bio/p
+000003d0: 756c 6c2f 3138 3632 2929 0a0a 2323 2320  ull/1862))..### 
+000003e0: 496d 706f 7274 616e 740a 0a2a 2043 6f6d  Important..* Com
+000003f0: 706c 6578 206d 756c 7469 706c 6520 6178  plex multiple ax
+00000400: 6973 2069 6e64 6578 696e 6720 6f70 6572  is indexing oper
+00000410: 6174 696f 6e73 2077 6974 6820 6054 6162  ations with `Tab
+00000420: 756c 6172 4d53 4160 2068 6176 6520 6265  ularMSA` have be
+00000430: 656e 2072 656d 6f76 6564 2066 726f 6d20  en removed from 
+00000440: 7465 7374 696e 6720 6475 6520 746f 2069  testing due to i
+00000450: 6e63 6f6d 7061 7469 6269 6c69 7469 6573  ncompatibilities
+00000460: 2077 6974 6820 6d6f 6465 726e 2076 6572   with modern ver
+00000470: 7369 6f6e 7320 6f66 2050 616e 6461 732e  sions of Pandas.
+00000480: 2028 5b23 3138 3531 5d28 6874 7470 733a   ([#1851](https:
+00000490: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
+000004a0: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
+000004b0: 7075 6c6c 2f31 3835 3129 290a 2a20 5069  pull/1851)).* Pi
+000004c0: 6e6e 696e 6720 6073 6369 7079 203c 3d20  nning `scipy <= 
+000004d0: 312e 3130 2e31 6020 285b 2331 3835 315d  1.10.1` ([#1851]
+000004e0: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+000004f0: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+00000500: 6974 2d62 696f 2f70 756c 6c2f 3138 3637  it-bio/pull/1867
+00000510: 2929 0a0a 2323 2320 4275 6720 6669 7865  ))..### Bug fixe
+00000520: 730a 0a2a 2046 6978 6564 2061 2062 7567  s..* Fixed a bug
+00000530: 2074 6861 7420 6361 7573 6564 2062 7569   that caused bui
+00000540: 6c64 2066 6169 6c75 7265 206f 6e20 7468  ld failure on th
+00000550: 6520 4152 4d36 3420 6d69 6372 6f61 7263  e ARM64 microarc
+00000560: 6869 7465 6374 7572 6520 6475 6520 746f  hitecture due to
+00000570: 2066 6c6f 6174 696e 672d 706f 696e 7420   floating-point 
+00000580: 6e75 6d62 6572 2068 616e 646c 696e 672e  number handling.
+00000590: 2028 5b23 3138 3539 5d28 6874 7470 733a   ([#1859](https:
+000005a0: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
+000005b0: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
+000005c0: 7075 6c6c 2f31 3835 3929 290a 2a20 4e65  pull/1859)).* Ne
+000005d0: 7665 7220 6c65 7420 7468 6520 4769 6e69  ver let the Gini
+000005e0: 2069 6e64 6578 2067 6f20 6265 6c6f 7720   index go below 
+000005f0: 302e 302c 2073 6565 205b 2331 3834 345d  0.0, see [#1844]
+00000600: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+00000610: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+00000620: 6974 2d62 696f 2f69 7373 7565 2f31 3834  it-bio/issue/184
+00000630: 3429 2e0a 2a20 4669 7865 6420 6275 6720  4)..* Fixed bug 
+00000640: 5b23 3138 3437 5d28 6874 7470 733a 2f2f  [#1847](https://
+00000650: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
+00000660: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
+00000670: 7375 6573 2f31 3834 3729 2069 6e20 7768  sues/1847) in wh
+00000680: 6963 6820 7468 6520 6564 6765 2066 726f  ich the edge fro
+00000690: 6d20 7468 6520 726f 6f74 2077 6173 2069  m the root was i
+000006a0: 6e61 6476 6572 7461 6e74 6c79 2069 6e63  nadvertantly inc
+000006b0: 6c75 6465 6420 696e 2074 6865 2063 616c  luded in the cal
+000006c0: 6375 6c61 7469 6f6e 2066 6f72 2060 6465  culation for `de
+000006d0: 7363 656e 6469 6e67 5f62 7261 6e63 685f  scending_branch_
+000006e0: 6c65 6e67 7468 600a 0a23 2323 204d 6973  length`..### Mis
+000006f0: 6365 6c6c 616e 656f 7573 0a0a 2a20 5265  cellaneous..* Re
+00000700: 706c 6163 6564 2064 6570 656e 6465 6e63  placed dependenc
+00000710: 6965 7320 6043 6163 6865 436f 6e74 726f  ies `CacheContro
+00000720: 6c60 2061 6e64 2060 6c6f 636b 6669 6c65  l` and `lockfile
+00000730: 6020 7769 7468 2060 7265 7175 6573 7473  ` with `requests
+00000740: 6020 746f 2061 766f 6964 2061 2064 6570  ` to avoid a dep
+00000750: 656e 6465 6e63 7920 696e 636f 6e73 6973  endency inconsis
+00000760: 7465 6e63 7920 6973 7375 6520 6f66 2074  tency issue of t
+00000770: 6865 2066 6f72 6d65 722e 2028 5365 6520  he former. (See 
+00000780: 5b23 3138 3633 5d28 6874 7470 733a 2f2f  [#1863](https://
+00000790: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
+000007a0: 7265 2f73 6369 6b69 742d 6269 6f2f 7075  re/scikit-bio/pu
+000007b0: 6c6c 2f31 3836 3329 2c20 6d65 7267 6564  ll/1863), merged
+000007c0: 2069 6e20 5b23 3138 3539 5d28 6874 7470   in [#1859](http
+000007d0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+000007e0: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+000007f0: 6f2f 7075 6c6c 2f31 3835 3929 290a 2a20  o/pull/1859)).* 
+00000800: 5570 6461 7465 6420 696e 7374 616c 6c61  Updated installa
+00000810: 7469 6f6e 2069 6e73 7472 7563 7469 6f6e  tion instruction
+00000820: 7320 666f 7220 6465 7665 6c6f 7065 7273  s for developers
+00000830: 2069 6e20 6043 4f4e 5452 4942 5554 494e   in `CONTRIBUTIN
+00000840: 472e 6d64 6020 285b 2331 3836 305d 2868  G.md` ([#1860](h
+00000850: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+00000860: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
+00000870: 2d62 696f 2f70 756c 6c2f 3138 3630 2929  -bio/pull/1860))
+00000880: 0a0a 2323 2056 6572 7369 6f6e 2030 2e35  ..## Version 0.5
+00000890: 2e38 0a0a 2323 2320 4665 6174 7572 6573  .8..### Features
+000008a0: 0a0a 2a20 4164 6465 6420 4e43 4249 2074  ..* Added NCBI t
+000008b0: 6178 6f6e 6f6d 7920 6461 7461 6261 7365  axonomy database
+000008c0: 2064 756d 7020 666f 726d 6174 2028 6074   dump format (`t
+000008d0: 6178 6475 6d70 6029 2028 5b23 3138 3130  axdump`) ([#1810
+000008e0: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+000008f0: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
+00000900: 6b69 742d 6269 6f2f 7075 6c6c 2f31 3831  kit-bio/pull/181
+00000910: 3029 292e 0a2a 2041 6464 6564 2060 5472  0))..* Added `Tr
+00000920: 6565 4e6f 6465 2e66 726f 6d5f 7461 7864  eeNode.from_taxd
+00000930: 756d 7060 2066 6f72 2063 6f6e 7665 7274  ump` for convert
+00000940: 696e 6720 7461 7864 756d 7020 696e 746f  ing taxdump into
+00000950: 2061 2074 7265 6520 285b 2331 3831 305d   a tree ([#1810]
+00000960: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+00000970: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+00000980: 6974 2d62 696f 2f70 756c 6c2f 3138 3130  it-bio/pull/1810
+00000990: 2929 2e0a 2a20 7363 696b 6974 2d6c 6561  ))..* scikit-lea
+000009a0: 726e 2068 6173 2062 6565 6e20 7265 6d6f  rn has been remo
+000009b0: 7665 6420 6173 2061 2064 6570 656e 6465  ved as a depende
+000009c0: 6e63 792e 2054 6869 7320 7761 7320 6120  ncy. This was a 
+000009d0: 6661 6972 6c79 2068 6561 7679 2d77 6569  fairly heavy-wei
+000009e0: 6768 7420 6465 7065 6e64 656e 6379 2074  ght dependency t
+000009f0: 6861 7420 7761 7320 7072 6f76 6964 696e  hat was providin
+00000a00: 6720 6d69 6e6f 7220 6675 6e63 7469 6f6e  g minor function
+00000a10: 616c 6974 7920 746f 2073 6369 6b69 742d  ality to scikit-
+00000a20: 6269 6f2e 2054 6865 2063 7269 7469 6361  bio. The critica
+00000a30: 6c20 636f 6d70 6f6e 656e 7473 2068 6176  l components hav
+00000a40: 6520 6265 656e 2069 6d70 6c65 6d65 6e74  e been implement
+00000a50: 6564 2069 6e20 7363 696b 6974 2d62 696f  ed in scikit-bio
+00000a60: 2064 6972 6563 746c 792c 2061 6e64 2074   directly, and t
+00000a70: 6865 206e 6f6e 2d63 7269 7469 6369 616c  he non-criticial
+00000a80: 2063 6f6d 706f 6e65 6e74 7320 6172 6520   components are 
+00000a90: 6c69 7374 6564 2075 6e64 6572 2022 4261  listed under "Ba
+00000aa0: 636b 7761 7264 2d69 6e63 6f6d 7061 7469  ckward-incompati
+00000ab0: 626c 6520 6368 616e 6765 7320 5b65 7870  ble changes [exp
+00000ac0: 6572 696d 656e 7461 6c5d 222e 0a2a 2050  erimental]"..* P
+00000ad0: 7974 686f 6e20 332e 3131 2069 7320 6e6f  ython 3.11 is no
+00000ae0: 7720 7375 7070 6f72 7465 642e 0a0a 2323  w supported...##
+00000af0: 2320 4261 636b 7761 7264 2d69 6e63 6f6d  # Backward-incom
+00000b00: 7061 7469 626c 6520 6368 616e 6765 7320  patible changes 
+00000b10: 5b65 7870 6572 696d 656e 7461 6c5d 0a2a  [experimental].*
+00000b20: 2057 6974 6820 7468 6520 7265 6d6f 7661   With the remova
+00000b30: 6c20 6f66 2074 6865 2073 6369 6b69 742d  l of the scikit-
+00000b40: 6c65 6172 6e20 6465 7065 6e64 656e 6379  learn dependency
+00000b50: 2c20 7468 7265 6520 6265 7461 2064 6976  , three beta div
+00000b60: 6572 7369 7479 206d 6574 7269 6320 6e61  ersity metric na
+00000b70: 6d65 7320 6361 6e20 6e6f 206c 6f6e 6765  mes can no longe
+00000b80: 7220 6265 2073 7065 6369 6669 6564 2e20  r be specified. 
+00000b90: 5468 6573 6520 6172 6520 6077 6d69 6e6b  These are `wmink
+00000ba0: 6f77 736b 6960 2c20 606e 616e 5f65 7563  owski`, `nan_euc
+00000bb0: 6c69 6465 616e 602c 2061 6e64 2060 6861  lidean`, and `ha
+00000bc0: 7665 7273 696e 6560 2e20 4f6e 2074 6573  versine`. On tes
+00000bd0: 7469 6e67 2c20 6077 6d69 6e6b 6f77 736b  ting, `wminkowsk
+00000be0: 6960 2061 6e64 2060 6861 7665 7273 696e  i` and `haversin
+00000bf0: 6560 2064 6964 206e 6f74 2077 6f72 6b20  e` did not work 
+00000c00: 7468 726f 7567 6820 6073 6b62 696f 2e64  through `skbio.d
+00000c10: 6976 6572 7369 7479 2e62 6574 615f 6469  iversity.beta_di
+00000c20: 7665 7273 6974 7960 2028 6f72 2060 736b  versity` (or `sk
+00000c30: 6c65 6172 6e2e 6d65 7472 6963 732e 7061  learn.metrics.pa
+00000c40: 6972 7769 7365 5f64 6973 7461 6e63 6573  irwise_distances
+00000c50: 6029 2e20 5468 6520 666f 726d 6572 2077  `). The former w
+00000c60: 6173 2064 6570 7265 6361 7465 6420 696e  as deprecated in
+00000c70: 2066 6176 6f72 206f 6620 6361 6c6c 696e   favor of callin
+00000c80: 6720 606d 696e 6b6f 7773 6b69 6020 7769  g `minkowski` wi
+00000c90: 7468 2061 2076 6563 746f 7220 6f66 2077  th a vector of w
+00000ca0: 6569 6768 7473 2070 726f 7669 6465 6420  eights provided 
+00000cb0: 6173 206b 7761 7267 2060 7760 2028 6578  as kwarg `w` (ex
+00000cc0: 616d 706c 6520 6265 6c6f 7729 2c20 616e  ample below), an
+00000cd0: 6420 7468 6520 6c61 7474 6572 2064 6f65  d the latter doe
+00000ce0: 7320 6e6f 7420 776f 726b 2077 6974 6820  s not work with 
+00000cf0: 6461 7461 206f 6620 7468 6973 2073 6861  data of this sha
+00000d00: 7065 2e20 606e 616e 5f65 7563 6c69 6465  pe. `nan_euclide
+00000d10: 616e 6020 6361 6e20 7374 696c 6c20 6265  an` can still be
+00000d20: 2061 6363 6573 7365 6420 6672 6f6e 2073   accessed fron s
+00000d30: 6369 6b69 742d 6c65 6172 6e20 6469 7265  cikit-learn dire
+00000d40: 6374 6c79 2069 6620 6e65 6564 6564 2c20  ctly if needed, 
+00000d50: 6966 2061 2075 7365 7220 696e 7374 616c  if a user instal
+00000d60: 6c73 2073 6369 6b69 742d 6c65 6172 6e20  ls scikit-learn 
+00000d70: 696e 2074 6865 6972 2065 6e76 6972 6f6e  in their environ
+00000d80: 6d65 6e74 2028 6578 616d 706c 6520 6265  ment (example be
+00000d90: 6c6f 7729 2e0a 0a20 2020 2060 6060 0a20  low)...    ```. 
+00000da0: 2020 2063 6f75 6e74 7320 3d20 5b5b 3233     counts = [[23
+00000db0: 2c20 3634 2c20 3134 2c20 302c 2030 2c20  , 64, 14, 0, 0, 
+00000dc0: 332c 2031 5d2c 0a20 2020 2020 2020 2020  3, 1],.         
+00000dd0: 2020 205b 302c 2033 2c20 3335 2c20 3432     [0, 3, 35, 42
+00000de0: 2c20 302c 2031 322c 2031 5d2c 0a20 2020  , 0, 12, 1],.   
+00000df0: 2020 2020 2020 2020 205b 302c 2035 2c20           [0, 5, 
+00000e00: 352c 2030 2c20 3430 2c20 3430 2c20 305d  5, 0, 40, 40, 0]
+00000e10: 2c0a 2020 2020 2020 2020 2020 2020 5b34  ,.            [4
+00000e20: 342c 2033 352c 2039 2c20 302c 2031 2c20  4, 35, 9, 0, 1, 
+00000e30: 302c 2030 5d2c 0a20 2020 2020 2020 2020  0, 0],.         
+00000e40: 2020 205b 302c 2032 2c20 382c 2030 2c20     [0, 2, 8, 0, 
+00000e50: 3335 2c20 3435 2c20 315d 2c0a 2020 2020  35, 45, 1],.    
+00000e60: 2020 2020 2020 2020 5b30 2c20 302c 2032          [0, 0, 2
+00000e70: 352c 2033 352c 2030 2c20 3139 2c20 305d  5, 35, 0, 19, 0]
+00000e80: 2c0a 2020 2020 2020 2020 2020 2020 5b38  ,.            [8
+00000e90: 382c 2033 312c 2030 2c20 352c 2035 2c20  8, 31, 0, 5, 5, 
+00000ea0: 352c 2035 5d2c 0a20 2020 2020 2020 2020  5, 5],.         
+00000eb0: 2020 205b 3434 2c20 3339 2c20 302c 2030     [44, 39, 0, 0
+00000ec0: 2c20 302c 2030 2c20 305d 5d0a 0a20 2020  , 0, 0, 0]]..   
+00000ed0: 2023 206e 6577 206d 6563 6861 6e69 736d   # new mechanism
+00000ee0: 206f 6620 6163 6365 7373 696e 6720 776d   of accessing wm
+00000ef0: 696e 6b6f 7773 6b69 0a20 2020 2066 726f  inkowski.    fro
+00000f00: 6d20 736b 6269 6f2e 6469 7665 7273 6974  m skbio.diversit
+00000f10: 7920 696d 706f 7274 2062 6574 615f 6469  y import beta_di
+00000f20: 7665 7273 6974 790a 2020 2020 6265 7461  versity.    beta
+00000f30: 5f64 6976 6572 7369 7479 2822 6d69 6e6b  _diversity("mink
+00000f40: 6f77 736b 6922 2c20 636f 756e 7473 2c20  owski", counts, 
+00000f50: 773d 5b31 2c31 2c31 2c31 2c31 2c31 2c32  w=[1,1,1,1,1,1,2
+00000f60: 5d29 0a0a 2020 2020 2320 6163 6365 7373  ])..    # access
+00000f70: 696e 6720 6e61 6e5f 6575 636c 6964 6561  ing nan_euclidea
+00000f80: 6e20 7468 726f 7567 6820 7363 696b 6974  n through scikit
+00000f90: 2d6c 6561 726e 2064 6972 6563 746c 790a  -learn directly.
+00000fa0: 2020 2020 696d 706f 7274 2073 6b62 696f      import skbio
+00000fb0: 0a20 2020 2066 726f 6d20 736b 6c65 6172  .    from sklear
+00000fc0: 6e2e 6d65 7472 6963 7320 696d 706f 7274  n.metrics import
+00000fd0: 2070 6169 7277 6973 655f 6469 7374 616e   pairwise_distan
+00000fe0: 6365 730a 2020 2020 736b 6c65 6172 6e5f  ces.    sklearn_
+00000ff0: 646d 203d 2070 6169 7277 6973 655f 6469  dm = pairwise_di
+00001000: 7374 616e 6365 7328 636f 756e 7473 2c20  stances(counts, 
+00001010: 6d65 7472 6963 3d22 6e61 6e5f 6575 636c  metric="nan_eucl
+00001020: 6964 6561 6e22 290a 2020 2020 736b 6269  idean").    skbi
+00001030: 6f5f 646d 203d 2073 6b62 696f 2e44 6973  o_dm = skbio.Dis
+00001040: 7461 6e63 654d 6174 7269 7828 736b 6c65  tanceMatrix(skle
+00001050: 6172 6e5f 646d 290a 2020 2020 6060 600a  arn_dm).    ```.
+00001060: 0a23 2323 2044 6570 7265 6361 7465 6420  .### Deprecated 
+00001070: 6675 6e63 7469 6f6e 616c 6974 7920 5b65  functionality [e
+00001080: 7870 6572 696d 656e 7461 6c5d 0a2a 2060  xperimental].* `
+00001090: 736b 6269 6f2e 616c 6967 6e6d 656e 742e  skbio.alignment.
+000010a0: 6c6f 6361 6c5f 7061 6972 7769 7365 5f61  local_pairwise_a
+000010b0: 6c69 676e 5f73 7377 6020 6861 7320 6265  lign_ssw` has be
+000010c0: 656e 2064 6570 7265 6361 7465 6420 285b  en deprecated ([
+000010d0: 2331 3831 345d 2868 7474 7073 3a2f 2f67  #1814](https://g
+000010e0: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
+000010f0: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
+00001100: 7565 732f 3138 3134 2929 2061 6e64 2077  ues/1814)) and w
+00001110: 696c 6c20 6265 2072 656d 6f76 6564 206f  ill be removed o
+00001120: 7220 7265 706c 6163 6564 2069 6e20 7363  r replaced in sc
+00001130: 696b 6974 2d62 696f 2030 2e36 2e30 2e0a  ikit-bio 0.6.0..
+00001140: 0a23 2323 2042 7567 2066 6978 6573 0a2a  .### Bug fixes.*
+00001150: 2055 7365 2060 6f6c 6465 7374 2d73 7570   Use `oldest-sup
+00001160: 706f 7274 6564 2d6e 756d 7079 6020 6173  ported-numpy` as
+00001170: 2062 7569 6c64 2064 6570 656e 6465 6e63   build dependenc
+00001180: 792e 2054 6869 7320 6669 7865 7320 7072  y. This fixes pr
+00001190: 6f62 6c65 6d73 2077 6974 6820 656e 7669  oblems with envi
+000011a0: 726f 6e6d 656e 7473 2074 6861 7420 7573  ronments that us
+000011b0: 6520 616e 206f 6c64 6572 2076 6572 7369  e an older versi
+000011c0: 6f6e 206f 6620 6e75 6d70 7920 7468 616e  on of numpy than
+000011d0: 2074 6865 206f 6e65 2075 7365 6420 746f   the one used to
+000011e0: 2062 7569 6c64 2073 6369 6b69 742d 6269   build scikit-bi
+000011f0: 6f20 285b 2331 3831 335d 2868 7474 7073  o ([#1813](https
+00001200: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
+00001210: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
+00001220: 2f70 756c 6c2f 3138 3133 2929 2e0a 0a23  /pull/1813))...#
+00001230: 2320 5665 7273 696f 6e20 302e 352e 370a  # Version 0.5.7.
+00001240: 0a23 2323 2046 6561 7475 7265 730a 0a2a  .### Features..*
+00001250: 2049 6e74 726f 6475 6365 2073 7570 706f   Introduce suppo
+00001260: 7274 2066 6f72 2050 7974 686f 6e20 332e  rt for Python 3.
+00001270: 3130 2028 5b23 3138 3031 5d28 6874 7470  10 ([#1801](http
+00001280: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+00001290: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+000012a0: 6f2f 7075 6c6c 2f31 3830 3129 292e 0a2a  o/pull/1801))..*
+000012b0: 2054 656e 7461 7469 7665 2073 7570 706f   Tentative suppo
+000012c0: 7274 2066 6f72 2041 7070 6c65 204d 3120  rt for Apple M1 
+000012d0: 285b 2331 3730 395d 2868 7474 7073 3a2f  ([#1709](https:/
+000012e0: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
+000012f0: 6f72 652f 7363 696b 6974 2d62 696f 2f70  ore/scikit-bio/p
+00001300: 756c 6c2f 3137 3039 2929 2e0a 2a20 4164  ull/1709))..* Ad
+00001310: 6465 6420 7375 7070 6f72 7420 666f 7220  ded support for 
+00001320: 7265 6164 696e 6720 616e 6420 7772 6974  reading and writ
+00001330: 696e 6720 6120 6269 6e61 7279 2064 6973  ing a binary dis
+00001340: 7461 6e63 6520 6d61 7472 6978 206f 626a  tance matrix obj
+00001350: 6563 7420 666f 726d 6174 2e20 285b 2331  ect format. ([#1
+00001360: 3731 365d 2868 7474 7073 3a2f 2f67 6974  716](https://git
+00001370: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+00001380: 7363 696b 6974 2d62 696f 2f70 756c 6c2f  scikit-bio/pull/
+00001390: 3137 3136 2929 0a2a 2041 6464 6564 2073  1716)).* Added s
+000013a0: 7570 706f 7274 2066 6f72 2060 6e70 2e66  upport for `np.f
+000013b0: 6c6f 6174 3332 6020 7769 7468 2060 4469  loat32` with `Di
+000013c0: 7373 696d 696c 6172 6974 794d 6174 7269  ssimilarityMatri
+000013d0: 7860 206f 626a 6563 7473 2e0a 2a20 4164  x` objects..* Ad
+000013e0: 6465 6420 7375 7070 6f72 7420 666f 7220  ded support for 
+000013f0: 6d65 7468 6f64 2061 6e64 206e 756d 6265  method and numbe
+00001400: 725f 6f66 5f64 696d 656e 7369 6f6e 7320  r_of_dimensions 
+00001410: 746f 2070 6572 6d64 6973 7020 7265 6475  to permdisp redu
+00001420: 6369 6e67 2074 6865 2072 756e 7469 6d65  cing the runtime
+00001430: 2062 7920 3130 3078 2061 7420 3430 3030   by 100x at 4000
+00001440: 2073 616d 706c 6573 2c20 5b69 7373 7565   samples, [issue
+00001450: 2023 3137 3639 5d28 6874 7470 733a 2f2f   #1769](https://
+00001460: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
+00001470: 7265 2f73 6369 6b69 742d 6269 6f2f 7075  re/scikit-bio/pu
+00001480: 6c6c 2f31 3736 3929 2e0a 2a20 4f72 6469  ll/1769)..* Ordi
+00001490: 6e61 7469 6f6e 5265 7375 6c74 7320 6f62  nationResults ob
+000014a0: 6a65 6374 2069 7320 6e6f 7720 6163 6365  ject is now acce
+000014b0: 7074 6564 2061 7320 696e 7075 7420 666f  pted as input fo
+000014c0: 7220 7065 726d 6469 7370 2e0a 0a23 2323  r permdisp...###
+000014d0: 2050 6572 666f 726d 616e 6365 2065 6e68   Performance enh
+000014e0: 616e 6365 6d65 6e74 730a 0a2a 2041 766f  ancements..* Avo
+000014f0: 6964 2061 6e20 696d 706c 6963 6974 2064  id an implicit d
+00001500: 6174 6120 636f 7079 206f 6e20 636f 6e73  ata copy on cons
+00001510: 7472 7563 7469 6f6e 206f 6620 6044 6973  truction of `Dis
+00001520: 7369 6d69 6c61 7269 7479 4d61 7472 6978  similarityMatrix
+00001530: 6020 6f62 6a65 6374 732e 0a2a 2041 766f  ` objects..* Avo
+00001540: 6964 2076 616c 6964 6174 696f 6e20 6f6e  id validation on
+00001550: 2063 6f70 7920 6f66 2060 4469 7373 696d   copy of `Dissim
+00001560: 696c 6172 6974 794d 6174 7269 7860 2061  ilarityMatrix` a
+00001570: 6e64 2060 4469 7374 616e 6365 4d61 7472  nd `DistanceMatr
+00001580: 6978 6020 6f62 6a65 6374 732c 2073 6565  ix` objects, see
+00001590: 205b 5052 2023 3137 3437 5d28 6874 7470   [PR #1747](http
+000015a0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+000015b0: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+000015c0: 6f2f 7075 6c6c 2f31 3734 3729 0a2a 2055  o/pull/1747).* U
+000015d0: 7365 2061 6e20 6f70 7469 6d69 7a65 6420  se an optimized 
+000015e0: 7665 7273 696f 6e20 6f66 2073 796d 6d65  version of symme
+000015f0: 7472 7920 6368 6563 6b20 696e 2044 6973  try check in Dis
+00001600: 7461 6e63 654d 6174 7269 782c 2073 6565  tanceMatrix, see
+00001610: 205b 5052 2023 3137 3437 5d28 6874 7470   [PR #1747](http
+00001620: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+00001630: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+00001640: 6f2f 7075 6c6c 2f31 3734 3729 0a2a 2041  o/pull/1747).* A
+00001650: 766f 6964 2070 6572 666f 726d 696e 6720  void performing 
+00001660: 6669 6c74 6572 696e 6720 7768 656e 2069  filtering when i
+00001670: 6473 2061 7265 2069 6465 6e74 6963 616c  ds are identical
+00001680: 2c20 7365 6520 5b50 5220 2331 3735 325d  , see [PR #1752]
+00001690: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+000016a0: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+000016b0: 6974 2d62 696f 2f70 756c 6c2f 3137 3532  it-bio/pull/1752
+000016c0: 290a 2a20 6365 6e74 6572 5f64 6973 7461  ).* center_dista
+000016d0: 6e63 655f 6d61 7472 6978 2068 6173 2062  nce_matrix has b
+000016e0: 6565 6e20 7265 2d69 6d70 6c65 6d65 6e74  een re-implement
+000016f0: 6564 2069 6e20 6379 7468 6f6e 2066 6f72  ed in cython for
+00001700: 2062 6f74 6820 7370 6565 6420 616e 6420   both speed and 
+00001710: 6d65 6d6f 7279 2075 7365 2e20 496e 6469  memory use. Indi
+00001720: 7265 6374 6c79 2073 7065 6564 7320 7570  rectly speeds up
+00001730: 2070 636f 6120 5b50 5220 2331 3734 395d   pcoa [PR #1749]
+00001740: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+00001750: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+00001760: 6974 2d62 696f 2f70 756c 6c2f 3137 3439  it-bio/pull/1749
+00001770: 290a 2a20 5573 6520 6120 6d65 6d6f 7279  ).* Use a memory
+00001780: 2d6f 7074 696d 697a 6564 2076 6572 7369  -optimized versi
+00001790: 6f6e 206f 6620 7065 726d 7574 6520 696e  on of permute in
+000017a0: 2044 6973 7461 6e63 654d 6174 7269 782c   DistanceMatrix,
+000017b0: 2073 6565 205b 5052 2023 3137 3536 5d28   see [PR #1756](
+000017c0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+000017d0: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+000017e0: 742d 6269 6f2f 7075 6c6c 2f31 3735 3629  t-bio/pull/1756)
+000017f0: 2e0a 2a20 5265 6661 6374 6f72 2070 6561  ..* Refactor pea
+00001800: 7273 6f6e 2061 6e64 2073 7065 6172 6d61  rson and spearma
+00001810: 6e20 736b 6269 6f2e 7374 6174 732e 6469  n skbio.stats.di
+00001820: 7374 616e 6365 2e6d 616e 7465 6c20 696d  stance.mantel im
+00001830: 706c 656d 656e 7461 7469 6f6e 7320 746f  plementations to
+00001840: 2064 7261 7374 6963 616c 6c79 2069 6d70   drastically imp
+00001850: 726f 7665 206d 656d 6f72 7920 6c6f 6361  rove memory loca
+00001860: 6c69 7479 2e20 416c 736f 2063 6163 6865  lity. Also cache
+00001870: 2069 6e74 6572 6d65 6469 6174 6520 7265   intermediate re
+00001880: 7375 6c74 7320 7468 6174 2061 7265 2069  sults that are i
+00001890: 6e76 6172 6961 6e74 2061 6372 6f73 7320  nvariant across 
+000018a0: 7065 726d 7574 6174 696f 6e73 2c20 7365  permutations, se
+000018b0: 6520 5b50 5220 2331 3735 365d 2868 7474  e [PR #1756](htt
+000018c0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+000018d0: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
+000018e0: 696f 2f70 756c 6c2f 3137 3536 292e 0a2a  io/pull/1756)..*
+000018f0: 2052 6566 6163 746f 7220 7065 726d 616e   Refactor perman
+00001900: 6f76 6120 746f 2072 656d 6f76 6520 696e  ova to remove in
+00001910: 7465 726d 6564 6961 7465 2062 7566 6665  termediate buffe
+00001920: 7273 2061 6e64 2063 7974 686f 6e69 7a65  rs and cythonize
+00001930: 2074 6865 2069 6e74 6572 6e61 6c73 2c20   the internals, 
+00001940: 7365 6520 5b50 5220 2331 3736 385d 2868  see [PR #1768](h
+00001950: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+00001960: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
+00001970: 2d62 696f 2f70 756c 6c2f 3137 3638 292e  -bio/pull/1768).
+00001980: 0a0a 2323 2320 4275 6720 6669 7865 730a  ..### Bug fixes.
+00001990: 0a2a 2046 6978 2077 696e 646f 7773 2061  .* Fix windows a
+000019a0: 6e64 2033 3262 6974 2069 6e63 6f6d 7061  nd 32bit incompa
+000019b0: 7469 6269 6c69 7479 2069 6e20 6075 6e77  tibility in `unw
+000019c0: 6569 6768 7465 645f 756e 6966 7261 6360  eighted_unifrac`
+000019d0: 2e0a 0a23 2323 204d 6973 6365 6c6c 616e  ...### Miscellan
+000019e0: 656f 7573 0a0a 2a20 5079 7468 6f6e 2033  eous..* Python 3
+000019f0: 2e36 2068 6173 2062 6565 6e20 7265 6d6f  .6 has been remo
+00001a00: 7665 6420 6672 6f6d 206f 7572 2074 6573  ved from our tes
+00001a10: 7469 6e67 206d 6174 7269 782e 0a2a 2053  ting matrix..* S
+00001a20: 7065 6369 6679 2062 7569 6c64 2064 6570  pecify build dep
+00001a30: 656e 6465 6e63 6965 7320 696e 2070 7970  endencies in pyp
+00001a40: 726f 6a65 6374 2e74 6f6d 6c2e 2054 6869  roject.toml. Thi
+00001a50: 7320 616c 6c6f 7773 2074 6865 2070 6163  s allows the pac
+00001a60: 6b61 6765 2074 6f20 6265 2069 6e73 7461  kage to be insta
+00001a70: 6c6c 6564 2077 6974 686f 7574 2068 6176  lled without hav
+00001a80: 696e 6720 746f 2066 6972 7374 206d 616e  ing to first man
+00001a90: 7561 6c6c 7920 696e 7374 616c 6c20 6e75  ually install nu
+00001aa0: 6d70 792e 0a2a 2055 7064 6174 6520 6864  mpy..* Update hd
+00001ab0: 6d65 6469 616e 7320 7061 636b 6167 6520  medians package 
+00001ac0: 746f 2061 2076 6572 7369 6f6e 2077 6869  to a version whi
+00001ad0: 6368 2064 6f65 736e 2774 2072 6571 7569  ch doesn't requi
+00001ae0: 7265 2061 6e20 696e 6974 6961 6c20 6d61  re an initial ma
+00001af0: 6e75 616c 206e 756d 7079 2069 6e73 7461  nual numpy insta
+00001b00: 6c6c 2e0a 2a20 4e6f 7720 6275 696c 6461  ll..* Now builda
+00001b10: 626c 6520 6f6e 206e 6f6e 2d78 3836 2070  ble on non-x86 p
+00001b20: 6c61 7466 6f72 6d73 2064 7565 2074 6f20  latforms due to 
+00001b30: 7573 6520 6f66 2074 6865 205b 5349 4d44  use of the [SIMD
+00001b40: 2045 7665 7279 7768 6572 655d 2868 7474   Everywhere](htt
+00001b50: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+00001b60: 7369 6d64 2d65 7665 7279 7768 6572 652f  simd-everywhere/
+00001b70: 7369 6d64 6529 206c 6962 7261 7279 2e0a  simde) library..
+00001b80: 2a20 5265 6765 6e65 7261 7465 2043 7974  * Regenerate Cyt
+00001b90: 686f 6e20 7772 6170 7065 7220 6279 2064  hon wrapper by d
+00001ba0: 6566 6175 6c74 2074 6f20 6176 6f69 6420  efault to avoid 
+00001bb0: 696e 636f 6d70 6174 6962 696c 6974 6965  incompatibilitie
+00001bc0: 7320 7769 7468 2069 6e73 7461 6c6c 6564  s with installed
+00001bd0: 2043 5079 7468 6f6e 2e0a 2a20 5570 6461   CPython..* Upda
+00001be0: 7465 2064 6f63 756d 656e 7461 7469 6f6e  te documentation
+00001bf0: 2066 6f72 2074 6865 2060 736b 6269 6f2e   for the `skbio.
+00001c00: 7374 6174 732e 636f 6d70 6f73 6974 696f  stats.compositio
+00001c10: 6e2e 616e 636f 6d60 2066 756e 6374 696f  n.ancom` functio
+00001c20: 6e2e 2028 5b23 3137 3431 5d28 6874 7470  n. ([#1741](http
+00001c30: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+00001c40: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+00001c50: 6f2f 7075 6c6c 2f31 3734 3129 290a 0a23  o/pull/1741))..#
+00001c60: 2320 5665 7273 696f 6e20 302e 352e 360a  # Version 0.5.6.
+00001c70: 0a23 2323 2046 6561 7475 7265 730a 0a2a  .### Features..*
+00001c80: 2041 6464 6564 206f 7074 696f 6e20 746f   Added option to
+00001c90: 2072 6574 7572 6e20 6120 6361 7074 7572   return a captur
+00001ca0: 6520 6772 6f75 7020 636f 6d70 696c 6564  e group compiled
+00001cb0: 2072 6567 6578 2070 6174 7465 726e 2074   regex pattern t
+00001cc0: 6f20 616e 7920 636c 6173 7320 696e 6865  o any class inhe
+00001cd0: 7269 7469 6e67 2060 6047 7261 6d6d 6172  riting ``Grammar
+00001ce0: 6564 5365 7175 656e 6365 6060 2074 6872  edSequence`` thr
+00001cf0: 6f75 6768 2074 6865 2060 6074 6f5f 7265  ough the ``to_re
+00001d00: 6765 7860 6020 6d65 7468 6f64 2e20 285b  gex`` method. ([
+00001d10: 2331 3433 315d 2868 7474 7073 3a2f 2f67  #1431](https://g
+00001d20: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
+00001d30: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
+00001d40: 7565 732f 3134 3331 2929 0a0a 2a20 4164  ues/1431))..* Ad
+00001d50: 6465 6420 6044 6973 7369 6d69 6c61 7269  ded `Dissimilari
+00001d60: 7479 2e77 6974 6869 6e60 2061 6e64 2060  ty.within` and `
+00001d70: 2e62 6574 7765 656e 6020 746f 206f 6274  .between` to obt
+00001d80: 6169 6e20 7468 6520 7265 7370 6563 7469  ain the respecti
+00001d90: 7665 2064 6973 7461 6e63 6573 2061 6e64  ve distances and
+00001da0: 2065 7870 7265 7373 2074 6865 6d20 6173   express them as
+00001db0: 2061 2060 4461 7461 4672 616d 6560 2e20   a `DataFrame`. 
+00001dc0: 285b 2331 3636 325d 2868 7474 7073 3a2f  ([#1662](https:/
+00001dd0: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
+00001de0: 6f72 652f 7363 696b 6974 2d62 696f 2f70  ore/scikit-bio/p
+00001df0: 756c 6c2f 3136 3632 2929 0a0a 2a20 4164  ull/1662))..* Ad
+00001e00: 6465 6420 4b65 6e64 616c 6c20 5461 7520  ded Kendall Tau 
+00001e10: 6173 2070 6f73 7369 626c 6520 636f 7272  as possible corr
+00001e20: 656c 6174 696f 6e20 6d65 7468 6f64 2069  elation method i
+00001e30: 6e20 7468 6520 6073 6b62 696f 2e73 7461  n the `skbio.sta
+00001e40: 7473 2e64 6973 7461 6e63 652e 6d61 6e74  ts.distance.mant
+00001e50: 656c 6020 6675 6e63 7469 6f6e 2028 5b23  el` function ([#
+00001e60: 3136 3735 5d28 6874 7470 733a 2f2f 6769  1675](https://gi
+00001e70: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
+00001e80: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
+00001e90: 6573 2f31 3637 3529 292e 0a0a 2a20 4164  es/1675))...* Ad
+00001ea0: 6465 6420 7375 7070 6f72 7420 666f 7220  ded support for 
+00001eb0: 4955 5041 4320 616d 696e 6f20 6163 6964  IUPAC amino acid
+00001ec0: 2063 6f64 6573 2055 2028 7365 6c65 6e6f   codes U (seleno
+00001ed0: 6379 7374 6569 6e65 292c 204f 2028 7079  cysteine), O (py
+00001ee0: 7272 6f6c 7973 696e 6529 2c20 616e 6420  rrolysine), and 
+00001ef0: 4a20 286c 6575 6369 6e65 206f 7220 6973  J (leucine or is
+00001f00: 6f6c 6575 6369 6e65 292e 2028 5b23 3135  oleucine). ([#15
+00001f10: 3736 5d28 6874 7470 733a 2f2f 6769 7468  76](https://gith
+00001f20: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+00001f30: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
+00001f40: 2f31 3537 3629 0a0a 2323 2320 4261 636b  /1576)..### Back
 00001f50: 7761 7264 2d69 6e63 6f6d 7061 7469 626c  ward-incompatibl
 00001f60: 6520 6368 616e 6765 7320 5b73 7461 626c  e changes [stabl
 00001f70: 655d 0a0a 2323 2320 4261 636b 7761 7264  e]..### Backward
 00001f80: 2d69 6e63 6f6d 7061 7469 626c 6520 6368  -incompatible ch
 00001f90: 616e 6765 7320 5b65 7870 6572 696d 656e  anges [experimen
-00001fa0: 7461 6c5d 0a0a 2323 2320 5065 7266 6f72  tal]..### Perfor
-00001fb0: 6d61 6e63 6520 656e 6861 6e63 656d 656e  mance enhancemen
-00001fc0: 7473 0a0a 2323 2320 4275 6720 6669 7865  ts..### Bug fixe
-00001fd0: 730a 0a23 2323 2044 6570 7265 6361 7465  s..### Deprecate
-00001fe0: 6420 6675 6e63 7469 6f6e 616c 6974 7920  d functionality 
-00001ff0: 5b73 7461 626c 655d 0a0a 2323 2320 4465  [stable]..### De
-00002000: 7072 6563 6174 6564 2066 756e 6374 696f  precated functio
-00002010: 6e61 6c69 7479 205b 6578 7065 7269 6d65  nality [experime
-00002020: 6e74 616c 5d0a 0a23 2323 204d 6973 6365  ntal]..### Misce
-00002030: 6c6c 616e 656f 7573 0a2a 2050 7974 686f  llaneous.* Pytho
-00002040: 6e20 332e 3620 616e 6420 332e 3720 636f  n 3.6 and 3.7 co
-00002050: 6d70 6174 6962 696c 6974 7920 6973 206e  mpatibility is n
-00002060: 6f77 2073 7570 706f 7274 6564 0a0a 2a20  ow supported..* 
-00002070: 4120 7079 7465 7374 2072 756e 6e65 7220  A pytest runner 
-00002080: 6973 2073 6869 7070 6564 2077 6974 6820  is shipped with 
-00002090: 6576 6572 7920 696e 7374 616c 6c61 7469  every installati
-000020a0: 6f6e 2028 5b23 3136 3333 5d28 6874 7470  on ([#1633](http
-000020b0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-000020c0: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-000020d0: 6f2f 7075 6c6c 2f31 3633 3329 290a 0a2a  o/pull/1633))..*
-000020e0: 2054 6865 206e 6f73 6574 6573 7420 6672   The nosetest fr
-000020f0: 616d 6577 6f72 6b20 6861 7320 6265 656e  amework has been
-00002100: 2072 6570 6c61 6365 6420 696e 2066 6176   replaced in fav
-00002110: 6f72 206f 6620 7079 7465 7374 2028 5b23  or of pytest ([#
-00002120: 3136 3234 5d28 6874 7470 733a 2f2f 6769  1624](https://gi
-00002130: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
-00002140: 2f73 6369 6b69 742d 6269 6f2f 7075 6c6c  /scikit-bio/pull
-00002150: 2f31 3632 3429 290a 0a2a 2054 6865 206e  /1624))..* The n
-00002160: 756d 7079 2064 6f63 7320 6172 6520 6465  umpy docs are de
-00002170: 7072 6563 6174 6564 2069 6e20 6661 766f  precated in favo
-00002180: 7220 6f66 205b 4e61 706f 6c65 6f6e 5d28  r of [Napoleon](
-00002190: 6874 7470 3a2f 2f77 7777 2e73 7068 696e  http://www.sphin
-000021a0: 782d 646f 632e 6f72 672f 656e 2f6d 6173  x-doc.org/en/mas
-000021b0: 7465 722f 7573 6167 652f 6578 7465 6e73  ter/usage/extens
-000021c0: 696f 6e73 2f6e 6170 6f6c 656f 6e2e 6874  ions/napoleon.ht
-000021d0: 6d6c 2920 285b 2331 3632 395d 2868 7474  ml) ([#1629](htt
-000021e0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-000021f0: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-00002200: 696f 2f70 756c 6c2f 3136 3239 2929 0a0a  io/pull/1629))..
-00002210: 2a20 5468 6973 2076 6572 7369 6f6e 2069  * This version i
-00002220: 7320 6e6f 7720 636f 6d70 6174 6962 6c65  s now compatible
-00002230: 2077 6974 6820 4e75 6d50 7920 3e3d 2031   with NumPy >= 1
-00002240: 2e39 2e32 2061 6e64 2050 616e 6461 7320  .9.2 and Pandas 
-00002250: 3e3d 2030 2e32 332e 2028 5b23 3136 3237  >= 0.23. ([#1627
-00002260: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
-00002270: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
-00002280: 6b69 742d 6269 6f2f 7075 6c6c 2f31 3632  kit-bio/pull/162
-00002290: 3729 290a 0a23 2320 5665 7273 696f 6e20  7))..## Version 
-000022a0: 302e 352e 3420 2832 3031 382d 3038 2d32  0.5.4 (2018-08-2
-000022b0: 3329 0a0a 2323 2320 4665 6174 7572 6573  3)..### Features
-000022c0: 0a0a 2a20 4164 6465 6420 6046 5356 4460  ..* Added `FSVD`
-000022d0: 2c20 616e 2061 6c74 6572 6e61 7469 7665  , an alternative
-000022e0: 2066 6173 7420 6865 7572 6973 7469 6320   fast heuristic 
-000022f0: 6d65 7468 6f64 2074 6f20 7065 7266 6f72  method to perfor
-00002300: 6d20 5072 696e 6369 7061 6c20 436f 6f72  m Principal Coor
-00002310: 6469 6e61 7465 7320 416e 616c 7973 6973  dinates Analysis
-00002320: 2c20 746f 2060 736b 6269 6f2e 7374 6174  , to `skbio.stat
-00002330: 732e 6f72 6469 6e61 7469 6f6e 2e70 636f  s.ordination.pco
-00002340: 6160 2e0a 0a23 2323 2042 6163 6b77 6172  a`...### Backwar
-00002350: 642d 696e 636f 6d70 6174 6962 6c65 2063  d-incompatible c
-00002360: 6861 6e67 6573 205b 7374 6162 6c65 5d0a  hanges [stable].
-00002370: 0a23 2323 2042 6163 6b77 6172 642d 696e  .### Backward-in
-00002380: 636f 6d70 6174 6962 6c65 2063 6861 6e67  compatible chang
-00002390: 6573 205b 6578 7065 7269 6d65 6e74 616c  es [experimental
-000023a0: 5d0a 0a23 2323 2050 6572 666f 726d 616e  ]..### Performan
-000023b0: 6365 2065 6e68 616e 6365 6d65 6e74 730a  ce enhancements.
-000023c0: 0a2a 2041 6464 6564 206f 7074 696d 697a  .* Added optimiz
-000023d0: 6564 2075 7469 6c69 7479 206d 6574 686f  ed utility metho
-000023e0: 6473 2060 665f 6d61 7472 6978 5f69 6e70  ds `f_matrix_inp
-000023f0: 6c61 6365 6020 616e 6420 6065 5f6d 6174  lace` and `e_mat
-00002400: 7269 785f 696e 706c 6163 6560 2077 6869  rix_inplace` whi
-00002410: 6368 2070 6572 666f 726d 2060 665f 6d61  ch perform `f_ma
-00002420: 7472 6978 6020 616e 6420 6065 5f6d 6174  trix` and `e_mat
-00002430: 7269 7860 2063 6f6d 7075 7461 7469 6f6e  rix` computation
-00002440: 7320 696e 2d70 6c61 6365 2061 6e64 2061  s in-place and a
-00002450: 7265 2075 7365 6420 6279 2074 6865 206e  re used by the n
-00002460: 6577 2060 6365 6e74 6572 5f64 6973 7461  ew `center_dista
-00002470: 6e63 655f 6d61 7472 6978 6020 6d65 7468  nce_matrix` meth
-00002480: 6f64 2069 6e20 6073 6b62 696f 2e73 7461  od in `skbio.sta
-00002490: 7473 2e6f 7264 696e 6174 696f 6e60 2e0a  ts.ordination`..
-000024a0: 0a23 2323 2042 7567 2066 6978 6573 0a0a  .### Bug fixes..
-000024b0: 2323 2320 4465 7072 6563 6174 6564 2066  ### Deprecated f
-000024c0: 756e 6374 696f 6e61 6c69 7479 205b 7374  unctionality [st
-000024d0: 6162 6c65 5d0a 0a23 2323 2044 6570 7265  able]..### Depre
-000024e0: 6361 7465 6420 6675 6e63 7469 6f6e 616c  cated functional
-000024f0: 6974 7920 5b65 7870 6572 696d 656e 7461  ity [experimenta
-00002500: 6c5d 0a0a 2323 2320 4d69 7363 656c 6c61  l]..### Miscella
-00002510: 6e65 6f75 730a 0a23 2320 5665 7273 696f  neous..## Versio
-00002520: 6e20 302e 352e 3320 2832 3031 382d 3038  n 0.5.3 (2018-08
-00002530: 2d30 3729 0a0a 2323 2320 4665 6174 7572  -07)..### Featur
-00002540: 6573 0a2a 2041 6464 6564 2060 756e 7061  es.* Added `unpa
-00002550: 636b 6020 616e 6420 6075 6e70 6163 6b5f  ck` and `unpack_
-00002560: 6279 5f66 756e 6360 206d 6574 686f 6473  by_func` methods
-00002570: 2074 6f20 6073 6b62 696f 2e74 7265 652e   to `skbio.tree.
-00002580: 5472 6565 4e6f 6465 6020 746f 2075 6e70  TreeNode` to unp
-00002590: 6163 6b20 6f6e 6520 6f72 206d 756c 7469  ack one or multi
-000025a0: 706c 6520 696e 7465 726e 616c 206e 6f64  ple internal nod
-000025b0: 6573 2e20 5468 6520 6075 6e70 6163 6b60  es. The `unpack`
-000025c0: 206f 7065 7261 7469 6f6e 2072 656d 6f76   operation remov
-000025d0: 6573 2061 6e20 696e 7465 726e 616c 206e  es an internal n
-000025e0: 6f64 6520 616e 6420 7265 6772 6166 7473  ode and regrafts
-000025f0: 2069 7473 2063 6869 6c64 7265 6e20 746f   its children to
-00002600: 2069 7473 2070 6172 656e 7420 7768 696c   its parent whil
-00002610: 6520 7265 7461 696e 696e 6720 7468 6520  e retaining the 
-00002620: 6f76 6572 616c 6c20 6c65 6e67 7468 2e20  overall length. 
-00002630: 285b 2331 3537 325d 2868 7474 7073 3a2f  ([#1572](https:/
-00002640: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
-00002650: 6f72 652f 7363 696b 6974 2d62 696f 2f70  ore/scikit-bio/p
-00002660: 756c 6c2f 3135 3732 2929 0a2a 2041 6464  ull/1572)).* Add
-00002670: 6564 2060 7375 7070 6f72 7460 2074 6f20  ed `support` to 
-00002680: 6073 6b62 696f 2e74 7265 652e 5472 6565  `skbio.tree.Tree
-00002690: 4e6f 6465 6020 746f 2072 6574 7572 6e20  Node` to return 
-000026a0: 7468 6520 7375 7070 6f72 7420 7661 6c75  the support valu
-000026b0: 6520 6f66 2061 206e 6f64 652e 0a2a 2041  e of a node..* A
-000026c0: 6464 6564 2060 7065 726d 6469 7370 6020  dded `permdisp` 
-000026d0: 746f 2060 736b 6269 6f2e 7374 6174 732e  to `skbio.stats.
-000026e0: 6469 7374 616e 6365 6020 746f 2074 6573  distance` to tes
-000026f0: 7420 666f 7220 7468 6520 686f 6d6f 6765  t for the homoge
-00002700: 6e69 6574 7920 6f66 2067 726f 7570 732e  niety of groups.
-00002710: 2028 5b23 3132 3238 5d28 6874 7470 733a   ([#1228](https:
-00002720: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
-00002730: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
-00002740: 6973 7375 6573 2f31 3232 3829 292e 0a0a  issues/1228))...
-00002750: 2a20 4164 6465 6420 6070 636f 615f 6269  * Added `pcoa_bi
-00002760: 706c 6f74 6020 746f 2060 736b 6269 6f2e  plot` to `skbio.
-00002770: 7374 6174 732e 6f72 6469 6e61 7469 6f6e  stats.ordination
-00002780: 6020 746f 2070 726f 6a65 6374 2064 6573  ` to project des
-00002790: 6372 6970 746f 7273 2069 6e74 6f20 6120  criptors into a 
-000027a0: 5043 6f41 2070 6c6f 742e 0a0a 2a20 4669  PCoA plot...* Fi
-000027b0: 7865 6420 7061 6e64 6173 2074 6f20 302e  xed pandas to 0.
-000027c0: 3232 2e30 2064 7565 2074 6f20 7468 6973  22.0 due to this
-000027d0: 3a20 6874 7470 733a 2f2f 6769 7468 7562  : https://github
-000027e0: 2e63 6f6d 2f70 616e 6461 732d 6465 762f  .com/pandas-dev/
-000027f0: 7061 6e64 6173 2f69 7373 7565 732f 3230  pandas/issues/20
-00002800: 3532 370a 0a23 2323 2042 6163 6b77 6172  527..### Backwar
-00002810: 642d 696e 636f 6d70 6174 6962 6c65 2063  d-incompatible c
-00002820: 6861 6e67 6573 205b 7374 6162 6c65 5d0a  hanges [stable].
-00002830: 0a23 2323 2042 6163 6b77 6172 642d 696e  .### Backward-in
-00002840: 636f 6d70 6174 6962 6c65 2063 6861 6e67  compatible chang
-00002850: 6573 205b 6578 7065 7269 6d65 6e74 616c  es [experimental
-00002860: 5d0a 0a23 2323 2050 6572 666f 726d 616e  ]..### Performan
-00002870: 6365 2065 6e68 616e 6365 6d65 6e74 730a  ce enhancements.
-00002880: 0a23 2323 2042 7567 2066 6978 6573 0a0a  .### Bug fixes..
-00002890: 2a20 5265 6c61 7869 6e67 2074 7970 6520  * Relaxing type 
-000028a0: 6368 6563 6b69 6e67 2069 6e20 6469 7665  checking in dive
-000028b0: 7273 6974 7920 6361 6c63 756c 6174 696f  rsity calculatio
-000028c0: 6e73 2e20 2028 5b23 3135 3833 5d28 6874  ns.  ([#1583](ht
-000028d0: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-000028e0: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
-000028f0: 6269 6f2f 6973 7375 6573 2f31 3538 3329  bio/issues/1583)
-00002900: 292e 0a0a 2323 2320 4465 7072 6563 6174  )...### Deprecat
-00002910: 6564 2066 756e 6374 696f 6e61 6c69 7479  ed functionality
-00002920: 205b 7374 6162 6c65 5d0a 0a23 2323 2044   [stable]..### D
+00001fa0: 7461 6c5d 0a0a 2a20 4368 616e 6765 6420  tal]..* Changed 
+00001fb0: 6073 6b62 696f 2e74 7265 652e 5472 6565  `skbio.tree.Tree
+00001fc0: 4e6f 6465 2e73 7570 706f 7274 6020 6672  Node.support` fr
+00001fd0: 6f6d 2061 206d 6574 686f 6420 746f 2061  om a method to a
+00001fe0: 2070 726f 7065 7274 792e 0a2a 2041 6464   property..* Add
+00001ff0: 6564 2060 6173 7369 676e 5f73 7570 706f  ed `assign_suppo
+00002000: 7274 7360 206d 6574 686f 6420 746f 2060  rts` method to `
+00002010: 736b 6269 6f2e 7472 6565 2e54 7265 654e  skbio.tree.TreeN
+00002020: 6f64 6560 2074 6f20 6578 7472 6163 7420  ode` to extract 
+00002030: 6272 616e 6368 2073 7570 706f 7274 2076  branch support v
+00002040: 616c 7565 7320 6672 6f6d 206e 6f64 6520  alues from node 
+00002050: 6c61 6265 6c73 2e0a 2a20 4d6f 6469 6669  labels..* Modifi
+00002060: 6564 2074 6865 2077 6179 2061 206e 6f64  ed the way a nod
+00002070: 6527 7320 6c61 6265 6c20 6973 2070 7269  e's label is pri
+00002080: 6e74 6564 3a20 6073 7570 706f 7274 3a6e  nted: `support:n
+00002090: 616d 6560 2069 6620 626f 7468 2065 7869  ame` if both exi
+000020a0: 7374 2c20 6f72 2060 7375 7070 6f72 7460  st, or `support`
+000020b0: 206f 7220 606e 616d 6560 2069 6620 6569   or `name` if ei
+000020c0: 7468 6572 2065 7869 7374 732e 0a0a 2323  ther exists...##
+000020d0: 2320 5065 7266 6f72 6d61 6e63 6520 656e  # Performance en
+000020e0: 6861 6e63 656d 656e 7473 0a0a 2323 2320  hancements..### 
+000020f0: 4275 6720 6669 7865 730a 2a20 5265 7175  Bug fixes.* Requ
+00002100: 6972 6520 6053 7068 696e 7820 3c3d 2033  ire `Sphinx <= 3
+00002110: 2e30 602e 204e 6577 6572 2053 7068 696e  .0`. Newer Sphin
+00002120: 7820 7665 7273 696f 6e73 2063 6175 7365  x versions cause
+00002130: 6420 6275 696c 6420 6572 726f 7273 2e20  d build errors. 
+00002140: 5b23 3137 3139 5d28 6874 7470 733a 2f2f  [#1719](https://
+00002150: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
+00002160: 7265 2f73 6369 6b69 742d 6269 6f2f 7075  re/scikit-bio/pu
+00002170: 6c6c 2f31 3731 3929 0a0a 2a20 2a20 6073  ll/1719)..* * `s
+00002180: 6b62 696f 2e73 7461 7473 2e6f 7264 696e  kbio.stats.ordin
+00002190: 6174 696f 6e60 2074 6573 7473 2068 6176  ation` tests hav
+000021a0: 6520 6265 656e 2072 656c 6178 6564 2e20  e been relaxed. 
+000021b0: 285b 2331 3731 335d 2868 7474 7073 3a2f  ([#1713](https:/
+000021c0: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
+000021d0: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
+000021e0: 7373 7565 732f 3137 3133 2929 0a0a 2a20  ssues/1713))..* 
+000021f0: 4669 7865 7320 6275 696c 6420 6572 726f  Fixes build erro
+00002200: 7273 2066 6f72 206e 6577 6572 2076 6572  rs for newer ver
+00002210: 7369 6f6e 7320 6f66 204e 756d 5079 2c20  sions of NumPy, 
+00002220: 5061 6e64 6173 2c20 616e 6420 5363 6950  Pandas, and SciP
+00002230: 792e 0a0a 2a20 436f 7272 6563 7465 6420  y...* Corrected 
+00002240: 6120 6372 6974 6963 6961 6c20 6275 6720  a criticial bug 
+00002250: 696e 2060 736b 6269 6f2e 616c 6967 6e6d  in `skbio.alignm
+00002260: 656e 742e 5374 7269 7065 6453 6d69 7468  ent.StripedSmith
+00002270: 5761 7465 726d 616e 602f 6073 6b62 696f  Waterman`/`skbio
+00002280: 2e61 6c69 676e 6d65 6e74 2e6c 6f63 616c  .alignment.local
+00002290: 5f70 6169 7277 6973 655f 616c 6967 6e5f  _pairwise_align_
+000022a0: 7373 7760 2077 6869 6368 2077 6f75 6c64  ssw` which would
+000022b0: 2063 6175 7365 2074 6865 2066 6f72 6d61   cause the forma
+000022c0: 7474 696e 6720 6f66 2074 6865 2061 6c69  tting of the ali
+000022d0: 676e 6564 2073 6571 7565 6e63 6573 2074  gned sequences t
+000022e0: 6f20 6d69 7370 6c61 6365 2067 6170 2063  o misplace gap c
+000022f0: 6861 7261 6374 6572 7320 6279 2074 6865  haracters by the
+00002300: 206e 756d 6265 7220 6f66 2067 6170 2063   number of gap c
+00002310: 6861 7261 6374 6572 7320 7072 6573 656e  haracters presen
+00002320: 7420 696e 2074 6865 206f 7070 6f73 696e  t in the opposin
+00002330: 6720 616c 6967 6e65 6420 7365 7175 656e  g aligned sequen
+00002340: 6365 2075 7020 746f 2074 6861 7420 706f  ce up to that po
+00002350: 696e 742e 2054 6869 7320 7761 7320 6361  int. This was ca
+00002360: 7573 6564 2062 7920 6120 6661 756c 7479  used by a faulty
+00002370: 2069 6d70 6c65 6d65 6e74 6174 696f 6e20   implementation 
+00002380: 6f66 2043 4947 4152 2073 7472 696e 6720  of CIGAR string 
+00002390: 7061 7273 696e 672c 2073 6565 205b 2331  parsing, see [#1
+000023a0: 3637 395d 2868 7474 7073 3a2f 2f67 6974  679](https://git
+000023b0: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+000023c0: 7363 696b 6974 2d62 696f 2f70 756c 6c2f  scikit-bio/pull/
+000023d0: 3136 3739 2920 666f 7220 6675 6c6c 2064  1679) for full d
+000023e0: 6574 6169 6c73 2e0a 0a2a 2046 6978 6573  etails...* Fixes
+000023f0: 2062 7569 6c64 2065 7272 6f72 7320 666f   build errors fo
+00002400: 7220 6e65 7765 7220 7665 7273 696f 6e73  r newer versions
+00002410: 206f 6620 4e75 6d50 792c 2050 616e 6461   of NumPy, Panda
+00002420: 732c 2061 6e64 2053 6369 5079 2e0a 0a2a  s, and SciPy...*
+00002430: 2043 6f72 7265 6374 6564 2061 2063 7269   Corrected a cri
+00002440: 7469 6369 616c 2062 7567 2069 6e20 6073  ticial bug in `s
+00002450: 6b62 696f 2e61 6c69 676e 6d65 6e74 2e53  kbio.alignment.S
+00002460: 7472 6970 6564 536d 6974 6857 6174 6572  tripedSmithWater
+00002470: 6d61 6e60 2f60 736b 6269 6f2e 616c 6967  man`/`skbio.alig
+00002480: 6e6d 656e 742e 6c6f 6361 6c5f 7061 6972  nment.local_pair
+00002490: 7769 7365 5f61 6c69 676e 5f73 7377 6020  wise_align_ssw` 
+000024a0: 7768 6963 6820 776f 756c 6420 6361 7573  which would caus
+000024b0: 6520 7468 6520 666f 726d 6174 7469 6e67  e the formatting
+000024c0: 206f 6620 7468 6520 616c 6967 6e65 6420   of the aligned 
+000024d0: 7365 7175 656e 6365 7320 746f 206d 6973  sequences to mis
+000024e0: 706c 6163 6520 6761 7020 6368 6172 6163  place gap charac
+000024f0: 7465 7273 2062 7920 7468 6520 6e75 6d62  ters by the numb
+00002500: 6572 206f 6620 6761 7020 6368 6172 6163  er of gap charac
+00002510: 7465 7273 2070 7265 7365 6e74 2069 6e20  ters present in 
+00002520: 7468 6520 6f70 706f 7369 6e67 2061 6c69  the opposing ali
+00002530: 676e 6564 2073 6571 7565 6e63 6520 7570  gned sequence up
+00002540: 2074 6f20 7468 6174 2070 6f69 6e74 2e20   to that point. 
+00002550: 5468 6973 2077 6173 2063 6175 7365 6420  This was caused 
+00002560: 6279 2061 2066 6175 6c74 7920 696d 706c  by a faulty impl
+00002570: 656d 656e 7461 7469 6f6e 206f 6620 4349  ementation of CI
+00002580: 4741 5220 7374 7269 6e67 2070 6172 7369  GAR string parsi
+00002590: 6e67 2c20 7365 6520 5b23 3136 3739 5d28  ng, see [#1679](
+000025a0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+000025b0: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+000025c0: 742d 6269 6f2f 7075 6c6c 2f31 3637 3929  t-bio/pull/1679)
+000025d0: 2066 6f72 2066 756c 6c20 6465 7461 696c   for full detail
+000025e0: 732e 0a0a 2323 2320 4465 7072 6563 6174  s...### Deprecat
+000025f0: 6564 2066 756e 6374 696f 6e61 6c69 7479  ed functionality
+00002600: 205b 7374 6162 6c65 5d0a 0a23 2323 2044   [stable]..### D
+00002610: 6570 7265 6361 7465 6420 6675 6e63 7469  eprecated functi
+00002620: 6f6e 616c 6974 7920 5b65 7870 6572 696d  onality [experim
+00002630: 656e 7461 6c5d 0a0a 2323 2320 4d69 7363  ental]..### Misc
+00002640: 656c 6c61 6e65 6f75 730a 0a2a 2060 736b  ellaneous..* `sk
+00002650: 6269 6f2e 6469 7665 7273 6974 792e 6265  bio.diversity.be
+00002660: 7461 5f64 6976 6572 7369 7479 6020 6e6f  ta_diversity` no
+00002670: 7720 6163 6365 7074 7320 6120 7061 6e64  w accepts a pand
+00002680: 6173 2044 6174 6146 7261 6d65 2061 7320  as DataFrame as 
+00002690: 696e 7075 742e 0a0a 2a20 4176 6f69 6420  input...* Avoid 
+000026a0: 7061 6e64 6173 2031 2e30 2e30 2069 6d70  pandas 1.0.0 imp
+000026b0: 6f72 7420 7761 726e 696e 6720 285b 2331  ort warning ([#1
+000026c0: 3638 385d 2868 7474 7073 3a2f 2f67 6974  688](https://git
+000026d0: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+000026e0: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
+000026f0: 732f 3136 3838 2929 0a0a 2a20 4164 6465  s/1688))..* Adde
+00002700: 6420 7375 7070 6f72 7420 666f 7220 5079  d support for Py
+00002710: 7468 6f6e 2033 2e38 2061 6e64 2064 726f  thon 3.8 and dro
+00002720: 7070 6564 2073 7570 706f 7274 2066 6f72  pped support for
+00002730: 2050 7974 686f 6e20 332e 352e 0a0a 2a20   Python 3.5...* 
+00002740: 5468 6973 2076 6572 7369 6f6e 206e 6f77  This version now
+00002750: 2064 6570 656e 6473 206f 6e20 6073 6369   depends on `sci
+00002760: 7079 203e 3d20 312e 3360 2061 6e64 2060  py >= 1.3` and `
+00002770: 7061 6e64 6173 203e 3d20 312e 3060 2e0a  pandas >= 1.0`..
+00002780: 0a23 2320 5665 7273 696f 6e20 302e 352e  .## Version 0.5.
+00002790: 3520 2832 3031 382d 3132 2d31 3029 0a0a  5 (2018-12-10)..
+000027a0: 2323 2320 4665 6174 7572 6573 0a0a 2a20  ### Features..* 
+000027b0: 6073 6b62 696f 2e73 7461 7473 2e63 6f6d  `skbio.stats.com
+000027c0: 706f 7369 7469 6f6e 6020 6e6f 7720 6861  position` now ha
+000027d0: 7320 6d65 7468 6f64 7320 746f 2063 6f6d  s methods to com
+000027e0: 7075 7465 2061 6464 6974 6976 6520 6c6f  pute additive lo
+000027f0: 672d 7261 7469 6f20 7472 616e 7366 6f72  g-ratio transfor
+00002800: 6d61 7469 6f6e 2061 6e64 2069 6e76 6572  mation and inver
+00002810: 7365 2061 6464 6974 6976 6520 6c6f 672d  se additive log-
+00002820: 7261 7469 6f20 7472 616e 7366 6f72 6d61  ratio transforma
+00002830: 7469 6f6e 2028 6061 6c72 602c 2060 616c  tion (`alr`, `al
+00002840: 725f 696e 7660 2920 6173 2077 656c 6c20  r_inv`) as well 
+00002850: 6173 2061 206d 6574 686f 6420 746f 2062  as a method to b
+00002860: 7569 6c64 2061 2062 6173 6973 2066 726f  uild a basis fro
+00002870: 6d20 6120 7365 7175 656e 7469 616c 2062  m a sequential b
+00002880: 696e 6172 7920 7061 7274 6974 696f 6e20  inary partition 
+00002890: 2860 7362 705f 6261 7369 7360 292e 0a0a  (`sbp_basis`)...
+000028a0: 2323 2320 4261 636b 7761 7264 2d69 6e63  ### Backward-inc
+000028b0: 6f6d 7061 7469 626c 6520 6368 616e 6765  ompatible change
+000028c0: 7320 5b73 7461 626c 655d 0a0a 2323 2320  s [stable]..### 
+000028d0: 4261 636b 7761 7264 2d69 6e63 6f6d 7061  Backward-incompa
+000028e0: 7469 626c 6520 6368 616e 6765 7320 5b65  tible changes [e
+000028f0: 7870 6572 696d 656e 7461 6c5d 0a0a 2323  xperimental]..##
+00002900: 2320 5065 7266 6f72 6d61 6e63 6520 656e  # Performance en
+00002910: 6861 6e63 656d 656e 7473 0a0a 2323 2320  hancements..### 
+00002920: 4275 6720 6669 7865 730a 0a23 2323 2044  Bug fixes..### D
 00002930: 6570 7265 6361 7465 6420 6675 6e63 7469  eprecated functi
-00002940: 6f6e 616c 6974 7920 5b65 7870 6572 696d  onality [experim
-00002950: 656e 7461 6c5d 0a0a 2323 2320 4d69 7363  ental]..### Misc
-00002960: 656c 6c61 6e65 6f75 730a 0a0a 2323 2056  ellaneous...## V
-00002970: 6572 7369 6f6e 2030 2e35 2e32 2028 3230  ersion 0.5.2 (20
-00002980: 3138 2d30 342d 3138 290a 0a23 2323 2046  18-04-18)..### F
-00002990: 6561 7475 7265 730a 2a20 4164 6465 6420  eatures.* Added 
-000029a0: 6060 736b 6269 6f2e 696f 2e66 6f72 6d61  ``skbio.io.forma
-000029b0: 742e 656d 626c 6060 2066 6f72 2072 6561  t.embl`` for rea
-000029c0: 6469 6e67 2061 6e64 2077 7269 7469 6e67  ding and writing
-000029d0: 2045 4d42 4c20 6669 6c65 7320 666f 7220   EMBL files for 
-000029e0: 6060 444e 4160 602c 2060 6052 4e41 6060  ``DNA``, ``RNA``
-000029f0: 2061 6e64 2060 6053 6571 7565 6e63 6560   and ``Sequence`
-00002a00: 6020 636c 6173 7365 732e 0a0a 2a20 5265  ` classes...* Re
-00002a10: 6d6f 7669 6e67 2056 616c 7565 4572 726f  moving ValueErro
-00002a20: 7220 6368 6563 6b20 696e 2060 736b 6269  r check in `skbi
-00002a30: 6f2e 7374 6174 732e 5f73 7562 7361 6d70  o.stats._subsamp
-00002a40: 6c65 2e73 7562 7361 6d70 6c65 5f63 6f75  le.subsample_cou
-00002a50: 6e74 7360 2077 6865 6e20 6072 6570 6c61  nts` when `repla
-00002a60: 6365 3d54 7275 6560 2061 6e64 2060 6e60  ce=True` and `n`
-00002a70: 2069 7320 6772 6561 7465 7220 7468 616e   is greater than
-00002a80: 2074 6865 206e 756d 6265 7220 6f66 2069   the number of i
-00002a90: 7465 6d73 2069 6e20 636f 756e 7473 2e20  tems in counts. 
-00002aa0: 205b 2331 3532 375d 2868 7474 7073 3a2f   [#1527](https:/
-00002ab0: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
-00002ac0: 6f72 652f 7363 696b 6974 2d62 696f 2f70  ore/scikit-bio/p
-00002ad0: 756c 6c2f 3135 3237 290a 0a2a 2041 6464  ull/1527)..* Add
-00002ae0: 6564 2060 6073 6b62 696f 2e69 6f2e 666f  ed ``skbio.io.fo
-00002af0: 726d 6174 2e67 6666 3360 6020 666f 7220  rmat.gff3`` for 
-00002b00: 7265 6164 696e 6720 616e 6420 7772 6974  reading and writ
-00002b10: 696e 6720 4746 4633 2066 696c 6573 2066  ing GFF3 files f
-00002b20: 6f72 2060 6044 4e41 6060 2c20 6060 5365  or ``DNA``, ``Se
-00002b30: 7175 656e 6365 6060 2c20 616e 6420 6060  quence``, and ``
-00002b40: 496e 7465 7276 616c 4d65 7461 6461 7461  IntervalMetadata
-00002b50: 6060 2063 6c61 7373 6573 2e20 285b 2331  `` classes. ([#1
-00002b60: 3435 305d 2868 7474 7073 3a2f 2f67 6974  450](https://git
-00002b70: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
-00002b80: 7363 696b 6974 2d62 696f 2f70 756c 6c2f  scikit-bio/pull/
-00002b90: 3134 3530 2929 0a0a 2a20 6073 6b62 696f  1450))..* `skbio
-00002ba0: 2e6d 6574 6164 6174 612e 496e 7465 7276  .metadata.Interv
-00002bb0: 616c 4d65 7461 6461 7461 6020 636f 6e73  alMetadata` cons
-00002bc0: 7472 7563 746f 7220 6861 7320 6120 6e65  tructor has a ne
-00002bd0: 7720 6b65 7977 6f72 6420 6172 6775 6d65  w keyword argume
-00002be0: 6e74 2c20 6063 6f70 795f 6672 6f6d 602c  nt, `copy_from`,
-00002bf0: 2066 6f72 2063 7265 6174 696e 6720 616e   for creating an
-00002c00: 2060 496e 7465 7276 616c 4d65 7461 6461   `IntervalMetada
-00002c10: 7461 6020 6f62 6a65 6374 2066 726f 6d20  ta` object from 
-00002c20: 616e 2065 7869 7374 696e 6720 6049 6e74  an existing `Int
-00002c30: 6572 7661 6c4d 6574 6164 6174 6160 206f  ervalMetadata` o
-00002c40: 626a 6563 7420 7769 7468 2073 7065 6369  bject with speci
-00002c50: 6669 6564 2060 7570 7065 725f 626f 756e  fied `upper_boun
-00002c60: 6460 2e0a 0a2a 2060 736b 6269 6f2e 6d65  d`...* `skbio.me
-00002c70: 7461 6461 7461 2e49 6e74 6572 7661 6c4d  tadata.IntervalM
-00002c80: 6574 6164 6174 6160 2063 6f6e 7374 7275  etadata` constru
-00002c90: 6374 6f72 2061 6c6c 6f77 7320 604e 6f6e  ctor allows `Non
-00002ca0: 6560 2061 7320 6120 7661 6c69 6420 7661  e` as a valid va
-00002cb0: 6c75 6520 666f 7220 6075 7070 6572 5f62  lue for `upper_b
-00002cc0: 6f75 6e64 602e 2041 6e20 6075 7070 6572  ound`. An `upper
-00002cd0: 5f62 6f75 6e64 6020 6f66 2060 4e6f 6e65  _bound` of `None
-00002ce0: 6020 6d65 616e 7320 7468 6174 2074 6865  ` means that the
-00002cf0: 2060 496e 7465 7276 616c 4d65 7461 6461   `IntervalMetada
-00002d00: 7461 6020 6f62 6a65 6374 2068 6173 206e  ta` object has n
-00002d10: 6f20 7570 7065 7220 626f 756e 642e 0a0a  o upper bound...
-00002d20: 2a20 6073 6b62 696f 2e6d 6574 6164 6174  * `skbio.metadat
-00002d30: 612e 496e 7465 7276 616c 4d65 7461 6461  a.IntervalMetada
-00002d40: 7461 2e64 726f 7060 2068 6173 2061 206e  ta.drop` has a n
-00002d50: 6577 2062 6f6f 6c65 616e 2070 6172 616d  ew boolean param
-00002d60: 6574 6572 2060 6e65 6761 7465 6020 746f  eter `negate` to
-00002d70: 2069 6e64 6963 6174 6520 7768 6574 6865   indicate whethe
-00002d80: 7220 746f 2064 726f 7020 6f72 206b 6565  r to drop or kee
-00002d90: 7020 7468 6520 7370 6563 6966 6965 6420  p the specified 
-00002da0: 6049 6e74 6572 7661 6c60 206f 626a 6563  `Interval` objec
-00002db0: 7473 2e0a 0a23 2323 2042 6163 6b77 6172  ts...### Backwar
-00002dc0: 642d 696e 636f 6d70 6174 6962 6c65 2063  d-incompatible c
-00002dd0: 6861 6e67 6573 205b 7374 6162 6c65 5d0a  hanges [stable].
-00002de0: 0a23 2323 2042 6163 6b77 6172 642d 696e  .### Backward-in
-00002df0: 636f 6d70 6174 6962 6c65 2063 6861 6e67  compatible chang
-00002e00: 6573 205b 6578 7065 7269 6d65 6e74 616c  es [experimental
-00002e10: 5d0a 0a23 2323 2050 6572 666f 726d 616e  ]..### Performan
-00002e20: 6365 2065 6e68 616e 6365 6d65 6e74 730a  ce enhancements.
-00002e30: 2a20 6073 6b62 696f 2e74 7265 652e 6e6a  * `skbio.tree.nj
-00002e40: 6020 7761 6c6c 2d63 6c6f 636b 2072 756e  ` wall-clock run
-00002e50: 7469 6d65 2077 6173 2064 6563 7265 6173  time was decreas
-00002e60: 6564 2062 7920 3939 2520 666f 7220 6120  ed by 99% for a 
-00002e70: 3530 3078 3530 3020 6469 7374 616e 6365  500x500 distance
-00002e80: 206d 6174 7269 7820 616e 6420 3933 2520   matrix and 93% 
-00002e90: 666f 7220 6120 3130 3078 3130 3020 6469  for a 100x100 di
-00002ea0: 7374 616e 6365 206d 6174 7269 782e 2028  stance matrix. (
-00002eb0: 5b23 3135 3132 5d28 6874 7470 733a 2f2f  [#1512](https://
-00002ec0: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-00002ed0: 7265 2f73 6369 6b69 742d 6269 6f2f 7075  re/scikit-bio/pu
-00002ee0: 6c6c 2f31 3531 3229 2c20 5b23 3135 3133  ll/1512), [#1513
-00002ef0: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
-00002f00: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
-00002f10: 6b69 742d 6269 6f2f 7075 6c6c 2f31 3531  kit-bio/pull/151
-00002f20: 3329 290a 0a23 2323 2042 7567 2066 6978  3))..### Bug fix
-00002f30: 6573 0a2a 2054 6865 2060 696e 636c 7564  es.* The `includ
-00002f40: 655f 7365 6c66 6020 7061 7261 6d65 7465  e_self` paramete
-00002f50: 7220 7761 7320 6e6f 7420 6265 696e 6720  r was not being 
-00002f60: 686f 6e6f 7265 6420 696e 2060 736b 6269  honored in `skbi
-00002f70: 6f2e 5472 6565 4e6f 6465 2e74 6970 7360  o.TreeNode.tips`
-00002f80: 2e20 5468 6520 7363 6f70 6520 6f66 2074  . The scope of t
-00002f90: 6869 7320 6275 6720 7761 7320 7468 6174  his bug was that
-00002fa0: 2069 6620 6054 7265 654e 6f64 652e 7469   if `TreeNode.ti
-00002fb0: 7073 6020 7761 7320 6361 6c6c 6564 206f  ps` was called o
-00002fc0: 6e20 6120 7469 702c 2069 7420 776f 756c  n a tip, it woul
-00002fd0: 6420 616c 7761 7973 2072 6573 756c 7420  d always result 
-00002fe0: 696e 2061 6e20 656d 7074 7920 606c 6973  in an empty `lis
-00002ff0: 7460 2077 6865 6e20 756e 726f 6c6c 6564  t` when unrolled
-00003000: 2e0a 0a2a 2049 6e20 6073 6b62 696f 2e73  ...* In `skbio.s
-00003010: 7461 7473 2e6f 7264 696e 6174 696f 6e2e  tats.ordination.
-00003020: 6361 602c 2060 7072 6f70 6f72 7469 6f6e  ca`, `proportion
-00003030: 5f65 7870 6c61 696e 6564 6020 7761 7320  _explained` was 
-00003040: 6d69 7373 696e 6720 696e 2074 6865 2072  missing in the r
-00003050: 6574 7572 6e65 6420 604f 7264 696e 6174  eturned `Ordinat
-00003060: 696f 6e52 6573 756c 7473 6020 6f62 6a65  ionResults` obje
-00003070: 6374 2e20 285b 2331 3334 355d 2868 7474  ct. ([#1345](htt
-00003080: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00003090: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-000030a0: 696f 2f69 7373 7565 732f 3133 3435 2929  io/issues/1345))
-000030b0: 0a0a 2a20 6073 6b62 696f 2e64 6976 6572  ..* `skbio.diver
-000030c0: 7369 7479 2e62 6574 615f 6469 7665 7273  sity.beta_divers
-000030d0: 6974 7960 206e 6f77 2068 616e 646c 6573  ity` now handles
-000030e0: 2071 7561 6c69 7461 7469 7665 206d 6574   qualitative met
-000030f0: 7269 6373 2061 7320 6578 7065 6374 6564  rics as expected
-00003100: 2073 7563 6820 7468 6174 2060 6265 7461   such that `beta
-00003110: 5f64 6976 6572 7369 7479 2827 6a61 6363  _diversity('jacc
-00003120: 6172 6427 2c20 6d61 7429 203d 3d20 6265  ard', mat) == be
-00003130: 7461 5f64 6976 6572 7369 7479 2827 6a61  ta_diversity('ja
-00003140: 6363 6172 6427 2c20 6d61 7420 3e20 3029  ccard', mat > 0)
-00003150: 602e 2050 6c65 6173 6520 7365 6520 5b23  `. Please see [#
-00003160: 3135 3439 5d28 6874 7470 733a 2f2f 6769  1549](https://gi
-00003170: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
-00003180: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
-00003190: 6573 2f31 3534 3929 2066 6f72 2066 7572  es/1549) for fur
-000031a0: 7468 6572 2064 6574 6169 6c2e 0a0a 2a20  ther detail...* 
-000031b0: 6073 6b62 696f 2e73 7461 7473 2e6f 7264  `skbio.stats.ord
-000031c0: 696e 6174 696f 6e2e 7264 6160 2054 6865  ination.rda` The
-000031d0: 206f 6363 6173 696f 6e61 6c20 636f 6c75   occasional colu
-000031e0: 6d6e 206d 6973 6d61 7463 6820 696e 206f  mn mismatch in o
-000031f0: 7574 7075 7420 6062 6970 6c6f 745f 7363  utput `biplot_sc
-00003200: 6f72 6573 6020 6973 2066 6978 6564 2028  ores` is fixed (
-00003210: 5b23 3135 3139 5d28 6874 7470 733a 2f2f  [#1519](https://
-00003220: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-00003230: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
-00003240: 7375 6573 2f31 3531 3929 292e 0a0a 2323  sues/1519))...##
-00003250: 2320 4465 7072 6563 6174 6564 2066 756e  # Deprecated fun
-00003260: 6374 696f 6e61 6c69 7479 205b 7374 6162  ctionality [stab
-00003270: 6c65 5d0a 0a23 2323 2044 6570 7265 6361  le]..### Depreca
-00003280: 7465 6420 6675 6e63 7469 6f6e 616c 6974  ted functionalit
-00003290: 7920 5b65 7870 6572 696d 656e 7461 6c5d  y [experimental]
-000032a0: 0a0a 2323 2320 4d69 7363 656c 6c61 6e65  ..### Miscellane
-000032b0: 6f75 730a 2a20 7363 696b 6974 2d62 696f  ous.* scikit-bio
-000032c0: 206e 6f77 2064 6570 656e 6473 206f 6e20   now depends on 
-000032d0: 7061 6e64 6173 203e 3d20 302e 3139 2e32  pandas >= 0.19.2
-000032e0: 2c20 616e 6420 6973 2063 6f6d 7061 7469  , and is compati
-000032f0: 626c 6520 7769 7468 206e 6577 6572 2070  ble with newer p
-00003300: 616e 6461 7320 7665 7273 696f 6e73 2028  andas versions (
-00003310: 652e 672e 2030 2e32 302e 3329 2074 6861  e.g. 0.20.3) tha
-00003320: 7420 7765 7265 2070 7265 7669 6f75 736c  t were previousl
-00003330: 7920 696e 636f 6d70 6174 6962 6c65 2e0a  y incompatible..
-00003340: 2a20 7363 696b 6974 2d62 696f 206e 6f77  * scikit-bio now
-00003350: 2064 6570 656e 6473 206f 6e20 606e 756d   depends on `num
-00003360: 7079 203e 3d20 312e 392e 322c 203c 2031  py >= 1.9.2, < 1
-00003370: 2e31 342e 3060 2066 6f72 2063 6f6d 7061  .14.0` for compa
-00003380: 7469 6269 6c69 7479 2077 6974 6820 5079  tibility with Py
-00003390: 7468 6f6e 2033 2e34 2c20 332e 352c 2061  thon 3.4, 3.5, a
-000033a0: 6e64 2033 2e36 2061 6e64 2074 6865 2061  nd 3.6 and the a
-000033b0: 7661 696c 6162 6c65 206e 756d 7079 2063  vailable numpy c
-000033c0: 6f6e 6461 2070 6163 6b61 6765 7320 696e  onda packages in
-000033d0: 2060 6465 6661 756c 7473 6020 616e 6420   `defaults` and 
-000033e0: 6063 6f6e 6461 2d66 6f72 6765 6020 6368  `conda-forge` ch
-000033f0: 616e 6e65 6c73 2e0a 2a20 6164 6465 6420  annels..* added 
-00003400: 7375 7070 6f72 7420 666f 7220 7275 6e6e  support for runn
-00003410: 696e 6720 7465 7374 7320 6672 6f6d 2060  ing tests from `
-00003420: 7365 7475 702e 7079 602e 2042 6f74 6820  setup.py`. Both 
-00003430: 6070 7974 686f 6e20 7365 7475 702e 7079  `python setup.py
-00003440: 206e 6f73 6574 6573 7473 6020 616e 6420   nosetests` and 
-00003450: 6070 7974 686f 6e20 7365 7475 702e 7079  `python setup.py
-00003460: 2074 6573 7460 2061 7265 206e 6f77 2073   test` are now s
-00003470: 7570 706f 7274 6564 2c20 686f 7765 7665  upported, howeve
-00003480: 7220 6070 7974 686f 6e20 7365 7475 702e  r `python setup.
-00003490: 7079 2074 6573 7460 2077 696c 6c20 6f6e  py test` will on
-000034a0: 6c79 2072 756e 2061 2073 7562 7365 7420  ly run a subset 
-000034b0: 6f66 2074 6865 2066 756c 6c20 7465 7374  of the full test
-000034c0: 2073 7569 7465 2e20 285b 2331 3334 315d   suite. ([#1341]
-000034d0: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-000034e0: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
-000034f0: 6974 2d62 696f 2f69 7373 7565 732f 3133  it-bio/issues/13
-00003500: 3431 2929 0a0a 2323 2056 6572 7369 6f6e  41))..## Version
-00003510: 2030 2e35 2e31 2028 3230 3136 2d31 312d   0.5.1 (2016-11-
-00003520: 3132 290a 0a23 2323 2046 6561 7475 7265  12)..### Feature
-00003530: 730a 2a20 4164 6465 6420 6049 6e74 6572  s.* Added `Inter
-00003540: 7661 6c4d 6574 6164 6174 6160 2061 6e64  valMetadata` and
-00003550: 2060 496e 7465 7276 616c 6020 636c 6173   `Interval` clas
-00003560: 7365 7320 696e 2060 736b 6269 6f2e 6d65  ses in `skbio.me
-00003570: 7461 6461 7461 6020 746f 2073 746f 7265  tadata` to store
-00003580: 2c20 7175 6572 792c 2061 6e64 206d 616e  , query, and man
-00003590: 6970 756c 6174 6520 696e 666f 726d 6174  ipulate informat
-000035a0: 696f 6e20 6f66 2061 2073 7562 2d72 6567  ion of a sub-reg
-000035b0: 696f 6e20 6f66 2061 2073 6571 7565 6e63  ion of a sequenc
-000035c0: 652e 2028 5b23 3134 3134 5d28 6874 7470  e. ([#1414](http
-000035d0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-000035e0: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-000035f0: 6f2f 6973 7375 6573 2f31 3431 3429 290a  o/issues/1414)).
-00003600: 2a20 6053 6571 7565 6e63 6560 2061 6e64  * `Sequence` and
-00003610: 2069 7473 2063 6869 6c64 2063 6c61 7373   its child class
-00003620: 6573 2028 696e 636c 7564 696e 6720 6047  es (including `G
-00003630: 7261 6d6d 6172 6564 5365 7175 656e 6365  rammaredSequence
-00003640: 602c 2060 524e 4160 2c20 6044 4e41 602c  `, `RNA`, `DNA`,
-00003650: 2060 5072 6f74 6569 6e60 2920 6e6f 7720   `Protein`) now 
-00003660: 6163 6365 7074 2060 496e 7465 7276 616c  accept `Interval
-00003670: 4d65 7461 6461 7461 6020 696e 2074 6865  Metadata` in the
-00003680: 6972 2063 6f6e 7374 7275 6374 6f72 2041  ir constructor A
-00003690: 5049 2e20 536f 6d65 206f 6620 7468 6569  PI. Some of thei
-000036a0: 7220 7265 6c65 7661 6e74 206d 6574 686f  r relevant metho
-000036b0: 6473 2061 7265 2061 6c73 6f20 7570 6461  ds are also upda
-000036c0: 7465 6420 6163 636f 7264 696e 676c 792e  ted accordingly.
-000036d0: 2028 5b23 3134 3330 5d28 6874 7470 733a   ([#1430](https:
-000036e0: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
-000036f0: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
-00003700: 7075 6c6c 2f31 3433 3029 290a 2a20 4765  pull/1430)).* Ge
-00003710: 6e42 616e 6b20 7061 7273 6572 206e 6f77  nBank parser now
-00003720: 2072 6561 6473 2061 6e64 2077 7269 7465   reads and write
-00003730: 7320 6053 6571 7565 6e63 6560 206f 7220  s `Sequence` or 
-00003740: 6974 7320 7375 6263 6c61 7373 206f 626a  its subclass obj
-00003750: 6563 7473 2077 6974 6820 6049 6e74 6572  ects with `Inter
-00003760: 7661 6c4d 6574 6164 6174 6160 2e20 285b  valMetadata`. ([
-00003770: 2331 3434 305d 2868 7474 7073 3a2f 2f67  #1440](https://g
-00003780: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
-00003790: 652f 7363 696b 6974 2d62 696f 2f70 756c  e/scikit-bio/pul
-000037a0: 6c2f 3134 3430 2929 0a2a 2060 4469 7373  l/1440)).* `Diss
-000037b0: 696d 696c 6172 6974 794d 6174 7269 7860  imilarityMatrix`
-000037c0: 206e 6f77 2068 6173 2061 206e 6577 2063   now has a new c
-000037d0: 6f6e 7374 7275 6374 6f72 206d 6574 686f  onstructor metho
-000037e0: 6420 6361 6c6c 6564 2060 6672 6f6d 5f69  d called `from_i
-000037f0: 7465 7261 626c 6560 2e20 285b 2331 3334  terable`. ([#134
-00003800: 335d 2868 7474 7073 3a2f 2f67 6974 6875  3](https://githu
-00003810: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
-00003820: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
-00003830: 3133 3433 2929 2e0a 2a20 6044 6973 7369  1343))..* `Dissi
-00003840: 6d69 6c61 7269 7479 4d61 7472 6978 6020  milarityMatrix` 
-00003850: 6e6f 7720 616c 6c6f 7773 206e 6f6e 2d68  now allows non-h
-00003860: 6f6c 6c6f 7720 6d61 7472 6963 6573 2e20  ollow matrices. 
-00003870: 285b 2331 3334 335d 2868 7474 7073 3a2f  ([#1343](https:/
-00003880: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
-00003890: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
-000038a0: 7373 7565 732f 3133 3433 2929 2e0a 2a20  ssues/1343))..* 
-000038b0: 6044 6973 7461 6e63 654d 6174 7269 782e  `DistanceMatrix.
-000038c0: 6672 6f6d 5f69 7465 7261 626c 6560 206e  from_iterable` n
-000038d0: 6f77 2061 6363 6570 7473 2061 2060 7661  ow accepts a `va
-000038e0: 6c69 6461 7465 3d54 7275 6560 2070 6172  lidate=True` par
-000038f0: 616d 6574 6572 2e20 285b 2331 3334 335d  ameter. ([#1343]
-00003900: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-00003910: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
-00003920: 6974 2d62 696f 2f69 7373 7565 732f 3133  it-bio/issues/13
-00003930: 3433 2929 2e0a 2a20 6060 4469 7374 616e  43))..* ``Distan
-00003940: 6365 4d61 7472 6978 6060 206e 6f77 2068  ceMatrix`` now h
-00003950: 6173 2061 206e 6577 206d 6574 686f 6420  as a new method 
-00003960: 6361 6c6c 6564 2060 6074 6f5f 7365 7269  called ``to_seri
-00003970: 6573 6060 2074 6f20 6372 6561 7465 2061  es`` to create a
-00003980: 2060 6070 616e 6461 732e 5365 7269 6573   ``pandas.Series
-00003990: 6060 2066 726f 6d20 6120 6060 4469 7374  `` from a ``Dist
-000039a0: 616e 6365 4d61 7472 6978 6060 2028 5b23  anceMatrix`` ([#
-000039b0: 3133 3937 5d28 6874 7470 733a 2f2f 6769  1397](https://gi
-000039c0: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
-000039d0: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
-000039e0: 6573 2f31 3339 3729 292e 0a2a 2041 6464  es/1397))..* Add
-000039f0: 6564 2070 6172 616c 6c65 6c20 6265 7461  ed parallel beta
-00003a00: 2064 6976 6572 7369 7479 2063 616c 6375   diversity calcu
-00003a10: 6c61 7469 6f6e 2073 7570 706f 7274 2076  lation support v
-00003a20: 6961 2060 736b 6269 6f2e 6469 7665 7273  ia `skbio.divers
-00003a30: 6974 792e 626c 6f63 6b5f 6265 7461 5f64  ity.block_beta_d
-00003a40: 6976 6572 7369 7479 602e 2054 6865 2069  iversity`. The i
-00003a50: 7373 7565 2061 6e64 2069 6465 6120 6973  ssue and idea is
-00003a60: 2064 6973 6375 7373 6564 2069 6e20 285b   discussed in ([
-00003a70: 2331 3138 315d 2868 7474 7073 3a2f 2f67  #1181](https://g
-00003a80: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
-00003a90: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
-00003aa0: 7565 732f 3131 3831 292c 2077 6869 6c65  ues/1181), while
-00003ab0: 2074 6865 2061 6374 7561 6c20 636f 6465   the actual code
-00003ac0: 2063 6861 6e67 6573 2061 7265 2069 6e20   changes are in 
-00003ad0: 5b23 3133 3532 5d28 6874 7470 733a 2f2f  [#1352](https://
-00003ae0: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-00003af0: 7265 2f73 6369 6b69 742d 6269 6f2f 7075  re/scikit-bio/pu
-00003b00: 6c6c 2f31 3335 3229 292e 0a0a 0a23 2323  ll/1352))....###
-00003b10: 2042 6163 6b77 6172 642d 696e 636f 6d70   Backward-incomp
-00003b20: 6174 6962 6c65 2063 6861 6e67 6573 205b  atible changes [
-00003b30: 7374 6162 6c65 5d0a 2a20 5468 6520 636f  stable].* The co
-00003b40: 6e73 7472 7563 746f 7220 4150 4920 666f  nstructor API fo
-00003b50: 7220 6053 6571 7565 6e63 6560 2061 6e64  r `Sequence` and
-00003b60: 2069 7473 2063 6869 6c64 2063 6c61 7373   its child class
-00003b70: 6573 2028 696e 636c 7564 696e 6720 6047  es (including `G
-00003b80: 7261 6d6d 6172 6564 5365 7175 656e 6365  rammaredSequence
-00003b90: 602c 2060 524e 4160 2c20 6044 4e41 602c  `, `RNA`, `DNA`,
-00003ba0: 2060 5072 6f74 6569 6e60 2920 6172 6520   `Protein`) are 
-00003bb0: 6368 616e 6765 6420 6672 6f6d 2060 2873  changed from `(s
-00003bc0: 6571 7565 6e63 652c 206d 6574 6164 6174  equence, metadat
-00003bd0: 613d 4e6f 6e65 2c20 706f 7369 7469 6f6e  a=None, position
-00003be0: 616c 5f6d 6574 6164 6174 613d 4e6f 6e65  al_metadata=None
-00003bf0: 2c20 6c6f 7765 7263 6173 653d 4661 6c73  , lowercase=Fals
-00003c00: 6529 6020 746f 2060 2873 6571 7565 6e63  e)` to `(sequenc
-00003c10: 652c 206d 6574 6164 6174 613d 4e6f 6e65  e, metadata=None
-00003c20: 2c20 706f 7369 7469 6f6e 616c 5f6d 6574  , positional_met
-00003c30: 6164 6174 613d 4e6f 6e65 2c20 696e 7465  adata=None, inte
-00003c40: 7276 616c 5f6d 6574 6164 6174 613d 4e6f  rval_metadata=No
-00003c50: 6e65 2c20 6c6f 7765 7263 6173 653d 4661  ne, lowercase=Fa
-00003c60: 6c73 6529 600a 0a20 2054 6865 2063 6861  lse)`..  The cha
-00003c70: 6e67 6573 2061 7265 206d 6164 6520 746f  nges are made to
-00003c80: 2061 6c6c 6f77 2074 6865 7365 2063 6c61   allow these cla
-00003c90: 7373 6573 2074 6f20 6164 6f70 7420 6049  sses to adopt `I
-00003ca0: 6e74 6572 7661 6c4d 6574 6164 6174 6160  ntervalMetadata`
-00003cb0: 206f 626a 6563 7420 666f 7220 696e 7465   object for inte
-00003cc0: 7276 616c 2066 6561 7475 7265 7320 6f6e  rval features on
-00003cd0: 2074 6865 2073 6571 7565 6e63 652e 2054   the sequence. T
-00003ce0: 6865 2060 696e 7465 7276 616c 5f6d 6574  he `interval_met
-00003cf0: 6164 6174 6160 2070 6172 616d 6574 6572  adata` parameter
-00003d00: 2069 7320 6164 6465 6420 696d 6564 6961   is added imedia
-00003d10: 7465 6c79 2061 6674 6572 2060 706f 7369  tely after `posi
-00003d20: 7469 6f6e 616c 5f6d 6574 6164 6174 6160  tional_metadata`
-00003d30: 2069 6e73 7465 6164 206f 6620 6170 7065   instead of appe
-00003d40: 6e64 6564 2074 6f20 7468 6520 656e 642c  nded to the end,
-00003d50: 2062 6563 6175 7365 2069 7420 6973 206d   because it is m
-00003d60: 6f72 6520 6e61 7475 7261 6c20 616e 6420  ore natural and 
-00003d70: 6c6f 6769 6361 6c20 616e 642c 206d 6f72  logical and, mor
-00003d80: 6520 696d 706f 7274 616e 746c 792c 2062  e importantly, b
-00003d90: 6563 6175 7365 2069 7420 6973 2075 6e6c  ecause it is unl
-00003da0: 696b 656c 7920 696e 2070 7261 6374 6963  ikely in practic
-00003db0: 6520 746f 2062 7265 616b 2075 7365 7220  e to break user 
-00003dc0: 636f 6465 2e20 4120 7573 6572 2773 2063  code. A user's c
-00003dd0: 6f64 6520 776f 756c 6420 6272 6561 6b20  ode would break 
-00003de0: 6f6e 6c79 2069 6620 7468 6579 2068 6164  only if they had
-00003df0: 2073 7570 706c 6965 6420 606d 6574 6164   supplied `metad
-00003e00: 6174 6160 2c20 6070 6f73 7469 6f6e 616c  ata`, `postional
-00003e10: 5f6d 6574 6164 6174 6160 2c20 616e 6420  _metadata`, and 
-00003e20: 606c 6f77 6572 6361 7365 6020 7061 7261  `lowercase` para
-00003e30: 6d65 7465 7273 2070 6f73 6974 696f 6e61  meters positiona
-00003e40: 6c6c 792e 2049 6e20 7468 6520 756e 6c69  lly. In the unli
-00003e50: 6b65 6c79 2065 7665 6e74 2074 6861 7420  kely event that 
-00003e60: 7468 6973 2068 6170 7065 6e73 2c20 7573  this happens, us
-00003e70: 6572 7320 7769 6c6c 2067 6574 2061 6e20  ers will get an 
-00003e80: 6572 726f 7220 7465 6c6c 696e 6720 7468  error telling th
-00003e90: 656d 2061 2062 6f6f 6c20 6973 6e27 7420  em a bool isn't 
-00003ea0: 6120 7661 6c69 6420 6049 6e74 6572 7661  a valid `Interva
-00003eb0: 6c4d 6574 6164 6174 6160 2074 7970 652c  lMetadata` type,
-00003ec0: 2073 6f20 6974 2077 6f6e 2774 2073 696c   so it won't sil
-00003ed0: 656e 746c 7920 7072 6f64 7563 6520 6275  ently produce bu
-00003ee0: 6767 7920 6265 6861 7669 6f72 2e0a 0a23  ggy behavior...#
-00003ef0: 2323 2042 6163 6b77 6172 642d 696e 636f  ## Backward-inco
-00003f00: 6d70 6174 6962 6c65 2063 6861 6e67 6573  mpatible changes
-00003f10: 205b 6578 7065 7269 6d65 6e74 616c 5d0a   [experimental].
-00003f20: 2a20 4d6f 6469 6679 696e 6720 6261 7369  * Modifying basi
-00003f30: 7320 6861 6e64 6c69 6e67 2069 6e20 6073  s handling in `s
-00003f40: 6b62 696f 2e73 7461 7473 2e63 6f6d 706f  kbio.stats.compo
-00003f50: 7369 7469 6f6e 2e69 6c72 5f69 6e76 6020  sition.ilr_inv` 
-00003f60: 7072 696f 7220 746f 2063 6865 636b 696e  prior to checkin
-00003f70: 6720 666f 7220 6f72 7468 6f67 6f6e 616c  g for orthogonal
-00003f80: 6974 792e 2020 4e6f 7720 7468 6520 6261  ity.  Now the ba
-00003f90: 7369 7320 6973 2073 7472 6963 746c 7920  sis is strictly 
-00003fa0: 6173 7375 6d65 6420 746f 2062 6520 696e  assumed to be in
-00003fb0: 2074 6865 2041 6974 6368 6973 6f6e 2073   the Aitchison s
-00003fc0: 696d 706c 6578 2e0a 2a20 6044 6973 7461  implex..* `Dista
-00003fd0: 6e63 654d 6174 7269 782e 6672 6f6d 5f69  nceMatrix.from_i
-00003fe0: 7465 7261 626c 6560 2064 6566 6175 6c74  terable` default
-00003ff0: 2062 6568 6176 696f 7220 6973 206e 6f77   behavior is now
-00004000: 2074 6f20 7661 6c69 6461 7465 206d 6174   to validate mat
-00004010: 7269 7820 6279 2063 6f6d 7075 7469 6e67  rix by computing
-00004020: 2061 6c6c 2070 6169 7277 6973 6520 6469   all pairwise di
-00004030: 7374 616e 6365 732e 2050 6173 7320 6076  stances. Pass `v
-00004040: 616c 6964 6174 653d 4661 6c73 6560 2074  alidate=False` t
-00004050: 6f20 6765 7420 7468 6520 7072 6576 696f  o get the previo
-00004060: 7573 2062 6568 6176 696f 7220 286e 6f20  us behavior (no 
-00004070: 7661 6c69 6461 7469 6f6e 2c20 6275 7420  validation, but 
-00004080: 6661 7374 6572 2065 7865 6375 7469 6f6e  faster execution
-00004090: 292e 285b 2331 3334 335d 2868 7474 7073  ).([#1343](https
-000040a0: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
-000040b0: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
-000040c0: 2f69 7373 7565 732f 3133 3433 2929 2e0a  /issues/1343))..
-000040d0: 2a20 4765 6e42 616e 6b20 492f 4f20 6e6f  * GenBank I/O no
-000040e0: 7720 7061 7273 6573 2073 6571 7565 6e63  w parses sequenc
-000040f0: 6520 6665 6174 7572 6573 2069 6e74 6f20  e features into 
-00004100: 7468 6520 6174 7472 6962 7574 6520 6f66  the attribute of
-00004110: 2060 696e 7465 7276 616c 5f6d 6574 6164   `interval_metad
-00004120: 6174 6160 2069 6e73 7465 6164 206f 6620  ata` instead of 
-00004130: 6070 6f73 6974 696f 6e61 5f6d 6574 6164  `positiona_metad
-00004140: 6174 6160 2e20 416e 6420 7468 6520 6b65  ata`. And the ke
-00004150: 7920 6f66 2060 4645 4154 5552 4553 6020  y of `FEATURES` 
-00004160: 6973 2072 656d 6f76 6564 2066 726f 6d20  is removed from 
-00004170: 606d 6574 6164 6174 6160 2061 7474 7269  `metadata` attri
-00004180: 6275 7465 2e0a 0a23 2323 2050 6572 666f  bute...### Perfo
-00004190: 726d 616e 6365 2065 6e68 616e 6365 6d65  rmance enhanceme
-000041a0: 6e74 730a 2a20 6054 7265 654e 6f64 652e  nts.* `TreeNode.
-000041b0: 7368 6561 7260 2077 6173 2072 6577 7269  shear` was rewri
-000041c0: 7474 656e 2066 6f72 2061 7070 726f 7869  tten for approxi
-000041d0: 6d61 7465 6c79 2061 2032 3525 2070 6572  mately a 25% per
-000041e0: 666f 726d 616e 6365 2069 6e63 7265 6173  formance increas
-000041f0: 652e 2028 5b23 3133 3939 5d28 6874 7470  e. ([#1399](http
-00004200: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-00004210: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-00004220: 6f2f 7075 6c6c 2f31 3339 3929 290a 2a20  o/pull/1399)).* 
-00004230: 5468 6520 6049 6e74 6572 7661 6c4d 6574  The `IntervalMet
-00004240: 6164 6174 6160 2061 6c6c 6f77 7320 6472  adata` allows dr
-00004250: 616d 6174 6963 2064 6563 7265 6173 6520  amatic decrease 
-00004260: 696e 206d 656d 6f72 7920 7573 6167 6520  in memory usage 
-00004270: 696e 2072 6561 6469 6e67 2047 656e 4261  in reading GenBa
-00004280: 6e6b 2066 696c 6573 206f 6620 6665 6174  nk files of feat
-00004290: 7572 6520 7269 6368 2073 6571 7565 6e63  ure rich sequenc
-000042a0: 6573 2e20 285b 2331 3135 395d 2868 7474  es. ([#1159](htt
-000042b0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-000042c0: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-000042d0: 696f 2f69 7373 7565 732f 3131 3539 2929  io/issues/1159))
-000042e0: 0a0a 2323 2320 4275 6720 6669 7865 730a  ..### Bug fixes.
-000042f0: 0a2a 2060 736b 6269 6f2e 7472 6565 2e54  .* `skbio.tree.T
-00004300: 7265 654e 6f64 652e 7072 756e 6560 2061  reeNode.prune` a
-00004310: 6e64 2069 6d70 6c69 6369 746c 7920 6073  nd implicitly `s
-00004320: 6b62 696f 2e74 7265 652e 5472 6565 4e6f  kbio.tree.TreeNo
-00004330: 6465 2e73 6865 6172 6020 7765 7265 206e  de.shear` were n
-00004340: 6f74 2068 616e 646c 696e 6720 6120 7369  ot handling a si
-00004350: 7475 6174 696f 6e20 696e 2077 6869 6368  tuation in which
-00004360: 2061 2070 6172 656e 7420 7761 7320 7661   a parent was va
-00004370: 6c69 646c 7920 7265 6d6f 7665 6420 6475  lidly removed du
-00004380: 7269 6e67 2070 7275 6e69 6e67 206f 7065  ring pruning ope
-00004390: 7261 7469 6f6e 7320 6173 206d 6179 2068  rations as may h
-000043a0: 6170 7065 6e20 6966 2074 6865 2072 6573  appen if the res
-000043b0: 756c 7469 6e67 2073 7562 7472 6565 2064  ulting subtree d
-000043c0: 6f65 7320 6e6f 7420 696e 636c 7564 6520  oes not include 
-000043d0: 7468 6520 726f 6f74 2e20 5072 6576 696f  the root. Previo
-000043e0: 7573 6c79 2c20 616e 2060 4174 7472 6962  usly, an `Attrib
-000043f0: 7574 6545 7272 6f72 6020 776f 756c 6420  uteError` would 
-00004400: 7261 6973 6520 6173 2060 7061 7265 6e74  raise as `parent
-00004410: 6020 776f 756c 6420 6265 2060 4e6f 6e65  ` would be `None
-00004420: 6020 696e 2074 6869 7320 7369 7475 6174  ` in this situat
-00004430: 696f 6e2e 0a2a 206e 756d 7079 206c 696e  ion..* numpy lin
-00004440: 6b69 6e67 2077 6173 2066 6978 6564 2066  king was fixed f
-00004450: 6f72 2069 6e73 7461 6c6c 6174 696f 6e20  or installation 
-00004460: 756e 6465 7220 456c 2043 6170 6974 616e  under El Capitan
-00004470: 2e0a 2a20 4120 6275 6720 7761 7320 696e  ..* A bug was in
-00004480: 7472 6f64 7563 6564 2069 6e20 2331 3339  troduced in #139
-00004490: 3820 696e 746f 2060 5472 6565 4e6f 6465  8 into `TreeNode
-000044a0: 2e70 7275 6e65 6020 616e 6420 6669 7865  .prune` and fixe
-000044b0: 6420 696e 2023 3134 3136 2069 6e20 7768  d in #1416 in wh
-000044c0: 6963 682c 2075 6e64 6572 2074 6865 2073  ich, under the s
-000044d0: 7065 6369 616c 2063 6173 6520 6f66 2061  pecial case of a
-000044e0: 2073 696e 676c 6520 6465 7363 656e 6465   single descende
-000044f0: 6e74 2065 7869 7374 696e 6720 6672 6f6d  nt existing from
-00004500: 2074 6865 2072 6f6f 742c 2074 6865 2072   the root, the r
-00004510: 6573 756c 7469 6e67 2063 6869 6c64 7265  esulting childre
-00004520: 6e20 7061 7265 6e74 2072 6566 6572 656e  n parent referen
-00004530: 6365 7320 7765 7265 206e 6f74 2075 7064  ces were not upd
-00004540: 6174 6564 2e20 5468 6520 6361 7573 6520  ated. The cause 
-00004550: 6f66 2074 6865 2062 7567 2077 6173 2061  of the bug was a
-00004560: 2063 616c 6c20 6d61 6465 2074 6f20 6073   call made to `s
-00004570: 656c 662e 6368 696c 6472 656e 2e65 7874  elf.children.ext
-00004580: 656e 6460 2061 7320 6f70 706f 7365 6420  end` as opposed 
-00004590: 746f 2060 7365 6c66 2e65 7874 656e 6460  to `self.extend`
-000045a0: 2077 6865 7265 2074 6865 2066 6f72 6d65   where the forme
-000045b0: 7220 6973 2061 2060 6c69 7374 2e65 7874  r is a `list.ext
-000045c0: 656e 6460 2077 6974 686f 7574 206b 6e6f  end` without kno
-000045d0: 776c 6564 6765 206f 6620 7468 6520 7472  wledge of the tr
-000045e0: 6565 2c20 7768 696c 6520 7468 6520 6c61  ee, while the la
-000045f0: 7474 6572 2069 7320 6054 7265 654e 6f64  tter is `TreeNod
-00004600: 652e 6578 7465 6e64 6020 7768 6963 6820  e.extend` which 
-00004610: 6973 2061 626c 6520 746f 2061 646a 7573  is able to adjus
-00004620: 7420 7265 6665 7265 6e63 6573 2074 6f20  t references to 
-00004630: 6073 656c 662e 7061 7265 6e74 602e 0a0a  `self.parent`...
-00004640: 2323 2320 4d69 7363 656c 6c61 6e65 6f75  ### Miscellaneou
-00004650: 730a 0a2a 2052 656d 6f76 6564 2064 6570  s..* Removed dep
-00004660: 7265 6361 7465 6420 6675 6e63 7469 6f6e  recated function
-00004670: 7320 6672 6f6d 2060 736b 6269 6f2e 7574  s from `skbio.ut
-00004680: 696c 603a 2060 6973 5f63 6173 6176 615f  il`: `is_casava_
-00004690: 7631 3830 5f6f 725f 6c61 7465 7260 2c20  v180_or_later`, 
-000046a0: 6072 656d 6f76 655f 6669 6c65 7360 2c20  `remove_files`, 
-000046b0: 616e 6420 6063 7265 6174 655f 6469 7260  and `create_dir`
-000046c0: 2e0a 2a20 5265 6d6f 7665 6420 6465 7072  ..* Removed depr
-000046d0: 6563 6174 6564 2060 736b 6269 6f2e 5365  ecated `skbio.Se
-000046e0: 7175 656e 6365 2e63 6f70 7960 206d 6574  quence.copy` met
-000046f0: 686f 642e 0a0a 2323 2056 6572 7369 6f6e  hod...## Version
-00004700: 2030 2e35 2e30 2028 3230 3136 2d30 362d   0.5.0 (2016-06-
-00004710: 3134 290a 0a2a 2a49 4d50 4f52 5441 4e54  14)..**IMPORTANT
-00004720: 2a2a 3a20 7363 696b 6974 2d62 696f 2069  **: scikit-bio i
-00004730: 7320 6e6f 206c 6f6e 6765 7220 636f 6d70  s no longer comp
-00004740: 6174 6962 6c65 2077 6974 6820 5079 7468  atible with Pyth
-00004750: 6f6e 2032 2e20 7363 696b 6974 2d62 696f  on 2. scikit-bio
-00004760: 2069 7320 636f 6d70 6174 6962 6c65 2077   is compatible w
-00004770: 6974 6820 5079 7468 6f6e 2033 2e34 2061  ith Python 3.4 a
-00004780: 6e64 206c 6174 6572 2e0a 0a23 2323 2046  nd later...### F
-00004790: 6561 7475 7265 730a 2a20 4164 6465 6420  eatures.* Added 
-000047a0: 6d6f 7265 2064 6573 6372 6970 7469 7665  more descriptive
-000047b0: 2065 7272 6f72 206d 6573 7361 6765 2074   error message t
-000047c0: 6f20 6073 6b62 696f 2e69 6f2e 7265 6769  o `skbio.io.regi
-000047d0: 7374 7279 6020 7768 656e 2061 7474 656d  stry` when attem
-000047e0: 7074 696e 6720 746f 2072 6561 6420 7769  pting to read wi
-000047f0: 7468 6f75 7420 7370 6563 6966 7969 6e67  thout specifying
-00004800: 2060 696e 746f 6020 616e 6420 7768 656e   `into` and when
-00004810: 2074 6865 7265 2069 7320 6e6f 2067 656e   there is no gen
-00004820: 6572 6174 6f72 2072 6561 6465 722e 2028  erator reader. (
-00004830: 5b23 3133 3236 5d28 6874 7470 733a 2f2f  [#1326](https://
-00004840: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-00004850: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
-00004860: 7375 6573 2f31 3332 3629 290a 2a20 4164  sues/1326)).* Ad
-00004870: 6465 6420 7375 7070 6f72 7420 666f 7220  ded support for 
-00004880: 7265 6665 7265 6e63 6520 7461 6773 2074  reference tags t
-00004890: 6f20 6073 6b62 696f 2e69 6f2e 666f 726d  o `skbio.io.form
-000048a0: 6174 2e73 746f 636b 686f 6c6d 6020 7265  at.stockholm` re
-000048b0: 6164 6572 2061 6e64 2077 7269 7465 722e  ader and writer.
-000048c0: 2028 5b23 3133 3438 5d28 6874 7470 733a   ([#1348](https:
-000048d0: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
-000048e0: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
-000048f0: 6973 7375 6573 2f31 3334 3829 290a 2a20  issues/1348)).* 
-00004900: 4578 7061 6e64 6564 2065 7272 6f72 206d  Expanded error m
-00004910: 6573 7361 6765 2069 6e20 6073 6b62 696f  essage in `skbio
-00004920: 2e69 6f2e 666f 726d 6174 2e73 746f 636b  .io.format.stock
-00004930: 686f 6c6d 6020 7265 6164 6572 2077 6865  holm` reader whe
-00004940: 6e20 6063 6f6e 7374 7275 6374 6f72 6020  n `constructor` 
-00004950: 6973 206e 6f74 2070 6173 7365 642c 2069  is not passed, i
-00004960: 6e20 6f72 6465 7220 746f 2070 726f 7669  n order to provi
-00004970: 6465 2062 6574 7465 7220 6578 706c 616e  de better explan
-00004980: 6174 696f 6e20 746f 2075 7365 722e 2028  ation to user. (
-00004990: 5b23 3133 3237 5d28 6874 7470 733a 2f2f  [#1327](https://
-000049a0: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-000049b0: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
-000049c0: 7375 6573 2f31 3332 3729 290a 2a20 4164  sues/1327)).* Ad
-000049d0: 6465 6420 6073 6b62 696f 2e73 6571 7565  ded `skbio.seque
-000049e0: 6e63 652e 6469 7374 616e 6365 2e6b 6d65  nce.distance.kme
-000049f0: 725f 6469 7374 616e 6365 6020 666f 7220  r_distance` for 
-00004a00: 636f 6d70 7574 696e 6720 7468 6520 6b6d  computing the km
-00004a10: 6572 2064 6973 7461 6e63 6520 6265 7477  er distance betw
-00004a20: 6565 6e20 7477 6f20 7365 7175 656e 6365  een two sequence
-00004a30: 732e 2028 5b23 3931 335d 2868 7474 7073  s. ([#913](https
-00004a40: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
-00004a50: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
-00004a60: 2f69 7373 7565 732f 3931 3329 290a 2a20  /issues/913)).* 
-00004a70: 4164 6465 6420 6073 6b62 696f 2e73 6571  Added `skbio.seq
-00004a80: 7565 6e63 652e 5365 7175 656e 6365 2e72  uence.Sequence.r
-00004a90: 6570 6c61 6365 6020 666f 7220 6173 7369  eplace` for assi
-00004aa0: 676e 696e 6720 6120 6368 6172 6163 7465  gning a characte
-00004ab0: 7220 746f 2070 6f73 6974 696f 6e73 2069  r to positions i
-00004ac0: 6e20 6120 6053 6571 7565 6e63 6560 2e20  n a `Sequence`. 
-00004ad0: 285b 2331 3232 325d 2868 7474 7073 3a2f  ([#1222](https:/
-00004ae0: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
-00004af0: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
-00004b00: 7373 7565 732f 3132 3232 2929 0a2a 2041  ssues/1222)).* A
-00004b10: 6464 6564 2073 7570 706f 7274 2066 6f72  dded support for
-00004b20: 2060 7061 6e64 6173 2e52 616e 6765 496e   `pandas.RangeIn
-00004b30: 6465 7860 2c20 6c6f 7765 7269 6e67 2074  dex`, lowering t
-00004b40: 6865 206d 656d 6f72 7920 666f 6f74 7072  he memory footpr
-00004b50: 696e 7420 6f66 2064 6566 6175 6c74 2069  int of default i
-00004b60: 6e74 6567 6572 2069 6e64 6578 206f 626a  nteger index obj
-00004b70: 6563 7473 2e20 6053 6571 7565 6e63 652e  ects. `Sequence.
-00004b80: 706f 7369 7469 6f6e 616c 5f6d 6574 6164  positional_metad
-00004b90: 6174 6160 2061 6e64 2060 5461 6275 6c61  ata` and `Tabula
-00004ba0: 724d 5341 2e70 6f73 6974 696f 6e61 6c5f  rMSA.positional_
-00004bb0: 6d65 7461 6461 7461 6020 6e6f 7720 7573  metadata` now us
-00004bc0: 6520 6070 642e 5261 6e67 6549 6e64 6578  e `pd.RangeIndex
-00004bd0: 6020 6173 2074 6865 2070 6f73 6974 696f  ` as the positio
-00004be0: 6e61 6c20 6d65 7461 6461 7461 2069 6e64  nal metadata ind
-00004bf0: 6578 2e20 6054 6162 756c 6172 4d53 4160  ex. `TabularMSA`
-00004c00: 206e 6f77 2075 7365 7320 6070 642e 5261   now uses `pd.Ra
-00004c10: 6e67 6549 6e64 6578 6020 6173 2074 6865  ngeIndex` as the
-00004c20: 2064 6566 6175 6c74 2069 6e64 6578 2e20   default index. 
-00004c30: 5573 6167 6520 6f66 2060 7064 2e52 616e  Usage of `pd.Ran
-00004c40: 6765 496e 6465 7860 206f 7665 7220 7468  geIndex` over th
-00004c50: 6520 7072 6576 696f 7573 2060 7064 2e49  e previous `pd.I
-00004c60: 6e74 3634 496e 6465 7860 205b 7368 6f75  nt64Index` [shou
-00004c70: 6c64 2062 6520 7472 616e 7370 6172 656e  ld be transparen
-00004c80: 745d 2868 7474 703a 2f2f 7061 6e64 6173  t](http://pandas
-00004c90: 2e70 7964 6174 612e 6f72 672f 7061 6e64  .pydata.org/pand
-00004ca0: 6173 2d64 6f63 732f 7665 7273 696f 6e2f  as-docs/version/
-00004cb0: 302e 3138 2e30 2f77 6861 7473 6e65 772e  0.18.0/whatsnew.
-00004cc0: 6874 6d6c 2372 616e 6765 2d69 6e64 6578  html#range-index
-00004cd0: 292c 2073 6f20 7468 6573 6520 6368 616e  ), so these chan
-00004ce0: 6765 7320 7368 6f75 6c64 2062 6520 6e6f  ges should be no
-00004cf0: 6e2d 6272 6561 6b69 6e67 2074 6f20 7573  n-breaking to us
-00004d00: 6572 732e 2073 6369 6b69 742d 6269 6f20  ers. scikit-bio 
-00004d10: 6e6f 7720 6465 7065 6e64 7320 6f6e 2070  now depends on p
-00004d20: 616e 6461 7320 3e3d 2030 2e31 382e 3020  andas >= 0.18.0 
-00004d30: 285b 2331 3330 385d 2868 7474 7073 3a2f  ([#1308](https:/
-00004d40: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
-00004d50: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
-00004d60: 7373 7565 732f 3133 3038 2929 0a2a 2041  ssues/1308)).* A
-00004d70: 6464 6564 2060 7265 7365 745f 696e 6465  dded `reset_inde
-00004d80: 783d 4661 6c73 6560 2070 6172 616d 6574  x=False` paramet
-00004d90: 6572 2074 6f20 6054 6162 756c 6172 4d53  er to `TabularMS
-00004da0: 412e 6170 7065 6e64 6020 616e 6420 6054  A.append` and `T
-00004db0: 6162 756c 6172 4d53 412e 6578 7465 6e64  abularMSA.extend
-00004dc0: 6020 666f 7220 7265 7365 7474 696e 6720  ` for resetting 
-00004dd0: 7468 6520 4d53 4127 7320 696e 6465 7820  the MSA's index 
-00004de0: 746f 2074 6865 2064 6566 6175 6c74 2069  to the default i
-00004df0: 6e64 6578 2061 6674 6572 2061 7070 656e  ndex after appen
-00004e00: 6469 6e67 2f65 7874 656e 6469 6e67 2e0a  ding/extending..
-00004e10: 2a20 4164 6465 6420 7375 7070 6f72 7420  * Added support 
-00004e20: 666f 7220 7061 7274 6961 6c20 7061 6972  for partial pair
-00004e30: 7769 7365 2063 616c 6375 6c61 7469 6f6e  wise calculation
-00004e40: 7320 7669 6120 6073 6b62 696f 2e64 6976  s via `skbio.div
-00004e50: 6572 7369 7479 2e70 6172 7469 616c 5f62  ersity.partial_b
-00004e60: 6574 615f 6469 7665 7273 6974 7960 2e20  eta_diversity`. 
-00004e70: 285b 2331 3232 315d 2868 7474 7073 3a2f  ([#1221](https:/
-00004e80: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
-00004e90: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
-00004ea0: 7373 7565 732f 3132 3231 292c 205b 2331  ssues/1221), [#1
-00004eb0: 3333 375d 2868 7474 7073 3a2f 2f67 6974  337](https://git
-00004ec0: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
-00004ed0: 7363 696b 6974 2d62 696f 2f70 756c 6c2f  scikit-bio/pull/
-00004ee0: 3133 3337 2929 2e20 5468 6973 2066 756e  1337)). This fun
-00004ef0: 6374 696f 6e20 6973 2069 6d6d 6564 6961  ction is immedia
-00004f00: 7465 6c79 2064 6570 7265 6361 7465 6420  tely deprecated 
-00004f10: 6173 2069 7473 2072 6574 7572 6e20 7479  as its return ty
-00004f20: 7065 2077 696c 6c20 6368 616e 6765 2069  pe will change i
-00004f30: 6e20 7468 6520 6675 7475 7265 2061 6e64  n the future and
-00004f40: 2073 686f 756c 6420 6265 2075 7365 6420   should be used 
-00004f50: 7769 7468 2063 6175 7469 6f6e 2069 6e20  with caution in 
-00004f60: 6974 7320 7072 6573 656e 7420 666f 726d  its present form
-00004f70: 2028 7365 6520 7468 6520 6675 6e63 7469   (see the functi
-00004f80: 6f6e 2773 2064 6f63 756d 656e 7461 7469  on's documentati
-00004f90: 6f6e 2066 6f72 2064 6574 6169 6c73 292e  on for details).
-00004fa0: 0a2a 2060 5465 6d70 6f72 6172 7946 696c  .* `TemporaryFil
-00004fb0: 6560 2061 6e64 2060 4e61 6d65 6454 656d  e` and `NamedTem
-00004fc0: 706f 7261 7279 4669 6c65 6020 6172 6520  poraryFile` are 
-00004fd0: 6e6f 7720 7375 7070 6f72 7465 6420 494f  now supported IO
-00004fe0: 2073 6f75 7263 6573 2066 6f72 2060 736b   sources for `sk
-00004ff0: 6269 6f2e 696f 6020 616e 6420 7265 6c61  bio.io` and rela
-00005000: 7465 6420 6675 6e63 7469 6f6e 616c 6974  ted functionalit
-00005010: 792e 2020 285b 2331 3239 315d 2868 7474  y.  ([#1291](htt
-00005020: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00005030: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-00005040: 696f 2f69 7373 7565 732f 3132 3931 2929  io/issues/1291))
-00005050: 0a2a 2041 6464 6564 2060 7472 6565 5f6e  .* Added `tree_n
-00005060: 6f64 655f 636c 6173 733d 5472 6565 4e6f  ode_class=TreeNo
-00005070: 6465 6020 7061 7261 6d65 7465 7220 746f  de` parameter to
-00005080: 2060 736b 6269 6f2e 7472 6565 2e6d 616a   `skbio.tree.maj
-00005090: 6f72 6974 795f 7275 6c65 6020 746f 2073  ority_rule` to s
-000050a0: 7570 706f 7274 2072 6574 7572 6e69 6e67  upport returning
-000050b0: 2063 6f6e 7365 6e73 7573 2074 7265 6573   consensus trees
-000050c0: 206f 6620 7479 7065 2060 5472 6565 4e6f   of type `TreeNo
-000050d0: 6465 6020 2874 6865 2064 6566 6175 6c74  de` (the default
-000050e0: 2920 6f72 2061 2074 7970 6520 7468 6174  ) or a type that
-000050f0: 2068 6173 2074 6865 2073 616d 6520 696e   has the same in
-00005100: 7465 7266 6163 6520 6173 2060 5472 6565  terface as `Tree
-00005110: 4e6f 6465 6020 2865 2e67 2e20 6054 7265  Node` (e.g. `Tre
-00005120: 654e 6f64 6560 2073 7562 636c 6173 7365  eNode` subclasse
-00005130: 7329 2028 5b23 3131 3933 5d28 6874 7470  s) ([#1193](http
-00005140: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-00005150: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-00005160: 6f2f 7075 6c6c 2f31 3139 3329 290a 2a20  o/pull/1193)).* 
-00005170: 6054 7265 654e 6f64 652e 6672 6f6d 5f6c  `TreeNode.from_l
-00005180: 696e 6b61 6765 5f6d 6174 7269 7860 2061  inkage_matrix` a
-00005190: 6e64 2060 5472 6565 4e6f 6465 2e66 726f  nd `TreeNode.fro
-000051a0: 6d5f 7461 786f 6e6f 6d79 6020 6e6f 7720  m_taxonomy` now 
-000051b0: 7375 7070 6f72 7420 636f 6e73 7472 7563  support construc
-000051c0: 7469 6e67 2060 5472 6565 4e6f 6465 6020  ting `TreeNode` 
-000051d0: 7375 6263 6c61 7373 6573 2e20 6054 7265  subclasses. `Tre
-000051e0: 654e 6f64 652e 6269 6675 7263 6174 6560  eNode.bifurcate`
-000051f0: 206e 6f77 2073 7570 706f 7274 7320 6054   now supports `T
-00005200: 7265 654e 6f64 6560 2073 7562 636c 6173  reeNode` subclas
-00005210: 7365 7320 285b 2331 3139 335d 2868 7474  ses ([#1193](htt
-00005220: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00005230: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-00005240: 696f 2f70 756c 6c2f 3131 3933 2929 0a2a  io/pull/1193)).*
-00005250: 2054 6865 2060 6967 6e6f 7265 5f6d 6574   The `ignore_met
-00005260: 6164 6174 6160 206b 6579 776f 7264 2068  adata` keyword h
-00005270: 6173 2062 6565 6e20 6164 6465 6420 746f  as been added to
-00005280: 2060 5461 6275 6c61 724d 5341 2e69 7465   `TabularMSA.ite
-00005290: 725f 706f 7369 7469 6f6e 7360 2074 6f20  r_positions` to 
-000052a0: 696d 7072 6f76 6520 7065 7266 6f72 6d61  improve performa
-000052b0: 6e63 6520 7768 656e 206d 6574 6164 6174  nce when metadat
-000052c0: 6120 6973 206e 6f74 206e 6563 6573 7361  a is not necessa
-000052d0: 7279 2e0a 2a20 5061 6972 7769 7365 2061  ry..* Pairwise a
-000052e0: 6c69 676e 6572 7320 696e 2060 736b 6269  ligners in `skbi
-000052f0: 6f2e 616c 6967 6e6d 656e 7460 206e 6f77  o.alignment` now
-00005300: 2070 726f 7061 6761 7465 2070 6572 2d73   propagate per-s
-00005310: 6571 7565 6e63 6520 606d 6574 6164 6174  equence `metadat
-00005320: 6160 206f 626a 6563 7473 2028 7468 6973  a` objects (this
-00005330: 2064 6f65 7320 6e6f 7420 696e 636c 7564   does not includ
-00005340: 6520 6070 6f73 6974 696f 6e61 6c5f 6d65  e `positional_me
-00005350: 7461 6461 7461 6029 2e0a 0a23 2323 2042  tadata`)...### B
-00005360: 6163 6b77 6172 642d 696e 636f 6d70 6174  ackward-incompat
-00005370: 6962 6c65 2063 6861 6e67 6573 205b 7374  ible changes [st
-00005380: 6162 6c65 5d0a 0a23 2323 2042 6163 6b77  able]..### Backw
-00005390: 6172 642d 696e 636f 6d70 6174 6962 6c65  ard-incompatible
-000053a0: 2063 6861 6e67 6573 205b 6578 7065 7269   changes [experi
-000053b0: 6d65 6e74 616c 5d0a 2a20 6054 6162 756c  mental].* `Tabul
-000053c0: 6172 4d53 412e 6170 7065 6e64 6020 616e  arMSA.append` an
-000053d0: 6420 6054 6162 756c 6172 4d53 412e 6578  d `TabularMSA.ex
-000053e0: 7465 6e64 6020 6e6f 7720 7265 7175 6972  tend` now requir
-000053f0: 6520 6f6e 6520 6f66 2060 6d69 6e74 6572  e one of `minter
-00005400: 602c 2060 696e 6465 7860 2c20 6f72 2060  `, `index`, or `
-00005410: 7265 7365 745f 696e 6465 7860 2074 6f20  reset_index` to 
-00005420: 6265 2070 726f 7669 6465 6420 7768 656e  be provided when
-00005430: 2069 6e63 6f72 706f 7261 7469 6e67 206e   incorporating n
-00005440: 6577 2073 6571 7565 6e63 6573 2069 6e74  ew sequences int
-00005450: 6f20 616e 204d 5341 2e20 5072 6576 696f  o an MSA. Previo
-00005460: 7573 2062 6568 6176 696f 7220 7761 7320  us behavior was 
-00005470: 746f 2061 7574 6f2d 696e 6372 656d 656e  to auto-incremen
-00005480: 7420 7468 6520 696e 6465 7820 6c61 6265  t the index labe
-00005490: 6c73 2069 6620 606d 696e 7465 7260 2061  ls if `minter` a
-000054a0: 6e64 2060 696e 6465 7860 2077 6572 656e  nd `index` weren
-000054b0: 2774 2070 726f 7669 6465 6420 616e 6420  't provided and 
-000054c0: 7468 6520 4d53 4120 6861 6420 6120 6465  the MSA had a de
-000054d0: 6661 756c 7420 696e 7465 6765 7220 696e  fault integer in
-000054e0: 6465 782c 206f 7468 6572 7769 7365 2065  dex, otherwise e
-000054f0: 7272 6f72 2e20 5573 6520 6072 6573 6574  rror. Use `reset
-00005500: 5f69 6e64 6578 3d54 7275 6560 2074 6f20  _index=True` to 
-00005510: 6f62 7461 696e 2074 6865 2070 7265 7669  obtain the previ
-00005520: 6f75 7320 6265 6861 7669 6f72 2069 6e20  ous behavior in 
-00005530: 6120 6d6f 7265 2065 7870 6c69 6369 7420  a more explicit 
-00005540: 7761 792e 0a2a 2060 736b 6269 6f2e 7374  way..* `skbio.st
-00005550: 6174 732e 636f 6d70 6f73 6974 696f 6e2e  ats.composition.
-00005560: 616e 636f 6d60 206e 6f77 2072 6574 7572  ancom` now retur
-00005570: 6e73 2074 776f 2060 7064 2e44 6174 6146  ns two `pd.DataF
-00005580: 7261 6d65 6020 6f62 6a65 6374 732c 2077  rame` objects, w
-00005590: 6865 7265 2069 7420 7072 6576 696f 7573  here it previous
-000055a0: 6c79 2072 6574 7572 6e65 6420 6f6e 652e  ly returned one.
-000055b0: 2054 6865 2066 6972 7374 2063 6f6e 7461   The first conta
-000055c0: 696e 7320 7468 6520 414e 434f 4d20 7465  ins the ANCOM te
-000055d0: 7374 2072 6573 756c 7473 2c20 6173 2062  st results, as b
-000055e0: 6566 6f72 652c 2061 6e64 2074 6865 2073  efore, and the s
-000055f0: 6563 6f6e 6420 636f 6e74 6169 6e73 2070  econd contains p
-00005600: 6572 6365 6e74 696c 6520 6162 756e 6461  ercentile abunda
-00005610: 6e63 6573 206f 6620 6561 6368 2066 6561  nces of each fea
-00005620: 7475 7265 2069 6e20 6561 6368 2067 726f  ture in each gro
-00005630: 7570 2e20 5468 6520 7370 6563 6966 6963  up. The specific
-00005640: 2070 6572 6365 6e74 696c 6573 2074 6861   percentiles tha
-00005650: 7420 6172 6520 636f 6d70 7574 6564 2061  t are computed a
-00005660: 6e64 2072 6574 7572 6e65 6420 6973 2063  nd returned is c
-00005670: 6f6e 7472 6f6c 6c65 6420 6279 2074 6865  ontrolled by the
-00005680: 206e 6577 2060 7065 7263 656e 7469 6c65   new `percentile
-00005690: 7360 2070 6172 616d 6574 6572 2074 6f20  s` parameter to 
-000056a0: 6073 6b62 696f 2e73 7461 7473 2e63 6f6d  `skbio.stats.com
-000056b0: 706f 7369 7469 6f6e 2e61 6e63 6f6d 602e  position.ancom`.
-000056c0: 2049 6e20 7468 6520 6675 7475 7265 2c20   In the future, 
-000056d0: 7468 6973 2073 6563 6f6e 6420 6070 642e  this second `pd.
-000056e0: 4461 7461 4672 616d 6560 2077 696c 6c20  DataFrame` will 
-000056f0: 6e6f 7420 6265 2072 6574 7572 6e65 6420  not be returned 
-00005700: 6279 2074 6869 7320 6675 6e63 7469 6f6e  by this function
-00005710: 2c20 6275 7420 7769 6c6c 2062 6520 6176  , but will be av
-00005720: 6169 6c61 626c 6520 7468 726f 7567 6820  ailable through 
-00005730: 7468 6520 5b63 6f6e 7469 6e67 656e 6379  the [contingency
-00005740: 2074 6162 6c65 2041 5049 5d28 6874 7470   table API](http
-00005750: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-00005760: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-00005770: 6f2f 6973 7375 6573 2f38 3438 292e 2028  o/issues/848). (
-00005780: 5b23 3132 3933 5d28 6874 7470 733a 2f2f  [#1293](https://
-00005790: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-000057a0: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
-000057b0: 7375 6573 2f31 3239 3329 290a 2a20 6073  sues/1293)).* `s
-000057c0: 6b62 696f 2e73 7461 7473 2e63 6f6d 706f  kbio.stats.compo
-000057d0: 7369 7469 6f6e 2e61 6e63 6f6d 6020 6e6f  sition.ancom` no
-000057e0: 7720 7065 7266 6f72 6d73 206d 756c 7469  w performs multi
-000057f0: 706c 6520 636f 6d70 6172 6973 6f6e 7320  ple comparisons 
-00005800: 636f 7272 6563 7469 6f6e 2062 7920 6465  correction by de
-00005810: 6661 756c 742e 2054 6865 2070 7265 7669  fault. The previ
-00005820: 6f75 7320 6265 6861 7669 6f72 206f 6620  ous behavior of 
-00005830: 6e6f 7420 7065 7266 6f72 6d69 6e67 206d  not performing m
-00005840: 756c 7469 706c 6520 636f 6d70 6172 6973  ultiple comparis
-00005850: 6f6e 7320 636f 7272 6563 7469 6f6e 2063  ons correction c
-00005860: 616e 2062 6520 6163 6869 6576 6564 2062  an be achieved b
-00005870: 7920 7061 7373 696e 6720 6060 6d75 6c74  y passing ``mult
-00005880: 6970 6c65 5f63 6f6d 7061 7269 736f 6e73  iple_comparisons
-00005890: 5f63 6f72 7265 6374 696f 6e3d 4e6f 6e65  _correction=None
-000058a0: 6060 2e0a 2a20 5468 6520 6060 7265 6a65  ``..* The ``reje
-000058b0: 6374 6060 2063 6f6c 756d 6e20 696e 2074  ct`` column in t
-000058c0: 6865 2066 6972 7374 2060 6070 642e 4461  he first ``pd.Da
-000058d0: 7461 4672 616d 6560 6020 7265 7475 726e  taFrame`` return
-000058e0: 6564 2066 726f 6d20 6073 6b62 696f 2e73  ed from `skbio.s
-000058f0: 7461 7473 2e63 6f6d 706f 7369 7469 6f6e  tats.composition
-00005900: 2e61 6e63 6f6d 6020 6861 7320 6265 656e  .ancom` has been
-00005910: 2072 656e 616d 6564 2060 6052 656a 6563   renamed ``Rejec
-00005920: 7420 6e75 6c6c 2068 7970 6f74 6865 7369  t null hypothesi
-00005930: 7360 6020 666f 7220 636c 6172 6974 792e  s`` for clarity.
-00005940: 2028 5b23 3133 3735 5d28 6874 7470 733a   ([#1375](https:
-00005950: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
-00005960: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
-00005970: 6973 7375 6573 2f31 3337 3529 290a 0a23  issues/1375))..#
-00005980: 2323 2042 7567 2066 6978 6573 0a2a 2046  ## Bug fixes.* F
-00005990: 6978 6564 2072 6f77 2061 6e64 2063 6f6c  ixed row and col
-000059a0: 756d 6e20 6e61 6d65 7320 746f 2060 6269  umn names to `bi
-000059b0: 706c 6f74 5f73 636f 7265 7360 2069 6e20  plot_scores` in 
-000059c0: 7468 6520 604f 7264 696e 6174 696f 6e52  the `OrdinationR
-000059d0: 6573 756c 7473 6020 6f62 6a65 6374 2066  esults` object f
-000059e0: 726f 6d20 6073 6b62 696f 2e73 7461 7473  rom `skbio.stats
-000059f0: 2e6f 7264 696e 6174 696f 6e60 2e20 5468  .ordination`. Th
-00005a00: 6973 2066 6978 2061 6666 6563 7420 7468  is fix affect th
-00005a10: 6520 6063 6361 6020 616e 6420 6072 6461  e `cca` and `rda
-00005a20: 6020 6d65 7468 6f64 732e 2028 5b23 3133  ` methods. ([#13
-00005a30: 3232 5d28 6874 7470 733a 2f2f 6769 7468  22](https://gith
-00005a40: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
-00005a50: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
-00005a60: 2f31 3332 3229 290a 2a20 4669 7865 6420  /1322)).* Fixed 
-00005a70: 6275 6720 7768 656e 2075 7369 6e67 2060  bug when using `
-00005a80: 736b 6269 6f2e 696f 2e66 6f72 6d61 742e  skbio.io.format.
-00005a90: 7374 6f63 6b68 6f6c 6d60 2072 6561 6465  stockholm` reade
-00005aa0: 7220 6f6e 2066 696c 6520 7769 7468 206d  r on file with m
-00005ab0: 756c 7469 2d6c 696e 6520 7472 6565 2077  ulti-line tree w
-00005ac0: 6974 6820 6e6f 2069 642e 2050 7265 7669  ith no id. Previ
-00005ad0: 6f75 736c 7920 7468 6973 2072 6169 7365  ously this raise
-00005ae0: 6420 616e 2060 4174 7472 6962 7574 6545  d an `AttributeE
-00005af0: 7272 6f72 602c 206e 6f77 2069 7420 636f  rror`, now it co
-00005b00: 7272 6563 746c 7920 6861 6e64 6c65 7320  rrectly handles 
-00005b10: 7468 6973 2074 7970 6520 6f66 2074 7265  this type of tre
-00005b20: 652e 2028 5b23 3133 3334 5d28 6874 7470  e. ([#1334](http
-00005b30: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-00005b40: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-00005b50: 6f2f 6973 7375 6573 2f31 3333 3429 290a  o/issues/1334)).
-00005b60: 2a20 4669 7865 6420 6275 6720 7768 656e  * Fixed bug when
-00005b70: 2072 6561 6469 6e67 2053 746f 636b 686f   reading Stockho
-00005b80: 6c6d 2066 696c 6573 2077 6974 6820 4746  lm files with GF
-00005b90: 206f 7220 4753 2066 6561 7475 7265 7320   or GS features 
-00005ba0: 7370 6c69 7420 6f76 6572 206d 756c 7469  split over multi
-00005bb0: 706c 6520 6c69 6e65 732e 2050 7265 7669  ple lines. Previ
-00005bc0: 6f75 736c 792c 2074 6865 2066 6561 7475  ously, the featu
-00005bd0: 7265 2074 6578 7420 7761 7320 7369 6d70  re text was simp
-00005be0: 6c79 2063 6f6e 6361 7465 6e61 7465 6420  ly concatenated 
-00005bf0: 6265 6361 7573 6520 6974 2077 6173 2061  because it was a
-00005c00: 7373 756d 6564 2074 6f20 6861 7665 2074  ssumed to have t
-00005c10: 7261 696c 696e 6720 7768 6974 6573 7061  railing whitespa
-00005c20: 6365 2e20 5468 6572 6520 6172 6520 6578  ce. There are ex
-00005c30: 616d 706c 6573 206f 6620 5374 6f63 6b68  amples of Stockh
-00005c40: 6f6c 6d20 6669 6c65 7320 7769 7468 2061  olm files with a
-00005c50: 6e64 2077 6974 686f 7574 2074 7261 696c  nd without trail
-00005c60: 696e 6720 7768 6974 6573 7061 6365 2066  ing whitespace f
-00005c70: 6f72 206d 756c 7469 2d6c 696e 6520 6665  or multi-line fe
-00005c80: 6174 7572 6573 2c20 736f 2074 6865 2060  atures, so the `
-00005c90: 736b 6269 6f2e 696f 2e66 6f72 6d61 742e  skbio.io.format.
-00005ca0: 7374 6f63 6b68 6f6c 6d60 2072 6561 6465  stockholm` reade
-00005cb0: 7220 6e6f 7720 6164 6473 2061 2073 696e  r now adds a sin
-00005cc0: 676c 6520 7370 6163 6520 7768 656e 2063  gle space when c
-00005cd0: 6f6e 6361 7465 6e61 7469 6e67 2066 6561  oncatenating fea
-00005ce0: 7475 7265 2074 6578 7420 7769 7468 6f75  ture text withou
-00005cf0: 7420 7472 6169 6c69 6e67 2077 6869 7465  t trailing white
-00005d00: 7370 6163 6520 746f 2061 766f 6964 206a  space to avoid j
-00005d10: 6f69 6e69 6e67 2077 6f72 6473 2074 6f67  oining words tog
-00005d20: 6574 6865 722e 204d 756c 7469 2d6c 696e  ether. Multi-lin
-00005d30: 6520 7472 6565 7320 7374 6f72 6564 2061  e trees stored a
-00005d40: 7320 4746 206d 6574 6164 6174 6120 6172  s GF metadata ar
-00005d50: 6520 636f 6e63 6174 656e 6174 6564 2061  e concatenated a
-00005d60: 7320 7468 6579 2061 7070 6561 7220 696e  s they appear in
-00005d70: 2074 6865 2066 696c 653b 2061 2073 7061   the file; a spa
-00005d80: 6365 2069 7320 6e6f 7420 6164 6465 6420  ce is not added 
-00005d90: 7768 656e 2063 6f6e 6361 7465 6e61 7469  when concatenati
-00005da0: 6e67 2e20 285b 2331 3332 385d 2868 7474  ng. ([#1328](htt
-00005db0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-00005dc0: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-00005dd0: 696f 2f69 7373 7565 732f 3133 3238 2929  io/issues/1328))
-00005de0: 0a2a 2046 6978 6564 2062 7567 2077 6865  .* Fixed bug whe
-00005df0: 6e20 7573 696e 6720 6053 6571 7565 6e63  n using `Sequenc
-00005e00: 652e 6974 6572 5f6b 6d65 7273 6020 6f6e  e.iter_kmers` on
-00005e10: 2065 6d70 7479 2060 5365 7175 656e 6365   empty `Sequence
-00005e20: 6020 6f62 6a65 6374 2e20 5072 6576 696f  ` object. Previo
-00005e30: 7573 6c79 2074 6869 7320 7261 6973 6564  usly this raised
-00005e40: 2061 2060 5661 6c75 6545 7272 6f72 602c   a `ValueError`,
-00005e50: 206e 6f77 2069 7420 7265 7475 726e 730a   now it returns.
-00005e60: 616e 2065 6d70 7479 2067 656e 6572 6174  an empty generat
-00005e70: 6f72 2e0a 2a20 4669 7865 6420 6d69 6e6f  or..* Fixed mino
-00005e80: 7220 6275 6720 7768 6572 6520 6164 6469  r bug where addi
-00005e90: 6e67 2073 6571 7565 6e63 6573 2074 6f20  ng sequences to 
-00005ea0: 616e 2065 6d70 7479 2060 5461 6275 6c61  an empty `Tabula
-00005eb0: 724d 5341 6020 7769 7468 204d 5341 2d77  rMSA` with MSA-w
-00005ec0: 6964 6520 6070 6f73 6974 696f 6e61 6c5f  ide `positional_
-00005ed0: 6d65 7461 6461 7461 6020 776f 756c 6420  metadata` would 
-00005ee0: 7265 7375 6c74 2069 6e20 6120 6054 6162  result in a `Tab
-00005ef0: 756c 6172 4d53 4160 206f 626a 6563 7420  ularMSA` object 
-00005f00: 696e 2061 6e20 696e 636f 6e73 6973 7465  in an inconsiste
-00005f10: 6e74 2073 7461 7465 2e20 5468 6973 2063  nt state. This c
-00005f20: 6f75 6c64 2068 6170 7065 6e20 7573 696e  ould happen usin
-00005f30: 6720 6054 6162 756c 6172 4d53 412e 6170  g `TabularMSA.ap
-00005f40: 7065 6e64 6020 6f72 2060 5461 6275 6c61  pend` or `Tabula
-00005f50: 724d 5341 2e65 7874 656e 6460 2e20 5468  rMSA.extend`. Th
-00005f60: 6973 2062 7567 206f 6e6c 7920 6166 6665  is bug only affe
-00005f70: 6374 7320 6120 6054 6162 756c 6172 4d53  cts a `TabularMS
-00005f80: 4160 206f 626a 6563 7420 2a77 6974 686f  A` object *witho
-00005f90: 7574 2a20 7365 7175 656e 6365 7320 7468  ut* sequences th
-00005fa0: 6174 2068 6173 204d 5341 2d77 6964 6520  at has MSA-wide 
-00005fb0: 6070 6f73 6974 696f 6e61 6c5f 6d65 7461  `positional_meta
-00005fc0: 6461 7461 6020 2866 6f72 2065 7861 6d70  data` (for examp
-00005fd0: 6c65 2c20 6054 6162 756c 6172 4d53 4128  le, `TabularMSA(
-00005fe0: 5b5d 2c20 706f 7369 7469 6f6e 616c 5f6d  [], positional_m
-00005ff0: 6574 6164 6174 613d 7b27 636f 6c75 6d6e  etadata={'column
-00006000: 273a 205b 5d7d 2960 292e 0a2a 2060 5472  ': []})`)..* `Tr
-00006010: 6565 4e6f 6465 2e64 6973 7461 6e63 6560  eeNode.distance`
-00006020: 206e 6f77 2068 616e 646c 6573 2074 6865   now handles the
-00006030: 2073 6974 7561 7469 6f6e 2069 6e20 7768   situation in wh
-00006040: 6963 6820 6073 656c 6660 206f 7220 606f  ich `self` or `o
-00006050: 7468 6572 6020 6172 6520 616e 6365 7374  ther` are ancest
-00006060: 6f72 732e 2050 7265 7669 6f73 6c79 2c20  ors. Previosly, 
-00006070: 6120 6e6f 6465 2066 7572 7468 6572 2075  a node further u
-00006080: 7020 7468 6520 7472 6565 2077 6173 2075  p the tree was u
-00006090: 7365 6420 7265 7375 6c74 696e 6720 696e  sed resulting in
-000060a0: 2069 6e66 6c61 7465 6420 6469 7374 616e   inflated distan
-000060b0: 6365 732e 2028 5b23 3830 375d 2868 7474  ces. ([#807](htt
-000060c0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-000060d0: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-000060e0: 696f 2f69 7373 7565 732f 3830 3729 290a  io/issues/807)).
-000060f0: 2a20 6054 7265 654e 6f64 652e 7072 756e  * `TreeNode.prun
-00006100: 6560 2063 616e 206e 6f77 2068 616e 646c  e` can now handl
-00006110: 6520 6120 726f 6f74 2077 6974 6820 6120  e a root with a 
-00006120: 7369 6e67 6c65 2064 6573 6365 6e64 656e  single descenden
-00006130: 742e 2050 7265 7669 6f75 736c 792c 2074  t. Previously, t
-00006140: 6865 2072 6f6f 7420 7761 7320 6967 6e6f  he root was igno
-00006150: 7265 6420 6672 6f6d 2070 6f73 7369 626c  red from possibl
-00006160: 7920 6861 7669 6e67 2061 2073 696e 676c  y having a singl
-00006170: 6520 6465 7363 656e 6465 6e74 2e20 285b  e descendent. ([
-00006180: 2331 3234 375d 2868 7474 7073 3a2f 2f67  #1247](https://g
-00006190: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
-000061a0: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
-000061b0: 7565 732f 3132 3437 2929 0a2a 2050 726f  ues/1247)).* Pro
-000061c0: 7669 6469 6e67 2074 6865 2060 666f 726d  viding the `form
-000061d0: 6174 6020 6b65 7977 6f72 6420 746f 2060  at` keyword to `
-000061e0: 736b 6269 6f2e 696f 2e72 6561 6460 2077  skbio.io.read` w
-000061f0: 6865 6e20 6372 6561 7469 6e67 2061 2067  hen creating a g
-00006200: 656e 6572 6174 6f72 2077 6974 6820 616e  enerator with an
-00006210: 2065 6d70 7479 2066 696c 6520 7769 6c6c   empty file will
-00006220: 206e 6f77 2072 6574 7572 6e20 616e 2065   now return an e
-00006230: 6d70 7479 2067 656e 6572 6174 6f72 2069  mpty generator i
-00006240: 6e73 7465 6164 206f 6620 7261 6973 696e  nstead of raisin
-00006250: 6720 6053 746f 7049 7465 7261 7469 6f6e  g `StopIteration
-00006260: 602e 2028 5b23 3133 3133 5d28 6874 7470  `. ([#1313](http
-00006270: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-00006280: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-00006290: 6f2f 6973 7375 6573 2f31 3331 3329 290a  o/issues/1313)).
-000062a0: 2a20 604f 7264 696e 6174 696f 6e52 6573  * `OrdinationRes
-000062b0: 756c 7473 6020 6973 206e 6f77 2069 6d70  ults` is now imp
-000062c0: 6f72 7461 626c 6520 6672 6f6d 2060 736b  ortable from `sk
-000062d0: 6269 6f60 2061 6e64 2060 736b 6269 6f2e  bio` and `skbio.
-000062e0: 7374 6174 732e 6f72 6469 6e61 7469 6f6e  stats.ordination
-000062f0: 6020 616e 6420 636f 7272 6563 746c 7920  ` and correctly 
-00006300: 6c69 6e6b 6564 2066 726f 6d20 7468 6520  linked from the 
-00006310: 646f 6375 6d65 6e74 6174 696f 6e20 285b  documentation ([
-00006320: 2331 3230 355d 2868 7474 7073 3a2f 2f67  #1205](https://g
-00006330: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
-00006340: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
-00006350: 7565 732f 3132 3035 2929 0a2a 2046 6978  ues/1205)).* Fix
-00006360: 6564 2070 6572 666f 726d 616e 6365 2062  ed performance b
-00006370: 7567 2069 6e20 7061 6972 7769 7365 2061  ug in pairwise a
-00006380: 6c69 676e 6572 7320 7265 7375 6c74 696e  ligners resultin
-00006390: 6720 696e 2031 3030 7820 776f 7273 6520  g in 100x worse 
-000063a0: 7065 7266 6f72 6d61 6e63 6520 7468 616e  performance than
-000063b0: 2069 6e20 302e 322e 342e 0a0a 2323 2320   in 0.2.4...### 
-000063c0: 4465 7072 6563 6174 6564 2066 756e 6374  Deprecated funct
-000063d0: 696f 6e61 6c69 7479 205b 7374 6162 6c65  ionality [stable
-000063e0: 5d0a 2a20 4465 7072 6563 6174 6564 2075  ].* Deprecated u
-000063f0: 7365 206f 6620 7468 6520 7465 726d 2022  se of the term "
-00006400: 6e6f 6e2d 6465 6765 6e65 7261 7465 222c  non-degenerate",
-00006410: 2069 6e20 6661 766f 7220 6f66 2022 6465   in favor of "de
-00006420: 6669 6e69 7465 222e 2060 4772 616d 6d61  finite". `Gramma
-00006430: 7265 6453 6571 7565 6e63 652e 6e6f 6e64  redSequence.nond
-00006440: 6567 656e 6572 6174 655f 6368 6172 7360  egenerate_chars`
-00006450: 2c20 6047 7261 6d6d 6172 6564 5365 7175  , `GrammaredSequ
-00006460: 656e 6365 2e6e 6f6e 6465 6765 6e65 7261  ence.nondegenera
-00006470: 7465 7360 2c20 616e 6420 6047 7261 6d6d  tes`, and `Gramm
-00006480: 6172 6564 5365 7175 656e 6365 2e68 6173  aredSequence.has
-00006490: 5f6e 6f6e 6465 6765 6e65 7261 7465 7360  _nondegenerates`
-000064a0: 2068 6176 6520 6265 656e 2072 656e 616d   have been renam
-000064b0: 6564 2074 6f20 6047 7261 6d6d 6172 6564  ed to `Grammared
-000064c0: 5365 7175 656e 6365 2e64 6566 696e 6974  Sequence.definit
-000064d0: 655f 6368 6172 7360 2c20 6047 7261 6d6d  e_chars`, `Gramm
-000064e0: 6172 6564 5365 7175 656e 6365 2e64 6566  aredSequence.def
-000064f0: 696e 6974 6573 602c 2061 6e64 2060 4772  inites`, and `Gr
-00006500: 616d 6d61 7265 6453 6571 7565 6e63 652e  ammaredSequence.
-00006510: 6861 735f 6465 6669 6e69 7465 7360 2c20  has_definites`, 
-00006520: 7265 7370 6563 7469 7665 6c79 2e20 5468  respectively. Th
-00006530: 6520 6f6c 6420 6e61 6d65 7320 7769 6c6c  e old names will
-00006540: 2062 6520 7265 6d6f 7665 6420 696e 2073   be removed in s
-00006550: 6369 6b69 742d 6269 6f20 302e 352e 322e  cikit-bio 0.5.2.
-00006560: 2052 656c 6576 616e 7420 6166 6665 6374   Relevant affect
-00006570: 6564 2070 7562 6c69 6320 636c 6173 7365  ed public classe
-00006580: 7320 696e 636c 7564 6520 6047 7261 6d6d  s include `Gramm
-00006590: 6172 6564 5365 7175 656e 6365 602c 2060  aredSequence`, `
-000065a0: 444e 4160 2c20 6052 4e41 602c 2061 6e64  DNA`, `RNA`, and
-000065b0: 2060 5072 6f74 6569 6e60 2e0a 0a23 2323   `Protein`...###
-000065c0: 2044 6570 7265 6361 7465 6420 6675 6e63   Deprecated func
-000065d0: 7469 6f6e 616c 6974 7920 5b65 7870 6572  tionality [exper
-000065e0: 696d 656e 7461 6c5d 0a2a 2044 6570 7265  imental].* Depre
-000065f0: 6361 7465 6420 6675 6e63 7469 6f6e 2060  cated function `
-00006600: 736b 6269 6f2e 7574 696c 2e63 7265 6174  skbio.util.creat
-00006610: 655f 6469 7260 2e20 5468 6973 2066 756e  e_dir`. This fun
-00006620: 6374 696f 6e20 7769 6c6c 2062 6520 7265  ction will be re
-00006630: 6d6f 7665 6420 696e 2073 6369 6b69 742d  moved in scikit-
-00006640: 6269 6f20 302e 352e 312e 2050 6c65 6173  bio 0.5.1. Pleas
-00006650: 6520 7573 6520 7468 6520 5079 7468 6f6e  e use the Python
-00006660: 2073 7461 6e64 6172 6420 6c69 6272 6172   standard librar
-00006670: 790a 6675 6e63 7469 6f6e 616c 6974 7920  y.functionality 
-00006680: 6465 7363 7269 6265 6420 5b68 6572 655d  described [here]
-00006690: 2868 7474 7073 3a2f 2f64 6f63 732e 7079  (https://docs.py
-000066a0: 7468 6f6e 2e6f 7267 2f32 2f6c 6962 7261  thon.org/2/libra
-000066b0: 7279 2f6f 732e 6874 6d6c 236f 732e 6d61  ry/os.html#os.ma
-000066c0: 6b65 6469 7273 292e 2028 5b23 3833 335d  kedirs). ([#833]
-000066d0: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-000066e0: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
-000066f0: 6974 2d62 696f 2f69 7373 7565 732f 3833  it-bio/issues/83
-00006700: 3329 290a 2a20 4465 7072 6563 6174 6564  3)).* Deprecated
-00006710: 2066 756e 6374 696f 6e20 6073 6b62 696f   function `skbio
-00006720: 2e75 7469 6c2e 7265 6d6f 7665 5f66 696c  .util.remove_fil
-00006730: 6573 602e 2054 6869 7320 6675 6e63 7469  es`. This functi
-00006740: 6f6e 2077 696c 6c20 6265 2072 656d 6f76  on will be remov
-00006750: 6564 2069 6e20 7363 696b 6974 2d62 696f  ed in scikit-bio
-00006760: 2030 2e35 2e31 2e20 506c 6561 7365 2075   0.5.1. Please u
-00006770: 7365 2074 6865 2050 7974 686f 6e20 7374  se the Python st
-00006780: 616e 6461 7264 206c 6962 7261 7279 0a66  andard library.f
-00006790: 756e 6374 696f 6e61 6c69 7479 2064 6573  unctionality des
-000067a0: 6372 6962 6564 205b 6865 7265 5d28 6874  cribed [here](ht
-000067b0: 7470 733a 2f2f 646f 6373 2e70 7974 686f  tps://docs.pytho
-000067c0: 6e2e 6f72 672f 322f 6c69 6272 6172 792f  n.org/2/library/
-000067d0: 6f73 2e68 746d 6c23 6f73 2e72 656d 6f76  os.html#os.remov
-000067e0: 6529 2e20 285b 2338 3333 5d28 6874 7470  e). ([#833](http
-000067f0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-00006800: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-00006810: 6f2f 6973 7375 6573 2f38 3333 2929 0a2a  o/issues/833)).*
-00006820: 2044 6570 7265 6361 7465 6420 6675 6e63   Deprecated func
-00006830: 7469 6f6e 2060 736b 6269 6f2e 7574 696c  tion `skbio.util
-00006840: 2e69 735f 6361 7361 7661 5f76 3138 305f  .is_casava_v180_
-00006850: 6f72 5f6c 6174 6572 602e 2054 6869 7320  or_later`. This 
-00006860: 6675 6e63 7469 6f6e 2077 696c 6c20 6265  function will be
-00006870: 2072 656d 6f76 6564 2069 6e20 302e 352e   removed in 0.5.
-00006880: 312e 2046 756e 6374 696f 6e61 6c69 7479  1. Functionality
-00006890: 206d 6f76 6564 2074 6f20 4641 5354 5120   moved to FASTQ 
-000068a0: 736e 6966 6665 722e 0a28 5b23 3833 335d  sniffer..([#833]
-000068b0: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-000068c0: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
-000068d0: 6974 2d62 696f 2f69 7373 7565 732f 3833  it-bio/issues/83
-000068e0: 3329 290a 0a23 2323 204d 6973 6365 6c6c  3))..### Miscell
-000068f0: 616e 656f 7573 0a2a 2057 6865 6e20 696e  aneous.* When in
-00006900: 7374 616c 6c69 6e67 2073 6369 6b69 742d  stalling scikit-
-00006910: 6269 6f20 7669 6120 6070 6970 602c 206e  bio via `pip`, n
-00006920: 756d 7079 206d 7573 7420 6e6f 7720 6265  umpy must now be
-00006930: 2069 6e73 7461 6c6c 6564 2066 6972 7374   installed first
-00006940: 2028 5b23 3132 3936 5d28 6874 7470 733a   ([#1296](https:
-00006950: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
-00006960: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
-00006970: 6973 7375 6573 2f31 3239 3629 290a 0a23  issues/1296))..#
-00006980: 2320 5665 7273 696f 6e20 302e 342e 3220  # Version 0.4.2 
-00006990: 2832 3031 362d 3032 2d31 3729 0a0a 4d69  (2016-02-17)..Mi
-000069a0: 6e6f 7220 6d61 696e 7465 6e61 6e63 6520  nor maintenance 
-000069b0: 7265 6c65 6173 652e 202a 2a54 6869 7320  release. **This 
-000069c0: 6973 2074 6865 206c 6173 7420 5079 7468  is the last Pyth
-000069d0: 6f6e 2032 2e37 2063 6f6d 7061 7469 626c  on 2.7 compatibl
-000069e0: 6520 7265 6c65 6173 652e 2046 7574 7572  e release. Futur
-000069f0: 6520 7363 696b 6974 2d62 696f 2072 656c  e scikit-bio rel
-00006a00: 6561 7365 7320 7769 6c6c 206f 6e6c 7920  eases will only 
-00006a10: 7375 7070 6f72 7420 5079 7468 6f6e 2033  support Python 3
-00006a20: 2e2a 2a0a 0a23 2323 2046 6561 7475 7265  .**..### Feature
-00006a30: 730a 2a20 4164 6465 6420 6073 6b62 696f  s.* Added `skbio
-00006a40: 2e74 7265 652e 5472 6565 4e6f 6465 2e62  .tree.TreeNode.b
-00006a50: 6966 7572 6361 7465 6020 666f 7220 636f  ifurcate` for co
-00006a60: 6e76 6572 7469 6e67 206d 756c 7469 6675  nverting multifu
-00006a70: 7263 6174 696e 6720 7472 6565 7320 696e  rcating trees in
-00006a80: 746f 2062 6966 7572 6361 7469 6e67 2074  to bifurcating t
-00006a90: 7265 6573 2e20 285b 2338 3936 5d28 6874  rees. ([#896](ht
-00006aa0: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-00006ab0: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
-00006ac0: 6269 6f2f 6973 7375 6573 2f38 3936 2929  bio/issues/896))
-00006ad0: 0a2a 2041 6464 6564 2060 736b 6269 6f2e  .* Added `skbio.
-00006ae0: 696f 2e66 6f72 6d61 742e 7374 6f63 6b68  io.format.stockh
-00006af0: 6f6c 6d60 2066 6f72 2072 6561 6469 6e67  olm` for reading
-00006b00: 2053 746f 636b 686f 6c6d 2066 696c 6573   Stockholm files
-00006b10: 2069 6e74 6f20 6120 6054 6162 756c 6172   into a `Tabular
-00006b20: 4d53 4160 2061 6e64 2077 7269 7469 6e67  MSA` and writing
-00006b30: 2066 726f 6d20 6120 6054 6162 756c 6172   from a `Tabular
-00006b40: 4d53 4160 2e20 285b 2339 3637 5d28 6874  MSA`. ([#967](ht
-00006b50: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-00006b60: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
-00006b70: 6269 6f2f 6973 7375 6573 2f39 3637 2929  bio/issues/967))
-00006b80: 0a2a 2073 6369 6b69 742d 6269 6f20 6053  .* scikit-bio `S
-00006b90: 6571 7565 6e63 6560 206f 626a 6563 7473  equence` objects
-00006ba0: 2068 6176 6520 6265 7474 6572 2063 6f6d   have better com
-00006bb0: 7061 7469 6269 6c69 7479 2077 6974 6820  patibility with 
-00006bc0: 6e75 6d70 792e 2046 6f72 2065 7861 6d70  numpy. For examp
-00006bd0: 6c65 2c20 6361 6c6c 696e 6720 606e 702e  le, calling `np.
-00006be0: 6173 6172 7261 7928 7365 7175 656e 6365  asarray(sequence
-00006bf0: 2960 206e 6f77 2063 6f6e 7665 7274 7320  )` now converts 
-00006c00: 7468 6520 7365 7175 656e 6365 2074 6f20  the sequence to 
-00006c10: 6120 6e75 6d70 7920 6172 7261 7920 6f66  a numpy array of
-00006c20: 2063 6861 7261 6374 6572 7320 2874 6865   characters (the
-00006c30: 2073 616d 6520 6173 2063 616c 6c69 6e67   same as calling
-00006c40: 2060 7365 7175 656e 6365 2e76 616c 7565   `sequence.value
-00006c50: 7360 292e 0a2a 2041 6464 6564 2060 736b  s`)..* Added `sk
-00006c60: 6269 6f2e 7365 7175 656e 6365 2e64 6973  bio.sequence.dis
-00006c70: 7461 6e63 6560 2073 7562 7061 636b 6167  tance` subpackag
-00006c80: 6520 666f 7220 636f 6d70 7574 696e 6720  e for computing 
-00006c90: 6469 7374 616e 6365 7320 6265 7477 6565  distances betwee
-00006ca0: 6e20 7363 696b 6974 2d62 696f 2060 5365  n scikit-bio `Se
-00006cb0: 7175 656e 6365 6020 6f62 6a65 6374 7320  quence` objects 
-00006cc0: 285b 2339 3133 5d28 6874 7470 733a 2f2f  ([#913](https://
-00006cd0: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-00006ce0: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
-00006cf0: 7375 6573 2f39 3133 2929 0a2a 2041 6464  sues/913)).* Add
-00006d00: 6564 2060 6073 6b62 696f 2e73 6571 7565  ed ``skbio.seque
-00006d10: 6e63 652e 4772 616d 6d61 7265 6453 6571  nce.GrammaredSeq
-00006d20: 7565 6e63 6560 602c 2077 6869 6368 2063  uence``, which c
-00006d30: 616e 2062 6520 696e 6865 7269 7465 6420  an be inherited 
-00006d40: 6672 6f6d 2074 6f20 6372 6561 7465 2067  from to create g
-00006d50: 7261 6d6d 6172 6564 2073 6571 7565 6e63  rammared sequenc
-00006d60: 6573 2077 6974 6820 6375 7374 6f6d 2061  es with custom a
-00006d70: 6c70 6861 6265 7473 2028 652e 672e 2c20  lphabets (e.g., 
-00006d80: 666f 7220 7573 6520 7769 7468 2054 6162  for use with Tab
-00006d90: 756c 6172 4d53 4129 2028 5b23 3131 3735  ularMSA) ([#1175
-00006da0: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
-00006db0: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
-00006dc0: 6b69 742d 6269 6f2f 6973 7375 6573 2f31  kit-bio/issues/1
-00006dd0: 3137 3529 290a 2a20 4164 6465 6420 6060  175)).* Added ``
-00006de0: 736b 6269 6f2e 7574 696c 2e63 6c61 7373  skbio.util.class
-00006df0: 7072 6f70 6572 7479 6060 2064 6563 6f72  property`` decor
-00006e00: 6174 6f72 0a0a 2323 2320 4261 636b 7761  ator..### Backwa
-00006e10: 7264 2d69 6e63 6f6d 7061 7469 626c 6520  rd-incompatible 
-00006e20: 6368 616e 6765 7320 5b73 7461 626c 655d  changes [stable]
-00006e30: 0a2a 2057 6865 6e20 736e 6966 6669 6e67  .* When sniffing
-00006e40: 206f 7220 7265 6164 696e 6720 6120 6669   or reading a fi
-00006e50: 6c65 2028 6073 6b62 696f 2e69 6f2e 736e  le (`skbio.io.sn
-00006e60: 6966 6660 2c20 6073 6b62 696f 2e69 6f2e  iff`, `skbio.io.
-00006e70: 7265 6164 602c 206f 7220 7468 6520 6f62  read`, or the ob
-00006e80: 6a65 6374 2d6f 7269 656e 7465 6420 602e  ject-oriented `.
-00006e90: 7265 6164 2829 6020 696e 7465 7266 6163  read()` interfac
-00006ea0: 6529 2c20 7061 7373 696e 6720 606e 6577  e), passing `new
-00006eb0: 6c69 6e65 6020 6173 2061 206b 6579 776f  line` as a keywo
-00006ec0: 7264 2061 7267 756d 656e 7420 746f 2060  rd argument to `
-00006ed0: 736b 6269 6f2e 696f 2e6f 7065 6e60 206e  skbio.io.open` n
-00006ee0: 6f77 2072 6169 7365 7320 6120 6054 7970  ow raises a `Typ
-00006ef0: 6545 7272 6f72 602e 2054 6869 7320 6261  eError`. This ba
-00006f00: 636b 7761 7264 2d69 6e63 6f6d 7061 7469  ckward-incompati
-00006f10: 626c 6520 6368 616e 6765 2074 6f20 6120  ble change to a 
-00006f20: 7374 6162 6c65 2041 5049 2069 7320 6e65  stable API is ne
-00006f30: 6365 7373 6172 7920 6265 6361 7573 6520  cessary because 
-00006f40: 6974 2066 6978 6573 2061 2062 7567 2028  it fixes a bug (
-00006f50: 6d6f 7265 2064 6574 6169 6c73 2069 6e20  more details in 
-00006f60: 6275 6720 6669 7820 7365 6374 696f 6e20  bug fix section 
-00006f70: 6265 6c6f 7729 2e0a 2a20 5768 656e 2072  below)..* When r
-00006f80: 6561 6469 6e67 2061 2046 4153 5451 206f  eading a FASTQ o
-00006f90: 7220 5153 4551 2066 696c 6520 616e 6420  r QSEQ file and 
-00006fa0: 7061 7373 696e 6720 6076 6172 6961 6e74  passing `variant
-00006fb0: 3d27 736f 6c65 7861 2760 2c20 6056 616c  ='solexa'`, `Val
-00006fc0: 7565 4572 726f 7260 2069 7320 6e6f 7720  ueError` is now 
-00006fd0: 7261 6973 6564 2069 6e73 7465 6164 206f  raised instead o
-00006fe0: 6620 604e 6f74 496d 706c 656d 656e 7465  f `NotImplemente
-00006ff0: 6445 7272 6f72 602e 2054 6869 7320 6261  dError`. This ba
-00007000: 636b 7761 7264 2d69 6e63 6f6d 7061 7469  ckward-incompati
-00007010: 626c 6520 6368 616e 6765 2074 6f20 6120  ble change to a 
-00007020: 7374 6162 6c65 2041 5049 2069 7320 6e65  stable API is ne
-00007030: 6365 7373 6172 7920 746f 2061 766f 6964  cessary to avoid
-00007040: 2063 7265 6174 696e 6720 6120 7370 696e   creating a spin
-00007050: 2d6c 6f63 6b65 6420 7072 6f63 6573 7320  -locked process 
-00007060: 6475 6520 746f 205b 6120 6275 6720 696e  due to [a bug in
-00007070: 2050 7974 686f 6e5d 2868 7474 7073 3a2f   Python](https:/
-00007080: 2f62 7567 732e 7079 7468 6f6e 2e6f 7267  /bugs.python.org
-00007090: 2f69 7373 7565 3235 3738 3629 2e20 5365  /issue25786). Se
-000070a0: 6520 5b23 3132 3536 5d28 6874 7470 733a  e [#1256](https:
-000070b0: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
-000070c0: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
-000070d0: 6973 7375 6573 2f31 3235 3629 2066 6f72  issues/1256) for
-000070e0: 2064 6574 6169 6c73 2e20 5468 6973 2063   details. This c
-000070f0: 6861 6e67 6520 6973 2074 656d 706f 7261  hange is tempora
-00007100: 7279 2061 6e64 2077 696c 6c20 6265 2072  ry and will be r
-00007110: 6576 6572 7465 6420 746f 2060 4e6f 7449  everted to `NotI
-00007120: 6d70 6c65 6d65 6e74 6564 4572 726f 7260  mplementedError`
-00007130: 2077 6865 6e20 7468 6520 6275 6720 6973   when the bug is
-00007140: 2066 6978 6564 2069 6e20 5079 7468 6f6e   fixed in Python
-00007150: 2e0a 0a23 2323 2042 6163 6b77 6172 642d  ...### Backward-
-00007160: 696e 636f 6d70 6174 6962 6c65 2063 6861  incompatible cha
-00007170: 6e67 6573 205b 6578 7065 7269 6d65 6e74  nges [experiment
-00007180: 616c 5d0a 2a20 6073 6b62 696f 2e69 6f2e  al].* `skbio.io.
-00007190: 666f 726d 6174 2e67 656e 6261 6e6b 603a  format.genbank`:
-000071a0: 2057 6865 6e20 7265 6164 696e 6720 4765   When reading Ge
-000071b0: 6e42 616e 6b20 6669 6c65 732c 2074 6865  nBank files, the
-000071c0: 2064 6174 6520 6669 656c 6420 6f66 2074   date field of t
-000071d0: 6865 204c 4f43 5553 206c 696e 6520 6973  he LOCUS line is
-000071e0: 206e 6f20 6c6f 6e67 6572 2070 6172 7365   no longer parse
-000071f0: 6420 696e 746f 2061 2060 6461 7465 7469  d into a `dateti
-00007200: 6d65 2e64 6174 6574 696d 6560 206f 626a  me.datetime` obj
-00007210: 6563 7420 616e 6420 6973 206c 6566 7420  ect and is left 
-00007220: 6173 2061 2073 7472 696e 672e 2057 6865  as a string. Whe
-00007230: 6e20 7772 6974 696e 6720 4765 6e42 616e  n writing GenBan
-00007240: 6b20 6669 6c65 732c 2074 6865 206c 6f63  k files, the loc
-00007250: 7573 2064 6174 6520 6d65 7461 6461 7461  us date metadata
-00007260: 2069 7320 6578 7065 6374 6564 2074 6f20   is expected to 
-00007270: 6265 2061 2073 7472 696e 6720 696e 7374  be a string inst
-00007280: 6561 6420 6f66 2061 2060 6461 7465 7469  ead of a `dateti
-00007290: 6d65 2e64 6174 6574 696d 6560 206f 626a  me.datetime` obj
-000072a0: 6563 7420 285b 2331 3135 335d 2868 7474  ect ([#1153](htt
-000072b0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-000072c0: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-000072d0: 696f 2f69 7373 7565 732f 3131 3533 2929  io/issues/1153))
-000072e0: 0a2a 2060 5365 7175 656e 6365 2e64 6973  .* `Sequence.dis
-000072f0: 7461 6e63 6560 206e 6f77 2063 6f6e 7665  tance` now conve
-00007300: 7274 7320 7468 6520 696e 7075 7420 7365  rts the input se
-00007310: 7175 656e 6365 2028 606f 7468 6572 6029  quence (`other`)
-00007320: 2074 6f20 6974 7320 7479 7065 2062 6566   to its type bef
-00007330: 6f72 6520 7061 7373 696e 6720 626f 7468  ore passing both
-00007340: 2073 6571 7565 6e63 6573 2074 6f20 606d   sequences to `m
-00007350: 6574 7269 6360 2e20 5072 6576 696f 7573  etric`. Previous
-00007360: 2062 6568 6176 696f 7220 7761 7320 746f   behavior was to
-00007370: 2061 6c77 6179 7320 636f 6e76 6572 7420   always convert 
-00007380: 746f 2060 5365 7175 656e 6365 602e 0a0a  to `Sequence`...
-00007390: 2323 2320 4275 6720 6669 7865 730a 2a20  ### Bug fixes.* 
-000073a0: 4669 7865 6420 6275 6720 7768 656e 2075  Fixed bug when u
-000073b0: 7369 6e67 2060 5365 7175 656e 6365 2e64  sing `Sequence.d
-000073c0: 6973 7461 6e63 6560 206f 7220 6044 6973  istance` or `Dis
-000073d0: 7461 6e63 654d 6174 7269 782e 6672 6f6d  tanceMatrix.from
-000073e0: 5f69 7465 7261 626c 6560 2074 6f20 636f  _iterable` to co
-000073f0: 6d70 7574 6520 6469 7374 616e 6365 7320  mpute distances 
-00007400: 6265 7477 6565 6e20 6053 6571 7565 6e63  between `Sequenc
-00007410: 6560 206f 626a 6563 7473 2077 6974 6820  e` objects with 
-00007420: 6469 6666 6572 696e 6720 606d 6574 6164  differing `metad
-00007430: 6174 6160 2f60 706f 7369 7469 6f6e 616c  ata`/`positional
-00007440: 5f6d 6574 6164 6174 6160 2061 6e64 2070  _metadata` and p
-00007450: 6173 7369 6e67 2060 6d65 7472 6963 3d73  assing `metric=s
-00007460: 6369 7079 2e73 7061 7469 616c 2e64 6973  cipy.spatial.dis
-00007470: 7461 6e63 652e 6861 6d6d 696e 6760 2028  tance.hamming` (
-00007480: 5b23 3132 3534 5d28 6874 7470 733a 2f2f  [#1254](https://
-00007490: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-000074a0: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
-000074b0: 7375 6573 2f31 3235 3429 290a 2a20 4669  sues/1254)).* Fi
-000074c0: 7865 6420 7065 7266 6f72 6d61 6e63 6520  xed performance 
-000074d0: 6275 6720 7768 656e 2063 6f6d 7075 7469  bug when computi
-000074e0: 6e67 2048 616d 6d69 6e67 2064 6973 7461  ng Hamming dista
-000074f0: 6e63 6573 2062 6574 7765 656e 2060 5365  nces between `Se
-00007500: 7175 656e 6365 6020 6f62 6a65 6374 7320  quence` objects 
-00007510: 696e 2060 4469 7374 616e 6365 4d61 7472  in `DistanceMatr
-00007520: 6978 2e66 726f 6d5f 6974 6572 6162 6c65  ix.from_iterable
-00007530: 6020 285b 2331 3235 305d 2868 7474 7073  ` ([#1250](https
-00007540: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
-00007550: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
-00007560: 2f69 7373 7565 732f 3132 3530 2929 0a2a  /issues/1250)).*
-00007570: 2043 6861 6e67 6564 2060 736b 6269 6f2e   Changed `skbio.
-00007580: 7374 6174 732e 636f 6d70 6f73 6974 696f  stats.compositio
-00007590: 6e2e 6d75 6c74 6970 6c69 6361 7469 7665  n.multiplicative
-000075a0: 5f72 6570 6c61 6365 6d65 6e74 6020 746f  _replacement` to
-000075b0: 2072 6169 7365 2061 6e20 6572 726f 7220   raise an error 
-000075c0: 7768 656e 6576 6572 2061 206c 6172 6765  whenever a large
-000075d0: 2076 616c 7565 206f 6620 6064 656c 7461   value of `delta
-000075e0: 6020 6973 2063 686f 7365 6e20 285b 2331  ` is chosen ([#1
-000075f0: 3234 315d 2868 7474 7073 3a2f 2f67 6974  241](https://git
-00007600: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
-00007610: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
-00007620: 732f 3132 3431 2929 0a2a 2057 6865 6e20  s/1241)).* When 
-00007630: 736e 6966 6669 6e67 206f 7220 7265 6164  sniffing or read
-00007640: 696e 6720 6120 6669 6c65 2028 6073 6b62  ing a file (`skb
-00007650: 696f 2e69 6f2e 736e 6966 6660 2c20 6073  io.io.sniff`, `s
-00007660: 6b62 696f 2e69 6f2e 7265 6164 602c 206f  kbio.io.read`, o
-00007670: 7220 7468 6520 6f62 6a65 6374 2d6f 7269  r the object-ori
-00007680: 656e 7465 6420 602e 7265 6164 2829 6020  ented `.read()` 
-00007690: 696e 7465 7266 6163 6529 2c20 7061 7373  interface), pass
-000076a0: 696e 6720 606e 6577 6c69 6e65 6020 6173  ing `newline` as
-000076b0: 2061 206b 6579 776f 7264 2061 7267 756d   a keyword argum
-000076c0: 656e 7420 746f 2060 736b 6269 6f2e 696f  ent to `skbio.io
-000076d0: 2e6f 7065 6e60 206e 6f77 2072 6169 7365  .open` now raise
-000076e0: 7320 6120 6054 7970 6545 7272 6f72 602e  s a `TypeError`.
-000076f0: 2054 6865 2066 696c 6520 666f 726d 6174   The file format
-00007700: 2773 2060 6e65 776c 696e 6560 2063 6861  's `newline` cha
-00007710: 7261 6374 6572 2077 696c 6c20 6265 2075  racter will be u
-00007720: 7365 6420 7768 656e 206f 7065 6e69 6e67  sed when opening
-00007730: 2074 6865 2066 696c 652e 2050 7265 7669   the file. Previ
-00007740: 6f75 7320 6265 6861 7669 6f72 2061 6c6c  ous behavior all
-00007750: 6f77 6564 206f 7665 7272 6964 696e 6720  owed overriding 
-00007760: 7468 6520 666f 726d 6174 2773 2060 6e65  the format's `ne
-00007770: 776c 696e 6560 2063 6861 7261 6374 6572  wline` character
-00007780: 2062 7574 2074 6869 7320 636f 756c 6420   but this could 
-00007790: 6361 7573 6520 6973 7375 6573 2077 6974  cause issues wit
-000077a0: 6820 7265 6164 6572 7320 7468 6174 2061  h readers that a
-000077b0: 7373 756d 6520 6e65 776c 696e 6520 6368  ssume newline ch
-000077c0: 6172 6163 7465 7273 2061 7265 2074 686f  aracters are tho
-000077d0: 7365 2064 6566 696e 6564 2062 7920 7468  se defined by th
-000077e0: 6520 6669 6c65 2066 6f72 6d61 7420 2877  e file format (w
-000077f0: 6869 6368 2069 7320 616e 2065 6e74 6972  hich is an entir
-00007800: 656c 7920 7265 6173 6f6e 6162 6c65 2061  ely reasonable a
-00007810: 7373 756d 7074 696f 6e29 2e20 5468 6973  ssumption). This
-00007820: 2062 7567 2069 7320 7665 7279 2075 6e6c   bug is very unl
-00007830: 696b 656c 7920 746f 2068 6176 6520 7375  ikely to have su
-00007840: 7266 6163 6564 2069 6e20 7072 6163 7469  rfaced in practi
-00007850: 6365 2061 7320 7468 6520 6465 6661 756c  ce as the defaul
-00007860: 7420 606e 6577 6c69 6e65 6020 6265 6861  t `newline` beha
-00007870: 7669 6f72 2069 7320 2a75 6e69 7665 7273  vior is *univers
-00007880: 616c 206e 6577 6c69 6e65 7320 6d6f 6465  al newlines mode
-00007890: 2a2e 0a2a 2044 4e41 2c20 524e 412c 2061  *..* DNA, RNA, a
-000078a0: 6e64 2050 726f 7465 696e 2061 7265 206e  nd Protein are n
-000078b0: 6f20 6c6f 6e67 6572 2069 6e68 6572 6974  o longer inherit
-000078c0: 6162 6c65 2062 6563 6175 7365 2074 6865  able because the
-000078d0: 7920 6173 7375 6d65 2061 6e20 4955 5041  y assume an IUPA
-000078e0: 4320 616c 7068 6162 6574 2e0a 2a20 6044  C alphabet..* `D
-000078f0: 6973 7461 6e63 654d 6174 7269 7860 2063  istanceMatrix` c
-00007900: 6f6e 7374 7275 6374 6f72 2070 726f 7669  onstructor provi
-00007910: 6465 7320 6d6f 7265 2069 6e66 6f72 6d61  des more informa
-00007920: 7469 7665 2065 7272 6f72 206d 6573 7361  tive error messa
-00007930: 6765 2077 6865 6e20 6461 7461 2063 6f6e  ge when data con
-00007940: 7461 696e 7320 4e61 4e73 2028 5b23 3132  tains NaNs ([#12
-00007950: 3736 5d28 6874 7470 733a 2f2f 6769 7468  76](https://gith
-00007960: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
-00007970: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
-00007980: 2f31 3237 3629 290a 0a23 2323 204d 6973  /1276))..### Mis
-00007990: 6365 6c6c 616e 656f 7573 0a2a 2057 6172  cellaneous.* War
-000079a0: 6e69 6e67 7320 7261 6973 6564 2062 7920  nings raised by 
-000079b0: 7363 696b 6974 2d62 696f 206e 6f77 2073  scikit-bio now s
-000079c0: 6861 7265 2061 2063 6f6d 6d6f 6e20 7375  hare a common su
-000079d0: 6263 6c61 7373 2060 6073 6b62 696f 2e75  bclass ``skbio.u
-000079e0: 7469 6c2e 536b 6269 6f57 6172 6e69 6e67  til.SkbioWarning
-000079f0: 6060 2e0a 0a23 2320 5665 7273 696f 6e20  ``...## Version 
-00007a00: 302e 342e 3120 2832 3031 352d 3132 2d30  0.4.1 (2015-12-0
-00007a10: 3929 0a0a 2323 2320 4665 6174 7572 6573  9)..### Features
-00007a20: 0a2a 2054 6865 2060 6054 6162 756c 6172  .* The ``Tabular
-00007a30: 4d53 4160 6020 6f62 6a65 6374 2077 6173  MSA`` object was
-00007a40: 2061 6464 6564 2074 6f20 7265 7072 6573   added to repres
-00007a50: 656e 7420 616e 6420 6f70 6572 6174 6520  ent and operate 
-00007a60: 6f6e 2074 6162 756c 6172 206d 756c 7469  on tabular multi
-00007a70: 706c 6520 7365 7175 656e 6365 2061 6c69  ple sequence ali
-00007a80: 676e 6d65 6e74 732e 2054 6869 7320 7361  gnments. This sa
-00007a90: 7469 7366 6965 7320 5b52 4643 2031 5d28  tisfies [RFC 1](
-00007aa0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-00007ab0: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
-00007ac0: 742d 6269 6f2d 7266 6373 2f62 6c6f 622f  t-bio-rfcs/blob/
-00007ad0: 6d61 7374 6572 2f61 6374 6976 652f 3030  master/active/00
-00007ae0: 312d 7461 6275 6c61 722d 6d73 612e 6d64  1-tabular-msa.md
-00007af0: 292e 2053 6565 2074 6865 2060 6054 6162  ). See the ``Tab
-00007b00: 756c 6172 4d53 4160 6020 646f 6373 2066  ularMSA`` docs f
-00007b10: 6f72 2066 756c 6c20 6465 7461 696c 732e  or full details.
-00007b20: 0a2a 2041 6464 6564 2070 6879 6c6f 6765  .* Added phyloge
-00007b30: 6e65 7469 6320 6469 7665 7273 6974 7920  netic diversity 
-00007b40: 6d65 7472 6963 732c 2069 6e63 6c75 6469  metrics, includi
-00007b50: 6e67 2077 6569 6768 7465 6420 556e 6946  ng weighted UniF
-00007b60: 7261 632c 2075 6e77 6569 6768 7465 6420  rac, unweighted 
-00007b70: 556e 6946 7261 632c 2061 6e64 2046 6169  UniFrac, and Fai
-00007b80: 7468 2773 2050 6879 6c6f 6765 6e65 7469  th's Phylogeneti
-00007b90: 6320 4469 7665 7273 6974 792e 2054 6865  c Diversity. The
-00007ba0: 7365 2061 7265 2061 6363 6573 7369 626c  se are accessibl
-00007bb0: 6520 6173 2060 6073 6b62 696f 2e64 6976  e as ``skbio.div
-00007bc0: 6572 7369 7479 2e62 6574 612e 756e 7765  ersity.beta.unwe
-00007bd0: 6967 6874 6564 5f75 6e69 6672 6163 6060  ighted_unifrac``
-00007be0: 2c20 6060 736b 6269 6f2e 6469 7665 7273  , ``skbio.divers
-00007bf0: 6974 792e 6265 7461 2e77 6569 6768 7465  ity.beta.weighte
-00007c00: 645f 756e 6966 7261 6360 602c 2061 6e64  d_unifrac``, and
-00007c10: 2060 6073 6b62 696f 2e64 6976 6572 7369   ``skbio.diversi
-00007c20: 7479 2e61 6c70 6861 2e66 6169 7468 5f70  ty.alpha.faith_p
-00007c30: 6460 602c 2072 6573 7065 6374 6976 656c  d``, respectivel
-00007c40: 792e 0a2a 2041 6464 6974 696f 6e20 6f66  y..* Addition of
-00007c50: 2074 6865 2066 756e 6374 696f 6e20 6060   the function ``
-00007c60: 736b 6269 6f2e 6469 7665 7273 6974 792e  skbio.diversity.
-00007c70: 616c 7068 615f 6469 7665 7273 6974 7960  alpha_diversity`
-00007c80: 6020 746f 2073 7570 706f 7274 2061 7070  ` to support app
-00007c90: 6c79 696e 6720 616e 2061 6c70 6861 2064  lying an alpha d
-00007ca0: 6976 6572 7369 7479 206d 6574 7269 6320  iversity metric 
-00007cb0: 746f 206d 756c 7469 706c 6520 7361 6d70  to multiple samp
-00007cc0: 6c65 7320 696e 206f 6e65 2063 616c 6c2e  les in one call.
-00007cd0: 0a2a 2041 6464 6974 696f 6e20 6f66 2074  .* Addition of t
-00007ce0: 6865 2066 756e 6374 696f 6e73 2060 6073  he functions ``s
-00007cf0: 6b62 696f 2e64 6976 6572 7369 7479 2e67  kbio.diversity.g
-00007d00: 6574 5f61 6c70 6861 5f64 6976 6572 7369  et_alpha_diversi
-00007d10: 7479 5f6d 6574 7269 6373 6060 2061 6e64  ty_metrics`` and
-00007d20: 2060 6073 6b62 696f 2e64 6976 6572 7369   ``skbio.diversi
-00007d30: 7479 2e67 6574 5f62 6574 615f 6469 7665  ty.get_beta_dive
-00007d40: 7273 6974 795f 6d65 7472 6963 7360 6020  rsity_metrics`` 
-00007d50: 746f 2073 7570 706f 7274 2064 6973 636f  to support disco
-00007d60: 7665 7279 206f 6620 7468 6520 616c 7068  very of the alph
-00007d70: 6120 616e 6420 6265 7461 2064 6976 6572  a and beta diver
-00007d80: 7369 7479 206d 6574 7269 6373 2069 6d70  sity metrics imp
-00007d90: 6c65 6d65 6e74 6564 2069 6e20 7363 696b  lemented in scik
-00007da0: 6974 2d62 696f 2e0a 2a20 4164 6465 6420  it-bio..* Added 
-00007db0: 6073 6b62 696f 2e73 7461 7473 2e63 6f6d  `skbio.stats.com
-00007dc0: 706f 7369 7469 6f6e 2e61 6e63 6f6d 6020  position.ancom` 
-00007dd0: 6675 6e63 7469 6f6e 2c20 6120 7465 7374  function, a test
-00007de0: 2066 6f72 204f 5455 2064 6966 6665 7265   for OTU differe
-00007df0: 6e74 6961 6c20 6162 756e 6461 6e63 6520  ntial abundance 
-00007e00: 6163 726f 7373 2073 616d 706c 6520 6361  across sample ca
-00007e10: 7465 676f 7269 6573 2e20 285b 2331 3035  tegories. ([#105
-00007e20: 345d 2868 7474 7073 3a2f 2f67 6974 6875  4](https://githu
-00007e30: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
-00007e40: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
-00007e50: 3130 3534 2929 0a2a 2041 6464 6564 2060  1054)).* Added `
-00007e60: 736b 6269 6f2e 696f 2e66 6f72 6d61 742e  skbio.io.format.
-00007e70: 626c 6173 7437 6020 666f 7220 7265 6164  blast7` for read
-00007e80: 696e 6720 424c 4153 542b 206f 7574 7075  ing BLAST+ outpu
-00007e90: 7420 666f 726d 6174 2037 206f 7220 424c  t format 7 or BL
-00007ea0: 4153 5420 6f75 7470 7574 2066 6f72 6d61  AST output forma
-00007eb0: 7420 3920 6669 6c65 7320 696e 746f 2061  t 9 files into a
-00007ec0: 2060 7064 2e44 6174 6146 7261 6d65 602e   `pd.DataFrame`.
-00007ed0: 2028 5b23 3131 3130 5d28 6874 7470 733a   ([#1110](https:
-00007ee0: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
-00007ef0: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
-00007f00: 6973 7375 6573 2f31 3131 3029 290a 2a20  issues/1110)).* 
-00007f10: 4164 6465 6420 6073 6b62 696f 2e44 6973  Added `skbio.Dis
-00007f20: 7369 6d69 6c61 7269 7479 4d61 7472 6978  similarityMatrix
-00007f30: 2e74 6f5f 6461 7461 5f66 7261 6d65 6020  .to_data_frame` 
-00007f40: 6d65 7468 6f64 2066 6f72 2063 7265 6174  method for creat
-00007f50: 696e 6720 6120 6060 7061 6e64 6173 2e44  ing a ``pandas.D
-00007f60: 6174 6146 7261 6d65 6060 2066 726f 6d20  ataFrame`` from 
-00007f70: 6120 6044 6973 7369 6d69 6c61 7269 7479  a `Dissimilarity
-00007f80: 4d61 7472 6978 6020 6f72 2060 4469 7374  Matrix` or `Dist
-00007f90: 616e 6365 4d61 7472 6978 602e 2028 5b23  anceMatrix`. ([#
-00007fa0: 3735 375d 2868 7474 7073 3a2f 2f67 6974  757](https://git
-00007fb0: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
-00007fc0: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
-00007fd0: 732f 3735 3729 290a 2a20 4164 6465 6420  s/757)).* Added 
-00007fe0: 7375 7070 6f72 7420 666f 7220 6f6e 652d  support for one-
-00007ff0: 6469 6d65 6e73 696f 6e61 6c20 7665 6374  dimensional vect
-00008000: 6f72 206f 6620 6469 7373 696d 696c 6172  or of dissimilar
-00008010: 6974 6965 7320 696e 2060 736b 6269 6f2e  ities in `skbio.
-00008020: 7374 6174 732e 6469 7374 616e 6365 2e44  stats.distance.D
-00008030: 6973 7369 6d69 6c61 7269 7479 4d61 7472  issimilarityMatr
-00008040: 6978 600a 636f 6e73 7472 7563 746f 722e  ix`.constructor.
-00008050: 2028 5b23 3632 3430 5d28 6874 7470 733a   ([#6240](https:
-00008060: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
-00008070: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
-00008080: 6973 7375 6573 2f36 3234 2929 0a2a 2041  issues/624)).* A
-00008090: 6464 6564 2060 736b 6269 6f2e 696f 2e66  dded `skbio.io.f
-000080a0: 6f72 6d61 742e 626c 6173 7436 6020 666f  ormat.blast6` fo
-000080b0: 7220 7265 6164 696e 6720 424c 4153 542b  r reading BLAST+
-000080c0: 206f 7574 7075 7420 666f 726d 6174 2036   output format 6
-000080d0: 206f 7220 424c 4153 5420 6f75 7470 7574   or BLAST output
-000080e0: 2066 6f72 6d61 7420 3820 6669 6c65 7320   format 8 files 
-000080f0: 696e 746f 2061 2060 7064 2e44 6174 6146  into a `pd.DataF
-00008100: 7261 6d65 602e 2028 5b23 3131 3130 5d28  rame`. ([#1110](
-00008110: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-00008120: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
-00008130: 742d 6269 6f2f 6973 7375 6573 2f31 3131  t-bio/issues/111
-00008140: 3029 290a 2a20 4164 6465 6420 6069 6e6e  0)).* Added `inn
-00008150: 6572 602c 2060 696c 7260 2c20 6069 6c72  er`, `ilr`, `ilr
-00008160: 5f69 6e76 6020 616e 6420 6063 6c72 5f69  _inv` and `clr_i
-00008170: 6e76 602c 2060 6073 6b62 696f 2e73 7461  nv`, ``skbio.sta
-00008180: 7473 2e63 6f6d 706f 7369 7469 6f6e 6060  ts.composition``
-00008190: 2c20 7768 6963 6820 656e 6162 6c65 7320  , which enables 
-000081a0: 6c69 6e65 6172 2074 7261 6e73 666f 726d  linear transform
-000081b0: 6174 696f 6e73 206f 6e20 636f 6d70 6f73  ations on compos
-000081c0: 6974 696f 6e73 2028 5b23 3839 325d 2868  itions ([#892](h
-000081d0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-000081e0: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
-000081f0: 2d62 696f 2f69 7373 7565 732f 3839 3229  -bio/issues/892)
-00008200: 0a2a 2041 6464 6564 2060 6073 6b62 696f  .* Added ``skbio
-00008210: 2e64 6976 6572 7369 7479 2e61 6c70 6861  .diversity.alpha
-00008220: 2e70 6965 6c6f 755f 6560 6020 6675 6e63  .pielou_e`` func
-00008230: 7469 6f6e 2061 7320 616e 2065 7665 6e6e  tion as an evenn
-00008240: 6573 7320 6d65 7472 6963 206f 6620 616c  ess metric of al
-00008250: 7068 6120 6469 7665 7273 6974 792e 2028  pha diversity. (
-00008260: 5b23 3130 3638 5d28 6874 7470 733a 2f2f  [#1068](https://
-00008270: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
-00008280: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
-00008290: 7375 6573 2f31 3036 3829 290a 2a20 4164  sues/1068)).* Ad
-000082a0: 6465 6420 6074 6f5f 7265 6765 7860 206d  ded `to_regex` m
-000082b0: 6574 686f 6420 746f 2060 736b 6269 6f2e  ethod to `skbio.
-000082c0: 7365 7175 656e 6365 2e5f 6975 7061 635f  sequence._iupac_
-000082d0: 7365 7175 656e 6365 6020 4142 4320 2d20  sequence` ABC - 
-000082e0: 6974 2072 6574 7572 6e73 2061 2072 6567  it returns a reg
-000082f0: 6578 206f 626a 6563 7420 7468 6174 206d  ex object that m
-00008300: 6174 6368 6573 2061 6c6c 206e 6f6e 2d64  atches all non-d
-00008310: 6567 656e 6572 6174 6520 7665 7273 696f  egenerate versio
-00008320: 6e73 206f 6620 7468 6520 7365 7175 656e  ns of the sequen
-00008330: 6365 2e0a 2a20 4164 6465 6420 6060 736b  ce..* Added ``sk
-00008340: 6269 6f2e 7574 696c 2e61 7373 6572 745f  bio.util.assert_
-00008350: 6f72 6469 6e61 7469 6f6e 5f72 6573 756c  ordination_resul
-00008360: 7473 5f65 7175 616c 6060 2066 756e 6374  ts_equal`` funct
-00008370: 696f 6e20 666f 7220 636f 6d70 6172 696e  ion for comparin
-00008380: 6720 6060 4f72 6469 6e61 7469 6f6e 5265  g ``OrdinationRe
-00008390: 7375 6c74 7360 6020 6f62 6a65 6374 7320  sults`` objects 
-000083a0: 696e 2075 6e69 7420 7465 7374 732e 0a2a  in unit tests..*
-000083b0: 2041 6464 6564 2060 6073 6b62 696f 2e69   Added ``skbio.i
-000083c0: 6f2e 666f 726d 6174 2e67 656e 6261 6e6b  o.format.genbank
-000083d0: 6060 2066 6f72 2072 6561 6469 6e67 2061  `` for reading a
-000083e0: 6e64 2077 7269 7469 6e67 2047 656e 4261  nd writing GenBa
-000083f0: 6e6b 2f47 656e 5065 7074 2066 6f72 2060  nk/GenPept for `
-00008400: 6044 4e41 6060 2c20 6060 524e 4160 602c  `DNA``, ``RNA``,
-00008410: 2060 6050 726f 7465 696e 6060 2061 6e64   ``Protein`` and
-00008420: 2060 6053 6571 7565 6e63 6560 6020 636c   ``Sequence`` cl
-00008430: 6173 7365 732e 0a2a 2041 6464 6564 2060  asses..* Added `
-00008440: 6073 6b62 696f 2e75 7469 6c2e 5265 7072  `skbio.util.Repr
-00008450: 6573 656e 7461 7469 6f6e 5761 726e 696e  esentationWarnin
-00008460: 6760 6020 666f 7220 7761 726e 696e 6720  g`` for warning 
-00008470: 6162 6f75 7420 7375 6273 7469 7475 7469  about substituti
-00008480: 6f6e 732c 2061 7373 756d 7074 696f 6e73  ons, assumptions
-00008490: 2c20 6f72 2070 6172 7469 6375 6c61 7220  , or particular 
-000084a0: 616c 7465 7261 7469 6f6e 7320 7468 6174  alterations that
-000084b0: 2077 6572 6520 6d61 6465 2066 6f72 2074   were made for t
-000084c0: 6865 2073 7563 6365 7373 6675 6c20 636f  he successful co
-000084d0: 6d70 6c65 7469 6f6e 206f 6620 6120 7072  mpletion of a pr
-000084e0: 6f63 6573 732e 0a2a 2060 6054 7265 654e  ocess..* ``TreeN
-000084f0: 6f64 652e 7469 705f 7469 705f 6469 7374  ode.tip_tip_dist
-00008500: 616e 6365 7360 6020 6e6f 7720 7375 7070  ances`` now supp
-00008510: 6f72 7473 206e 6f64 6573 2077 6974 686f  orts nodes witho
-00008520: 7574 2061 6e20 6173 736f 6369 6174 6564  ut an associated
-00008530: 206c 656e 6774 682e 2049 6e20 7468 6973   length. In this
-00008540: 2063 6173 652c 2061 206c 656e 6774 6820   case, a length 
-00008550: 6f66 2030 2e30 2069 7320 6173 7375 6d65  of 0.0 is assume
-00008560: 6420 616e 6420 616e 2060 6073 6b62 696f  d and an ``skbio
-00008570: 2e75 7469 6c2e 5265 7072 6573 656e 7461  .util.Representa
-00008580: 7469 6f6e 5761 726e 696e 6760 6020 6973  tionWarning`` is
-00008590: 2072 6169 7365 642e 2050 7265 7669 6f75   raised. Previou
-000085a0: 7320 6265 6861 7669 6f72 2077 6173 2074  s behavior was t
-000085b0: 6f20 7261 6973 6520 6120 6060 4e6f 4c65  o raise a ``NoLe
-000085c0: 6e67 7468 4572 726f 7260 602e 2028 5b23  ngthError``. ([#
-000085d0: 3739 315d 2868 7474 7073 3a2f 2f67 6974  791](https://git
-000085e0: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
-000085f0: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
-00008600: 732f 3739 3129 290a 2a20 6060 4469 7374  s/791)).* ``Dist
-00008610: 616e 6365 4d61 7472 6978 6060 206e 6f77  anceMatrix`` now
-00008620: 2068 6173 2061 206e 6577 2063 6f6e 7374   has a new const
-00008630: 7275 6374 6f72 206d 6574 686f 6420 6361  ructor method ca
-00008640: 6c6c 6564 2060 6672 6f6d 5f69 7465 7261  lled `from_itera
-00008650: 626c 6560 2e0a 2a20 6060 5365 7175 656e  ble`..* ``Sequen
-00008660: 6365 6060 206e 6f77 2061 6363 6570 7473  ce`` now accepts
-00008670: 2060 606c 6f77 6572 6361 7365 6060 206b   ``lowercase`` k
-00008680: 6579 776f 7264 206c 696b 6520 6060 444e  eyword like ``DN
-00008690: 4160 6020 616e 6420 6f74 6865 7273 2e20  A`` and others. 
-000086a0: 5570 6461 7465 6420 6060 6661 7374 6160  Updated ``fasta`
-000086b0: 602c 2060 6066 6173 7471 6060 2c20 616e  `, ``fastq``, an
-000086c0: 6420 6060 7173 6571 6060 2072 6561 6465  d ``qseq`` reade
-000086d0: 7273 2f77 7269 7465 7273 2066 6f72 2060  rs/writers for `
-000086e0: 6053 6571 7565 6e63 6560 6020 746f 2072  `Sequence`` to r
-000086f0: 6566 6c65 6374 2074 6869 732e 0a2a 2054  eflect this..* T
-00008700: 6865 2060 606c 6f77 6572 6361 7365 6060  he ``lowercase``
-00008710: 206d 6574 686f 6420 6861 7320 6265 656e   method has been
-00008720: 206d 6f76 6564 2075 7020 746f 2060 6053   moved up to ``S
-00008730: 6571 7565 6e63 6560 6020 6d65 616e 696e  equence`` meanin
-00008740: 6720 616c 6c20 7365 7175 656e 6365 206f  g all sequence o
-00008750: 626a 6563 7473 206e 6f77 2068 6176 6520  bjects now have 
-00008760: 6120 6060 6c6f 7765 7263 6173 6560 6020  a ``lowercase`` 
-00008770: 6d65 7468 6f64 2e0a 2a20 4164 6465 6420  method..* Added 
-00008780: 6060 7265 7665 7273 655f 7472 616e 7363  ``reverse_transc
-00008790: 7269 6265 6060 2063 6c61 7373 206d 6574  ribe`` class met
-000087a0: 686f 6420 746f 2060 6052 4e41 6060 2e0a  hod to ``RNA``..
-000087b0: 2a20 4164 6465 6420 6053 6571 7565 6e63  * Added `Sequenc
-000087c0: 652e 6f62 7365 7276 6564 5f63 6861 7273  e.observed_chars
-000087d0: 6020 7072 6f70 6572 7479 2066 6f72 206f  ` property for o
-000087e0: 6274 6169 6e69 6e67 2074 6865 2073 6574  btaining the set
-000087f0: 206f 6620 6f62 7365 7276 6564 2063 6861   of observed cha
-00008800: 7261 6374 6572 7320 696e 2061 2073 6571  racters in a seq
-00008810: 7565 6e63 652e 2028 5b23 3130 3735 5d28  uence. ([#1075](
-00008820: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-00008830: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
-00008840: 742d 6269 6f2f 6973 7375 6573 2f31 3037  t-bio/issues/107
-00008850: 3529 290a 2a20 4164 6465 6420 6053 6571  5)).* Added `Seq
-00008860: 7565 6e63 652e 6672 6571 7565 6e63 6965  uence.frequencie
-00008870: 7360 206d 6574 686f 6420 666f 7220 636f  s` method for co
-00008880: 6d70 7574 696e 6720 6368 6172 6163 7465  mputing characte
-00008890: 7220 6672 6571 7565 6e63 6965 7320 696e  r frequencies in
-000088a0: 2061 2073 6571 7565 6e63 652e 2028 5b23   a sequence. ([#
-000088b0: 3130 3734 5d28 6874 7470 733a 2f2f 6769  1074](https://gi
-000088c0: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
-000088d0: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
-000088e0: 6573 2f31 3037 3429 290a 2a20 4164 6465  es/1074)).* Adde
-000088f0: 6420 6578 7065 7269 6d65 6e74 616c 2063  d experimental c
-00008900: 6c61 7373 2d6d 6574 686f 6420 6060 5365  lass-method ``Se
-00008910: 7175 656e 6365 2e63 6f6e 6361 7460 6020  quence.concat`` 
-00008920: 7768 6963 6820 7769 6c6c 2070 726f 6475  which will produ
-00008930: 6365 2061 206e 6577 2073 6571 7565 6e63  ce a new sequenc
-00008940: 6520 6672 6f6d 2061 6e20 6974 6572 6162  e from an iterab
-00008950: 6c65 206f 6620 6578 6973 7469 6e67 2073  le of existing s
-00008960: 6571 7565 6e63 6573 2e20 5061 7261 6d65  equences. Parame
-00008970: 7465 7273 2063 6f6e 7472 6f6c 2068 6f77  ters control how
-00008980: 2070 6f73 6974 696f 6e61 6c20 6d65 7461   positional meta
-00008990: 6461 7461 2069 7320 7072 6f70 6167 6174  data is propagat
-000089a0: 6564 2064 7572 696e 6720 6120 636f 6e63  ed during a conc
-000089b0: 6174 656e 6174 696f 6e2e 0a2a 2060 6054  atenation..* ``T
-000089c0: 7265 654e 6f64 652e 746f 5f61 7272 6179  reeNode.to_array
-000089d0: 6060 206e 6f77 2073 7570 706f 7274 7320  `` now supports 
-000089e0: 7265 706c 6163 696e 6720 6060 6e61 6e60  replacing ``nan`
-000089f0: 6020 6272 616e 6368 206c 656e 6774 6873  ` branch lengths
-00008a00: 2069 6e20 7468 6520 7265 7375 6c74 696e   in the resultin
-00008a10: 6720 6272 616e 6368 206c 656e 6774 6820  g branch length 
-00008a20: 7665 6374 6f72 2077 6974 6820 7468 6520  vector with the 
-00008a30: 7661 6c75 6520 7072 6f76 6964 6564 2061  value provided a
-00008a40: 7320 6060 6e61 6e5f 6c65 6e67 7468 5f76  s ``nan_length_v
-00008a50: 616c 7565 6060 2e0a 2a20 6060 736b 6269  alue``..* ``skbi
-00008a60: 6f2e 696f 2e66 6f72 6d61 742e 7068 796c  o.io.format.phyl
-00008a70: 6970 6060 206e 6f77 2073 7570 706f 7274  ip`` now support
-00008a80: 7320 736e 6966 6669 6e67 2061 6e64 2072  s sniffing and r
-00008a90: 6561 6469 6e67 2073 7472 6963 742c 2073  eading strict, s
-00008aa0: 6571 7565 6e74 6961 6c20 5048 594c 4950  equential PHYLIP
-00008ab0: 2d66 6f72 6d61 7474 6564 2066 696c 6573  -formatted files
-00008ac0: 2069 6e74 6f20 6060 736b 6269 6f2e 416c   into ``skbio.Al
-00008ad0: 6967 6e6d 656e 7460 6020 6f62 6a65 6374  ignment`` object
-00008ae0: 732e 2028 5b23 3130 3036 5d28 6874 7470  s. ([#1006](http
-00008af0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-00008b00: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-00008b10: 6f2f 6973 7375 6573 2f31 3030 3629 290a  o/issues/1006)).
-00008b20: 2a20 4164 6465 6420 6064 6566 6175 6c74  * Added `default
-00008b30: 5f67 6170 5f63 6861 7260 2063 6c61 7373  _gap_char` class
-00008b40: 2070 726f 7065 7274 7920 746f 2060 6044   property to ``D
-00008b50: 4e41 6060 2c20 6060 524e 4160 602c 2061  NA``, ``RNA``, a
-00008b60: 6e64 2060 6050 726f 7465 696e 6060 2066  nd ``Protein`` f
-00008b70: 6f72 2072 6570 7265 7365 6e74 696e 6720  or representing 
-00008b80: 6761 7020 6368 6172 6163 7465 7273 2069  gap characters i
-00008b90: 6e20 6120 6e65 7720 7365 7175 656e 6365  n a new sequence
-00008ba0: 2e0a 0a23 2323 2042 6163 6b77 6172 642d  ...### Backward-
-00008bb0: 696e 636f 6d70 6174 6962 6c65 2063 6861  incompatible cha
-00008bc0: 6e67 6573 205b 7374 6162 6c65 5d0a 2a20  nges [stable].* 
-00008bd0: 6053 6571 7565 6e63 652e 6b6d 6572 5f66  `Sequence.kmer_f
-00008be0: 7265 7175 656e 6369 6573 6020 6e6f 7720  requencies` now 
-00008bf0: 7265 7475 726e 7320 6120 6064 6963 7460  returns a `dict`
-00008c00: 2e20 5072 6576 696f 7573 2062 6568 6176  . Previous behav
-00008c10: 696f 7220 7761 7320 746f 2072 6574 7572  ior was to retur
-00008c20: 6e20 6120 6063 6f6c 6c65 6374 696f 6e73  n a `collections
-00008c30: 2e43 6f75 6e74 6572 6020 6966 2060 7265  .Counter` if `re
-00008c40: 6c61 7469 7665 3d46 616c 7365 6020 7761  lative=False` wa
-00008c50: 7320 7061 7373 6564 2c20 616e 6420 6120  s passed, and a 
-00008c60: 6063 6f6c 6c65 6374 696f 6e73 2e64 6566  `collections.def
-00008c70: 6175 6c74 6469 6374 6020 6966 2060 7265  aultdict` if `re
-00008c80: 6c61 7469 7665 3d54 7275 6560 2077 6173  lative=True` was
-00008c90: 2070 6173 7365 642e 2049 6e20 7468 6520   passed. In the 
-00008ca0: 6361 7365 206f 6620 6120 6d69 7373 696e  case of a missin
-00008cb0: 6720 6b65 792c 2074 6865 2060 436f 756e  g key, the `Coun
-00008cc0: 7465 7260 2077 6f75 6c64 2072 6574 7572  ter` would retur
-00008cd0: 6e20 3020 616e 6420 7468 6520 6064 6566  n 0 and the `def
-00008ce0: 6175 6c74 6469 6374 6020 776f 756c 6420  aultdict` would 
-00008cf0: 7265 7475 726e 2030 2e30 2e20 4265 6361  return 0.0. Beca
-00008d00: 7573 6520 7468 6520 7265 7475 726e 2074  use the return t
-00008d10: 7970 6520 6973 206e 6f77 2061 6c77 6179  ype is now alway
-00008d20: 7320 6120 6064 6963 7460 2c20 6174 7465  s a `dict`, atte
-00008d30: 6d70 7469 6e67 2074 6f20 6163 6365 7373  mpting to access
-00008d40: 2061 206d 6973 7369 6e67 206b 6579 2077   a missing key w
-00008d50: 696c 6c20 7261 6973 6520 6120 604b 6579  ill raise a `Key
-00008d60: 4572 726f 7260 2e20 5468 6973 2063 6861  Error`. This cha
-00008d70: 6e67 6520 2a6d 6179 2a20 6272 6561 6b20  nge *may* break 
-00008d80: 6261 636b 7761 7264 732d 636f 6d70 6174  backwards-compat
-00008d90: 6962 696c 6974 7920 6465 7065 6e64 696e  ibility dependin
-00008da0: 6720 6f6e 2068 6f77 2074 6865 2060 436f  g on how the `Co
-00008db0: 756e 7465 7260 2f60 6465 6661 756c 7464  unter`/`defaultd
-00008dc0: 6963 7460 2069 7320 6265 696e 6720 7573  ict` is being us
-00008dd0: 6564 2e20 5765 2068 6f70 6520 7468 6174  ed. We hope that
-00008de0: 2069 6e20 6d6f 7374 2063 6173 6573 2074   in most cases t
-00008df0: 6869 7320 6368 616e 6765 2077 696c 6c20  his change will 
-00008e00: 6e6f 7420 6272 6561 6b20 6261 636b 7761  not break backwa
-00008e10: 7264 732d 636f 6d70 6174 6962 696c 6974  rds-compatibilit
-00008e20: 7920 6265 6361 7573 6520 626f 7468 2060  y because both `
-00008e30: 436f 756e 7465 7260 2061 6e64 2060 6465  Counter` and `de
-00008e40: 6661 756c 7464 6963 7460 2061 7265 2060  faultdict` are `
-00008e50: 6469 6374 6020 7375 6263 6c61 7373 6573  dict` subclasses
-00008e60: 2e0a 0a20 2020 4966 2074 6865 2070 7265  ...   If the pre
-00008e70: 7669 6f75 7320 6265 6861 7669 6f72 2069  vious behavior i
-00008e80: 7320 6465 7369 7265 642c 2063 6f6e 7665  s desired, conve
-00008e90: 7274 2074 6865 2060 6469 6374 6020 696e  rt the `dict` in
-00008ea0: 746f 2061 2060 436f 756e 7465 7260 2f60  to a `Counter`/`
-00008eb0: 6465 6661 756c 7464 6963 7460 3a0a 0a20  defaultdict`:.. 
-00008ec0: 2020 2060 6060 7079 7468 6f6e 0a20 2020     ```python.   
-00008ed0: 2069 6d70 6f72 7420 636f 6c6c 6563 7469   import collecti
-00008ee0: 6f6e 730a 2020 2020 6672 6f6d 2073 6b62  ons.    from skb
-00008ef0: 696f 2069 6d70 6f72 7420 5365 7175 656e  io import Sequen
-00008f00: 6365 0a20 2020 2073 6571 203d 2053 6571  ce.    seq = Seq
-00008f10: 7565 6e63 6528 2741 4343 4741 4754 5454  uence('ACCGAGTTT
-00008f20: 4141 4343 4741 4154 4127 290a 0a20 2020  AACCGAATA')..   
-00008f30: 2023 2043 6f75 6e74 6572 0a20 2020 2066   # Counter.    f
-00008f40: 7265 7173 5f64 6963 7420 3d20 7365 712e  reqs_dict = seq.
-00008f50: 6b6d 6572 5f66 7265 7175 656e 6369 6573  kmer_frequencies
-00008f60: 286b 3d38 290a 2020 2020 6672 6571 735f  (k=8).    freqs_
-00008f70: 636f 756e 7465 7220 3d20 636f 6c6c 6563  counter = collec
-00008f80: 7469 6f6e 732e 436f 756e 7465 7228 6672  tions.Counter(fr
-00008f90: 6571 735f 6469 6374 290a 0a20 2020 2023  eqs_dict)..    #
-00008fa0: 2064 6566 6175 6c74 6469 6374 0a20 2020   defaultdict.   
-00008fb0: 2066 7265 7173 5f64 6963 7420 3d20 7365   freqs_dict = se
-00008fc0: 712e 6b6d 6572 5f66 7265 7175 656e 6369  q.kmer_frequenci
-00008fd0: 6573 286b 3d38 2c20 7265 6c61 7469 7665  es(k=8, relative
-00008fe0: 3d54 7275 6529 0a20 2020 2066 7265 7173  =True).    freqs
-00008ff0: 5f64 6566 6175 6c74 5f64 6963 7420 3d20  _default_dict = 
-00009000: 636f 6c6c 6563 7469 6f6e 732e 6465 6661  collections.defa
-00009010: 756c 7464 6963 7428 666c 6f61 742c 2066  ultdict(float, f
-00009020: 7265 7173 5f64 6963 7429 0a20 2020 2060  reqs_dict).    `
-00009030: 6060 0a0a 2020 202a 2a52 6174 696f 6e61  ``..   **Rationa
-00009040: 6c65 3a2a 2a20 5765 2062 656c 6965 7665  le:** We believe
-00009050: 2069 7420 6973 2073 6166 6572 2074 6f20   it is safer to 
-00009060: 7265 7475 726e 2060 6469 6374 6020 696e  return `dict` in
-00009070: 7374 6561 6420 6f66 2060 436f 756e 7465  stead of `Counte
-00009080: 7260 2f60 6465 6661 756c 7464 6963 7460  r`/`defaultdict`
-00009090: 2061 7320 7468 6973 206d 6179 2070 7265   as this may pre
-000090a0: 7665 6e74 2065 7272 6f72 2d70 726f 6e65  vent error-prone
-000090b0: 2075 7361 6765 206f 6620 7468 6520 7265   usage of the re
-000090c0: 7475 726e 2076 616c 7565 2e20 5072 6576  turn value. Prev
-000090d0: 696f 7573 2062 6568 6176 696f 7220 616c  ious behavior al
-000090e0: 6c6f 7765 6420 6163 6365 7373 696e 6720  lowed accessing 
-000090f0: 6d69 7373 696e 6720 6b6d 6572 732c 2072  missing kmers, r
-00009100: 6574 7572 6e69 6e67 2030 206f 7220 302e  eturning 0 or 0.
-00009110: 3020 6465 7065 6e64 696e 6720 6f6e 2074  0 depending on t
-00009120: 6865 2060 7265 6c61 7469 7665 6020 7061  he `relative` pa
-00009130: 7261 6d65 7465 722e 2054 6869 7320 6973  rameter. This is
-00009140: 2063 6f6e 7665 6e69 656e 7420 696e 206d   convenient in m
-00009150: 616e 7920 6361 7365 7320 6275 7420 616c  any cases but al
-00009160: 736f 2070 6f74 656e 7469 616c 6c79 206d  so potentially m
-00009170: 6973 6c65 6164 696e 672e 2046 6f72 2065  isleading. For e
-00009180: 7861 6d70 6c65 2c20 636f 6e73 6964 6572  xample, consider
-00009190: 2074 6865 2066 6f6c 6c6f 7769 6e67 2063   the following c
-000091a0: 6f64 653a 0a0a 2020 2020 6060 6070 7974  ode:..    ```pyt
-000091b0: 686f 6e0a 2020 2020 6672 6f6d 2073 6b62  hon.    from skb
-000091c0: 696f 2069 6d70 6f72 7420 5365 7175 656e  io import Sequen
-000091d0: 6365 0a20 2020 2073 6571 203d 2053 6571  ce.    seq = Seq
-000091e0: 7565 6e63 6528 2741 4343 4741 4754 5454  uence('ACCGAGTTT
-000091f0: 4141 4343 4741 4154 4127 290a 2020 2020  AACCGAATA').    
-00009200: 6672 6571 7320 3d20 7365 712e 6b6d 6572  freqs = seq.kmer
-00009210: 5f66 7265 7175 656e 6369 6573 286b 3d38  _frequencies(k=8
-00009220: 290a 2020 2020 6672 6571 735b 2741 4343  ).    freqs['ACC
-00009230: 4741 275d 0a20 2020 2060 6060 0a0a 2020  GA'].    ```..  
-00009240: 2020 5072 6576 696f 7573 2062 6568 6176    Previous behav
-00009250: 696f 7220 776f 756c 6420 7265 7475 726e  ior would return
-00009260: 2030 2062 6563 6175 7365 2074 6865 206b   0 because the k
-00009270: 6d65 7220 6027 4143 4347 4127 6020 6973  mer `'ACCGA'` is
-00009280: 206e 6f74 2070 7265 7365 6e74 2069 6e20   not present in 
-00009290: 7468 6520 6043 6f75 6e74 6572 602e 2049  the `Counter`. I
-000092a0: 6e20 6f6e 6520 7265 7370 6563 7420 7468  n one respect th
-000092b0: 6973 2069 7320 7468 6520 636f 7272 6563  is is the correc
-000092c0: 7420 616e 7377 6572 2062 6563 6175 7365  t answer because
-000092d0: 2077 6520 6173 6b65 6420 666f 7220 6b6d   we asked for km
-000092e0: 6572 7320 6f66 206c 656e 6774 6820 383b  ers of length 8;
-000092f0: 2060 2741 4343 4741 2760 2069 7320 6120   `'ACCGA'` is a 
-00009300: 6469 6666 6572 656e 7420 6c65 6e67 7468  different length
-00009310: 2073 6f20 6974 2069 7320 6e6f 7420 696e   so it is not in
-00009320: 636c 7564 6564 2069 6e20 7468 6520 7265  cluded in the re
-00009330: 7375 6c74 732e 2048 6f77 6576 6572 2c20  sults. However, 
-00009340: 7765 2062 656c 6965 7665 2069 7420 6973  we believe it is
-00009350: 2073 6166 6572 2074 6f20 6176 6f69 6420   safer to avoid 
-00009360: 7468 6973 2069 6d70 6c69 6369 7420 6265  this implicit be
-00009370: 6861 7669 6f72 2069 6e20 6361 7365 2074  havior in case t
-00009380: 6865 2075 7365 7220 6173 7375 6d65 7320  he user assumes 
-00009390: 7468 6572 6520 6172 6520 6e6f 2060 2741  there are no `'A
-000093a0: 4343 4741 2760 206b 6d65 7273 2069 6e20  CCGA'` kmers in 
-000093b0: 7468 6520 7365 7175 656e 6365 2028 7768  the sequence (wh
-000093c0: 6963 6820 7468 6572 6520 6172 6521 292e  ich there are!).
-000093d0: 2041 2060 4b65 7945 7272 6f72 6020 696e   A `KeyError` in
-000093e0: 2074 6869 7320 6361 7365 2069 7320 6d6f   this case is mo
-000093f0: 7265 2065 7870 6c69 6369 7420 616e 6420  re explicit and 
-00009400: 666f 7263 6573 2074 6865 2075 7365 7220  forces the user 
-00009410: 746f 2063 6f6e 7369 6465 7220 7468 6569  to consider thei
-00009420: 7220 7175 6572 792e 2052 6574 7572 6e69  r query. Returni
-00009430: 6e67 2061 2060 6469 6374 6020 7769 6c6c  ng a `dict` will
-00009440: 2061 6c73 6f20 6265 2063 6f6e 7369 7374   also be consist
-00009450: 656e 7420 7769 7468 2060 5365 7175 656e  ent with `Sequen
-00009460: 6365 2e66 7265 7175 656e 6369 6573 602e  ce.frequencies`.
-00009470: 0a0a 2323 2320 4261 636b 7761 7264 2d69  ..### Backward-i
-00009480: 6e63 6f6d 7061 7469 626c 6520 6368 616e  ncompatible chan
-00009490: 6765 7320 5b65 7870 6572 696d 656e 7461  ges [experimenta
-000094a0: 6c5d 0a2a 2052 6570 6c61 6365 6420 6060  l].* Replaced ``
-000094b0: 5043 6f41 6060 2c20 6060 4343 4160 602c  PCoA``, ``CCA``,
-000094c0: 2060 6043 4160 6020 616e 6420 6060 5244   ``CA`` and ``RD
-000094d0: 4160 6020 696e 2060 6073 6b62 696f 2e73  A`` in ``skbio.s
-000094e0: 7461 7473 2e6f 7264 696e 6174 696f 6e60  tats.ordination`
-000094f0: 6020 7769 7468 2065 7175 6976 616c 656e  ` with equivalen
-00009500: 7420 6675 6e63 7469 6f6e 7320 6060 7063  t functions ``pc
-00009510: 6f61 6060 2c20 6060 6363 6160 602c 2060  oa``, ``cca``, `
-00009520: 6063 6160 6020 616e 6420 6060 7264 6160  `ca`` and ``rda`
-00009530: 602e 2054 6865 7365 2066 756e 6374 696f  `. These functio
-00009540: 6e73 206e 6f77 2074 616b 6520 6060 7064  ns now take ``pd
-00009550: 2e44 6174 6146 7261 6d65 6060 206f 626a  .DataFrame`` obj
-00009560: 6563 7473 2e0a 2a20 4368 616e 6765 2060  ects..* Change `
-00009570: 604f 7264 696e 6174 696f 6e52 6573 756c  `OrdinationResul
-00009580: 7473 6060 2074 6f20 6861 7665 2069 7473  ts`` to have its
-00009590: 2061 7474 7269 6275 7465 7320 6261 7365   attributes base
-000095a0: 6420 6f6e 2060 6070 642e 4461 7461 4672  d on ``pd.DataFr
-000095b0: 616d 6560 6020 616e 6420 6060 7064 2e53  ame`` and ``pd.S
-000095c0: 6572 6965 7360 6020 6f62 6a65 6374 732c  eries`` objects,
-000095d0: 2069 6e73 7465 6164 206f 6620 7061 6972   instead of pair
-000095e0: 7320 6f66 2069 6465 6e74 6966 6965 7273  s of identifiers
-000095f0: 2061 6e64 2076 616c 7565 732e 2054 6865   and values. The
-00009600: 2063 6861 6e67 6573 2061 7265 2061 7320   changes are as 
-00009610: 666f 6c6c 6f77 733a 0a20 2020 202d 2060  follows:.    - `
-00009620: 6073 7065 6369 6573 6060 2061 6e64 2060  `species`` and `
-00009630: 6073 7065 6369 6573 5f69 6473 6060 2068  `species_ids`` h
-00009640: 6176 6520 6265 656e 2072 6570 6c61 6365  ave been replace
-00009650: 6420 6279 2061 2060 6070 642e 4461 7461  d by a ``pd.Data
-00009660: 4672 616d 6560 6020 6e61 6d65 6420 6060  Frame`` named ``
-00009670: 6665 6174 7572 6573 6060 2e0a 2020 2020  features``..    
-00009680: 2d20 6060 7369 7465 6060 2061 6e64 2060  - ``site`` and `
-00009690: 6073 6974 655f 6964 7360 6020 6861 7665  `site_ids`` have
-000096a0: 2062 6565 6e20 7265 706c 6163 6564 2062   been replaced b
-000096b0: 7920 6120 6060 7064 2e44 6174 6146 7261  y a ``pd.DataFra
-000096c0: 6d65 6060 206e 616d 6564 2060 6073 616d  me`` named ``sam
-000096d0: 706c 6573 6060 2e0a 2020 2020 2d20 6060  ples``..    - ``
-000096e0: 6569 6776 616c 7360 6020 6973 206e 6f77  eigvals`` is now
-000096f0: 2061 2060 6070 642e 5365 7269 6573 6060   a ``pd.Series``
-00009700: 206f 626a 6563 742e 0a20 2020 202d 2060   object..    - `
-00009710: 6070 726f 706f 7274 696f 6e5f 6578 706c  `proportion_expl
-00009720: 6169 6e65 6460 6020 6973 206e 6f77 2061  ained`` is now a
-00009730: 2060 6070 642e 5365 7269 6573 6060 206f   ``pd.Series`` o
-00009740: 626a 6563 742e 0a20 2020 202d 2060 6062  bject..    - ``b
-00009750: 6970 6c6f 7460 6020 6973 206e 6f77 2061  iplot`` is now a
-00009760: 2060 6070 642e 4461 7461 4672 616d 6560   ``pd.DataFrame`
-00009770: 6020 6f62 6a65 6374 206e 616d 6564 2060  ` object named `
-00009780: 6062 6970 6c6f 745f 7363 6f72 6573 6060  `biplot_scores``
-00009790: 2e0a 2020 2020 2d20 6060 7369 7465 5f63  ..    - ``site_c
-000097a0: 6f6e 7374 7261 696e 7473 6060 2069 7320  onstraints`` is 
-000097b0: 6e6f 7720 6120 6060 7064 2e44 6174 6146  now a ``pd.DataF
-000097c0: 7261 6d65 6060 206f 626a 6563 7420 6e61  rame`` object na
-000097d0: 6d65 6420 6060 7361 6d70 6c65 5f63 6f6e  med ``sample_con
-000097e0: 7374 7261 696e 7473 6060 2e0a 2a20 6060  straints``..* ``
-000097f0: 7368 6f72 745f 6d65 7468 6f64 5f6e 616d  short_method_nam
-00009800: 6560 6020 616e 6420 6060 6c6f 6e67 5f6d  e`` and ``long_m
-00009810: 6574 686f 645f 6e61 6d65 6060 2061 7265  ethod_name`` are
-00009820: 206e 6f77 2072 6571 7569 7265 6420 6172   now required ar
-00009830: 6775 6d65 6e74 7320 6f66 2074 6865 2060  guments of the `
-00009840: 604f 7264 696e 6174 696f 6e52 6573 756c  `OrdinationResul
-00009850: 7473 6060 206f 626a 6563 742e 0a2a 2052  ts`` object..* R
-00009860: 656d 6f76 6564 2060 736b 6269 6f2e 6469  emoved `skbio.di
-00009870: 7665 7273 6974 792e 616c 7068 612e 6571  versity.alpha.eq
-00009880: 7569 7461 6269 6c69 7479 602e 2050 6c65  uitability`. Ple
-00009890: 6173 6520 7573 6520 6073 6b62 696f 2e64  ase use `skbio.d
-000098a0: 6976 6572 7369 7479 2e61 6c70 6861 2e70  iversity.alpha.p
-000098b0: 6965 6c6f 755f 6560 2c20 7768 6963 6820  ielou_e`, which 
-000098c0: 6973 206d 6f72 6520 6163 6375 7261 7465  is more accurate
-000098d0: 6c79 206e 616d 6564 2061 6e64 2062 6574  ly named and bet
-000098e0: 7465 7220 646f 6375 6d65 6e74 6564 2e20  ter documented. 
-000098f0: 4e6f 7465 2074 6861 7420 6065 7175 6974  Note that `equit
-00009900: 6162 696c 6974 7960 2062 7920 6465 6661  ability` by defa
-00009910: 756c 7420 7573 6564 206c 6f67 6172 6974  ult used logarit
-00009920: 686d 2062 6173 6520 3220 7768 696c 6520  hm base 2 while 
-00009930: 6070 6965 6c6f 755f 6560 2075 7365 7320  `pielou_e` uses 
-00009940: 6c6f 6761 7269 7468 6d20 6261 7365 2060  logarithm base `
-00009950: 6560 2061 7320 6465 7363 7269 6265 6420  e` as described 
-00009960: 696e 2048 6569 7020 3139 3734 2e0a 2a20  in Heip 1974..* 
-00009970: 6060 736b 6269 6f2e 6469 7665 7273 6974  ``skbio.diversit
-00009980: 792e 6265 7461 2e70 775f 6469 7374 616e  y.beta.pw_distan
-00009990: 6365 7360 6020 6973 206e 6f77 2063 616c  ces`` is now cal
-000099a0: 6c65 6420 6060 736b 6269 6f2e 6469 7665  led ``skbio.dive
-000099b0: 7273 6974 792e 6265 7461 5f64 6976 6572  rsity.beta_diver
-000099c0: 7369 7479 6060 2e20 5468 6973 2066 756e  sity``. This fun
-000099d0: 6374 696f 6e20 6e6f 206c 6f6e 6765 7220  ction no longer 
-000099e0: 6465 6669 6e65 7320 6120 6465 6661 756c  defines a defaul
-000099f0: 7420 6d65 7472 6963 2c20 616e 6420 6060  t metric, and ``
-00009a00: 6d65 7472 6963 6060 2069 7320 6e6f 7720  metric`` is now 
-00009a10: 7468 6520 6669 7273 7420 6172 6775 6d65  the first argume
-00009a20: 6e74 2074 6f20 7468 6973 2066 756e 6374  nt to this funct
-00009a30: 696f 6e2e 2054 6869 7320 6675 6e63 7469  ion. This functi
-00009a40: 6f6e 2063 616e 2061 6c73 6f20 6e6f 7720  on can also now 
-00009a50: 7461 6b65 2061 2070 6169 7277 6973 6520  take a pairwise 
-00009a60: 6469 7374 616e 6365 7320 6675 6e63 7469  distances functi
-00009a70: 6f6e 2061 7320 6060 7061 6972 7769 7365  on as ``pairwise
-00009a80: 5f66 756e 6360 602e 0a2a 2044 6570 7265  _func``..* Depre
-00009a90: 6361 7465 6420 6675 6e63 7469 6f6e 2060  cated function `
-00009aa0: 6073 6b62 696f 2e64 6976 6572 7369 7479  `skbio.diversity
-00009ab0: 2e62 6574 612e 7077 5f64 6973 7461 6e63  .beta.pw_distanc
-00009ac0: 6573 5f66 726f 6d5f 7461 626c 6560 6020  es_from_table`` 
-00009ad0: 6861 7320 6265 656e 2072 656d 6f76 6564  has been removed
-00009ae0: 2066 726f 6d20 7363 696b 6974 2d62 696f   from scikit-bio
-00009af0: 2061 7320 7363 6865 6475 6c65 642e 2043   as scheduled. C
-00009b00: 6f64 6520 7468 6174 2075 7365 6420 7468  ode that used th
-00009b10: 6973 2073 686f 756c 6420 6265 2061 6461  is should be ada
-00009b20: 7074 6564 2074 6f20 7573 6520 6060 736b  pted to use ``sk
-00009b30: 6269 6f2e 6469 7665 7273 6974 792e 6265  bio.diversity.be
-00009b40: 7461 5f64 6976 6572 7369 7479 6060 2e0a  ta_diversity``..
-00009b50: 2a20 6060 5472 6565 4e6f 6465 2e69 6e64  * ``TreeNode.ind
-00009b60: 6578 5f74 7265 6560 6020 6e6f 7720 7265  ex_tree`` now re
-00009b70: 7475 726e 7320 6120 322d 4420 6e75 6d70  turns a 2-D nump
-00009b80: 7920 6172 7261 7920 6173 2069 7473 2073  y array as its s
-00009b90: 6563 6f6e 6420 7265 7475 726e 2076 616c  econd return val
-00009ba0: 7565 2028 7468 6520 6368 696c 6420 6e6f  ue (the child no
-00009bb0: 6465 2069 6e64 6578 2920 696e 7374 6561  de index) instea
-00009bc0: 6420 6f66 2061 2031 2d44 206e 756d 7079  d of a 1-D numpy
-00009bd0: 2061 7272 6179 2e0a 2a20 4465 7072 6563   array..* Deprec
-00009be0: 6174 6564 2066 756e 6374 696f 6e73 2060  ated functions `
-00009bf0: 736b 6269 6f2e 6472 6177 2e62 6f78 706c  skbio.draw.boxpl
-00009c00: 6f74 7360 2061 6e64 2060 736b 6269 6f2e  ots` and `skbio.
-00009c10: 6472 6177 2e67 726f 7570 6564 5f64 6973  draw.grouped_dis
-00009c20: 7472 6962 7574 696f 6e73 6020 6861 7665  tributions` have
-00009c30: 2062 6565 6e20 7265 6d6f 7665 6420 6672   been removed fr
-00009c40: 6f6d 2073 6369 6b69 742d 6269 6f20 6173  om scikit-bio as
-00009c50: 2073 6368 6564 756c 6564 2e20 5468 6573   scheduled. Thes
-00009c60: 6520 6675 6e63 7469 6f6e 7320 6765 6e65  e functions gene
-00009c70: 7261 7465 6420 706c 6f74 7320 7468 6174  rated plots that
-00009c80: 2077 6572 6520 6e6f 7420 7370 6563 6966   were not specif
-00009c90: 6963 2074 6f20 6269 6f69 6e66 6f72 6d61  ic to bioinforma
-00009ca0: 7469 6373 2e20 5468 6573 6520 7479 7065  tics. These type
-00009cb0: 7320 6f66 2070 6c6f 7473 2063 616e 2062  s of plots can b
-00009cc0: 6520 6765 6e65 7261 7465 6420 7769 7468  e generated with
-00009cd0: 2073 6561 626f 726e 206f 7220 616e 6f74   seaborn or anot
-00009ce0: 6865 7220 6765 6e65 7261 6c2d 7075 7270  her general-purp
-00009cf0: 6f73 6520 706c 6f74 7469 6e67 2070 6163  ose plotting pac
-00009d00: 6b61 6765 2e0a 2a20 4465 7072 6563 6174  kage..* Deprecat
-00009d10: 6564 2066 756e 6374 696f 6e20 6073 6b62  ed function `skb
-00009d20: 696f 2e73 7461 7473 2e70 6f77 6572 2e62  io.stats.power.b
-00009d30: 6f6f 7473 7472 6170 5f70 6f77 6572 5f63  ootstrap_power_c
-00009d40: 7572 7665 6020 6861 7320 6265 656e 2072  urve` has been r
-00009d50: 656d 6f76 6564 2066 726f 6d20 7363 696b  emoved from scik
-00009d60: 6974 2d62 696f 2061 7320 7363 6865 6475  it-bio as schedu
-00009d70: 6c65 642e 2055 7365 2060 736b 6269 6f2e  led. Use `skbio.
-00009d80: 7374 6174 732e 706f 7765 722e 7375 6273  stats.power.subs
-00009d90: 616d 706c 655f 706f 7765 7260 206f 7220  ample_power` or 
-00009da0: 6073 6b62 696f 2e73 7461 7473 2e70 6f77  `skbio.stats.pow
-00009db0: 6572 2e73 7562 7361 6d70 6c65 5f70 6169  er.subsample_pai
-00009dc0: 7265 645f 706f 7765 7260 2066 6f6c 6c6f  red_power` follo
-00009dd0: 7765 6420 6279 2060 736b 6269 6f2e 7374  wed by `skbio.st
-00009de0: 6174 732e 706f 7765 722e 636f 6e66 6964  ats.power.confid
-00009df0: 656e 6365 5f62 6f75 6e64 602e 0a2a 2044  ence_bound`..* D
-00009e00: 6570 7265 6361 7465 6420 6675 6e63 7469  eprecated functi
-00009e10: 6f6e 2060 736b 6269 6f2e 7374 6174 732e  on `skbio.stats.
-00009e20: 7370 6174 6961 6c2e 7072 6f63 7275 7374  spatial.procrust
-00009e30: 6573 6020 6861 7320 6265 656e 2072 656d  es` has been rem
-00009e40: 6f76 6564 2066 726f 6d20 7363 696b 6974  oved from scikit
-00009e50: 2d62 696f 2061 7320 7363 6865 6475 6c65  -bio as schedule
-00009e60: 6420 696e 2066 6176 6f72 206f 6620 6073  d in favor of `s
-00009e70: 6369 7079 2e73 7061 7469 616c 2e70 726f  cipy.spatial.pro
-00009e80: 6372 7573 7465 7360 2e0a 2a20 4465 7072  crustes`..* Depr
-00009e90: 6563 6174 6564 2063 6c61 7373 2060 736b  ecated class `sk
-00009ea0: 6269 6f2e 7472 6565 2e43 6f6d 7072 6573  bio.tree.Compres
-00009eb0: 7365 6454 7269 6560 2061 6e64 2066 756e  sedTrie` and fun
-00009ec0: 6374 696f 6e20 6073 6b62 696f 2e74 7265  ction `skbio.tre
-00009ed0: 652e 6661 7374 615f 746f 5f70 6169 726c  e.fasta_to_pairl
-00009ee0: 6973 7460 2068 6176 6520 6265 656e 2072  ist` have been r
-00009ef0: 656d 6f76 6564 2066 726f 6d20 7363 696b  emoved from scik
-00009f00: 6974 2d62 696f 2061 7320 7363 6865 6475  it-bio as schedu
-00009f10: 6c65 6420 696e 2066 6176 6f72 206f 6620  led in favor of 
-00009f20: 6578 6973 7469 6e67 2067 656e 6572 616c  existing general
-00009f30: 2d70 7572 706f 7365 2050 7974 686f 6e20  -purpose Python 
-00009f40: 7472 6965 2070 6163 6b61 6765 732e 0a2a  trie packages..*
-00009f50: 2044 6570 7265 6361 7465 6420 6675 6e63   Deprecated func
-00009f60: 7469 6f6e 2060 736b 6269 6f2e 7574 696c  tion `skbio.util
-00009f70: 2e66 6c61 7474 656e 6020 6861 7320 6265  .flatten` has be
-00009f80: 656e 2072 656d 6f76 6564 2066 726f 6d20  en removed from 
-00009f90: 7363 696b 6974 2d62 696f 2061 7320 7363  scikit-bio as sc
-00009fa0: 6865 6475 6c65 6420 696e 2066 6176 6f72  heduled in favor
-00009fb0: 206f 6620 736f 6c75 7469 6f6e 7320 6176   of solutions av
-00009fc0: 6169 6c61 626c 6520 696e 2074 6865 2050  ailable in the P
-00009fd0: 7974 686f 6e20 7374 616e 6461 7264 206c  ython standard l
-00009fe0: 6962 7261 7279 2028 7365 6520 5b68 6572  ibrary (see [her
-00009ff0: 655d 2868 7474 703a 2f2f 7374 6163 6b6f  e](http://stacko
-0000a000: 7665 7266 6c6f 772e 636f 6d2f 612f 3935  verflow.com/a/95
-0000a010: 3239 3532 2f33 3633 3930 3233 2920 616e  2952/3639023) an
-0000a020: 6420 5b68 6572 655d 2868 7474 703a 2f2f  d [here](http://
-0000a030: 7374 6163 6b6f 7665 7266 6c6f 772e 636f  stackoverflow.co
-0000a040: 6d2f 612f 3430 3631 3939 2f33 3633 3930  m/a/406199/36390
-0000a050: 3233 2920 666f 7220 6578 616d 706c 6573  23) for examples
-0000a060: 292e 0a2a 2050 6169 7277 6973 6520 616c  )..* Pairwise al
-0000a070: 6967 6e6d 656e 7420 6675 6e63 7469 6f6e  ignment function
-0000a080: 7320 696e 2060 736b 6269 6f2e 616c 6967  s in `skbio.alig
-0000a090: 6e6d 656e 7460 206e 6f77 2072 6574 7572  nment` now retur
-0000a0a0: 6e20 6120 7475 706c 6520 636f 6e74 6169  n a tuple contai
-0000a0b0: 6e69 6e67 2074 6865 2060 5461 6275 6c61  ning the `Tabula
-0000a0c0: 724d 5341 6020 616c 6967 6e6d 656e 742c  rMSA` alignment,
-0000a0d0: 2061 6c69 676e 6d65 6e74 2073 636f 7265   alignment score
-0000a0e0: 2c20 616e 6420 7374 6172 742f 656e 6420  , and start/end 
-0000a0f0: 706f 7369 7469 6f6e 732e 2054 6865 2072  positions. The r
-0000a100: 6574 7572 6e65 6420 6054 6162 756c 6172  eturned `Tabular
-0000a110: 4d53 4160 2773 2060 696e 6465 7860 2069  MSA`'s `index` i
-0000a120: 7320 616c 7761 7973 2074 6865 2064 6566  s always the def
-0000a130: 6175 6c74 2069 6e74 6567 6572 2069 6e64  ault integer ind
-0000a140: 6578 3b20 7365 7175 656e 6365 2049 4473  ex; sequence IDs
-0000a150: 2061 7265 206e 6f20 6c6f 6e67 6572 2070   are no longer p
-0000a160: 726f 7061 6761 7465 6420 746f 2074 6865  ropagated to the
-0000a170: 204d 5341 2e20 4164 6469 7469 6f6e 616c   MSA. Additional
-0000a180: 6c79 2c20 7468 6520 7061 6972 7769 7365  ly, the pairwise
-0000a190: 2061 6c69 676e 6d65 6e74 2066 756e 6374   alignment funct
-0000a1a0: 696f 6e73 206e 6f77 2061 6363 6570 7420  ions now accept 
-0000a1b0: 7468 6520 666f 6c6c 6f77 696e 6720 696e  the following in
-0000a1c0: 7075 7420 7479 7065 7320 746f 2061 6c69  put types to ali
-0000a1d0: 676e 3a0a 2020 2020 2d20 606c 6f63 616c  gn:.    - `local
-0000a1e0: 5f70 6169 7277 6973 655f 616c 6967 6e5f  _pairwise_align_
-0000a1f0: 6e75 636c 656f 7469 6465 603a 2060 444e  nucleotide`: `DN
-0000a200: 4160 206f 7220 6052 4e41 600a 2020 2020  A` or `RNA`.    
-0000a210: 2d20 606c 6f63 616c 5f70 6169 7277 6973  - `local_pairwis
-0000a220: 655f 616c 6967 6e5f 7072 6f74 6569 6e60  e_align_protein`
-0000a230: 3a20 6050 726f 7465 696e 600a 2020 2020  : `Protein`.    
-0000a240: 2d20 606c 6f63 616c 5f70 6169 7277 6973  - `local_pairwis
-0000a250: 655f 616c 6967 6e60 3a20 6049 5550 4143  e_align`: `IUPAC
-0000a260: 5365 7175 656e 6365 600a 2020 2020 2d20  Sequence`.    - 
-0000a270: 6067 6c6f 6261 6c5f 7061 6972 7769 7365  `global_pairwise
-0000a280: 5f61 6c69 676e 5f6e 7563 6c65 6f74 6964  _align_nucleotid
-0000a290: 6560 3a20 6044 4e41 602c 2060 524e 4160  e`: `DNA`, `RNA`
-0000a2a0: 2c20 6f72 2060 5461 6275 6c61 724d 5341  , or `TabularMSA
-0000a2b0: 5b44 4e41 7c52 4e41 5d60 0a20 2020 202d  [DNA|RNA]`.    -
-0000a2c0: 2060 676c 6f62 616c 5f70 6169 7277 6973   `global_pairwis
-0000a2d0: 655f 616c 6967 6e5f 7072 6f74 6569 6e60  e_align_protein`
-0000a2e0: 3a20 6050 726f 7465 696e 6020 6f72 2060  : `Protein` or `
-0000a2f0: 5461 6275 6c61 724d 5341 5b50 726f 7465  TabularMSA[Prote
-0000a300: 696e 5d60 0a20 2020 202d 2060 676c 6f62  in]`.    - `glob
-0000a310: 616c 5f70 6169 7277 6973 655f 616c 6967  al_pairwise_alig
-0000a320: 6e60 3a20 6049 5550 4143 5365 7175 656e  n`: `IUPACSequen
-0000a330: 6365 6020 6f72 2060 5461 6275 6c61 724d  ce` or `TabularM
-0000a340: 5341 600a 2020 2020 2d20 606c 6f63 616c  SA`.    - `local
-0000a350: 5f70 6169 7277 6973 655f 616c 6967 6e5f  _pairwise_align_
-0000a360: 7373 7760 3a20 6044 4e41 602c 2060 524e  ssw`: `DNA`, `RN
-0000a370: 4160 2c20 6f72 2060 5072 6f74 6569 6e60  A`, or `Protein`
-0000a380: 2e20 4164 6469 7469 6f6e 616c 6c79 2c20  . Additionally, 
-0000a390: 7468 6973 2066 756e 6374 696f 6e20 6e6f  this function no
-0000a3a0: 7720 6f76 6572 7269 6465 7320 7468 6520  w overrides the 
-0000a3b0: 6070 726f 7465 696e 6020 6b77 6172 6720  `protein` kwarg 
-0000a3c0: 6261 7365 6420 6f6e 2069 6e70 7574 2074  based on input t
-0000a3d0: 7970 652e 2060 636f 6e73 7472 7563 746f  ype. `constructo
-0000a3e0: 7260 2070 6172 616d 6574 6572 2077 6173  r` parameter was
-0000a3f0: 2072 656d 6f76 6564 2062 6563 6175 7365   removed because
-0000a400: 2074 6865 2066 756e 6374 696f 6e20 6e6f   the function no
-0000a410: 7720 6465 7465 726d 696e 6573 2074 6865  w determines the
-0000a420: 2072 6574 7572 6e20 7479 7065 2062 6173   return type bas
-0000a430: 6564 206f 6e20 696e 7075 7420 7479 7065  ed on input type
-0000a440: 2e0a 2a20 5265 6d6f 7665 6420 6073 6b62  ..* Removed `skb
-0000a450: 696f 2e61 6c69 676e 6d65 6e74 2e53 6571  io.alignment.Seq
-0000a460: 7565 6e63 6543 6f6c 6c65 6374 696f 6e60  uenceCollection`
-0000a470: 2069 6e20 6661 766f 7220 6f66 2075 7369   in favor of usi
-0000a480: 6e67 2061 206c 6973 7420 6f72 206f 7468  ng a list or oth
-0000a490: 6572 2073 7461 6e64 6172 6420 6c69 6272  er standard libr
-0000a4a0: 6172 7920 636f 6e74 6169 6e65 7273 2074  ary containers t
-0000a4b0: 6f20 7374 6f72 6520 7363 696b 6974 2d62  o store scikit-b
-0000a4c0: 696f 2073 6571 7565 6e63 6520 6f62 6a65  io sequence obje
-0000a4d0: 6374 7320 286d 6f73 7420 6053 6571 7565  cts (most `Seque
-0000a4e0: 6e63 6543 6f6c 6c65 6374 696f 6e60 206f  nceCollection` o
-0000a4f0: 7065 7261 7469 6f6e 7320 7765 7265 2073  perations were s
-0000a500: 696d 706c 6520 6c69 7374 2063 6f6d 7072  imple list compr
-0000a510: 6568 656e 7369 6f6e 7329 2e20 5573 6520  ehensions). Use 
-0000a520: 6044 6973 7461 6e63 654d 6174 7269 782e  `DistanceMatrix.
-0000a530: 6672 6f6d 5f69 7465 7261 626c 6560 2069  from_iterable` i
-0000a540: 6e73 7465 6164 206f 6620 6053 6571 7565  nstead of `Seque
-0000a550: 6e63 6543 6f6c 6c65 6374 696f 6e2e 6469  nceCollection.di
-0000a560: 7374 616e 6365 7360 2028 7061 7373 2060  stances` (pass `
-0000a570: 6b65 793d 2269 6422 6020 746f 2065 7861  key="id"` to exa
-0000a580: 6374 6c79 206d 6174 6368 206f 7269 6769  ctly match origi
-0000a590: 6e61 6c20 6265 6861 7669 6f72 292e 0a2a  nal behavior)..*
-0000a5a0: 2052 656d 6f76 6564 2060 736b 6269 6f2e   Removed `skbio.
-0000a5b0: 616c 6967 6e6d 656e 742e 416c 6967 6e6d  alignment.Alignm
-0000a5c0: 656e 7460 2069 6e20 6661 766f 7220 6f66  ent` in favor of
-0000a5d0: 2060 736b 6269 6f2e 616c 6967 6e6d 656e   `skbio.alignmen
-0000a5e0: 742e 5461 6275 6c61 724d 5341 602e 0a2a  t.TabularMSA`..*
-0000a5f0: 2052 656d 6f76 6564 2060 736b 6269 6f2e   Removed `skbio.
-0000a600: 616c 6967 6e6d 656e 742e 5365 7175 656e  alignment.Sequen
-0000a610: 6365 436f 6c6c 6563 7469 6f6e 4572 726f  ceCollectionErro
-0000a620: 7260 2061 6e64 2060 736b 6269 6f2e 616c  r` and `skbio.al
-0000a630: 6967 6e6d 656e 742e 416c 6967 6e6d 656e  ignment.Alignmen
-0000a640: 7445 7272 6f72 6020 6578 6365 7074 696f  tError` exceptio
-0000a650: 6e73 2061 7320 7468 6569 7220 636f 7272  ns as their corr
-0000a660: 6573 706f 6e64 696e 6720 636c 6173 7365  esponding classe
-0000a670: 7320 6e6f 206c 6f6e 6765 7220 6578 6973  s no longer exis
-0000a680: 742e 0a0a 2323 2320 4275 6720 4669 7865  t...### Bug Fixe
-0000a690: 730a 0a2a 2060 6053 6571 7565 6e63 6560  s..* ``Sequence`
-0000a6a0: 6020 6f62 6a65 6374 7320 6e6f 7720 6861  ` objects now ha
-0000a6b0: 6e64 6c65 2073 6c69 6369 6e67 206f 6620  ndle slicing of 
-0000a6c0: 656d 7074 7920 706f 7369 7469 6f6e 616c  empty positional
-0000a6d0: 206d 6574 6164 6174 6120 636f 7272 6563   metadata correc
-0000a6e0: 746c 792e 2041 6e79 206d 6574 6164 6174  tly. Any metadat
-0000a6f0: 6120 7468 6174 2069 7320 656d 7074 7920  a that is empty 
-0000a700: 7769 6c6c 206e 6f20 6c6f 6e67 6572 2062  will no longer b
-0000a710: 6520 7072 6f70 6167 6174 6564 2062 7920  e propagated by 
-0000a720: 7468 6520 696e 7465 726e 616c 2060 605f  the internal ``_
-0000a730: 746f 6060 2063 6f6e 7374 7275 6374 6f72  to`` constructor
-0000a740: 2e20 285b 2331 3133 335d 2868 7474 7073  . ([#1133](https
-0000a750: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
-0000a760: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
-0000a770: 2f69 7373 7565 732f 3131 3333 2929 0a2a  /issues/1133)).*
-0000a780: 2060 6044 6973 7369 6d69 6c61 7269 7479   ``Dissimilarity
-0000a790: 4d61 7472 6978 2e70 6c6f 7428 2960 6020  Matrix.plot()`` 
-0000a7a0: 6e6f 206c 6f6e 6765 7220 6c65 6176 6573  no longer leaves
-0000a7b0: 2061 2077 6869 7465 2062 6f72 6465 7220   a white border 
-0000a7c0: 6172 6f75 6e64 2074 6865 0a20 2068 6561  around the.  hea
-0000a7d0: 746d 6170 2069 7420 706c 6f74 7320 2850  tmap it plots (P
-0000a7e0: 5220 2331 3037 3029 2e0a 2a20 5472 6565  R #1070)..* Tree
-0000a7f0: 4e6f 6465 2e72 6f6f 745f 6174 5f6d 6964  Node.root_at_mid
-0000a800: 706f 696e 7460 6020 6e6f 206c 6f6e 6765  point`` no longe
-0000a810: 7220 6661 696c 7320 7768 656e 2061 206e  r fails when a n
-0000a820: 6f64 6520 7769 7468 2074 776f 2065 7175  ode with two equ
-0000a830: 616c 206c 656e 6774 6820 6368 696c 6420  al length child 
-0000a840: 6272 616e 6368 6573 2065 7869 7374 7320  branches exists 
-0000a850: 696e 2074 6865 2074 7265 652e 2028 5b23  in the tree. ([#
-0000a860: 3130 3737 5d28 6874 7470 733a 2f2f 6769  1077](https://gi
-0000a870: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
-0000a880: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
-0000a890: 6573 2f31 3037 3729 290a 2a20 6060 5472  es/1077)).* ``Tr
-0000a8a0: 6565 4e6f 6465 2e5f 7365 745f 6d61 785f  eeNode._set_max_
-0000a8b0: 6469 7374 616e 6365 6060 2c20 6173 2063  distance``, as c
-0000a8c0: 616c 6c65 6420 7468 726f 7567 6820 6060  alled through ``
-0000a8d0: 5472 6565 4e6f 6465 2e67 6574 5f6d 6178  TreeNode.get_max
-0000a8e0: 5f64 6973 7461 6e63 6560 6020 6f72 2060  _distance`` or `
-0000a8f0: 6054 7265 654e 6f64 652e 726f 6f74 5f61  `TreeNode.root_a
-0000a900: 745f 6d69 6470 6f69 6e74 6060 2077 6f75  t_midpoint`` wou
-0000a910: 6c64 2073 746f 7265 2064 6973 7461 6e63  ld store distanc
-0000a920: 6520 696e 666f 726d 6174 696f 6e20 6173  e information as
-0000a930: 2060 606c 6973 7460 6073 2069 6e20 7468   ``list``s in th
-0000a940: 6520 6174 7472 6962 7574 6520 6060 4d61  e attribute ``Ma
-0000a950: 7844 6973 7454 6970 7360 6020 6f6e 2065  xDistTips`` on e
-0000a960: 6163 6820 6e6f 6465 2069 6e20 7468 6520  ach node in the 
-0000a970: 7472 6565 2c20 686f 7765 7665 722c 2074  tree, however, t
-0000a980: 6865 7365 2064 6973 7461 6e63 6573 2077  hese distances w
-0000a990: 6572 6520 6f6e 6c79 2076 616c 6964 2066  ere only valid f
-0000a9a0: 6f72 2074 6865 206e 6f64 6520 696e 2077  or the node in w
-0000a9b0: 6869 6368 2074 6865 2063 616c 6c20 746f  hich the call to
-0000a9c0: 2060 605f 7365 745f 6d61 785f 6469 7374   ``_set_max_dist
-0000a9d0: 616e 6365 6060 2077 6173 206d 6164 652e  ance`` was made.
-0000a9e0: 2054 6865 2076 616c 7565 7320 636f 6e74   The values cont
-0000a9f0: 6169 6e65 6420 696e 2060 604d 6178 4469  ained in ``MaxDi
-0000aa00: 7374 5469 7073 6060 2061 7265 206e 6f77  stTips`` are now
-0000aa10: 2063 6f72 7265 6374 2061 6372 6f73 7320   correct across 
-0000aa20: 7468 6520 7472 6565 2066 6f6c 6c6f 7769  the tree followi
-0000aa30: 6e67 2061 2063 616c 6c20 746f 2060 6067  ng a call to ``g
-0000aa40: 6574 5f6d 6178 5f64 6973 7461 6e63 6560  et_max_distance`
-0000aa50: 602e 2054 6865 2073 636f 7065 206f 6620  `. The scope of 
-0000aa60: 696d 7061 6374 206f 6620 7468 6973 2062  impact of this b
-0000aa70: 7567 2069 7320 6c69 6d69 7465 6420 746f  ug is limited to
-0000aa80: 2075 7365 7273 2074 6861 7420 7765 7265   users that were
-0000aa90: 2069 6e74 6572 6163 7469 6e67 2064 6972   interacting dir
-0000aaa0: 6563 746c 7920 7769 7468 2060 604d 6178  ectly with ``Max
-0000aab0: 4469 7374 5469 7073 6060 206f 6e20 6465  DistTips`` on de
-0000aac0: 7363 656e 6461 6e74 206e 6f64 6573 3b20  scendant nodes; 
-0000aad0: 7468 6973 2062 7567 2064 6f65 7320 6e6f  this bug does no
-0000aae0: 7420 696d 7061 6374 2061 6e79 206b 6e6f  t impact any kno
-0000aaf0: 776e 206d 6574 686f 6420 7769 7468 696e  wn method within
-0000ab00: 2073 6369 6b69 742d 6269 6f2e 2028 5b23   scikit-bio. ([#
-0000ab10: 3132 3233 5d28 6874 7470 733a 2f2f 6769  1223](https://gi
-0000ab20: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
-0000ab30: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
-0000ab40: 6573 2f31 3232 3329 290a 2a20 4164 6465  es/1223)).* Adde
-0000ab50: 6420 6d69 7373 696e 6720 606e 6f73 6560  d missing `nose`
-0000ab60: 2064 6570 656e 6465 6e63 7920 746f 2073   dependency to s
-0000ab70: 6574 7570 2e70 7927 7320 6069 6e73 7461  etup.py's `insta
-0000ab80: 6c6c 5f72 6571 7569 7265 7360 2e20 285b  ll_requires`. ([
-0000ab90: 2331 3231 345d 2868 7474 7073 3a2f 2f67  #1214](https://g
-0000aba0: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
-0000abb0: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
-0000abc0: 7565 732f 3132 3134 2929 0a2a 2046 6978  ues/1214)).* Fix
-0000abd0: 6564 2069 7373 7565 2074 6861 7420 7265  ed issue that re
-0000abe0: 7375 6c74 6564 2069 6e20 6c65 6765 6e64  sulted in legend
-0000abf0: 7320 6f66 2060 604f 7264 696e 6174 696f  s of ``Ordinatio
-0000ac00: 6e52 6573 756c 7460 6020 706c 6f74 7320  nResult`` plots 
-0000ac10: 736f 6d65 7469 6d65 7320 6265 696e 6720  sometimes being 
-0000ac20: 7472 756e 6361 7465 642e 2028 5b23 3132  truncated. ([#12
-0000ac30: 3130 5d28 6874 7470 733a 2f2f 6769 7468  10](https://gith
-0000ac40: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
-0000ac50: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
-0000ac60: 2f31 3231 3029 290a 0a23 2323 2044 6570  /1210))..### Dep
-0000ac70: 7265 6361 7465 6420 6675 6e63 7469 6f6e  recated function
-0000ac80: 616c 6974 7920 5b73 7461 626c 655d 0a2a  ality [stable].*
-0000ac90: 2060 736b 6269 6f2e 5365 7175 656e 6365   `skbio.Sequence
-0000aca0: 2e63 6f70 7960 2068 6173 2062 6565 6e20  .copy` has been 
-0000acb0: 6465 7072 6563 6174 6564 2069 6e20 6661  deprecated in fa
-0000acc0: 766f 7220 6f66 2060 636f 7079 2e63 6f70  vor of `copy.cop
-0000acd0: 7928 7365 7129 6020 616e 6420 6063 6f70  y(seq)` and `cop
-0000ace0: 792e 6465 6570 636f 7079 2873 6571 2960  y.deepcopy(seq)`
-0000acf0: 2e0a 0a23 2323 204d 6973 6365 6c6c 616e  ...### Miscellan
-0000ad00: 656f 7573 0a2a 2044 6f63 7465 7374 7320  eous.* Doctests 
-0000ad10: 6172 6520 6e6f 7720 7772 6974 7465 6e20  are now written 
-0000ad20: 696e 2050 7974 686f 6e20 332e 0a2a 2060  in Python 3..* `
-0000ad30: 606d 616b 6520 7465 7374 6060 206e 6f77  `make test`` now
-0000ad40: 2076 616c 6964 6174 6573 204d 414e 4946   validates MANIF
-0000ad50: 4553 542e 696e 2075 7369 6e67 205b 6368  EST.in using [ch
-0000ad60: 6563 6b2d 6d61 6e69 6665 7374 5d28 6874  eck-manifest](ht
-0000ad70: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
-0000ad80: 2f6d 6765 646d 696e 2f63 6865 636b 2d6d  /mgedmin/check-m
-0000ad90: 616e 6966 6573 7429 2e20 285b 2334 3631  anifest). ([#461
-0000ada0: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
-0000adb0: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
-0000adc0: 6b69 742d 6269 6f2f 6973 7375 6573 2f34  kit-bio/issues/4
-0000add0: 3631 2929 0a2a 204d 616e 7920 6e65 7720  61)).* Many new 
-0000ade0: 616c 7068 6120 6469 7665 7273 6974 7920  alpha diversity 
-0000adf0: 6571 7561 7469 6f6e 7320 6164 6465 6420  equations added 
-0000ae00: 746f 2060 6073 6b62 696f 2e64 6976 6572  to ``skbio.diver
-0000ae10: 7369 7479 2e61 6c70 6861 6060 2064 6f63  sity.alpha`` doc
-0000ae20: 756d 656e 7461 7469 6f6e 2e20 285b 2333  umentation. ([#3
-0000ae30: 3231 5d28 6874 7470 733a 2f2f 6769 7468  21](https://gith
-0000ae40: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
-0000ae50: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
-0000ae60: 2f33 3231 2929 0a2a 204f 7264 6572 206f  /321)).* Order o
-0000ae70: 6620 6060 6c6f 7765 7263 6173 6560 6020  f ``lowercase`` 
-0000ae80: 616e 6420 6060 7661 6c69 6461 7465 6060  and ``validate``
-0000ae90: 206b 6579 776f 7264 7320 7377 6170 7065   keywords swappe
-0000aea0: 6420 696e 2060 6044 4e41 6060 2c20 6060  d in ``DNA``, ``
-0000aeb0: 524e 4160 602c 2061 6e64 2060 6050 726f  RNA``, and ``Pro
-0000aec0: 7465 696e 6060 2e0a 0a23 2320 5665 7273  tein``...## Vers
-0000aed0: 696f 6e20 302e 342e 3020 2832 3031 352d  ion 0.4.0 (2015-
-0000aee0: 3037 2d30 3829 0a0a 496e 6974 6961 6c20  07-08)..Initial 
-0000aef0: 6265 7461 2072 656c 6561 7365 2e20 496e  beta release. In
-0000af00: 2061 6464 6974 696f 6e20 746f 2074 6865   addition to the
-0000af10: 2063 6861 6e67 6573 2064 6574 6169 6c65   changes detaile
-0000af20: 6420 6265 6c6f 772c 2074 6865 2066 6f6c  d below, the fol
-0000af30: 6c6f 7769 6e67 0a73 7562 7061 636b 6167  lowing.subpackag
-0000af40: 6573 2068 6176 6520 6265 656e 206d 6f73  es have been mos
-0000af50: 746c 7920 6f72 2065 6e74 6972 656c 7920  tly or entirely 
-0000af60: 7265 7772 6974 7465 6e20 616e 6420 6d6f  rewritten and mo
-0000af70: 7374 206f 6620 7468 6569 7220 4150 4973  st of their APIs
-0000af80: 2061 7265 0a73 7562 7374 616e 7469 616c   are.substantial
-0000af90: 6c79 2064 6966 6665 7265 6e74 2028 616e  ly different (an
-0000afa0: 6420 696d 7072 6f76 6564 2129 3a0a 0a2a  d improved!):..*
-0000afb0: 2060 736b 6269 6f2e 7365 7175 656e 6365   `skbio.sequence
-0000afc0: 600a 2a20 6073 6b62 696f 2e69 6f60 0a0a  `.* `skbio.io`..
-0000afd0: 5468 6520 4150 4973 206f 6620 7468 6573  The APIs of thes
-0000afe0: 6520 7375 6270 6163 6b61 6765 7320 6172  e subpackages ar
-0000aff0: 6520 6e6f 7720 7374 6162 6c65 2c20 616e  e now stable, an
-0000b000: 6420 616c 6c20 6f74 6865 7273 2061 7265  d all others are
-0000b010: 2065 7870 6572 696d 656e 7461 6c2e 2053   experimental. S
-0000b020: 6565 2074 6865 205b 4150 4920 7374 6162  ee the [API stab
-0000b030: 696c 6974 7920 646f 6373 5d28 6874 7470  ility docs](http
-0000b040: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-0000b050: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-0000b060: 6f2f 7472 6565 2f30 2e34 2e30 2f64 6f63  o/tree/0.4.0/doc
-0000b070: 2f73 6f75 7263 652f 7573 6572 2f61 7069  /source/user/api
-0000b080: 5f73 7461 6269 6c69 7479 2e72 7374 2920  _stability.rst) 
-0000b090: 666f 7220 6d6f 7265 2064 6574 6169 6c73  for more details
-0000b0a0: 2c20 696e 636c 7564 696e 6720 7768 6174  , including what
-0000b0b0: 2077 6520 6d65 616e 2062 7920 2a73 7461   we mean by *sta
-0000b0c0: 626c 652a 2061 6e64 202a 6578 7065 7269  ble* and *experi
-0000b0d0: 6d65 6e74 616c 2a20 696e 2074 6869 7320  mental* in this 
-0000b0e0: 636f 6e74 6578 742e 2057 6520 7265 636f  context. We reco
-0000b0f0: 676e 697a 6520 7468 6174 2074 6869 7320  gnize that this 
-0000b100: 6973 2061 206c 6f74 206f 6620 6261 636b  is a lot of back
-0000b110: 7761 7264 2d69 6e63 6f6d 7061 7469 626c  ward-incompatibl
-0000b120: 6520 6368 616e 6765 732e 2054 6f20 6176  e changes. To av
-0000b130: 6f69 6420 7468 6573 6520 7479 7065 7320  oid these types 
-0000b140: 6f66 2063 6861 6e67 6573 2062 6569 6e67  of changes being
-0000b150: 2061 2073 7572 7072 6973 6520 746f 206f   a surprise to o
-0000b160: 7572 2075 7365 7273 2c20 6f75 7220 7075  ur users, our pu
-0000b170: 626c 6963 2041 5049 7320 6172 6520 6e6f  blic APIs are no
-0000b180: 7720 6465 636f 7261 7465 6420 746f 206d  w decorated to m
-0000b190: 616b 6520 6974 2063 6c65 6172 2074 6f20  ake it clear to 
-0000b1a0: 6465 7665 6c6f 7065 7273 2077 6865 6e20  developers when 
-0000b1b0: 616e 2041 5049 2063 616e 2062 6520 7265  an API can be re
-0000b1c0: 6c69 6564 2075 706f 6e20 2873 7461 626c  lied upon (stabl
-0000b1d0: 6529 2061 6e64 2077 6865 6e20 6974 206d  e) and when it m
-0000b1e0: 6179 2062 6520 7375 626a 6563 7420 746f  ay be subject to
-0000b1f0: 2063 6861 6e67 6520 2865 7870 6572 696d   change (experim
-0000b200: 656e 7461 6c29 2e0a 0a23 2323 2046 6561  ental)...### Fea
-0000b210: 7475 7265 730a 2a20 4164 6465 6420 6073  tures.* Added `s
-0000b220: 6b62 696f 2e73 7461 7473 2e63 6f6d 706f  kbio.stats.compo
-0000b230: 7369 7469 6f6e 6020 666f 7220 616e 616c  sition` for anal
-0000b240: 797a 696e 6720 6461 7461 206d 6164 6520  yzing data made 
-0000b250: 7570 206f 6620 7072 6f70 6f72 7469 6f6e  up of proportion
-0000b260: 730a 2a20 4164 6465 6420 6e65 7720 6060  s.* Added new ``
-0000b270: 736b 6269 6f2e 7374 6174 732e 6576 6f6c  skbio.stats.evol
-0000b280: 7665 6060 2073 7562 7061 636b 6167 6520  ve`` subpackage 
-0000b290: 666f 7220 6576 6f6c 7574 696f 6e61 7279  for evolutionary
-0000b2a0: 2073 7461 7469 7374 6963 732e 2043 7572   statistics. Cur
-0000b2b0: 7265 6e74 6c79 2063 6f6e 7461 696e 7320  rently contains 
-0000b2c0: 6120 7369 6e67 6c65 2066 756e 6374 696f  a single functio
-0000b2d0: 6e2c 2060 6068 6f6d 6d6f 6c61 5f63 6f73  n, ``hommola_cos
-0000b2e0: 7065 6369 6174 696f 6e60 602c 2077 6869  peciation``, whi
-0000b2f0: 6368 2069 6d70 6c65 6d65 6e74 7320 6120  ch implements a 
-0000b300: 7065 726d 7574 6174 696f 6e2d 6261 7365  permutation-base
-0000b310: 6420 7465 7374 206f 6620 636f 7272 656c  d test of correl
-0000b320: 6174 696f 6e20 6265 7477 6565 6e20 7477  ation between tw
-0000b330: 6f20 6469 7374 616e 6365 206d 6174 7269  o distance matri
-0000b340: 6365 732e 0a2a 2041 6464 6564 2073 7570  ces..* Added sup
-0000b350: 706f 7274 2066 6f72 2060 6073 6b62 696f  port for ``skbio
-0000b360: 2e69 6f2e 7574 696c 2e6f 7065 6e5f 6669  .io.util.open_fi
-0000b370: 6c65 6060 2061 6e64 2060 6073 6b62 696f  le`` and ``skbio
-0000b380: 2e69 6f2e 7574 696c 2e6f 7065 6e5f 6669  .io.util.open_fi
-0000b390: 6c65 7360 6020 746f 2070 756c 6c20 6669  les`` to pull fi
-0000b3a0: 6c65 7320 6672 6f6d 2048 5454 5020 616e  les from HTTP an
-0000b3b0: 6420 4854 5450 5320 5552 4c73 2e20 5468  d HTTPS URLs. Th
-0000b3c0: 6973 2062 6568 6176 696f 7220 7072 6f70  is behavior prop
-0000b3d0: 6167 6174 6573 2074 6f20 7468 6520 492f  agates to the I/
-0000b3e0: 4f20 7265 6769 7374 7279 2e0a 2a20 4641  O registry..* FA
-0000b3f0: 5354 412f 5155 414c 2028 6060 736b 6269  STA/QUAL (``skbi
-0000b400: 6f2e 696f 2e66 6f72 6d61 742e 6661 7374  o.io.format.fast
-0000b410: 6160 6029 2061 6e64 2046 4153 5451 2028  a``) and FASTQ (
-0000b420: 6060 736b 6269 6f2e 696f 2e66 6f72 6d61  ``skbio.io.forma
-0000b430: 742e 6661 7374 7160 6029 2072 6561 6465  t.fastq``) reade
-0000b440: 7273 206e 6f77 2061 6c6c 6f77 2062 6c61  rs now allow bla
-0000b450: 6e6b 206f 7220 7768 6974 6573 7061 6365  nk or whitespace
-0000b460: 2d6f 6e6c 7920 6c69 6e65 7320 6174 2074  -only lines at t
-0000b470: 6865 2062 6567 696e 6e69 6e67 206f 6620  he beginning of 
-0000b480: 7468 6520 6669 6c65 2c20 6265 7477 6565  the file, betwee
-0000b490: 6e20 7265 636f 7264 732c 206f 7220 6174  n records, or at
-0000b4a0: 2074 6865 2065 6e64 206f 6620 7468 6520   the end of the 
-0000b4b0: 6669 6c65 2e20 4120 626c 616e 6b20 6f72  file. A blank or
-0000b4c0: 2077 6869 7465 7370 6163 652d 6f6e 6c79   whitespace-only
-0000b4d0: 206c 696e 6520 696e 2061 6e79 206f 7468   line in any oth
-0000b4e0: 6572 206c 6f63 6174 696f 6e20 7769 6c6c  er location will
-0000b4f0: 2063 6f6e 7469 6e75 6520 746f 2072 6169   continue to rai
-0000b500: 7365 2061 6e20 6572 726f 7220 5b23 3738  se an error [#78
-0000b510: 315d 2868 7474 7073 3a2f 2f67 6974 6875  1](https://githu
-0000b520: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
-0000b530: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
-0000b540: 3738 3129 2e0a 2a20 7363 696b 6974 2d62  781)..* scikit-b
-0000b550: 696f 206e 6f77 2069 676e 6f72 6573 206c  io now ignores l
-0000b560: 6561 6469 6e67 2061 6e64 2074 7261 696c  eading and trail
-0000b570: 696e 6720 7768 6974 6573 7061 6365 2063  ing whitespace c
-0000b580: 6861 7261 6374 6572 7320 6f6e 2065 6163  haracters on eac
-0000b590: 6820 6c69 6e65 2077 6869 6c65 2072 6561  h line while rea
-0000b5a0: 6469 6e67 2046 4153 5441 2f51 5541 4c20  ding FASTA/QUAL 
-0000b5b0: 616e 6420 4641 5354 5120 6669 6c65 732e  and FASTQ files.
-0000b5c0: 0a2a 2041 6464 6564 2060 7261 7469 6f60  .* Added `ratio`
-0000b5d0: 2070 6172 616d 6574 6572 2074 6f20 6073   parameter to `s
-0000b5e0: 6b62 696f 2e73 7461 7473 2e70 6f77 6572  kbio.stats.power
-0000b5f0: 2e73 7562 7361 6d70 6c65 5f70 6f77 6572  .subsample_power
-0000b600: 602e 2054 6869 7320 616c 6c6f 7773 2074  `. This allows t
-0000b610: 6865 2075 7365 7220 746f 2063 616c 6375  he user to calcu
-0000b620: 6c61 7465 2070 6f77 6572 206f 6e20 6772  late power on gr
-0000b630: 6f75 7073 2066 6f72 2075 6e65 7665 6e20  oups for uneven 
-0000b640: 7369 7a65 2028 466f 7220 6578 616d 706c  size (For exampl
-0000b650: 652c 2064 7261 7720 7477 6963 6520 6173  e, draw twice as
-0000b660: 206d 616e 7920 7361 6d70 6c65 7320 6672   many samples fr
-0000b670: 6f6d 2047 726f 7570 2042 2074 6861 6e20  om Group B than 
-0000b680: 4772 6f75 7020 4129 2e20 4966 2060 7261  Group A). If `ra
-0000b690: 7469 6f60 2069 7320 6e6f 7420 7365 742c  tio` is not set,
-0000b6a0: 2067 726f 7570 2073 697a 6573 2077 696c   group sizes wil
-0000b6b0: 6c20 7265 6d61 696e 2065 7175 616c 2061  l remain equal a
-0000b6c0: 6372 6f73 7320 616c 6c20 6772 6f75 7073  cross all groups
-0000b6d0: 2e0a 2a20 506f 7765 7220 6361 6c63 756c  ..* Power calcul
-0000b6e0: 6174 696f 6e73 2028 6073 6b62 696f 2e73  ations (`skbio.s
-0000b6f0: 7461 7473 2e70 6f77 6572 2e73 7562 7361  tats.power.subsa
-0000b700: 6d70 6c65 5f70 6f77 6572 6020 616e 6420  mple_power` and 
-0000b710: 6073 6b62 696f 2e73 7461 7473 2e70 6f77  `skbio.stats.pow
-0000b720: 6572 2e73 7562 7361 6d70 6c65 5f70 6169  er.subsample_pai
-0000b730: 7265 645f 706f 7765 7260 2920 6361 6e20  red_power`) can 
-0000b740: 7573 6520 7465 7374 2066 756e 6374 696f  use test functio
-0000b750: 6e73 2074 6861 7420 7265 7475 726e 206d  ns that return m
-0000b760: 756c 7469 706c 6520 7020 7661 6c75 6573  ultiple p values
-0000b770: 2c20 6c69 6b65 2073 6f6d 6520 6d75 6c74  , like some mult
-0000b780: 6976 6172 6961 7465 206c 696e 6561 7220  ivariate linear 
-0000b790: 7265 6772 6573 7369 6f6e 206d 6f64 656c  regression model
-0000b7a0: 732e 2050 7265 7669 6f75 736c 792c 2074  s. Previously, t
-0000b7b0: 6865 2070 6f77 6572 2063 616c 6375 6c61  he power calcula
-0000b7c0: 7469 6f6e 7320 7265 7175 6972 6564 2074  tions required t
-0000b7d0: 6865 2074 6573 7420 746f 2072 6574 7572  he test to retur
-0000b7e0: 6e20 6120 7369 6e67 6c65 2070 2076 616c  n a single p val
-0000b7f0: 7565 2e0a 2a20 4164 6465 6420 6060 736b  ue..* Added ``sk
-0000b800: 6269 6f2e 7574 696c 2e61 7373 6572 745f  bio.util.assert_
-0000b810: 6461 7461 5f66 7261 6d65 5f61 6c6d 6f73  data_frame_almos
-0000b820: 745f 6571 7561 6c60 6020 6675 6e63 7469  t_equal`` functi
-0000b830: 6f6e 2066 6f72 2063 6f6d 7061 7269 6e67  on for comparing
-0000b840: 2060 6070 642e 4461 7461 4672 616d 6560   ``pd.DataFrame`
-0000b850: 6020 6f62 6a65 6374 7320 696e 2075 6e69  ` objects in uni
-0000b860: 7420 7465 7374 732e 0a0a 2323 2320 5065  t tests...### Pe
-0000b870: 7266 6f72 6d61 6e63 6520 656e 6861 6e63  rformance enhanc
-0000b880: 656d 656e 7473 0a2a 2054 6865 2073 7065  ements.* The spe
-0000b890: 6564 206f 6620 7175 616c 6974 7920 7363  ed of quality sc
-0000b8a0: 6f72 6520 6465 636f 6469 6e67 2068 6173  ore decoding has
-0000b8b0: 2062 6565 6e20 7369 676e 6966 6963 616e   been significan
-0000b8c0: 746c 7920 696d 7072 6f76 6564 2028 7e32  tly improved (~2
-0000b8d0: 7829 2077 6865 6e20 7265 6164 696e 6720  x) when reading 
-0000b8e0: 6066 6173 7471 6020 6669 6c65 732e 0a2a  `fastq` files..*
-0000b8f0: 2054 6865 2073 7065 6564 206f 6620 604e   The speed of `N
-0000b900: 7563 6c65 6f74 6964 6553 6571 7565 6e63  ucleotideSequenc
-0000b910: 652e 7265 7665 7273 655f 636f 6d70 6c65  e.reverse_comple
-0000b920: 6d65 6e74 6020 6861 7320 6265 656e 2069  ment` has been i
-0000b930: 6d70 726f 7665 6420 287e 3678 292e 0a0a  mproved (~6x)...
-0000b940: 2323 2320 4275 6720 6669 7865 730a 2a20  ### Bug fixes.* 
-0000b950: 4368 616e 6765 6420 6053 6571 7565 6e63  Changed `Sequenc
-0000b960: 652e 6469 7374 616e 6365 6020 746f 2072  e.distance` to r
-0000b970: 6169 7365 2061 6e20 6572 726f 7220 616e  aise an error an
-0000b980: 7920 7469 6d65 2074 776f 2073 6571 7565  y time two seque
-0000b990: 6e63 6573 2061 7265 2070 6173 7365 6420  nces are passed 
-0000b9a0: 6f66 2064 6966 6665 7265 6e74 206c 656e  of different len
-0000b9b0: 6774 6873 2072 6567 6172 646c 6573 7320  gths regardless 
-0000b9c0: 6f66 2074 6865 2060 6469 7374 616e 6365  of the `distance
-0000b9d0: 5f66 6e60 2062 6569 6e67 2070 6173 7365  _fn` being passe
-0000b9e0: 642e 205b 2823 3531 3429 5d28 6874 7470  d. [(#514)](http
-0000b9f0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-0000ba00: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-0000ba10: 6f2f 6973 7375 6573 2f35 3134 290a 2a20  o/issues/514).* 
-0000ba20: 4669 7865 6420 6973 7375 6520 7769 7468  Fixed issue with
-0000ba30: 2060 6054 7265 654e 6f64 652e 6578 7465   ``TreeNode.exte
-0000ba40: 6e64 6060 2077 6865 7265 2069 6620 6769  nd`` where if gi
-0000ba50: 7665 6e20 7468 6520 6368 696c 6472 656e  ven the children
-0000ba60: 206f 6620 616e 6f74 6865 7220 6060 5472   of another ``Tr
-0000ba70: 6565 4e6f 6465 6060 206f 626a 6563 7420  eeNode`` object 
-0000ba80: 2860 6074 7265 652e 6368 696c 6472 656e  (``tree.children
-0000ba90: 6060 292c 2062 6f74 6820 7472 6565 7320  ``), both trees 
-0000baa0: 776f 756c 6420 6265 206c 6566 7420 696e  would be left in
-0000bab0: 2061 6e20 696e 636f 7272 6563 7420 616e   an incorrect an
-0000bac0: 6420 756e 7072 6564 6963 7461 626c 6520  d unpredictable 
-0000bad0: 7374 6174 652e 2028 5b23 3838 395d 2868  state. ([#889](h
-0000bae0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-0000baf0: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
-0000bb00: 2d62 696f 2f69 7373 7565 732f 3838 3929  -bio/issues/889)
-0000bb10: 290a 2a20 4368 616e 6765 6420 7468 6520  ).* Changed the 
-0000bb20: 7761 7920 706f 7765 7220 7761 7320 6361  way power was ca
-0000bb30: 6c63 756c 6174 6564 2069 6e20 6073 7562  lculated in `sub
-0000bb40: 7361 6d70 6c65 5f70 6169 7265 645f 706f  sample_paired_po
-0000bb50: 7765 7260 2074 6f20 6d6f 7665 2074 6865  wer` to move the
-0000bb60: 2073 7562 7361 6d70 6c65 2073 656c 6563   subsample selec
-0000bb70: 7469 6f6e 2062 6566 6f72 6520 7468 6520  tion before the 
-0000bb80: 7465 7374 2069 7320 7065 7266 6f72 6d65  test is performe
-0000bb90: 642e 2054 6869 7320 696e 6372 6561 7365  d. This increase
-0000bba0: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
-0000bbb0: 4d6f 6e74 6520 4361 726c 6f20 7369 6d75  Monte Carlo simu
-0000bbc0: 6c61 7469 6f6e 7320 7065 7266 6f72 6d65  lations performe
-0000bbd0: 6420 6475 7269 6e67 2070 6f77 6572 2065  d during power e
-0000bbe0: 7374 696d 6174 696f 6e2c 2061 6e64 2069  stimation, and i
-0000bbf0: 6d70 726f 7665 7320 7468 6520 6163 6375  mproves the accu
-0000bc00: 7261 6379 206f 6620 7468 6520 7265 7475  racy of the retu
-0000bc10: 726e 6564 2065 7374 696d 6174 652e 2050  rned estimate. P
-0000bc20: 7265 7669 6f75 7320 706f 7765 7220 6573  revious power es
-0000bc30: 7469 6d61 7465 7320 6672 6f6d 2060 7375  timates from `su
-0000bc40: 6273 616d 706c 655f 7061 6972 6564 5f70  bsample_paired_p
-0000bc50: 6f77 6572 6020 7368 6f75 6c64 2062 6520  ower` should be 
-0000bc60: 6469 7372 6567 6172 6465 6420 616e 6420  disregarded and 
-0000bc70: 7265 2d63 616c 6375 6c61 7465 642e 2028  re-calculated. (
-0000bc80: 5b23 3931 305d 2868 7474 7073 3a2f 2f67  [#910](https://g
-0000bc90: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
-0000bca0: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
-0000bcb0: 7565 732f 3931 3029 290a 2a20 4669 7865  ues/910)).* Fixe
-0000bcc0: 6420 6973 7375 6520 7768 6572 6520 6072  d issue where `r
-0000bcd0: 616e 6464 6d60 2077 6173 2061 7474 656d  anddm` was attem
-0000bce0: 7074 696e 6720 746f 2063 7265 6174 6520  pting to create 
-0000bcf0: 6173 796d 6d65 7472 6963 2064 6973 7461  asymmetric dista
-0000bd00: 6e63 6520 6d61 7472 6963 6573 2e54 6869  nce matrices.Thi
-0000bd10: 7320 7761 7320 6361 7573 696e 6720 616e  s was causing an
-0000bd20: 2065 7272 6f72 2074 6f20 6265 2072 6169   error to be rai
-0000bd30: 7365 6420 6279 2074 6865 2060 4469 7374  sed by the `Dist
-0000bd40: 616e 6365 4d61 7472 6978 6020 636f 6e73  anceMatrix` cons
-0000bd50: 7472 7563 746f 7220 696e 7369 6465 206f  tructor inside o
-0000bd60: 6620 7468 6520 6072 616e 6464 6d60 2066  f the `randdm` f
-0000bd70: 756e 6374 696f 6e2c 2073 6f20 7468 6174  unction, so that
-0000bd80: 2060 7261 6e64 646d 6020 776f 756c 6420   `randdm` would 
-0000bd90: 6661 696c 2077 6865 6e20 6174 7465 6d70  fail when attemp
-0000bda0: 7469 6e67 2074 6f20 6372 6561 7465 206c  ting to create l
-0000bdb0: 6172 6765 2064 6973 7461 6e63 6520 6d61  arge distance ma
-0000bdc0: 7472 6963 6573 2e20 285b 2339 3433 5d28  trices. ([#943](
-0000bdd0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-0000bde0: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
-0000bdf0: 742d 6269 6f2f 6973 7375 6573 2f39 3433  t-bio/issues/943
-0000be00: 2929 0a0a 2323 2320 4465 7072 6563 6174  ))..### Deprecat
-0000be10: 6564 2066 756e 6374 696f 6e61 6c69 7479  ed functionality
-0000be20: 0a2a 2044 6570 7265 6361 7465 6420 6073  .* Deprecated `s
-0000be30: 6b62 696f 2e75 7469 6c2e 666c 6174 7465  kbio.util.flatte
-0000be40: 6e60 2e20 5468 6973 2066 756e 6374 696f  n`. This functio
-0000be50: 6e20 7769 6c6c 2062 6520 7265 6d6f 7665  n will be remove
-0000be60: 6420 696e 2073 6369 6b69 742d 6269 6f20  d in scikit-bio 
-0000be70: 302e 332e 312e 2050 6c65 6173 6520 7573  0.3.1. Please us
-0000be80: 6520 7374 616e 6461 7264 2070 7974 686f  e standard pytho
-0000be90: 6e20 6c69 6272 6172 7920 6675 6e63 7469  n library functi
-0000bea0: 6f6e 616c 6974 790a 6465 7363 7269 6265  onality.describe
-0000beb0: 6420 6865 7265 205b 4d61 6b69 6e67 2061  d here [Making a
-0000bec0: 2066 6c61 7420 6c69 7374 206f 7574 206f   flat list out o
-0000bed0: 6620 6c69 7374 7320 6f66 206c 6973 7473  f lists of lists
-0000bee0: 5d28 6874 7470 3a2f 2f73 7461 636b 6f76  ](http://stackov
-0000bef0: 6572 666c 6f77 2e63 6f6d 2f61 2f39 3532  erflow.com/a/952
-0000bf00: 3935 322f 3336 3339 3032 3329 2c20 5b46  952/3639023), [F
-0000bf10: 6c61 7474 656e 696e 6720 6120 7368 616c  lattening a shal
-0000bf20: 6c6f 7720 6c69 7374 5d28 6874 7470 3a2f  low list](http:/
-0000bf30: 2f73 7461 636b 6f76 6572 666c 6f77 2e63  /stackoverflow.c
-0000bf40: 6f6d 2f61 2f34 3036 3139 392f 3336 3339  om/a/406199/3639
-0000bf50: 3032 3329 2028 5b23 3833 335d 2868 7474  023) ([#833](htt
-0000bf60: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-0000bf70: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-0000bf80: 696f 2f69 7373 7565 732f 3833 3329 290a  io/issues/833)).
-0000bf90: 2a20 4465 7072 6563 6174 6564 2060 736b  * Deprecated `sk
-0000bfa0: 6269 6f2e 7374 6174 732e 706f 7765 722e  bio.stats.power.
-0000bfb0: 626f 6f74 7374 7261 705f 706f 7765 725f  bootstrap_power_
-0000bfc0: 6375 7276 6560 2077 696c 6c20 6265 2072  curve` will be r
-0000bfd0: 656d 6f76 6564 2069 6e20 7363 696b 6974  emoved in scikit
-0000bfe0: 2d62 696f 2030 2e34 2e31 2e20 4974 2069  -bio 0.4.1. It i
-0000bff0: 7320 6465 7072 6563 6174 6564 2069 6e20  s deprecated in 
-0000c000: 6661 766f 7220 6f66 2075 7369 6e67 2060  favor of using `
-0000c010: 6073 7562 7361 6d70 6c65 5f70 6f77 6572  `subsample_power
-0000c020: 6060 206f 7220 6060 7361 6d70 6c65 5f70  `` or ``sample_p
-0000c030: 6169 7265 645f 706f 7765 7260 6020 746f  aired_power`` to
-0000c040: 2063 616c 6375 6c61 7465 2061 2070 6f77   calculate a pow
-0000c050: 6572 206d 6174 7269 782c 2061 6e64 2074  er matrix, and t
-0000c060: 6865 6e20 7468 6520 7573 6520 6f66 2060  hen the use of `
-0000c070: 6063 6f6e 6669 6465 6e63 655f 626f 756e  `confidence_boun
-0000c080: 6473 6060 2074 6f20 6361 6c63 756c 6174  ds`` to calculat
-0000c090: 6520 7468 6520 6176 6572 6167 6520 616e  e the average an
-0000c0a0: 6420 636f 6e66 6964 656e 6365 2069 6e74  d confidence int
-0000c0b0: 6572 7661 6c73 2e0a 0a23 2323 2042 6163  ervals...### Bac
-0000c0c0: 6b77 6172 642d 696e 636f 6d70 6174 6962  kward-incompatib
-0000c0d0: 6c65 2063 6861 6e67 6573 0a2a 2052 656d  le changes.* Rem
-0000c0e0: 6f76 6564 2074 6865 2066 6f6c 6c6f 7769  oved the followi
-0000c0f0: 6e67 2064 6570 7265 6361 7465 6420 6675  ng deprecated fu
-0000c100: 6e63 7469 6f6e 616c 6974 793a 0a20 2020  nctionality:.   
-0000c110: 202d 2060 736b 6269 6f2e 7061 7273 6560   - `skbio.parse`
-0000c120: 2073 7562 7061 636b 6167 652c 2069 6e63   subpackage, inc
-0000c130: 6c75 6469 6e67 2060 5365 7175 656e 6365  luding `Sequence
-0000c140: 4974 6572 6174 6f72 602c 2060 4661 7374  Iterator`, `Fast
-0000c150: 6149 7465 7261 746f 7260 2c20 6046 6173  aIterator`, `Fas
-0000c160: 7471 4974 6572 6174 6f72 602c 2060 6c6f  tqIterator`, `lo
-0000c170: 6164 602c 2060 7061 7273 655f 6661 7374  ad`, `parse_fast
-0000c180: 6160 2c20 6070 6172 7365 5f66 6173 7471  a`, `parse_fastq
-0000c190: 602c 2060 7061 7273 655f 7175 616c 602c  `, `parse_qual`,
-0000c1a0: 2060 7772 6974 655f 636c 7573 7461 6c60   `write_clustal`
-0000c1b0: 2c20 6070 6172 7365 5f63 6c75 7374 616c  , `parse_clustal
-0000c1c0: 602c 2061 6e64 2060 4661 7374 7150 6172  `, and `FastqPar
-0000c1d0: 7365 4572 726f 7260 3b20 706c 6561 7365  seError`; please
-0000c1e0: 2075 7365 2060 736b 6269 6f2e 696f 6020   use `skbio.io` 
-0000c1f0: 696e 7374 6561 642e 0a20 2020 202d 2060  instead..    - `
-0000c200: 736b 6269 6f2e 666f 726d 6174 6020 7375  skbio.format` su
-0000c210: 6270 6163 6b61 6765 2c20 696e 636c 7564  bpackage, includ
-0000c220: 696e 6720 6066 6173 7461 5f66 726f 6d5f  ing `fasta_from_
-0000c230: 7365 7175 656e 6365 602c 2060 6661 7374  sequence`, `fast
-0000c240: 615f 6672 6f6d 5f61 6c69 676e 6d65 6e74  a_from_alignment
-0000c250: 602c 2061 6e64 2060 666f 726d 6174 5f66  `, and `format_f
-0000c260: 6173 7471 5f72 6563 6f72 6460 3b20 706c  astq_record`; pl
-0000c270: 6561 7365 2075 7365 2060 736b 6269 6f2e  ease use `skbio.
-0000c280: 696f 6020 696e 7374 6561 642e 0a20 2020  io` instead..   
-0000c290: 202d 2060 736b 6269 6f2e 616c 6967 6e6d   - `skbio.alignm
-0000c2a0: 656e 742e 5365 7175 656e 6365 436f 6c6c  ent.SequenceColl
-0000c2b0: 6563 7469 6f6e 2e69 6e74 5f6d 6170 603b  ection.int_map`;
-0000c2c0: 2070 6c65 6173 6520 7573 6520 6053 6571   please use `Seq
-0000c2d0: 7565 6e63 6543 6f6c 6c65 6374 696f 6e2e  uenceCollection.
-0000c2e0: 7570 6461 7465 5f69 6473 6020 696e 7374  update_ids` inst
-0000c2f0: 6561 642e 0a20 2020 202d 2060 736b 6269  ead..    - `skbi
-0000c300: 6f2e 616c 6967 6e6d 656e 742e 5365 7175  o.alignment.Sequ
-0000c310: 656e 6365 436f 6c6c 6563 7469 6f6e 6020  enceCollection` 
-0000c320: 6d65 7468 6f64 7320 6074 6f5f 6661 7374  methods `to_fast
-0000c330: 6160 2061 6e64 2060 746f 4661 7374 6160  a` and `toFasta`
-0000c340: 3b20 706c 6561 7365 2075 7365 2060 5365  ; please use `Se
-0000c350: 7175 656e 6365 436f 6c6c 6563 7469 6f6e  quenceCollection
-0000c360: 2e77 7269 7465 6020 696e 7374 6561 642e  .write` instead.
-0000c370: 0a20 2020 202d 2060 636f 6e73 7472 7563  .    - `construc
-0000c380: 746f 7260 2070 6172 616d 6574 6572 2069  tor` parameter i
-0000c390: 6e20 6073 6b62 696f 2e61 6c69 676e 6d65  n `skbio.alignme
-0000c3a0: 6e74 2e41 6c69 676e 6d65 6e74 2e6d 616a  nt.Alignment.maj
-0000c3b0: 6f72 6974 795f 636f 6e73 656e 7375 7360  ority_consensus`
-0000c3c0: 3b20 706c 6561 7365 2063 6f6e 7665 7274  ; please convert
-0000c3d0: 2072 6574 7572 6e65 6420 6269 6f6c 6f67   returned biolog
-0000c3e0: 6963 616c 2073 6571 7565 6e63 6520 6f62  ical sequence ob
-0000c3f0: 6a65 6374 206d 616e 7561 6c6c 7920 6173  ject manually as
-0000c400: 2064 6573 6972 6564 2028 652e 672e 2c20   desired (e.g., 
-0000c410: 6073 7472 2873 6571 2960 292e 0a20 2020  `str(seq)`)..   
-0000c420: 202d 2060 736b 6269 6f2e 616c 6967 6e6d   - `skbio.alignm
-0000c430: 656e 742e 416c 6967 6e6d 656e 742e 746f  ent.Alignment.to
-0000c440: 5f70 6879 6c69 7060 3b20 706c 6561 7365  _phylip`; please
-0000c450: 2075 7365 2060 416c 6967 6e6d 656e 742e   use `Alignment.
-0000c460: 7772 6974 6560 2069 6e73 7465 6164 2e0a  write` instead..
-0000c470: 2020 2020 2d20 6073 6b62 696f 2e73 6571      - `skbio.seq
-0000c480: 7565 6e63 652e 4269 6f6c 6f67 6963 616c  uence.Biological
-0000c490: 5365 7175 656e 6365 2e74 6f5f 6661 7374  Sequence.to_fast
-0000c4a0: 6160 3b20 706c 6561 7365 2075 7365 2060  a`; please use `
-0000c4b0: 4269 6f6c 6f67 6963 616c 5365 7175 656e  BiologicalSequen
-0000c4c0: 6365 2e77 7269 7465 6020 696e 7374 6561  ce.write` instea
-0000c4d0: 642e 0a20 2020 202d 2060 736b 6269 6f2e  d..    - `skbio.
-0000c4e0: 7472 6565 2e54 7265 654e 6f64 6560 206d  tree.TreeNode` m
-0000c4f0: 6574 686f 6473 2060 6672 6f6d 5f6e 6577  ethods `from_new
-0000c500: 6963 6b60 2c20 6066 726f 6d5f 6669 6c65  ick`, `from_file
-0000c510: 602c 2061 6e64 2060 746f 5f6e 6577 6963  `, and `to_newic
-0000c520: 6b60 3b20 706c 6561 7365 2075 7365 2060  k`; please use `
-0000c530: 5472 6565 4e6f 6465 2e72 6561 6460 2061  TreeNode.read` a
-0000c540: 6e64 2060 5472 6565 4e6f 6465 2e77 7269  nd `TreeNode.wri
-0000c550: 7465 6020 696e 7374 6561 642e 0a20 2020  te` instead..   
-0000c560: 202d 2060 736b 6269 6f2e 7374 6174 732e   - `skbio.stats.
-0000c570: 6469 7374 616e 6365 2e44 6973 7369 6d69  distance.Dissimi
-0000c580: 6c61 7269 7479 4d61 7472 6978 6020 6d65  larityMatrix` me
-0000c590: 7468 6f64 7320 6066 726f 6d5f 6669 6c65  thods `from_file
-0000c5a0: 6020 616e 6420 6074 6f5f 6669 6c65 603b  ` and `to_file`;
-0000c5b0: 2070 6c65 6173 6520 7573 6520 6044 6973   please use `Dis
-0000c5c0: 7369 6d69 6c61 7269 7479 4d61 7472 6978  similarityMatrix
-0000c5d0: 2e72 6561 6460 2061 6e64 2060 4469 7373  .read` and `Diss
-0000c5e0: 696d 696c 6172 6974 794d 6174 7269 782e  imilarityMatrix.
-0000c5f0: 7772 6974 6560 2069 6e73 7465 6164 2e0a  write` instead..
-0000c600: 2020 2020 2d20 6073 6b62 696f 2e73 7461      - `skbio.sta
-0000c610: 7473 2e6f 7264 696e 6174 696f 6e2e 4f72  ts.ordination.Or
-0000c620: 6469 6e61 7469 6f6e 5265 7375 6c74 7360  dinationResults`
-0000c630: 206d 6574 686f 6473 2060 6672 6f6d 5f66   methods `from_f
-0000c640: 696c 6560 2061 6e64 2060 746f 5f66 696c  ile` and `to_fil
-0000c650: 6560 3b20 706c 6561 7365 2075 7365 2060  e`; please use `
-0000c660: 4f72 6469 6e61 7469 6f6e 5265 7375 6c74  OrdinationResult
-0000c670: 732e 7265 6164 6020 616e 6420 604f 7264  s.read` and `Ord
-0000c680: 696e 6174 696f 6e52 6573 756c 7473 2e77  inationResults.w
-0000c690: 7269 7465 6020 696e 7374 6561 642e 0a20  rite` instead.. 
-0000c6a0: 2020 202d 2060 736b 6269 6f2e 7374 6174     - `skbio.stat
-0000c6b0: 732e 705f 7661 6c75 655f 746f 5f73 7472  s.p_value_to_str
-0000c6c0: 603b 2074 6865 7265 2069 7320 6e6f 2072  `; there is no r
-0000c6d0: 6570 6c61 6365 6d65 6e74 2e0a 2020 2020  eplacement..    
-0000c6e0: 2d20 6073 6b62 696f 2e73 7461 7473 2e73  - `skbio.stats.s
-0000c6f0: 7562 7361 6d70 6c65 603b 2070 6c65 6173  ubsample`; pleas
-0000c700: 6520 7573 6520 6073 6b62 696f 2e73 7461  e use `skbio.sta
-0000c710: 7473 2e73 7562 7361 6d70 6c65 5f63 6f75  ts.subsample_cou
-0000c720: 6e74 7360 2069 6e73 7465 6164 2e0a 2020  nts` instead..  
-0000c730: 2020 2d20 6073 6b62 696f 2e73 7461 7473    - `skbio.stats
-0000c740: 2e64 6973 7461 6e63 652e 414e 4f53 494d  .distance.ANOSIM
-0000c750: 603b 2070 6c65 6173 6520 7573 6520 6073  `; please use `s
-0000c760: 6b62 696f 2e73 7461 7473 2e64 6973 7461  kbio.stats.dista
-0000c770: 6e63 652e 616e 6f73 696d 6020 696e 7374  nce.anosim` inst
-0000c780: 6561 642e 0a20 2020 202d 2060 736b 6269  ead..    - `skbi
-0000c790: 6f2e 7374 6174 732e 6469 7374 616e 6365  o.stats.distance
-0000c7a0: 2e50 4552 4d41 4e4f 5641 603b 2070 6c65  .PERMANOVA`; ple
-0000c7b0: 6173 6520 7573 6520 6073 6b62 696f 2e73  ase use `skbio.s
-0000c7c0: 7461 7473 2e64 6973 7461 6e63 652e 7065  tats.distance.pe
-0000c7d0: 726d 616e 6f76 6160 2069 6e73 7465 6164  rmanova` instead
-0000c7e0: 2e0a 2020 2020 2d20 6073 6b62 696f 2e73  ..    - `skbio.s
-0000c7f0: 7461 7473 2e64 6973 7461 6e63 652e 4361  tats.distance.Ca
-0000c800: 7465 676f 7269 6361 6c53 7461 7473 5265  tegoricalStatsRe
-0000c810: 7375 6c74 7360 3b20 7468 6572 6520 6973  sults`; there is
-0000c820: 206e 6f20 7265 706c 6163 656d 656e 742c   no replacement,
-0000c830: 2070 6c65 6173 6520 7573 6520 6073 6b62   please use `skb
-0000c840: 696f 2e73 7461 7473 2e64 6973 7461 6e63  io.stats.distanc
-0000c850: 652e 616e 6f73 696d 6020 6f72 2060 736b  e.anosim` or `sk
-0000c860: 6269 6f2e 7374 6174 732e 6469 7374 616e  bio.stats.distan
-0000c870: 6365 2e70 6572 6d61 6e6f 7661 602c 2077  ce.permanova`, w
-0000c880: 6869 6368 2077 696c 6c20 7265 7475 726e  hich will return
-0000c890: 2061 2060 7061 6e64 6173 2e53 6572 6965   a `pandas.Serie
-0000c8a0: 7360 206f 626a 6563 742e 0a2a 2060 736b  s` object..* `sk
-0000c8b0: 6269 6f2e 616c 6967 6e6d 656e 742e 416c  bio.alignment.Al
-0000c8c0: 6967 6e6d 656e 742e 6d61 6a6f 7269 7479  ignment.majority
-0000c8d0: 5f63 6f6e 7365 6e73 7573 6020 6e6f 7720  _consensus` now 
-0000c8e0: 7265 7475 726e 7320 6042 696f 6c6f 6769  returns `Biologi
-0000c8f0: 6361 6c53 6571 7565 6e63 6528 2727 2960  calSequence('')`
-0000c900: 2069 6620 7468 6520 616c 6967 6e6d 656e   if the alignmen
-0000c910: 7420 6973 2065 6d70 7479 2e20 5072 6576  t is empty. Prev
-0000c920: 696f 7573 6c79 2c20 6027 2760 2077 6173  iously, `''` was
-0000c930: 2072 6574 7572 6e65 642e 0a2a 2060 6d69   returned..* `mi
-0000c940: 6e5f 6f62 7365 7276 6174 696f 6e73 6020  n_observations` 
-0000c950: 7761 7320 7265 6d6f 7665 6420 6672 6f6d  was removed from
-0000c960: 2060 736b 6269 6f2e 7374 6174 732e 706f   `skbio.stats.po
-0000c970: 7765 722e 7375 6273 616d 706c 655f 706f  wer.subsample_po
-0000c980: 7765 7260 2061 6e64 2060 736b 6269 6f2e  wer` and `skbio.
-0000c990: 7374 6174 732e 706f 7765 722e 7375 6273  stats.power.subs
-0000c9a0: 616d 706c 655f 7061 6972 6564 5f70 6f77  ample_paired_pow
-0000c9b0: 6572 602e 2054 6865 206d 696e 696d 756d  er`. The minimum
-0000c9c0: 206e 756d 6265 7220 6f66 2073 616d 706c   number of sampl
-0000c9d0: 6573 2066 6f72 2073 7562 7361 6d70 6c69  es for subsampli
-0000c9e0: 6e67 2064 6570 656e 6473 206f 6e20 7468  ng depends on th
-0000c9f0: 6520 6461 7461 2073 6574 2061 6e64 2073  e data set and s
-0000ca00: 7461 7469 7374 6963 616c 2074 6573 7473  tatistical tests
-0000ca10: 2e20 4861 7669 6e67 2061 2064 6566 6175  . Having a defau
-0000ca20: 6c74 2070 6172 616d 6574 6572 2074 6f20  lt parameter to 
-0000ca30: 7365 7420 756e 6e65 6365 7373 6172 7920  set unnecessary 
-0000ca40: 6c69 6d69 7461 7469 6f6e 7320 6f6e 2074  limitations on t
-0000ca50: 6865 2074 6563 686e 6971 7565 2e0a 0a23  he technique...#
-0000ca60: 2323 204d 6973 6365 6c6c 616e 656f 7573  ## Miscellaneous
-0000ca70: 0a2a 2043 6861 6e67 6564 2074 6573 7469  .* Changed testi
-0000ca80: 6e67 2070 726f 6365 6475 7265 730a 2020  ng procedures.  
-0000ca90: 2020 2d20 4465 7665 6c6f 7065 7273 2073    - Developers s
-0000caa0: 686f 756c 6420 6e6f 7720 7573 6520 606d  hould now use `m
-0000cab0: 616b 6520 7465 7374 600a 2020 2020 2d20  ake test`.    - 
-0000cac0: 5573 6572 7320 6361 6e20 7573 6520 6070  Users can use `p
-0000cad0: 7974 686f 6e20 2d6d 2073 6b62 696f 2e74  ython -m skbio.t
-0000cae0: 6573 7460 0a20 2020 202d 2041 6464 6564  est`.    - Added
-0000caf0: 2060 736b 6269 6f2e 7574 696c 2e5f 7465   `skbio.util._te
-0000cb00: 7374 696e 672e 5465 7374 5275 6e6e 6572  sting.TestRunner
-0000cb10: 6020 2861 7661 696c 6162 6c65 2074 6872  ` (available thr
-0000cb20: 6f75 6768 2060 736b 6269 6f2e 7574 696c  ough `skbio.util
-0000cb30: 2e54 6573 7452 756e 6e65 7260 292e 2055  .TestRunner`). U
-0000cb40: 7365 6420 746f 2070 726f 7669 6465 2061  sed to provide a
-0000cb50: 2060 7465 7374 6020 6d65 7468 6f64 2066   `test` method f
-0000cb60: 6f72 2065 6163 6820 6d6f 6475 6c65 2069  or each module i
-0000cb70: 6e69 7420 6669 6c65 2e20 5468 6973 2063  nit file. This c
-0000cb80: 6c61 7373 2072 6570 7265 7365 6e74 7320  lass represents 
-0000cb90: 6120 756e 6966 6965 6420 7465 7374 696e  a unified testin
-0000cba0: 6720 7061 7468 2077 6869 6368 2077 7261  g path which wra
-0000cbb0: 7073 2061 6c6c 2060 736b 6269 6f60 2074  ps all `skbio` t
-0000cbc0: 6573 7469 6e67 2066 756e 6374 696f 6e61  esting functiona
-0000cbd0: 6c69 7479 2e0a 2020 2020 2d20 4175 746f  lity..    - Auto
-0000cbe0: 6465 7465 6374 2050 7974 686f 6e20 7665  detect Python ve
-0000cbf0: 7273 696f 6e20 616e 6420 6469 7361 626c  rsion and disabl
-0000cc00: 6520 646f 6374 6573 7473 2066 6f72 2050  e doctests for P
-0000cc10: 7974 686f 6e20 332e 0a2a 2060 6e75 6d70  ython 3..* `nump
-0000cc20: 7960 2069 7320 6e6f 206c 6f6e 6765 7220  y` is no longer 
-0000cc30: 7265 7175 6972 6564 2074 6f20 6265 2069  required to be i
-0000cc40: 6e73 7461 6c6c 6564 2062 6566 6f72 6520  nstalled before 
-0000cc50: 696e 7374 616c 6c69 6e67 2073 6369 6b69  installing sciki
-0000cc60: 742d 6269 6f21 0a2a 2055 7067 7261 6465  t-bio!.* Upgrade
-0000cc70: 6420 6368 6563 6b6c 6973 742e 7079 2074  d checklist.py t
-0000cc80: 6f20 6368 6563 6b20 736f 7572 6365 2066  o check source f
-0000cc90: 696c 6573 206e 6f6e 2d63 6f6e 666f 726d  iles non-conform
-0000cca0: 696e 6720 746f 205b 6e65 7720 6865 6164  ing to [new head
-0000ccb0: 6572 2073 7479 6c65 5d28 6874 7470 3a2f  er style](http:/
-0000ccc0: 2f73 6369 6b69 742d 6269 6f2e 6f72 672f  /scikit-bio.org/
-0000ccd0: 646f 6373 2f6c 6174 6573 742f 6465 7665  docs/latest/deve
-0000cce0: 6c6f 706d 656e 742f 6e65 775f 6d6f 6475  lopment/new_modu
-0000ccf0: 6c65 2e68 746d 6c29 2e20 285b 2338 3535  le.html). ([#855
-0000cd00: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
-0000cd10: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
-0000cd20: 6b69 742d 6269 6f2f 6973 7375 6573 2f38  kit-bio/issues/8
-0000cd30: 3535 2929 0a2a 2055 7064 6174 6564 2074  55)).* Updated t
-0000cd40: 6f20 7573 6520 606e 6174 736f 7274 6020  o use `natsort` 
-0000cd50: 3e3d 2034 2e30 2e30 2e0a 2a20 5468 6520  >= 4.0.0..* The 
-0000cd60: 6d65 7468 6f64 206f 6620 7375 6273 616d  method of subsam
-0000cd70: 706c 696e 6720 7761 7320 6368 616e 6765  pling was change
-0000cd80: 6420 666f 7220 6060 736b 6269 6f2e 7374  d for ``skbio.st
-0000cd90: 6174 732e 706f 7765 722e 7375 6273 616d  ats.power.subsam
-0000cda0: 706c 655f 7061 6972 6564 5f70 6f77 6572  ple_paired_power
-0000cdb0: 6060 2e20 5261 7468 6572 2074 6861 6e20  ``. Rather than 
-0000cdc0: 6472 6177 696e 6720 6120 7061 6972 6564  drawing a paired
-0000cdd0: 2073 616d 706c 6520 666f 7220 7468 6520   sample for the 
-0000cde0: 7275 6e20 616e 6420 7468 656e 2073 7562  run and then sub
-0000cdf0: 7361 6d70 6c69 6e67 2066 6f72 2065 6163  sampling for eac
-0000ce00: 6820 636f 756e 742c 2074 6865 2073 7562  h count, the sub
-0000ce10: 7361 6d70 6c65 2069 7320 6e6f 7720 6472  sample is now dr
-0000ce20: 6177 6e20 666f 7220 6561 6368 2073 616d  awn for each sam
-0000ce30: 706c 6520 616e 6420 6561 6368 2072 756e  ple and each run
-0000ce40: 2e20 496e 2074 6573 7420 6461 7461 2c20  . In test data, 
-0000ce50: 7468 6973 2064 6964 206e 6f74 2073 6967  this did not sig
-0000ce60: 6e69 6669 6361 6e74 6c79 2061 6c74 6572  nificantly alter
-0000ce70: 2074 6865 2070 6f77 6572 2072 6573 756c   the power resul
-0000ce80: 7473 2e0a 2a20 6368 6563 6b6c 6973 742e  ts..* checklist.
-0000ce90: 7079 206e 6f77 2065 6e66 6f72 6365 7320  py now enforces 
-0000cea0: 605f 5f66 7574 7572 655f 5f60 2069 6d70  `__future__` imp
-0000ceb0: 6f72 7473 2069 6e20 2e70 7920 6669 6c65  orts in .py file
-0000cec0: 732e 0a0a 2323 2056 6572 7369 6f6e 2030  s...## Version 0
-0000ced0: 2e32 2e33 2028 3230 3135 2d30 322d 3133  .2.3 (2015-02-13
-0000cee0: 290a 0a23 2323 2046 6561 7475 7265 730a  )..### Features.
-0000cef0: 2a20 4d6f 6469 6669 6564 2060 6073 6b62  * Modified ``skb
-0000cf00: 696f 2e73 7461 7473 2e64 6973 7461 6e63  io.stats.distanc
-0000cf10: 652e 7077 6d61 6e74 656c 6060 2074 6f20  e.pwmantel`` to 
-0000cf20: 6163 6365 7074 2061 206c 6973 7420 6f66  accept a list of
-0000cf30: 2066 696c 6570 6174 6873 2e20 5468 6973   filepaths. This
-0000cf40: 2069 7320 7573 6566 756c 2061 7320 6974   is useful as it
-0000cf50: 2061 6c6c 6f77 7320 666f 7220 6120 736d   allows for a sm
-0000cf60: 616c 6c65 7220 616d 6f75 6e74 206f 6620  aller amount of 
-0000cf70: 6d65 6d6f 7279 2063 6f6e 7375 6d70 7469  memory consumpti
-0000cf80: 6f6e 2061 7320 6974 206f 6e6c 7920 6c6f  on as it only lo
-0000cf90: 6164 7320 7477 6f20 6d61 7472 6963 6573  ads two matrices
-0000cfa0: 2061 7420 6120 7469 6d65 2061 7320 6f70   at a time as op
-0000cfb0: 706f 7365 6420 746f 2072 6571 7569 7269  posed to requiri
-0000cfc0: 6e67 2074 6861 7420 616c 6c20 6469 7374  ng that all dist
-0000cfd0: 616e 6365 206d 6174 7269 6365 7320 6172  ance matrices ar
-0000cfe0: 6520 6c6f 6164 6564 2069 6e74 6f20 6d65  e loaded into me
-0000cff0: 6d6f 7279 2e0a 2a20 4164 6465 6420 6060  mory..* Added ``
-0000d000: 736b 6269 6f2e 7574 696c 2e66 696e 645f  skbio.util.find_
-0000d010: 6475 706c 6963 6174 6573 6060 2066 6f72  duplicates`` for
-0000d020: 2066 696e 6469 6e67 2064 7570 6c69 6361   finding duplica
-0000d030: 7465 2065 6c65 6d65 6e74 7320 696e 2061  te elements in a
-0000d040: 6e20 6974 6572 6162 6c65 2e0a 0a23 2323  n iterable...###
-0000d050: 2042 7567 2066 6978 6573 0a2a 2046 6978   Bug fixes.* Fix
-0000d060: 6564 2066 6c6f 6174 696e 6720 706f 696e  ed floating poin
-0000d070: 7420 7072 6563 6973 696f 6e20 6275 6773  t precision bugs
-0000d080: 2069 6e20 6060 416c 6967 6e6d 656e 742e   in ``Alignment.
-0000d090: 706f 7369 7469 6f6e 5f66 7265 7175 656e  position_frequen
-0000d0a0: 6369 6573 6060 2c20 6060 416c 6967 6e6d  cies``, ``Alignm
-0000d0b0: 656e 742e 706f 7369 7469 6f6e 5f65 6e74  ent.position_ent
-0000d0c0: 726f 7069 6573 6060 2c20 6060 416c 6967  ropies``, ``Alig
-0000d0d0: 6e6d 656e 742e 6f6d 6974 5f67 6170 5f70  nment.omit_gap_p
-0000d0e0: 6f73 6974 696f 6e73 6060 2c20 6060 416c  ositions``, ``Al
-0000d0f0: 6967 6e6d 656e 742e 6f6d 6974 5f67 6170  ignment.omit_gap
-0000d100: 5f73 6571 7565 6e63 6573 6060 2c20 6060  _sequences``, ``
-0000d110: 4269 6f6c 6f67 6963 616c 5365 7175 656e  BiologicalSequen
-0000d120: 6365 2e6b 5f77 6f72 645f 6672 6571 7565  ce.k_word_freque
-0000d130: 6e63 6965 7360 602c 2061 6e64 2060 6053  ncies``, and ``S
-0000d140: 6571 7565 6e63 6543 6f6c 6c65 6374 696f  equenceCollectio
-0000d150: 6e2e 6b5f 776f 7264 5f66 7265 7175 656e  n.k_word_frequen
-0000d160: 6369 6573 6060 2028 5b23 3830 315d 2868  cies`` ([#801](h
-0000d170: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-0000d180: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
-0000d190: 2d62 696f 2f69 7373 7565 732f 3830 3129  -bio/issues/801)
-0000d1a0: 292e 0a0a 2323 2320 4261 636b 7761 7264  )...### Backward
-0000d1b0: 2d69 6e63 6f6d 7061 7469 626c 6520 6368  -incompatible ch
-0000d1c0: 616e 6765 730a 2a20 5265 6d6f 7665 6420  anges.* Removed 
-0000d1d0: 6060 6665 6174 7572 655f 7479 7065 7360  ``feature_types`
-0000d1e0: 6020 6174 7472 6962 7574 6520 6672 6f6d  ` attribute from
-0000d1f0: 2060 6042 696f 6c6f 6769 6361 6c53 6571   ``BiologicalSeq
-0000d200: 7565 6e63 6560 6020 616e 6420 616c 6c20  uence`` and all 
-0000d210: 7375 6263 6c61 7373 6573 2028 5b23 3739  subclasses ([#79
-0000d220: 375d 2868 7474 7073 3a2f 2f67 6974 6875  7](https://githu
-0000d230: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
-0000d240: 696b 6974 2d62 696f 2f70 756c 6c2f 3739  ikit-bio/pull/79
-0000d250: 3729 292e 0a2a 2052 656d 6f76 6564 2060  7))..* Removed `
-0000d260: 6066 696e 645f 6665 6174 7572 6573 6060  `find_features``
-0000d270: 206d 6574 686f 6420 6672 6f6d 2060 6042   method from ``B
-0000d280: 696f 6c6f 6769 6361 6c53 6571 7565 6e63  iologicalSequenc
-0000d290: 6560 6020 616e 6420 6060 5072 6f74 6569  e`` and ``Protei
-0000d2a0: 6e53 6571 7565 6e63 6560 6020 285b 2337  nSequence`` ([#7
-0000d2b0: 3937 5d28 6874 7470 733a 2f2f 6769 7468  97](https://gith
-0000d2c0: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
-0000d2d0: 6369 6b69 742d 6269 6f2f 7075 6c6c 2f37  cikit-bio/pull/7
-0000d2e0: 3937 2929 2e0a 2a20 6060 4269 6f6c 6f67  97))..* ``Biolog
-0000d2f0: 6963 616c 5365 7175 656e 6365 2e6b 5f77  icalSequence.k_w
-0000d300: 6f72 645f 6672 6571 7565 6e63 6965 7360  ord_frequencies`
-0000d310: 6020 6e6f 7720 7265 7475 726e 7320 6120  ` now returns a 
-0000d320: 6060 636f 6c6c 6563 7469 6f6e 732e 6465  ``collections.de
-0000d330: 6661 756c 7464 6963 7460 6020 6f66 2074  faultdict`` of t
-0000d340: 7970 6520 6060 666c 6f61 7460 6020 696e  ype ``float`` in
-0000d350: 7374 6561 6420 6f66 2074 7970 6520 6060  stead of type ``
-0000d360: 696e 7460 602e 2054 6869 7320 6f6e 6c79  int``. This only
-0000d370: 2061 6666 6563 7473 2074 6865 2022 6465   affects the "de
-0000d380: 6661 756c 7422 2063 6173 652c 2077 6865  fault" case, whe
-0000d390: 6e20 6120 6b65 7920 6973 6e27 7420 7072  n a key isn't pr
-0000d3a0: 6573 656e 7420 696e 2074 6865 2064 6963  esent in the dic
-0000d3b0: 7469 6f6e 6172 792e 2050 7265 7669 6f75  tionary. Previou
-0000d3c0: 7320 6265 6861 7669 6f72 2077 6f75 6c64  s behavior would
-0000d3d0: 2072 6574 7572 6e20 6060 3060 6020 6173   return ``0`` as
-0000d3e0: 2061 6e20 6060 696e 7460 602c 2077 6869   an ``int``, whi
-0000d3f0: 6c65 2074 6865 206e 6577 2062 6568 6176  le the new behav
-0000d400: 696f 7220 6973 2074 6f20 7265 7475 726e  ior is to return
-0000d410: 2060 6030 2e30 6060 2061 7320 6120 6060   ``0.0`` as a ``
-0000d420: 666c 6f61 7460 602e 2054 6869 7320 6368  float``. This ch
-0000d430: 616e 6765 2061 6c73 6f20 6166 6665 6374  ange also affect
-0000d440: 7320 7468 6520 6060 6465 6661 756c 7464  s the ``defaultd
-0000d450: 6963 7460 6073 2074 6861 7420 6172 6520  ict``s that are 
-0000d460: 7265 7475 726e 6564 2062 7920 6060 5365  returned by ``Se
-0000d470: 7175 656e 6365 436f 6c6c 6563 7469 6f6e  quenceCollection
-0000d480: 2e6b 5f77 6f72 645f 6672 6571 7565 6e63  .k_word_frequenc
-0000d490: 6965 7360 602e 0a0a 2323 2320 4d69 7363  ies``...### Misc
-0000d4a0: 656c 6c61 6e65 6f75 730a 2a20 6060 4469  ellaneous.* ``Di
-0000d4b0: 7373 696d 696c 6172 6974 794d 6174 7269  ssimilarityMatri
-0000d4c0: 7860 6020 616e 6420 6060 4469 7374 616e  x`` and ``Distan
-0000d4d0: 6365 4d61 7472 6978 6060 206e 6f77 2072  ceMatrix`` now r
-0000d4e0: 6570 6f72 7420 6475 706c 6963 6174 6520  eport duplicate 
-0000d4f0: 4944 7320 696e 2074 6865 2060 6044 6973  IDs in the ``Dis
-0000d500: 7369 6d69 6c61 7269 7479 4d61 7472 6978  similarityMatrix
-0000d510: 4572 726f 7260 6020 6d65 7373 6167 6520  Error`` message 
-0000d520: 7468 6174 2063 616e 2062 6520 7261 6973  that can be rais
-0000d530: 6564 2064 7572 696e 6720 7661 6c69 6461  ed during valida
-0000d540: 7469 6f6e 2e0a 0a23 2320 5665 7273 696f  tion...## Versio
-0000d550: 6e20 302e 322e 3220 2832 3031 342d 3132  n 0.2.2 (2014-12
-0000d560: 2d30 3429 0a0a 2323 2320 4665 6174 7572  -04)..### Featur
-0000d570: 6573 0a2a 2041 6464 6564 2060 6070 6c6f  es.* Added ``plo
-0000d580: 7460 6020 6d65 7468 6f64 2074 6f20 6060  t`` method to ``
-0000d590: 736b 6269 6f2e 7374 6174 732e 6469 7374  skbio.stats.dist
-0000d5a0: 616e 6365 2e44 6973 7369 6d69 6c61 7269  ance.Dissimilari
-0000d5b0: 7479 4d61 7472 6978 6060 2066 6f72 2063  tyMatrix`` for c
-0000d5c0: 7265 6174 696e 6720 6261 7369 6320 6865  reating basic he
-0000d5d0: 6174 6d61 7073 206f 6620 6120 6469 7373  atmaps of a diss
-0000d5e0: 696d 696c 6172 6974 792f 6469 7374 616e  imilarity/distan
-0000d5f0: 6365 206d 6174 7269 7820 2873 6565 205b  ce matrix (see [
-0000d600: 2336 3834 5d28 6874 7470 733a 2f2f 6769  #684](https://gi
-0000d610: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
-0000d620: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
-0000d630: 6573 2f36 3834 2929 2e20 416c 736f 2061  es/684)). Also a
-0000d640: 6464 6564 2020 6060 5f72 6570 725f 706e  dded  ``_repr_pn
-0000d650: 675f 6060 2061 6e64 2060 605f 7265 7072  g_`` and ``_repr
-0000d660: 5f73 7667 5f60 6020 6d65 7468 6f64 7320  _svg_`` methods 
-0000d670: 666f 7220 6175 746f 6d61 7469 6320 6469  for automatic di
-0000d680: 7370 6c61 7920 696e 2074 6865 2049 5079  splay in the IPy
-0000d690: 7468 6f6e 204e 6f74 6562 6f6f 6b2c 2077  thon Notebook, w
-0000d6a0: 6974 6820 6060 706e 6760 6020 616e 6420  ith ``png`` and 
-0000d6b0: 6060 7376 6760 6020 7072 6f70 6572 7469  ``svg`` properti
-0000d6c0: 6573 2066 6f72 2064 6972 6563 7420 6163  es for direct ac
-0000d6d0: 6365 7373 2e0a 2a20 4164 6465 6420 605f  cess..* Added `_
-0000d6e0: 5f73 7472 5f5f 6020 6d65 7468 6f64 2074  _str__` method t
-0000d6f0: 6f20 6073 6b62 696f 2e73 7461 7473 2e6f  o `skbio.stats.o
-0000d700: 7264 696e 6174 696f 6e2e 4f72 6469 6e61  rdination.Ordina
-0000d710: 7469 6f6e 5265 7375 6c74 7360 2e0a 2a20  tionResults`..* 
-0000d720: 4164 6465 6420 6060 736b 6269 6f2e 7374  Added ``skbio.st
-0000d730: 6174 732e 6469 7374 616e 6365 2e61 6e6f  ats.distance.ano
-0000d740: 7369 6d60 6020 616e 6420 6060 736b 6269  sim`` and ``skbi
-0000d750: 6f2e 7374 6174 732e 6469 7374 616e 6365  o.stats.distance
-0000d760: 2e70 6572 6d61 6e6f 7661 6060 2066 756e  .permanova`` fun
-0000d770: 6374 696f 6e73 2c20 7768 6963 6820 7265  ctions, which re
-0000d780: 706c 6163 6520 7468 6520 6060 736b 6269  place the ``skbi
-0000d790: 6f2e 7374 6174 732e 6469 7374 616e 6365  o.stats.distance
-0000d7a0: 2e41 4e4f 5349 4d60 6020 616e 6420 6060  .ANOSIM`` and ``
-0000d7b0: 736b 6269 6f2e 7374 6174 732e 6469 7374  skbio.stats.dist
-0000d7c0: 616e 6365 2e50 4552 4d41 4e4f 5641 6060  ance.PERMANOVA``
-0000d7d0: 2063 6c61 7373 6573 2e20 5468 6573 6520   classes. These 
-0000d7e0: 6e65 7720 6675 6e63 7469 6f6e 7320 7072  new functions pr
-0000d7f0: 6f76 6964 6520 7369 6d70 6c65 7220 7072  ovide simpler pr
-0000d800: 6f63 6564 7572 616c 2069 6e74 6572 6661  ocedural interfa
-0000d810: 6365 7320 746f 2072 756e 6e69 6e67 2074  ces to running t
-0000d820: 6865 7365 2073 7461 7469 7374 6963 616c  hese statistical
-0000d830: 206d 6574 686f 6473 2e20 5468 6579 2061   methods. They a
-0000d840: 6c73 6f20 7072 6f76 6964 6520 6d6f 7265  lso provide more
-0000d850: 2063 6f6e 7665 6e69 656e 7420 6163 6365   convenient acce
-0000d860: 7373 2074 6f20 7265 7375 6c74 7320 6279  ss to results by
-0000d870: 2072 6574 7572 6e69 6e67 2061 2060 6070   returning a ``p
-0000d880: 616e 6461 732e 5365 7269 6573 6060 2069  andas.Series`` i
-0000d890: 6e73 7465 6164 206f 6620 6120 6060 4361  nstead of a ``Ca
-0000d8a0: 7465 676f 7269 6361 6c53 7461 7473 5265  tegoricalStatsRe
-0000d8b0: 7375 6c74 7360 6020 6f62 6a65 6374 2e20  sults`` object. 
-0000d8c0: 5468 6573 6520 6675 6e63 7469 6f6e 7320  These functions 
-0000d8d0: 6861 7665 206d 6f72 6520 6578 7465 6e73  have more extens
-0000d8e0: 6976 6520 646f 6375 6d65 6e74 6174 696f  ive documentatio
-0000d8f0: 6e20 7468 616e 2074 6865 6972 2070 7265  n than their pre
-0000d900: 7669 6f75 7320 7665 7273 696f 6e73 2e20  vious versions. 
-0000d910: 4966 2073 6967 6e69 6669 6361 6e63 6520  If significance 
-0000d920: 7465 7374 7320 6172 6520 7375 7070 7265  tests are suppre
-0000d930: 7373 6564 2c20 702d 7661 6c75 6573 2061  ssed, p-values a
-0000d940: 7265 2072 6574 7572 6e65 6420 6173 2060  re returned as `
-0000d950: 606e 702e 6e61 6e60 6020 696e 7374 6561  `np.nan`` instea
-0000d960: 6420 6f66 2060 604e 6f6e 6560 6020 666f  d of ``None`` fo
-0000d970: 7220 636f 6e73 6973 7465 6e63 7920 7769  r consistency wi
-0000d980: 7468 206f 7468 6572 2073 7461 7469 7374  th other statist
-0000d990: 6963 616c 206d 6574 686f 6473 2069 6e20  ical methods in 
-0000d9a0: 7363 696b 6974 2d62 696f 2e20 5b23 3735  scikit-bio. [#75
-0000d9b0: 345d 2868 7474 7073 3a2f 2f67 6974 6875  4](https://githu
-0000d9c0: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
-0000d9d0: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
-0000d9e0: 3735 3429 0a2a 2041 6464 6564 2060 736b  754).* Added `sk
-0000d9f0: 6269 6f2e 7374 6174 732e 706f 7765 7260  bio.stats.power`
-0000da00: 2066 6f72 2070 6572 666f 726d 696e 6720   for performing 
-0000da10: 656d 7069 7269 6361 6c20 706f 7765 7220  empirical power 
-0000da20: 616e 616c 7973 6973 2e20 5468 6520 6d6f  analysis. The mo
-0000da30: 6475 6c65 2075 7365 7320 6578 6973 7469  dule uses existi
-0000da40: 6e67 2064 6174 6173 6574 7320 616e 6420  ng datasets and 
-0000da50: 6974 6572 6174 6976 656c 7920 6472 6177  iteratively draw
-0000da60: 7320 7361 6d70 6c65 7320 746f 2065 7374  s samples to est
-0000da70: 696d 6174 6520 7468 6520 6e75 6d62 6572  imate the number
-0000da80: 206f 6620 7361 6d70 6c65 7320 6e65 6564   of samples need
-0000da90: 6564 2074 6f20 7365 6520 6120 7369 676e  ed to see a sign
-0000daa0: 6966 6963 616e 7420 6469 6666 6572 656e  ificant differen
-0000dab0: 6365 2066 6f72 2061 2067 6976 656e 2063  ce for a given c
-0000dac0: 7269 7469 6361 6c20 7661 6c75 652e 0a2a  ritical value..*
-0000dad0: 2041 6464 6564 2060 736b 6269 6f2e 7374   Added `skbio.st
-0000dae0: 6174 732e 6973 7562 7361 6d70 6c65 6020  ats.isubsample` 
-0000daf0: 666f 7220 7375 6273 616d 706c 696e 6720  for subsampling 
-0000db00: 6672 6f6d 2061 6e20 756e 6b6e 6f77 6e20  from an unknown 
-0000db10: 6e75 6d62 6572 206f 6620 7661 6c75 6573  number of values
-0000db20: 2e20 5468 6973 206d 6574 686f 6420 7375  . This method su
-0000db30: 7070 6f72 7473 2073 7562 7361 6d70 6c69  pports subsampli
-0000db40: 6e67 2066 726f 6d20 6d75 6c74 6970 6c65  ng from multiple
-0000db50: 2070 6172 7469 7469 6f6e 7320 616e 6420   partitions and 
-0000db60: 646f 6573 206e 6f74 2072 6571 7569 7265  does not require
-0000db70: 2074 6861 7420 616c 6c20 6974 656d 7320   that all items 
-0000db80: 6265 2073 746f 7265 6420 696e 206d 656d  be stored in mem
-0000db90: 6f72 792c 2072 6571 7569 7269 6e67 2061  ory, requiring a
-0000dba0: 7070 726f 7869 6d61 7465 6c79 2060 4f28  pproximately `O(
-0000dbb0: 4e2a 4d29 6060 2073 7061 6365 2077 6865  N*M)`` space whe
-0000dbc0: 7265 2060 4e60 2069 7320 7468 6520 6e75  re `N` is the nu
-0000dbd0: 6d62 6572 206f 6620 7061 7274 6974 696f  mber of partitio
-0000dbe0: 6e73 2061 6e64 2060 4d60 2069 7320 7468  ns and `M` is th
-0000dbf0: 6520 6d61 7869 6d75 6d20 7375 6273 616d  e maximum subsam
-0000dc00: 706c 6520 7369 7a65 2e0a 2a20 4164 6465  ple size..* Adde
-0000dc10: 6420 6060 736b 6269 6f2e 7374 6174 732e  d ``skbio.stats.
-0000dc20: 7375 6273 616d 706c 655f 636f 756e 7473  subsample_counts
-0000dc30: 6060 2c20 7768 6963 6820 7265 706c 6163  ``, which replac
-0000dc40: 6573 2060 6073 6b62 696f 2e73 7461 7473  es ``skbio.stats
-0000dc50: 2e73 7562 7361 6d70 6c65 6060 2e20 5365  .subsample``. Se
-0000dc60: 6520 6465 7072 6563 6174 696f 6e20 7365  e deprecation se
-0000dc70: 6374 696f 6e20 6265 6c6f 7720 666f 7220  ction below for 
-0000dc80: 6d6f 7265 2064 6574 6169 6c73 2028 5b23  more details ([#
-0000dc90: 3737 305d 2868 7474 7073 3a2f 2f67 6974  770](https://git
-0000dca0: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
-0000dcb0: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
-0000dcc0: 732f 3737 3029 292e 0a0a 2323 2320 4275  s/770))...### Bu
-0000dcd0: 6720 6669 7865 730a 2a20 4669 7865 6420  g fixes.* Fixed 
-0000dce0: 6973 7375 6520 7768 6572 6520 5353 5720  issue where SSW 
-0000dcf0: 776f 756c 646e 2774 2063 6f6d 7069 6c65  wouldn't compile
-0000dd00: 206f 6e20 6936 3836 2061 7263 6869 7465   on i686 archite
-0000dd10: 6374 7572 6573 2028 5b23 3430 395d 2868  ctures ([#409](h
-0000dd20: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-0000dd30: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
-0000dd40: 2d62 696f 2f69 7373 7565 732f 3430 3929  -bio/issues/409)
-0000dd50: 292e 0a0a 2323 2320 4465 7072 6563 6174  )...### Deprecat
-0000dd60: 6564 2066 756e 6374 696f 6e61 6c69 7479  ed functionality
-0000dd70: 0a2a 2044 6570 7265 6361 7465 6420 6060  .* Deprecated ``
-0000dd80: 736b 6269 6f2e 7374 6174 732e 705f 7661  skbio.stats.p_va
-0000dd90: 6c75 655f 746f 5f73 7472 6060 2e20 5468  lue_to_str``. Th
-0000dda0: 6973 2066 756e 6374 696f 6e20 7769 6c6c  is function will
-0000ddb0: 2062 6520 7265 6d6f 7665 6420 696e 2073   be removed in s
-0000ddc0: 6369 6b69 742d 6269 6f20 302e 332e 302e  cikit-bio 0.3.0.
-0000ddd0: 2050 6572 6d75 7461 7469 6f6e 2d62 6173   Permutation-bas
-0000dde0: 6564 2070 2d76 616c 7565 7320 696e 2073  ed p-values in s
-0000ddf0: 6369 6b69 742d 6269 6f20 6172 6520 6361  cikit-bio are ca
-0000de00: 6c63 756c 6174 6564 2061 7320 6060 286e  lculated as ``(n
-0000de10: 756d 5f65 7874 7265 6d65 202b 2031 2920  um_extreme + 1) 
-0000de20: 2f20 286e 756d 5f70 6572 6d75 7461 7469  / (num_permutati
-0000de30: 6f6e 7320 2b20 3129 6060 2c20 736f 2069  ons + 1)``, so i
-0000de40: 7420 6973 2069 6d70 6f73 7369 626c 6520  t is impossible 
-0000de50: 746f 206f 6274 6169 6e20 6120 702d 7661  to obtain a p-va
-0000de60: 6c75 6520 6f66 207a 6572 6f2e 2054 6869  lue of zero. Thi
-0000de70: 7320 6675 6e63 7469 6f6e 2068 6973 746f  s function histo
-0000de80: 7269 6361 6c6c 7920 6578 6973 7465 6420  rically existed 
-0000de90: 666f 7220 636f 7272 6563 7469 6e67 2074  for correcting t
-0000dea0: 6865 206e 756d 6265 7220 6f66 2064 6967  he number of dig
-0000deb0: 6974 7320 6469 7370 6c61 7965 6420 7768  its displayed wh
-0000dec0: 656e 206f 6274 6169 6e69 6e67 2061 2070  en obtaining a p
-0000ded0: 2d76 616c 7565 206f 6620 7a65 726f 2e20  -value of zero. 
-0000dee0: 5369 6e63 6520 7468 6973 2069 7320 6e6f  Since this is no
-0000def0: 206c 6f6e 6765 7220 706f 7373 6962 6c65   longer possible
-0000df00: 2c20 7468 6973 2066 756e 6374 696f 6e61  , this functiona
-0000df10: 6c69 7479 2077 696c 6c20 6265 2072 656d  lity will be rem
-0000df20: 6f76 6564 2e0a 2a20 4465 7072 6563 6174  oved..* Deprecat
-0000df30: 6564 2060 6073 6b62 696f 2e73 7461 7473  ed ``skbio.stats
-0000df40: 2e64 6973 7461 6e63 652e 414e 4f53 494d  .distance.ANOSIM
-0000df50: 6060 2061 6e64 2060 6073 6b62 696f 2e73  `` and ``skbio.s
-0000df60: 7461 7473 2e64 6973 7461 6e63 652e 5045  tats.distance.PE
-0000df70: 524d 414e 4f56 4160 6020 696e 2066 6176  RMANOVA`` in fav
-0000df80: 6f72 206f 6620 6060 736b 6269 6f2e 7374  or of ``skbio.st
-0000df90: 6174 732e 6469 7374 616e 6365 2e61 6e6f  ats.distance.ano
-0000dfa0: 7369 6d60 6020 616e 6420 6060 736b 6269  sim`` and ``skbi
-0000dfb0: 6f2e 7374 6174 732e 6469 7374 616e 6365  o.stats.distance
-0000dfc0: 2e70 6572 6d61 6e6f 7661 6060 2c20 7265  .permanova``, re
-0000dfd0: 7370 6563 7469 7665 6c79 2e0a 2a20 4465  spectively..* De
-0000dfe0: 7072 6563 6174 6564 2060 6073 6b62 696f  precated ``skbio
-0000dff0: 2e73 7461 7473 2e64 6973 7461 6e63 652e  .stats.distance.
-0000e000: 4361 7465 676f 7269 6361 6c53 7461 7473  CategoricalStats
-0000e010: 5265 7375 6c74 7360 6020 696e 2066 6176  Results`` in fav
-0000e020: 6f72 206f 6620 7573 696e 6720 6060 7061  or of using ``pa
-0000e030: 6e64 6173 2e53 6572 6965 7360 6020 746f  ndas.Series`` to
-0000e040: 2073 746f 7265 2073 7461 7469 7374 6963   store statistic
-0000e050: 616c 206d 6574 686f 6420 7265 7375 6c74  al method result
-0000e060: 732e 2060 6061 6e6f 7369 6d60 6020 616e  s. ``anosim`` an
-0000e070: 6420 6060 7065 726d 616e 6f76 6160 6020  d ``permanova`` 
-0000e080: 7265 7475 726e 2060 6070 616e 6461 732e  return ``pandas.
-0000e090: 5365 7269 6573 6060 2069 6e73 7465 6164  Series`` instead
-0000e0a0: 206f 6620 6060 4361 7465 676f 7269 6361   of ``Categorica
-0000e0b0: 6c53 7461 7473 5265 7375 6c74 7360 602e  lStatsResults``.
-0000e0c0: 0a2a 2044 6570 7265 6361 7465 6420 6060  .* Deprecated ``
-0000e0d0: 736b 6269 6f2e 7374 6174 732e 7375 6273  skbio.stats.subs
-0000e0e0: 616d 706c 6560 6020 696e 2066 6176 6f72  ample`` in favor
-0000e0f0: 206f 6620 6060 736b 6269 6f2e 7374 6174   of ``skbio.stat
-0000e100: 732e 7375 6273 616d 706c 655f 636f 756e  s.subsample_coun
-0000e110: 7473 6060 2c20 7768 6963 6820 7072 6f76  ts``, which prov
-0000e120: 6964 6573 2061 6e20 6964 656e 7469 6361  ides an identica
-0000e130: 6c20 696e 7465 7266 6163 653b 206f 6e6c  l interface; onl
-0000e140: 7920 7468 6520 6675 6e63 7469 6f6e 206e  y the function n
-0000e150: 616d 6520 6861 7320 6368 616e 6765 642e  ame has changed.
-0000e160: 2060 6073 6b62 696f 2e73 7461 7473 2e73   ``skbio.stats.s
-0000e170: 7562 7361 6d70 6c65 6060 2077 696c 6c20  ubsample`` will 
-0000e180: 6265 2072 656d 6f76 6564 2069 6e20 7363  be removed in sc
-0000e190: 696b 6974 2d62 696f 2030 2e33 2e30 2e0a  ikit-bio 0.3.0..
-0000e1a0: 0a23 2323 2042 6163 6b77 6172 642d 696e  .### Backward-in
-0000e1b0: 636f 6d70 6174 6962 6c65 2063 6861 6e67  compatible chang
-0000e1c0: 6573 0a2a 2044 6570 7265 6361 7469 6f6e  es.* Deprecation
-0000e1d0: 2077 6172 6e69 6e67 7320 6172 6520 6e6f   warnings are no
-0000e1e0: 7720 7261 6973 6564 2075 7369 6e67 2060  w raised using `
-0000e1f0: 6044 6570 7265 6361 7469 6f6e 5761 726e  `DeprecationWarn
-0000e200: 696e 6760 6020 696e 7374 6561 6420 6f66  ing`` instead of
-0000e210: 2060 6055 7365 7257 6172 6e69 6e67 6060   ``UserWarning``
-0000e220: 2028 5b23 3737 345d 2868 7474 7073 3a2f   ([#774](https:/
-0000e230: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
-0000e240: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
-0000e250: 7373 7565 732f 3737 3429 292e 0a0a 2323  ssues/774))...##
-0000e260: 2320 4d69 7363 656c 6c61 6e65 6f75 730a  # Miscellaneous.
-0000e270: 2a20 5468 6520 6060 7061 6e64 6173 2e44  * The ``pandas.D
-0000e280: 6174 6146 7261 6d65 6060 2072 6574 7572  ataFrame`` retur
-0000e290: 6e65 6420 6279 2060 6073 6b62 696f 2e73  ned by ``skbio.s
-0000e2a0: 7461 7473 2e64 6973 7461 6e63 652e 7077  tats.distance.pw
-0000e2b0: 6d61 6e74 656c 6060 206e 6f77 2073 746f  mantel`` now sto
-0000e2c0: 7265 7320 702d 7661 6c75 6573 2061 7320  res p-values as 
-0000e2d0: 666c 6f61 7473 2061 6e64 2064 6f65 7320  floats and does 
-0000e2e0: 6e6f 7420 636f 6e76 6572 7420 7468 656d  not convert them
-0000e2f0: 2074 6f20 7374 7269 6e67 7320 7769 7468   to strings with
-0000e300: 2061 2073 7065 6369 6669 6320 6e75 6d62   a specific numb
-0000e310: 6572 206f 6620 6469 6769 7473 2e20 702d  er of digits. p-
-0000e320: 7661 6c75 6573 2074 6861 7420 7765 7265  values that were
-0000e330: 2070 7265 7669 6f75 736c 7920 7374 6f72   previously stor
-0000e340: 6564 2061 7320 224e 2f41 2220 6172 6520  ed as "N/A" are 
-0000e350: 6e6f 7720 7374 6f72 6564 2061 7320 6060  now stored as ``
-0000e360: 6e70 2e6e 616e 6060 2066 6f72 2063 6f6e  np.nan`` for con
-0000e370: 7369 7374 656e 6379 2077 6974 6820 6f74  sistency with ot
-0000e380: 6865 7220 7374 6174 6973 7469 6361 6c20  her statistical 
-0000e390: 6d65 7468 6f64 7320 696e 2073 6369 6b69  methods in sciki
-0000e3a0: 742d 6269 6f2e 2053 6565 206e 6f74 6520  t-bio. See note 
-0000e3b0: 696e 2022 4465 7072 6563 6174 6564 2066  in "Deprecated f
-0000e3c0: 756e 6374 696f 6e61 6c69 7479 2220 6162  unctionality" ab
-0000e3d0: 6f76 6520 7265 6761 7264 696e 6720 6060  ove regarding ``
-0000e3e0: 705f 7661 6c75 655f 746f 5f73 7472 6060  p_value_to_str``
-0000e3f0: 2066 6f72 2064 6574 6169 6c73 2e0a 2a20   for details..* 
-0000e400: 7363 696b 6974 2d62 696f 206e 6f77 2073  scikit-bio now s
-0000e410: 7570 706f 7274 7320 7665 7273 696f 6e73  upports versions
-0000e420: 206f 6620 4950 7974 686f 6e20 3c20 322e   of IPython < 2.
-0000e430: 302e 3020 285b 2337 3637 5d28 6874 7470  0.0 ([#767](http
-0000e440: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-0000e450: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-0000e460: 6f2f 6973 7375 6573 2f37 3637 2929 2e0a  o/issues/767))..
-0000e470: 0a23 2320 5665 7273 696f 6e20 302e 322e  .## Version 0.2.
-0000e480: 3120 2832 3031 342d 3130 2d32 3729 0a0a  1 (2014-10-27)..
-0000e490: 5468 6973 2069 7320 616e 2061 6c70 6861  This is an alpha
-0000e4a0: 2072 656c 6561 7365 206f 6620 7363 696b   release of scik
-0000e4b0: 6974 2d62 696f 2e20 4174 2074 6869 7320  it-bio. At this 
-0000e4c0: 7374 6167 652c 206d 616a 6f72 2062 6163  stage, major bac
-0000e4d0: 6b77 6172 6473 2d69 6e63 6f6d 7061 7469  kwards-incompati
-0000e4e0: 626c 6520 4150 4920 6368 616e 6765 7320  ble API changes 
-0000e4f0: 6361 6e20 616e 6420 7769 6c6c 2068 6170  can and will hap
-0000e500: 7065 6e2e 2055 6e69 6669 6564 2049 2f4f  pen. Unified I/O
-0000e510: 2077 6974 6820 7468 6520 7363 696b 6974   with the scikit
-0000e520: 2d62 696f 2049 2f4f 2072 6567 6973 7472  -bio I/O registr
-0000e530: 7920 7761 7320 7468 6520 666f 6375 7320  y was the focus 
-0000e540: 6f66 2074 6869 7320 7265 6c65 6173 652e  of this release.
-0000e550: 0a0a 2323 2320 4665 6174 7572 6573 0a2a  ..### Features.*
-0000e560: 2041 6464 6564 2060 6073 7472 6963 7460   Added ``strict`
-0000e570: 6020 616e 6420 6060 6c6f 6f6b 7570 6060  ` and ``lookup``
-0000e580: 206f 7074 696f 6e61 6c20 7061 7261 6d65   optional parame
-0000e590: 7465 7273 2074 6f20 6060 736b 6269 6f2e  ters to ``skbio.
-0000e5a0: 7374 6174 732e 6469 7374 616e 6365 2e6d  stats.distance.m
-0000e5b0: 616e 7465 6c60 6020 666f 7220 6861 6e64  antel`` for hand
-0000e5c0: 6c69 6e67 2072 656f 7264 6572 696e 6720  ling reordering 
-0000e5d0: 616e 6420 6d61 7463 6869 6e67 206f 6620  and matching of 
-0000e5e0: 4944 7320 7768 656e 2070 726f 7669 6465  IDs when provide
-0000e5f0: 6420 6060 4469 7374 616e 6365 4d61 7472  d ``DistanceMatr
-0000e600: 6978 6060 2069 6e73 7461 6e63 6573 2061  ix`` instances a
-0000e610: 7320 696e 7075 7420 2874 6865 7365 2070  s input (these p
-0000e620: 6172 616d 6574 6572 7320 7765 7265 2070  arameters were p
-0000e630: 7265 7669 6f75 736c 7920 6f6e 6c79 2061  reviously only a
-0000e640: 7661 696c 6162 6c65 2069 6e20 6060 736b  vailable in ``sk
-0000e650: 6269 6f2e 7374 6174 732e 6469 7374 616e  bio.stats.distan
-0000e660: 6365 2e70 776d 616e 7465 6c60 6029 2e0a  ce.pwmantel``)..
-0000e670: 2a20 6060 736b 6269 6f2e 7374 6174 732e  * ``skbio.stats.
-0000e680: 6469 7374 616e 6365 2e70 776d 616e 7465  distance.pwmante
-0000e690: 6c60 6020 6e6f 7720 6163 6365 7074 7320  l`` now accepts 
-0000e6a0: 616e 2069 7465 7261 626c 6520 6f66 2060  an iterable of `
-0000e6b0: 6061 7272 6179 5f6c 696b 6560 6020 6f62  `array_like`` ob
-0000e6c0: 6a65 6374 732e 2050 7265 7669 6f75 736c  jects. Previousl
-0000e6d0: 792c 206f 6e6c 7920 6060 4469 7374 616e  y, only ``Distan
-0000e6e0: 6365 4d61 7472 6978 6060 2069 6e73 7461  ceMatrix`` insta
-0000e6f0: 6e63 6573 2077 6572 6520 616c 6c6f 7765  nces were allowe
-0000e700: 642e 0a2a 2041 6464 6564 2060 6070 6c6f  d..* Added ``plo
-0000e710: 7460 6020 6d65 7468 6f64 2074 6f20 6060  t`` method to ``
-0000e720: 736b 6269 6f2e 7374 6174 732e 6f72 6469  skbio.stats.ordi
-0000e730: 6e61 7469 6f6e 2e4f 7264 696e 6174 696f  nation.Ordinatio
-0000e740: 6e52 6573 756c 7473 6060 2066 6f72 2063  nResults`` for c
-0000e750: 7265 6174 696e 6720 6261 7369 6320 332d  reating basic 3-
-0000e760: 4420 6d61 7470 6c6f 746c 6962 2073 6361  D matplotlib sca
-0000e770: 7474 6572 706c 6f74 7320 6f66 206f 7264  tterplots of ord
-0000e780: 696e 6174 696f 6e20 7265 7375 6c74 732c  ination results,
-0000e790: 206f 7074 696f 6e61 6c6c 7920 636f 6c6f   optionally colo
-0000e7a0: 7265 6420 6279 206d 6574 6164 6174 6120  red by metadata 
-0000e7b0: 696e 2061 2060 6070 616e 6461 732e 4461  in a ``pandas.Da
-0000e7c0: 7461 4672 616d 6560 6020 2873 6565 205b  taFrame`` (see [
-0000e7d0: 2335 3138 5d28 6874 7470 733a 2f2f 6769  #518](https://gi
-0000e7e0: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
-0000e7f0: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
-0000e800: 6573 2f35 3138 2929 2e20 416c 736f 2061  es/518)). Also a
-0000e810: 6464 6564 2020 6060 5f72 6570 725f 706e  dded  ``_repr_pn
-0000e820: 675f 6060 2061 6e64 2060 605f 7265 7072  g_`` and ``_repr
-0000e830: 5f73 7667 5f60 6020 6d65 7468 6f64 7320  _svg_`` methods 
-0000e840: 666f 7220 6175 746f 6d61 7469 6320 6469  for automatic di
-0000e850: 7370 6c61 7920 696e 2074 6865 2049 5079  splay in the IPy
-0000e860: 7468 6f6e 204e 6f74 6562 6f6f 6b2c 2077  thon Notebook, w
-0000e870: 6974 6820 6060 706e 6760 6020 616e 6420  ith ``png`` and 
-0000e880: 6060 7376 6760 6020 7072 6f70 6572 7469  ``svg`` properti
-0000e890: 6573 2066 6f72 2064 6972 6563 7420 6163  es for direct ac
-0000e8a0: 6365 7373 2e0a 2a20 4164 6465 6420 6060  cess..* Added ``
-0000e8b0: 736b 6269 6f2e 7374 6174 732e 6f72 6469  skbio.stats.ordi
-0000e8c0: 6e61 7469 6f6e 2e61 7373 6572 745f 6f72  nation.assert_or
-0000e8d0: 6469 6e61 7469 6f6e 5f72 6573 756c 7473  dination_results
-0000e8e0: 5f65 7175 616c 6060 2066 6f72 2063 6f6d  _equal`` for com
-0000e8f0: 7061 7269 6e67 2060 604f 7264 696e 6174  paring ``Ordinat
-0000e900: 696f 6e52 6573 756c 7473 6060 206f 626a  ionResults`` obj
-0000e910: 6563 7473 2066 6f72 2065 7175 616c 6974  ects for equalit
-0000e920: 7920 696e 2075 6e69 7420 7465 7374 732e  y in unit tests.
-0000e930: 0a2a 2060 6042 696f 6c6f 6769 6361 6c53  .* ``BiologicalS
-0000e940: 6571 7565 6e63 6560 6020 2861 6e64 2069  equence`` (and i
-0000e950: 7473 2073 7562 636c 6173 7365 7329 206e  ts subclasses) n
-0000e960: 6f77 206f 7074 696f 6e61 6c6c 7920 7374  ow optionally st
-0000e970: 6f72 6520 5068 7265 6420 7175 616c 6974  ore Phred qualit
-0000e980: 7920 7363 6f72 6573 2e20 4120 6269 6f6c  y scores. A biol
-0000e990: 6f67 6963 616c 2073 6571 7565 6e63 6527  ogical sequence'
-0000e9a0: 7320 7175 616c 6974 7920 7363 6f72 6573  s quality scores
-0000e9b0: 2061 7265 2073 746f 7265 6420 6173 2061   are stored as a
-0000e9c0: 2031 2d44 2060 606e 756d 7079 2e6e 6461   1-D ``numpy.nda
-0000e9d0: 7272 6179 6060 206f 6620 6e6f 6e6e 6567  rray`` of nonneg
-0000e9e0: 6174 6976 6520 696e 7465 6765 7273 2074  ative integers t
-0000e9f0: 6861 7420 6973 2074 6865 2073 616d 6520  hat is the same 
-0000ea00: 6c65 6e67 7468 2061 7320 7468 6520 6269  length as the bi
-0000ea10: 6f6c 6f67 6963 616c 2073 6571 7565 6e63  ological sequenc
-0000ea20: 652e 2051 7561 6c69 7479 2073 636f 7265  e. Quality score
-0000ea30: 7320 6361 6e20 6265 2070 726f 7669 6465  s can be provide
-0000ea40: 6420 7570 6f6e 206f 626a 6563 7420 696e  d upon object in
-0000ea50: 7374 616e 7469 6174 696f 6e20 7669 6120  stantiation via 
-0000ea60: 7468 6520 6b65 7977 6f72 6420 6172 6775  the keyword argu
-0000ea70: 6d65 6e74 2060 6071 7561 6c69 7479 6060  ment ``quality``
-0000ea80: 2c20 616e 6420 6361 6e20 6265 2072 6574  , and can be ret
-0000ea90: 7269 6576 6564 2076 6961 2074 6865 2060  rieved via the `
-0000eaa0: 6042 696f 6c6f 6769 6361 6c53 6571 7565  `BiologicalSeque
-0000eab0: 6e63 652e 7175 616c 6974 7960 6020 7072  nce.quality`` pr
-0000eac0: 6f70 6572 7479 2e20 6060 4269 6f6c 6f67  operty. ``Biolog
-0000ead0: 6963 616c 5365 7175 656e 6365 2e68 6173  icalSequence.has
-0000eae0: 5f71 7561 6c69 7479 6060 2069 7320 616c  _quality`` is al
-0000eaf0: 736f 2070 726f 7669 6465 6420 666f 7220  so provided for 
-0000eb00: 6465 7465 726d 696e 696e 6720 7768 6574  determining whet
-0000eb10: 6865 7220 6120 6269 6f6c 6f67 6963 616c  her a biological
-0000eb20: 2073 6571 7565 6e63 6520 6861 7320 7175   sequence has qu
-0000eb30: 616c 6974 7920 7363 6f72 6573 206f 7220  ality scores or 
-0000eb40: 6e6f 742e 2053 6565 205b 2336 3136 5d28  not. See [#616](
-0000eb50: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-0000eb60: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
-0000eb70: 742d 6269 6f2f 6973 7375 6573 2f36 3136  t-bio/issues/616
-0000eb80: 2920 666f 7220 6d6f 7265 2064 6574 6169  ) for more detai
-0000eb90: 6c73 2e0a 2a20 4164 6465 6420 6060 4269  ls..* Added ``Bi
-0000eba0: 6f6c 6f67 6963 616c 5365 7175 656e 6365  ologicalSequence
-0000ebb0: 2e73 6571 7565 6e63 6560 6020 7072 6f70  .sequence`` prop
-0000ebc0: 6572 7479 2066 6f72 2072 6574 7269 6576  erty for retriev
-0000ebd0: 696e 6720 7468 6520 756e 6465 726c 7969  ing the underlyi
-0000ebe0: 6e67 2073 7472 696e 6720 7265 7072 6573  ng string repres
-0000ebf0: 656e 7469 6e67 2074 6865 2073 6571 7565  enting the seque
-0000ec00: 6e63 6520 6368 6172 6163 7465 7273 2e20  nce characters. 
-0000ec10: 5468 6973 2077 6173 2070 7265 7669 6f75  This was previou
-0000ec20: 736c 7920 2861 6e64 2073 7469 6c6c 2069  sly (and still i
-0000ec30: 7329 2061 6363 6573 7369 626c 6520 7669  s) accessible vi
-0000ec40: 6120 6060 4269 6f6c 6f67 6963 616c 5365  a ``BiologicalSe
-0000ec50: 7175 656e 6365 2e5f 5f73 7472 5f5f 6060  quence.__str__``
-0000ec60: 2e20 4974 2069 7320 7072 6f76 6964 6564  . It is provided
-0000ec70: 2076 6961 2061 2070 726f 7065 7274 7920   via a property 
-0000ec80: 666f 7220 636f 6e76 656e 6965 6e63 6520  for convenience 
-0000ec90: 616e 6420 6578 706c 6963 6974 6e65 7373  and explicitness
-0000eca0: 2e0a 2a20 4164 6465 6420 6060 4269 6f6c  ..* Added ``Biol
-0000ecb0: 6f67 6963 616c 5365 7175 656e 6365 2e65  ogicalSequence.e
-0000ecc0: 7175 616c 7360 6020 666f 7220 6675 6c6c  quals`` for full
-0000ecd0: 2063 6f6e 7472 6f6c 206f 7665 7220 6571   control over eq
-0000ece0: 7561 6c69 7479 2074 6573 7469 6e67 206f  uality testing o
-0000ecf0: 6620 6269 6f6c 6f67 6963 616c 2073 6571  f biological seq
-0000ed00: 7565 6e63 6573 2e20 4279 2064 6566 6175  uences. By defau
-0000ed10: 6c74 2c20 6269 6f6c 6f67 6963 616c 2073  lt, biological s
-0000ed20: 6571 7565 6e63 6573 206d 7573 7420 6861  equences must ha
-0000ed30: 7665 2074 6865 2073 616d 6520 7479 7065  ve the same type
-0000ed40: 2c20 756e 6465 726c 7969 6e67 2073 6571  , underlying seq
-0000ed50: 7565 6e63 6520 6f66 2063 6861 7261 6374  uence of charact
-0000ed60: 6572 732c 2069 6465 6e74 6966 6965 722c  ers, identifier,
-0000ed70: 2064 6573 6372 6970 7469 6f6e 2c20 616e   description, an
-0000ed80: 6420 7175 616c 6974 7920 7363 6f72 6573  d quality scores
-0000ed90: 2074 6f20 636f 6d70 6172 6520 6571 7561   to compare equa
-0000eda0: 6c2e 2054 6865 7365 2070 726f 7065 7274  l. These propert
-0000edb0: 6965 7320 6361 6e20 6265 2069 676e 6f72  ies can be ignor
-0000edc0: 6564 2076 6961 2074 6865 206b 6579 776f  ed via the keywo
-0000edd0: 7264 2061 7267 756d 656e 7420 6060 6967  rd argument ``ig
-0000ede0: 6e6f 7265 6060 2e20 5468 6520 6265 6861  nore``. The beha
-0000edf0: 7669 6f72 206f 6620 6060 4269 6f6c 6f67  vior of ``Biolog
-0000ee00: 6963 616c 5365 7175 656e 6365 2e5f 5f65  icalSequence.__e
-0000ee10: 715f 5f60 602f 6060 5f5f 6e65 5f5f 6060  q__``/``__ne__``
-0000ee20: 2072 656d 6169 6e73 2075 6e63 6861 6e67   remains unchang
-0000ee30: 6564 2028 6f6e 6c79 2074 7970 6520 616e  ed (only type an
-0000ee40: 6420 756e 6465 726c 7969 6e67 2073 6571  d underlying seq
-0000ee50: 7565 6e63 6520 6f66 2063 6861 7261 6374  uence of charact
-0000ee60: 6572 7320 6172 6520 636f 6d70 6172 6564  ers are compared
-0000ee70: 292e 0a2a 2041 6464 6564 2060 6042 696f  )..* Added ``Bio
-0000ee80: 6c6f 6769 6361 6c53 6571 7565 6e63 652e  logicalSequence.
-0000ee90: 636f 7079 6060 2066 6f72 2063 7265 6174  copy`` for creat
-0000eea0: 696e 6720 6120 636f 7079 206f 6620 6120  ing a copy of a 
-0000eeb0: 6269 6f6c 6f67 6963 616c 2073 6571 7565  biological seque
-0000eec0: 6e63 652c 206f 7074 696f 6e61 6c6c 7920  nce, optionally 
-0000eed0: 7769 7468 206f 6e65 206f 7220 6d6f 7265  with one or more
-0000eee0: 2061 7474 7269 6275 7465 7320 7570 6461   attributes upda
-0000eef0: 7465 642e 0a2a 2060 6042 696f 6c6f 6769  ted..* ``Biologi
-0000ef00: 6361 6c53 6571 7565 6e63 652e 5f5f 6765  calSequence.__ge
-0000ef10: 7469 7465 6d5f 5f60 6020 6e6f 7720 7375  titem__`` now su
-0000ef20: 7070 6f72 7473 2073 7065 6369 6679 696e  pports specifyin
-0000ef30: 6720 6120 7365 7175 656e 6365 206f 6620  g a sequence of 
-0000ef40: 696e 6469 6365 7320 746f 2074 616b 6520  indices to take 
-0000ef50: 6672 6f6d 2074 6865 2062 696f 6c6f 6769  from the biologi
-0000ef60: 6361 6c20 7365 7175 656e 6365 2e0a 2a20  cal sequence..* 
-0000ef70: 4d65 7468 6f64 7320 746f 2072 6561 6420  Methods to read 
-0000ef80: 616e 6420 7772 6974 6520 7461 786f 6e6f  and write taxono
-0000ef90: 6d69 6573 2061 7265 206e 6f77 2061 7661  mies are now ava
-0000efa0: 696c 6162 6c65 2075 6e64 6572 2060 6073  ilable under ``s
-0000efb0: 6b62 696f 2e74 7265 652e 5472 6565 4e6f  kbio.tree.TreeNo
-0000efc0: 6465 2e66 726f 6d5f 7461 786f 6e6f 6d79  de.from_taxonomy
-0000efd0: 6060 2061 6e64 2060 6073 6b62 696f 2e74  `` and ``skbio.t
-0000efe0: 7265 652e 5472 6565 4e6f 6465 2e74 6f5f  ree.TreeNode.to_
-0000eff0: 7461 786f 6e6f 6d79 6060 2072 6573 7065  taxonomy`` respe
-0000f000: 6374 6976 656c 792e 0a2a 2041 6464 6564  ctively..* Added
-0000f010: 2060 6053 6571 7565 6e63 6543 6f6c 6c65   ``SequenceColle
-0000f020: 6374 696f 6e2e 7570 6461 7465 5f69 6473  ction.update_ids
-0000f030: 6060 2c20 7768 6963 6820 7072 6f76 6964  ``, which provid
-0000f040: 6573 2061 2066 6c65 7869 626c 6520 7761  es a flexible wa
-0000f050: 7920 6f66 2075 7064 6174 696e 6720 7365  y of updating se
-0000f060: 7175 656e 6365 2049 4473 206f 6e20 6120  quence IDs on a 
-0000f070: 6060 5365 7175 656e 6365 436f 6c6c 6563  ``SequenceCollec
-0000f080: 7469 6f6e 6060 206f 7220 6060 416c 6967  tion`` or ``Alig
-0000f090: 6e6d 656e 7460 6020 286e 6f74 6520 7468  nment`` (note th
-0000f0a0: 6174 2061 206e 6577 206f 626a 6563 7420  at a new object 
-0000f0b0: 6973 2072 6574 7572 6e65 642c 2073 696e  is returned, sin
-0000f0c0: 6365 2069 6e73 7461 6e63 6573 206f 6620  ce instances of 
-0000f0d0: 7468 6573 6520 636c 6173 7365 7320 6172  these classes ar
-0000f0e0: 6520 696d 6d75 7461 626c 6529 2e20 4465  e immutable). De
-0000f0f0: 7072 6563 6174 6564 2060 6053 6571 7565  precated ``Seque
-0000f100: 6e63 6543 6f6c 6c65 6374 696f 6e2e 696e  nceCollection.in
-0000f110: 745f 6d61 7060 6020 696e 2066 6176 6f72  t_map`` in favor
-0000f120: 206f 6620 7468 6973 206e 6577 206d 6574   of this new met
-0000f130: 686f 643b 2069 7420 7769 6c6c 2062 6520  hod; it will be 
-0000f140: 7265 6d6f 7665 6420 696e 2073 6369 6b69  removed in sciki
-0000f150: 742d 6269 6f20 302e 332e 302e 0a2a 2041  t-bio 0.3.0..* A
-0000f160: 6464 6564 2060 6073 6b62 696f 2e75 7469  dded ``skbio.uti
-0000f170: 6c2e 6361 7264 696e 616c 5f74 6f5f 6f72  l.cardinal_to_or
-0000f180: 6469 6e61 6c60 6020 666f 7220 636f 6e76  dinal`` for conv
-0000f190: 6572 7469 6e67 2061 2063 6172 6469 6e61  erting a cardina
-0000f1a0: 6c20 6e75 6d62 6572 2074 6f20 6f72 6469  l number to ordi
-0000f1b0: 6e61 6c20 7374 7269 6e67 2028 652e 672e  nal string (e.g.
-0000f1c0: 2c20 7573 6566 756c 2066 6f72 2065 7272  , useful for err
-0000f1d0: 6f72 206d 6573 7361 6765 7329 2e0a 2a20  or messages)..* 
-0000f1e0: 4e65 7720 492f 4f20 5265 6769 7374 7279  New I/O Registry
-0000f1f0: 3a20 7375 7070 6f72 7473 206d 756c 7469  : supports multi
-0000f200: 706c 6520 6669 6c65 2066 6f72 6d61 7473  ple file formats
-0000f210: 2c20 6175 746f 6d61 7469 6320 6669 6c65  , automatic file
-0000f220: 2066 6f72 6d61 7420 6465 7465 6374 696f   format detectio
-0000f230: 6e20 7768 656e 2072 6561 6469 6e67 2c20  n when reading, 
-0000f240: 756e 6966 6965 6420 7072 6f63 6564 7572  unified procedur
-0000f250: 616c 2060 6073 6b62 696f 2e69 6f2e 7265  al ``skbio.io.re
-0000f260: 6164 6060 2061 6e64 2060 6073 6b62 696f  ad`` and ``skbio
-0000f270: 2e69 6f2e 7772 6974 6560 6020 696e 2061  .io.write`` in a
-0000f280: 6464 6974 696f 6e20 746f 204f 4f50 2069  ddition to OOP i
-0000f290: 6e74 6572 6661 6365 7320 2860 6072 6561  nterfaces (``rea
-0000f2a0: 642f 7772 6974 6560 6020 6d65 7468 6f64  d/write`` method
-0000f2b0: 7329 206f 6e20 7468 6520 6265 6c6f 7720  s) on the below 
-0000f2c0: 6f62 6a65 6374 732e 2053 6565 2060 6073  objects. See ``s
-0000f2d0: 6b62 696f 2e69 6f60 6020 666f 7220 6d6f  kbio.io`` for mo
-0000f2e0: 7265 2064 6574 6169 6c73 2e0a 2020 2020  re details..    
-0000f2f0: 2d20 4164 6465 6420 2263 6c75 7374 616c  - Added "clustal
-0000f300: 2220 666f 726d 6174 2073 7570 706f 7274  " format support
-0000f310: 3a0a 2020 2020 2020 2020 2a20 4861 7320  :.        * Has 
-0000f320: 736e 6966 6665 720a 2020 2020 2020 2020  sniffer.        
-0000f330: 2a20 5265 6164 6572 733a 2060 6041 6c69  * Readers: ``Ali
-0000f340: 676e 6d65 6e74 6060 0a20 2020 2020 2020  gnment``.       
-0000f350: 202a 2057 7269 7465 7273 3a20 6060 416c   * Writers: ``Al
-0000f360: 6967 6e6d 656e 7460 600a 2020 2020 2d20  ignment``.    - 
-0000f370: 4164 6465 6420 226c 736d 6174 2220 666f  Added "lsmat" fo
-0000f380: 726d 6174 2073 7570 706f 7274 3a0a 2020  rmat support:.  
-0000f390: 2020 2020 2020 2a20 4861 7320 736e 6966        * Has snif
-0000f3a0: 6665 720a 2020 2020 2020 2020 2a20 5265  fer.        * Re
-0000f3b0: 6164 6572 733a 2060 6044 6973 7369 6d69  aders: ``Dissimi
-0000f3c0: 6c61 7269 7479 4d61 7472 6978 6060 2c20  larityMatrix``, 
-0000f3d0: 6060 4469 7374 616e 6365 4d61 7472 6978  ``DistanceMatrix
-0000f3e0: 6060 0a20 2020 2020 2020 202a 2057 7269  ``.        * Wri
-0000f3f0: 7465 7273 3a20 6060 4469 7373 696d 696c  ters: ``Dissimil
-0000f400: 6172 6974 794d 6174 7269 7860 602c 2060  arityMatrix``, `
-0000f410: 6044 6973 7461 6e63 654d 6174 7269 7860  `DistanceMatrix`
-0000f420: 600a 2020 2020 2d20 4164 6465 6420 226f  `.    - Added "o
-0000f430: 7264 696e 6174 696f 6e22 2066 6f72 6d61  rdination" forma
-0000f440: 7420 7375 7070 6f72 743a 0a20 2020 2020  t support:.     
-0000f450: 2020 202a 2048 6173 2073 6e69 6666 6572     * Has sniffer
-0000f460: 0a20 2020 2020 2020 202a 2052 6561 6465  .        * Reade
-0000f470: 7273 3a20 6060 4f72 6469 6e61 7469 6f6e  rs: ``Ordination
-0000f480: 5265 7375 6c74 7360 600a 2020 2020 2020  Results``.      
-0000f490: 2020 2a20 5772 6974 6572 733a 2060 604f    * Writers: ``O
-0000f4a0: 7264 696e 6174 696f 6e52 6573 756c 7473  rdinationResults
-0000f4b0: 6060 0a20 2020 202d 2041 6464 6564 2022  ``.    - Added "
-0000f4c0: 6e65 7769 636b 2220 666f 726d 6174 2073  newick" format s
-0000f4d0: 7570 706f 7274 3a0a 2020 2020 2020 2020  upport:.        
-0000f4e0: 2a20 4861 7320 736e 6966 6665 720a 2020  * Has sniffer.  
-0000f4f0: 2020 2020 2020 2a20 5265 6164 6572 733a        * Readers:
-0000f500: 2060 6054 7265 654e 6f64 6560 600a 2020   ``TreeNode``.  
-0000f510: 2020 2020 2020 2a20 5772 6974 6572 733a        * Writers:
-0000f520: 2060 6054 7265 654e 6f64 6560 600a 2020   ``TreeNode``.  
-0000f530: 2020 2d20 4164 6465 6420 2270 6879 6c69    - Added "phyli
-0000f540: 7022 2066 6f72 6d61 7420 7375 7070 6f72  p" format suppor
-0000f550: 743a 0a20 2020 2020 2020 202a 204e 6f20  t:.        * No 
-0000f560: 736e 6966 6665 720a 2020 2020 2020 2020  sniffer.        
-0000f570: 2a20 5265 6164 6572 733a 204e 6f6e 650a  * Readers: None.
-0000f580: 2020 2020 2020 2020 2a20 5772 6974 6572          * Writer
-0000f590: 733a 2060 6041 6c69 676e 6d65 6e74 6060  s: ``Alignment``
-0000f5a0: 0a20 2020 202d 2041 6464 6564 2022 7173  .    - Added "qs
-0000f5b0: 6571 2220 666f 726d 6174 2073 7570 706f  eq" format suppo
-0000f5c0: 7274 3a0a 2020 2020 2020 2020 2a20 4861  rt:.        * Ha
-0000f5d0: 7320 736e 6966 6665 720a 2020 2020 2020  s sniffer.      
-0000f5e0: 2020 2a20 5265 6164 6572 733a 2067 656e    * Readers: gen
-0000f5f0: 6572 6174 6f72 206f 6620 6060 4269 6f6c  erator of ``Biol
-0000f600: 6f67 6963 616c 5365 7175 656e 6365 6060  ogicalSequence``
-0000f610: 206f 7220 6974 7320 7375 6263 6c61 7373   or its subclass
-0000f620: 6573 2c20 6060 5365 7175 656e 6365 436f  es, ``SequenceCo
-0000f630: 6c6c 6563 7469 6f6e 6060 2c20 6060 4269  llection``, ``Bi
-0000f640: 6f6c 6f67 6963 616c 5365 7175 656e 6365  ologicalSequence
-0000f650: 6060 2c20 6060 4e75 636c 656f 7469 6465  ``, ``Nucleotide
-0000f660: 5365 7175 656e 6365 6060 2c20 6060 444e  Sequence``, ``DN
-0000f670: 4153 6571 7565 6e63 6560 602c 2060 6052  ASequence``, ``R
-0000f680: 4e41 5365 7175 656e 6365 6060 2c20 6060  NASequence``, ``
-0000f690: 5072 6f74 6569 6e53 6571 7565 6e63 6560  ProteinSequence`
-0000f6a0: 600a 2020 2020 2020 2020 2a20 5772 6974  `.        * Writ
-0000f6b0: 6572 733a 204e 6f6e 650a 2020 2020 2d20  ers: None.    - 
-0000f6c0: 4164 6465 6420 2266 6173 7461 222f 5155  Added "fasta"/QU
-0000f6d0: 414c 2066 6f72 6d61 7420 7375 7070 6f72  AL format suppor
-0000f6e0: 743a 0a20 2020 2020 2020 202a 2048 6173  t:.        * Has
-0000f6f0: 2073 6e69 6666 6572 0a20 2020 2020 2020   sniffer.       
-0000f700: 202a 2052 6561 6465 7273 3a20 6765 6e65   * Readers: gene
-0000f710: 7261 746f 7220 6f66 2060 6042 696f 6c6f  rator of ``Biolo
-0000f720: 6769 6361 6c53 6571 7565 6e63 6560 6020  gicalSequence`` 
-0000f730: 6f72 2069 7473 2073 7562 636c 6173 7365  or its subclasse
-0000f740: 732c 2060 6053 6571 7565 6e63 6543 6f6c  s, ``SequenceCol
-0000f750: 6c65 6374 696f 6e60 602c 2060 6041 6c69  lection``, ``Ali
-0000f760: 676e 6d65 6e74 6060 2c20 6060 4269 6f6c  gnment``, ``Biol
-0000f770: 6f67 6963 616c 5365 7175 656e 6365 6060  ogicalSequence``
-0000f780: 2c20 6060 4e75 636c 656f 7469 6465 5365  , ``NucleotideSe
-0000f790: 7175 656e 6365 6060 2c20 6060 444e 4153  quence``, ``DNAS
-0000f7a0: 6571 7565 6e63 6560 602c 2060 6052 4e41  equence``, ``RNA
-0000f7b0: 5365 7175 656e 6365 6060 2c20 6060 5072  Sequence``, ``Pr
-0000f7c0: 6f74 6569 6e53 6571 7565 6e63 6560 600a  oteinSequence``.
-0000f7d0: 2020 2020 2020 2020 2a20 5772 6974 6572          * Writer
-0000f7e0: 733a 2073 616d 6520 6173 2072 6561 6465  s: same as reade
-0000f7f0: 7273 0a20 2020 202d 2041 6464 6564 2022  rs.    - Added "
-0000f800: 6661 7374 7122 2066 6f72 6d61 7420 7375  fastq" format su
-0000f810: 7070 6f72 743a 0a20 2020 2020 2020 202a  pport:.        *
-0000f820: 2048 6173 2073 6e69 6666 6572 0a20 2020   Has sniffer.   
-0000f830: 2020 2020 202a 2052 6561 6465 7273 3a20       * Readers: 
-0000f840: 6765 6e65 7261 746f 7220 6f66 2060 6042  generator of ``B
-0000f850: 696f 6c6f 6769 6361 6c53 6571 7565 6e63  iologicalSequenc
-0000f860: 6560 6020 6f72 2069 7473 2073 7562 636c  e`` or its subcl
-0000f870: 6173 7365 732c 2060 6053 6571 7565 6e63  asses, ``Sequenc
-0000f880: 6543 6f6c 6c65 6374 696f 6e60 602c 2060  eCollection``, `
-0000f890: 6041 6c69 676e 6d65 6e74 6060 2c20 6060  `Alignment``, ``
-0000f8a0: 4269 6f6c 6f67 6963 616c 5365 7175 656e  BiologicalSequen
-0000f8b0: 6365 6060 2c20 6060 4e75 636c 656f 7469  ce``, ``Nucleoti
-0000f8c0: 6465 5365 7175 656e 6365 6060 2c20 6060  deSequence``, ``
-0000f8d0: 444e 4153 6571 7565 6e63 6560 602c 2060  DNASequence``, `
-0000f8e0: 6052 4e41 5365 7175 656e 6365 6060 2c20  `RNASequence``, 
-0000f8f0: 6060 5072 6f74 6569 6e53 6571 7565 6e63  ``ProteinSequenc
-0000f900: 6560 600a 2020 2020 2020 2020 2a20 5772  e``.        * Wr
-0000f910: 6974 6572 733a 2073 616d 6520 6173 2072  iters: same as r
-0000f920: 6561 6465 7273 0a0a 2323 2320 4275 6720  eaders..### Bug 
-0000f930: 6669 7865 730a 0a2a 2052 656d 6f76 6564  fixes..* Removed
-0000f940: 2060 6063 6f6e 7374 7275 6374 6f72 6060   ``constructor``
-0000f950: 2070 6172 616d 6574 6572 2066 726f 6d20   parameter from 
-0000f960: 6060 416c 6967 6e6d 656e 742e 6b5f 776f  ``Alignment.k_wo
-0000f970: 7264 5f66 7265 7175 656e 6369 6573 6060  rd_frequencies``
-0000f980: 2c20 6060 4269 6f6c 6f67 6963 616c 5365  , ``BiologicalSe
-0000f990: 7175 656e 6365 2e6b 5f77 6f72 6473 6060  quence.k_words``
-0000f9a0: 2c20 6060 4269 6f6c 6f67 6963 616c 5365  , ``BiologicalSe
-0000f9b0: 7175 656e 6365 2e6b 5f77 6f72 645f 636f  quence.k_word_co
-0000f9c0: 756e 7473 6060 2c20 616e 6420 6060 4269  unts``, and ``Bi
-0000f9d0: 6f6c 6f67 6963 616c 5365 7175 656e 6365  ologicalSequence
-0000f9e0: 2e6b 5f77 6f72 645f 6672 6571 7565 6e63  .k_word_frequenc
-0000f9f0: 6965 7360 6020 6173 2069 7420 6861 6420  ies`` as it had 
-0000fa00: 6e6f 2065 6666 6563 7420 2869 7420 7761  no effect (it wa
-0000fa10: 7320 6e65 7665 7220 686f 6f6b 6564 2075  s never hooked u
-0000fa20: 7020 696e 2074 6865 2075 6e64 6572 6c79  p in the underly
-0000fa30: 696e 6720 636f 6465 292e 2060 6042 696f  ing code). ``Bio
-0000fa40: 6c6f 6769 6361 6c53 6571 7565 6e63 652e  logicalSequence.
-0000fa50: 6b5f 776f 7264 7360 6020 6e6f 7720 7265  k_words`` now re
-0000fa60: 7475 726e 7320 6120 6765 6e65 7261 746f  turns a generato
-0000fa70: 7220 6f66 2060 6042 696f 6c6f 6769 6361  r of ``Biologica
-0000fa80: 6c53 6571 7565 6e63 6560 6020 6f62 6a65  lSequence`` obje
-0000fa90: 6374 7320 696e 7374 6561 6420 6f66 2073  cts instead of s
-0000faa0: 7472 696e 6773 2e0a 2a20 4d6f 6469 6669  trings..* Modifi
-0000fab0: 6564 2074 6865 2060 6041 6c69 676e 6d65  ed the ``Alignme
-0000fac0: 6e74 6060 2063 6f6e 7374 7275 6374 6f72  nt`` constructor
-0000fad0: 2074 6f20 7665 7269 6679 2074 6861 7420   to verify that 
-0000fae0: 616c 6c20 7365 7175 656e 6365 7320 6861  all sequences ha
-0000faf0: 7665 2074 6865 2073 616d 6520 6c65 6e67  ve the same leng
-0000fb00: 7468 2c20 6966 206e 6f74 2c20 7261 6973  th, if not, rais
-0000fb10: 6520 616e 2060 6041 6c69 676e 6d65 6e74  e an ``Alignment
-0000fb20: 4572 726f 7260 6020 6578 6365 7074 696f  Error`` exceptio
-0000fb30: 6e2e 2020 5570 6461 7465 6420 7468 6520  n.  Updated the 
-0000fb40: 6d65 7468 6f64 2060 6041 6c69 676e 6d65  method ``Alignme
-0000fb50: 6e74 2e73 7562 616c 6967 6e6d 656e 7460  nt.subalignment`
-0000fb60: 6020 746f 2063 616c 6375 6c61 7465 2074  ` to calculate t
-0000fb70: 6865 2069 6e64 6963 6573 206f 6e6c 7920  he indices only 
-0000fb80: 6f6e 6365 206e 6f77 2074 6861 7420 6964  once now that id
-0000fb90: 656e 7469 6361 6c20 7365 7175 656e 6365  entical sequence
-0000fba0: 206c 656e 6774 6820 6973 2067 7561 7261   length is guara
-0000fbb0: 6e74 6565 642e 0a0a 2323 2320 4465 7072  nteed...### Depr
-0000fbc0: 6563 6174 6564 2066 756e 6374 696f 6e61  ecated functiona
-0000fbd0: 6c69 7479 0a2a 2044 6570 7265 6361 7465  lity.* Deprecate
-0000fbe0: 6420 6060 636f 6e73 7472 7563 746f 7260  d ``constructor`
-0000fbf0: 6020 7061 7261 6d65 7465 7220 696e 2060  ` parameter in `
-0000fc00: 6041 6c69 676e 6d65 6e74 2e6d 616a 6f72  `Alignment.major
-0000fc10: 6974 795f 636f 6e73 656e 7375 7360 6020  ity_consensus`` 
-0000fc20: 696e 2066 6176 6f72 206f 6620 6861 7669  in favor of havi
-0000fc30: 6e67 2075 7365 7273 2063 616c 6c20 6060  ng users call ``
-0000fc40: 7374 7260 6020 6f6e 2074 6865 2072 6574  str`` on the ret
-0000fc50: 7572 6e65 6420 6060 4269 6f6c 6f67 6963  urned ``Biologic
-0000fc60: 616c 5365 7175 656e 6365 6060 2e20 5468  alSequence``. Th
-0000fc70: 6973 2070 6172 616d 6574 6572 2077 696c  is parameter wil
-0000fc80: 6c20 6265 2072 656d 6f76 6564 2069 6e20  l be removed in 
-0000fc90: 7363 696b 6974 2d62 696f 2030 2e33 2e30  scikit-bio 0.3.0
-0000fca0: 2e0a 0a2a 2045 7869 7374 696e 6720 492f  ...* Existing I/
-0000fcb0: 4f20 6675 6e63 7469 6f6e 616c 6974 7920  O functionality 
-0000fcc0: 6465 7072 6563 6174 6564 2069 6e20 6661  deprecated in fa
-0000fcd0: 766f 7220 6f66 2049 2f4f 2072 6567 6973  vor of I/O regis
-0000fce0: 7472 792c 206f 6c64 2066 756e 6374 696f  try, old functio
-0000fcf0: 6e61 6c69 7479 2077 696c 6c20 6265 2072  nality will be r
-0000fd00: 656d 6f76 6564 2069 6e20 7363 696b 6974  emoved in scikit
-0000fd10: 2d62 696f 2030 2e33 2e30 2e20 416c 6c20  -bio 0.3.0. All 
-0000fd20: 6675 6e63 7469 6f6e 616c 6974 7920 6361  functionality ca
-0000fd30: 6e20 6265 2066 6f75 6e64 2061 7420 6060  n be found at ``
-0000fd40: 736b 6269 6f2e 696f 2e72 6561 6460 602c  skbio.io.read``,
-0000fd50: 2060 6073 6b62 696f 2e69 6f2e 7772 6974   ``skbio.io.writ
-0000fd60: 6560 602c 2061 6e64 2074 6865 206d 6574  e``, and the met
-0000fd70: 686f 6473 206c 6973 7465 6420 6265 6c6f  hods listed belo
-0000fd80: 773a 0a20 2020 202a 2044 6570 7265 6361  w:.    * Depreca
-0000fd90: 7465 6420 7468 6520 666f 6c6c 6f77 696e  ted the followin
-0000fda0: 6720 2263 6c75 7374 616c 2220 7265 6164  g "clustal" read
-0000fdb0: 6572 732f 7772 6974 6572 733a 0a20 2020  ers/writers:.   
-0000fdc0: 2020 2020 202d 2060 6077 7269 7465 5f63       - ``write_c
-0000fdd0: 6c75 7374 616c 6060 202d 3e20 6060 416c  lustal`` -> ``Al
-0000fde0: 6967 6e6d 656e 742e 7772 6974 6560 600a  ignment.write``.
-0000fdf0: 2020 2020 2020 2020 2d20 6060 7061 7273          - ``pars
-0000fe00: 655f 636c 7573 7461 6c60 6020 2d3e 2060  e_clustal`` -> `
-0000fe10: 6041 6c69 676e 6d65 6e74 2e72 6561 6460  `Alignment.read`
-0000fe20: 600a 0a20 2020 202a 2044 6570 7265 6361  `..    * Depreca
-0000fe30: 7465 6420 7468 6520 666f 6c6c 6f77 696e  ted the followin
-0000fe40: 6720 6469 7374 616e 6365 206d 6174 7269  g distance matri
-0000fe50: 7820 666f 726d 6174 2028 226c 736d 6174  x format ("lsmat
-0000fe60: 2229 2072 6561 6465 7273 2f77 7269 7465  ") readers/write
-0000fe70: 7273 3a0a 2020 2020 2020 2020 2d20 6060  rs:.        - ``
-0000fe80: 4469 7373 696d 696c 6172 6974 794d 6174  DissimilarityMat
-0000fe90: 7269 782e 6672 6f6d 5f66 696c 6560 6020  rix.from_file`` 
-0000fea0: 2d3e 2060 6044 6973 7369 6d69 6c61 7269  -> ``Dissimilari
-0000feb0: 7479 4d61 7472 6978 2e72 6561 6460 600a  tyMatrix.read``.
-0000fec0: 2020 2020 2020 2020 2d20 6060 4469 7373          - ``Diss
-0000fed0: 696d 696c 6172 6974 794d 6174 7269 782e  imilarityMatrix.
-0000fee0: 746f 5f66 696c 6560 6020 2d3e 2060 6044  to_file`` -> ``D
-0000fef0: 6973 7369 6d69 6c61 7269 7479 4d61 7472  issimilarityMatr
-0000ff00: 6978 2e77 7269 7465 6060 0a20 2020 2020  ix.write``.     
-0000ff10: 2020 202d 2060 6044 6973 7461 6e63 654d     - ``DistanceM
-0000ff20: 6174 7269 782e 6672 6f6d 5f66 696c 6560  atrix.from_file`
-0000ff30: 6020 2d3e 2060 6044 6973 7461 6e63 654d  ` -> ``DistanceM
-0000ff40: 6174 7269 782e 7265 6164 6060 0a20 2020  atrix.read``.   
-0000ff50: 2020 2020 202d 2060 6044 6973 7461 6e63       - ``Distanc
-0000ff60: 654d 6174 7269 782e 746f 5f66 696c 6560  eMatrix.to_file`
-0000ff70: 6020 2d3e 2060 6044 6973 7461 6e63 654d  ` -> ``DistanceM
-0000ff80: 6174 7269 782e 7772 6974 6560 600a 0a20  atrix.write``.. 
-0000ff90: 2020 202a 2044 6570 7265 6361 7465 6420     * Deprecated 
-0000ffa0: 7468 6520 666f 6c6c 6f77 696e 6720 6f72  the following or
-0000ffb0: 6469 6e61 7469 6f6e 2066 6f72 6d61 7420  dination format 
-0000ffc0: 2822 6f72 6469 6e61 7469 6f6e 2229 2072  ("ordination") r
-0000ffd0: 6561 6465 7273 2f77 7269 7465 7273 3a0a  eaders/writers:.
-0000ffe0: 2020 2020 2020 2020 2d20 6060 4f72 6469          - ``Ordi
-0000fff0: 6e61 7469 6f6e 5265 7375 6c74 732e 6672  nationResults.fr
-00010000: 6f6d 5f66 696c 6560 6020 2d3e 2060 604f  om_file`` -> ``O
-00010010: 7264 696e 6174 696f 6e52 6573 756c 7473  rdinationResults
-00010020: 2e72 6561 6460 600a 2020 2020 2020 2020  .read``.        
-00010030: 2d20 6060 4f72 6469 6e61 7469 6f6e 5265  - ``OrdinationRe
-00010040: 7375 6c74 732e 746f 5f66 696c 6560 6020  sults.to_file`` 
-00010050: 2d3e 2060 604f 7264 696e 6174 696f 6e52  -> ``OrdinationR
-00010060: 6573 756c 7473 2e77 7269 7465 6060 0a0a  esults.write``..
-00010070: 2020 2020 2a20 4465 7072 6563 6174 6564      * Deprecated
-00010080: 2074 6865 2066 6f6c 6c6f 7769 6e67 2022   the following "
-00010090: 6e65 7769 636b 2220 7265 6164 6572 732f  newick" readers/
-000100a0: 7772 6974 6572 733a 0a20 2020 2020 2020  writers:.       
-000100b0: 202d 2060 6054 7265 654e 6f64 652e 6672   - ``TreeNode.fr
-000100c0: 6f6d 5f66 696c 6560 6020 2d3e 2060 6054  om_file`` -> ``T
-000100d0: 7265 654e 6f64 652e 7265 6164 6060 0a20  reeNode.read``. 
-000100e0: 2020 2020 2020 202d 2060 6054 7265 654e         - ``TreeN
-000100f0: 6f64 652e 6672 6f6d 5f6e 6577 6963 6b60  ode.from_newick`
-00010100: 6020 2d3e 2060 6054 7265 654e 6f64 652e  ` -> ``TreeNode.
-00010110: 7265 6164 6060 0a20 2020 2020 2020 202d  read``.        -
-00010120: 2060 6054 7265 654e 6f64 652e 746f 5f6e   ``TreeNode.to_n
-00010130: 6577 6963 6b60 6020 2d3e 2060 6054 7265  ewick`` -> ``Tre
-00010140: 654e 6f64 652e 7772 6974 6560 600a 0a20  eNode.write``.. 
-00010150: 2020 202a 2044 6570 7265 6361 7465 6420     * Deprecated 
-00010160: 7468 6520 666f 6c6c 6f77 696e 6720 2270  the following "p
-00010170: 6879 6c69 7022 2077 7269 7465 7273 3a0a  hylip" writers:.
-00010180: 2020 2020 2020 2020 2d20 6060 416c 6967          - ``Alig
-00010190: 6e6d 656e 742e 746f 5f70 6879 6c69 7060  nment.to_phylip`
-000101a0: 6020 2d3e 2060 6041 6c69 676e 6d65 6e74  ` -> ``Alignment
-000101b0: 2e77 7269 7465 6060 0a0a 2020 2020 2a20  .write``..    * 
-000101c0: 4465 7072 6563 6174 6564 2074 6865 2066  Deprecated the f
-000101d0: 6f6c 6c6f 7769 6e67 2022 6661 7374 6122  ollowing "fasta"
-000101e0: 2f51 5541 4c20 7265 6164 6572 732f 7772  /QUAL readers/wr
-000101f0: 6974 6572 733a 0a20 2020 2020 2020 202d  iters:.        -
-00010200: 2060 6053 6571 7565 6e63 6543 6f6c 6c65   ``SequenceColle
-00010210: 6374 696f 6e2e 6672 6f6d 5f66 6173 7461  ction.from_fasta
-00010220: 5f72 6563 6f72 6473 6060 202d 3e20 6060  _records`` -> ``
-00010230: 5365 7175 656e 6365 436f 6c6c 6563 7469  SequenceCollecti
-00010240: 6f6e 2e72 6561 6460 600a 2020 2020 2020  on.read``.      
-00010250: 2020 2d20 6060 5365 7175 656e 6365 436f    - ``SequenceCo
-00010260: 6c6c 6563 7469 6f6e 2e74 6f5f 6661 7374  llection.to_fast
-00010270: 6160 6020 2d3e 2060 6053 6571 7565 6e63  a`` -> ``Sequenc
-00010280: 6543 6f6c 6c65 6374 696f 6e2e 7772 6974  eCollection.writ
-00010290: 6560 600a 2020 2020 2020 2020 2d20 6060  e``.        - ``
-000102a0: 6661 7374 615f 6672 6f6d 5f73 6571 7565  fasta_from_seque
-000102b0: 6e63 6573 6060 202d 3e20 6060 736b 6269  nces`` -> ``skbi
-000102c0: 6f2e 696f 2e77 7269 7465 286f 626a 2c20  o.io.write(obj, 
-000102d0: 696e 746f 3d3c 6669 6c65 3e2c 2066 6f72  into=<file>, for
-000102e0: 6d61 743d 2766 6173 7461 2729 6060 0a20  mat='fasta')``. 
-000102f0: 2020 2020 2020 202d 2060 6066 6173 7461         - ``fasta
-00010300: 5f66 726f 6d5f 616c 6967 6e6d 656e 7460  _from_alignment`
-00010310: 6020 2d3e 2060 6041 6c69 676e 6d65 6e74  ` -> ``Alignment
-00010320: 2e77 7269 7465 6060 0a20 2020 2020 2020  .write``.       
-00010330: 202d 2060 6070 6172 7365 5f66 6173 7461   - ``parse_fasta
-00010340: 6060 202d 3e20 6060 736b 6269 6f2e 696f  `` -> ``skbio.io
-00010350: 2e72 6561 6428 3c66 6173 7461 3e2c 2066  .read(<fasta>, f
-00010360: 6f72 6d61 743d 2766 6173 7461 2729 6060  ormat='fasta')``
-00010370: 0a20 2020 2020 2020 202d 2060 6070 6172  .        - ``par
-00010380: 7365 5f71 7561 6c60 6020 2d3e 2060 6073  se_qual`` -> ``s
-00010390: 6b62 696f 2e69 6f2e 7265 6164 283c 6661  kbio.io.read(<fa
-000103a0: 7374 613e 2c20 666f 726d 6174 3d27 6661  sta>, format='fa
-000103b0: 7374 6127 2c20 7175 616c 3d3c 6669 6c65  sta', qual=<file
-000103c0: 3e29 6060 0a20 2020 2020 2020 202d 2060  >)``.        - `
-000103d0: 6042 696f 6c6f 6769 6361 6c53 6571 7565  `BiologicalSeque
-000103e0: 6e63 652e 746f 5f66 6173 7461 6060 202d  nce.to_fasta`` -
-000103f0: 3e20 6060 4269 6f6c 6f67 6963 616c 5365  > ``BiologicalSe
-00010400: 7175 656e 6365 2e77 7269 7465 6060 0a0a  quence.write``..
-00010410: 2020 2020 2a20 4465 7072 6563 6174 6564      * Deprecated
-00010420: 2074 6865 2066 6f6c 6c6f 7769 6e67 2022   the following "
-00010430: 6661 7374 7122 2072 6561 6465 7273 2f77  fastq" readers/w
-00010440: 7269 7465 7273 3a0a 2020 2020 2020 2020  riters:.        
-00010450: 2d20 6060 7061 7273 655f 6661 7374 7160  - ``parse_fastq`
-00010460: 6020 2d3e 2060 6073 6b62 696f 2e69 6f2e  ` -> ``skbio.io.
-00010470: 7265 6164 283c 6661 7374 713e 2c20 666f  read(<fastq>, fo
-00010480: 726d 6174 3d27 6661 7374 7127 2960 600a  rmat='fastq')``.
-00010490: 2020 2020 2020 2020 2d20 6060 666f 726d          - ``form
-000104a0: 6174 5f66 6173 7471 5f72 6563 6f72 6460  at_fastq_record`
-000104b0: 6020 2d3e 2060 6073 6b62 696f 2e69 6f2e  ` -> ``skbio.io.
-000104c0: 7772 6974 6528 3c66 6173 7471 3e2c 2066  write(<fastq>, f
-000104d0: 6f72 6d61 743d 2766 6173 7471 2729 6060  ormat='fastq')``
-000104e0: 0a0a 2323 2320 4261 636b 7761 7264 2d69  ..### Backward-i
-000104f0: 6e63 6f6d 7061 7469 626c 6520 6368 616e  ncompatible chan
-00010500: 6765 730a 0a2a 2060 6073 6b62 696f 2e73  ges..* ``skbio.s
-00010510: 7461 7473 2e64 6973 7461 6e63 652e 6d61  tats.distance.ma
-00010520: 6e74 656c 6060 206e 6f77 2072 6574 7572  ntel`` now retur
-00010530: 6e73 2061 2033 2d65 6c65 6d65 6e74 2074  ns a 3-element t
-00010540: 7570 6c65 2063 6f6e 7461 696e 696e 6720  uple containing 
-00010550: 636f 7272 656c 6174 696f 6e20 636f 6566  correlation coef
-00010560: 6669 6369 656e 742c 2070 2d76 616c 7565  ficient, p-value
-00010570: 2c20 616e 6420 7468 6520 6e75 6d62 6572  , and the number
-00010580: 206f 6620 6d61 7463 6869 6e67 2072 6f77   of matching row
-00010590: 732f 636f 6c73 2069 6e20 7468 6520 6469  s/cols in the di
-000105a0: 7374 616e 6365 206d 6174 7269 6365 7320  stance matrices 
-000105b0: 2860 606e 6060 292e 2054 6865 2072 6574  (``n``). The ret
-000105c0: 7572 6e20 7661 6c75 6520 7761 7320 7072  urn value was pr
-000105d0: 6576 696f 7573 6c79 2061 2032 2d65 6c65  eviously a 2-ele
-000105e0: 6d65 6e74 2074 7570 6c65 2063 6f6e 7461  ment tuple conta
-000105f0: 696e 696e 6720 6f6e 6c79 2074 6865 2063  ining only the c
-00010600: 6f72 7265 6c61 7469 6f6e 2063 6f65 6666  orrelation coeff
-00010610: 6963 6965 6e74 2061 6e64 2070 2d76 616c  icient and p-val
-00010620: 7565 2e0a 2a20 6060 736b 6269 6f2e 7374  ue..* ``skbio.st
-00010630: 6174 732e 6469 7374 616e 6365 2e6d 616e  ats.distance.man
-00010640: 7465 6c60 6020 7265 6f72 6465 7273 2069  tel`` reorders i
-00010650: 6e70 7574 2060 6044 6973 7461 6e63 654d  nput ``DistanceM
-00010660: 6174 7269 7860 6020 696e 7374 616e 6365  atrix`` instance
-00010670: 7320 6261 7365 6420 6f6e 206d 6174 6368  s based on match
-00010680: 696e 6720 4944 7320 2873 6565 206f 7074  ing IDs (see opt
-00010690: 696f 6e61 6c20 7061 7261 6d65 7465 7273  ional parameters
-000106a0: 2060 6073 7472 6963 7460 6020 616e 6420   ``strict`` and 
-000106b0: 6060 6c6f 6f6b 7570 6060 2066 6f72 2063  ``lookup`` for c
-000106c0: 6f6e 7472 6f6c 6c69 6e67 2074 6869 7320  ontrolling this 
-000106d0: 6265 6861 7669 6f72 292e 2049 6e20 7468  behavior). In th
-000106e0: 6520 7061 7374 2c20 6060 4469 7374 616e  e past, ``Distan
-000106f0: 6365 4d61 7472 6978 6060 2069 6e73 7461  ceMatrix`` insta
-00010700: 6e63 6573 2077 6572 6520 7472 6561 7465  nces were treate
-00010710: 6420 7468 6520 7361 6d65 2061 7320 6060  d the same as ``
-00010720: 6172 7261 795f 6c69 6b65 6060 2069 6e70  array_like`` inp
-00010730: 7574 2061 6e64 206e 6f20 7265 6f72 6465  ut and no reorde
-00010740: 7269 6e67 2074 6f6f 6b20 706c 6163 652c  ring took place,
-00010750: 2072 6567 6172 646c 6573 7320 6f66 2049   regardless of I
-00010760: 4420 286d 6973 296d 6174 6368 6573 2e20  D (mis)matches. 
-00010770: 6060 6172 7261 795f 6c69 6b65 6060 2069  ``array_like`` i
-00010780: 6e70 7574 2062 6568 6176 696f 7220 7265  nput behavior re
-00010790: 6d61 696e 7320 7468 6520 7361 6d65 2e0a  mains the same..
-000107a0: 2a20 4966 206d 6973 6d61 7463 6865 6420  * If mismatched 
-000107b0: 7479 7065 7320 6172 6520 7072 6f76 6964  types are provid
-000107c0: 6564 2074 6f20 6060 736b 6269 6f2e 7374  ed to ``skbio.st
-000107d0: 6174 732e 6469 7374 616e 6365 2e6d 616e  ats.distance.man
-000107e0: 7465 6c60 6020 2865 2e67 2e2c 2061 2060  tel`` (e.g., a `
-000107f0: 6044 6973 7461 6e63 654d 6174 7269 7860  `DistanceMatrix`
-00010800: 6020 616e 6420 6060 6172 7261 795f 6c69  ` and ``array_li
-00010810: 6b65 6060 292c 2061 2060 6054 7970 6545  ke``), a ``TypeE
-00010820: 7272 6f72 6060 2077 696c 6c20 6265 2072  rror`` will be r
-00010830: 6169 7365 642e 0a0a 2323 2320 4d69 7363  aised...### Misc
-00010840: 656c 6c61 6e65 6f75 730a 0a2a 2041 6464  ellaneous..* Add
-00010850: 6564 2067 6974 2074 696d 6573 7461 6d70  ed git timestamp
-00010860: 2063 6865 636b 696e 6720 746f 2063 6865   checking to che
-00010870: 636b 6c69 7374 2e70 792c 2065 6e73 7572  cklist.py, ensur
-00010880: 696e 6720 7468 6174 2077 6865 6e20 6368  ing that when ch
-00010890: 616e 6765 7320 6172 6520 6d61 6465 2074  anges are made t
-000108a0: 6f20 4379 7468 6f6e 2028 2e70 7978 2920  o Cython (.pyx) 
-000108b0: 6669 6c65 732c 2074 6865 6972 2063 6f72  files, their cor
-000108c0: 7265 7370 6f6e 6469 6e67 2067 656e 6572  responding gener
-000108d0: 6174 6564 2043 2066 696c 6573 2061 7265  ated C files are
-000108e0: 2061 6c73 6f20 7570 6461 7465 642e 0a2a   also updated..*
-000108f0: 2046 6978 6564 2070 6572 666f 726d 616e   Fixed performan
-00010900: 6365 2062 7567 2077 6865 6e20 696e 7374  ce bug when inst
-00010910: 616e 7469 6174 696e 6720 6060 4269 6f6c  antiating ``Biol
-00010920: 6f67 6963 616c 5365 7175 656e 6365 6060  ogicalSequence``
-00010930: 206f 626a 6563 7473 2e20 5468 6520 7072   objects. The pr
-00010940: 6576 696f 7573 2072 756e 7469 6d65 2073  evious runtime s
-00010950: 6361 6c65 6420 6c69 6e65 6172 6c79 2077  caled linearly w
-00010960: 6974 6820 7365 7175 656e 6365 206c 656e  ith sequence len
-00010970: 6774 683b 2069 7420 6973 206e 6f77 2063  gth; it is now c
-00010980: 6f6e 7374 616e 7420 7469 6d65 2077 6865  onstant time whe
-00010990: 6e20 7468 6520 7365 7175 656e 6365 2069  n the sequence i
-000109a0: 7320 616c 7265 6164 7920 6120 7374 7269  s already a stri
-000109b0: 6e67 2e20 5365 6520 5b23 3632 335d 2868  ng. See [#623](h
-000109c0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
-000109d0: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
-000109e0: 2d62 696f 2f69 7373 7565 732f 3632 3329  -bio/issues/623)
-000109f0: 2066 6f72 2064 6574 6169 6c73 2e0a 2a20   for details..* 
-00010a00: 4950 7974 686f 6e20 616e 6420 7369 7820  IPython and six 
-00010a10: 6172 6520 6e6f 7720 7265 7175 6972 6564  are now required
-00010a20: 2064 6570 656e 6465 6e63 6965 732e 0a0a   dependencies...
-00010a30: 2323 2056 6572 7369 6f6e 2030 2e32 2e30  ## Version 0.2.0
-00010a40: 2028 3230 3134 2d30 382d 3037 290a 0a54   (2014-08-07)..T
-00010a50: 6869 7320 6973 2061 6e20 696e 6974 6961  his is an initia
-00010a60: 6c20 616c 7068 6120 7265 6c65 6173 6520  l alpha release 
-00010a70: 6f66 2073 6369 6b69 742d 6269 6f2e 2041  of scikit-bio. A
-00010a80: 7420 7468 6973 2073 7461 6765 2c20 6d61  t this stage, ma
-00010a90: 6a6f 7220 6261 636b 7761 7264 732d 696e  jor backwards-in
-00010aa0: 636f 6d70 6174 6962 6c65 2041 5049 2063  compatible API c
-00010ab0: 6861 6e67 6573 2063 616e 2061 6e64 2077  hanges can and w
-00010ac0: 696c 6c20 6861 7070 656e 2e20 4d61 6e79  ill happen. Many
-00010ad0: 2062 6163 6b77 6172 6473 2d69 6e63 6f6d   backwards-incom
-00010ae0: 7061 7469 626c 6520 4150 4920 6368 616e  patible API chan
-00010af0: 6765 7320 7765 7265 206d 6164 6520 7369  ges were made si
-00010b00: 6e63 6520 7468 6520 7072 6576 696f 7573  nce the previous
-00010b10: 2072 656c 6561 7365 2e0a 0a23 2323 2046   release...### F
-00010b20: 6561 7475 7265 730a 0a2a 2041 6464 6564  eatures..* Added
-00010b30: 2061 6269 6c69 7479 2074 6f20 636f 6d70   ability to comp
-00010b40: 7574 6520 6469 7374 616e 6365 7320 6265  ute distances be
-00010b50: 7477 6565 6e20 7365 7175 656e 6365 7320  tween sequences 
-00010b60: 696e 2061 2060 6053 6571 7565 6e63 6543  in a ``SequenceC
-00010b70: 6f6c 6c65 6374 696f 6e60 6020 6f62 6a65  ollection`` obje
-00010b80: 6374 2028 5b23 3530 395d 2868 7474 7073  ct ([#509](https
-00010b90: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
-00010ba0: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
-00010bb0: 2f69 7373 7565 732f 3530 3929 292c 2061  /issues/509)), a
-00010bc0: 6e64 2065 7870 616e 6465 6420 6060 416c  nd expanded ``Al
-00010bd0: 6967 6e6d 656e 742e 6469 7374 616e 6365  ignment.distance
-00010be0: 6060 2074 6f20 616c 6c6f 7720 7468 6520  `` to allow the 
-00010bf0: 7573 6572 2074 6f20 7061 7373 2061 2066  user to pass a f
-00010c00: 756e 6374 696f 6e20 666f 7220 636f 6d70  unction for comp
-00010c10: 7574 696e 6720 6469 7374 616e 6365 7320  uting distances 
-00010c20: 2874 6865 2064 6566 6175 6c74 2064 6973  (the default dis
-00010c30: 7461 6e63 6520 6d65 7472 6963 2069 7320  tance metric is 
-00010c40: 7374 696c 6c20 6060 7363 6970 792e 7370  still ``scipy.sp
-00010c50: 6174 6961 6c2e 6469 7374 616e 6365 2e68  atial.distance.h
-00010c60: 616d 6d69 6e67 6060 2920 285b 2331 3934  amming``) ([#194
-00010c70: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
-00010c80: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
-00010c90: 6b69 742d 6269 6f2f 6973 7375 6573 2f31  kit-bio/issues/1
-00010ca0: 3934 2929 2e0a 2a20 4164 6465 6420 6675  94))..* Added fu
-00010cb0: 6e63 7469 6f6e 616c 6974 7920 746f 206e  nctionality to n
-00010cc0: 6f74 2070 656e 616c 697a 6520 7465 726d  ot penalize term
-00010cd0: 696e 616c 2067 6170 7320 696e 2067 6c6f  inal gaps in glo
-00010ce0: 6261 6c20 616c 6967 6e6d 656e 742e 2054  bal alignment. T
-00010cf0: 6869 7320 6675 6e63 7469 6f6e 616c 6974  his functionalit
-00010d00: 7920 7265 7375 6c74 7320 696e 206d 6f72  y results in mor
-00010d10: 6520 6269 6f6c 6f67 6963 616c 6c79 2072  e biologically r
-00010d20: 656c 6576 616e 7420 676c 6f62 616c 2061  elevant global a
-00010d30: 6c69 676e 6d65 6e74 7320 2873 6565 205b  lignments (see [
-00010d40: 2335 3337 5d28 6874 7470 733a 2f2f 6769  #537](https://gi
-00010d50: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
-00010d60: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
-00010d70: 6573 2f35 3337 2920 666f 7220 6469 7363  es/537) for disc
-00010d80: 7573 7369 6f6e 206f 6620 7468 6520 6973  ussion of the is
-00010d90: 7375 6529 2061 6e64 2069 7320 6e6f 7720  sue) and is now 
-00010da0: 7468 6520 6465 6661 756c 7420 6265 6861  the default beha
-00010db0: 7669 6f72 2066 6f72 2067 6c6f 6261 6c20  vior for global 
-00010dc0: 616c 6967 6e6d 656e 742e 0a2a 2054 6865  alignment..* The
-00010dd0: 2070 7974 686f 6e20 676c 6f62 616c 2061   python global a
-00010de0: 6c69 676e 6572 7320 2860 6067 6c6f 6261  ligners (``globa
-00010df0: 6c5f 7061 6972 7769 7365 5f61 6c69 676e  l_pairwise_align
-00010e00: 6060 2c20 6060 676c 6f62 616c 5f70 6169  ``, ``global_pai
-00010e10: 7277 6973 655f 616c 6967 6e5f 6e75 636c  rwise_align_nucl
-00010e20: 656f 7469 6465 6060 2c20 616e 6420 6060  eotide``, and ``
-00010e30: 676c 6f62 616c 5f70 6169 7277 6973 655f  global_pairwise_
-00010e40: 616c 6967 6e5f 7072 6f74 6569 6e60 6029  align_protein``)
-00010e50: 206e 6f77 2073 7570 706f 7274 2061 6c69   now support ali
-00010e60: 676e 696e 6720 7061 6972 7320 6f66 2073  gning pairs of s
-00010e70: 6571 7565 6e63 6573 2c20 7061 6972 7320  equences, pairs 
-00010e80: 6f66 2061 6c69 676e 6d65 6e74 732c 2061  of alignments, a
-00010e90: 6e64 2061 2073 6571 7565 6e63 6520 616e  nd a sequence an
-00010ea0: 6420 616e 2061 6c69 676e 6d65 6e74 2028  d an alignment (
-00010eb0: 7365 6520 5b23 3535 305d 2868 7474 7073  see [#550](https
-00010ec0: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
-00010ed0: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
-00010ee0: 2f69 7373 7565 732f 3535 3029 292e 2054  /issues/550)). T
-00010ef0: 6869 7320 6675 6e63 7469 6f6e 616c 6974  his functionalit
-00010f00: 7920 7375 7070 6f72 7473 2070 726f 6772  y supports progr
-00010f10: 6573 7369 7665 206d 756c 7469 706c 6520  essive multiple 
-00010f20: 7365 7175 656e 6365 2061 6c69 676e 6d65  sequence alignme
-00010f30: 6e74 2c20 616d 6f6e 6720 6f74 6865 7220  nt, among other 
-00010f40: 7468 696e 6773 2073 7563 6820 6173 2061  things such as a
-00010f50: 6464 696e 6720 6120 7365 7175 656e 6365  dding a sequence
-00010f60: 2074 6f20 616e 2065 7869 7374 696e 6720   to an existing 
-00010f70: 616c 6967 6e6d 656e 742e 0a2a 2041 6464  alignment..* Add
-00010f80: 6564 2060 6053 746f 636b 686f 6c6d 416c  ed ``StockholmAl
-00010f90: 6967 6e6d 656e 742e 746f 5f66 696c 6560  ignment.to_file`
-00010fa0: 6020 666f 7220 7772 6974 696e 6720 5374  ` for writing St
-00010fb0: 6f63 6b68 6f6c 6d2d 666f 726d 6174 7465  ockholm-formatte
-00010fc0: 6420 6669 6c65 732e 0a2a 2041 6464 6564  d files..* Added
-00010fd0: 2060 6073 7472 6963 743d 5472 7565 6060   ``strict=True``
-00010fe0: 206f 7074 696f 6e61 6c20 7061 7261 6d65   optional parame
-00010ff0: 7465 7220 746f 2060 6044 6973 7369 6d69  ter to ``Dissimi
-00011000: 6c61 7269 7479 4d61 7472 6978 2e66 696c  larityMatrix.fil
-00011010: 7465 7260 602e 0a2a 2041 6464 6564 2060  ter``..* Added `
-00011020: 6054 7265 654e 6f64 652e 6669 6e64 5f61  `TreeNode.find_a
-00011030: 6c6c 6060 2066 6f72 2066 696e 6469 6e67  ll`` for finding
-00011040: 2061 6c6c 2074 7265 6520 6e6f 6465 7320   all tree nodes 
-00011050: 7468 6174 206d 6174 6368 2061 2067 6976  that match a giv
-00011060: 656e 206e 616d 652e 0a0a 0a23 2323 2042  en name....### B
-00011070: 7567 2066 6978 6573 0a0a 2a20 4669 7865  ug fixes..* Fixe
-00011080: 6420 6275 6720 7468 6174 2072 6573 756c  d bug that resul
-00011090: 7465 6420 696e 2061 2060 6056 616c 7565  ted in a ``Value
-000110a0: 4572 726f 7260 6020 6672 6f6d 2060 606c  Error`` from ``l
-000110b0: 6f63 616c 5f61 6c69 676e 5f70 6169 7277  ocal_align_pairw
-000110c0: 6973 655f 6e75 636c 656f 7469 6465 6060  ise_nucleotide``
-000110d0: 2028 7365 6520 5b23 3530 345d 2868 7474   (see [#504](htt
-000110e0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
-000110f0: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
-00011100: 696f 2f69 7373 7565 732f 3530 3429 2920  io/issues/504)) 
-00011110: 756e 6465 7220 6d61 6e79 2063 6972 6375  under many circu
-00011120: 6d73 7461 6e63 6573 2e20 5468 6973 2077  mstances. This w
-00011130: 6f75 6c64 206e 6f74 2067 656e 6572 6174  ould not generat
-00011140: 6520 696e 636f 7272 6563 7420 7265 7375  e incorrect resu
-00011150: 6c74 732c 2062 7574 2077 6f75 6c64 2063  lts, but would c
-00011160: 6175 7365 2074 6865 2063 6f64 6520 746f  ause the code to
-00011170: 2066 6169 6c2e 0a0a 2323 2320 4261 636b   fail...### Back
-00011180: 7761 7264 2d69 6e63 6f6d 7061 7469 626c  ward-incompatibl
-00011190: 6520 6368 616e 6765 730a 0a2a 2052 656d  e changes..* Rem
-000111a0: 6f76 6564 2060 6073 6b62 696f 2e6d 6174  oved ``skbio.mat
-000111b0: 6860 602c 206c 6561 7669 6e67 2060 6073  h``, leaving ``s
-000111c0: 7461 7473 6060 2061 6e64 2060 6064 6976  tats`` and ``div
-000111d0: 6572 7369 7479 6060 2074 6f20 6265 636f  ersity`` to beco
-000111e0: 6d65 2074 6f70 206c 6576 656c 2070 6163  me top level pac
-000111f0: 6b61 6765 732e 2046 6f72 2065 7861 6d70  kages. For examp
-00011200: 6c65 2c20 696e 7374 6561 6420 6f66 2060  le, instead of `
-00011210: 6066 726f 6d20 736b 6269 6f2e 6d61 7468  `from skbio.math
-00011220: 2e73 7461 7473 2e6f 7264 696e 6174 696f  .stats.ordinatio
-00011230: 6e20 696d 706f 7274 2050 436f 4160 6020  n import PCoA`` 
-00011240: 796f 7520 776f 756c 6420 6e6f 7720 696d  you would now im
-00011250: 706f 7274 2060 6066 726f 6d20 736b 6269  port ``from skbi
-00011260: 6f2e 7374 6174 732e 6f72 6469 6e61 7469  o.stats.ordinati
-00011270: 6f6e 2069 6d70 6f72 7420 5043 6f41 6060  on import PCoA``
-00011280: 2e0a 2a20 5468 6520 6d6f 6475 6c65 2060  ..* The module `
-00011290: 6073 6b62 696f 2e6d 6174 682e 6772 6164  `skbio.math.grad
-000112a0: 6965 6e74 6060 2061 7320 7765 6c6c 2061  ient`` as well a
-000112b0: 7320 7468 6520 636f 6e74 656e 7473 206f  s the contents o
-000112c0: 6620 6060 736b 6269 6f2e 6d61 7468 2e73  f ``skbio.math.s
-000112d0: 7562 7361 6d70 6c65 6060 2061 6e64 2060  ubsample`` and `
-000112e0: 6073 6b62 696f 2e6d 6174 682e 7374 6174  `skbio.math.stat
-000112f0: 732e 6d69 7363 6060 2061 7265 206e 6f77  s.misc`` are now
-00011300: 2066 6f75 6e64 2069 6e20 6060 736b 6269   found in ``skbi
-00011310: 6f2e 7374 6174 7360 602e 2041 7320 616e  o.stats``. As an
-00011320: 2065 7861 6d70 6c65 2c20 746f 2069 6d70   example, to imp
-00011330: 6f72 7420 7375 6273 616d 706c 653a 2060  ort subsample: `
-00011340: 6066 726f 6d20 736b 6269 6f2e 7374 6174  `from skbio.stat
-00011350: 7320 696d 706f 7274 2073 7562 7361 6d70  s import subsamp
-00011360: 6c65 6060 3b20 746f 2069 6d70 6f72 7420  le``; to import 
-00011370: 6576 6572 7974 6869 6e67 2066 726f 6d20  everything from 
-00011380: 6772 6164 6965 6e74 3a20 6060 6672 6f6d  gradient: ``from
-00011390: 2073 6b62 696f 2e73 7461 7473 2e67 7261   skbio.stats.gra
-000113a0: 6469 656e 7420 696d 706f 7274 202a 6060  dient import *``
-000113b0: 2e0a 2a20 5468 6520 636f 6e74 656e 7473  ..* The contents
-000113c0: 206f 6620 6060 736b 6269 6f2e 6d61 7468   of ``skbio.math
-000113d0: 2e73 7461 7473 2e6f 7264 696e 6174 696f  .stats.ordinatio
-000113e0: 6e2e 7574 696c 7360 6020 6172 6520 6e6f  n.utils`` are no
-000113f0: 7720 696e 2060 6073 6b62 696f 2e73 7461  w in ``skbio.sta
-00011400: 7473 2e6f 7264 696e 6174 696f 6e60 602e  ts.ordination``.
-00011410: 0a2a 2052 656d 6f76 6564 2060 6073 6b62  .* Removed ``skb
-00011420: 696f 2e61 7070 6060 2073 7562 7061 636b  io.app`` subpack
-00011430: 6167 6520 2869 2e65 2e2c 2074 6865 202a  age (i.e., the *
-00011440: 6170 706c 6963 6174 696f 6e20 636f 6e74  application cont
-00011450: 726f 6c6c 6572 2066 7261 6d65 776f 726b  roller framework
-00011460: 2a29 2061 7320 7468 6973 2063 6f64 6520  *) as this code 
-00011470: 6861 7320 6265 656e 2070 6f72 7465 6420  has been ported 
-00011480: 746f 2074 6865 2073 7461 6e64 616c 6f6e  to the standalon
-00011490: 6520 5b62 7572 7269 746f 5d28 6874 7470  e [burrito](http
-000114a0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-000114b0: 696f 636f 7265 2f62 7572 7269 746f 2920  iocore/burrito) 
-000114c0: 5079 7468 6f6e 2070 6163 6b61 6765 2e20  Python package. 
-000114d0: 5468 6973 2063 6f64 6520 7761 7320 6e6f  This code was no
-000114e0: 7420 7370 6563 6966 6963 2074 6f20 6269  t specific to bi
-000114f0: 6f69 6e66 6f72 6d61 7469 6373 2061 6e64  oinformatics and
-00011500: 2069 7320 7573 6566 756c 2066 6f72 2077   is useful for w
-00011510: 7261 7070 696e 6720 636f 6d6d 616e 642d  rapping command-
-00011520: 6c69 6e65 2061 7070 6c69 6361 7469 6f6e  line application
-00011530: 7320 696e 2067 656e 6572 616c 2e0a 2a20  s in general..* 
-00011540: 5265 6d6f 7665 6420 6060 736b 6269 6f2e  Removed ``skbio.
-00011550: 636f 7265 6060 2c20 6c65 6176 696e 6720  core``, leaving 
-00011560: 6060 616c 6967 6e6d 656e 7460 602c 2060  ``alignment``, `
-00011570: 6067 656e 6574 6963 5f63 6f64 6560 602c  `genetic_code``,
-00011580: 2060 6073 6571 7565 6e63 6560 602c 2060   ``sequence``, `
-00011590: 6074 7265 6560 602c 2061 6e64 2060 6077  `tree``, and ``w
-000115a0: 6f72 6b66 6c6f 7760 6020 746f 2062 6563  orkflow`` to bec
-000115b0: 6f6d 6520 746f 7020 6c65 7665 6c20 7061  ome top level pa
-000115c0: 636b 6167 6573 2e20 466f 7220 6578 616d  ckages. For exam
-000115d0: 706c 652c 2069 6e73 7465 6164 206f 6620  ple, instead of 
-000115e0: 6060 6672 6f6d 2073 6b62 696f 2e63 6f72  ``from skbio.cor
-000115f0: 652e 7365 7175 656e 6365 2069 6d70 6f72  e.sequence impor
-00011600: 7420 444e 4160 6020 796f 7520 776f 756c  t DNA`` you woul
-00011610: 6420 6e6f 7720 696d 706f 7274 2060 6066  d now import ``f
-00011620: 726f 6d20 736b 6269 6f2e 7365 7175 656e  rom skbio.sequen
-00011630: 6365 2069 6d70 6f72 7420 444e 4160 602e  ce import DNA``.
-00011640: 0a2a 2052 656d 6f76 6564 2060 6073 6b62  .* Removed ``skb
-00011650: 696f 2e75 7469 6c2e 6578 6365 7074 696f  io.util.exceptio
-00011660: 6e60 6020 616e 6420 6060 736b 6269 6f2e  n`` and ``skbio.
-00011670: 7574 696c 2e77 6172 6e69 6e67 6060 2028  util.warning`` (
-00011680: 7365 6520 5b23 3537 375d 2868 7474 7073  see [#577](https
-00011690: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
-000116a0: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
-000116b0: 2f69 7373 7565 732f 3537 3729 2066 6f72  /issues/577) for
-000116c0: 2074 6865 2072 6561 736f 6e69 6e67 2062   the reasoning b
-000116d0: 6568 696e 6420 7468 6973 2063 6861 6e67  ehind this chang
-000116e0: 6529 2e20 5468 6520 6578 6365 7074 696f  e). The exceptio
-000116f0: 6e73 2f77 6172 6e69 6e67 7320 7765 7265  ns/warnings were
-00011700: 206d 6f76 6564 2074 6f20 7468 6520 666f   moved to the fo
-00011710: 6c6c 6f77 696e 6720 6c6f 6361 7469 6f6e  llowing location
-00011720: 733a 0a20 2d20 6060 4669 6c65 466f 726d  s:. - ``FileForm
-00011730: 6174 4572 726f 7260 602c 2060 6052 6563  atError``, ``Rec
-00011740: 6f72 6445 7272 6f72 6060 2c20 6060 4669  ordError``, ``Fi
-00011750: 656c 6445 7272 6f72 6060 2c20 616e 6420  eldError``, and 
-00011760: 6060 4566 6669 6369 656e 6379 5761 726e  ``EfficiencyWarn
-00011770: 696e 6760 6020 6861 7665 2062 6565 6e20  ing`` have been 
-00011780: 6d6f 7665 6420 746f 2060 6073 6b62 696f  moved to ``skbio
-00011790: 2e75 7469 6c60 600a 202d 2060 6042 696f  .util``. - ``Bio
-000117a0: 6c6f 6769 6361 6c53 6571 7565 6e63 6545  logicalSequenceE
-000117b0: 7272 6f72 6060 2068 6173 2062 6565 6e20  rror`` has been 
-000117c0: 6d6f 7665 6420 746f 2060 6073 6b62 696f  moved to ``skbio
-000117d0: 2e73 6571 7565 6e63 6560 600a 202d 2060  .sequence``. - `
-000117e0: 6053 6571 7565 6e63 6543 6f6c 6c65 6374  `SequenceCollect
-000117f0: 696f 6e45 7272 6f72 6060 2061 6e64 2060  ionError`` and `
-00011800: 6053 746f 636b 686f 6c6d 5061 7273 6545  `StockholmParseE
-00011810: 7272 6f72 6060 2068 6176 6520 6265 656e  rror`` have been
-00011820: 206d 6f76 6564 2074 6f20 6060 736b 6269   moved to ``skbi
-00011830: 6f2e 616c 6967 6e6d 656e 7460 600a 202d  o.alignment``. -
-00011840: 2060 6044 6973 7369 6d69 6c61 7269 7479   ``Dissimilarity
-00011850: 4d61 7472 6978 4572 726f 7260 602c 2060  MatrixError``, `
-00011860: 6044 6973 7461 6e63 654d 6174 7269 7845  `DistanceMatrixE
-00011870: 7272 6f72 6060 2c20 6060 4469 7373 696d  rror``, ``Dissim
-00011880: 696c 6172 6974 794d 6174 7269 7846 6f72  ilarityMatrixFor
-00011890: 6d61 7445 7272 6f72 6060 2c20 616e 6420  matError``, and 
-000118a0: 6060 4d69 7373 696e 6749 4445 7272 6f72  ``MissingIDError
-000118b0: 6060 2068 6176 6520 6265 656e 206d 6f76  `` have been mov
-000118c0: 6564 2074 6f20 6060 736b 6269 6f2e 7374  ed to ``skbio.st
-000118d0: 6174 732e 6469 7374 616e 6365 6060 0a20  ats.distance``. 
-000118e0: 2d20 6060 5472 6565 4572 726f 7260 602c  - ``TreeError``,
-000118f0: 2060 604e 6f4c 656e 6774 6845 7272 6f72   ``NoLengthError
-00011900: 6060 2c20 6060 4475 706c 6963 6174 654e  ``, ``DuplicateN
-00011910: 6f64 6545 7272 6f72 6060 2c20 6060 4d69  odeError``, ``Mi
-00011920: 7373 696e 674e 6f64 6545 7272 6f72 6060  ssingNodeError``
-00011930: 2c20 616e 6420 6060 4e6f 5061 7265 6e74  , and ``NoParent
-00011940: 4572 726f 7260 6020 6861 7665 2062 6565  Error`` have bee
-00011950: 6e20 6d6f 7665 6420 746f 2060 6073 6b62  n moved to ``skb
-00011960: 696f 2e74 7265 6560 600a 202d 2060 6046  io.tree``. - ``F
-00011970: 6173 7471 5061 7273 6545 7272 6f72 6060  astqParseError``
-00011980: 2068 6173 2062 6565 6e20 6d6f 7665 6420   has been moved 
-00011990: 746f 2060 6073 6b62 696f 2e70 6172 7365  to ``skbio.parse
-000119a0: 2e73 6571 7565 6e63 6573 6060 0a20 2d20  .sequences``. - 
-000119b0: 6060 4765 6e65 7469 6343 6f64 6545 7272  ``GeneticCodeErr
-000119c0: 6f72 6060 2c20 6060 4765 6e65 7469 6343  or``, ``GeneticC
-000119d0: 6f64 6549 6e69 7445 7272 6f72 6060 2c20  odeInitError``, 
-000119e0: 616e 6420 6060 496e 7661 6c69 6443 6f64  and ``InvalidCod
-000119f0: 6f6e 4572 726f 7260 6020 6861 7665 2062  onError`` have b
-00011a00: 6565 6e20 6d6f 7665 6420 746f 2060 6073  een moved to ``s
-00011a10: 6b62 696f 2e67 656e 6574 6963 5f63 6f64  kbio.genetic_cod
-00011a20: 6560 600a 2a20 5468 6520 636f 6e74 656e  e``.* The conten
-00011a30: 7473 206f 6620 6060 736b 6269 6f2e 6765  ts of ``skbio.ge
-00011a40: 6e65 7469 635f 636f 6465 6060 2066 6f72  netic_code`` for
-00011a50: 6d65 726c 7920 6060 736b 6269 6f2e 636f  merly ``skbio.co
-00011a60: 7265 2e67 656e 6574 6963 5f63 6f64 6560  re.genetic_code`
-00011a70: 6020 6172 6520 6e6f 7720 696e 2060 6073  ` are now in ``s
-00011a80: 6b62 696f 2e73 6571 7565 6e63 6560 602e  kbio.sequence``.
-00011a90: 2054 6865 2060 6047 656e 6574 6963 436f   The ``GeneticCo
-00011aa0: 6465 7360 6020 6469 6374 696f 6e61 7279  des`` dictionary
-00011ab0: 2069 7320 6e6f 7720 6120 6675 6e63 7469   is now a functi
-00011ac0: 6f6e 2060 6067 656e 6574 6963 5f63 6f64  on ``genetic_cod
-00011ad0: 6560 602e 2054 6865 2066 756e 6374 696f  e``. The functio
-00011ae0: 6e61 6c69 7479 2069 7320 7468 6520 7361  nality is the sa
-00011af0: 6d65 2c20 6578 6365 7074 2074 6861 7420  me, except that 
-00011b00: 6265 6361 7573 6520 7468 6973 2069 7320  because this is 
-00011b10: 6e6f 7720 6120 6675 6e63 7469 6f6e 2072  now a function r
-00011b20: 6174 6865 7220 7468 616e 2061 2064 6963  ather than a dic
-00011b30: 742c 2072 6574 7269 6576 696e 6720 6120  t, retrieving a 
-00011b40: 6765 6e65 7469 6320 636f 6465 2069 7320  genetic code is 
-00011b50: 646f 6e65 2075 7369 6e67 2061 2066 756e  done using a fun
-00011b60: 6374 696f 6e20 6361 6c6c 2072 6174 6865  ction call rathe
-00011b70: 7220 7468 616e 2061 206c 6f6f 6b75 7020  r than a lookup 
-00011b80: 2873 6f2c 2066 6f72 2065 7861 6d70 6c65  (so, for example
-00011b90: 2c20 6060 4765 6e65 7469 6343 6f64 6573  , ``GeneticCodes
-00011ba0: 5b32 5d60 6020 6265 636f 6d65 7320 6060  [2]`` becomes ``
-00011bb0: 6765 6e65 7469 635f 636f 6465 2832 2960  genetic_code(2)`
-00011bc0: 602e 0a2a 204d 616e 7920 7375 626d 6f64  `..* Many submod
-00011bd0: 756c 6573 2068 6176 6520 6265 656e 206d  ules have been m
-00011be0: 6164 6520 7072 6976 6174 6520 7769 7468  ade private with
-00011bf0: 2074 6865 2069 6e74 656e 7469 6f6e 206f   the intention o
-00011c00: 6620 7369 6d70 6c69 6679 696e 6720 696d  f simplifying im
-00011c10: 706f 7274 7320 666f 7220 7573 6572 732e  ports for users.
-00011c20: 2053 6565 205b 2335 3632 5d28 6874 7470   See [#562](http
-00011c30: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
-00011c40: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
-00011c50: 6f2f 6973 7375 6573 2f35 3632 2920 666f  o/issues/562) fo
-00011c60: 7220 6469 7363 7573 7369 6f6e 206f 6620  r discussion of 
-00011c70: 7468 6973 2063 6861 6e67 652e 2054 6865  this change. The
-00011c80: 2066 6f6c 6c6f 7769 6e67 206c 6973 7420   following list 
-00011c90: 636f 6e74 6169 6e73 2074 6865 2070 7265  contains the pre
-00011ca0: 7669 6f75 7320 6d6f 6475 6c65 206e 616d  vious module nam
-00011cb0: 6520 616e 6420 7768 6572 6520 696d 706f  e and where impo
-00011cc0: 7274 7320 6672 6f6d 2074 6861 7420 6d6f  rts from that mo
-00011cd0: 6475 6c65 2073 686f 756c 6420 6e6f 7720  dule should now 
-00011ce0: 636f 6d65 2066 726f 6d2e 0a20 2d20 6060  come from.. - ``
-00011cf0: 736b 6269 6f2e 616c 6967 6e6d 656e 742e  skbio.alignment.
-00011d00: 7373 7760 6020 746f 2060 6073 6b62 696f  ssw`` to ``skbio
-00011d10: 2e61 6c69 676e 6d65 6e74 6060 0a20 2d20  .alignment``. - 
-00011d20: 6060 736b 6269 6f2e 616c 6967 6e6d 656e  ``skbio.alignmen
-00011d30: 742e 616c 6967 6e6d 656e 7460 6020 746f  t.alignment`` to
-00011d40: 2060 6073 6b62 696f 2e61 6c69 676e 6d65   ``skbio.alignme
-00011d50: 6e74 6060 0a20 2d20 6060 736b 6269 6f2e  nt``. - ``skbio.
-00011d60: 616c 6967 6e6d 656e 742e 7061 6972 7769  alignment.pairwi
-00011d70: 7365 6060 2074 6f20 6060 736b 6269 6f2e  se`` to ``skbio.
-00011d80: 616c 6967 6e6d 656e 7460 600a 202d 2060  alignment``. - `
-00011d90: 6073 6b62 696f 2e64 6976 6572 7369 7479  `skbio.diversity
-00011da0: 2e61 6c70 6861 2e62 6173 6560 6020 746f  .alpha.base`` to
-00011db0: 2060 6073 6b62 696f 2e64 6976 6572 7369   ``skbio.diversi
-00011dc0: 7479 2e61 6c70 6861 6060 0a20 2d20 6060  ty.alpha``. - ``
-00011dd0: 736b 6269 6f2e 6469 7665 7273 6974 792e  skbio.diversity.
-00011de0: 616c 7068 612e 6769 6e69 6060 2074 6f20  alpha.gini`` to 
-00011df0: 6060 736b 6269 6f2e 6469 7665 7273 6974  ``skbio.diversit
-00011e00: 792e 616c 7068 6160 600a 202d 2060 6073  y.alpha``. - ``s
-00011e10: 6b62 696f 2e64 6976 6572 7369 7479 2e61  kbio.diversity.a
-00011e20: 6c70 6861 2e6c 6c61 6473 6572 6060 2074  lpha.lladser`` t
-00011e30: 6f20 6060 736b 6269 6f2e 6469 7665 7273  o ``skbio.divers
-00011e40: 6974 792e 616c 7068 6160 600a 202d 2060  ity.alpha``. - `
-00011e50: 6073 6b62 696f 2e64 6976 6572 7369 7479  `skbio.diversity
-00011e60: 2e62 6574 612e 6261 7365 6060 2074 6f20  .beta.base`` to 
-00011e70: 6060 736b 6269 6f2e 6469 7665 7273 6974  ``skbio.diversit
-00011e80: 792e 6265 7461 6060 0a20 2d20 6060 736b  y.beta``. - ``sk
-00011e90: 6269 6f2e 6472 6177 2e64 6973 7472 6962  bio.draw.distrib
-00011ea0: 7574 696f 6e73 6060 2074 6f20 6060 736b  utions`` to ``sk
-00011eb0: 6269 6f2e 6472 6177 6060 0a20 2d20 6060  bio.draw``. - ``
-00011ec0: 736b 6269 6f2e 7374 6174 732e 6469 7374  skbio.stats.dist
-00011ed0: 616e 6365 2e61 6e6f 7369 6d60 6020 746f  ance.anosim`` to
-00011ee0: 2060 6073 6b62 696f 2e73 7461 7473 2e64   ``skbio.stats.d
-00011ef0: 6973 7461 6e63 6560 600a 202d 2060 6073  istance``. - ``s
-00011f00: 6b62 696f 2e73 7461 7473 2e64 6973 7461  kbio.stats.dista
-00011f10: 6e63 652e 6261 7365 6060 2074 6f20 6060  nce.base`` to ``
-00011f20: 736b 6269 6f2e 7374 6174 732e 6469 7374  skbio.stats.dist
-00011f30: 616e 6365 6060 0a20 2d20 6060 736b 6269  ance``. - ``skbi
-00011f40: 6f2e 7374 6174 732e 6469 7374 616e 6365  o.stats.distance
-00011f50: 2e70 6572 6d61 6e6f 7661 6060 2074 6f20  .permanova`` to 
-00011f60: 6060 736b 6269 6f2e 7374 6174 732e 6469  ``skbio.stats.di
-00011f70: 7374 616e 6365 6060 0a20 2d20 6060 736b  stance``. - ``sk
-00011f80: 6269 6f2e 6469 7374 616e 6365 6060 2074  bio.distance`` t
-00011f90: 6f20 6060 736b 6269 6f2e 7374 6174 732e  o ``skbio.stats.
-00011fa0: 6469 7374 616e 6365 6060 0a20 2d20 6060  distance``. - ``
-00011fb0: 736b 6269 6f2e 7374 6174 732e 6f72 6469  skbio.stats.ordi
-00011fc0: 6e61 7469 6f6e 2e62 6173 6560 6020 746f  nation.base`` to
-00011fd0: 2060 6073 6b62 696f 2e73 7461 7473 2e6f   ``skbio.stats.o
-00011fe0: 7264 696e 6174 696f 6e60 600a 202d 2060  rdination``. - `
-00011ff0: 6073 6b62 696f 2e73 7461 7473 2e6f 7264  `skbio.stats.ord
-00012000: 696e 6174 696f 6e2e 6361 6e6f 6e69 6361  ination.canonica
-00012010: 6c5f 636f 7272 6573 706f 6e64 656e 6365  l_correspondence
-00012020: 5f61 6e61 6c79 7369 7360 6020 746f 2060  _analysis`` to `
-00012030: 6073 6b62 696f 2e73 7461 7473 2e6f 7264  `skbio.stats.ord
-00012040: 696e 6174 696f 6e60 600a 202d 2060 6073  ination``. - ``s
-00012050: 6b62 696f 2e73 7461 7473 2e6f 7264 696e  kbio.stats.ordin
-00012060: 6174 696f 6e2e 636f 7272 6573 706f 6e64  ation.correspond
-00012070: 656e 6365 5f61 6e61 6c79 7369 7360 6020  ence_analysis`` 
-00012080: 746f 2060 6073 6b62 696f 2e73 7461 7473  to ``skbio.stats
-00012090: 2e6f 7264 696e 6174 696f 6e60 600a 202d  .ordination``. -
-000120a0: 2060 6073 6b62 696f 2e73 7461 7473 2e6f   ``skbio.stats.o
-000120b0: 7264 696e 6174 696f 6e2e 7072 696e 6369  rdination.princi
-000120c0: 7061 6c5f 636f 6f72 6469 6e61 7465 5f61  pal_coordinate_a
-000120d0: 6e61 6c79 7369 7360 6020 746f 2060 6073  nalysis`` to ``s
-000120e0: 6b62 696f 2e73 7461 7473 2e6f 7264 696e  kbio.stats.ordin
-000120f0: 6174 696f 6e60 600a 202d 2060 6073 6b62  ation``. - ``skb
-00012100: 696f 2e73 7461 7473 2e6f 7264 696e 6174  io.stats.ordinat
-00012110: 696f 6e2e 7265 6475 6e64 616e 6379 5f61  ion.redundancy_a
-00012120: 6e61 6c79 7369 7360 6020 746f 2060 6073  nalysis`` to ``s
-00012130: 6b62 696f 2e73 7461 7473 2e6f 7264 696e  kbio.stats.ordin
-00012140: 6174 696f 6e60 600a 202d 2060 6073 6b62  ation``. - ``skb
-00012150: 696f 2e74 7265 652e 7472 6565 6060 2074  io.tree.tree`` t
-00012160: 6f20 6060 736b 6269 6f2e 7472 6565 6060  o ``skbio.tree``
-00012170: 0a20 2d20 6060 736b 6269 6f2e 7472 6565  . - ``skbio.tree
-00012180: 2e74 7269 6560 6020 746f 2060 6073 6b62  .trie`` to ``skb
-00012190: 696f 2e74 7265 6560 600a 202d 2060 6073  io.tree``. - ``s
-000121a0: 6b62 696f 2e75 7469 6c2e 6d69 7363 6060  kbio.util.misc``
-000121b0: 2074 6f20 6060 736b 6269 6f2e 7574 696c   to ``skbio.util
-000121c0: 6060 0a20 2d20 6060 736b 6269 6f2e 7574  ``. - ``skbio.ut
-000121d0: 696c 2e74 6573 7469 6e67 6060 2074 6f20  il.testing`` to 
-000121e0: 6060 736b 6269 6f2e 7574 696c 6060 0a20  ``skbio.util``. 
-000121f0: 2d20 6060 736b 6269 6f2e 7574 696c 2e65  - ``skbio.util.e
-00012200: 7863 6570 7469 6f6e 6060 2074 6f20 6060  xception`` to ``
-00012210: 736b 6269 6f2e 7574 696c 6060 0a20 2d20  skbio.util``. - 
-00012220: 6060 736b 6269 6f2e 7574 696c 2e77 6172  ``skbio.util.war
-00012230: 6e69 6e67 6060 2074 6f20 6060 736b 6269  ning`` to ``skbi
-00012240: 6f2e 7574 696c 6060 0a2a 204d 6f76 6564  o.util``.* Moved
-00012250: 2060 6073 6b62 696f 2e64 6973 7461 6e63   ``skbio.distanc
-00012260: 6560 6020 636f 6e74 656e 7473 2069 6e74  e`` contents int
-00012270: 6f20 6060 736b 6269 6f2e 7374 6174 732e  o ``skbio.stats.
-00012280: 6469 7374 616e 6365 6060 2e0a 0a23 2323  distance``...###
-00012290: 204d 6973 6365 6c6c 616e 656f 7573 0a0a   Miscellaneous..
-000122a0: 2a20 5265 6c61 7865 6420 7265 7175 6972  * Relaxed requir
-000122b0: 656d 656e 7420 696e 2060 6042 696f 6c6f  ement in ``Biolo
-000122c0: 6769 6361 6c53 6571 7565 6e63 652e 6469  gicalSequence.di
-000122d0: 7374 616e 6365 6060 2074 6861 7420 7365  stance`` that se
-000122e0: 7175 656e 6365 7320 6265 696e 6720 636f  quences being co
-000122f0: 6d70 6172 6564 2061 7265 206f 6620 6571  mpared are of eq
-00012300: 7561 6c20 6c65 6e67 7468 2e20 5468 6973  ual length. This
-00012310: 2069 7320 7265 6c65 7661 6e74 2066 6f72   is relevant for
-00012320: 2048 616d 6d69 6e67 2064 6973 7461 6e63   Hamming distanc
-00012330: 652c 2073 6f20 7468 6520 6368 6563 6b20  e, so the check 
-00012340: 6973 2073 7469 6c6c 2070 6572 666f 726d  is still perform
-00012350: 6564 2069 6e20 7468 6174 2063 6173 652c  ed in that case,
-00012360: 2062 7574 206f 7468 6572 2064 6973 7461   but other dista
-00012370: 6e63 6520 6d65 7472 6963 7320 6d61 7920  nce metrics may 
-00012380: 6e6f 7420 6861 7665 2074 6861 7420 7265  not have that re
-00012390: 7175 6972 656d 656e 742e 2053 6565 205b  quirement. See [
-000123a0: 2335 3034 5d28 6874 7470 733a 2f2f 6769  #504](https://gi
-000123b0: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
-000123c0: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
-000123d0: 6573 2f35 3037 2929 2e0a 2a20 5265 6e61  es/507))..* Rena
-000123e0: 6d65 6420 6060 706f 7765 7274 7269 702e  med ``powertrip.
-000123f0: 7079 6060 2072 6570 6f2d 6368 6563 6b69  py`` repo-checki
-00012400: 6e67 2073 6372 6970 7420 746f 2060 6063  ng script to ``c
-00012410: 6865 636b 6c69 7374 2e70 7960 6020 666f  hecklist.py`` fo
-00012420: 7220 636c 6172 6974 792e 0a2a 2060 6063  r clarity..* ``c
-00012430: 6865 636b 6c69 7374 2e70 7960 6020 6e6f  hecklist.py`` no
-00012440: 7720 656e 7375 7265 7320 7468 6174 2061  w ensures that a
-00012450: 6c6c 2075 6e69 7420 7465 7374 7320 696d  ll unit tests im
-00012460: 706f 7274 2066 726f 6d20 6120 6d69 6e69  port from a mini
-00012470: 6d61 6c6c 7920 6465 6570 2041 5049 2e20  mally deep API. 
-00012480: 466f 7220 6578 616d 706c 652c 2069 7420  For example, it 
-00012490: 7769 6c6c 2070 726f 6475 6365 2061 6e20  will produce an 
-000124a0: 6572 726f 7220 6966 2060 6073 6b62 696f  error if ``skbio
-000124b0: 2e63 6f72 652e 6469 7374 616e 6365 2e44  .core.distance.D
-000124c0: 6973 7461 6e63 654d 6174 7269 7860 6020  istanceMatrix`` 
-000124d0: 6973 2075 7365 6420 6f76 6572 2060 6073  is used over ``s
-000124e0: 6b62 696f 2e44 6973 7461 6e63 654d 6174  kbio.DistanceMat
-000124f0: 7269 7860 602e 0a2a 2045 7874 7261 2064  rix``..* Extra d
-00012500: 696d 656e 7369 6f6e 2069 7320 6e6f 206c  imension is no l
-00012510: 6f6e 6765 7220 6361 6c63 756c 6174 6564  onger calculated
-00012520: 2069 6e20 6060 736b 6269 6f2e 7374 6174   in ``skbio.stat
-00012530: 732e 7370 6174 6961 6c2e 7072 6f63 7275  s.spatial.procru
-00012540: 7374 6573 6060 2e0a 2a20 4578 7061 6e64  stes``..* Expand
-00012550: 6564 2064 6f63 756d 656e 7461 7469 6f6e  ed documentation
-00012560: 2069 6e20 7661 7269 6f75 7320 7375 6270   in various subp
-00012570: 6163 6b61 6765 732e 0a2a 2041 6464 6564  ackages..* Added
-00012580: 206e 6577 2073 6369 6b69 742d 6269 6f20   new scikit-bio 
-00012590: 6c6f 676f 2e20 5468 616e 6b73 205b 416c  logo. Thanks [Al
-000125a0: 696e 6120 5072 6173 7361 735d 2868 7474  ina Prassas](htt
-000125b0: 703a 2f2f 6361 7267 6f63 6f6c 6c65 6374  p://cargocollect
-000125c0: 6976 652e 636f 6d2f 616c 696e 6170 7261  ive.com/alinapra
-000125d0: 7373 6173 2921 0a0a 2323 2056 6572 7369  ssas)!..## Versi
-000125e0: 6f6e 2030 2e31 2e34 2028 3230 3134 2d30  on 0.1.4 (2014-0
-000125f0: 362d 3235 290a 0a54 6869 7320 6973 2061  6-25)..This is a
-00012600: 2070 7265 2d61 6c70 6861 2072 656c 6561   pre-alpha relea
-00012610: 7365 2e20 4174 2074 6869 7320 7374 6167  se. At this stag
-00012620: 652c 206d 616a 6f72 2062 6163 6b77 6172  e, major backwar
-00012630: 6473 2d69 6e63 6f6d 7061 7469 626c 6520  ds-incompatible 
-00012640: 4150 4920 6368 616e 6765 7320 6361 6e20  API changes can 
-00012650: 616e 6420 7769 6c6c 2068 6170 7065 6e2e  and will happen.
-00012660: 0a0a 2323 2320 4665 6174 7572 6573 0a0a  ..### Features..
-00012670: 2a20 4164 6465 6420 5079 7468 6f6e 2069  * Added Python i
-00012680: 6d70 6c65 6d65 6e74 6174 696f 6e73 206f  mplementations o
-00012690: 6620 536d 6974 682d 5761 7465 726d 616e  f Smith-Waterman
-000126a0: 2061 6e64 204e 6565 646c 656d 616e 2d57   and Needleman-W
-000126b0: 756e 7363 6820 616c 6967 6e6d 656e 7420  unsch alignment 
-000126c0: 6173 2060 6073 6b62 696f 2e63 6f72 652e  as ``skbio.core.
-000126d0: 616c 6967 6e6d 656e 742e 7061 6972 7769  alignment.pairwi
-000126e0: 7365 2e6c 6f63 616c 5f70 6169 7277 6973  se.local_pairwis
-000126f0: 655f 616c 6967 6e60 6020 616e 6420 6060  e_align`` and ``
-00012700: 736b 6269 6f2e 636f 7265 2e61 6c69 676e  skbio.core.align
-00012710: 6d65 6e74 2e70 6169 7277 6973 652e 676c  ment.pairwise.gl
-00012720: 6f62 616c 5f70 6169 7277 6973 655f 616c  obal_pairwise_al
-00012730: 6967 6e60 602e 2054 6865 7365 2061 7265  ign``. These are
-00012740: 206d 7563 6820 736c 6f77 6572 2074 6861   much slower tha
-00012750: 6e20 6e61 7469 7665 2043 2069 6d70 6c65  n native C imple
-00012760: 6d65 6e74 6174 696f 6e73 2028 652e 672e  mentations (e.g.
-00012770: 2c20 6060 736b 6269 6f2e 636f 7265 2e61  , ``skbio.core.a
-00012780: 6c69 676e 6d65 6e74 2e6c 6f63 616c 5f70  lignment.local_p
-00012790: 6169 7277 6973 655f 616c 6967 6e5f 7373  airwise_align_ss
-000127a0: 7760 6029 2061 6e64 2061 7320 6120 7265  w``) and as a re
-000127b0: 7375 6c74 2072 6169 7365 2061 6e20 6060  sult raise an ``
-000127c0: 4566 6669 6365 6e63 7957 6172 6e69 6e67  EfficencyWarning
-000127d0: 6060 2077 6865 6e20 6361 6c6c 6564 2c20  `` when called, 
-000127e0: 6275 7420 6172 6520 696e 636c 7564 6564  but are included
-000127f0: 2061 7320 7468 6579 2073 6572 7665 2061   as they serve a
-00012800: 7320 7573 6566 756c 2065 6475 6361 7469  s useful educati
-00012810: 6f6e 616c 2065 7861 6d70 6c65 7320 6173  onal examples as
-00012820: 2074 6865 79e2 8099 7265 2073 696d 706c   they...re simpl
-00012830: 6520 746f 2065 7870 6572 696d 656e 7420  e to experiment 
-00012840: 7769 7468 2e0a 2a20 4164 6465 6420 6060  with..* Added ``
-00012850: 736b 6269 6f2e 636f 7265 2e64 6976 6572  skbio.core.diver
-00012860: 7369 7479 2e62 6574 612e 7077 5f64 6973  sity.beta.pw_dis
-00012870: 7461 6e63 6573 6060 2061 6e64 2060 6073  tances`` and ``s
-00012880: 6b62 696f 2e63 6f72 652e 6469 7665 7273  kbio.core.divers
-00012890: 6974 792e 6265 7461 2e70 775f 6469 7374  ity.beta.pw_dist
-000128a0: 616e 6365 735f 6672 6f6d 5f74 6162 6c65  ances_from_table
-000128b0: 6060 2e20 5468 6573 6520 7072 6f76 6964  ``. These provid
-000128c0: 6520 636f 6e76 656e 6965 6e74 2061 6363  e convenient acc
-000128d0: 6573 7320 746f 2074 6865 2060 6073 6369  ess to the ``sci
-000128e0: 7079 2e73 7061 7469 616c 2e64 6973 7461  py.spatial.dista
-000128f0: 6e63 652e 7064 6973 7460 6020 2a62 6574  nce.pdist`` *bet
-00012900: 6120 6469 7665 7273 6974 792a 206d 6574  a diversity* met
-00012910: 7269 6373 2066 726f 6d20 7769 7468 696e  rics from within
-00012920: 2073 6369 6b69 742d 6269 6f2e 2054 6865   scikit-bio. The
-00012930: 2060 6073 6b62 696f 2e63 6f72 652e 6469   ``skbio.core.di
-00012940: 7665 7273 6974 792e 6265 7461 2e70 775f  versity.beta.pw_
-00012950: 6469 7374 616e 6365 735f 6672 6f6d 5f74  distances_from_t
-00012960: 6162 6c65 6060 2066 756e 6374 696f 6e20  able`` function 
-00012970: 7769 6c6c 206f 6e6c 7920 6265 2061 7661  will only be ava
-00012980: 696c 6162 6c65 2074 656d 706f 7261 7269  ilable temporari
-00012990: 6c79 2c20 756e 7469 6c20 7468 6520 6060  ly, until the ``
-000129a0: 6269 6f6d 2e74 6162 6c65 2e54 6162 6c65  biom.table.Table
-000129b0: 6060 206f 626a 6563 7420 6973 206d 6572  `` object is mer
-000129c0: 6765 6420 696e 746f 2073 6369 6b69 742d  ged into scikit-
-000129d0: 6269 6f20 2873 6565 205b 2334 3839 5d28  bio (see [#489](
-000129e0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
-000129f0: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
-00012a00: 742d 6269 6f2f 6973 7375 6573 2f34 3839  t-bio/issues/489
-00012a10: 2929 2c20 6174 2077 6869 6368 2070 6f69  )), at which poi
-00012a20: 6e74 2060 6073 6b62 696f 2e63 6f72 652e  nt ``skbio.core.
-00012a30: 6469 7665 7273 6974 792e 6265 7461 2e70  diversity.beta.p
-00012a40: 775f 6469 7374 616e 6365 7360 6020 7769  w_distances`` wi
-00012a50: 6c6c 2062 6520 7570 6461 7465 6420 746f  ll be updated to
-00012a60: 2075 7365 2074 6861 742e 0a2a 2041 6464   use that..* Add
-00012a70: 6564 2060 6073 6b62 696f 2e63 6f72 652e  ed ``skbio.core.
-00012a80: 616c 6967 6e6d 656e 742e 5374 6f63 6b68  alignment.Stockh
-00012a90: 6f6c 6d41 6c69 676e 6d65 6e74 6060 2c20  olmAlignment``, 
-00012aa0: 7768 6963 6820 7072 6f76 6964 6573 2073  which provides s
-00012ab0: 7570 706f 7274 2066 6f72 2070 6172 7369  upport for parsi
-00012ac0: 6e67 205b 5374 6f63 6b68 6f6c 6d2d 666f  ng [Stockholm-fo
-00012ad0: 726d 6174 7465 6420 616c 6967 6e6d 656e  rmatted alignmen
-00012ae0: 7420 6669 6c65 735d 2868 7474 703a 2f2f  t files](http://
-00012af0: 736f 6e6e 6861 6d6d 6572 2e73 6263 2e73  sonnhammer.sbc.s
-00012b00: 752e 7365 2f53 746f 636b 686f 6c6d 2e68  u.se/Stockholm.h
-00012b10: 746d 6c29 2061 6e64 2077 6f72 6b69 6e67  tml) and working
-00012b20: 2077 6974 6820 7468 6f73 6520 616c 6967   with those alig
-00012b30: 6e6d 656e 7473 2069 6e20 7468 6520 636f  nments in the co
-00012b40: 6e74 6578 7420 524e 4120 7365 636f 6e64  ntext RNA second
-00012b50: 6172 7920 7374 7275 6374 7572 616c 2069  ary structural i
-00012b60: 6e66 6f72 6d61 7469 6f6e 2e0a 2a20 4164  nformation..* Ad
-00012b70: 6465 6420 6060 736b 6269 6f2e 636f 7265  ded ``skbio.core
-00012b80: 2e74 7265 652e 6d61 6a6f 7269 7479 5f72  .tree.majority_r
-00012b90: 756c 6560 6020 6675 6e63 7469 6f6e 2066  ule`` function f
-00012ba0: 6f72 2063 6f6d 7075 7469 6e67 2063 6f6e  or computing con
-00012bb0: 7365 6e73 7573 2074 7265 6573 2066 726f  sensus trees fro
-00012bc0: 6d20 6120 6c69 7374 206f 6620 7472 6565  m a list of tree
-00012bd0: 732e 0a0a 2323 2320 4261 636b 7761 7264  s...### Backward
-00012be0: 2d69 6e63 6f6d 7061 7469 626c 6520 6368  -incompatible ch
-00012bf0: 616e 6765 730a 0a2a 2046 756e 6374 696f  anges..* Functio
-00012c00: 6e20 6060 736b 6269 6f2e 636f 7265 2e61  n ``skbio.core.a
-00012c10: 6c69 676e 6d65 6e74 2e61 6c69 676e 5f73  lignment.align_s
-00012c20: 7472 6970 6564 5f73 6d69 7468 5f77 6174  triped_smith_wat
-00012c30: 6572 6d61 6e60 6020 7265 6e61 6d65 6420  erman`` renamed 
-00012c40: 746f 2060 606c 6f63 616c 5f70 6169 7277  to ``local_pairw
-00012c50: 6973 655f 616c 6967 6e5f 7373 7760 6020  ise_align_ssw`` 
-00012c60: 616e 6420 6e6f 7720 7265 7475 726e 7320  and now returns 
-00012c70: 616e 2060 6041 6c69 676e 6d65 6e74 6060  an ``Alignment``
-00012c80: 206f 626a 6563 7420 696e 7374 6561 6420   object instead 
-00012c90: 6f66 2061 6e20 6060 416c 6967 6e6d 656e  of an ``Alignmen
-00012ca0: 7453 7472 7563 7475 7265 6060 0a2a 2054  tStructure``.* T
-00012cb0: 6865 2066 6f6c 6c6f 7769 6e67 206b 6579  he following key
-00012cc0: 776f 7264 2d61 7267 756d 656e 7473 2066  word-arguments f
-00012cd0: 6f72 2060 6053 7472 6970 6564 536d 6974  or ``StripedSmit
-00012ce0: 6857 6174 6572 6d61 6e60 6020 616e 6420  hWaterman`` and 
-00012cf0: 6060 6c6f 6361 6c5f 7061 6972 7769 7365  ``local_pairwise
-00012d00: 5f61 6c69 676e 5f73 7377 6060 2068 6176  _align_ssw`` hav
-00012d10: 6520 6265 656e 2072 656e 616d 6564 3a0a  e been renamed:.
-00012d20: 2020 2020 2a20 6060 6761 705f 6f70 656e      * ``gap_open
-00012d30: 6060 202d 3e20 6060 6761 705f 6f70 656e  `` -> ``gap_open
-00012d40: 5f70 656e 616c 7479 6060 0a20 2020 202a  _penalty``.    *
-00012d50: 2060 6067 6170 5f65 7874 656e 6460 6020   ``gap_extend`` 
-00012d60: 2d3e 2060 6067 6170 5f65 7874 656e 645f  -> ``gap_extend_
-00012d70: 7065 6e61 6c74 7960 600a 2020 2020 2a20  penalty``.    * 
-00012d80: 6060 6d61 7463 6860 6020 2d3e 2060 606d  ``match`` -> ``m
-00012d90: 6174 6368 5f73 636f 7265 6060 0a20 2020  atch_score``.   
-00012da0: 202a 2060 606d 6973 6d61 7463 6860 6020   * ``mismatch`` 
-00012db0: 2d3e 2060 606d 6973 6d61 7463 685f 7363  -> ``mismatch_sc
-00012dc0: 6f72 6560 600a 2a20 5265 6d6f 7665 6420  ore``.* Removed 
-00012dd0: 6060 736b 6269 6f2e 7574 696c 2e73 6f72  ``skbio.util.sor
-00012de0: 7460 6020 6d6f 6475 6c65 2069 6e20 6661  t`` module in fa
-00012df0: 766f 7220 6f66 205b 6e61 7473 6f72 745d  vor of [natsort]
-00012e00: 2868 7474 7073 3a2f 2f70 7970 692e 7079  (https://pypi.py
-00012e10: 7468 6f6e 2e6f 7267 2f70 7970 692f 6e61  thon.org/pypi/na
-00012e20: 7473 6f72 7429 2070 6163 6b61 6765 2e0a  tsort) package..
-00012e30: 0a23 2323 204d 6973 6365 6c6c 616e 656f  .### Miscellaneo
-00012e40: 7573 0a0a 2a20 4164 6465 6420 706f 7765  us..* Added powe
-00012e50: 7274 7269 702e 7079 2073 6372 6970 7420  rtrip.py script 
-00012e60: 746f 2070 6572 666f 726d 2062 6173 6963  to perform basic
-00012e70: 2073 616e 6974 792d 6368 6563 6b69 6e67   sanity-checking
-00012e80: 206f 6620 7468 6520 7265 706f 2062 6173   of the repo bas
-00012e90: 6564 206f 6e20 7265 6375 7272 696e 6720  ed on recurring 
-00012ea0: 6973 7375 6573 2074 6861 7420 7765 7265  issues that were
-00012eb0: 6e27 7420 6265 696e 6720 6361 7567 6874  n't being caught
-00012ec0: 2075 6e74 696c 2072 656c 6561 7365 2074   until release t
-00012ed0: 696d 653b 2061 6464 6564 2074 6f20 5472  ime; added to Tr
-00012ee0: 6176 6973 2062 7569 6c64 2e0a 2a20 4164  avis build..* Ad
-00012ef0: 6465 6420 5245 4c45 4153 452e 6d64 2077  ded RELEASE.md w
-00012f00: 6974 6820 7265 6c65 6173 6520 696e 7374  ith release inst
-00012f10: 7275 6374 696f 6e73 2e0a 2a20 4164 6465  ructions..* Adde
-00012f20: 6420 696e 7465 7273 7068 696e 7820 6d61  d intersphinx ma
-00012f30: 7070 696e 6773 2074 6f20 646f 6373 2073  ppings to docs s
-00012f40: 6f20 7468 6174 2022 5365 6520 416c 736f  o that "See Also
-00012f50: 2220 7265 6665 7265 6e63 6573 2074 6f20  " references to 
-00012f60: 6e75 6d70 792c 2073 6369 7079 2c20 6d61  numpy, scipy, ma
-00012f70: 7470 6c6f 746c 6962 2c20 616e 6420 7061  tplotlib, and pa
-00012f80: 6e64 6173 2061 7265 2068 7970 6572 6c69  ndas are hyperli
-00012f90: 6e6b 732e 0a2a 2054 6865 2066 6f6c 6c6f  nks..* The follo
-00012fa0: 7769 6e67 2063 6c61 7373 6573 2061 7265  wing classes are
-00012fb0: 206e 6f20 6c6f 6e67 6572 2060 606e 616d   no longer ``nam
-00012fc0: 6564 7475 706c 6560 6020 7375 6263 6c61  edtuple`` subcla
-00012fd0: 7373 6573 2028 7365 6520 5b23 3335 395d  sses (see [#359]
-00012fe0: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
-00012ff0: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
-00013000: 6974 2d62 696f 2f69 7373 7565 732f 3335  it-bio/issues/35
-00013010: 3929 2066 6f72 2074 6865 2072 6174 696f  9) for the ratio
-00013020: 6e61 6c65 293a 0a20 2020 202a 2060 6073  nale):.    * ``s
-00013030: 6b62 696f 2e6d 6174 682e 7374 6174 732e  kbio.math.stats.
-00013040: 6f72 6469 6e61 7469 6f6e 2e4f 7264 696e  ordination.Ordin
-00013050: 6174 696f 6e52 6573 756c 7473 6060 0a20  ationResults``. 
-00013060: 2020 202a 2060 6073 6b62 696f 2e6d 6174     * ``skbio.mat
-00013070: 682e 6772 6164 6965 6e74 2e47 726f 7570  h.gradient.Group
-00013080: 5265 7375 6c74 7360 600a 2020 2020 2a20  Results``.    * 
-00013090: 6060 736b 6269 6f2e 6d61 7468 2e67 7261  ``skbio.math.gra
-000130a0: 6469 656e 742e 4361 7465 676f 7279 5265  dient.CategoryRe
-000130b0: 7375 6c74 7360 600a 2020 2020 2a20 6060  sults``.    * ``
-000130c0: 736b 6269 6f2e 6d61 7468 2e67 7261 6469  skbio.math.gradi
-000130d0: 656e 742e 4772 6164 6965 6e74 414e 4f56  ent.GradientANOV
-000130e0: 4152 6573 756c 7473 6060 0a2a 2041 6464  AResults``.* Add
-000130f0: 6564 2063 6f64 696e 6720 6775 6964 656c  ed coding guidel
-00013100: 696e 6573 2064 7261 6674 2e0a 2a20 4164  ines draft..* Ad
-00013110: 6465 6420 6e65 7720 616c 7068 6120 6469  ded new alpha di
-00013120: 7665 7273 6974 7920 666f 726d 756c 6173  versity formulas
-00013130: 2074 6f20 7468 6520 6060 736b 6269 6f2e   to the ``skbio.
-00013140: 6d61 7468 2e64 6976 6572 7369 7479 2e61  math.diversity.a
-00013150: 6c70 6861 6060 2064 6f63 756d 656e 7461  lpha`` documenta
-00013160: 7469 6f6e 2e0a 0a23 2320 5665 7273 696f  tion...## Versio
-00013170: 6e20 302e 312e 3320 2832 3031 342d 3036  n 0.1.3 (2014-06
-00013180: 2d31 3229 0a0a 5468 6973 2069 7320 6120  -12)..This is a 
-00013190: 7072 652d 616c 7068 6120 7265 6c65 6173  pre-alpha releas
-000131a0: 652e 2041 7420 7468 6973 2073 7461 6765  e. At this stage
-000131b0: 2c20 6d61 6a6f 7220 6261 636b 7761 7264  , major backward
-000131c0: 732d 696e 636f 6d70 6174 6962 6c65 2041  s-incompatible A
-000131d0: 5049 2063 6861 6e67 6573 2063 616e 2061  PI changes can a
-000131e0: 6e64 2077 696c 6c20 6861 7070 656e 2e0a  nd will happen..
-000131f0: 0a23 2323 2046 6561 7475 7265 730a 0a2a  .### Features..*
-00013200: 2041 6464 6564 2060 6065 6e66 6f72 6365   Added ``enforce
-00013210: 5f71 7561 6c5f 7261 6e67 6560 6020 7061  _qual_range`` pa
-00013220: 7261 6d65 7465 7220 746f 2060 6070 6172  rameter to ``par
-00013230: 7365 5f66 6173 7471 6060 2028 6f6e 2062  se_fastq`` (on b
-00013240: 7920 6465 6661 756c 742c 206d 6169 6e74  y default, maint
-00013250: 6169 6e69 6e67 2062 6163 6b77 6172 6420  aining backward 
-00013260: 636f 6d70 6174 6962 696c 6974 7929 2e20  compatibility). 
-00013270: 5468 6973 2061 6c6c 6f77 7320 6469 7361  This allows disa
-00013280: 626c 696e 6720 6f66 2074 6865 2071 7561  bling of the qua
-00013290: 6c69 7479 2073 636f 7265 2072 616e 6765  lity score range
-000132a0: 2d63 6865 636b 696e 672e 0a2a 2041 6464  -checking..* Add
-000132b0: 6564 2060 6073 6b62 696f 2e63 6f72 652e  ed ``skbio.core.
-000132c0: 7472 6565 2e6e 6a60 602c 2077 6869 6368  tree.nj``, which
-000132d0: 2061 7070 6c69 6573 206e 6569 6768 626f   applies neighbo
-000132e0: 722d 6a6f 696e 696e 6720 666f 7220 7068  r-joining for ph
-000132f0: 796c 6f67 656e 6574 6963 2072 6563 6f6e  ylogenetic recon
-00013300: 7374 7275 6374 696f 6e2e 0a2a 2041 6464  struction..* Add
-00013310: 6564 2060 6062 696f 656e 7660 602c 2060  ed ``bioenv``, `
-00013320: 606d 616e 7465 6c60 602c 2061 6e64 2060  `mantel``, and `
-00013330: 6070 776d 616e 7465 6c60 6020 6469 7374  `pwmantel`` dist
-00013340: 616e 6365 2d62 6173 6564 2073 7461 7469  ance-based stati
-00013350: 7374 6963 7320 746f 2060 6073 6b62 696f  stics to ``skbio
-00013360: 2e6d 6174 682e 7374 6174 732e 6469 7374  .math.stats.dist
-00013370: 616e 6365 6060 2073 7562 7061 636b 6167  ance`` subpackag
-00013380: 652e 0a2a 2041 6464 6564 2060 6073 6b62  e..* Added ``skb
-00013390: 696f 2e6d 6174 682e 7374 6174 732e 6d69  io.math.stats.mi
-000133a0: 7363 6060 206d 6f64 756c 6520 666f 7220  sc`` module for 
-000133b0: 6d69 7363 656c 6c61 6e65 6f75 7320 7374  miscellaneous st
-000133c0: 6174 7320 7574 696c 6974 7920 6675 6e63  ats utility func
-000133d0: 7469 6f6e 732e 0a2a 2049 4473 2061 7265  tions..* IDs are
-000133e0: 206e 6f77 206f 7074 696f 6e61 6c20 7768   now optional wh
-000133f0: 656e 2063 6f6e 7374 7275 6374 696e 6720  en constructing 
-00013400: 6120 6060 4469 7373 696d 696c 6172 6974  a ``Dissimilarit
-00013410: 794d 6174 7269 7860 6020 6f72 2060 6044  yMatrix`` or ``D
-00013420: 6973 7461 6e63 654d 6174 7269 7860 6020  istanceMatrix`` 
-00013430: 286d 6f6e 6f74 6f6e 6963 616c 6c79 2d69  (monotonically-i
-00013440: 6e63 7265 6173 696e 6720 696e 7465 6765  ncreasing intege
-00013450: 7273 2063 6173 7420 6173 2073 7472 696e  rs cast as strin
-00013460: 6773 2061 7265 2061 7574 6f6d 6174 6963  gs are automatic
-00013470: 616c 6c79 2075 7365 6429 2e0a 2a20 4164  ally used)..* Ad
-00013480: 6465 6420 6060 4469 7374 616e 6365 4d61  ded ``DistanceMa
-00013490: 7472 6978 2e70 6572 6d75 7465 6060 206d  trix.permute`` m
-000134a0: 6574 686f 6420 666f 7220 7261 6e64 6f6d  ethod for random
-000134b0: 6c79 2070 6572 6d75 7469 6e67 2072 6f77  ly permuting row
-000134c0: 7320 616e 6420 636f 6c75 6d6e 7320 6f66  s and columns of
-000134d0: 2061 2064 6973 7461 6e63 6520 6d61 7472   a distance matr
-000134e0: 6978 2e0a 2a20 4164 6465 6420 7468 6520  ix..* Added the 
-000134f0: 666f 6c6c 6f77 696e 6720 6d65 7468 6f64  following method
-00013500: 7320 746f 2060 6044 6973 7369 6d69 6c61  s to ``Dissimila
-00013510: 7269 7479 4d61 7472 6978 6060 3a20 6060  rityMatrix``: ``
-00013520: 6669 6c74 6572 6060 2c20 6060 696e 6465  filter``, ``inde
-00013530: 7860 602c 2061 6e64 2060 605f 5f63 6f6e  x``, and ``__con
-00013540: 7461 696e 735f 5f60 6020 666f 7220 4944  tains__`` for ID
-00013550: 2d62 6173 6564 2066 696c 7465 7269 6e67  -based filtering
-00013560: 2c20 696e 6465 7820 6c6f 6f6b 7570 2c20  , index lookup, 
-00013570: 616e 6420 6d65 6d62 6572 7368 6970 2074  and membership t
-00013580: 6573 7469 6e67 2c20 7265 7370 6563 7469  esting, respecti
-00013590: 7665 6c79 2e0a 2a20 4164 6465 6420 6060  vely..* Added ``
-000135a0: 6967 6e6f 7265 5f63 6f6d 6d65 6e74 6060  ignore_comment``
-000135b0: 2070 6172 616d 6574 6572 2074 6f20 6060   parameter to ``
-000135c0: 7061 7273 655f 6661 7374 6160 6020 286f  parse_fasta`` (o
-000135d0: 6666 2062 7920 6465 6661 756c 742c 206d  ff by default, m
-000135e0: 6169 6e74 6169 6e69 6e67 2062 6163 6b77  aintaining backw
-000135f0: 6172 6420 636f 6d70 6174 6962 696c 6974  ard compatibilit
-00013600: 7929 2e20 5468 6973 2068 616e 646c 6573  y). This handles
-00013610: 2073 7472 6970 7069 6e67 2074 6865 2063   stripping the c
-00013620: 6f6d 6d65 6e74 2066 6965 6c64 2066 726f  omment field fro
-00013630: 6d20 7468 6520 6865 6164 6572 206c 696e  m the header lin
-00013640: 6520 2869 2e65 2e2c 2061 6c6c 2063 6861  e (i.e., all cha
-00013650: 7261 6374 6572 7320 6265 6769 6e6e 696e  racters beginnin
-00013660: 6720 7769 7468 2074 6865 2066 6972 7374  g with the first
-00013670: 2073 7061 6365 2920 6265 666f 7265 2072   space) before r
-00013680: 6574 7572 6e69 6e67 2074 6865 206c 6162  eturning the lab
-00013690: 656c 2e0a 2a20 4164 6465 6420 696d 706f  el..* Added impo
-000136a0: 7274 7320 6f66 2060 6042 696f 6c6f 6769  rts of ``Biologi
-000136b0: 6361 6c53 6571 7565 6e63 6560 602c 2060  calSequence``, `
-000136c0: 604e 7563 6c65 6f74 6964 6553 6571 7565  `NucleotideSeque
-000136d0: 6e63 6560 602c 2060 6044 4e41 6060 2c20  nce``, ``DNA``, 
-000136e0: 6060 444e 4153 6571 7565 6e63 6560 602c  ``DNASequence``,
-000136f0: 2060 6052 4e41 6060 2c20 6060 524e 4153   ``RNA``, ``RNAS
-00013700: 6571 7565 6e63 6560 602c 2060 6050 726f  equence``, ``Pro
-00013710: 7465 696e 6060 2c20 6060 5072 6f74 6569  tein``, ``Protei
-00013720: 6e53 6571 7565 6e63 6560 602c 2060 6044  nSequence``, ``D
-00013730: 6973 7461 6e63 654d 6174 7269 7860 602c  istanceMatrix``,
-00013740: 2060 6061 6c69 676e 5f73 7472 6970 6564   ``align_striped
-00013750: 5f73 6d69 7468 5f77 6174 6572 6d61 6e60  _smith_waterman`
-00013760: 602c 2060 6020 5365 7175 656e 6365 436f  `, `` SequenceCo
-00013770: 6c6c 6563 7469 6f6e 6060 2c20 6060 416c  llection``, ``Al
-00013780: 6967 6e6d 656e 7460 602c 2060 6054 7265  ignment``, ``Tre
-00013790: 654e 6f64 6560 602c 2060 606e 6a60 602c  eNode``, ``nj``,
-000137a0: 2060 6070 6172 7365 5f66 6173 7461 6060   ``parse_fasta``
-000137b0: 2c20 6060 7061 7273 655f 6661 7374 7160  , ``parse_fastq`
-000137c0: 602c 2060 6070 6172 7365 5f71 7561 6c60  `, ``parse_qual`
-000137d0: 602c 2060 6046 6173 7461 4974 6572 6174  `, ``FastaIterat
-000137e0: 6f72 6060 2c20 6060 4661 7374 7149 7465  or``, ``FastqIte
-000137f0: 7261 746f 7260 602c 2060 6053 6571 7565  rator``, ``Seque
-00013800: 6e63 6549 7465 7261 746f 7260 6020 696e  nceIterator`` in
-00013810: 2060 6073 6b62 696f 2f5f 5f69 6e69 745f   ``skbio/__init_
-00013820: 5f2e 7079 6060 2066 6f72 2063 6f6e 7665  _.py`` for conve
-00013830: 6e69 656e 7420 696d 706f 7274 696e 672e  nient importing.
-00013840: 2046 6f72 2065 7861 6d70 6c65 2c20 6974   For example, it
-00013850: 2773 206e 6f77 2070 6f73 7369 626c 6520  's now possible 
-00013860: 746f 2060 6066 726f 6d20 736b 6269 6f20  to ``from skbio 
-00013870: 696d 706f 7274 2041 6c69 676e 6d65 6e74  import Alignment
-00013880: 6060 2c20 7261 7468 6572 2074 6861 6e20  ``, rather than 
-00013890: 6060 6672 6f6d 2073 6b62 696f 2e63 6f72  ``from skbio.cor
-000138a0: 652e 616c 6967 6e6d 656e 7420 696d 706f  e.alignment impo
-000138b0: 7274 2041 6c69 676e 6d65 6e74 6060 2e0a  rt Alignment``..
-000138c0: 0a23 2323 2042 7567 2066 6978 6573 0a0a  .### Bug fixes..
-000138d0: 2a20 4669 7865 6420 6120 636f 7570 6c65  * Fixed a couple
-000138e0: 206f 6620 756e 6974 2074 6573 7473 2074   of unit tests t
-000138f0: 6861 7420 636f 756c 6420 6661 696c 2073  hat could fail s
-00013900: 746f 6368 6173 7469 6361 6c6c 792e 0a2a  tochastically..*
-00013910: 2041 6464 6564 206d 6973 7369 6e67 2060   Added missing `
-00013920: 605f 5f69 6e69 745f 5f2e 7079 6060 2066  `__init__.py`` f
-00013930: 696c 6573 2074 6f20 6120 636f 7570 6c65  iles to a couple
-00013940: 206f 6620 7465 7374 2064 6972 6563 746f   of test directo
-00013950: 7269 6573 2073 6f20 7468 6174 2074 6865  ries so that the
-00013960: 7365 2074 6573 7473 2077 6f6e 2774 2062  se tests won't b
-00013970: 6520 736b 6970 7065 642e 0a2a 2060 6070  e skipped..* ``p
-00013980: 6172 7365 5f66 6173 7471 6060 206e 6f77  arse_fastq`` now
-00013990: 2072 6169 7365 7320 616e 2065 7272 6f72   raises an error
-000139a0: 206f 6e20 6461 6e67 6c69 6e67 2072 6563   on dangling rec
-000139b0: 6f72 6473 2e0a 2a20 4669 7865 6420 7365  ords..* Fixed se
-000139c0: 7665 7261 6c20 7761 726e 696e 6773 2074  veral warnings t
-000139d0: 6861 7420 7765 7265 2072 6169 7365 6420  hat were raised 
-000139e0: 7768 696c 6520 7275 6e6e 696e 6720 7468  while running th
-000139f0: 6520 7465 7374 2073 7569 7465 2077 6974  e test suite wit
-00013a00: 6820 5079 7468 6f6e 2033 2e34 2e0a 0a23  h Python 3.4...#
-00013a10: 2323 2042 6163 6b77 6172 642d 696e 636f  ## Backward-inco
-00013a20: 6d70 6174 6962 6c65 2063 6861 6e67 6573  mpatible changes
-00013a30: 0a0a 2a20 4675 6e63 7469 6f6e 616c 6974  ..* Functionalit
-00013a40: 7920 696d 706f 7274 6564 2066 726f 6d20  y imported from 
-00013a50: 6060 736b 6269 6f2e 636f 7265 2e73 7377  ``skbio.core.ssw
-00013a60: 6060 206d 7573 7420 6e6f 7720 6265 2069  `` must now be i
-00013a70: 6d70 6f72 7465 6420 6672 6f6d 2060 6073  mported from ``s
-00013a80: 6b62 696f 2e63 6f72 652e 616c 6967 6e6d  kbio.core.alignm
-00013a90: 656e 7460 6020 696e 7374 6561 642e 0a0a  ent`` instead...
-00013aa0: 2323 2320 4d69 7363 656c 6c61 6e65 6f75  ### Miscellaneou
-00013ab0: 730a 0a2a 2043 6f64 6520 6973 206e 6f77  s..* Code is now
-00013ac0: 2066 6c61 6b65 382d 636f 6d70 6c69 616e   flake8-complian
-00013ad0: 743b 2061 6464 6564 2066 6c61 6b65 3820  t; added flake8 
-00013ae0: 6368 6563 6b69 6e67 2074 6f20 5472 6176  checking to Trav
-00013af0: 6973 2062 7569 6c64 2e0a 2a20 5661 7269  is build..* Vari
-00013b00: 6f75 7320 6164 6469 7469 6f6e 7320 616e  ous additions an
-00013b10: 6420 696d 7072 6f76 656d 656e 7473 2074  d improvements t
-00013b20: 6f20 646f 6375 6d65 6e74 6174 696f 6e20  o documentation 
-00013b30: 2841 5049 2c20 696e 7374 616c 6c61 7469  (API, installati
-00013b40: 6f6e 2069 6e73 7472 7563 7469 6f6e 732c  on instructions,
-00013b50: 2064 6576 656c 6f70 6572 2069 6e73 7472   developer instr
-00013b60: 7563 7469 6f6e 732c 2065 7463 2e29 2e0a  uctions, etc.)..
-00013b70: 2a20 6060 5f5f 6675 7475 7265 5f5f 6060  * ``__future__``
-00013b80: 2069 6d70 6f72 7473 2061 7265 206e 6f77   imports are now
-00013b90: 2073 7461 6e64 6172 6469 7a65 6420 6163   standardized ac
-00013ba0: 726f 7373 2074 6865 2063 6f64 6562 6173  ross the codebas
-00013bb0: 652e 0a2a 204e 6577 2077 6562 7369 7465  e..* New website
-00013bc0: 2066 726f 6e74 2070 6167 6520 616e 6420   front page and 
-00013bd0: 7374 796c 696e 6720 6368 616e 6765 7320  styling changes 
-00013be0: 7468 726f 7567 686f 7574 2e20 4d6f 7665  throughout. Move
-00013bf0: 6420 646f 6373 2073 6974 6520 746f 2069  d docs site to i
-00013c00: 7473 206f 776e 2076 6572 7369 6f6e 6564  ts own versioned
-00013c10: 2073 7562 6469 7265 6374 6f72 6965 732e   subdirectories.
-00013c20: 0a2a 2052 656f 7267 616e 697a 6564 2061  .* Reorganized a
-00013c30: 6c69 676e 6d65 6e74 2064 6174 6120 7374  lignment data st
-00013c40: 7275 6374 7572 6573 2061 6e64 2061 6c67  ructures and alg
-00013c50: 6f72 6974 686d 7320 2865 2e67 2e2c 2053  orithms (e.g., S
-00013c60: 5357 2063 6f64 652c 2060 6041 6c69 676e  SW code, ``Align
-00013c70: 6d65 6e74 6060 2063 6c61 7373 2c20 6574  ment`` class, et
-00013c80: 632e 2920 696e 746f 2061 6e20 6060 736b  c.) into an ``sk
-00013c90: 6269 6f2e 636f 7265 2e61 6c69 676e 6d65  bio.core.alignme
-00013ca0: 6e74 6060 2073 7562 7061 636b 6167 652e  nt`` subpackage.
-00013cb0: 0a0a 2323 2056 6572 7369 6f6e 2030 2e31  ..## Version 0.1
-00013cc0: 2e31 2028 3230 3134 2d30 352d 3136 290a  .1 (2014-05-16).
-00013cd0: 0a46 6978 6573 2074 6f20 7365 7475 702e  .Fixes to setup.
-00013ce0: 7079 2e20 5468 6973 2069 7320 6120 7072  py. This is a pr
-00013cf0: 652d 616c 7068 6120 7265 6c65 6173 652e  e-alpha release.
-00013d00: 2041 7420 7468 6973 2073 7461 6765 2c20   At this stage, 
-00013d10: 6d61 6a6f 7220 6261 636b 7761 7264 732d  major backwards-
-00013d20: 696e 636f 6d70 6174 6962 6c65 2041 5049  incompatible API
-00013d30: 2063 6861 6e67 6573 2063 616e 2061 6e64   changes can and
-00013d40: 2077 696c 6c20 6861 7070 656e 2e0a 0a23   will happen...#
-00013d50: 2320 5665 7273 696f 6e20 302e 312e 3020  # Version 0.1.0 
-00013d60: 2832 3031 342d 3035 2d31 3529 0a0a 496e  (2014-05-15)..In
-00013d70: 6974 6961 6c20 7072 652d 616c 7068 6120  itial pre-alpha 
-00013d80: 7265 6c65 6173 652e 2041 7420 7468 6973  release. At this
-00013d90: 2073 7461 6765 2c20 6d61 6a6f 7220 6261   stage, major ba
-00013da0: 636b 7761 7264 732d 696e 636f 6d70 6174  ckwards-incompat
-00013db0: 6962 6c65 2041 5049 2063 6861 6e67 6573  ible API changes
-00013dc0: 2063 616e 2061 6e64 2077 696c 6c20 6861   can and will ha
-00013dd0: 7070 656e 2e0a                           ppen..
+00002940: 6f6e 616c 6974 7920 5b73 7461 626c 655d  onality [stable]
+00002950: 0a0a 2323 2320 4465 7072 6563 6174 6564  ..### Deprecated
+00002960: 2066 756e 6374 696f 6e61 6c69 7479 205b   functionality [
+00002970: 6578 7065 7269 6d65 6e74 616c 5d0a 0a23  experimental]..#
+00002980: 2323 204d 6973 6365 6c6c 616e 656f 7573  ## Miscellaneous
+00002990: 0a2a 2050 7974 686f 6e20 332e 3620 616e  .* Python 3.6 an
+000029a0: 6420 332e 3720 636f 6d70 6174 6962 696c  d 3.7 compatibil
+000029b0: 6974 7920 6973 206e 6f77 2073 7570 706f  ity is now suppo
+000029c0: 7274 6564 0a0a 2a20 4120 7079 7465 7374  rted..* A pytest
+000029d0: 2072 756e 6e65 7220 6973 2073 6869 7070   runner is shipp
+000029e0: 6564 2077 6974 6820 6576 6572 7920 696e  ed with every in
+000029f0: 7374 616c 6c61 7469 6f6e 2028 5b23 3136  stallation ([#16
+00002a00: 3333 5d28 6874 7470 733a 2f2f 6769 7468  33](https://gith
+00002a10: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+00002a20: 6369 6b69 742d 6269 6f2f 7075 6c6c 2f31  cikit-bio/pull/1
+00002a30: 3633 3329 290a 0a2a 2054 6865 206e 6f73  633))..* The nos
+00002a40: 6574 6573 7420 6672 616d 6577 6f72 6b20  etest framework 
+00002a50: 6861 7320 6265 656e 2072 6570 6c61 6365  has been replace
+00002a60: 6420 696e 2066 6176 6f72 206f 6620 7079  d in favor of py
+00002a70: 7465 7374 2028 5b23 3136 3234 5d28 6874  test ([#1624](ht
+00002a80: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+00002a90: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
+00002aa0: 6269 6f2f 7075 6c6c 2f31 3632 3429 290a  bio/pull/1624)).
+00002ab0: 0a2a 2054 6865 206e 756d 7079 2064 6f63  .* The numpy doc
+00002ac0: 7320 6172 6520 6465 7072 6563 6174 6564  s are deprecated
+00002ad0: 2069 6e20 6661 766f 7220 6f66 205b 4e61   in favor of [Na
+00002ae0: 706f 6c65 6f6e 5d28 6874 7470 3a2f 2f77  poleon](http://w
+00002af0: 7777 2e73 7068 696e 782d 646f 632e 6f72  ww.sphinx-doc.or
+00002b00: 672f 656e 2f6d 6173 7465 722f 7573 6167  g/en/master/usag
+00002b10: 652f 6578 7465 6e73 696f 6e73 2f6e 6170  e/extensions/nap
+00002b20: 6f6c 656f 6e2e 6874 6d6c 2920 285b 2331  oleon.html) ([#1
+00002b30: 3632 395d 2868 7474 7073 3a2f 2f67 6974  629](https://git
+00002b40: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+00002b50: 7363 696b 6974 2d62 696f 2f70 756c 6c2f  scikit-bio/pull/
+00002b60: 3136 3239 2929 0a0a 2a20 5468 6973 2076  1629))..* This v
+00002b70: 6572 7369 6f6e 2069 7320 6e6f 7720 636f  ersion is now co
+00002b80: 6d70 6174 6962 6c65 2077 6974 6820 4e75  mpatible with Nu
+00002b90: 6d50 7920 3e3d 2031 2e39 2e32 2061 6e64  mPy >= 1.9.2 and
+00002ba0: 2050 616e 6461 7320 3e3d 2030 2e32 332e   Pandas >= 0.23.
+00002bb0: 2028 5b23 3136 3237 5d28 6874 7470 733a   ([#1627](https:
+00002bc0: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
+00002bd0: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
+00002be0: 7075 6c6c 2f31 3632 3729 290a 0a23 2320  pull/1627))..## 
+00002bf0: 5665 7273 696f 6e20 302e 352e 3420 2832  Version 0.5.4 (2
+00002c00: 3031 382d 3038 2d32 3329 0a0a 2323 2320  018-08-23)..### 
+00002c10: 4665 6174 7572 6573 0a0a 2a20 4164 6465  Features..* Adde
+00002c20: 6420 6046 5356 4460 2c20 616e 2061 6c74  d `FSVD`, an alt
+00002c30: 6572 6e61 7469 7665 2066 6173 7420 6865  ernative fast he
+00002c40: 7572 6973 7469 6320 6d65 7468 6f64 2074  uristic method t
+00002c50: 6f20 7065 7266 6f72 6d20 5072 696e 6369  o perform Princi
+00002c60: 7061 6c20 436f 6f72 6469 6e61 7465 7320  pal Coordinates 
+00002c70: 416e 616c 7973 6973 2c20 746f 2060 736b  Analysis, to `sk
+00002c80: 6269 6f2e 7374 6174 732e 6f72 6469 6e61  bio.stats.ordina
+00002c90: 7469 6f6e 2e70 636f 6160 2e0a 0a23 2323  tion.pcoa`...###
+00002ca0: 2042 6163 6b77 6172 642d 696e 636f 6d70   Backward-incomp
+00002cb0: 6174 6962 6c65 2063 6861 6e67 6573 205b  atible changes [
+00002cc0: 7374 6162 6c65 5d0a 0a23 2323 2042 6163  stable]..### Bac
+00002cd0: 6b77 6172 642d 696e 636f 6d70 6174 6962  kward-incompatib
+00002ce0: 6c65 2063 6861 6e67 6573 205b 6578 7065  le changes [expe
+00002cf0: 7269 6d65 6e74 616c 5d0a 0a23 2323 2050  rimental]..### P
+00002d00: 6572 666f 726d 616e 6365 2065 6e68 616e  erformance enhan
+00002d10: 6365 6d65 6e74 730a 0a2a 2041 6464 6564  cements..* Added
+00002d20: 206f 7074 696d 697a 6564 2075 7469 6c69   optimized utili
+00002d30: 7479 206d 6574 686f 6473 2060 665f 6d61  ty methods `f_ma
+00002d40: 7472 6978 5f69 6e70 6c61 6365 6020 616e  trix_inplace` an
+00002d50: 6420 6065 5f6d 6174 7269 785f 696e 706c  d `e_matrix_inpl
+00002d60: 6163 6560 2077 6869 6368 2070 6572 666f  ace` which perfo
+00002d70: 726d 2060 665f 6d61 7472 6978 6020 616e  rm `f_matrix` an
+00002d80: 6420 6065 5f6d 6174 7269 7860 2063 6f6d  d `e_matrix` com
+00002d90: 7075 7461 7469 6f6e 7320 696e 2d70 6c61  putations in-pla
+00002da0: 6365 2061 6e64 2061 7265 2075 7365 6420  ce and are used 
+00002db0: 6279 2074 6865 206e 6577 2060 6365 6e74  by the new `cent
+00002dc0: 6572 5f64 6973 7461 6e63 655f 6d61 7472  er_distance_matr
+00002dd0: 6978 6020 6d65 7468 6f64 2069 6e20 6073  ix` method in `s
+00002de0: 6b62 696f 2e73 7461 7473 2e6f 7264 696e  kbio.stats.ordin
+00002df0: 6174 696f 6e60 2e0a 0a23 2323 2042 7567  ation`...### Bug
+00002e00: 2066 6978 6573 0a0a 2323 2320 4465 7072   fixes..### Depr
+00002e10: 6563 6174 6564 2066 756e 6374 696f 6e61  ecated functiona
+00002e20: 6c69 7479 205b 7374 6162 6c65 5d0a 0a23  lity [stable]..#
+00002e30: 2323 2044 6570 7265 6361 7465 6420 6675  ## Deprecated fu
+00002e40: 6e63 7469 6f6e 616c 6974 7920 5b65 7870  nctionality [exp
+00002e50: 6572 696d 656e 7461 6c5d 0a0a 2323 2320  erimental]..### 
+00002e60: 4d69 7363 656c 6c61 6e65 6f75 730a 0a23  Miscellaneous..#
+00002e70: 2320 5665 7273 696f 6e20 302e 352e 3320  # Version 0.5.3 
+00002e80: 2832 3031 382d 3038 2d30 3729 0a0a 2323  (2018-08-07)..##
+00002e90: 2320 4665 6174 7572 6573 0a2a 2041 6464  # Features.* Add
+00002ea0: 6564 2060 756e 7061 636b 6020 616e 6420  ed `unpack` and 
+00002eb0: 6075 6e70 6163 6b5f 6279 5f66 756e 6360  `unpack_by_func`
+00002ec0: 206d 6574 686f 6473 2074 6f20 6073 6b62   methods to `skb
+00002ed0: 696f 2e74 7265 652e 5472 6565 4e6f 6465  io.tree.TreeNode
+00002ee0: 6020 746f 2075 6e70 6163 6b20 6f6e 6520  ` to unpack one 
+00002ef0: 6f72 206d 756c 7469 706c 6520 696e 7465  or multiple inte
+00002f00: 726e 616c 206e 6f64 6573 2e20 5468 6520  rnal nodes. The 
+00002f10: 6075 6e70 6163 6b60 206f 7065 7261 7469  `unpack` operati
+00002f20: 6f6e 2072 656d 6f76 6573 2061 6e20 696e  on removes an in
+00002f30: 7465 726e 616c 206e 6f64 6520 616e 6420  ternal node and 
+00002f40: 7265 6772 6166 7473 2069 7473 2063 6869  regrafts its chi
+00002f50: 6c64 7265 6e20 746f 2069 7473 2070 6172  ldren to its par
+00002f60: 656e 7420 7768 696c 6520 7265 7461 696e  ent while retain
+00002f70: 696e 6720 7468 6520 6f76 6572 616c 6c20  ing the overall 
+00002f80: 6c65 6e67 7468 2e20 285b 2331 3537 325d  length. ([#1572]
+00002f90: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+00002fa0: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+00002fb0: 6974 2d62 696f 2f70 756c 6c2f 3135 3732  it-bio/pull/1572
+00002fc0: 2929 0a2a 2041 6464 6564 2060 7375 7070  )).* Added `supp
+00002fd0: 6f72 7460 2074 6f20 6073 6b62 696f 2e74  ort` to `skbio.t
+00002fe0: 7265 652e 5472 6565 4e6f 6465 6020 746f  ree.TreeNode` to
+00002ff0: 2072 6574 7572 6e20 7468 6520 7375 7070   return the supp
+00003000: 6f72 7420 7661 6c75 6520 6f66 2061 206e  ort value of a n
+00003010: 6f64 652e 0a2a 2041 6464 6564 2060 7065  ode..* Added `pe
+00003020: 726d 6469 7370 6020 746f 2060 736b 6269  rmdisp` to `skbi
+00003030: 6f2e 7374 6174 732e 6469 7374 616e 6365  o.stats.distance
+00003040: 6020 746f 2074 6573 7420 666f 7220 7468  ` to test for th
+00003050: 6520 686f 6d6f 6765 6e69 6574 7920 6f66  e homogeniety of
+00003060: 2067 726f 7570 732e 2028 5b23 3132 3238   groups. ([#1228
+00003070: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+00003080: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
+00003090: 6b69 742d 6269 6f2f 6973 7375 6573 2f31  kit-bio/issues/1
+000030a0: 3232 3829 292e 0a0a 2a20 4164 6465 6420  228))...* Added 
+000030b0: 6070 636f 615f 6269 706c 6f74 6020 746f  `pcoa_biplot` to
+000030c0: 2060 736b 6269 6f2e 7374 6174 732e 6f72   `skbio.stats.or
+000030d0: 6469 6e61 7469 6f6e 6020 746f 2070 726f  dination` to pro
+000030e0: 6a65 6374 2064 6573 6372 6970 746f 7273  ject descriptors
+000030f0: 2069 6e74 6f20 6120 5043 6f41 2070 6c6f   into a PCoA plo
+00003100: 742e 0a0a 2a20 4669 7865 6420 7061 6e64  t...* Fixed pand
+00003110: 6173 2074 6f20 302e 3232 2e30 2064 7565  as to 0.22.0 due
+00003120: 2074 6f20 7468 6973 3a20 6874 7470 733a   to this: https:
+00003130: 2f2f 6769 7468 7562 2e63 6f6d 2f70 616e  //github.com/pan
+00003140: 6461 732d 6465 762f 7061 6e64 6173 2f69  das-dev/pandas/i
+00003150: 7373 7565 732f 3230 3532 370a 0a23 2323  ssues/20527..###
+00003160: 2042 6163 6b77 6172 642d 696e 636f 6d70   Backward-incomp
+00003170: 6174 6962 6c65 2063 6861 6e67 6573 205b  atible changes [
+00003180: 7374 6162 6c65 5d0a 0a23 2323 2042 6163  stable]..### Bac
+00003190: 6b77 6172 642d 696e 636f 6d70 6174 6962  kward-incompatib
+000031a0: 6c65 2063 6861 6e67 6573 205b 6578 7065  le changes [expe
+000031b0: 7269 6d65 6e74 616c 5d0a 0a23 2323 2050  rimental]..### P
+000031c0: 6572 666f 726d 616e 6365 2065 6e68 616e  erformance enhan
+000031d0: 6365 6d65 6e74 730a 0a23 2323 2042 7567  cements..### Bug
+000031e0: 2066 6978 6573 0a0a 2a20 5265 6c61 7869   fixes..* Relaxi
+000031f0: 6e67 2074 7970 6520 6368 6563 6b69 6e67  ng type checking
+00003200: 2069 6e20 6469 7665 7273 6974 7920 6361   in diversity ca
+00003210: 6c63 756c 6174 696f 6e73 2e20 2028 5b23  lculations.  ([#
+00003220: 3135 3833 5d28 6874 7470 733a 2f2f 6769  1583](https://gi
+00003230: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
+00003240: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
+00003250: 6573 2f31 3538 3329 292e 0a0a 2323 2320  es/1583))...### 
+00003260: 4465 7072 6563 6174 6564 2066 756e 6374  Deprecated funct
+00003270: 696f 6e61 6c69 7479 205b 7374 6162 6c65  ionality [stable
+00003280: 5d0a 0a23 2323 2044 6570 7265 6361 7465  ]..### Deprecate
+00003290: 6420 6675 6e63 7469 6f6e 616c 6974 7920  d functionality 
+000032a0: 5b65 7870 6572 696d 656e 7461 6c5d 0a0a  [experimental]..
+000032b0: 2323 2320 4d69 7363 656c 6c61 6e65 6f75  ### Miscellaneou
+000032c0: 730a 0a0a 2323 2056 6572 7369 6f6e 2030  s...## Version 0
+000032d0: 2e35 2e32 2028 3230 3138 2d30 342d 3138  .5.2 (2018-04-18
+000032e0: 290a 0a23 2323 2046 6561 7475 7265 730a  )..### Features.
+000032f0: 2a20 4164 6465 6420 6060 736b 6269 6f2e  * Added ``skbio.
+00003300: 696f 2e66 6f72 6d61 742e 656d 626c 6060  io.format.embl``
+00003310: 2066 6f72 2072 6561 6469 6e67 2061 6e64   for reading and
+00003320: 2077 7269 7469 6e67 2045 4d42 4c20 6669   writing EMBL fi
+00003330: 6c65 7320 666f 7220 6060 444e 4160 602c  les for ``DNA``,
+00003340: 2060 6052 4e41 6060 2061 6e64 2060 6053   ``RNA`` and ``S
+00003350: 6571 7565 6e63 6560 6020 636c 6173 7365  equence`` classe
+00003360: 732e 0a0a 2a20 5265 6d6f 7669 6e67 2056  s...* Removing V
+00003370: 616c 7565 4572 726f 7220 6368 6563 6b20  alueError check 
+00003380: 696e 2060 736b 6269 6f2e 7374 6174 732e  in `skbio.stats.
+00003390: 5f73 7562 7361 6d70 6c65 2e73 7562 7361  _subsample.subsa
+000033a0: 6d70 6c65 5f63 6f75 6e74 7360 2077 6865  mple_counts` whe
+000033b0: 6e20 6072 6570 6c61 6365 3d54 7275 6560  n `replace=True`
+000033c0: 2061 6e64 2060 6e60 2069 7320 6772 6561   and `n` is grea
+000033d0: 7465 7220 7468 616e 2074 6865 206e 756d  ter than the num
+000033e0: 6265 7220 6f66 2069 7465 6d73 2069 6e20  ber of items in 
+000033f0: 636f 756e 7473 2e20 205b 2331 3532 375d  counts.  [#1527]
+00003400: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+00003410: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+00003420: 6974 2d62 696f 2f70 756c 6c2f 3135 3237  it-bio/pull/1527
+00003430: 290a 0a2a 2041 6464 6564 2060 6073 6b62  )..* Added ``skb
+00003440: 696f 2e69 6f2e 666f 726d 6174 2e67 6666  io.io.format.gff
+00003450: 3360 6020 666f 7220 7265 6164 696e 6720  3`` for reading 
+00003460: 616e 6420 7772 6974 696e 6720 4746 4633  and writing GFF3
+00003470: 2066 696c 6573 2066 6f72 2060 6044 4e41   files for ``DNA
+00003480: 6060 2c20 6060 5365 7175 656e 6365 6060  ``, ``Sequence``
+00003490: 2c20 616e 6420 6060 496e 7465 7276 616c  , and ``Interval
+000034a0: 4d65 7461 6461 7461 6060 2063 6c61 7373  Metadata`` class
+000034b0: 6573 2e20 285b 2331 3435 305d 2868 7474  es. ([#1450](htt
+000034c0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+000034d0: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
+000034e0: 696f 2f70 756c 6c2f 3134 3530 2929 0a0a  io/pull/1450))..
+000034f0: 2a20 6073 6b62 696f 2e6d 6574 6164 6174  * `skbio.metadat
+00003500: 612e 496e 7465 7276 616c 4d65 7461 6461  a.IntervalMetada
+00003510: 7461 6020 636f 6e73 7472 7563 746f 7220  ta` constructor 
+00003520: 6861 7320 6120 6e65 7720 6b65 7977 6f72  has a new keywor
+00003530: 6420 6172 6775 6d65 6e74 2c20 6063 6f70  d argument, `cop
+00003540: 795f 6672 6f6d 602c 2066 6f72 2063 7265  y_from`, for cre
+00003550: 6174 696e 6720 616e 2060 496e 7465 7276  ating an `Interv
+00003560: 616c 4d65 7461 6461 7461 6020 6f62 6a65  alMetadata` obje
+00003570: 6374 2066 726f 6d20 616e 2065 7869 7374  ct from an exist
+00003580: 696e 6720 6049 6e74 6572 7661 6c4d 6574  ing `IntervalMet
+00003590: 6164 6174 6160 206f 626a 6563 7420 7769  adata` object wi
+000035a0: 7468 2073 7065 6369 6669 6564 2060 7570  th specified `up
+000035b0: 7065 725f 626f 756e 6460 2e0a 0a2a 2060  per_bound`...* `
+000035c0: 736b 6269 6f2e 6d65 7461 6461 7461 2e49  skbio.metadata.I
+000035d0: 6e74 6572 7661 6c4d 6574 6164 6174 6160  ntervalMetadata`
+000035e0: 2063 6f6e 7374 7275 6374 6f72 2061 6c6c   constructor all
+000035f0: 6f77 7320 604e 6f6e 6560 2061 7320 6120  ows `None` as a 
+00003600: 7661 6c69 6420 7661 6c75 6520 666f 7220  valid value for 
+00003610: 6075 7070 6572 5f62 6f75 6e64 602e 2041  `upper_bound`. A
+00003620: 6e20 6075 7070 6572 5f62 6f75 6e64 6020  n `upper_bound` 
+00003630: 6f66 2060 4e6f 6e65 6020 6d65 616e 7320  of `None` means 
+00003640: 7468 6174 2074 6865 2060 496e 7465 7276  that the `Interv
+00003650: 616c 4d65 7461 6461 7461 6020 6f62 6a65  alMetadata` obje
+00003660: 6374 2068 6173 206e 6f20 7570 7065 7220  ct has no upper 
+00003670: 626f 756e 642e 0a0a 2a20 6073 6b62 696f  bound...* `skbio
+00003680: 2e6d 6574 6164 6174 612e 496e 7465 7276  .metadata.Interv
+00003690: 616c 4d65 7461 6461 7461 2e64 726f 7060  alMetadata.drop`
+000036a0: 2068 6173 2061 206e 6577 2062 6f6f 6c65   has a new boole
+000036b0: 616e 2070 6172 616d 6574 6572 2060 6e65  an parameter `ne
+000036c0: 6761 7465 6020 746f 2069 6e64 6963 6174  gate` to indicat
+000036d0: 6520 7768 6574 6865 7220 746f 2064 726f  e whether to dro
+000036e0: 7020 6f72 206b 6565 7020 7468 6520 7370  p or keep the sp
+000036f0: 6563 6966 6965 6420 6049 6e74 6572 7661  ecified `Interva
+00003700: 6c60 206f 626a 6563 7473 2e0a 0a23 2323  l` objects...###
+00003710: 2042 6163 6b77 6172 642d 696e 636f 6d70   Backward-incomp
+00003720: 6174 6962 6c65 2063 6861 6e67 6573 205b  atible changes [
+00003730: 7374 6162 6c65 5d0a 0a23 2323 2042 6163  stable]..### Bac
+00003740: 6b77 6172 642d 696e 636f 6d70 6174 6962  kward-incompatib
+00003750: 6c65 2063 6861 6e67 6573 205b 6578 7065  le changes [expe
+00003760: 7269 6d65 6e74 616c 5d0a 0a23 2323 2050  rimental]..### P
+00003770: 6572 666f 726d 616e 6365 2065 6e68 616e  erformance enhan
+00003780: 6365 6d65 6e74 730a 2a20 6073 6b62 696f  cements.* `skbio
+00003790: 2e74 7265 652e 6e6a 6020 7761 6c6c 2d63  .tree.nj` wall-c
+000037a0: 6c6f 636b 2072 756e 7469 6d65 2077 6173  lock runtime was
+000037b0: 2064 6563 7265 6173 6564 2062 7920 3939   decreased by 99
+000037c0: 2520 666f 7220 6120 3530 3078 3530 3020  % for a 500x500 
+000037d0: 6469 7374 616e 6365 206d 6174 7269 7820  distance matrix 
+000037e0: 616e 6420 3933 2520 666f 7220 6120 3130  and 93% for a 10
+000037f0: 3078 3130 3020 6469 7374 616e 6365 206d  0x100 distance m
+00003800: 6174 7269 782e 2028 5b23 3135 3132 5d28  atrix. ([#1512](
+00003810: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+00003820: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+00003830: 742d 6269 6f2f 7075 6c6c 2f31 3531 3229  t-bio/pull/1512)
+00003840: 2c20 5b23 3135 3133 5d28 6874 7470 733a  , [#1513](https:
+00003850: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
+00003860: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
+00003870: 7075 6c6c 2f31 3531 3329 290a 0a23 2323  pull/1513))..###
+00003880: 2042 7567 2066 6978 6573 0a2a 2054 6865   Bug fixes.* The
+00003890: 2060 696e 636c 7564 655f 7365 6c66 6020   `include_self` 
+000038a0: 7061 7261 6d65 7465 7220 7761 7320 6e6f  parameter was no
+000038b0: 7420 6265 696e 6720 686f 6e6f 7265 6420  t being honored 
+000038c0: 696e 2060 736b 6269 6f2e 5472 6565 4e6f  in `skbio.TreeNo
+000038d0: 6465 2e74 6970 7360 2e20 5468 6520 7363  de.tips`. The sc
+000038e0: 6f70 6520 6f66 2074 6869 7320 6275 6720  ope of this bug 
+000038f0: 7761 7320 7468 6174 2069 6620 6054 7265  was that if `Tre
+00003900: 654e 6f64 652e 7469 7073 6020 7761 7320  eNode.tips` was 
+00003910: 6361 6c6c 6564 206f 6e20 6120 7469 702c  called on a tip,
+00003920: 2069 7420 776f 756c 6420 616c 7761 7973   it would always
+00003930: 2072 6573 756c 7420 696e 2061 6e20 656d   result in an em
+00003940: 7074 7920 606c 6973 7460 2077 6865 6e20  pty `list` when 
+00003950: 756e 726f 6c6c 6564 2e0a 0a2a 2049 6e20  unrolled...* In 
+00003960: 6073 6b62 696f 2e73 7461 7473 2e6f 7264  `skbio.stats.ord
+00003970: 696e 6174 696f 6e2e 6361 602c 2060 7072  ination.ca`, `pr
+00003980: 6f70 6f72 7469 6f6e 5f65 7870 6c61 696e  oportion_explain
+00003990: 6564 6020 7761 7320 6d69 7373 696e 6720  ed` was missing 
+000039a0: 696e 2074 6865 2072 6574 7572 6e65 6420  in the returned 
+000039b0: 604f 7264 696e 6174 696f 6e52 6573 756c  `OrdinationResul
+000039c0: 7473 6020 6f62 6a65 6374 2e20 285b 2331  ts` object. ([#1
+000039d0: 3334 355d 2868 7474 7073 3a2f 2f67 6974  345](https://git
+000039e0: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+000039f0: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
+00003a00: 732f 3133 3435 2929 0a0a 2a20 6073 6b62  s/1345))..* `skb
+00003a10: 696f 2e64 6976 6572 7369 7479 2e62 6574  io.diversity.bet
+00003a20: 615f 6469 7665 7273 6974 7960 206e 6f77  a_diversity` now
+00003a30: 2068 616e 646c 6573 2071 7561 6c69 7461   handles qualita
+00003a40: 7469 7665 206d 6574 7269 6373 2061 7320  tive metrics as 
+00003a50: 6578 7065 6374 6564 2073 7563 6820 7468  expected such th
+00003a60: 6174 2060 6265 7461 5f64 6976 6572 7369  at `beta_diversi
+00003a70: 7479 2827 6a61 6363 6172 6427 2c20 6d61  ty('jaccard', ma
+00003a80: 7429 203d 3d20 6265 7461 5f64 6976 6572  t) == beta_diver
+00003a90: 7369 7479 2827 6a61 6363 6172 6427 2c20  sity('jaccard', 
+00003aa0: 6d61 7420 3e20 3029 602e 2050 6c65 6173  mat > 0)`. Pleas
+00003ab0: 6520 7365 6520 5b23 3135 3439 5d28 6874  e see [#1549](ht
+00003ac0: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+00003ad0: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
+00003ae0: 6269 6f2f 6973 7375 6573 2f31 3534 3929  bio/issues/1549)
+00003af0: 2066 6f72 2066 7572 7468 6572 2064 6574   for further det
+00003b00: 6169 6c2e 0a0a 2a20 6073 6b62 696f 2e73  ail...* `skbio.s
+00003b10: 7461 7473 2e6f 7264 696e 6174 696f 6e2e  tats.ordination.
+00003b20: 7264 6160 2054 6865 206f 6363 6173 696f  rda` The occasio
+00003b30: 6e61 6c20 636f 6c75 6d6e 206d 6973 6d61  nal column misma
+00003b40: 7463 6820 696e 206f 7574 7075 7420 6062  tch in output `b
+00003b50: 6970 6c6f 745f 7363 6f72 6573 6020 6973  iplot_scores` is
+00003b60: 2066 6978 6564 2028 5b23 3135 3139 5d28   fixed ([#1519](
+00003b70: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+00003b80: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+00003b90: 742d 6269 6f2f 6973 7375 6573 2f31 3531  t-bio/issues/151
+00003ba0: 3929 292e 0a0a 2323 2320 4465 7072 6563  9))...### Deprec
+00003bb0: 6174 6564 2066 756e 6374 696f 6e61 6c69  ated functionali
+00003bc0: 7479 205b 7374 6162 6c65 5d0a 0a23 2323  ty [stable]..###
+00003bd0: 2044 6570 7265 6361 7465 6420 6675 6e63   Deprecated func
+00003be0: 7469 6f6e 616c 6974 7920 5b65 7870 6572  tionality [exper
+00003bf0: 696d 656e 7461 6c5d 0a0a 2323 2320 4d69  imental]..### Mi
+00003c00: 7363 656c 6c61 6e65 6f75 730a 2a20 7363  scellaneous.* sc
+00003c10: 696b 6974 2d62 696f 206e 6f77 2064 6570  ikit-bio now dep
+00003c20: 656e 6473 206f 6e20 7061 6e64 6173 203e  ends on pandas >
+00003c30: 3d20 302e 3139 2e32 2c20 616e 6420 6973  = 0.19.2, and is
+00003c40: 2063 6f6d 7061 7469 626c 6520 7769 7468   compatible with
+00003c50: 206e 6577 6572 2070 616e 6461 7320 7665   newer pandas ve
+00003c60: 7273 696f 6e73 2028 652e 672e 2030 2e32  rsions (e.g. 0.2
+00003c70: 302e 3329 2074 6861 7420 7765 7265 2070  0.3) that were p
+00003c80: 7265 7669 6f75 736c 7920 696e 636f 6d70  reviously incomp
+00003c90: 6174 6962 6c65 2e0a 2a20 7363 696b 6974  atible..* scikit
+00003ca0: 2d62 696f 206e 6f77 2064 6570 656e 6473  -bio now depends
+00003cb0: 206f 6e20 606e 756d 7079 203e 3d20 312e   on `numpy >= 1.
+00003cc0: 392e 322c 203c 2031 2e31 342e 3060 2066  9.2, < 1.14.0` f
+00003cd0: 6f72 2063 6f6d 7061 7469 6269 6c69 7479  or compatibility
+00003ce0: 2077 6974 6820 5079 7468 6f6e 2033 2e34   with Python 3.4
+00003cf0: 2c20 332e 352c 2061 6e64 2033 2e36 2061  , 3.5, and 3.6 a
+00003d00: 6e64 2074 6865 2061 7661 696c 6162 6c65  nd the available
+00003d10: 206e 756d 7079 2063 6f6e 6461 2070 6163   numpy conda pac
+00003d20: 6b61 6765 7320 696e 2060 6465 6661 756c  kages in `defaul
+00003d30: 7473 6020 616e 6420 6063 6f6e 6461 2d66  ts` and `conda-f
+00003d40: 6f72 6765 6020 6368 616e 6e65 6c73 2e0a  orge` channels..
+00003d50: 2a20 6164 6465 6420 7375 7070 6f72 7420  * added support 
+00003d60: 666f 7220 7275 6e6e 696e 6720 7465 7374  for running test
+00003d70: 7320 6672 6f6d 2060 7365 7475 702e 7079  s from `setup.py
+00003d80: 602e 2042 6f74 6820 6070 7974 686f 6e20  `. Both `python 
+00003d90: 7365 7475 702e 7079 206e 6f73 6574 6573  setup.py nosetes
+00003da0: 7473 6020 616e 6420 6070 7974 686f 6e20  ts` and `python 
+00003db0: 7365 7475 702e 7079 2074 6573 7460 2061  setup.py test` a
+00003dc0: 7265 206e 6f77 2073 7570 706f 7274 6564  re now supported
+00003dd0: 2c20 686f 7765 7665 7220 6070 7974 686f  , however `pytho
+00003de0: 6e20 7365 7475 702e 7079 2074 6573 7460  n setup.py test`
+00003df0: 2077 696c 6c20 6f6e 6c79 2072 756e 2061   will only run a
+00003e00: 2073 7562 7365 7420 6f66 2074 6865 2066   subset of the f
+00003e10: 756c 6c20 7465 7374 2073 7569 7465 2e20  ull test suite. 
+00003e20: 285b 2331 3334 315d 2868 7474 7073 3a2f  ([#1341](https:/
+00003e30: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
+00003e40: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
+00003e50: 7373 7565 732f 3133 3431 2929 0a0a 2323  ssues/1341))..##
+00003e60: 2056 6572 7369 6f6e 2030 2e35 2e31 2028   Version 0.5.1 (
+00003e70: 3230 3136 2d31 312d 3132 290a 0a23 2323  2016-11-12)..###
+00003e80: 2046 6561 7475 7265 730a 2a20 4164 6465   Features.* Adde
+00003e90: 6420 6049 6e74 6572 7661 6c4d 6574 6164  d `IntervalMetad
+00003ea0: 6174 6160 2061 6e64 2060 496e 7465 7276  ata` and `Interv
+00003eb0: 616c 6020 636c 6173 7365 7320 696e 2060  al` classes in `
+00003ec0: 736b 6269 6f2e 6d65 7461 6461 7461 6020  skbio.metadata` 
+00003ed0: 746f 2073 746f 7265 2c20 7175 6572 792c  to store, query,
+00003ee0: 2061 6e64 206d 616e 6970 756c 6174 6520   and manipulate 
+00003ef0: 696e 666f 726d 6174 696f 6e20 6f66 2061  information of a
+00003f00: 2073 7562 2d72 6567 696f 6e20 6f66 2061   sub-region of a
+00003f10: 2073 6571 7565 6e63 652e 2028 5b23 3134   sequence. ([#14
+00003f20: 3134 5d28 6874 7470 733a 2f2f 6769 7468  14](https://gith
+00003f30: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+00003f40: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
+00003f50: 2f31 3431 3429 290a 2a20 6053 6571 7565  /1414)).* `Seque
+00003f60: 6e63 6560 2061 6e64 2069 7473 2063 6869  nce` and its chi
+00003f70: 6c64 2063 6c61 7373 6573 2028 696e 636c  ld classes (incl
+00003f80: 7564 696e 6720 6047 7261 6d6d 6172 6564  uding `Grammared
+00003f90: 5365 7175 656e 6365 602c 2060 524e 4160  Sequence`, `RNA`
+00003fa0: 2c20 6044 4e41 602c 2060 5072 6f74 6569  , `DNA`, `Protei
+00003fb0: 6e60 2920 6e6f 7720 6163 6365 7074 2060  n`) now accept `
+00003fc0: 496e 7465 7276 616c 4d65 7461 6461 7461  IntervalMetadata
+00003fd0: 6020 696e 2074 6865 6972 2063 6f6e 7374  ` in their const
+00003fe0: 7275 6374 6f72 2041 5049 2e20 536f 6d65  ructor API. Some
+00003ff0: 206f 6620 7468 6569 7220 7265 6c65 7661   of their releva
+00004000: 6e74 206d 6574 686f 6473 2061 7265 2061  nt methods are a
+00004010: 6c73 6f20 7570 6461 7465 6420 6163 636f  lso updated acco
+00004020: 7264 696e 676c 792e 2028 5b23 3134 3330  rdingly. ([#1430
+00004030: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+00004040: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
+00004050: 6b69 742d 6269 6f2f 7075 6c6c 2f31 3433  kit-bio/pull/143
+00004060: 3029 290a 2a20 4765 6e42 616e 6b20 7061  0)).* GenBank pa
+00004070: 7273 6572 206e 6f77 2072 6561 6473 2061  rser now reads a
+00004080: 6e64 2077 7269 7465 7320 6053 6571 7565  nd writes `Seque
+00004090: 6e63 6560 206f 7220 6974 7320 7375 6263  nce` or its subc
+000040a0: 6c61 7373 206f 626a 6563 7473 2077 6974  lass objects wit
+000040b0: 6820 6049 6e74 6572 7661 6c4d 6574 6164  h `IntervalMetad
+000040c0: 6174 6160 2e20 285b 2331 3434 305d 2868  ata`. ([#1440](h
+000040d0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+000040e0: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
+000040f0: 2d62 696f 2f70 756c 6c2f 3134 3430 2929  -bio/pull/1440))
+00004100: 0a2a 2060 4469 7373 696d 696c 6172 6974  .* `Dissimilarit
+00004110: 794d 6174 7269 7860 206e 6f77 2068 6173  yMatrix` now has
+00004120: 2061 206e 6577 2063 6f6e 7374 7275 6374   a new construct
+00004130: 6f72 206d 6574 686f 6420 6361 6c6c 6564  or method called
+00004140: 2060 6672 6f6d 5f69 7465 7261 626c 6560   `from_iterable`
+00004150: 2e20 285b 2331 3334 335d 2868 7474 7073  . ([#1343](https
+00004160: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
+00004170: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
+00004180: 2f69 7373 7565 732f 3133 3433 2929 2e0a  /issues/1343))..
+00004190: 2a20 6044 6973 7369 6d69 6c61 7269 7479  * `Dissimilarity
+000041a0: 4d61 7472 6978 6020 6e6f 7720 616c 6c6f  Matrix` now allo
+000041b0: 7773 206e 6f6e 2d68 6f6c 6c6f 7720 6d61  ws non-hollow ma
+000041c0: 7472 6963 6573 2e20 285b 2331 3334 335d  trices. ([#1343]
+000041d0: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+000041e0: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+000041f0: 6974 2d62 696f 2f69 7373 7565 732f 3133  it-bio/issues/13
+00004200: 3433 2929 2e0a 2a20 6044 6973 7461 6e63  43))..* `Distanc
+00004210: 654d 6174 7269 782e 6672 6f6d 5f69 7465  eMatrix.from_ite
+00004220: 7261 626c 6560 206e 6f77 2061 6363 6570  rable` now accep
+00004230: 7473 2061 2060 7661 6c69 6461 7465 3d54  ts a `validate=T
+00004240: 7275 6560 2070 6172 616d 6574 6572 2e20  rue` parameter. 
+00004250: 285b 2331 3334 335d 2868 7474 7073 3a2f  ([#1343](https:/
+00004260: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
+00004270: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
+00004280: 7373 7565 732f 3133 3433 2929 2e0a 2a20  ssues/1343))..* 
+00004290: 6060 4469 7374 616e 6365 4d61 7472 6978  ``DistanceMatrix
+000042a0: 6060 206e 6f77 2068 6173 2061 206e 6577  `` now has a new
+000042b0: 206d 6574 686f 6420 6361 6c6c 6564 2060   method called `
+000042c0: 6074 6f5f 7365 7269 6573 6060 2074 6f20  `to_series`` to 
+000042d0: 6372 6561 7465 2061 2060 6070 616e 6461  create a ``panda
+000042e0: 732e 5365 7269 6573 6060 2066 726f 6d20  s.Series`` from 
+000042f0: 6120 6060 4469 7374 616e 6365 4d61 7472  a ``DistanceMatr
+00004300: 6978 6060 2028 5b23 3133 3937 5d28 6874  ix`` ([#1397](ht
+00004310: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+00004320: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
+00004330: 6269 6f2f 6973 7375 6573 2f31 3339 3729  bio/issues/1397)
+00004340: 292e 0a2a 2041 6464 6564 2070 6172 616c  )..* Added paral
+00004350: 6c65 6c20 6265 7461 2064 6976 6572 7369  lel beta diversi
+00004360: 7479 2063 616c 6375 6c61 7469 6f6e 2073  ty calculation s
+00004370: 7570 706f 7274 2076 6961 2060 736b 6269  upport via `skbi
+00004380: 6f2e 6469 7665 7273 6974 792e 626c 6f63  o.diversity.bloc
+00004390: 6b5f 6265 7461 5f64 6976 6572 7369 7479  k_beta_diversity
+000043a0: 602e 2054 6865 2069 7373 7565 2061 6e64  `. The issue and
+000043b0: 2069 6465 6120 6973 2064 6973 6375 7373   idea is discuss
+000043c0: 6564 2069 6e20 285b 2331 3138 315d 2868  ed in ([#1181](h
+000043d0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+000043e0: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
+000043f0: 2d62 696f 2f69 7373 7565 732f 3131 3831  -bio/issues/1181
+00004400: 292c 2077 6869 6c65 2074 6865 2061 6374  ), while the act
+00004410: 7561 6c20 636f 6465 2063 6861 6e67 6573  ual code changes
+00004420: 2061 7265 2069 6e20 5b23 3133 3532 5d28   are in [#1352](
+00004430: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+00004440: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+00004450: 742d 6269 6f2f 7075 6c6c 2f31 3335 3229  t-bio/pull/1352)
+00004460: 292e 0a0a 0a23 2323 2042 6163 6b77 6172  )....### Backwar
+00004470: 642d 696e 636f 6d70 6174 6962 6c65 2063  d-incompatible c
+00004480: 6861 6e67 6573 205b 7374 6162 6c65 5d0a  hanges [stable].
+00004490: 2a20 5468 6520 636f 6e73 7472 7563 746f  * The constructo
+000044a0: 7220 4150 4920 666f 7220 6053 6571 7565  r API for `Seque
+000044b0: 6e63 6560 2061 6e64 2069 7473 2063 6869  nce` and its chi
+000044c0: 6c64 2063 6c61 7373 6573 2028 696e 636c  ld classes (incl
+000044d0: 7564 696e 6720 6047 7261 6d6d 6172 6564  uding `Grammared
+000044e0: 5365 7175 656e 6365 602c 2060 524e 4160  Sequence`, `RNA`
+000044f0: 2c20 6044 4e41 602c 2060 5072 6f74 6569  , `DNA`, `Protei
+00004500: 6e60 2920 6172 6520 6368 616e 6765 6420  n`) are changed 
+00004510: 6672 6f6d 2060 2873 6571 7565 6e63 652c  from `(sequence,
+00004520: 206d 6574 6164 6174 613d 4e6f 6e65 2c20   metadata=None, 
+00004530: 706f 7369 7469 6f6e 616c 5f6d 6574 6164  positional_metad
+00004540: 6174 613d 4e6f 6e65 2c20 6c6f 7765 7263  ata=None, lowerc
+00004550: 6173 653d 4661 6c73 6529 6020 746f 2060  ase=False)` to `
+00004560: 2873 6571 7565 6e63 652c 206d 6574 6164  (sequence, metad
+00004570: 6174 613d 4e6f 6e65 2c20 706f 7369 7469  ata=None, positi
+00004580: 6f6e 616c 5f6d 6574 6164 6174 613d 4e6f  onal_metadata=No
+00004590: 6e65 2c20 696e 7465 7276 616c 5f6d 6574  ne, interval_met
+000045a0: 6164 6174 613d 4e6f 6e65 2c20 6c6f 7765  adata=None, lowe
+000045b0: 7263 6173 653d 4661 6c73 6529 600a 0a20  rcase=False)`.. 
+000045c0: 2054 6865 2063 6861 6e67 6573 2061 7265   The changes are
+000045d0: 206d 6164 6520 746f 2061 6c6c 6f77 2074   made to allow t
+000045e0: 6865 7365 2063 6c61 7373 6573 2074 6f20  hese classes to 
+000045f0: 6164 6f70 7420 6049 6e74 6572 7661 6c4d  adopt `IntervalM
+00004600: 6574 6164 6174 6160 206f 626a 6563 7420  etadata` object 
+00004610: 666f 7220 696e 7465 7276 616c 2066 6561  for interval fea
+00004620: 7475 7265 7320 6f6e 2074 6865 2073 6571  tures on the seq
+00004630: 7565 6e63 652e 2054 6865 2060 696e 7465  uence. The `inte
+00004640: 7276 616c 5f6d 6574 6164 6174 6160 2070  rval_metadata` p
+00004650: 6172 616d 6574 6572 2069 7320 6164 6465  arameter is adde
+00004660: 6420 696d 6564 6961 7465 6c79 2061 6674  d imediately aft
+00004670: 6572 2060 706f 7369 7469 6f6e 616c 5f6d  er `positional_m
+00004680: 6574 6164 6174 6160 2069 6e73 7465 6164  etadata` instead
+00004690: 206f 6620 6170 7065 6e64 6564 2074 6f20   of appended to 
+000046a0: 7468 6520 656e 642c 2062 6563 6175 7365  the end, because
+000046b0: 2069 7420 6973 206d 6f72 6520 6e61 7475   it is more natu
+000046c0: 7261 6c20 616e 6420 6c6f 6769 6361 6c20  ral and logical 
+000046d0: 616e 642c 206d 6f72 6520 696d 706f 7274  and, more import
+000046e0: 616e 746c 792c 2062 6563 6175 7365 2069  antly, because i
+000046f0: 7420 6973 2075 6e6c 696b 656c 7920 696e  t is unlikely in
+00004700: 2070 7261 6374 6963 6520 746f 2062 7265   practice to bre
+00004710: 616b 2075 7365 7220 636f 6465 2e20 4120  ak user code. A 
+00004720: 7573 6572 2773 2063 6f64 6520 776f 756c  user's code woul
+00004730: 6420 6272 6561 6b20 6f6e 6c79 2069 6620  d break only if 
+00004740: 7468 6579 2068 6164 2073 7570 706c 6965  they had supplie
+00004750: 6420 606d 6574 6164 6174 6160 2c20 6070  d `metadata`, `p
+00004760: 6f73 7469 6f6e 616c 5f6d 6574 6164 6174  ostional_metadat
+00004770: 6160 2c20 616e 6420 606c 6f77 6572 6361  a`, and `lowerca
+00004780: 7365 6020 7061 7261 6d65 7465 7273 2070  se` parameters p
+00004790: 6f73 6974 696f 6e61 6c6c 792e 2049 6e20  ositionally. In 
+000047a0: 7468 6520 756e 6c69 6b65 6c79 2065 7665  the unlikely eve
+000047b0: 6e74 2074 6861 7420 7468 6973 2068 6170  nt that this hap
+000047c0: 7065 6e73 2c20 7573 6572 7320 7769 6c6c  pens, users will
+000047d0: 2067 6574 2061 6e20 6572 726f 7220 7465   get an error te
+000047e0: 6c6c 696e 6720 7468 656d 2061 2062 6f6f  lling them a boo
+000047f0: 6c20 6973 6e27 7420 6120 7661 6c69 6420  l isn't a valid 
+00004800: 6049 6e74 6572 7661 6c4d 6574 6164 6174  `IntervalMetadat
+00004810: 6160 2074 7970 652c 2073 6f20 6974 2077  a` type, so it w
+00004820: 6f6e 2774 2073 696c 656e 746c 7920 7072  on't silently pr
+00004830: 6f64 7563 6520 6275 6767 7920 6265 6861  oduce buggy beha
+00004840: 7669 6f72 2e0a 0a23 2323 2042 6163 6b77  vior...### Backw
+00004850: 6172 642d 696e 636f 6d70 6174 6962 6c65  ard-incompatible
+00004860: 2063 6861 6e67 6573 205b 6578 7065 7269   changes [experi
+00004870: 6d65 6e74 616c 5d0a 2a20 4d6f 6469 6679  mental].* Modify
+00004880: 696e 6720 6261 7369 7320 6861 6e64 6c69  ing basis handli
+00004890: 6e67 2069 6e20 6073 6b62 696f 2e73 7461  ng in `skbio.sta
+000048a0: 7473 2e63 6f6d 706f 7369 7469 6f6e 2e69  ts.composition.i
+000048b0: 6c72 5f69 6e76 6020 7072 696f 7220 746f  lr_inv` prior to
+000048c0: 2063 6865 636b 696e 6720 666f 7220 6f72   checking for or
+000048d0: 7468 6f67 6f6e 616c 6974 792e 2020 4e6f  thogonality.  No
+000048e0: 7720 7468 6520 6261 7369 7320 6973 2073  w the basis is s
+000048f0: 7472 6963 746c 7920 6173 7375 6d65 6420  trictly assumed 
+00004900: 746f 2062 6520 696e 2074 6865 2041 6974  to be in the Ait
+00004910: 6368 6973 6f6e 2073 696d 706c 6578 2e0a  chison simplex..
+00004920: 2a20 6044 6973 7461 6e63 654d 6174 7269  * `DistanceMatri
+00004930: 782e 6672 6f6d 5f69 7465 7261 626c 6560  x.from_iterable`
+00004940: 2064 6566 6175 6c74 2062 6568 6176 696f   default behavio
+00004950: 7220 6973 206e 6f77 2074 6f20 7661 6c69  r is now to vali
+00004960: 6461 7465 206d 6174 7269 7820 6279 2063  date matrix by c
+00004970: 6f6d 7075 7469 6e67 2061 6c6c 2070 6169  omputing all pai
+00004980: 7277 6973 6520 6469 7374 616e 6365 732e  rwise distances.
+00004990: 2050 6173 7320 6076 616c 6964 6174 653d   Pass `validate=
+000049a0: 4661 6c73 6560 2074 6f20 6765 7420 7468  False` to get th
+000049b0: 6520 7072 6576 696f 7573 2062 6568 6176  e previous behav
+000049c0: 696f 7220 286e 6f20 7661 6c69 6461 7469  ior (no validati
+000049d0: 6f6e 2c20 6275 7420 6661 7374 6572 2065  on, but faster e
+000049e0: 7865 6375 7469 6f6e 292e 285b 2331 3334  xecution).([#134
+000049f0: 335d 2868 7474 7073 3a2f 2f67 6974 6875  3](https://githu
+00004a00: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
+00004a10: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
+00004a20: 3133 3433 2929 2e0a 2a20 4765 6e42 616e  1343))..* GenBan
+00004a30: 6b20 492f 4f20 6e6f 7720 7061 7273 6573  k I/O now parses
+00004a40: 2073 6571 7565 6e63 6520 6665 6174 7572   sequence featur
+00004a50: 6573 2069 6e74 6f20 7468 6520 6174 7472  es into the attr
+00004a60: 6962 7574 6520 6f66 2060 696e 7465 7276  ibute of `interv
+00004a70: 616c 5f6d 6574 6164 6174 6160 2069 6e73  al_metadata` ins
+00004a80: 7465 6164 206f 6620 6070 6f73 6974 696f  tead of `positio
+00004a90: 6e61 5f6d 6574 6164 6174 6160 2e20 416e  na_metadata`. An
+00004aa0: 6420 7468 6520 6b65 7920 6f66 2060 4645  d the key of `FE
+00004ab0: 4154 5552 4553 6020 6973 2072 656d 6f76  ATURES` is remov
+00004ac0: 6564 2066 726f 6d20 606d 6574 6164 6174  ed from `metadat
+00004ad0: 6160 2061 7474 7269 6275 7465 2e0a 0a23  a` attribute...#
+00004ae0: 2323 2050 6572 666f 726d 616e 6365 2065  ## Performance e
+00004af0: 6e68 616e 6365 6d65 6e74 730a 2a20 6054  nhancements.* `T
+00004b00: 7265 654e 6f64 652e 7368 6561 7260 2077  reeNode.shear` w
+00004b10: 6173 2072 6577 7269 7474 656e 2066 6f72  as rewritten for
+00004b20: 2061 7070 726f 7869 6d61 7465 6c79 2061   approximately a
+00004b30: 2032 3525 2070 6572 666f 726d 616e 6365   25% performance
+00004b40: 2069 6e63 7265 6173 652e 2028 5b23 3133   increase. ([#13
+00004b50: 3939 5d28 6874 7470 733a 2f2f 6769 7468  99](https://gith
+00004b60: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+00004b70: 6369 6b69 742d 6269 6f2f 7075 6c6c 2f31  cikit-bio/pull/1
+00004b80: 3339 3929 290a 2a20 5468 6520 6049 6e74  399)).* The `Int
+00004b90: 6572 7661 6c4d 6574 6164 6174 6160 2061  ervalMetadata` a
+00004ba0: 6c6c 6f77 7320 6472 616d 6174 6963 2064  llows dramatic d
+00004bb0: 6563 7265 6173 6520 696e 206d 656d 6f72  ecrease in memor
+00004bc0: 7920 7573 6167 6520 696e 2072 6561 6469  y usage in readi
+00004bd0: 6e67 2047 656e 4261 6e6b 2066 696c 6573  ng GenBank files
+00004be0: 206f 6620 6665 6174 7572 6520 7269 6368   of feature rich
+00004bf0: 2073 6571 7565 6e63 6573 2e20 285b 2331   sequences. ([#1
+00004c00: 3135 395d 2868 7474 7073 3a2f 2f67 6974  159](https://git
+00004c10: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+00004c20: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
+00004c30: 732f 3131 3539 2929 0a0a 2323 2320 4275  s/1159))..### Bu
+00004c40: 6720 6669 7865 730a 0a2a 2060 736b 6269  g fixes..* `skbi
+00004c50: 6f2e 7472 6565 2e54 7265 654e 6f64 652e  o.tree.TreeNode.
+00004c60: 7072 756e 6560 2061 6e64 2069 6d70 6c69  prune` and impli
+00004c70: 6369 746c 7920 6073 6b62 696f 2e74 7265  citly `skbio.tre
+00004c80: 652e 5472 6565 4e6f 6465 2e73 6865 6172  e.TreeNode.shear
+00004c90: 6020 7765 7265 206e 6f74 2068 616e 646c  ` were not handl
+00004ca0: 696e 6720 6120 7369 7475 6174 696f 6e20  ing a situation 
+00004cb0: 696e 2077 6869 6368 2061 2070 6172 656e  in which a paren
+00004cc0: 7420 7761 7320 7661 6c69 646c 7920 7265  t was validly re
+00004cd0: 6d6f 7665 6420 6475 7269 6e67 2070 7275  moved during pru
+00004ce0: 6e69 6e67 206f 7065 7261 7469 6f6e 7320  ning operations 
+00004cf0: 6173 206d 6179 2068 6170 7065 6e20 6966  as may happen if
+00004d00: 2074 6865 2072 6573 756c 7469 6e67 2073   the resulting s
+00004d10: 7562 7472 6565 2064 6f65 7320 6e6f 7420  ubtree does not 
+00004d20: 696e 636c 7564 6520 7468 6520 726f 6f74  include the root
+00004d30: 2e20 5072 6576 696f 7573 6c79 2c20 616e  . Previously, an
+00004d40: 2060 4174 7472 6962 7574 6545 7272 6f72   `AttributeError
+00004d50: 6020 776f 756c 6420 7261 6973 6520 6173  ` would raise as
+00004d60: 2060 7061 7265 6e74 6020 776f 756c 6420   `parent` would 
+00004d70: 6265 2060 4e6f 6e65 6020 696e 2074 6869  be `None` in thi
+00004d80: 7320 7369 7475 6174 696f 6e2e 0a2a 206e  s situation..* n
+00004d90: 756d 7079 206c 696e 6b69 6e67 2077 6173  umpy linking was
+00004da0: 2066 6978 6564 2066 6f72 2069 6e73 7461   fixed for insta
+00004db0: 6c6c 6174 696f 6e20 756e 6465 7220 456c  llation under El
+00004dc0: 2043 6170 6974 616e 2e0a 2a20 4120 6275   Capitan..* A bu
+00004dd0: 6720 7761 7320 696e 7472 6f64 7563 6564  g was introduced
+00004de0: 2069 6e20 2331 3339 3820 696e 746f 2060   in #1398 into `
+00004df0: 5472 6565 4e6f 6465 2e70 7275 6e65 6020  TreeNode.prune` 
+00004e00: 616e 6420 6669 7865 6420 696e 2023 3134  and fixed in #14
+00004e10: 3136 2069 6e20 7768 6963 682c 2075 6e64  16 in which, und
+00004e20: 6572 2074 6865 2073 7065 6369 616c 2063  er the special c
+00004e30: 6173 6520 6f66 2061 2073 696e 676c 6520  ase of a single 
+00004e40: 6465 7363 656e 6465 6e74 2065 7869 7374  descendent exist
+00004e50: 696e 6720 6672 6f6d 2074 6865 2072 6f6f  ing from the roo
+00004e60: 742c 2074 6865 2072 6573 756c 7469 6e67  t, the resulting
+00004e70: 2063 6869 6c64 7265 6e20 7061 7265 6e74   children parent
+00004e80: 2072 6566 6572 656e 6365 7320 7765 7265   references were
+00004e90: 206e 6f74 2075 7064 6174 6564 2e20 5468   not updated. Th
+00004ea0: 6520 6361 7573 6520 6f66 2074 6865 2062  e cause of the b
+00004eb0: 7567 2077 6173 2061 2063 616c 6c20 6d61  ug was a call ma
+00004ec0: 6465 2074 6f20 6073 656c 662e 6368 696c  de to `self.chil
+00004ed0: 6472 656e 2e65 7874 656e 6460 2061 7320  dren.extend` as 
+00004ee0: 6f70 706f 7365 6420 746f 2060 7365 6c66  opposed to `self
+00004ef0: 2e65 7874 656e 6460 2077 6865 7265 2074  .extend` where t
+00004f00: 6865 2066 6f72 6d65 7220 6973 2061 2060  he former is a `
+00004f10: 6c69 7374 2e65 7874 656e 6460 2077 6974  list.extend` wit
+00004f20: 686f 7574 206b 6e6f 776c 6564 6765 206f  hout knowledge o
+00004f30: 6620 7468 6520 7472 6565 2c20 7768 696c  f the tree, whil
+00004f40: 6520 7468 6520 6c61 7474 6572 2069 7320  e the latter is 
+00004f50: 6054 7265 654e 6f64 652e 6578 7465 6e64  `TreeNode.extend
+00004f60: 6020 7768 6963 6820 6973 2061 626c 6520  ` which is able 
+00004f70: 746f 2061 646a 7573 7420 7265 6665 7265  to adjust refere
+00004f80: 6e63 6573 2074 6f20 6073 656c 662e 7061  nces to `self.pa
+00004f90: 7265 6e74 602e 0a0a 2323 2320 4d69 7363  rent`...### Misc
+00004fa0: 656c 6c61 6e65 6f75 730a 0a2a 2052 656d  ellaneous..* Rem
+00004fb0: 6f76 6564 2064 6570 7265 6361 7465 6420  oved deprecated 
+00004fc0: 6675 6e63 7469 6f6e 7320 6672 6f6d 2060  functions from `
+00004fd0: 736b 6269 6f2e 7574 696c 603a 2060 6973  skbio.util`: `is
+00004fe0: 5f63 6173 6176 615f 7631 3830 5f6f 725f  _casava_v180_or_
+00004ff0: 6c61 7465 7260 2c20 6072 656d 6f76 655f  later`, `remove_
+00005000: 6669 6c65 7360 2c20 616e 6420 6063 7265  files`, and `cre
+00005010: 6174 655f 6469 7260 2e0a 2a20 5265 6d6f  ate_dir`..* Remo
+00005020: 7665 6420 6465 7072 6563 6174 6564 2060  ved deprecated `
+00005030: 736b 6269 6f2e 5365 7175 656e 6365 2e63  skbio.Sequence.c
+00005040: 6f70 7960 206d 6574 686f 642e 0a0a 2323  opy` method...##
+00005050: 2056 6572 7369 6f6e 2030 2e35 2e30 2028   Version 0.5.0 (
+00005060: 3230 3136 2d30 362d 3134 290a 0a2a 2a49  2016-06-14)..**I
+00005070: 4d50 4f52 5441 4e54 2a2a 3a20 7363 696b  MPORTANT**: scik
+00005080: 6974 2d62 696f 2069 7320 6e6f 206c 6f6e  it-bio is no lon
+00005090: 6765 7220 636f 6d70 6174 6962 6c65 2077  ger compatible w
+000050a0: 6974 6820 5079 7468 6f6e 2032 2e20 7363  ith Python 2. sc
+000050b0: 696b 6974 2d62 696f 2069 7320 636f 6d70  ikit-bio is comp
+000050c0: 6174 6962 6c65 2077 6974 6820 5079 7468  atible with Pyth
+000050d0: 6f6e 2033 2e34 2061 6e64 206c 6174 6572  on 3.4 and later
+000050e0: 2e0a 0a23 2323 2046 6561 7475 7265 730a  ...### Features.
+000050f0: 2a20 4164 6465 6420 6d6f 7265 2064 6573  * Added more des
+00005100: 6372 6970 7469 7665 2065 7272 6f72 206d  criptive error m
+00005110: 6573 7361 6765 2074 6f20 6073 6b62 696f  essage to `skbio
+00005120: 2e69 6f2e 7265 6769 7374 7279 6020 7768  .io.registry` wh
+00005130: 656e 2061 7474 656d 7074 696e 6720 746f  en attempting to
+00005140: 2072 6561 6420 7769 7468 6f75 7420 7370   read without sp
+00005150: 6563 6966 7969 6e67 2060 696e 746f 6020  ecifying `into` 
+00005160: 616e 6420 7768 656e 2074 6865 7265 2069  and when there i
+00005170: 7320 6e6f 2067 656e 6572 6174 6f72 2072  s no generator r
+00005180: 6561 6465 722e 2028 5b23 3133 3236 5d28  eader. ([#1326](
+00005190: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+000051a0: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+000051b0: 742d 6269 6f2f 6973 7375 6573 2f31 3332  t-bio/issues/132
+000051c0: 3629 290a 2a20 4164 6465 6420 7375 7070  6)).* Added supp
+000051d0: 6f72 7420 666f 7220 7265 6665 7265 6e63  ort for referenc
+000051e0: 6520 7461 6773 2074 6f20 6073 6b62 696f  e tags to `skbio
+000051f0: 2e69 6f2e 666f 726d 6174 2e73 746f 636b  .io.format.stock
+00005200: 686f 6c6d 6020 7265 6164 6572 2061 6e64  holm` reader and
+00005210: 2077 7269 7465 722e 2028 5b23 3133 3438   writer. ([#1348
+00005220: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+00005230: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
+00005240: 6b69 742d 6269 6f2f 6973 7375 6573 2f31  kit-bio/issues/1
+00005250: 3334 3829 290a 2a20 4578 7061 6e64 6564  348)).* Expanded
+00005260: 2065 7272 6f72 206d 6573 7361 6765 2069   error message i
+00005270: 6e20 6073 6b62 696f 2e69 6f2e 666f 726d  n `skbio.io.form
+00005280: 6174 2e73 746f 636b 686f 6c6d 6020 7265  at.stockholm` re
+00005290: 6164 6572 2077 6865 6e20 6063 6f6e 7374  ader when `const
+000052a0: 7275 6374 6f72 6020 6973 206e 6f74 2070  ructor` is not p
+000052b0: 6173 7365 642c 2069 6e20 6f72 6465 7220  assed, in order 
+000052c0: 746f 2070 726f 7669 6465 2062 6574 7465  to provide bette
+000052d0: 7220 6578 706c 616e 6174 696f 6e20 746f  r explanation to
+000052e0: 2075 7365 722e 2028 5b23 3133 3237 5d28   user. ([#1327](
+000052f0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+00005300: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+00005310: 742d 6269 6f2f 6973 7375 6573 2f31 3332  t-bio/issues/132
+00005320: 3729 290a 2a20 4164 6465 6420 6073 6b62  7)).* Added `skb
+00005330: 696f 2e73 6571 7565 6e63 652e 6469 7374  io.sequence.dist
+00005340: 616e 6365 2e6b 6d65 725f 6469 7374 616e  ance.kmer_distan
+00005350: 6365 6020 666f 7220 636f 6d70 7574 696e  ce` for computin
+00005360: 6720 7468 6520 6b6d 6572 2064 6973 7461  g the kmer dista
+00005370: 6e63 6520 6265 7477 6565 6e20 7477 6f20  nce between two 
+00005380: 7365 7175 656e 6365 732e 2028 5b23 3931  sequences. ([#91
+00005390: 335d 2868 7474 7073 3a2f 2f67 6974 6875  3](https://githu
+000053a0: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
+000053b0: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
+000053c0: 3931 3329 290a 2a20 4164 6465 6420 6073  913)).* Added `s
+000053d0: 6b62 696f 2e73 6571 7565 6e63 652e 5365  kbio.sequence.Se
+000053e0: 7175 656e 6365 2e72 6570 6c61 6365 6020  quence.replace` 
+000053f0: 666f 7220 6173 7369 676e 696e 6720 6120  for assigning a 
+00005400: 6368 6172 6163 7465 7220 746f 2070 6f73  character to pos
+00005410: 6974 696f 6e73 2069 6e20 6120 6053 6571  itions in a `Seq
+00005420: 7565 6e63 6560 2e20 285b 2331 3232 325d  uence`. ([#1222]
+00005430: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+00005440: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+00005450: 6974 2d62 696f 2f69 7373 7565 732f 3132  it-bio/issues/12
+00005460: 3232 2929 0a2a 2041 6464 6564 2073 7570  22)).* Added sup
+00005470: 706f 7274 2066 6f72 2060 7061 6e64 6173  port for `pandas
+00005480: 2e52 616e 6765 496e 6465 7860 2c20 6c6f  .RangeIndex`, lo
+00005490: 7765 7269 6e67 2074 6865 206d 656d 6f72  wering the memor
+000054a0: 7920 666f 6f74 7072 696e 7420 6f66 2064  y footprint of d
+000054b0: 6566 6175 6c74 2069 6e74 6567 6572 2069  efault integer i
+000054c0: 6e64 6578 206f 626a 6563 7473 2e20 6053  ndex objects. `S
+000054d0: 6571 7565 6e63 652e 706f 7369 7469 6f6e  equence.position
+000054e0: 616c 5f6d 6574 6164 6174 6160 2061 6e64  al_metadata` and
+000054f0: 2060 5461 6275 6c61 724d 5341 2e70 6f73   `TabularMSA.pos
+00005500: 6974 696f 6e61 6c5f 6d65 7461 6461 7461  itional_metadata
+00005510: 6020 6e6f 7720 7573 6520 6070 642e 5261  ` now use `pd.Ra
+00005520: 6e67 6549 6e64 6578 6020 6173 2074 6865  ngeIndex` as the
+00005530: 2070 6f73 6974 696f 6e61 6c20 6d65 7461   positional meta
+00005540: 6461 7461 2069 6e64 6578 2e20 6054 6162  data index. `Tab
+00005550: 756c 6172 4d53 4160 206e 6f77 2075 7365  ularMSA` now use
+00005560: 7320 6070 642e 5261 6e67 6549 6e64 6578  s `pd.RangeIndex
+00005570: 6020 6173 2074 6865 2064 6566 6175 6c74  ` as the default
+00005580: 2069 6e64 6578 2e20 5573 6167 6520 6f66   index. Usage of
+00005590: 2060 7064 2e52 616e 6765 496e 6465 7860   `pd.RangeIndex`
+000055a0: 206f 7665 7220 7468 6520 7072 6576 696f   over the previo
+000055b0: 7573 2060 7064 2e49 6e74 3634 496e 6465  us `pd.Int64Inde
+000055c0: 7860 205b 7368 6f75 6c64 2062 6520 7472  x` [should be tr
+000055d0: 616e 7370 6172 656e 745d 2868 7474 703a  ansparent](http:
+000055e0: 2f2f 7061 6e64 6173 2e70 7964 6174 612e  //pandas.pydata.
+000055f0: 6f72 672f 7061 6e64 6173 2d64 6f63 732f  org/pandas-docs/
+00005600: 7665 7273 696f 6e2f 302e 3138 2e30 2f77  version/0.18.0/w
+00005610: 6861 7473 6e65 772e 6874 6d6c 2372 616e  hatsnew.html#ran
+00005620: 6765 2d69 6e64 6578 292c 2073 6f20 7468  ge-index), so th
+00005630: 6573 6520 6368 616e 6765 7320 7368 6f75  ese changes shou
+00005640: 6c64 2062 6520 6e6f 6e2d 6272 6561 6b69  ld be non-breaki
+00005650: 6e67 2074 6f20 7573 6572 732e 2073 6369  ng to users. sci
+00005660: 6b69 742d 6269 6f20 6e6f 7720 6465 7065  kit-bio now depe
+00005670: 6e64 7320 6f6e 2070 616e 6461 7320 3e3d  nds on pandas >=
+00005680: 2030 2e31 382e 3020 285b 2331 3330 385d   0.18.0 ([#1308]
+00005690: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+000056a0: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+000056b0: 6974 2d62 696f 2f69 7373 7565 732f 3133  it-bio/issues/13
+000056c0: 3038 2929 0a2a 2041 6464 6564 2060 7265  08)).* Added `re
+000056d0: 7365 745f 696e 6465 783d 4661 6c73 6560  set_index=False`
+000056e0: 2070 6172 616d 6574 6572 2074 6f20 6054   parameter to `T
+000056f0: 6162 756c 6172 4d53 412e 6170 7065 6e64  abularMSA.append
+00005700: 6020 616e 6420 6054 6162 756c 6172 4d53  ` and `TabularMS
+00005710: 412e 6578 7465 6e64 6020 666f 7220 7265  A.extend` for re
+00005720: 7365 7474 696e 6720 7468 6520 4d53 4127  setting the MSA'
+00005730: 7320 696e 6465 7820 746f 2074 6865 2064  s index to the d
+00005740: 6566 6175 6c74 2069 6e64 6578 2061 6674  efault index aft
+00005750: 6572 2061 7070 656e 6469 6e67 2f65 7874  er appending/ext
+00005760: 656e 6469 6e67 2e0a 2a20 4164 6465 6420  ending..* Added 
+00005770: 7375 7070 6f72 7420 666f 7220 7061 7274  support for part
+00005780: 6961 6c20 7061 6972 7769 7365 2063 616c  ial pairwise cal
+00005790: 6375 6c61 7469 6f6e 7320 7669 6120 6073  culations via `s
+000057a0: 6b62 696f 2e64 6976 6572 7369 7479 2e70  kbio.diversity.p
+000057b0: 6172 7469 616c 5f62 6574 615f 6469 7665  artial_beta_dive
+000057c0: 7273 6974 7960 2e20 285b 2331 3232 315d  rsity`. ([#1221]
+000057d0: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+000057e0: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+000057f0: 6974 2d62 696f 2f69 7373 7565 732f 3132  it-bio/issues/12
+00005800: 3231 292c 205b 2331 3333 375d 2868 7474  21), [#1337](htt
+00005810: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+00005820: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
+00005830: 696f 2f70 756c 6c2f 3133 3337 2929 2e20  io/pull/1337)). 
+00005840: 5468 6973 2066 756e 6374 696f 6e20 6973  This function is
+00005850: 2069 6d6d 6564 6961 7465 6c79 2064 6570   immediately dep
+00005860: 7265 6361 7465 6420 6173 2069 7473 2072  recated as its r
+00005870: 6574 7572 6e20 7479 7065 2077 696c 6c20  eturn type will 
+00005880: 6368 616e 6765 2069 6e20 7468 6520 6675  change in the fu
+00005890: 7475 7265 2061 6e64 2073 686f 756c 6420  ture and should 
+000058a0: 6265 2075 7365 6420 7769 7468 2063 6175  be used with cau
+000058b0: 7469 6f6e 2069 6e20 6974 7320 7072 6573  tion in its pres
+000058c0: 656e 7420 666f 726d 2028 7365 6520 7468  ent form (see th
+000058d0: 6520 6675 6e63 7469 6f6e 2773 2064 6f63  e function's doc
+000058e0: 756d 656e 7461 7469 6f6e 2066 6f72 2064  umentation for d
+000058f0: 6574 6169 6c73 292e 0a2a 2060 5465 6d70  etails)..* `Temp
+00005900: 6f72 6172 7946 696c 6560 2061 6e64 2060  oraryFile` and `
+00005910: 4e61 6d65 6454 656d 706f 7261 7279 4669  NamedTemporaryFi
+00005920: 6c65 6020 6172 6520 6e6f 7720 7375 7070  le` are now supp
+00005930: 6f72 7465 6420 494f 2073 6f75 7263 6573  orted IO sources
+00005940: 2066 6f72 2060 736b 6269 6f2e 696f 6020   for `skbio.io` 
+00005950: 616e 6420 7265 6c61 7465 6420 6675 6e63  and related func
+00005960: 7469 6f6e 616c 6974 792e 2020 285b 2331  tionality.  ([#1
+00005970: 3239 315d 2868 7474 7073 3a2f 2f67 6974  291](https://git
+00005980: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+00005990: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
+000059a0: 732f 3132 3931 2929 0a2a 2041 6464 6564  s/1291)).* Added
+000059b0: 2060 7472 6565 5f6e 6f64 655f 636c 6173   `tree_node_clas
+000059c0: 733d 5472 6565 4e6f 6465 6020 7061 7261  s=TreeNode` para
+000059d0: 6d65 7465 7220 746f 2060 736b 6269 6f2e  meter to `skbio.
+000059e0: 7472 6565 2e6d 616a 6f72 6974 795f 7275  tree.majority_ru
+000059f0: 6c65 6020 746f 2073 7570 706f 7274 2072  le` to support r
+00005a00: 6574 7572 6e69 6e67 2063 6f6e 7365 6e73  eturning consens
+00005a10: 7573 2074 7265 6573 206f 6620 7479 7065  us trees of type
+00005a20: 2060 5472 6565 4e6f 6465 6020 2874 6865   `TreeNode` (the
+00005a30: 2064 6566 6175 6c74 2920 6f72 2061 2074   default) or a t
+00005a40: 7970 6520 7468 6174 2068 6173 2074 6865  ype that has the
+00005a50: 2073 616d 6520 696e 7465 7266 6163 6520   same interface 
+00005a60: 6173 2060 5472 6565 4e6f 6465 6020 2865  as `TreeNode` (e
+00005a70: 2e67 2e20 6054 7265 654e 6f64 6560 2073  .g. `TreeNode` s
+00005a80: 7562 636c 6173 7365 7329 2028 5b23 3131  ubclasses) ([#11
+00005a90: 3933 5d28 6874 7470 733a 2f2f 6769 7468  93](https://gith
+00005aa0: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+00005ab0: 6369 6b69 742d 6269 6f2f 7075 6c6c 2f31  cikit-bio/pull/1
+00005ac0: 3139 3329 290a 2a20 6054 7265 654e 6f64  193)).* `TreeNod
+00005ad0: 652e 6672 6f6d 5f6c 696e 6b61 6765 5f6d  e.from_linkage_m
+00005ae0: 6174 7269 7860 2061 6e64 2060 5472 6565  atrix` and `Tree
+00005af0: 4e6f 6465 2e66 726f 6d5f 7461 786f 6e6f  Node.from_taxono
+00005b00: 6d79 6020 6e6f 7720 7375 7070 6f72 7420  my` now support 
+00005b10: 636f 6e73 7472 7563 7469 6e67 2060 5472  constructing `Tr
+00005b20: 6565 4e6f 6465 6020 7375 6263 6c61 7373  eeNode` subclass
+00005b30: 6573 2e20 6054 7265 654e 6f64 652e 6269  es. `TreeNode.bi
+00005b40: 6675 7263 6174 6560 206e 6f77 2073 7570  furcate` now sup
+00005b50: 706f 7274 7320 6054 7265 654e 6f64 6560  ports `TreeNode`
+00005b60: 2073 7562 636c 6173 7365 7320 285b 2331   subclasses ([#1
+00005b70: 3139 335d 2868 7474 7073 3a2f 2f67 6974  193](https://git
+00005b80: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+00005b90: 7363 696b 6974 2d62 696f 2f70 756c 6c2f  scikit-bio/pull/
+00005ba0: 3131 3933 2929 0a2a 2054 6865 2060 6967  1193)).* The `ig
+00005bb0: 6e6f 7265 5f6d 6574 6164 6174 6160 206b  nore_metadata` k
+00005bc0: 6579 776f 7264 2068 6173 2062 6565 6e20  eyword has been 
+00005bd0: 6164 6465 6420 746f 2060 5461 6275 6c61  added to `Tabula
+00005be0: 724d 5341 2e69 7465 725f 706f 7369 7469  rMSA.iter_positi
+00005bf0: 6f6e 7360 2074 6f20 696d 7072 6f76 6520  ons` to improve 
+00005c00: 7065 7266 6f72 6d61 6e63 6520 7768 656e  performance when
+00005c10: 206d 6574 6164 6174 6120 6973 206e 6f74   metadata is not
+00005c20: 206e 6563 6573 7361 7279 2e0a 2a20 5061   necessary..* Pa
+00005c30: 6972 7769 7365 2061 6c69 676e 6572 7320  irwise aligners 
+00005c40: 696e 2060 736b 6269 6f2e 616c 6967 6e6d  in `skbio.alignm
+00005c50: 656e 7460 206e 6f77 2070 726f 7061 6761  ent` now propaga
+00005c60: 7465 2070 6572 2d73 6571 7565 6e63 6520  te per-sequence 
+00005c70: 606d 6574 6164 6174 6160 206f 626a 6563  `metadata` objec
+00005c80: 7473 2028 7468 6973 2064 6f65 7320 6e6f  ts (this does no
+00005c90: 7420 696e 636c 7564 6520 6070 6f73 6974  t include `posit
+00005ca0: 696f 6e61 6c5f 6d65 7461 6461 7461 6029  ional_metadata`)
+00005cb0: 2e0a 0a23 2323 2042 6163 6b77 6172 642d  ...### Backward-
+00005cc0: 696e 636f 6d70 6174 6962 6c65 2063 6861  incompatible cha
+00005cd0: 6e67 6573 205b 7374 6162 6c65 5d0a 0a23  nges [stable]..#
+00005ce0: 2323 2042 6163 6b77 6172 642d 696e 636f  ## Backward-inco
+00005cf0: 6d70 6174 6962 6c65 2063 6861 6e67 6573  mpatible changes
+00005d00: 205b 6578 7065 7269 6d65 6e74 616c 5d0a   [experimental].
+00005d10: 2a20 6054 6162 756c 6172 4d53 412e 6170  * `TabularMSA.ap
+00005d20: 7065 6e64 6020 616e 6420 6054 6162 756c  pend` and `Tabul
+00005d30: 6172 4d53 412e 6578 7465 6e64 6020 6e6f  arMSA.extend` no
+00005d40: 7720 7265 7175 6972 6520 6f6e 6520 6f66  w require one of
+00005d50: 2060 6d69 6e74 6572 602c 2060 696e 6465   `minter`, `inde
+00005d60: 7860 2c20 6f72 2060 7265 7365 745f 696e  x`, or `reset_in
+00005d70: 6465 7860 2074 6f20 6265 2070 726f 7669  dex` to be provi
+00005d80: 6465 6420 7768 656e 2069 6e63 6f72 706f  ded when incorpo
+00005d90: 7261 7469 6e67 206e 6577 2073 6571 7565  rating new seque
+00005da0: 6e63 6573 2069 6e74 6f20 616e 204d 5341  nces into an MSA
+00005db0: 2e20 5072 6576 696f 7573 2062 6568 6176  . Previous behav
+00005dc0: 696f 7220 7761 7320 746f 2061 7574 6f2d  ior was to auto-
+00005dd0: 696e 6372 656d 656e 7420 7468 6520 696e  increment the in
+00005de0: 6465 7820 6c61 6265 6c73 2069 6620 606d  dex labels if `m
+00005df0: 696e 7465 7260 2061 6e64 2060 696e 6465  inter` and `inde
+00005e00: 7860 2077 6572 656e 2774 2070 726f 7669  x` weren't provi
+00005e10: 6465 6420 616e 6420 7468 6520 4d53 4120  ded and the MSA 
+00005e20: 6861 6420 6120 6465 6661 756c 7420 696e  had a default in
+00005e30: 7465 6765 7220 696e 6465 782c 206f 7468  teger index, oth
+00005e40: 6572 7769 7365 2065 7272 6f72 2e20 5573  erwise error. Us
+00005e50: 6520 6072 6573 6574 5f69 6e64 6578 3d54  e `reset_index=T
+00005e60: 7275 6560 2074 6f20 6f62 7461 696e 2074  rue` to obtain t
+00005e70: 6865 2070 7265 7669 6f75 7320 6265 6861  he previous beha
+00005e80: 7669 6f72 2069 6e20 6120 6d6f 7265 2065  vior in a more e
+00005e90: 7870 6c69 6369 7420 7761 792e 0a2a 2060  xplicit way..* `
+00005ea0: 736b 6269 6f2e 7374 6174 732e 636f 6d70  skbio.stats.comp
+00005eb0: 6f73 6974 696f 6e2e 616e 636f 6d60 206e  osition.ancom` n
+00005ec0: 6f77 2072 6574 7572 6e73 2074 776f 2060  ow returns two `
+00005ed0: 7064 2e44 6174 6146 7261 6d65 6020 6f62  pd.DataFrame` ob
+00005ee0: 6a65 6374 732c 2077 6865 7265 2069 7420  jects, where it 
+00005ef0: 7072 6576 696f 7573 6c79 2072 6574 7572  previously retur
+00005f00: 6e65 6420 6f6e 652e 2054 6865 2066 6972  ned one. The fir
+00005f10: 7374 2063 6f6e 7461 696e 7320 7468 6520  st contains the 
+00005f20: 414e 434f 4d20 7465 7374 2072 6573 756c  ANCOM test resul
+00005f30: 7473 2c20 6173 2062 6566 6f72 652c 2061  ts, as before, a
+00005f40: 6e64 2074 6865 2073 6563 6f6e 6420 636f  nd the second co
+00005f50: 6e74 6169 6e73 2070 6572 6365 6e74 696c  ntains percentil
+00005f60: 6520 6162 756e 6461 6e63 6573 206f 6620  e abundances of 
+00005f70: 6561 6368 2066 6561 7475 7265 2069 6e20  each feature in 
+00005f80: 6561 6368 2067 726f 7570 2e20 5468 6520  each group. The 
+00005f90: 7370 6563 6966 6963 2070 6572 6365 6e74  specific percent
+00005fa0: 696c 6573 2074 6861 7420 6172 6520 636f  iles that are co
+00005fb0: 6d70 7574 6564 2061 6e64 2072 6574 7572  mputed and retur
+00005fc0: 6e65 6420 6973 2063 6f6e 7472 6f6c 6c65  ned is controlle
+00005fd0: 6420 6279 2074 6865 206e 6577 2060 7065  d by the new `pe
+00005fe0: 7263 656e 7469 6c65 7360 2070 6172 616d  rcentiles` param
+00005ff0: 6574 6572 2074 6f20 6073 6b62 696f 2e73  eter to `skbio.s
+00006000: 7461 7473 2e63 6f6d 706f 7369 7469 6f6e  tats.composition
+00006010: 2e61 6e63 6f6d 602e 2049 6e20 7468 6520  .ancom`. In the 
+00006020: 6675 7475 7265 2c20 7468 6973 2073 6563  future, this sec
+00006030: 6f6e 6420 6070 642e 4461 7461 4672 616d  ond `pd.DataFram
+00006040: 6560 2077 696c 6c20 6e6f 7420 6265 2072  e` will not be r
+00006050: 6574 7572 6e65 6420 6279 2074 6869 7320  eturned by this 
+00006060: 6675 6e63 7469 6f6e 2c20 6275 7420 7769  function, but wi
+00006070: 6c6c 2062 6520 6176 6169 6c61 626c 6520  ll be available 
+00006080: 7468 726f 7567 6820 7468 6520 5b63 6f6e  through the [con
+00006090: 7469 6e67 656e 6379 2074 6162 6c65 2041  tingency table A
+000060a0: 5049 5d28 6874 7470 733a 2f2f 6769 7468  PI](https://gith
+000060b0: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+000060c0: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
+000060d0: 2f38 3438 292e 2028 5b23 3132 3933 5d28  /848). ([#1293](
+000060e0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+000060f0: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+00006100: 742d 6269 6f2f 6973 7375 6573 2f31 3239  t-bio/issues/129
+00006110: 3329 290a 2a20 6073 6b62 696f 2e73 7461  3)).* `skbio.sta
+00006120: 7473 2e63 6f6d 706f 7369 7469 6f6e 2e61  ts.composition.a
+00006130: 6e63 6f6d 6020 6e6f 7720 7065 7266 6f72  ncom` now perfor
+00006140: 6d73 206d 756c 7469 706c 6520 636f 6d70  ms multiple comp
+00006150: 6172 6973 6f6e 7320 636f 7272 6563 7469  arisons correcti
+00006160: 6f6e 2062 7920 6465 6661 756c 742e 2054  on by default. T
+00006170: 6865 2070 7265 7669 6f75 7320 6265 6861  he previous beha
+00006180: 7669 6f72 206f 6620 6e6f 7420 7065 7266  vior of not perf
+00006190: 6f72 6d69 6e67 206d 756c 7469 706c 6520  orming multiple 
+000061a0: 636f 6d70 6172 6973 6f6e 7320 636f 7272  comparisons corr
+000061b0: 6563 7469 6f6e 2063 616e 2062 6520 6163  ection can be ac
+000061c0: 6869 6576 6564 2062 7920 7061 7373 696e  hieved by passin
+000061d0: 6720 6060 6d75 6c74 6970 6c65 5f63 6f6d  g ``multiple_com
+000061e0: 7061 7269 736f 6e73 5f63 6f72 7265 6374  parisons_correct
+000061f0: 696f 6e3d 4e6f 6e65 6060 2e0a 2a20 5468  ion=None``..* Th
+00006200: 6520 6060 7265 6a65 6374 6060 2063 6f6c  e ``reject`` col
+00006210: 756d 6e20 696e 2074 6865 2066 6972 7374  umn in the first
+00006220: 2060 6070 642e 4461 7461 4672 616d 6560   ``pd.DataFrame`
+00006230: 6020 7265 7475 726e 6564 2066 726f 6d20  ` returned from 
+00006240: 6073 6b62 696f 2e73 7461 7473 2e63 6f6d  `skbio.stats.com
+00006250: 706f 7369 7469 6f6e 2e61 6e63 6f6d 6020  position.ancom` 
+00006260: 6861 7320 6265 656e 2072 656e 616d 6564  has been renamed
+00006270: 2060 6052 656a 6563 7420 6e75 6c6c 2068   ``Reject null h
+00006280: 7970 6f74 6865 7369 7360 6020 666f 7220  ypothesis`` for 
+00006290: 636c 6172 6974 792e 2028 5b23 3133 3735  clarity. ([#1375
+000062a0: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+000062b0: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
+000062c0: 6b69 742d 6269 6f2f 6973 7375 6573 2f31  kit-bio/issues/1
+000062d0: 3337 3529 290a 0a23 2323 2042 7567 2066  375))..### Bug f
+000062e0: 6978 6573 0a2a 2046 6978 6564 2072 6f77  ixes.* Fixed row
+000062f0: 2061 6e64 2063 6f6c 756d 6e20 6e61 6d65   and column name
+00006300: 7320 746f 2060 6269 706c 6f74 5f73 636f  s to `biplot_sco
+00006310: 7265 7360 2069 6e20 7468 6520 604f 7264  res` in the `Ord
+00006320: 696e 6174 696f 6e52 6573 756c 7473 6020  inationResults` 
+00006330: 6f62 6a65 6374 2066 726f 6d20 6073 6b62  object from `skb
+00006340: 696f 2e73 7461 7473 2e6f 7264 696e 6174  io.stats.ordinat
+00006350: 696f 6e60 2e20 5468 6973 2066 6978 2061  ion`. This fix a
+00006360: 6666 6563 7420 7468 6520 6063 6361 6020  ffect the `cca` 
+00006370: 616e 6420 6072 6461 6020 6d65 7468 6f64  and `rda` method
+00006380: 732e 2028 5b23 3133 3232 5d28 6874 7470  s. ([#1322](http
+00006390: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+000063a0: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+000063b0: 6f2f 6973 7375 6573 2f31 3332 3229 290a  o/issues/1322)).
+000063c0: 2a20 4669 7865 6420 6275 6720 7768 656e  * Fixed bug when
+000063d0: 2075 7369 6e67 2060 736b 6269 6f2e 696f   using `skbio.io
+000063e0: 2e66 6f72 6d61 742e 7374 6f63 6b68 6f6c  .format.stockhol
+000063f0: 6d60 2072 6561 6465 7220 6f6e 2066 696c  m` reader on fil
+00006400: 6520 7769 7468 206d 756c 7469 2d6c 696e  e with multi-lin
+00006410: 6520 7472 6565 2077 6974 6820 6e6f 2069  e tree with no i
+00006420: 642e 2050 7265 7669 6f75 736c 7920 7468  d. Previously th
+00006430: 6973 2072 6169 7365 6420 616e 2060 4174  is raised an `At
+00006440: 7472 6962 7574 6545 7272 6f72 602c 206e  tributeError`, n
+00006450: 6f77 2069 7420 636f 7272 6563 746c 7920  ow it correctly 
+00006460: 6861 6e64 6c65 7320 7468 6973 2074 7970  handles this typ
+00006470: 6520 6f66 2074 7265 652e 2028 5b23 3133  e of tree. ([#13
+00006480: 3334 5d28 6874 7470 733a 2f2f 6769 7468  34](https://gith
+00006490: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+000064a0: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
+000064b0: 2f31 3333 3429 290a 2a20 4669 7865 6420  /1334)).* Fixed 
+000064c0: 6275 6720 7768 656e 2072 6561 6469 6e67  bug when reading
+000064d0: 2053 746f 636b 686f 6c6d 2066 696c 6573   Stockholm files
+000064e0: 2077 6974 6820 4746 206f 7220 4753 2066   with GF or GS f
+000064f0: 6561 7475 7265 7320 7370 6c69 7420 6f76  eatures split ov
+00006500: 6572 206d 756c 7469 706c 6520 6c69 6e65  er multiple line
+00006510: 732e 2050 7265 7669 6f75 736c 792c 2074  s. Previously, t
+00006520: 6865 2066 6561 7475 7265 2074 6578 7420  he feature text 
+00006530: 7761 7320 7369 6d70 6c79 2063 6f6e 6361  was simply conca
+00006540: 7465 6e61 7465 6420 6265 6361 7573 6520  tenated because 
+00006550: 6974 2077 6173 2061 7373 756d 6564 2074  it was assumed t
+00006560: 6f20 6861 7665 2074 7261 696c 696e 6720  o have trailing 
+00006570: 7768 6974 6573 7061 6365 2e20 5468 6572  whitespace. Ther
+00006580: 6520 6172 6520 6578 616d 706c 6573 206f  e are examples o
+00006590: 6620 5374 6f63 6b68 6f6c 6d20 6669 6c65  f Stockholm file
+000065a0: 7320 7769 7468 2061 6e64 2077 6974 686f  s with and witho
+000065b0: 7574 2074 7261 696c 696e 6720 7768 6974  ut trailing whit
+000065c0: 6573 7061 6365 2066 6f72 206d 756c 7469  espace for multi
+000065d0: 2d6c 696e 6520 6665 6174 7572 6573 2c20  -line features, 
+000065e0: 736f 2074 6865 2060 736b 6269 6f2e 696f  so the `skbio.io
+000065f0: 2e66 6f72 6d61 742e 7374 6f63 6b68 6f6c  .format.stockhol
+00006600: 6d60 2072 6561 6465 7220 6e6f 7720 6164  m` reader now ad
+00006610: 6473 2061 2073 696e 676c 6520 7370 6163  ds a single spac
+00006620: 6520 7768 656e 2063 6f6e 6361 7465 6e61  e when concatena
+00006630: 7469 6e67 2066 6561 7475 7265 2074 6578  ting feature tex
+00006640: 7420 7769 7468 6f75 7420 7472 6169 6c69  t without traili
+00006650: 6e67 2077 6869 7465 7370 6163 6520 746f  ng whitespace to
+00006660: 2061 766f 6964 206a 6f69 6e69 6e67 2077   avoid joining w
+00006670: 6f72 6473 2074 6f67 6574 6865 722e 204d  ords together. M
+00006680: 756c 7469 2d6c 696e 6520 7472 6565 7320  ulti-line trees 
+00006690: 7374 6f72 6564 2061 7320 4746 206d 6574  stored as GF met
+000066a0: 6164 6174 6120 6172 6520 636f 6e63 6174  adata are concat
+000066b0: 656e 6174 6564 2061 7320 7468 6579 2061  enated as they a
+000066c0: 7070 6561 7220 696e 2074 6865 2066 696c  ppear in the fil
+000066d0: 653b 2061 2073 7061 6365 2069 7320 6e6f  e; a space is no
+000066e0: 7420 6164 6465 6420 7768 656e 2063 6f6e  t added when con
+000066f0: 6361 7465 6e61 7469 6e67 2e20 285b 2331  catenating. ([#1
+00006700: 3332 385d 2868 7474 7073 3a2f 2f67 6974  328](https://git
+00006710: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+00006720: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
+00006730: 732f 3133 3238 2929 0a2a 2046 6978 6564  s/1328)).* Fixed
+00006740: 2062 7567 2077 6865 6e20 7573 696e 6720   bug when using 
+00006750: 6053 6571 7565 6e63 652e 6974 6572 5f6b  `Sequence.iter_k
+00006760: 6d65 7273 6020 6f6e 2065 6d70 7479 2060  mers` on empty `
+00006770: 5365 7175 656e 6365 6020 6f62 6a65 6374  Sequence` object
+00006780: 2e20 5072 6576 696f 7573 6c79 2074 6869  . Previously thi
+00006790: 7320 7261 6973 6564 2061 2060 5661 6c75  s raised a `Valu
+000067a0: 6545 7272 6f72 602c 206e 6f77 2069 7420  eError`, now it 
+000067b0: 7265 7475 726e 730a 616e 2065 6d70 7479  returns.an empty
+000067c0: 2067 656e 6572 6174 6f72 2e0a 2a20 4669   generator..* Fi
+000067d0: 7865 6420 6d69 6e6f 7220 6275 6720 7768  xed minor bug wh
+000067e0: 6572 6520 6164 6469 6e67 2073 6571 7565  ere adding seque
+000067f0: 6e63 6573 2074 6f20 616e 2065 6d70 7479  nces to an empty
+00006800: 2060 5461 6275 6c61 724d 5341 6020 7769   `TabularMSA` wi
+00006810: 7468 204d 5341 2d77 6964 6520 6070 6f73  th MSA-wide `pos
+00006820: 6974 696f 6e61 6c5f 6d65 7461 6461 7461  itional_metadata
+00006830: 6020 776f 756c 6420 7265 7375 6c74 2069  ` would result i
+00006840: 6e20 6120 6054 6162 756c 6172 4d53 4160  n a `TabularMSA`
+00006850: 206f 626a 6563 7420 696e 2061 6e20 696e   object in an in
+00006860: 636f 6e73 6973 7465 6e74 2073 7461 7465  consistent state
+00006870: 2e20 5468 6973 2063 6f75 6c64 2068 6170  . This could hap
+00006880: 7065 6e20 7573 696e 6720 6054 6162 756c  pen using `Tabul
+00006890: 6172 4d53 412e 6170 7065 6e64 6020 6f72  arMSA.append` or
+000068a0: 2060 5461 6275 6c61 724d 5341 2e65 7874   `TabularMSA.ext
+000068b0: 656e 6460 2e20 5468 6973 2062 7567 206f  end`. This bug o
+000068c0: 6e6c 7920 6166 6665 6374 7320 6120 6054  nly affects a `T
+000068d0: 6162 756c 6172 4d53 4160 206f 626a 6563  abularMSA` objec
+000068e0: 7420 2a77 6974 686f 7574 2a20 7365 7175  t *without* sequ
+000068f0: 656e 6365 7320 7468 6174 2068 6173 204d  ences that has M
+00006900: 5341 2d77 6964 6520 6070 6f73 6974 696f  SA-wide `positio
+00006910: 6e61 6c5f 6d65 7461 6461 7461 6020 2866  nal_metadata` (f
+00006920: 6f72 2065 7861 6d70 6c65 2c20 6054 6162  or example, `Tab
+00006930: 756c 6172 4d53 4128 5b5d 2c20 706f 7369  ularMSA([], posi
+00006940: 7469 6f6e 616c 5f6d 6574 6164 6174 613d  tional_metadata=
+00006950: 7b27 636f 6c75 6d6e 273a 205b 5d7d 2960  {'column': []})`
+00006960: 292e 0a2a 2060 5472 6565 4e6f 6465 2e64  )..* `TreeNode.d
+00006970: 6973 7461 6e63 6560 206e 6f77 2068 616e  istance` now han
+00006980: 646c 6573 2074 6865 2073 6974 7561 7469  dles the situati
+00006990: 6f6e 2069 6e20 7768 6963 6820 6073 656c  on in which `sel
+000069a0: 6660 206f 7220 606f 7468 6572 6020 6172  f` or `other` ar
+000069b0: 6520 616e 6365 7374 6f72 732e 2050 7265  e ancestors. Pre
+000069c0: 7669 6f73 6c79 2c20 6120 6e6f 6465 2066  viosly, a node f
+000069d0: 7572 7468 6572 2075 7020 7468 6520 7472  urther up the tr
+000069e0: 6565 2077 6173 2075 7365 6420 7265 7375  ee was used resu
+000069f0: 6c74 696e 6720 696e 2069 6e66 6c61 7465  lting in inflate
+00006a00: 6420 6469 7374 616e 6365 732e 2028 5b23  d distances. ([#
+00006a10: 3830 375d 2868 7474 7073 3a2f 2f67 6974  807](https://git
+00006a20: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+00006a30: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
+00006a40: 732f 3830 3729 290a 2a20 6054 7265 654e  s/807)).* `TreeN
+00006a50: 6f64 652e 7072 756e 6560 2063 616e 206e  ode.prune` can n
+00006a60: 6f77 2068 616e 646c 6520 6120 726f 6f74  ow handle a root
+00006a70: 2077 6974 6820 6120 7369 6e67 6c65 2064   with a single d
+00006a80: 6573 6365 6e64 656e 742e 2050 7265 7669  escendent. Previ
+00006a90: 6f75 736c 792c 2074 6865 2072 6f6f 7420  ously, the root 
+00006aa0: 7761 7320 6967 6e6f 7265 6420 6672 6f6d  was ignored from
+00006ab0: 2070 6f73 7369 626c 7920 6861 7669 6e67   possibly having
+00006ac0: 2061 2073 696e 676c 6520 6465 7363 656e   a single descen
+00006ad0: 6465 6e74 2e20 285b 2331 3234 375d 2868  dent. ([#1247](h
+00006ae0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+00006af0: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
+00006b00: 2d62 696f 2f69 7373 7565 732f 3132 3437  -bio/issues/1247
+00006b10: 2929 0a2a 2050 726f 7669 6469 6e67 2074  )).* Providing t
+00006b20: 6865 2060 666f 726d 6174 6020 6b65 7977  he `format` keyw
+00006b30: 6f72 6420 746f 2060 736b 6269 6f2e 696f  ord to `skbio.io
+00006b40: 2e72 6561 6460 2077 6865 6e20 6372 6561  .read` when crea
+00006b50: 7469 6e67 2061 2067 656e 6572 6174 6f72  ting a generator
+00006b60: 2077 6974 6820 616e 2065 6d70 7479 2066   with an empty f
+00006b70: 696c 6520 7769 6c6c 206e 6f77 2072 6574  ile will now ret
+00006b80: 7572 6e20 616e 2065 6d70 7479 2067 656e  urn an empty gen
+00006b90: 6572 6174 6f72 2069 6e73 7465 6164 206f  erator instead o
+00006ba0: 6620 7261 6973 696e 6720 6053 746f 7049  f raising `StopI
+00006bb0: 7465 7261 7469 6f6e 602e 2028 5b23 3133  teration`. ([#13
+00006bc0: 3133 5d28 6874 7470 733a 2f2f 6769 7468  13](https://gith
+00006bd0: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+00006be0: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
+00006bf0: 2f31 3331 3329 290a 2a20 604f 7264 696e  /1313)).* `Ordin
+00006c00: 6174 696f 6e52 6573 756c 7473 6020 6973  ationResults` is
+00006c10: 206e 6f77 2069 6d70 6f72 7461 626c 6520   now importable 
+00006c20: 6672 6f6d 2060 736b 6269 6f60 2061 6e64  from `skbio` and
+00006c30: 2060 736b 6269 6f2e 7374 6174 732e 6f72   `skbio.stats.or
+00006c40: 6469 6e61 7469 6f6e 6020 616e 6420 636f  dination` and co
+00006c50: 7272 6563 746c 7920 6c69 6e6b 6564 2066  rrectly linked f
+00006c60: 726f 6d20 7468 6520 646f 6375 6d65 6e74  rom the document
+00006c70: 6174 696f 6e20 285b 2331 3230 355d 2868  ation ([#1205](h
+00006c80: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+00006c90: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
+00006ca0: 2d62 696f 2f69 7373 7565 732f 3132 3035  -bio/issues/1205
+00006cb0: 2929 0a2a 2046 6978 6564 2070 6572 666f  )).* Fixed perfo
+00006cc0: 726d 616e 6365 2062 7567 2069 6e20 7061  rmance bug in pa
+00006cd0: 6972 7769 7365 2061 6c69 676e 6572 7320  irwise aligners 
+00006ce0: 7265 7375 6c74 696e 6720 696e 2031 3030  resulting in 100
+00006cf0: 7820 776f 7273 6520 7065 7266 6f72 6d61  x worse performa
+00006d00: 6e63 6520 7468 616e 2069 6e20 302e 322e  nce than in 0.2.
+00006d10: 342e 0a0a 2323 2320 4465 7072 6563 6174  4...### Deprecat
+00006d20: 6564 2066 756e 6374 696f 6e61 6c69 7479  ed functionality
+00006d30: 205b 7374 6162 6c65 5d0a 2a20 4465 7072   [stable].* Depr
+00006d40: 6563 6174 6564 2075 7365 206f 6620 7468  ecated use of th
+00006d50: 6520 7465 726d 2022 6e6f 6e2d 6465 6765  e term "non-dege
+00006d60: 6e65 7261 7465 222c 2069 6e20 6661 766f  nerate", in favo
+00006d70: 7220 6f66 2022 6465 6669 6e69 7465 222e  r of "definite".
+00006d80: 2060 4772 616d 6d61 7265 6453 6571 7565   `GrammaredSeque
+00006d90: 6e63 652e 6e6f 6e64 6567 656e 6572 6174  nce.nondegenerat
+00006da0: 655f 6368 6172 7360 2c20 6047 7261 6d6d  e_chars`, `Gramm
+00006db0: 6172 6564 5365 7175 656e 6365 2e6e 6f6e  aredSequence.non
+00006dc0: 6465 6765 6e65 7261 7465 7360 2c20 616e  degenerates`, an
+00006dd0: 6420 6047 7261 6d6d 6172 6564 5365 7175  d `GrammaredSequ
+00006de0: 656e 6365 2e68 6173 5f6e 6f6e 6465 6765  ence.has_nondege
+00006df0: 6e65 7261 7465 7360 2068 6176 6520 6265  nerates` have be
+00006e00: 656e 2072 656e 616d 6564 2074 6f20 6047  en renamed to `G
+00006e10: 7261 6d6d 6172 6564 5365 7175 656e 6365  rammaredSequence
+00006e20: 2e64 6566 696e 6974 655f 6368 6172 7360  .definite_chars`
+00006e30: 2c20 6047 7261 6d6d 6172 6564 5365 7175  , `GrammaredSequ
+00006e40: 656e 6365 2e64 6566 696e 6974 6573 602c  ence.definites`,
+00006e50: 2061 6e64 2060 4772 616d 6d61 7265 6453   and `GrammaredS
+00006e60: 6571 7565 6e63 652e 6861 735f 6465 6669  equence.has_defi
+00006e70: 6e69 7465 7360 2c20 7265 7370 6563 7469  nites`, respecti
+00006e80: 7665 6c79 2e20 5468 6520 6f6c 6420 6e61  vely. The old na
+00006e90: 6d65 7320 7769 6c6c 2062 6520 7265 6d6f  mes will be remo
+00006ea0: 7665 6420 696e 2073 6369 6b69 742d 6269  ved in scikit-bi
+00006eb0: 6f20 302e 352e 322e 2052 656c 6576 616e  o 0.5.2. Relevan
+00006ec0: 7420 6166 6665 6374 6564 2070 7562 6c69  t affected publi
+00006ed0: 6320 636c 6173 7365 7320 696e 636c 7564  c classes includ
+00006ee0: 6520 6047 7261 6d6d 6172 6564 5365 7175  e `GrammaredSequ
+00006ef0: 656e 6365 602c 2060 444e 4160 2c20 6052  ence`, `DNA`, `R
+00006f00: 4e41 602c 2061 6e64 2060 5072 6f74 6569  NA`, and `Protei
+00006f10: 6e60 2e0a 0a23 2323 2044 6570 7265 6361  n`...### Depreca
+00006f20: 7465 6420 6675 6e63 7469 6f6e 616c 6974  ted functionalit
+00006f30: 7920 5b65 7870 6572 696d 656e 7461 6c5d  y [experimental]
+00006f40: 0a2a 2044 6570 7265 6361 7465 6420 6675  .* Deprecated fu
+00006f50: 6e63 7469 6f6e 2060 736b 6269 6f2e 7574  nction `skbio.ut
+00006f60: 696c 2e63 7265 6174 655f 6469 7260 2e20  il.create_dir`. 
+00006f70: 5468 6973 2066 756e 6374 696f 6e20 7769  This function wi
+00006f80: 6c6c 2062 6520 7265 6d6f 7665 6420 696e  ll be removed in
+00006f90: 2073 6369 6b69 742d 6269 6f20 302e 352e   scikit-bio 0.5.
+00006fa0: 312e 2050 6c65 6173 6520 7573 6520 7468  1. Please use th
+00006fb0: 6520 5079 7468 6f6e 2073 7461 6e64 6172  e Python standar
+00006fc0: 6420 6c69 6272 6172 790a 6675 6e63 7469  d library.functi
+00006fd0: 6f6e 616c 6974 7920 6465 7363 7269 6265  onality describe
+00006fe0: 6420 5b68 6572 655d 2868 7474 7073 3a2f  d [here](https:/
+00006ff0: 2f64 6f63 732e 7079 7468 6f6e 2e6f 7267  /docs.python.org
+00007000: 2f32 2f6c 6962 7261 7279 2f6f 732e 6874  /2/library/os.ht
+00007010: 6d6c 236f 732e 6d61 6b65 6469 7273 292e  ml#os.makedirs).
+00007020: 2028 5b23 3833 335d 2868 7474 7073 3a2f   ([#833](https:/
+00007030: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
+00007040: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
+00007050: 7373 7565 732f 3833 3329 290a 2a20 4465  ssues/833)).* De
+00007060: 7072 6563 6174 6564 2066 756e 6374 696f  precated functio
+00007070: 6e20 6073 6b62 696f 2e75 7469 6c2e 7265  n `skbio.util.re
+00007080: 6d6f 7665 5f66 696c 6573 602e 2054 6869  move_files`. Thi
+00007090: 7320 6675 6e63 7469 6f6e 2077 696c 6c20  s function will 
+000070a0: 6265 2072 656d 6f76 6564 2069 6e20 7363  be removed in sc
+000070b0: 696b 6974 2d62 696f 2030 2e35 2e31 2e20  ikit-bio 0.5.1. 
+000070c0: 506c 6561 7365 2075 7365 2074 6865 2050  Please use the P
+000070d0: 7974 686f 6e20 7374 616e 6461 7264 206c  ython standard l
+000070e0: 6962 7261 7279 0a66 756e 6374 696f 6e61  ibrary.functiona
+000070f0: 6c69 7479 2064 6573 6372 6962 6564 205b  lity described [
+00007100: 6865 7265 5d28 6874 7470 733a 2f2f 646f  here](https://do
+00007110: 6373 2e70 7974 686f 6e2e 6f72 672f 322f  cs.python.org/2/
+00007120: 6c69 6272 6172 792f 6f73 2e68 746d 6c23  library/os.html#
+00007130: 6f73 2e72 656d 6f76 6529 2e20 285b 2338  os.remove). ([#8
+00007140: 3333 5d28 6874 7470 733a 2f2f 6769 7468  33](https://gith
+00007150: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+00007160: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
+00007170: 2f38 3333 2929 0a2a 2044 6570 7265 6361  /833)).* Depreca
+00007180: 7465 6420 6675 6e63 7469 6f6e 2060 736b  ted function `sk
+00007190: 6269 6f2e 7574 696c 2e69 735f 6361 7361  bio.util.is_casa
+000071a0: 7661 5f76 3138 305f 6f72 5f6c 6174 6572  va_v180_or_later
+000071b0: 602e 2054 6869 7320 6675 6e63 7469 6f6e  `. This function
+000071c0: 2077 696c 6c20 6265 2072 656d 6f76 6564   will be removed
+000071d0: 2069 6e20 302e 352e 312e 2046 756e 6374   in 0.5.1. Funct
+000071e0: 696f 6e61 6c69 7479 206d 6f76 6564 2074  ionality moved t
+000071f0: 6f20 4641 5354 5120 736e 6966 6665 722e  o FASTQ sniffer.
+00007200: 0a28 5b23 3833 335d 2868 7474 7073 3a2f  .([#833](https:/
+00007210: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
+00007220: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
+00007230: 7373 7565 732f 3833 3329 290a 0a23 2323  ssues/833))..###
+00007240: 204d 6973 6365 6c6c 616e 656f 7573 0a2a   Miscellaneous.*
+00007250: 2057 6865 6e20 696e 7374 616c 6c69 6e67   When installing
+00007260: 2073 6369 6b69 742d 6269 6f20 7669 6120   scikit-bio via 
+00007270: 6070 6970 602c 206e 756d 7079 206d 7573  `pip`, numpy mus
+00007280: 7420 6e6f 7720 6265 2069 6e73 7461 6c6c  t now be install
+00007290: 6564 2066 6972 7374 2028 5b23 3132 3936  ed first ([#1296
+000072a0: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+000072b0: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
+000072c0: 6b69 742d 6269 6f2f 6973 7375 6573 2f31  kit-bio/issues/1
+000072d0: 3239 3629 290a 0a23 2320 5665 7273 696f  296))..## Versio
+000072e0: 6e20 302e 342e 3220 2832 3031 362d 3032  n 0.4.2 (2016-02
+000072f0: 2d31 3729 0a0a 4d69 6e6f 7220 6d61 696e  -17)..Minor main
+00007300: 7465 6e61 6e63 6520 7265 6c65 6173 652e  tenance release.
+00007310: 202a 2a54 6869 7320 6973 2074 6865 206c   **This is the l
+00007320: 6173 7420 5079 7468 6f6e 2032 2e37 2063  ast Python 2.7 c
+00007330: 6f6d 7061 7469 626c 6520 7265 6c65 6173  ompatible releas
+00007340: 652e 2046 7574 7572 6520 7363 696b 6974  e. Future scikit
+00007350: 2d62 696f 2072 656c 6561 7365 7320 7769  -bio releases wi
+00007360: 6c6c 206f 6e6c 7920 7375 7070 6f72 7420  ll only support 
+00007370: 5079 7468 6f6e 2033 2e2a 2a0a 0a23 2323  Python 3.**..###
+00007380: 2046 6561 7475 7265 730a 2a20 4164 6465   Features.* Adde
+00007390: 6420 6073 6b62 696f 2e74 7265 652e 5472  d `skbio.tree.Tr
+000073a0: 6565 4e6f 6465 2e62 6966 7572 6361 7465  eeNode.bifurcate
+000073b0: 6020 666f 7220 636f 6e76 6572 7469 6e67  ` for converting
+000073c0: 206d 756c 7469 6675 7263 6174 696e 6720   multifurcating 
+000073d0: 7472 6565 7320 696e 746f 2062 6966 7572  trees into bifur
+000073e0: 6361 7469 6e67 2074 7265 6573 2e20 285b  cating trees. ([
+000073f0: 2338 3936 5d28 6874 7470 733a 2f2f 6769  #896](https://gi
+00007400: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
+00007410: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
+00007420: 6573 2f38 3936 2929 0a2a 2041 6464 6564  es/896)).* Added
+00007430: 2060 736b 6269 6f2e 696f 2e66 6f72 6d61   `skbio.io.forma
+00007440: 742e 7374 6f63 6b68 6f6c 6d60 2066 6f72  t.stockholm` for
+00007450: 2072 6561 6469 6e67 2053 746f 636b 686f   reading Stockho
+00007460: 6c6d 2066 696c 6573 2069 6e74 6f20 6120  lm files into a 
+00007470: 6054 6162 756c 6172 4d53 4160 2061 6e64  `TabularMSA` and
+00007480: 2077 7269 7469 6e67 2066 726f 6d20 6120   writing from a 
+00007490: 6054 6162 756c 6172 4d53 4160 2e20 285b  `TabularMSA`. ([
+000074a0: 2339 3637 5d28 6874 7470 733a 2f2f 6769  #967](https://gi
+000074b0: 7468 7562 2e63 6f6d 2f62 696f 636f 7265  thub.com/biocore
+000074c0: 2f73 6369 6b69 742d 6269 6f2f 6973 7375  /scikit-bio/issu
+000074d0: 6573 2f39 3637 2929 0a2a 2073 6369 6b69  es/967)).* sciki
+000074e0: 742d 6269 6f20 6053 6571 7565 6e63 6560  t-bio `Sequence`
+000074f0: 206f 626a 6563 7473 2068 6176 6520 6265   objects have be
+00007500: 7474 6572 2063 6f6d 7061 7469 6269 6c69  tter compatibili
+00007510: 7479 2077 6974 6820 6e75 6d70 792e 2046  ty with numpy. F
+00007520: 6f72 2065 7861 6d70 6c65 2c20 6361 6c6c  or example, call
+00007530: 696e 6720 606e 702e 6173 6172 7261 7928  ing `np.asarray(
+00007540: 7365 7175 656e 6365 2960 206e 6f77 2063  sequence)` now c
+00007550: 6f6e 7665 7274 7320 7468 6520 7365 7175  onverts the sequ
+00007560: 656e 6365 2074 6f20 6120 6e75 6d70 7920  ence to a numpy 
+00007570: 6172 7261 7920 6f66 2063 6861 7261 6374  array of charact
+00007580: 6572 7320 2874 6865 2073 616d 6520 6173  ers (the same as
+00007590: 2063 616c 6c69 6e67 2060 7365 7175 656e   calling `sequen
+000075a0: 6365 2e76 616c 7565 7360 292e 0a2a 2041  ce.values`)..* A
+000075b0: 6464 6564 2060 736b 6269 6f2e 7365 7175  dded `skbio.sequ
+000075c0: 656e 6365 2e64 6973 7461 6e63 6560 2073  ence.distance` s
+000075d0: 7562 7061 636b 6167 6520 666f 7220 636f  ubpackage for co
+000075e0: 6d70 7574 696e 6720 6469 7374 616e 6365  mputing distance
+000075f0: 7320 6265 7477 6565 6e20 7363 696b 6974  s between scikit
+00007600: 2d62 696f 2060 5365 7175 656e 6365 6020  -bio `Sequence` 
+00007610: 6f62 6a65 6374 7320 285b 2339 3133 5d28  objects ([#913](
+00007620: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+00007630: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+00007640: 742d 6269 6f2f 6973 7375 6573 2f39 3133  t-bio/issues/913
+00007650: 2929 0a2a 2041 6464 6564 2060 6073 6b62  )).* Added ``skb
+00007660: 696f 2e73 6571 7565 6e63 652e 4772 616d  io.sequence.Gram
+00007670: 6d61 7265 6453 6571 7565 6e63 6560 602c  maredSequence``,
+00007680: 2077 6869 6368 2063 616e 2062 6520 696e   which can be in
+00007690: 6865 7269 7465 6420 6672 6f6d 2074 6f20  herited from to 
+000076a0: 6372 6561 7465 2067 7261 6d6d 6172 6564  create grammared
+000076b0: 2073 6571 7565 6e63 6573 2077 6974 6820   sequences with 
+000076c0: 6375 7374 6f6d 2061 6c70 6861 6265 7473  custom alphabets
+000076d0: 2028 652e 672e 2c20 666f 7220 7573 6520   (e.g., for use 
+000076e0: 7769 7468 2054 6162 756c 6172 4d53 4129  with TabularMSA)
+000076f0: 2028 5b23 3131 3735 5d28 6874 7470 733a   ([#1175](https:
+00007700: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
+00007710: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
+00007720: 6973 7375 6573 2f31 3137 3529 290a 2a20  issues/1175)).* 
+00007730: 4164 6465 6420 6060 736b 6269 6f2e 7574  Added ``skbio.ut
+00007740: 696c 2e63 6c61 7373 7072 6f70 6572 7479  il.classproperty
+00007750: 6060 2064 6563 6f72 6174 6f72 0a0a 2323  `` decorator..##
+00007760: 2320 4261 636b 7761 7264 2d69 6e63 6f6d  # Backward-incom
+00007770: 7061 7469 626c 6520 6368 616e 6765 7320  patible changes 
+00007780: 5b73 7461 626c 655d 0a2a 2057 6865 6e20  [stable].* When 
+00007790: 736e 6966 6669 6e67 206f 7220 7265 6164  sniffing or read
+000077a0: 696e 6720 6120 6669 6c65 2028 6073 6b62  ing a file (`skb
+000077b0: 696f 2e69 6f2e 736e 6966 6660 2c20 6073  io.io.sniff`, `s
+000077c0: 6b62 696f 2e69 6f2e 7265 6164 602c 206f  kbio.io.read`, o
+000077d0: 7220 7468 6520 6f62 6a65 6374 2d6f 7269  r the object-ori
+000077e0: 656e 7465 6420 602e 7265 6164 2829 6020  ented `.read()` 
+000077f0: 696e 7465 7266 6163 6529 2c20 7061 7373  interface), pass
+00007800: 696e 6720 606e 6577 6c69 6e65 6020 6173  ing `newline` as
+00007810: 2061 206b 6579 776f 7264 2061 7267 756d   a keyword argum
+00007820: 656e 7420 746f 2060 736b 6269 6f2e 696f  ent to `skbio.io
+00007830: 2e6f 7065 6e60 206e 6f77 2072 6169 7365  .open` now raise
+00007840: 7320 6120 6054 7970 6545 7272 6f72 602e  s a `TypeError`.
+00007850: 2054 6869 7320 6261 636b 7761 7264 2d69   This backward-i
+00007860: 6e63 6f6d 7061 7469 626c 6520 6368 616e  ncompatible chan
+00007870: 6765 2074 6f20 6120 7374 6162 6c65 2041  ge to a stable A
+00007880: 5049 2069 7320 6e65 6365 7373 6172 7920  PI is necessary 
+00007890: 6265 6361 7573 6520 6974 2066 6978 6573  because it fixes
+000078a0: 2061 2062 7567 2028 6d6f 7265 2064 6574   a bug (more det
+000078b0: 6169 6c73 2069 6e20 6275 6720 6669 7820  ails in bug fix 
+000078c0: 7365 6374 696f 6e20 6265 6c6f 7729 2e0a  section below)..
+000078d0: 2a20 5768 656e 2072 6561 6469 6e67 2061  * When reading a
+000078e0: 2046 4153 5451 206f 7220 5153 4551 2066   FASTQ or QSEQ f
+000078f0: 696c 6520 616e 6420 7061 7373 696e 6720  ile and passing 
+00007900: 6076 6172 6961 6e74 3d27 736f 6c65 7861  `variant='solexa
+00007910: 2760 2c20 6056 616c 7565 4572 726f 7260  '`, `ValueError`
+00007920: 2069 7320 6e6f 7720 7261 6973 6564 2069   is now raised i
+00007930: 6e73 7465 6164 206f 6620 604e 6f74 496d  nstead of `NotIm
+00007940: 706c 656d 656e 7465 6445 7272 6f72 602e  plementedError`.
+00007950: 2054 6869 7320 6261 636b 7761 7264 2d69   This backward-i
+00007960: 6e63 6f6d 7061 7469 626c 6520 6368 616e  ncompatible chan
+00007970: 6765 2074 6f20 6120 7374 6162 6c65 2041  ge to a stable A
+00007980: 5049 2069 7320 6e65 6365 7373 6172 7920  PI is necessary 
+00007990: 746f 2061 766f 6964 2063 7265 6174 696e  to avoid creatin
+000079a0: 6720 6120 7370 696e 2d6c 6f63 6b65 6420  g a spin-locked 
+000079b0: 7072 6f63 6573 7320 6475 6520 746f 205b  process due to [
+000079c0: 6120 6275 6720 696e 2050 7974 686f 6e5d  a bug in Python]
+000079d0: 2868 7474 7073 3a2f 2f62 7567 732e 7079  (https://bugs.py
+000079e0: 7468 6f6e 2e6f 7267 2f69 7373 7565 3235  thon.org/issue25
+000079f0: 3738 3629 2e20 5365 6520 5b23 3132 3536  786). See [#1256
+00007a00: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+00007a10: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
+00007a20: 6b69 742d 6269 6f2f 6973 7375 6573 2f31  kit-bio/issues/1
+00007a30: 3235 3629 2066 6f72 2064 6574 6169 6c73  256) for details
+00007a40: 2e20 5468 6973 2063 6861 6e67 6520 6973  . This change is
+00007a50: 2074 656d 706f 7261 7279 2061 6e64 2077   temporary and w
+00007a60: 696c 6c20 6265 2072 6576 6572 7465 6420  ill be reverted 
+00007a70: 746f 2060 4e6f 7449 6d70 6c65 6d65 6e74  to `NotImplement
+00007a80: 6564 4572 726f 7260 2077 6865 6e20 7468  edError` when th
+00007a90: 6520 6275 6720 6973 2066 6978 6564 2069  e bug is fixed i
+00007aa0: 6e20 5079 7468 6f6e 2e0a 0a23 2323 2042  n Python...### B
+00007ab0: 6163 6b77 6172 642d 696e 636f 6d70 6174  ackward-incompat
+00007ac0: 6962 6c65 2063 6861 6e67 6573 205b 6578  ible changes [ex
+00007ad0: 7065 7269 6d65 6e74 616c 5d0a 2a20 6073  perimental].* `s
+00007ae0: 6b62 696f 2e69 6f2e 666f 726d 6174 2e67  kbio.io.format.g
+00007af0: 656e 6261 6e6b 603a 2057 6865 6e20 7265  enbank`: When re
+00007b00: 6164 696e 6720 4765 6e42 616e 6b20 6669  ading GenBank fi
+00007b10: 6c65 732c 2074 6865 2064 6174 6520 6669  les, the date fi
+00007b20: 656c 6420 6f66 2074 6865 204c 4f43 5553  eld of the LOCUS
+00007b30: 206c 696e 6520 6973 206e 6f20 6c6f 6e67   line is no long
+00007b40: 6572 2070 6172 7365 6420 696e 746f 2061  er parsed into a
+00007b50: 2060 6461 7465 7469 6d65 2e64 6174 6574   `datetime.datet
+00007b60: 696d 6560 206f 626a 6563 7420 616e 6420  ime` object and 
+00007b70: 6973 206c 6566 7420 6173 2061 2073 7472  is left as a str
+00007b80: 696e 672e 2057 6865 6e20 7772 6974 696e  ing. When writin
+00007b90: 6720 4765 6e42 616e 6b20 6669 6c65 732c  g GenBank files,
+00007ba0: 2074 6865 206c 6f63 7573 2064 6174 6520   the locus date 
+00007bb0: 6d65 7461 6461 7461 2069 7320 6578 7065  metadata is expe
+00007bc0: 6374 6564 2074 6f20 6265 2061 2073 7472  cted to be a str
+00007bd0: 696e 6720 696e 7374 6561 6420 6f66 2061  ing instead of a
+00007be0: 2060 6461 7465 7469 6d65 2e64 6174 6574   `datetime.datet
+00007bf0: 696d 6560 206f 626a 6563 7420 285b 2331  ime` object ([#1
+00007c00: 3135 335d 2868 7474 7073 3a2f 2f67 6974  153](https://git
+00007c10: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+00007c20: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
+00007c30: 732f 3131 3533 2929 0a2a 2060 5365 7175  s/1153)).* `Sequ
+00007c40: 656e 6365 2e64 6973 7461 6e63 6560 206e  ence.distance` n
+00007c50: 6f77 2063 6f6e 7665 7274 7320 7468 6520  ow converts the 
+00007c60: 696e 7075 7420 7365 7175 656e 6365 2028  input sequence (
+00007c70: 606f 7468 6572 6029 2074 6f20 6974 7320  `other`) to its 
+00007c80: 7479 7065 2062 6566 6f72 6520 7061 7373  type before pass
+00007c90: 696e 6720 626f 7468 2073 6571 7565 6e63  ing both sequenc
+00007ca0: 6573 2074 6f20 606d 6574 7269 6360 2e20  es to `metric`. 
+00007cb0: 5072 6576 696f 7573 2062 6568 6176 696f  Previous behavio
+00007cc0: 7220 7761 7320 746f 2061 6c77 6179 7320  r was to always 
+00007cd0: 636f 6e76 6572 7420 746f 2060 5365 7175  convert to `Sequ
+00007ce0: 656e 6365 602e 0a0a 2323 2320 4275 6720  ence`...### Bug 
+00007cf0: 6669 7865 730a 2a20 4669 7865 6420 6275  fixes.* Fixed bu
+00007d00: 6720 7768 656e 2075 7369 6e67 2060 5365  g when using `Se
+00007d10: 7175 656e 6365 2e64 6973 7461 6e63 6560  quence.distance`
+00007d20: 206f 7220 6044 6973 7461 6e63 654d 6174   or `DistanceMat
+00007d30: 7269 782e 6672 6f6d 5f69 7465 7261 626c  rix.from_iterabl
+00007d40: 6560 2074 6f20 636f 6d70 7574 6520 6469  e` to compute di
+00007d50: 7374 616e 6365 7320 6265 7477 6565 6e20  stances between 
+00007d60: 6053 6571 7565 6e63 6560 206f 626a 6563  `Sequence` objec
+00007d70: 7473 2077 6974 6820 6469 6666 6572 696e  ts with differin
+00007d80: 6720 606d 6574 6164 6174 6160 2f60 706f  g `metadata`/`po
+00007d90: 7369 7469 6f6e 616c 5f6d 6574 6164 6174  sitional_metadat
+00007da0: 6160 2061 6e64 2070 6173 7369 6e67 2060  a` and passing `
+00007db0: 6d65 7472 6963 3d73 6369 7079 2e73 7061  metric=scipy.spa
+00007dc0: 7469 616c 2e64 6973 7461 6e63 652e 6861  tial.distance.ha
+00007dd0: 6d6d 696e 6760 2028 5b23 3132 3534 5d28  mming` ([#1254](
+00007de0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+00007df0: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+00007e00: 742d 6269 6f2f 6973 7375 6573 2f31 3235  t-bio/issues/125
+00007e10: 3429 290a 2a20 4669 7865 6420 7065 7266  4)).* Fixed perf
+00007e20: 6f72 6d61 6e63 6520 6275 6720 7768 656e  ormance bug when
+00007e30: 2063 6f6d 7075 7469 6e67 2048 616d 6d69   computing Hammi
+00007e40: 6e67 2064 6973 7461 6e63 6573 2062 6574  ng distances bet
+00007e50: 7765 656e 2060 5365 7175 656e 6365 6020  ween `Sequence` 
+00007e60: 6f62 6a65 6374 7320 696e 2060 4469 7374  objects in `Dist
+00007e70: 616e 6365 4d61 7472 6978 2e66 726f 6d5f  anceMatrix.from_
+00007e80: 6974 6572 6162 6c65 6020 285b 2331 3235  iterable` ([#125
+00007e90: 305d 2868 7474 7073 3a2f 2f67 6974 6875  0](https://githu
+00007ea0: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
+00007eb0: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
+00007ec0: 3132 3530 2929 0a2a 2043 6861 6e67 6564  1250)).* Changed
+00007ed0: 2060 736b 6269 6f2e 7374 6174 732e 636f   `skbio.stats.co
+00007ee0: 6d70 6f73 6974 696f 6e2e 6d75 6c74 6970  mposition.multip
+00007ef0: 6c69 6361 7469 7665 5f72 6570 6c61 6365  licative_replace
+00007f00: 6d65 6e74 6020 746f 2072 6169 7365 2061  ment` to raise a
+00007f10: 6e20 6572 726f 7220 7768 656e 6576 6572  n error whenever
+00007f20: 2061 206c 6172 6765 2076 616c 7565 206f   a large value o
+00007f30: 6620 6064 656c 7461 6020 6973 2063 686f  f `delta` is cho
+00007f40: 7365 6e20 285b 2331 3234 315d 2868 7474  sen ([#1241](htt
+00007f50: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+00007f60: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
+00007f70: 696f 2f69 7373 7565 732f 3132 3431 2929  io/issues/1241))
+00007f80: 0a2a 2057 6865 6e20 736e 6966 6669 6e67  .* When sniffing
+00007f90: 206f 7220 7265 6164 696e 6720 6120 6669   or reading a fi
+00007fa0: 6c65 2028 6073 6b62 696f 2e69 6f2e 736e  le (`skbio.io.sn
+00007fb0: 6966 6660 2c20 6073 6b62 696f 2e69 6f2e  iff`, `skbio.io.
+00007fc0: 7265 6164 602c 206f 7220 7468 6520 6f62  read`, or the ob
+00007fd0: 6a65 6374 2d6f 7269 656e 7465 6420 602e  ject-oriented `.
+00007fe0: 7265 6164 2829 6020 696e 7465 7266 6163  read()` interfac
+00007ff0: 6529 2c20 7061 7373 696e 6720 606e 6577  e), passing `new
+00008000: 6c69 6e65 6020 6173 2061 206b 6579 776f  line` as a keywo
+00008010: 7264 2061 7267 756d 656e 7420 746f 2060  rd argument to `
+00008020: 736b 6269 6f2e 696f 2e6f 7065 6e60 206e  skbio.io.open` n
+00008030: 6f77 2072 6169 7365 7320 6120 6054 7970  ow raises a `Typ
+00008040: 6545 7272 6f72 602e 2054 6865 2066 696c  eError`. The fil
+00008050: 6520 666f 726d 6174 2773 2060 6e65 776c  e format's `newl
+00008060: 696e 6560 2063 6861 7261 6374 6572 2077  ine` character w
+00008070: 696c 6c20 6265 2075 7365 6420 7768 656e  ill be used when
+00008080: 206f 7065 6e69 6e67 2074 6865 2066 696c   opening the fil
+00008090: 652e 2050 7265 7669 6f75 7320 6265 6861  e. Previous beha
+000080a0: 7669 6f72 2061 6c6c 6f77 6564 206f 7665  vior allowed ove
+000080b0: 7272 6964 696e 6720 7468 6520 666f 726d  rriding the form
+000080c0: 6174 2773 2060 6e65 776c 696e 6560 2063  at's `newline` c
+000080d0: 6861 7261 6374 6572 2062 7574 2074 6869  haracter but thi
+000080e0: 7320 636f 756c 6420 6361 7573 6520 6973  s could cause is
+000080f0: 7375 6573 2077 6974 6820 7265 6164 6572  sues with reader
+00008100: 7320 7468 6174 2061 7373 756d 6520 6e65  s that assume ne
+00008110: 776c 696e 6520 6368 6172 6163 7465 7273  wline characters
+00008120: 2061 7265 2074 686f 7365 2064 6566 696e   are those defin
+00008130: 6564 2062 7920 7468 6520 6669 6c65 2066  ed by the file f
+00008140: 6f72 6d61 7420 2877 6869 6368 2069 7320  ormat (which is 
+00008150: 616e 2065 6e74 6972 656c 7920 7265 6173  an entirely reas
+00008160: 6f6e 6162 6c65 2061 7373 756d 7074 696f  onable assumptio
+00008170: 6e29 2e20 5468 6973 2062 7567 2069 7320  n). This bug is 
+00008180: 7665 7279 2075 6e6c 696b 656c 7920 746f  very unlikely to
+00008190: 2068 6176 6520 7375 7266 6163 6564 2069   have surfaced i
+000081a0: 6e20 7072 6163 7469 6365 2061 7320 7468  n practice as th
+000081b0: 6520 6465 6661 756c 7420 606e 6577 6c69  e default `newli
+000081c0: 6e65 6020 6265 6861 7669 6f72 2069 7320  ne` behavior is 
+000081d0: 2a75 6e69 7665 7273 616c 206e 6577 6c69  *universal newli
+000081e0: 6e65 7320 6d6f 6465 2a2e 0a2a 2044 4e41  nes mode*..* DNA
+000081f0: 2c20 524e 412c 2061 6e64 2050 726f 7465  , RNA, and Prote
+00008200: 696e 2061 7265 206e 6f20 6c6f 6e67 6572  in are no longer
+00008210: 2069 6e68 6572 6974 6162 6c65 2062 6563   inheritable bec
+00008220: 6175 7365 2074 6865 7920 6173 7375 6d65  ause they assume
+00008230: 2061 6e20 4955 5041 4320 616c 7068 6162   an IUPAC alphab
+00008240: 6574 2e0a 2a20 6044 6973 7461 6e63 654d  et..* `DistanceM
+00008250: 6174 7269 7860 2063 6f6e 7374 7275 6374  atrix` construct
+00008260: 6f72 2070 726f 7669 6465 7320 6d6f 7265  or provides more
+00008270: 2069 6e66 6f72 6d61 7469 7665 2065 7272   informative err
+00008280: 6f72 206d 6573 7361 6765 2077 6865 6e20  or message when 
+00008290: 6461 7461 2063 6f6e 7461 696e 7320 4e61  data contains Na
+000082a0: 4e73 2028 5b23 3132 3736 5d28 6874 7470  Ns ([#1276](http
+000082b0: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+000082c0: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+000082d0: 6f2f 6973 7375 6573 2f31 3237 3629 290a  o/issues/1276)).
+000082e0: 0a23 2323 204d 6973 6365 6c6c 616e 656f  .### Miscellaneo
+000082f0: 7573 0a2a 2057 6172 6e69 6e67 7320 7261  us.* Warnings ra
+00008300: 6973 6564 2062 7920 7363 696b 6974 2d62  ised by scikit-b
+00008310: 696f 206e 6f77 2073 6861 7265 2061 2063  io now share a c
+00008320: 6f6d 6d6f 6e20 7375 6263 6c61 7373 2060  ommon subclass `
+00008330: 6073 6b62 696f 2e75 7469 6c2e 536b 6269  `skbio.util.Skbi
+00008340: 6f57 6172 6e69 6e67 6060 2e0a 0a23 2320  oWarning``...## 
+00008350: 5665 7273 696f 6e20 302e 342e 3120 2832  Version 0.4.1 (2
+00008360: 3031 352d 3132 2d30 3929 0a0a 2323 2320  015-12-09)..### 
+00008370: 4665 6174 7572 6573 0a2a 2054 6865 2060  Features.* The `
+00008380: 6054 6162 756c 6172 4d53 4160 6020 6f62  `TabularMSA`` ob
+00008390: 6a65 6374 2077 6173 2061 6464 6564 2074  ject was added t
+000083a0: 6f20 7265 7072 6573 656e 7420 616e 6420  o represent and 
+000083b0: 6f70 6572 6174 6520 6f6e 2074 6162 756c  operate on tabul
+000083c0: 6172 206d 756c 7469 706c 6520 7365 7175  ar multiple sequ
+000083d0: 656e 6365 2061 6c69 676e 6d65 6e74 732e  ence alignments.
+000083e0: 2054 6869 7320 7361 7469 7366 6965 7320   This satisfies 
+000083f0: 5b52 4643 2031 5d28 6874 7470 733a 2f2f  [RFC 1](https://
+00008400: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
+00008410: 7265 2f73 6369 6b69 742d 6269 6f2d 7266  re/scikit-bio-rf
+00008420: 6373 2f62 6c6f 622f 6d61 7374 6572 2f61  cs/blob/master/a
+00008430: 6374 6976 652f 3030 312d 7461 6275 6c61  ctive/001-tabula
+00008440: 722d 6d73 612e 6d64 292e 2053 6565 2074  r-msa.md). See t
+00008450: 6865 2060 6054 6162 756c 6172 4d53 4160  he ``TabularMSA`
+00008460: 6020 646f 6373 2066 6f72 2066 756c 6c20  ` docs for full 
+00008470: 6465 7461 696c 732e 0a2a 2041 6464 6564  details..* Added
+00008480: 2070 6879 6c6f 6765 6e65 7469 6320 6469   phylogenetic di
+00008490: 7665 7273 6974 7920 6d65 7472 6963 732c  versity metrics,
+000084a0: 2069 6e63 6c75 6469 6e67 2077 6569 6768   including weigh
+000084b0: 7465 6420 556e 6946 7261 632c 2075 6e77  ted UniFrac, unw
+000084c0: 6569 6768 7465 6420 556e 6946 7261 632c  eighted UniFrac,
+000084d0: 2061 6e64 2046 6169 7468 2773 2050 6879   and Faith's Phy
+000084e0: 6c6f 6765 6e65 7469 6320 4469 7665 7273  logenetic Divers
+000084f0: 6974 792e 2054 6865 7365 2061 7265 2061  ity. These are a
+00008500: 6363 6573 7369 626c 6520 6173 2060 6073  ccessible as ``s
+00008510: 6b62 696f 2e64 6976 6572 7369 7479 2e62  kbio.diversity.b
+00008520: 6574 612e 756e 7765 6967 6874 6564 5f75  eta.unweighted_u
+00008530: 6e69 6672 6163 6060 2c20 6060 736b 6269  nifrac``, ``skbi
+00008540: 6f2e 6469 7665 7273 6974 792e 6265 7461  o.diversity.beta
+00008550: 2e77 6569 6768 7465 645f 756e 6966 7261  .weighted_unifra
+00008560: 6360 602c 2061 6e64 2060 6073 6b62 696f  c``, and ``skbio
+00008570: 2e64 6976 6572 7369 7479 2e61 6c70 6861  .diversity.alpha
+00008580: 2e66 6169 7468 5f70 6460 602c 2072 6573  .faith_pd``, res
+00008590: 7065 6374 6976 656c 792e 0a2a 2041 6464  pectively..* Add
+000085a0: 6974 696f 6e20 6f66 2074 6865 2066 756e  ition of the fun
+000085b0: 6374 696f 6e20 6060 736b 6269 6f2e 6469  ction ``skbio.di
+000085c0: 7665 7273 6974 792e 616c 7068 615f 6469  versity.alpha_di
+000085d0: 7665 7273 6974 7960 6020 746f 2073 7570  versity`` to sup
+000085e0: 706f 7274 2061 7070 6c79 696e 6720 616e  port applying an
+000085f0: 2061 6c70 6861 2064 6976 6572 7369 7479   alpha diversity
+00008600: 206d 6574 7269 6320 746f 206d 756c 7469   metric to multi
+00008610: 706c 6520 7361 6d70 6c65 7320 696e 206f  ple samples in o
+00008620: 6e65 2063 616c 6c2e 0a2a 2041 6464 6974  ne call..* Addit
+00008630: 696f 6e20 6f66 2074 6865 2066 756e 6374  ion of the funct
+00008640: 696f 6e73 2060 6073 6b62 696f 2e64 6976  ions ``skbio.div
+00008650: 6572 7369 7479 2e67 6574 5f61 6c70 6861  ersity.get_alpha
+00008660: 5f64 6976 6572 7369 7479 5f6d 6574 7269  _diversity_metri
+00008670: 6373 6060 2061 6e64 2060 6073 6b62 696f  cs`` and ``skbio
+00008680: 2e64 6976 6572 7369 7479 2e67 6574 5f62  .diversity.get_b
+00008690: 6574 615f 6469 7665 7273 6974 795f 6d65  eta_diversity_me
+000086a0: 7472 6963 7360 6020 746f 2073 7570 706f  trics`` to suppo
+000086b0: 7274 2064 6973 636f 7665 7279 206f 6620  rt discovery of 
+000086c0: 7468 6520 616c 7068 6120 616e 6420 6265  the alpha and be
+000086d0: 7461 2064 6976 6572 7369 7479 206d 6574  ta diversity met
+000086e0: 7269 6373 2069 6d70 6c65 6d65 6e74 6564  rics implemented
+000086f0: 2069 6e20 7363 696b 6974 2d62 696f 2e0a   in scikit-bio..
+00008700: 2a20 4164 6465 6420 6073 6b62 696f 2e73  * Added `skbio.s
+00008710: 7461 7473 2e63 6f6d 706f 7369 7469 6f6e  tats.composition
+00008720: 2e61 6e63 6f6d 6020 6675 6e63 7469 6f6e  .ancom` function
+00008730: 2c20 6120 7465 7374 2066 6f72 204f 5455  , a test for OTU
+00008740: 2064 6966 6665 7265 6e74 6961 6c20 6162   differential ab
+00008750: 756e 6461 6e63 6520 6163 726f 7373 2073  undance across s
+00008760: 616d 706c 6520 6361 7465 676f 7269 6573  ample categories
+00008770: 2e20 285b 2331 3035 345d 2868 7474 7073  . ([#1054](https
+00008780: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
+00008790: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
+000087a0: 2f69 7373 7565 732f 3130 3534 2929 0a2a  /issues/1054)).*
+000087b0: 2041 6464 6564 2060 736b 6269 6f2e 696f   Added `skbio.io
+000087c0: 2e66 6f72 6d61 742e 626c 6173 7437 6020  .format.blast7` 
+000087d0: 666f 7220 7265 6164 696e 6720 424c 4153  for reading BLAS
+000087e0: 542b 206f 7574 7075 7420 666f 726d 6174  T+ output format
+000087f0: 2037 206f 7220 424c 4153 5420 6f75 7470   7 or BLAST outp
+00008800: 7574 2066 6f72 6d61 7420 3920 6669 6c65  ut format 9 file
+00008810: 7320 696e 746f 2061 2060 7064 2e44 6174  s into a `pd.Dat
+00008820: 6146 7261 6d65 602e 2028 5b23 3131 3130  aFrame`. ([#1110
+00008830: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+00008840: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
+00008850: 6b69 742d 6269 6f2f 6973 7375 6573 2f31  kit-bio/issues/1
+00008860: 3131 3029 290a 2a20 4164 6465 6420 6073  110)).* Added `s
+00008870: 6b62 696f 2e44 6973 7369 6d69 6c61 7269  kbio.Dissimilari
+00008880: 7479 4d61 7472 6978 2e74 6f5f 6461 7461  tyMatrix.to_data
+00008890: 5f66 7261 6d65 6020 6d65 7468 6f64 2066  _frame` method f
+000088a0: 6f72 2063 7265 6174 696e 6720 6120 6060  or creating a ``
+000088b0: 7061 6e64 6173 2e44 6174 6146 7261 6d65  pandas.DataFrame
+000088c0: 6060 2066 726f 6d20 6120 6044 6973 7369  `` from a `Dissi
+000088d0: 6d69 6c61 7269 7479 4d61 7472 6978 6020  milarityMatrix` 
+000088e0: 6f72 2060 4469 7374 616e 6365 4d61 7472  or `DistanceMatr
+000088f0: 6978 602e 2028 5b23 3735 375d 2868 7474  ix`. ([#757](htt
+00008900: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+00008910: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
+00008920: 696f 2f69 7373 7565 732f 3735 3729 290a  io/issues/757)).
+00008930: 2a20 4164 6465 6420 7375 7070 6f72 7420  * Added support 
+00008940: 666f 7220 6f6e 652d 6469 6d65 6e73 696f  for one-dimensio
+00008950: 6e61 6c20 7665 6374 6f72 206f 6620 6469  nal vector of di
+00008960: 7373 696d 696c 6172 6974 6965 7320 696e  ssimilarities in
+00008970: 2060 736b 6269 6f2e 7374 6174 732e 6469   `skbio.stats.di
+00008980: 7374 616e 6365 2e44 6973 7369 6d69 6c61  stance.Dissimila
+00008990: 7269 7479 4d61 7472 6978 600a 636f 6e73  rityMatrix`.cons
+000089a0: 7472 7563 746f 722e 2028 5b23 3632 3430  tructor. ([#6240
+000089b0: 5d28 6874 7470 733a 2f2f 6769 7468 7562  ](https://github
+000089c0: 2e63 6f6d 2f62 696f 636f 7265 2f73 6369  .com/biocore/sci
+000089d0: 6b69 742d 6269 6f2f 6973 7375 6573 2f36  kit-bio/issues/6
+000089e0: 3234 2929 0a2a 2041 6464 6564 2060 736b  24)).* Added `sk
+000089f0: 6269 6f2e 696f 2e66 6f72 6d61 742e 626c  bio.io.format.bl
+00008a00: 6173 7436 6020 666f 7220 7265 6164 696e  ast6` for readin
+00008a10: 6720 424c 4153 542b 206f 7574 7075 7420  g BLAST+ output 
+00008a20: 666f 726d 6174 2036 206f 7220 424c 4153  format 6 or BLAS
+00008a30: 5420 6f75 7470 7574 2066 6f72 6d61 7420  T output format 
+00008a40: 3820 6669 6c65 7320 696e 746f 2061 2060  8 files into a `
+00008a50: 7064 2e44 6174 6146 7261 6d65 602e 2028  pd.DataFrame`. (
+00008a60: 5b23 3131 3130 5d28 6874 7470 733a 2f2f  [#1110](https://
+00008a70: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
+00008a80: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
+00008a90: 7375 6573 2f31 3131 3029 290a 2a20 4164  sues/1110)).* Ad
+00008aa0: 6465 6420 6069 6e6e 6572 602c 2060 696c  ded `inner`, `il
+00008ab0: 7260 2c20 6069 6c72 5f69 6e76 6020 616e  r`, `ilr_inv` an
+00008ac0: 6420 6063 6c72 5f69 6e76 602c 2060 6073  d `clr_inv`, ``s
+00008ad0: 6b62 696f 2e73 7461 7473 2e63 6f6d 706f  kbio.stats.compo
+00008ae0: 7369 7469 6f6e 6060 2c20 7768 6963 6820  sition``, which 
+00008af0: 656e 6162 6c65 7320 6c69 6e65 6172 2074  enables linear t
+00008b00: 7261 6e73 666f 726d 6174 696f 6e73 206f  ransformations o
+00008b10: 6e20 636f 6d70 6f73 6974 696f 6e73 2028  n compositions (
+00008b20: 5b23 3839 325d 2868 7474 7073 3a2f 2f67  [#892](https://g
+00008b30: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
+00008b40: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
+00008b50: 7565 732f 3839 3229 0a2a 2041 6464 6564  ues/892).* Added
+00008b60: 2060 6073 6b62 696f 2e64 6976 6572 7369   ``skbio.diversi
+00008b70: 7479 2e61 6c70 6861 2e70 6965 6c6f 755f  ty.alpha.pielou_
+00008b80: 6560 6020 6675 6e63 7469 6f6e 2061 7320  e`` function as 
+00008b90: 616e 2065 7665 6e6e 6573 7320 6d65 7472  an evenness metr
+00008ba0: 6963 206f 6620 616c 7068 6120 6469 7665  ic of alpha dive
+00008bb0: 7273 6974 792e 2028 5b23 3130 3638 5d28  rsity. ([#1068](
+00008bc0: 6874 7470 733a 2f2f 6769 7468 7562 2e63  https://github.c
+00008bd0: 6f6d 2f62 696f 636f 7265 2f73 6369 6b69  om/biocore/sciki
+00008be0: 742d 6269 6f2f 6973 7375 6573 2f31 3036  t-bio/issues/106
+00008bf0: 3829 290a 2a20 4164 6465 6420 6074 6f5f  8)).* Added `to_
+00008c00: 7265 6765 7860 206d 6574 686f 6420 746f  regex` method to
+00008c10: 2060 736b 6269 6f2e 7365 7175 656e 6365   `skbio.sequence
+00008c20: 2e5f 6975 7061 635f 7365 7175 656e 6365  ._iupac_sequence
+00008c30: 6020 4142 4320 2d20 6974 2072 6574 7572  ` ABC - it retur
+00008c40: 6e73 2061 2072 6567 6578 206f 626a 6563  ns a regex objec
+00008c50: 7420 7468 6174 206d 6174 6368 6573 2061  t that matches a
+00008c60: 6c6c 206e 6f6e 2d64 6567 656e 6572 6174  ll non-degenerat
+00008c70: 6520 7665 7273 696f 6e73 206f 6620 7468  e versions of th
+00008c80: 6520 7365 7175 656e 6365 2e0a 2a20 4164  e sequence..* Ad
+00008c90: 6465 6420 6060 736b 6269 6f2e 7574 696c  ded ``skbio.util
+00008ca0: 2e61 7373 6572 745f 6f72 6469 6e61 7469  .assert_ordinati
+00008cb0: 6f6e 5f72 6573 756c 7473 5f65 7175 616c  on_results_equal
+00008cc0: 6060 2066 756e 6374 696f 6e20 666f 7220  `` function for 
+00008cd0: 636f 6d70 6172 696e 6720 6060 4f72 6469  comparing ``Ordi
+00008ce0: 6e61 7469 6f6e 5265 7375 6c74 7360 6020  nationResults`` 
+00008cf0: 6f62 6a65 6374 7320 696e 2075 6e69 7420  objects in unit 
+00008d00: 7465 7374 732e 0a2a 2041 6464 6564 2060  tests..* Added `
+00008d10: 6073 6b62 696f 2e69 6f2e 666f 726d 6174  `skbio.io.format
+00008d20: 2e67 656e 6261 6e6b 6060 2066 6f72 2072  .genbank`` for r
+00008d30: 6561 6469 6e67 2061 6e64 2077 7269 7469  eading and writi
+00008d40: 6e67 2047 656e 4261 6e6b 2f47 656e 5065  ng GenBank/GenPe
+00008d50: 7074 2066 6f72 2060 6044 4e41 6060 2c20  pt for ``DNA``, 
+00008d60: 6060 524e 4160 602c 2060 6050 726f 7465  ``RNA``, ``Prote
+00008d70: 696e 6060 2061 6e64 2060 6053 6571 7565  in`` and ``Seque
+00008d80: 6e63 6560 6020 636c 6173 7365 732e 0a2a  nce`` classes..*
+00008d90: 2041 6464 6564 2060 6073 6b62 696f 2e75   Added ``skbio.u
+00008da0: 7469 6c2e 5265 7072 6573 656e 7461 7469  til.Representati
+00008db0: 6f6e 5761 726e 696e 6760 6020 666f 7220  onWarning`` for 
+00008dc0: 7761 726e 696e 6720 6162 6f75 7420 7375  warning about su
+00008dd0: 6273 7469 7475 7469 6f6e 732c 2061 7373  bstitutions, ass
+00008de0: 756d 7074 696f 6e73 2c20 6f72 2070 6172  umptions, or par
+00008df0: 7469 6375 6c61 7220 616c 7465 7261 7469  ticular alterati
+00008e00: 6f6e 7320 7468 6174 2077 6572 6520 6d61  ons that were ma
+00008e10: 6465 2066 6f72 2074 6865 2073 7563 6365  de for the succe
+00008e20: 7373 6675 6c20 636f 6d70 6c65 7469 6f6e  ssful completion
+00008e30: 206f 6620 6120 7072 6f63 6573 732e 0a2a   of a process..*
+00008e40: 2060 6054 7265 654e 6f64 652e 7469 705f   ``TreeNode.tip_
+00008e50: 7469 705f 6469 7374 616e 6365 7360 6020  tip_distances`` 
+00008e60: 6e6f 7720 7375 7070 6f72 7473 206e 6f64  now supports nod
+00008e70: 6573 2077 6974 686f 7574 2061 6e20 6173  es without an as
+00008e80: 736f 6369 6174 6564 206c 656e 6774 682e  sociated length.
+00008e90: 2049 6e20 7468 6973 2063 6173 652c 2061   In this case, a
+00008ea0: 206c 656e 6774 6820 6f66 2030 2e30 2069   length of 0.0 i
+00008eb0: 7320 6173 7375 6d65 6420 616e 6420 616e  s assumed and an
+00008ec0: 2060 6073 6b62 696f 2e75 7469 6c2e 5265   ``skbio.util.Re
+00008ed0: 7072 6573 656e 7461 7469 6f6e 5761 726e  presentationWarn
+00008ee0: 696e 6760 6020 6973 2072 6169 7365 642e  ing`` is raised.
+00008ef0: 2050 7265 7669 6f75 7320 6265 6861 7669   Previous behavi
+00008f00: 6f72 2077 6173 2074 6f20 7261 6973 6520  or was to raise 
+00008f10: 6120 6060 4e6f 4c65 6e67 7468 4572 726f  a ``NoLengthErro
+00008f20: 7260 602e 2028 5b23 3739 315d 2868 7474  r``. ([#791](htt
+00008f30: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+00008f40: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
+00008f50: 696f 2f69 7373 7565 732f 3739 3129 290a  io/issues/791)).
+00008f60: 2a20 6060 4469 7374 616e 6365 4d61 7472  * ``DistanceMatr
+00008f70: 6978 6060 206e 6f77 2068 6173 2061 206e  ix`` now has a n
+00008f80: 6577 2063 6f6e 7374 7275 6374 6f72 206d  ew constructor m
+00008f90: 6574 686f 6420 6361 6c6c 6564 2060 6672  ethod called `fr
+00008fa0: 6f6d 5f69 7465 7261 626c 6560 2e0a 2a20  om_iterable`..* 
+00008fb0: 6060 5365 7175 656e 6365 6060 206e 6f77  ``Sequence`` now
+00008fc0: 2061 6363 6570 7473 2060 606c 6f77 6572   accepts ``lower
+00008fd0: 6361 7365 6060 206b 6579 776f 7264 206c  case`` keyword l
+00008fe0: 696b 6520 6060 444e 4160 6020 616e 6420  ike ``DNA`` and 
+00008ff0: 6f74 6865 7273 2e20 5570 6461 7465 6420  others. Updated 
+00009000: 6060 6661 7374 6160 602c 2060 6066 6173  ``fasta``, ``fas
+00009010: 7471 6060 2c20 616e 6420 6060 7173 6571  tq``, and ``qseq
+00009020: 6060 2072 6561 6465 7273 2f77 7269 7465  `` readers/write
+00009030: 7273 2066 6f72 2060 6053 6571 7565 6e63  rs for ``Sequenc
+00009040: 6560 6020 746f 2072 6566 6c65 6374 2074  e`` to reflect t
+00009050: 6869 732e 0a2a 2054 6865 2060 606c 6f77  his..* The ``low
+00009060: 6572 6361 7365 6060 206d 6574 686f 6420  ercase`` method 
+00009070: 6861 7320 6265 656e 206d 6f76 6564 2075  has been moved u
+00009080: 7020 746f 2060 6053 6571 7565 6e63 6560  p to ``Sequence`
+00009090: 6020 6d65 616e 696e 6720 616c 6c20 7365  ` meaning all se
+000090a0: 7175 656e 6365 206f 626a 6563 7473 206e  quence objects n
+000090b0: 6f77 2068 6176 6520 6120 6060 6c6f 7765  ow have a ``lowe
+000090c0: 7263 6173 6560 6020 6d65 7468 6f64 2e0a  rcase`` method..
+000090d0: 2a20 4164 6465 6420 6060 7265 7665 7273  * Added ``revers
+000090e0: 655f 7472 616e 7363 7269 6265 6060 2063  e_transcribe`` c
+000090f0: 6c61 7373 206d 6574 686f 6420 746f 2060  lass method to `
+00009100: 6052 4e41 6060 2e0a 2a20 4164 6465 6420  `RNA``..* Added 
+00009110: 6053 6571 7565 6e63 652e 6f62 7365 7276  `Sequence.observ
+00009120: 6564 5f63 6861 7273 6020 7072 6f70 6572  ed_chars` proper
+00009130: 7479 2066 6f72 206f 6274 6169 6e69 6e67  ty for obtaining
+00009140: 2074 6865 2073 6574 206f 6620 6f62 7365   the set of obse
+00009150: 7276 6564 2063 6861 7261 6374 6572 7320  rved characters 
+00009160: 696e 2061 2073 6571 7565 6e63 652e 2028  in a sequence. (
+00009170: 5b23 3130 3735 5d28 6874 7470 733a 2f2f  [#1075](https://
+00009180: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
+00009190: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
+000091a0: 7375 6573 2f31 3037 3529 290a 2a20 4164  sues/1075)).* Ad
+000091b0: 6465 6420 6053 6571 7565 6e63 652e 6672  ded `Sequence.fr
+000091c0: 6571 7565 6e63 6965 7360 206d 6574 686f  equencies` metho
+000091d0: 6420 666f 7220 636f 6d70 7574 696e 6720  d for computing 
+000091e0: 6368 6172 6163 7465 7220 6672 6571 7565  character freque
+000091f0: 6e63 6965 7320 696e 2061 2073 6571 7565  ncies in a seque
+00009200: 6e63 652e 2028 5b23 3130 3734 5d28 6874  nce. ([#1074](ht
+00009210: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+00009220: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
+00009230: 6269 6f2f 6973 7375 6573 2f31 3037 3429  bio/issues/1074)
+00009240: 290a 2a20 4164 6465 6420 6578 7065 7269  ).* Added experi
+00009250: 6d65 6e74 616c 2063 6c61 7373 2d6d 6574  mental class-met
+00009260: 686f 6420 6060 5365 7175 656e 6365 2e63  hod ``Sequence.c
+00009270: 6f6e 6361 7460 6020 7768 6963 6820 7769  oncat`` which wi
+00009280: 6c6c 2070 726f 6475 6365 2061 206e 6577  ll produce a new
+00009290: 2073 6571 7565 6e63 6520 6672 6f6d 2061   sequence from a
+000092a0: 6e20 6974 6572 6162 6c65 206f 6620 6578  n iterable of ex
+000092b0: 6973 7469 6e67 2073 6571 7565 6e63 6573  isting sequences
+000092c0: 2e20 5061 7261 6d65 7465 7273 2063 6f6e  . Parameters con
+000092d0: 7472 6f6c 2068 6f77 2070 6f73 6974 696f  trol how positio
+000092e0: 6e61 6c20 6d65 7461 6461 7461 2069 7320  nal metadata is 
+000092f0: 7072 6f70 6167 6174 6564 2064 7572 696e  propagated durin
+00009300: 6720 6120 636f 6e63 6174 656e 6174 696f  g a concatenatio
+00009310: 6e2e 0a2a 2060 6054 7265 654e 6f64 652e  n..* ``TreeNode.
+00009320: 746f 5f61 7272 6179 6060 206e 6f77 2073  to_array`` now s
+00009330: 7570 706f 7274 7320 7265 706c 6163 696e  upports replacin
+00009340: 6720 6060 6e61 6e60 6020 6272 616e 6368  g ``nan`` branch
+00009350: 206c 656e 6774 6873 2069 6e20 7468 6520   lengths in the 
+00009360: 7265 7375 6c74 696e 6720 6272 616e 6368  resulting branch
+00009370: 206c 656e 6774 6820 7665 6374 6f72 2077   length vector w
+00009380: 6974 6820 7468 6520 7661 6c75 6520 7072  ith the value pr
+00009390: 6f76 6964 6564 2061 7320 6060 6e61 6e5f  ovided as ``nan_
+000093a0: 6c65 6e67 7468 5f76 616c 7565 6060 2e0a  length_value``..
+000093b0: 2a20 6060 736b 6269 6f2e 696f 2e66 6f72  * ``skbio.io.for
+000093c0: 6d61 742e 7068 796c 6970 6060 206e 6f77  mat.phylip`` now
+000093d0: 2073 7570 706f 7274 7320 736e 6966 6669   supports sniffi
+000093e0: 6e67 2061 6e64 2072 6561 6469 6e67 2073  ng and reading s
+000093f0: 7472 6963 742c 2073 6571 7565 6e74 6961  trict, sequentia
+00009400: 6c20 5048 594c 4950 2d66 6f72 6d61 7474  l PHYLIP-formatt
+00009410: 6564 2066 696c 6573 2069 6e74 6f20 6060  ed files into ``
+00009420: 736b 6269 6f2e 416c 6967 6e6d 656e 7460  skbio.Alignment`
+00009430: 6020 6f62 6a65 6374 732e 2028 5b23 3130  ` objects. ([#10
+00009440: 3036 5d28 6874 7470 733a 2f2f 6769 7468  06](https://gith
+00009450: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+00009460: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
+00009470: 2f31 3030 3629 290a 2a20 4164 6465 6420  /1006)).* Added 
+00009480: 6064 6566 6175 6c74 5f67 6170 5f63 6861  `default_gap_cha
+00009490: 7260 2063 6c61 7373 2070 726f 7065 7274  r` class propert
+000094a0: 7920 746f 2060 6044 4e41 6060 2c20 6060  y to ``DNA``, ``
+000094b0: 524e 4160 602c 2061 6e64 2060 6050 726f  RNA``, and ``Pro
+000094c0: 7465 696e 6060 2066 6f72 2072 6570 7265  tein`` for repre
+000094d0: 7365 6e74 696e 6720 6761 7020 6368 6172  senting gap char
+000094e0: 6163 7465 7273 2069 6e20 6120 6e65 7720  acters in a new 
+000094f0: 7365 7175 656e 6365 2e0a 0a23 2323 2042  sequence...### B
+00009500: 6163 6b77 6172 642d 696e 636f 6d70 6174  ackward-incompat
+00009510: 6962 6c65 2063 6861 6e67 6573 205b 7374  ible changes [st
+00009520: 6162 6c65 5d0a 2a20 6053 6571 7565 6e63  able].* `Sequenc
+00009530: 652e 6b6d 6572 5f66 7265 7175 656e 6369  e.kmer_frequenci
+00009540: 6573 6020 6e6f 7720 7265 7475 726e 7320  es` now returns 
+00009550: 6120 6064 6963 7460 2e20 5072 6576 696f  a `dict`. Previo
+00009560: 7573 2062 6568 6176 696f 7220 7761 7320  us behavior was 
+00009570: 746f 2072 6574 7572 6e20 6120 6063 6f6c  to return a `col
+00009580: 6c65 6374 696f 6e73 2e43 6f75 6e74 6572  lections.Counter
+00009590: 6020 6966 2060 7265 6c61 7469 7665 3d46  ` if `relative=F
+000095a0: 616c 7365 6020 7761 7320 7061 7373 6564  alse` was passed
+000095b0: 2c20 616e 6420 6120 6063 6f6c 6c65 6374  , and a `collect
+000095c0: 696f 6e73 2e64 6566 6175 6c74 6469 6374  ions.defaultdict
+000095d0: 6020 6966 2060 7265 6c61 7469 7665 3d54  ` if `relative=T
+000095e0: 7275 6560 2077 6173 2070 6173 7365 642e  rue` was passed.
+000095f0: 2049 6e20 7468 6520 6361 7365 206f 6620   In the case of 
+00009600: 6120 6d69 7373 696e 6720 6b65 792c 2074  a missing key, t
+00009610: 6865 2060 436f 756e 7465 7260 2077 6f75  he `Counter` wou
+00009620: 6c64 2072 6574 7572 6e20 3020 616e 6420  ld return 0 and 
+00009630: 7468 6520 6064 6566 6175 6c74 6469 6374  the `defaultdict
+00009640: 6020 776f 756c 6420 7265 7475 726e 2030  ` would return 0
+00009650: 2e30 2e20 4265 6361 7573 6520 7468 6520  .0. Because the 
+00009660: 7265 7475 726e 2074 7970 6520 6973 206e  return type is n
+00009670: 6f77 2061 6c77 6179 7320 6120 6064 6963  ow always a `dic
+00009680: 7460 2c20 6174 7465 6d70 7469 6e67 2074  t`, attempting t
+00009690: 6f20 6163 6365 7373 2061 206d 6973 7369  o access a missi
+000096a0: 6e67 206b 6579 2077 696c 6c20 7261 6973  ng key will rais
+000096b0: 6520 6120 604b 6579 4572 726f 7260 2e20  e a `KeyError`. 
+000096c0: 5468 6973 2063 6861 6e67 6520 2a6d 6179  This change *may
+000096d0: 2a20 6272 6561 6b20 6261 636b 7761 7264  * break backward
+000096e0: 732d 636f 6d70 6174 6962 696c 6974 7920  s-compatibility 
+000096f0: 6465 7065 6e64 696e 6720 6f6e 2068 6f77  depending on how
+00009700: 2074 6865 2060 436f 756e 7465 7260 2f60   the `Counter`/`
+00009710: 6465 6661 756c 7464 6963 7460 2069 7320  defaultdict` is 
+00009720: 6265 696e 6720 7573 6564 2e20 5765 2068  being used. We h
+00009730: 6f70 6520 7468 6174 2069 6e20 6d6f 7374  ope that in most
+00009740: 2063 6173 6573 2074 6869 7320 6368 616e   cases this chan
+00009750: 6765 2077 696c 6c20 6e6f 7420 6272 6561  ge will not brea
+00009760: 6b20 6261 636b 7761 7264 732d 636f 6d70  k backwards-comp
+00009770: 6174 6962 696c 6974 7920 6265 6361 7573  atibility becaus
+00009780: 6520 626f 7468 2060 436f 756e 7465 7260  e both `Counter`
+00009790: 2061 6e64 2060 6465 6661 756c 7464 6963   and `defaultdic
+000097a0: 7460 2061 7265 2060 6469 6374 6020 7375  t` are `dict` su
+000097b0: 6263 6c61 7373 6573 2e0a 0a20 2020 4966  bclasses...   If
+000097c0: 2074 6865 2070 7265 7669 6f75 7320 6265   the previous be
+000097d0: 6861 7669 6f72 2069 7320 6465 7369 7265  havior is desire
+000097e0: 642c 2063 6f6e 7665 7274 2074 6865 2060  d, convert the `
+000097f0: 6469 6374 6020 696e 746f 2061 2060 436f  dict` into a `Co
+00009800: 756e 7465 7260 2f60 6465 6661 756c 7464  unter`/`defaultd
+00009810: 6963 7460 3a0a 0a20 2020 2060 6060 7079  ict`:..    ```py
+00009820: 7468 6f6e 0a20 2020 2069 6d70 6f72 7420  thon.    import 
+00009830: 636f 6c6c 6563 7469 6f6e 730a 2020 2020  collections.    
+00009840: 6672 6f6d 2073 6b62 696f 2069 6d70 6f72  from skbio impor
+00009850: 7420 5365 7175 656e 6365 0a20 2020 2073  t Sequence.    s
+00009860: 6571 203d 2053 6571 7565 6e63 6528 2741  eq = Sequence('A
+00009870: 4343 4741 4754 5454 4141 4343 4741 4154  CCGAGTTTAACCGAAT
+00009880: 4127 290a 0a20 2020 2023 2043 6f75 6e74  A')..    # Count
+00009890: 6572 0a20 2020 2066 7265 7173 5f64 6963  er.    freqs_dic
+000098a0: 7420 3d20 7365 712e 6b6d 6572 5f66 7265  t = seq.kmer_fre
+000098b0: 7175 656e 6369 6573 286b 3d38 290a 2020  quencies(k=8).  
+000098c0: 2020 6672 6571 735f 636f 756e 7465 7220    freqs_counter 
+000098d0: 3d20 636f 6c6c 6563 7469 6f6e 732e 436f  = collections.Co
+000098e0: 756e 7465 7228 6672 6571 735f 6469 6374  unter(freqs_dict
+000098f0: 290a 0a20 2020 2023 2064 6566 6175 6c74  )..    # default
+00009900: 6469 6374 0a20 2020 2066 7265 7173 5f64  dict.    freqs_d
+00009910: 6963 7420 3d20 7365 712e 6b6d 6572 5f66  ict = seq.kmer_f
+00009920: 7265 7175 656e 6369 6573 286b 3d38 2c20  requencies(k=8, 
+00009930: 7265 6c61 7469 7665 3d54 7275 6529 0a20  relative=True). 
+00009940: 2020 2066 7265 7173 5f64 6566 6175 6c74     freqs_default
+00009950: 5f64 6963 7420 3d20 636f 6c6c 6563 7469  _dict = collecti
+00009960: 6f6e 732e 6465 6661 756c 7464 6963 7428  ons.defaultdict(
+00009970: 666c 6f61 742c 2066 7265 7173 5f64 6963  float, freqs_dic
+00009980: 7429 0a20 2020 2060 6060 0a0a 2020 202a  t).    ```..   *
+00009990: 2a52 6174 696f 6e61 6c65 3a2a 2a20 5765  *Rationale:** We
+000099a0: 2062 656c 6965 7665 2069 7420 6973 2073   believe it is s
+000099b0: 6166 6572 2074 6f20 7265 7475 726e 2060  afer to return `
+000099c0: 6469 6374 6020 696e 7374 6561 6420 6f66  dict` instead of
+000099d0: 2060 436f 756e 7465 7260 2f60 6465 6661   `Counter`/`defa
+000099e0: 756c 7464 6963 7460 2061 7320 7468 6973  ultdict` as this
+000099f0: 206d 6179 2070 7265 7665 6e74 2065 7272   may prevent err
+00009a00: 6f72 2d70 726f 6e65 2075 7361 6765 206f  or-prone usage o
+00009a10: 6620 7468 6520 7265 7475 726e 2076 616c  f the return val
+00009a20: 7565 2e20 5072 6576 696f 7573 2062 6568  ue. Previous beh
+00009a30: 6176 696f 7220 616c 6c6f 7765 6420 6163  avior allowed ac
+00009a40: 6365 7373 696e 6720 6d69 7373 696e 6720  cessing missing 
+00009a50: 6b6d 6572 732c 2072 6574 7572 6e69 6e67  kmers, returning
+00009a60: 2030 206f 7220 302e 3020 6465 7065 6e64   0 or 0.0 depend
+00009a70: 696e 6720 6f6e 2074 6865 2060 7265 6c61  ing on the `rela
+00009a80: 7469 7665 6020 7061 7261 6d65 7465 722e  tive` parameter.
+00009a90: 2054 6869 7320 6973 2063 6f6e 7665 6e69   This is conveni
+00009aa0: 656e 7420 696e 206d 616e 7920 6361 7365  ent in many case
+00009ab0: 7320 6275 7420 616c 736f 2070 6f74 656e  s but also poten
+00009ac0: 7469 616c 6c79 206d 6973 6c65 6164 696e  tially misleadin
+00009ad0: 672e 2046 6f72 2065 7861 6d70 6c65 2c20  g. For example, 
+00009ae0: 636f 6e73 6964 6572 2074 6865 2066 6f6c  consider the fol
+00009af0: 6c6f 7769 6e67 2063 6f64 653a 0a0a 2020  lowing code:..  
+00009b00: 2020 6060 6070 7974 686f 6e0a 2020 2020    ```python.    
+00009b10: 6672 6f6d 2073 6b62 696f 2069 6d70 6f72  from skbio impor
+00009b20: 7420 5365 7175 656e 6365 0a20 2020 2073  t Sequence.    s
+00009b30: 6571 203d 2053 6571 7565 6e63 6528 2741  eq = Sequence('A
+00009b40: 4343 4741 4754 5454 4141 4343 4741 4154  CCGAGTTTAACCGAAT
+00009b50: 4127 290a 2020 2020 6672 6571 7320 3d20  A').    freqs = 
+00009b60: 7365 712e 6b6d 6572 5f66 7265 7175 656e  seq.kmer_frequen
+00009b70: 6369 6573 286b 3d38 290a 2020 2020 6672  cies(k=8).    fr
+00009b80: 6571 735b 2741 4343 4741 275d 0a20 2020  eqs['ACCGA'].   
+00009b90: 2060 6060 0a0a 2020 2020 5072 6576 696f   ```..    Previo
+00009ba0: 7573 2062 6568 6176 696f 7220 776f 756c  us behavior woul
+00009bb0: 6420 7265 7475 726e 2030 2062 6563 6175  d return 0 becau
+00009bc0: 7365 2074 6865 206b 6d65 7220 6027 4143  se the kmer `'AC
+00009bd0: 4347 4127 6020 6973 206e 6f74 2070 7265  CGA'` is not pre
+00009be0: 7365 6e74 2069 6e20 7468 6520 6043 6f75  sent in the `Cou
+00009bf0: 6e74 6572 602e 2049 6e20 6f6e 6520 7265  nter`. In one re
+00009c00: 7370 6563 7420 7468 6973 2069 7320 7468  spect this is th
+00009c10: 6520 636f 7272 6563 7420 616e 7377 6572  e correct answer
+00009c20: 2062 6563 6175 7365 2077 6520 6173 6b65   because we aske
+00009c30: 6420 666f 7220 6b6d 6572 7320 6f66 206c  d for kmers of l
+00009c40: 656e 6774 6820 383b 2060 2741 4343 4741  ength 8; `'ACCGA
+00009c50: 2760 2069 7320 6120 6469 6666 6572 656e  '` is a differen
+00009c60: 7420 6c65 6e67 7468 2073 6f20 6974 2069  t length so it i
+00009c70: 7320 6e6f 7420 696e 636c 7564 6564 2069  s not included i
+00009c80: 6e20 7468 6520 7265 7375 6c74 732e 2048  n the results. H
+00009c90: 6f77 6576 6572 2c20 7765 2062 656c 6965  owever, we belie
+00009ca0: 7665 2069 7420 6973 2073 6166 6572 2074  ve it is safer t
+00009cb0: 6f20 6176 6f69 6420 7468 6973 2069 6d70  o avoid this imp
+00009cc0: 6c69 6369 7420 6265 6861 7669 6f72 2069  licit behavior i
+00009cd0: 6e20 6361 7365 2074 6865 2075 7365 7220  n case the user 
+00009ce0: 6173 7375 6d65 7320 7468 6572 6520 6172  assumes there ar
+00009cf0: 6520 6e6f 2060 2741 4343 4741 2760 206b  e no `'ACCGA'` k
+00009d00: 6d65 7273 2069 6e20 7468 6520 7365 7175  mers in the sequ
+00009d10: 656e 6365 2028 7768 6963 6820 7468 6572  ence (which ther
+00009d20: 6520 6172 6521 292e 2041 2060 4b65 7945  e are!). A `KeyE
+00009d30: 7272 6f72 6020 696e 2074 6869 7320 6361  rror` in this ca
+00009d40: 7365 2069 7320 6d6f 7265 2065 7870 6c69  se is more expli
+00009d50: 6369 7420 616e 6420 666f 7263 6573 2074  cit and forces t
+00009d60: 6865 2075 7365 7220 746f 2063 6f6e 7369  he user to consi
+00009d70: 6465 7220 7468 6569 7220 7175 6572 792e  der their query.
+00009d80: 2052 6574 7572 6e69 6e67 2061 2060 6469   Returning a `di
+00009d90: 6374 6020 7769 6c6c 2061 6c73 6f20 6265  ct` will also be
+00009da0: 2063 6f6e 7369 7374 656e 7420 7769 7468   consistent with
+00009db0: 2060 5365 7175 656e 6365 2e66 7265 7175   `Sequence.frequ
+00009dc0: 656e 6369 6573 602e 0a0a 2323 2320 4261  encies`...### Ba
+00009dd0: 636b 7761 7264 2d69 6e63 6f6d 7061 7469  ckward-incompati
+00009de0: 626c 6520 6368 616e 6765 7320 5b65 7870  ble changes [exp
+00009df0: 6572 696d 656e 7461 6c5d 0a2a 2052 6570  erimental].* Rep
+00009e00: 6c61 6365 6420 6060 5043 6f41 6060 2c20  laced ``PCoA``, 
+00009e10: 6060 4343 4160 602c 2060 6043 4160 6020  ``CCA``, ``CA`` 
+00009e20: 616e 6420 6060 5244 4160 6020 696e 2060  and ``RDA`` in `
+00009e30: 6073 6b62 696f 2e73 7461 7473 2e6f 7264  `skbio.stats.ord
+00009e40: 696e 6174 696f 6e60 6020 7769 7468 2065  ination`` with e
+00009e50: 7175 6976 616c 656e 7420 6675 6e63 7469  quivalent functi
+00009e60: 6f6e 7320 6060 7063 6f61 6060 2c20 6060  ons ``pcoa``, ``
+00009e70: 6363 6160 602c 2060 6063 6160 6020 616e  cca``, ``ca`` an
+00009e80: 6420 6060 7264 6160 602e 2054 6865 7365  d ``rda``. These
+00009e90: 2066 756e 6374 696f 6e73 206e 6f77 2074   functions now t
+00009ea0: 616b 6520 6060 7064 2e44 6174 6146 7261  ake ``pd.DataFra
+00009eb0: 6d65 6060 206f 626a 6563 7473 2e0a 2a20  me`` objects..* 
+00009ec0: 4368 616e 6765 2060 604f 7264 696e 6174  Change ``Ordinat
+00009ed0: 696f 6e52 6573 756c 7473 6060 2074 6f20  ionResults`` to 
+00009ee0: 6861 7665 2069 7473 2061 7474 7269 6275  have its attribu
+00009ef0: 7465 7320 6261 7365 6420 6f6e 2060 6070  tes based on ``p
+00009f00: 642e 4461 7461 4672 616d 6560 6020 616e  d.DataFrame`` an
+00009f10: 6420 6060 7064 2e53 6572 6965 7360 6020  d ``pd.Series`` 
+00009f20: 6f62 6a65 6374 732c 2069 6e73 7465 6164  objects, instead
+00009f30: 206f 6620 7061 6972 7320 6f66 2069 6465   of pairs of ide
+00009f40: 6e74 6966 6965 7273 2061 6e64 2076 616c  ntifiers and val
+00009f50: 7565 732e 2054 6865 2063 6861 6e67 6573  ues. The changes
+00009f60: 2061 7265 2061 7320 666f 6c6c 6f77 733a   are as follows:
+00009f70: 0a20 2020 202d 2060 6073 7065 6369 6573  .    - ``species
+00009f80: 6060 2061 6e64 2060 6073 7065 6369 6573  `` and ``species
+00009f90: 5f69 6473 6060 2068 6176 6520 6265 656e  _ids`` have been
+00009fa0: 2072 6570 6c61 6365 6420 6279 2061 2060   replaced by a `
+00009fb0: 6070 642e 4461 7461 4672 616d 6560 6020  `pd.DataFrame`` 
+00009fc0: 6e61 6d65 6420 6060 6665 6174 7572 6573  named ``features
+00009fd0: 6060 2e0a 2020 2020 2d20 6060 7369 7465  ``..    - ``site
+00009fe0: 6060 2061 6e64 2060 6073 6974 655f 6964  `` and ``site_id
+00009ff0: 7360 6020 6861 7665 2062 6565 6e20 7265  s`` have been re
+0000a000: 706c 6163 6564 2062 7920 6120 6060 7064  placed by a ``pd
+0000a010: 2e44 6174 6146 7261 6d65 6060 206e 616d  .DataFrame`` nam
+0000a020: 6564 2060 6073 616d 706c 6573 6060 2e0a  ed ``samples``..
+0000a030: 2020 2020 2d20 6060 6569 6776 616c 7360      - ``eigvals`
+0000a040: 6020 6973 206e 6f77 2061 2060 6070 642e  ` is now a ``pd.
+0000a050: 5365 7269 6573 6060 206f 626a 6563 742e  Series`` object.
+0000a060: 0a20 2020 202d 2060 6070 726f 706f 7274  .    - ``proport
+0000a070: 696f 6e5f 6578 706c 6169 6e65 6460 6020  ion_explained`` 
+0000a080: 6973 206e 6f77 2061 2060 6070 642e 5365  is now a ``pd.Se
+0000a090: 7269 6573 6060 206f 626a 6563 742e 0a20  ries`` object.. 
+0000a0a0: 2020 202d 2060 6062 6970 6c6f 7460 6020     - ``biplot`` 
+0000a0b0: 6973 206e 6f77 2061 2060 6070 642e 4461  is now a ``pd.Da
+0000a0c0: 7461 4672 616d 6560 6020 6f62 6a65 6374  taFrame`` object
+0000a0d0: 206e 616d 6564 2060 6062 6970 6c6f 745f   named ``biplot_
+0000a0e0: 7363 6f72 6573 6060 2e0a 2020 2020 2d20  scores``..    - 
+0000a0f0: 6060 7369 7465 5f63 6f6e 7374 7261 696e  ``site_constrain
+0000a100: 7473 6060 2069 7320 6e6f 7720 6120 6060  ts`` is now a ``
+0000a110: 7064 2e44 6174 6146 7261 6d65 6060 206f  pd.DataFrame`` o
+0000a120: 626a 6563 7420 6e61 6d65 6420 6060 7361  bject named ``sa
+0000a130: 6d70 6c65 5f63 6f6e 7374 7261 696e 7473  mple_constraints
+0000a140: 6060 2e0a 2a20 6060 7368 6f72 745f 6d65  ``..* ``short_me
+0000a150: 7468 6f64 5f6e 616d 6560 6020 616e 6420  thod_name`` and 
+0000a160: 6060 6c6f 6e67 5f6d 6574 686f 645f 6e61  ``long_method_na
+0000a170: 6d65 6060 2061 7265 206e 6f77 2072 6571  me`` are now req
+0000a180: 7569 7265 6420 6172 6775 6d65 6e74 7320  uired arguments 
+0000a190: 6f66 2074 6865 2060 604f 7264 696e 6174  of the ``Ordinat
+0000a1a0: 696f 6e52 6573 756c 7473 6060 206f 626a  ionResults`` obj
+0000a1b0: 6563 742e 0a2a 2052 656d 6f76 6564 2060  ect..* Removed `
+0000a1c0: 736b 6269 6f2e 6469 7665 7273 6974 792e  skbio.diversity.
+0000a1d0: 616c 7068 612e 6571 7569 7461 6269 6c69  alpha.equitabili
+0000a1e0: 7479 602e 2050 6c65 6173 6520 7573 6520  ty`. Please use 
+0000a1f0: 6073 6b62 696f 2e64 6976 6572 7369 7479  `skbio.diversity
+0000a200: 2e61 6c70 6861 2e70 6965 6c6f 755f 6560  .alpha.pielou_e`
+0000a210: 2c20 7768 6963 6820 6973 206d 6f72 6520  , which is more 
+0000a220: 6163 6375 7261 7465 6c79 206e 616d 6564  accurately named
+0000a230: 2061 6e64 2062 6574 7465 7220 646f 6375   and better docu
+0000a240: 6d65 6e74 6564 2e20 4e6f 7465 2074 6861  mented. Note tha
+0000a250: 7420 6065 7175 6974 6162 696c 6974 7960  t `equitability`
+0000a260: 2062 7920 6465 6661 756c 7420 7573 6564   by default used
+0000a270: 206c 6f67 6172 6974 686d 2062 6173 6520   logarithm base 
+0000a280: 3220 7768 696c 6520 6070 6965 6c6f 755f  2 while `pielou_
+0000a290: 6560 2075 7365 7320 6c6f 6761 7269 7468  e` uses logarith
+0000a2a0: 6d20 6261 7365 2060 6560 2061 7320 6465  m base `e` as de
+0000a2b0: 7363 7269 6265 6420 696e 2048 6569 7020  scribed in Heip 
+0000a2c0: 3139 3734 2e0a 2a20 6060 736b 6269 6f2e  1974..* ``skbio.
+0000a2d0: 6469 7665 7273 6974 792e 6265 7461 2e70  diversity.beta.p
+0000a2e0: 775f 6469 7374 616e 6365 7360 6020 6973  w_distances`` is
+0000a2f0: 206e 6f77 2063 616c 6c65 6420 6060 736b   now called ``sk
+0000a300: 6269 6f2e 6469 7665 7273 6974 792e 6265  bio.diversity.be
+0000a310: 7461 5f64 6976 6572 7369 7479 6060 2e20  ta_diversity``. 
+0000a320: 5468 6973 2066 756e 6374 696f 6e20 6e6f  This function no
+0000a330: 206c 6f6e 6765 7220 6465 6669 6e65 7320   longer defines 
+0000a340: 6120 6465 6661 756c 7420 6d65 7472 6963  a default metric
+0000a350: 2c20 616e 6420 6060 6d65 7472 6963 6060  , and ``metric``
+0000a360: 2069 7320 6e6f 7720 7468 6520 6669 7273   is now the firs
+0000a370: 7420 6172 6775 6d65 6e74 2074 6f20 7468  t argument to th
+0000a380: 6973 2066 756e 6374 696f 6e2e 2054 6869  is function. Thi
+0000a390: 7320 6675 6e63 7469 6f6e 2063 616e 2061  s function can a
+0000a3a0: 6c73 6f20 6e6f 7720 7461 6b65 2061 2070  lso now take a p
+0000a3b0: 6169 7277 6973 6520 6469 7374 616e 6365  airwise distance
+0000a3c0: 7320 6675 6e63 7469 6f6e 2061 7320 6060  s function as ``
+0000a3d0: 7061 6972 7769 7365 5f66 756e 6360 602e  pairwise_func``.
+0000a3e0: 0a2a 2044 6570 7265 6361 7465 6420 6675  .* Deprecated fu
+0000a3f0: 6e63 7469 6f6e 2060 6073 6b62 696f 2e64  nction ``skbio.d
+0000a400: 6976 6572 7369 7479 2e62 6574 612e 7077  iversity.beta.pw
+0000a410: 5f64 6973 7461 6e63 6573 5f66 726f 6d5f  _distances_from_
+0000a420: 7461 626c 6560 6020 6861 7320 6265 656e  table`` has been
+0000a430: 2072 656d 6f76 6564 2066 726f 6d20 7363   removed from sc
+0000a440: 696b 6974 2d62 696f 2061 7320 7363 6865  ikit-bio as sche
+0000a450: 6475 6c65 642e 2043 6f64 6520 7468 6174  duled. Code that
+0000a460: 2075 7365 6420 7468 6973 2073 686f 756c   used this shoul
+0000a470: 6420 6265 2061 6461 7074 6564 2074 6f20  d be adapted to 
+0000a480: 7573 6520 6060 736b 6269 6f2e 6469 7665  use ``skbio.dive
+0000a490: 7273 6974 792e 6265 7461 5f64 6976 6572  rsity.beta_diver
+0000a4a0: 7369 7479 6060 2e0a 2a20 6060 5472 6565  sity``..* ``Tree
+0000a4b0: 4e6f 6465 2e69 6e64 6578 5f74 7265 6560  Node.index_tree`
+0000a4c0: 6020 6e6f 7720 7265 7475 726e 7320 6120  ` now returns a 
+0000a4d0: 322d 4420 6e75 6d70 7920 6172 7261 7920  2-D numpy array 
+0000a4e0: 6173 2069 7473 2073 6563 6f6e 6420 7265  as its second re
+0000a4f0: 7475 726e 2076 616c 7565 2028 7468 6520  turn value (the 
+0000a500: 6368 696c 6420 6e6f 6465 2069 6e64 6578  child node index
+0000a510: 2920 696e 7374 6561 6420 6f66 2061 2031  ) instead of a 1
+0000a520: 2d44 206e 756d 7079 2061 7272 6179 2e0a  -D numpy array..
+0000a530: 2a20 4465 7072 6563 6174 6564 2066 756e  * Deprecated fun
+0000a540: 6374 696f 6e73 2060 736b 6269 6f2e 6472  ctions `skbio.dr
+0000a550: 6177 2e62 6f78 706c 6f74 7360 2061 6e64  aw.boxplots` and
+0000a560: 2060 736b 6269 6f2e 6472 6177 2e67 726f   `skbio.draw.gro
+0000a570: 7570 6564 5f64 6973 7472 6962 7574 696f  uped_distributio
+0000a580: 6e73 6020 6861 7665 2062 6565 6e20 7265  ns` have been re
+0000a590: 6d6f 7665 6420 6672 6f6d 2073 6369 6b69  moved from sciki
+0000a5a0: 742d 6269 6f20 6173 2073 6368 6564 756c  t-bio as schedul
+0000a5b0: 6564 2e20 5468 6573 6520 6675 6e63 7469  ed. These functi
+0000a5c0: 6f6e 7320 6765 6e65 7261 7465 6420 706c  ons generated pl
+0000a5d0: 6f74 7320 7468 6174 2077 6572 6520 6e6f  ots that were no
+0000a5e0: 7420 7370 6563 6966 6963 2074 6f20 6269  t specific to bi
+0000a5f0: 6f69 6e66 6f72 6d61 7469 6373 2e20 5468  oinformatics. Th
+0000a600: 6573 6520 7479 7065 7320 6f66 2070 6c6f  ese types of plo
+0000a610: 7473 2063 616e 2062 6520 6765 6e65 7261  ts can be genera
+0000a620: 7465 6420 7769 7468 2073 6561 626f 726e  ted with seaborn
+0000a630: 206f 7220 616e 6f74 6865 7220 6765 6e65   or another gene
+0000a640: 7261 6c2d 7075 7270 6f73 6520 706c 6f74  ral-purpose plot
+0000a650: 7469 6e67 2070 6163 6b61 6765 2e0a 2a20  ting package..* 
+0000a660: 4465 7072 6563 6174 6564 2066 756e 6374  Deprecated funct
+0000a670: 696f 6e20 6073 6b62 696f 2e73 7461 7473  ion `skbio.stats
+0000a680: 2e70 6f77 6572 2e62 6f6f 7473 7472 6170  .power.bootstrap
+0000a690: 5f70 6f77 6572 5f63 7572 7665 6020 6861  _power_curve` ha
+0000a6a0: 7320 6265 656e 2072 656d 6f76 6564 2066  s been removed f
+0000a6b0: 726f 6d20 7363 696b 6974 2d62 696f 2061  rom scikit-bio a
+0000a6c0: 7320 7363 6865 6475 6c65 642e 2055 7365  s scheduled. Use
+0000a6d0: 2060 736b 6269 6f2e 7374 6174 732e 706f   `skbio.stats.po
+0000a6e0: 7765 722e 7375 6273 616d 706c 655f 706f  wer.subsample_po
+0000a6f0: 7765 7260 206f 7220 6073 6b62 696f 2e73  wer` or `skbio.s
+0000a700: 7461 7473 2e70 6f77 6572 2e73 7562 7361  tats.power.subsa
+0000a710: 6d70 6c65 5f70 6169 7265 645f 706f 7765  mple_paired_powe
+0000a720: 7260 2066 6f6c 6c6f 7765 6420 6279 2060  r` followed by `
+0000a730: 736b 6269 6f2e 7374 6174 732e 706f 7765  skbio.stats.powe
+0000a740: 722e 636f 6e66 6964 656e 6365 5f62 6f75  r.confidence_bou
+0000a750: 6e64 602e 0a2a 2044 6570 7265 6361 7465  nd`..* Deprecate
+0000a760: 6420 6675 6e63 7469 6f6e 2060 736b 6269  d function `skbi
+0000a770: 6f2e 7374 6174 732e 7370 6174 6961 6c2e  o.stats.spatial.
+0000a780: 7072 6f63 7275 7374 6573 6020 6861 7320  procrustes` has 
+0000a790: 6265 656e 2072 656d 6f76 6564 2066 726f  been removed fro
+0000a7a0: 6d20 7363 696b 6974 2d62 696f 2061 7320  m scikit-bio as 
+0000a7b0: 7363 6865 6475 6c65 6420 696e 2066 6176  scheduled in fav
+0000a7c0: 6f72 206f 6620 6073 6369 7079 2e73 7061  or of `scipy.spa
+0000a7d0: 7469 616c 2e70 726f 6372 7573 7465 7360  tial.procrustes`
+0000a7e0: 2e0a 2a20 4465 7072 6563 6174 6564 2063  ..* Deprecated c
+0000a7f0: 6c61 7373 2060 736b 6269 6f2e 7472 6565  lass `skbio.tree
+0000a800: 2e43 6f6d 7072 6573 7365 6454 7269 6560  .CompressedTrie`
+0000a810: 2061 6e64 2066 756e 6374 696f 6e20 6073   and function `s
+0000a820: 6b62 696f 2e74 7265 652e 6661 7374 615f  kbio.tree.fasta_
+0000a830: 746f 5f70 6169 726c 6973 7460 2068 6176  to_pairlist` hav
+0000a840: 6520 6265 656e 2072 656d 6f76 6564 2066  e been removed f
+0000a850: 726f 6d20 7363 696b 6974 2d62 696f 2061  rom scikit-bio a
+0000a860: 7320 7363 6865 6475 6c65 6420 696e 2066  s scheduled in f
+0000a870: 6176 6f72 206f 6620 6578 6973 7469 6e67  avor of existing
+0000a880: 2067 656e 6572 616c 2d70 7572 706f 7365   general-purpose
+0000a890: 2050 7974 686f 6e20 7472 6965 2070 6163   Python trie pac
+0000a8a0: 6b61 6765 732e 0a2a 2044 6570 7265 6361  kages..* Depreca
+0000a8b0: 7465 6420 6675 6e63 7469 6f6e 2060 736b  ted function `sk
+0000a8c0: 6269 6f2e 7574 696c 2e66 6c61 7474 656e  bio.util.flatten
+0000a8d0: 6020 6861 7320 6265 656e 2072 656d 6f76  ` has been remov
+0000a8e0: 6564 2066 726f 6d20 7363 696b 6974 2d62  ed from scikit-b
+0000a8f0: 696f 2061 7320 7363 6865 6475 6c65 6420  io as scheduled 
+0000a900: 696e 2066 6176 6f72 206f 6620 736f 6c75  in favor of solu
+0000a910: 7469 6f6e 7320 6176 6169 6c61 626c 6520  tions available 
+0000a920: 696e 2074 6865 2050 7974 686f 6e20 7374  in the Python st
+0000a930: 616e 6461 7264 206c 6962 7261 7279 2028  andard library (
+0000a940: 7365 6520 5b68 6572 655d 2868 7474 703a  see [here](http:
+0000a950: 2f2f 7374 6163 6b6f 7665 7266 6c6f 772e  //stackoverflow.
+0000a960: 636f 6d2f 612f 3935 3239 3532 2f33 3633  com/a/952952/363
+0000a970: 3930 3233 2920 616e 6420 5b68 6572 655d  9023) and [here]
+0000a980: 2868 7474 703a 2f2f 7374 6163 6b6f 7665  (http://stackove
+0000a990: 7266 6c6f 772e 636f 6d2f 612f 3430 3631  rflow.com/a/4061
+0000a9a0: 3939 2f33 3633 3930 3233 2920 666f 7220  99/3639023) for 
+0000a9b0: 6578 616d 706c 6573 292e 0a2a 2050 6169  examples)..* Pai
+0000a9c0: 7277 6973 6520 616c 6967 6e6d 656e 7420  rwise alignment 
+0000a9d0: 6675 6e63 7469 6f6e 7320 696e 2060 736b  functions in `sk
+0000a9e0: 6269 6f2e 616c 6967 6e6d 656e 7460 206e  bio.alignment` n
+0000a9f0: 6f77 2072 6574 7572 6e20 6120 7475 706c  ow return a tupl
+0000aa00: 6520 636f 6e74 6169 6e69 6e67 2074 6865  e containing the
+0000aa10: 2060 5461 6275 6c61 724d 5341 6020 616c   `TabularMSA` al
+0000aa20: 6967 6e6d 656e 742c 2061 6c69 676e 6d65  ignment, alignme
+0000aa30: 6e74 2073 636f 7265 2c20 616e 6420 7374  nt score, and st
+0000aa40: 6172 742f 656e 6420 706f 7369 7469 6f6e  art/end position
+0000aa50: 732e 2054 6865 2072 6574 7572 6e65 6420  s. The returned 
+0000aa60: 6054 6162 756c 6172 4d53 4160 2773 2060  `TabularMSA`'s `
+0000aa70: 696e 6465 7860 2069 7320 616c 7761 7973  index` is always
+0000aa80: 2074 6865 2064 6566 6175 6c74 2069 6e74   the default int
+0000aa90: 6567 6572 2069 6e64 6578 3b20 7365 7175  eger index; sequ
+0000aaa0: 656e 6365 2049 4473 2061 7265 206e 6f20  ence IDs are no 
+0000aab0: 6c6f 6e67 6572 2070 726f 7061 6761 7465  longer propagate
+0000aac0: 6420 746f 2074 6865 204d 5341 2e20 4164  d to the MSA. Ad
+0000aad0: 6469 7469 6f6e 616c 6c79 2c20 7468 6520  ditionally, the 
+0000aae0: 7061 6972 7769 7365 2061 6c69 676e 6d65  pairwise alignme
+0000aaf0: 6e74 2066 756e 6374 696f 6e73 206e 6f77  nt functions now
+0000ab00: 2061 6363 6570 7420 7468 6520 666f 6c6c   accept the foll
+0000ab10: 6f77 696e 6720 696e 7075 7420 7479 7065  owing input type
+0000ab20: 7320 746f 2061 6c69 676e 3a0a 2020 2020  s to align:.    
+0000ab30: 2d20 606c 6f63 616c 5f70 6169 7277 6973  - `local_pairwis
+0000ab40: 655f 616c 6967 6e5f 6e75 636c 656f 7469  e_align_nucleoti
+0000ab50: 6465 603a 2060 444e 4160 206f 7220 6052  de`: `DNA` or `R
+0000ab60: 4e41 600a 2020 2020 2d20 606c 6f63 616c  NA`.    - `local
+0000ab70: 5f70 6169 7277 6973 655f 616c 6967 6e5f  _pairwise_align_
+0000ab80: 7072 6f74 6569 6e60 3a20 6050 726f 7465  protein`: `Prote
+0000ab90: 696e 600a 2020 2020 2d20 606c 6f63 616c  in`.    - `local
+0000aba0: 5f70 6169 7277 6973 655f 616c 6967 6e60  _pairwise_align`
+0000abb0: 3a20 6049 5550 4143 5365 7175 656e 6365  : `IUPACSequence
+0000abc0: 600a 2020 2020 2d20 6067 6c6f 6261 6c5f  `.    - `global_
+0000abd0: 7061 6972 7769 7365 5f61 6c69 676e 5f6e  pairwise_align_n
+0000abe0: 7563 6c65 6f74 6964 6560 3a20 6044 4e41  ucleotide`: `DNA
+0000abf0: 602c 2060 524e 4160 2c20 6f72 2060 5461  `, `RNA`, or `Ta
+0000ac00: 6275 6c61 724d 5341 5b44 4e41 7c52 4e41  bularMSA[DNA|RNA
+0000ac10: 5d60 0a20 2020 202d 2060 676c 6f62 616c  ]`.    - `global
+0000ac20: 5f70 6169 7277 6973 655f 616c 6967 6e5f  _pairwise_align_
+0000ac30: 7072 6f74 6569 6e60 3a20 6050 726f 7465  protein`: `Prote
+0000ac40: 696e 6020 6f72 2060 5461 6275 6c61 724d  in` or `TabularM
+0000ac50: 5341 5b50 726f 7465 696e 5d60 0a20 2020  SA[Protein]`.   
+0000ac60: 202d 2060 676c 6f62 616c 5f70 6169 7277   - `global_pairw
+0000ac70: 6973 655f 616c 6967 6e60 3a20 6049 5550  ise_align`: `IUP
+0000ac80: 4143 5365 7175 656e 6365 6020 6f72 2060  ACSequence` or `
+0000ac90: 5461 6275 6c61 724d 5341 600a 2020 2020  TabularMSA`.    
+0000aca0: 2d20 606c 6f63 616c 5f70 6169 7277 6973  - `local_pairwis
+0000acb0: 655f 616c 6967 6e5f 7373 7760 3a20 6044  e_align_ssw`: `D
+0000acc0: 4e41 602c 2060 524e 4160 2c20 6f72 2060  NA`, `RNA`, or `
+0000acd0: 5072 6f74 6569 6e60 2e20 4164 6469 7469  Protein`. Additi
+0000ace0: 6f6e 616c 6c79 2c20 7468 6973 2066 756e  onally, this fun
+0000acf0: 6374 696f 6e20 6e6f 7720 6f76 6572 7269  ction now overri
+0000ad00: 6465 7320 7468 6520 6070 726f 7465 696e  des the `protein
+0000ad10: 6020 6b77 6172 6720 6261 7365 6420 6f6e  ` kwarg based on
+0000ad20: 2069 6e70 7574 2074 7970 652e 2060 636f   input type. `co
+0000ad30: 6e73 7472 7563 746f 7260 2070 6172 616d  nstructor` param
+0000ad40: 6574 6572 2077 6173 2072 656d 6f76 6564  eter was removed
+0000ad50: 2062 6563 6175 7365 2074 6865 2066 756e   because the fun
+0000ad60: 6374 696f 6e20 6e6f 7720 6465 7465 726d  ction now determ
+0000ad70: 696e 6573 2074 6865 2072 6574 7572 6e20  ines the return 
+0000ad80: 7479 7065 2062 6173 6564 206f 6e20 696e  type based on in
+0000ad90: 7075 7420 7479 7065 2e0a 2a20 5265 6d6f  put type..* Remo
+0000ada0: 7665 6420 6073 6b62 696f 2e61 6c69 676e  ved `skbio.align
+0000adb0: 6d65 6e74 2e53 6571 7565 6e63 6543 6f6c  ment.SequenceCol
+0000adc0: 6c65 6374 696f 6e60 2069 6e20 6661 766f  lection` in favo
+0000add0: 7220 6f66 2075 7369 6e67 2061 206c 6973  r of using a lis
+0000ade0: 7420 6f72 206f 7468 6572 2073 7461 6e64  t or other stand
+0000adf0: 6172 6420 6c69 6272 6172 7920 636f 6e74  ard library cont
+0000ae00: 6169 6e65 7273 2074 6f20 7374 6f72 6520  ainers to store 
+0000ae10: 7363 696b 6974 2d62 696f 2073 6571 7565  scikit-bio seque
+0000ae20: 6e63 6520 6f62 6a65 6374 7320 286d 6f73  nce objects (mos
+0000ae30: 7420 6053 6571 7565 6e63 6543 6f6c 6c65  t `SequenceColle
+0000ae40: 6374 696f 6e60 206f 7065 7261 7469 6f6e  ction` operation
+0000ae50: 7320 7765 7265 2073 696d 706c 6520 6c69  s were simple li
+0000ae60: 7374 2063 6f6d 7072 6568 656e 7369 6f6e  st comprehension
+0000ae70: 7329 2e20 5573 6520 6044 6973 7461 6e63  s). Use `Distanc
+0000ae80: 654d 6174 7269 782e 6672 6f6d 5f69 7465  eMatrix.from_ite
+0000ae90: 7261 626c 6560 2069 6e73 7465 6164 206f  rable` instead o
+0000aea0: 6620 6053 6571 7565 6e63 6543 6f6c 6c65  f `SequenceColle
+0000aeb0: 6374 696f 6e2e 6469 7374 616e 6365 7360  ction.distances`
+0000aec0: 2028 7061 7373 2060 6b65 793d 2269 6422   (pass `key="id"
+0000aed0: 6020 746f 2065 7861 6374 6c79 206d 6174  ` to exactly mat
+0000aee0: 6368 206f 7269 6769 6e61 6c20 6265 6861  ch original beha
+0000aef0: 7669 6f72 292e 0a2a 2052 656d 6f76 6564  vior)..* Removed
+0000af00: 2060 736b 6269 6f2e 616c 6967 6e6d 656e   `skbio.alignmen
+0000af10: 742e 416c 6967 6e6d 656e 7460 2069 6e20  t.Alignment` in 
+0000af20: 6661 766f 7220 6f66 2060 736b 6269 6f2e  favor of `skbio.
+0000af30: 616c 6967 6e6d 656e 742e 5461 6275 6c61  alignment.Tabula
+0000af40: 724d 5341 602e 0a2a 2052 656d 6f76 6564  rMSA`..* Removed
+0000af50: 2060 736b 6269 6f2e 616c 6967 6e6d 656e   `skbio.alignmen
+0000af60: 742e 5365 7175 656e 6365 436f 6c6c 6563  t.SequenceCollec
+0000af70: 7469 6f6e 4572 726f 7260 2061 6e64 2060  tionError` and `
+0000af80: 736b 6269 6f2e 616c 6967 6e6d 656e 742e  skbio.alignment.
+0000af90: 416c 6967 6e6d 656e 7445 7272 6f72 6020  AlignmentError` 
+0000afa0: 6578 6365 7074 696f 6e73 2061 7320 7468  exceptions as th
+0000afb0: 6569 7220 636f 7272 6573 706f 6e64 696e  eir correspondin
+0000afc0: 6720 636c 6173 7365 7320 6e6f 206c 6f6e  g classes no lon
+0000afd0: 6765 7220 6578 6973 742e 0a0a 2323 2320  ger exist...### 
+0000afe0: 4275 6720 4669 7865 730a 0a2a 2060 6053  Bug Fixes..* ``S
+0000aff0: 6571 7565 6e63 6560 6020 6f62 6a65 6374  equence`` object
+0000b000: 7320 6e6f 7720 6861 6e64 6c65 2073 6c69  s now handle sli
+0000b010: 6369 6e67 206f 6620 656d 7074 7920 706f  cing of empty po
+0000b020: 7369 7469 6f6e 616c 206d 6574 6164 6174  sitional metadat
+0000b030: 6120 636f 7272 6563 746c 792e 2041 6e79  a correctly. Any
+0000b040: 206d 6574 6164 6174 6120 7468 6174 2069   metadata that i
+0000b050: 7320 656d 7074 7920 7769 6c6c 206e 6f20  s empty will no 
+0000b060: 6c6f 6e67 6572 2062 6520 7072 6f70 6167  longer be propag
+0000b070: 6174 6564 2062 7920 7468 6520 696e 7465  ated by the inte
+0000b080: 726e 616c 2060 605f 746f 6060 2063 6f6e  rnal ``_to`` con
+0000b090: 7374 7275 6374 6f72 2e20 285b 2331 3133  structor. ([#113
+0000b0a0: 335d 2868 7474 7073 3a2f 2f67 6974 6875  3](https://githu
+0000b0b0: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
+0000b0c0: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
+0000b0d0: 3131 3333 2929 0a2a 2060 6044 6973 7369  1133)).* ``Dissi
+0000b0e0: 6d69 6c61 7269 7479 4d61 7472 6978 2e70  milarityMatrix.p
+0000b0f0: 6c6f 7428 2960 6020 6e6f 206c 6f6e 6765  lot()`` no longe
+0000b100: 7220 6c65 6176 6573 2061 2077 6869 7465  r leaves a white
+0000b110: 2062 6f72 6465 7220 6172 6f75 6e64 2074   border around t
+0000b120: 6865 0a20 2068 6561 746d 6170 2069 7420  he.  heatmap it 
+0000b130: 706c 6f74 7320 2850 5220 2331 3037 3029  plots (PR #1070)
+0000b140: 2e0a 2a20 5472 6565 4e6f 6465 2e72 6f6f  ..* TreeNode.roo
+0000b150: 745f 6174 5f6d 6964 706f 696e 7460 6020  t_at_midpoint`` 
+0000b160: 6e6f 206c 6f6e 6765 7220 6661 696c 7320  no longer fails 
+0000b170: 7768 656e 2061 206e 6f64 6520 7769 7468  when a node with
+0000b180: 2074 776f 2065 7175 616c 206c 656e 6774   two equal lengt
+0000b190: 6820 6368 696c 6420 6272 616e 6368 6573  h child branches
+0000b1a0: 2065 7869 7374 7320 696e 2074 6865 2074   exists in the t
+0000b1b0: 7265 652e 2028 5b23 3130 3737 5d28 6874  ree. ([#1077](ht
+0000b1c0: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+0000b1d0: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
+0000b1e0: 6269 6f2f 6973 7375 6573 2f31 3037 3729  bio/issues/1077)
+0000b1f0: 290a 2a20 6060 5472 6565 4e6f 6465 2e5f  ).* ``TreeNode._
+0000b200: 7365 745f 6d61 785f 6469 7374 616e 6365  set_max_distance
+0000b210: 6060 2c20 6173 2063 616c 6c65 6420 7468  ``, as called th
+0000b220: 726f 7567 6820 6060 5472 6565 4e6f 6465  rough ``TreeNode
+0000b230: 2e67 6574 5f6d 6178 5f64 6973 7461 6e63  .get_max_distanc
+0000b240: 6560 6020 6f72 2060 6054 7265 654e 6f64  e`` or ``TreeNod
+0000b250: 652e 726f 6f74 5f61 745f 6d69 6470 6f69  e.root_at_midpoi
+0000b260: 6e74 6060 2077 6f75 6c64 2073 746f 7265  nt`` would store
+0000b270: 2064 6973 7461 6e63 6520 696e 666f 726d   distance inform
+0000b280: 6174 696f 6e20 6173 2060 606c 6973 7460  ation as ``list`
+0000b290: 6073 2069 6e20 7468 6520 6174 7472 6962  `s in the attrib
+0000b2a0: 7574 6520 6060 4d61 7844 6973 7454 6970  ute ``MaxDistTip
+0000b2b0: 7360 6020 6f6e 2065 6163 6820 6e6f 6465  s`` on each node
+0000b2c0: 2069 6e20 7468 6520 7472 6565 2c20 686f   in the tree, ho
+0000b2d0: 7765 7665 722c 2074 6865 7365 2064 6973  wever, these dis
+0000b2e0: 7461 6e63 6573 2077 6572 6520 6f6e 6c79  tances were only
+0000b2f0: 2076 616c 6964 2066 6f72 2074 6865 206e   valid for the n
+0000b300: 6f64 6520 696e 2077 6869 6368 2074 6865  ode in which the
+0000b310: 2063 616c 6c20 746f 2060 605f 7365 745f   call to ``_set_
+0000b320: 6d61 785f 6469 7374 616e 6365 6060 2077  max_distance`` w
+0000b330: 6173 206d 6164 652e 2054 6865 2076 616c  as made. The val
+0000b340: 7565 7320 636f 6e74 6169 6e65 6420 696e  ues contained in
+0000b350: 2060 604d 6178 4469 7374 5469 7073 6060   ``MaxDistTips``
+0000b360: 2061 7265 206e 6f77 2063 6f72 7265 6374   are now correct
+0000b370: 2061 6372 6f73 7320 7468 6520 7472 6565   across the tree
+0000b380: 2066 6f6c 6c6f 7769 6e67 2061 2063 616c   following a cal
+0000b390: 6c20 746f 2060 6067 6574 5f6d 6178 5f64  l to ``get_max_d
+0000b3a0: 6973 7461 6e63 6560 602e 2054 6865 2073  istance``. The s
+0000b3b0: 636f 7065 206f 6620 696d 7061 6374 206f  cope of impact o
+0000b3c0: 6620 7468 6973 2062 7567 2069 7320 6c69  f this bug is li
+0000b3d0: 6d69 7465 6420 746f 2075 7365 7273 2074  mited to users t
+0000b3e0: 6861 7420 7765 7265 2069 6e74 6572 6163  hat were interac
+0000b3f0: 7469 6e67 2064 6972 6563 746c 7920 7769  ting directly wi
+0000b400: 7468 2060 604d 6178 4469 7374 5469 7073  th ``MaxDistTips
+0000b410: 6060 206f 6e20 6465 7363 656e 6461 6e74  `` on descendant
+0000b420: 206e 6f64 6573 3b20 7468 6973 2062 7567   nodes; this bug
+0000b430: 2064 6f65 7320 6e6f 7420 696d 7061 6374   does not impact
+0000b440: 2061 6e79 206b 6e6f 776e 206d 6574 686f   any known metho
+0000b450: 6420 7769 7468 696e 2073 6369 6b69 742d  d within scikit-
+0000b460: 6269 6f2e 2028 5b23 3132 3233 5d28 6874  bio. ([#1223](ht
+0000b470: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+0000b480: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
+0000b490: 6269 6f2f 6973 7375 6573 2f31 3232 3329  bio/issues/1223)
+0000b4a0: 290a 2a20 4164 6465 6420 6d69 7373 696e  ).* Added missin
+0000b4b0: 6720 606e 6f73 6560 2064 6570 656e 6465  g `nose` depende
+0000b4c0: 6e63 7920 746f 2073 6574 7570 2e70 7927  ncy to setup.py'
+0000b4d0: 7320 6069 6e73 7461 6c6c 5f72 6571 7569  s `install_requi
+0000b4e0: 7265 7360 2e20 285b 2331 3231 345d 2868  res`. ([#1214](h
+0000b4f0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+0000b500: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
+0000b510: 2d62 696f 2f69 7373 7565 732f 3132 3134  -bio/issues/1214
+0000b520: 2929 0a2a 2046 6978 6564 2069 7373 7565  )).* Fixed issue
+0000b530: 2074 6861 7420 7265 7375 6c74 6564 2069   that resulted i
+0000b540: 6e20 6c65 6765 6e64 7320 6f66 2060 604f  n legends of ``O
+0000b550: 7264 696e 6174 696f 6e52 6573 756c 7460  rdinationResult`
+0000b560: 6020 706c 6f74 7320 736f 6d65 7469 6d65  ` plots sometime
+0000b570: 7320 6265 696e 6720 7472 756e 6361 7465  s being truncate
+0000b580: 642e 2028 5b23 3132 3130 5d28 6874 7470  d. ([#1210](http
+0000b590: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+0000b5a0: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+0000b5b0: 6f2f 6973 7375 6573 2f31 3231 3029 290a  o/issues/1210)).
+0000b5c0: 0a23 2323 2044 6570 7265 6361 7465 6420  .### Deprecated 
+0000b5d0: 6675 6e63 7469 6f6e 616c 6974 7920 5b73  functionality [s
+0000b5e0: 7461 626c 655d 0a2a 2060 736b 6269 6f2e  table].* `skbio.
+0000b5f0: 5365 7175 656e 6365 2e63 6f70 7960 2068  Sequence.copy` h
+0000b600: 6173 2062 6565 6e20 6465 7072 6563 6174  as been deprecat
+0000b610: 6564 2069 6e20 6661 766f 7220 6f66 2060  ed in favor of `
+0000b620: 636f 7079 2e63 6f70 7928 7365 7129 6020  copy.copy(seq)` 
+0000b630: 616e 6420 6063 6f70 792e 6465 6570 636f  and `copy.deepco
+0000b640: 7079 2873 6571 2960 2e0a 0a23 2323 204d  py(seq)`...### M
+0000b650: 6973 6365 6c6c 616e 656f 7573 0a2a 2044  iscellaneous.* D
+0000b660: 6f63 7465 7374 7320 6172 6520 6e6f 7720  octests are now 
+0000b670: 7772 6974 7465 6e20 696e 2050 7974 686f  written in Pytho
+0000b680: 6e20 332e 0a2a 2060 606d 616b 6520 7465  n 3..* ``make te
+0000b690: 7374 6060 206e 6f77 2076 616c 6964 6174  st`` now validat
+0000b6a0: 6573 204d 414e 4946 4553 542e 696e 2075  es MANIFEST.in u
+0000b6b0: 7369 6e67 205b 6368 6563 6b2d 6d61 6e69  sing [check-mani
+0000b6c0: 6665 7374 5d28 6874 7470 733a 2f2f 6769  fest](https://gi
+0000b6d0: 7468 7562 2e63 6f6d 2f6d 6765 646d 696e  thub.com/mgedmin
+0000b6e0: 2f63 6865 636b 2d6d 616e 6966 6573 7429  /check-manifest)
+0000b6f0: 2e20 285b 2334 3631 5d28 6874 7470 733a  . ([#461](https:
+0000b700: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
+0000b710: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
+0000b720: 6973 7375 6573 2f34 3631 2929 0a2a 204d  issues/461)).* M
+0000b730: 616e 7920 6e65 7720 616c 7068 6120 6469  any new alpha di
+0000b740: 7665 7273 6974 7920 6571 7561 7469 6f6e  versity equation
+0000b750: 7320 6164 6465 6420 746f 2060 6073 6b62  s added to ``skb
+0000b760: 696f 2e64 6976 6572 7369 7479 2e61 6c70  io.diversity.alp
+0000b770: 6861 6060 2064 6f63 756d 656e 7461 7469  ha`` documentati
+0000b780: 6f6e 2e20 285b 2333 3231 5d28 6874 7470  on. ([#321](http
+0000b790: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+0000b7a0: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+0000b7b0: 6f2f 6973 7375 6573 2f33 3231 2929 0a2a  o/issues/321)).*
+0000b7c0: 204f 7264 6572 206f 6620 6060 6c6f 7765   Order of ``lowe
+0000b7d0: 7263 6173 6560 6020 616e 6420 6060 7661  rcase`` and ``va
+0000b7e0: 6c69 6461 7465 6060 206b 6579 776f 7264  lidate`` keyword
+0000b7f0: 7320 7377 6170 7065 6420 696e 2060 6044  s swapped in ``D
+0000b800: 4e41 6060 2c20 6060 524e 4160 602c 2061  NA``, ``RNA``, a
+0000b810: 6e64 2060 6050 726f 7465 696e 6060 2e0a  nd ``Protein``..
+0000b820: 0a23 2320 5665 7273 696f 6e20 302e 342e  .## Version 0.4.
+0000b830: 3020 2832 3031 352d 3037 2d30 3829 0a0a  0 (2015-07-08)..
+0000b840: 496e 6974 6961 6c20 6265 7461 2072 656c  Initial beta rel
+0000b850: 6561 7365 2e20 496e 2061 6464 6974 696f  ease. In additio
+0000b860: 6e20 746f 2074 6865 2063 6861 6e67 6573  n to the changes
+0000b870: 2064 6574 6169 6c65 6420 6265 6c6f 772c   detailed below,
+0000b880: 2074 6865 2066 6f6c 6c6f 7769 6e67 0a73   the following.s
+0000b890: 7562 7061 636b 6167 6573 2068 6176 6520  ubpackages have 
+0000b8a0: 6265 656e 206d 6f73 746c 7920 6f72 2065  been mostly or e
+0000b8b0: 6e74 6972 656c 7920 7265 7772 6974 7465  ntirely rewritte
+0000b8c0: 6e20 616e 6420 6d6f 7374 206f 6620 7468  n and most of th
+0000b8d0: 6569 7220 4150 4973 2061 7265 0a73 7562  eir APIs are.sub
+0000b8e0: 7374 616e 7469 616c 6c79 2064 6966 6665  stantially diffe
+0000b8f0: 7265 6e74 2028 616e 6420 696d 7072 6f76  rent (and improv
+0000b900: 6564 2129 3a0a 0a2a 2060 736b 6269 6f2e  ed!):..* `skbio.
+0000b910: 7365 7175 656e 6365 600a 2a20 6073 6b62  sequence`.* `skb
+0000b920: 696f 2e69 6f60 0a0a 5468 6520 4150 4973  io.io`..The APIs
+0000b930: 206f 6620 7468 6573 6520 7375 6270 6163   of these subpac
+0000b940: 6b61 6765 7320 6172 6520 6e6f 7720 7374  kages are now st
+0000b950: 6162 6c65 2c20 616e 6420 616c 6c20 6f74  able, and all ot
+0000b960: 6865 7273 2061 7265 2065 7870 6572 696d  hers are experim
+0000b970: 656e 7461 6c2e 2053 6565 2074 6865 205b  ental. See the [
+0000b980: 4150 4920 7374 6162 696c 6974 7920 646f  API stability do
+0000b990: 6373 5d28 6874 7470 733a 2f2f 6769 7468  cs](https://gith
+0000b9a0: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+0000b9b0: 6369 6b69 742d 6269 6f2f 7472 6565 2f30  cikit-bio/tree/0
+0000b9c0: 2e34 2e30 2f64 6f63 2f73 6f75 7263 652f  .4.0/doc/source/
+0000b9d0: 7573 6572 2f61 7069 5f73 7461 6269 6c69  user/api_stabili
+0000b9e0: 7479 2e72 7374 2920 666f 7220 6d6f 7265  ty.rst) for more
+0000b9f0: 2064 6574 6169 6c73 2c20 696e 636c 7564   details, includ
+0000ba00: 696e 6720 7768 6174 2077 6520 6d65 616e  ing what we mean
+0000ba10: 2062 7920 2a73 7461 626c 652a 2061 6e64   by *stable* and
+0000ba20: 202a 6578 7065 7269 6d65 6e74 616c 2a20   *experimental* 
+0000ba30: 696e 2074 6869 7320 636f 6e74 6578 742e  in this context.
+0000ba40: 2057 6520 7265 636f 676e 697a 6520 7468   We recognize th
+0000ba50: 6174 2074 6869 7320 6973 2061 206c 6f74  at this is a lot
+0000ba60: 206f 6620 6261 636b 7761 7264 2d69 6e63   of backward-inc
+0000ba70: 6f6d 7061 7469 626c 6520 6368 616e 6765  ompatible change
+0000ba80: 732e 2054 6f20 6176 6f69 6420 7468 6573  s. To avoid thes
+0000ba90: 6520 7479 7065 7320 6f66 2063 6861 6e67  e types of chang
+0000baa0: 6573 2062 6569 6e67 2061 2073 7572 7072  es being a surpr
+0000bab0: 6973 6520 746f 206f 7572 2075 7365 7273  ise to our users
+0000bac0: 2c20 6f75 7220 7075 626c 6963 2041 5049  , our public API
+0000bad0: 7320 6172 6520 6e6f 7720 6465 636f 7261  s are now decora
+0000bae0: 7465 6420 746f 206d 616b 6520 6974 2063  ted to make it c
+0000baf0: 6c65 6172 2074 6f20 6465 7665 6c6f 7065  lear to develope
+0000bb00: 7273 2077 6865 6e20 616e 2041 5049 2063  rs when an API c
+0000bb10: 616e 2062 6520 7265 6c69 6564 2075 706f  an be relied upo
+0000bb20: 6e20 2873 7461 626c 6529 2061 6e64 2077  n (stable) and w
+0000bb30: 6865 6e20 6974 206d 6179 2062 6520 7375  hen it may be su
+0000bb40: 626a 6563 7420 746f 2063 6861 6e67 6520  bject to change 
+0000bb50: 2865 7870 6572 696d 656e 7461 6c29 2e0a  (experimental)..
+0000bb60: 0a23 2323 2046 6561 7475 7265 730a 2a20  .### Features.* 
+0000bb70: 4164 6465 6420 6073 6b62 696f 2e73 7461  Added `skbio.sta
+0000bb80: 7473 2e63 6f6d 706f 7369 7469 6f6e 6020  ts.composition` 
+0000bb90: 666f 7220 616e 616c 797a 696e 6720 6461  for analyzing da
+0000bba0: 7461 206d 6164 6520 7570 206f 6620 7072  ta made up of pr
+0000bbb0: 6f70 6f72 7469 6f6e 730a 2a20 4164 6465  oportions.* Adde
+0000bbc0: 6420 6e65 7720 6060 736b 6269 6f2e 7374  d new ``skbio.st
+0000bbd0: 6174 732e 6576 6f6c 7665 6060 2073 7562  ats.evolve`` sub
+0000bbe0: 7061 636b 6167 6520 666f 7220 6576 6f6c  package for evol
+0000bbf0: 7574 696f 6e61 7279 2073 7461 7469 7374  utionary statist
+0000bc00: 6963 732e 2043 7572 7265 6e74 6c79 2063  ics. Currently c
+0000bc10: 6f6e 7461 696e 7320 6120 7369 6e67 6c65  ontains a single
+0000bc20: 2066 756e 6374 696f 6e2c 2060 6068 6f6d   function, ``hom
+0000bc30: 6d6f 6c61 5f63 6f73 7065 6369 6174 696f  mola_cospeciatio
+0000bc40: 6e60 602c 2077 6869 6368 2069 6d70 6c65  n``, which imple
+0000bc50: 6d65 6e74 7320 6120 7065 726d 7574 6174  ments a permutat
+0000bc60: 696f 6e2d 6261 7365 6420 7465 7374 206f  ion-based test o
+0000bc70: 6620 636f 7272 656c 6174 696f 6e20 6265  f correlation be
+0000bc80: 7477 6565 6e20 7477 6f20 6469 7374 616e  tween two distan
+0000bc90: 6365 206d 6174 7269 6365 732e 0a2a 2041  ce matrices..* A
+0000bca0: 6464 6564 2073 7570 706f 7274 2066 6f72  dded support for
+0000bcb0: 2060 6073 6b62 696f 2e69 6f2e 7574 696c   ``skbio.io.util
+0000bcc0: 2e6f 7065 6e5f 6669 6c65 6060 2061 6e64  .open_file`` and
+0000bcd0: 2060 6073 6b62 696f 2e69 6f2e 7574 696c   ``skbio.io.util
+0000bce0: 2e6f 7065 6e5f 6669 6c65 7360 6020 746f  .open_files`` to
+0000bcf0: 2070 756c 6c20 6669 6c65 7320 6672 6f6d   pull files from
+0000bd00: 2048 5454 5020 616e 6420 4854 5450 5320   HTTP and HTTPS 
+0000bd10: 5552 4c73 2e20 5468 6973 2062 6568 6176  URLs. This behav
+0000bd20: 696f 7220 7072 6f70 6167 6174 6573 2074  ior propagates t
+0000bd30: 6f20 7468 6520 492f 4f20 7265 6769 7374  o the I/O regist
+0000bd40: 7279 2e0a 2a20 4641 5354 412f 5155 414c  ry..* FASTA/QUAL
+0000bd50: 2028 6060 736b 6269 6f2e 696f 2e66 6f72   (``skbio.io.for
+0000bd60: 6d61 742e 6661 7374 6160 6029 2061 6e64  mat.fasta``) and
+0000bd70: 2046 4153 5451 2028 6060 736b 6269 6f2e   FASTQ (``skbio.
+0000bd80: 696f 2e66 6f72 6d61 742e 6661 7374 7160  io.format.fastq`
+0000bd90: 6029 2072 6561 6465 7273 206e 6f77 2061  `) readers now a
+0000bda0: 6c6c 6f77 2062 6c61 6e6b 206f 7220 7768  llow blank or wh
+0000bdb0: 6974 6573 7061 6365 2d6f 6e6c 7920 6c69  itespace-only li
+0000bdc0: 6e65 7320 6174 2074 6865 2062 6567 696e  nes at the begin
+0000bdd0: 6e69 6e67 206f 6620 7468 6520 6669 6c65  ning of the file
+0000bde0: 2c20 6265 7477 6565 6e20 7265 636f 7264  , between record
+0000bdf0: 732c 206f 7220 6174 2074 6865 2065 6e64  s, or at the end
+0000be00: 206f 6620 7468 6520 6669 6c65 2e20 4120   of the file. A 
+0000be10: 626c 616e 6b20 6f72 2077 6869 7465 7370  blank or whitesp
+0000be20: 6163 652d 6f6e 6c79 206c 696e 6520 696e  ace-only line in
+0000be30: 2061 6e79 206f 7468 6572 206c 6f63 6174   any other locat
+0000be40: 696f 6e20 7769 6c6c 2063 6f6e 7469 6e75  ion will continu
+0000be50: 6520 746f 2072 6169 7365 2061 6e20 6572  e to raise an er
+0000be60: 726f 7220 5b23 3738 315d 2868 7474 7073  ror [#781](https
+0000be70: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
+0000be80: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
+0000be90: 2f69 7373 7565 732f 3738 3129 2e0a 2a20  /issues/781)..* 
+0000bea0: 7363 696b 6974 2d62 696f 206e 6f77 2069  scikit-bio now i
+0000beb0: 676e 6f72 6573 206c 6561 6469 6e67 2061  gnores leading a
+0000bec0: 6e64 2074 7261 696c 696e 6720 7768 6974  nd trailing whit
+0000bed0: 6573 7061 6365 2063 6861 7261 6374 6572  espace character
+0000bee0: 7320 6f6e 2065 6163 6820 6c69 6e65 2077  s on each line w
+0000bef0: 6869 6c65 2072 6561 6469 6e67 2046 4153  hile reading FAS
+0000bf00: 5441 2f51 5541 4c20 616e 6420 4641 5354  TA/QUAL and FAST
+0000bf10: 5120 6669 6c65 732e 0a2a 2041 6464 6564  Q files..* Added
+0000bf20: 2060 7261 7469 6f60 2070 6172 616d 6574   `ratio` paramet
+0000bf30: 6572 2074 6f20 6073 6b62 696f 2e73 7461  er to `skbio.sta
+0000bf40: 7473 2e70 6f77 6572 2e73 7562 7361 6d70  ts.power.subsamp
+0000bf50: 6c65 5f70 6f77 6572 602e 2054 6869 7320  le_power`. This 
+0000bf60: 616c 6c6f 7773 2074 6865 2075 7365 7220  allows the user 
+0000bf70: 746f 2063 616c 6375 6c61 7465 2070 6f77  to calculate pow
+0000bf80: 6572 206f 6e20 6772 6f75 7073 2066 6f72  er on groups for
+0000bf90: 2075 6e65 7665 6e20 7369 7a65 2028 466f   uneven size (Fo
+0000bfa0: 7220 6578 616d 706c 652c 2064 7261 7720  r example, draw 
+0000bfb0: 7477 6963 6520 6173 206d 616e 7920 7361  twice as many sa
+0000bfc0: 6d70 6c65 7320 6672 6f6d 2047 726f 7570  mples from Group
+0000bfd0: 2042 2074 6861 6e20 4772 6f75 7020 4129   B than Group A)
+0000bfe0: 2e20 4966 2060 7261 7469 6f60 2069 7320  . If `ratio` is 
+0000bff0: 6e6f 7420 7365 742c 2067 726f 7570 2073  not set, group s
+0000c000: 697a 6573 2077 696c 6c20 7265 6d61 696e  izes will remain
+0000c010: 2065 7175 616c 2061 6372 6f73 7320 616c   equal across al
+0000c020: 6c20 6772 6f75 7073 2e0a 2a20 506f 7765  l groups..* Powe
+0000c030: 7220 6361 6c63 756c 6174 696f 6e73 2028  r calculations (
+0000c040: 6073 6b62 696f 2e73 7461 7473 2e70 6f77  `skbio.stats.pow
+0000c050: 6572 2e73 7562 7361 6d70 6c65 5f70 6f77  er.subsample_pow
+0000c060: 6572 6020 616e 6420 6073 6b62 696f 2e73  er` and `skbio.s
+0000c070: 7461 7473 2e70 6f77 6572 2e73 7562 7361  tats.power.subsa
+0000c080: 6d70 6c65 5f70 6169 7265 645f 706f 7765  mple_paired_powe
+0000c090: 7260 2920 6361 6e20 7573 6520 7465 7374  r`) can use test
+0000c0a0: 2066 756e 6374 696f 6e73 2074 6861 7420   functions that 
+0000c0b0: 7265 7475 726e 206d 756c 7469 706c 6520  return multiple 
+0000c0c0: 7020 7661 6c75 6573 2c20 6c69 6b65 2073  p values, like s
+0000c0d0: 6f6d 6520 6d75 6c74 6976 6172 6961 7465  ome multivariate
+0000c0e0: 206c 696e 6561 7220 7265 6772 6573 7369   linear regressi
+0000c0f0: 6f6e 206d 6f64 656c 732e 2050 7265 7669  on models. Previ
+0000c100: 6f75 736c 792c 2074 6865 2070 6f77 6572  ously, the power
+0000c110: 2063 616c 6375 6c61 7469 6f6e 7320 7265   calculations re
+0000c120: 7175 6972 6564 2074 6865 2074 6573 7420  quired the test 
+0000c130: 746f 2072 6574 7572 6e20 6120 7369 6e67  to return a sing
+0000c140: 6c65 2070 2076 616c 7565 2e0a 2a20 4164  le p value..* Ad
+0000c150: 6465 6420 6060 736b 6269 6f2e 7574 696c  ded ``skbio.util
+0000c160: 2e61 7373 6572 745f 6461 7461 5f66 7261  .assert_data_fra
+0000c170: 6d65 5f61 6c6d 6f73 745f 6571 7561 6c60  me_almost_equal`
+0000c180: 6020 6675 6e63 7469 6f6e 2066 6f72 2063  ` function for c
+0000c190: 6f6d 7061 7269 6e67 2060 6070 642e 4461  omparing ``pd.Da
+0000c1a0: 7461 4672 616d 6560 6020 6f62 6a65 6374  taFrame`` object
+0000c1b0: 7320 696e 2075 6e69 7420 7465 7374 732e  s in unit tests.
+0000c1c0: 0a0a 2323 2320 5065 7266 6f72 6d61 6e63  ..### Performanc
+0000c1d0: 6520 656e 6861 6e63 656d 656e 7473 0a2a  e enhancements.*
+0000c1e0: 2054 6865 2073 7065 6564 206f 6620 7175   The speed of qu
+0000c1f0: 616c 6974 7920 7363 6f72 6520 6465 636f  ality score deco
+0000c200: 6469 6e67 2068 6173 2062 6565 6e20 7369  ding has been si
+0000c210: 676e 6966 6963 616e 746c 7920 696d 7072  gnificantly impr
+0000c220: 6f76 6564 2028 7e32 7829 2077 6865 6e20  oved (~2x) when 
+0000c230: 7265 6164 696e 6720 6066 6173 7471 6020  reading `fastq` 
+0000c240: 6669 6c65 732e 0a2a 2054 6865 2073 7065  files..* The spe
+0000c250: 6564 206f 6620 604e 7563 6c65 6f74 6964  ed of `Nucleotid
+0000c260: 6553 6571 7565 6e63 652e 7265 7665 7273  eSequence.revers
+0000c270: 655f 636f 6d70 6c65 6d65 6e74 6020 6861  e_complement` ha
+0000c280: 7320 6265 656e 2069 6d70 726f 7665 6420  s been improved 
+0000c290: 287e 3678 292e 0a0a 2323 2320 4275 6720  (~6x)...### Bug 
+0000c2a0: 6669 7865 730a 2a20 4368 616e 6765 6420  fixes.* Changed 
+0000c2b0: 6053 6571 7565 6e63 652e 6469 7374 616e  `Sequence.distan
+0000c2c0: 6365 6020 746f 2072 6169 7365 2061 6e20  ce` to raise an 
+0000c2d0: 6572 726f 7220 616e 7920 7469 6d65 2074  error any time t
+0000c2e0: 776f 2073 6571 7565 6e63 6573 2061 7265  wo sequences are
+0000c2f0: 2070 6173 7365 6420 6f66 2064 6966 6665   passed of diffe
+0000c300: 7265 6e74 206c 656e 6774 6873 2072 6567  rent lengths reg
+0000c310: 6172 646c 6573 7320 6f66 2074 6865 2060  ardless of the `
+0000c320: 6469 7374 616e 6365 5f66 6e60 2062 6569  distance_fn` bei
+0000c330: 6e67 2070 6173 7365 642e 205b 2823 3531  ng passed. [(#51
+0000c340: 3429 5d28 6874 7470 733a 2f2f 6769 7468  4)](https://gith
+0000c350: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+0000c360: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
+0000c370: 2f35 3134 290a 2a20 4669 7865 6420 6973  /514).* Fixed is
+0000c380: 7375 6520 7769 7468 2060 6054 7265 654e  sue with ``TreeN
+0000c390: 6f64 652e 6578 7465 6e64 6060 2077 6865  ode.extend`` whe
+0000c3a0: 7265 2069 6620 6769 7665 6e20 7468 6520  re if given the 
+0000c3b0: 6368 696c 6472 656e 206f 6620 616e 6f74  children of anot
+0000c3c0: 6865 7220 6060 5472 6565 4e6f 6465 6060  her ``TreeNode``
+0000c3d0: 206f 626a 6563 7420 2860 6074 7265 652e   object (``tree.
+0000c3e0: 6368 696c 6472 656e 6060 292c 2062 6f74  children``), bot
+0000c3f0: 6820 7472 6565 7320 776f 756c 6420 6265  h trees would be
+0000c400: 206c 6566 7420 696e 2061 6e20 696e 636f   left in an inco
+0000c410: 7272 6563 7420 616e 6420 756e 7072 6564  rrect and unpred
+0000c420: 6963 7461 626c 6520 7374 6174 652e 2028  ictable state. (
+0000c430: 5b23 3838 395d 2868 7474 7073 3a2f 2f67  [#889](https://g
+0000c440: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
+0000c450: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
+0000c460: 7565 732f 3838 3929 290a 2a20 4368 616e  ues/889)).* Chan
+0000c470: 6765 6420 7468 6520 7761 7920 706f 7765  ged the way powe
+0000c480: 7220 7761 7320 6361 6c63 756c 6174 6564  r was calculated
+0000c490: 2069 6e20 6073 7562 7361 6d70 6c65 5f70   in `subsample_p
+0000c4a0: 6169 7265 645f 706f 7765 7260 2074 6f20  aired_power` to 
+0000c4b0: 6d6f 7665 2074 6865 2073 7562 7361 6d70  move the subsamp
+0000c4c0: 6c65 2073 656c 6563 7469 6f6e 2062 6566  le selection bef
+0000c4d0: 6f72 6520 7468 6520 7465 7374 2069 7320  ore the test is 
+0000c4e0: 7065 7266 6f72 6d65 642e 2054 6869 7320  performed. This 
+0000c4f0: 696e 6372 6561 7365 7320 7468 6520 6e75  increases the nu
+0000c500: 6d62 6572 206f 6620 4d6f 6e74 6520 4361  mber of Monte Ca
+0000c510: 726c 6f20 7369 6d75 6c61 7469 6f6e 7320  rlo simulations 
+0000c520: 7065 7266 6f72 6d65 6420 6475 7269 6e67  performed during
+0000c530: 2070 6f77 6572 2065 7374 696d 6174 696f   power estimatio
+0000c540: 6e2c 2061 6e64 2069 6d70 726f 7665 7320  n, and improves 
+0000c550: 7468 6520 6163 6375 7261 6379 206f 6620  the accuracy of 
+0000c560: 7468 6520 7265 7475 726e 6564 2065 7374  the returned est
+0000c570: 696d 6174 652e 2050 7265 7669 6f75 7320  imate. Previous 
+0000c580: 706f 7765 7220 6573 7469 6d61 7465 7320  power estimates 
+0000c590: 6672 6f6d 2060 7375 6273 616d 706c 655f  from `subsample_
+0000c5a0: 7061 6972 6564 5f70 6f77 6572 6020 7368  paired_power` sh
+0000c5b0: 6f75 6c64 2062 6520 6469 7372 6567 6172  ould be disregar
+0000c5c0: 6465 6420 616e 6420 7265 2d63 616c 6375  ded and re-calcu
+0000c5d0: 6c61 7465 642e 2028 5b23 3931 305d 2868  lated. ([#910](h
+0000c5e0: 7474 7073 3a2f 2f67 6974 6875 622e 636f  ttps://github.co
+0000c5f0: 6d2f 6269 6f63 6f72 652f 7363 696b 6974  m/biocore/scikit
+0000c600: 2d62 696f 2f69 7373 7565 732f 3931 3029  -bio/issues/910)
+0000c610: 290a 2a20 4669 7865 6420 6973 7375 6520  ).* Fixed issue 
+0000c620: 7768 6572 6520 6072 616e 6464 6d60 2077  where `randdm` w
+0000c630: 6173 2061 7474 656d 7074 696e 6720 746f  as attempting to
+0000c640: 2063 7265 6174 6520 6173 796d 6d65 7472   create asymmetr
+0000c650: 6963 2064 6973 7461 6e63 6520 6d61 7472  ic distance matr
+0000c660: 6963 6573 2e54 6869 7320 7761 7320 6361  ices.This was ca
+0000c670: 7573 696e 6720 616e 2065 7272 6f72 2074  using an error t
+0000c680: 6f20 6265 2072 6169 7365 6420 6279 2074  o be raised by t
+0000c690: 6865 2060 4469 7374 616e 6365 4d61 7472  he `DistanceMatr
+0000c6a0: 6978 6020 636f 6e73 7472 7563 746f 7220  ix` constructor 
+0000c6b0: 696e 7369 6465 206f 6620 7468 6520 6072  inside of the `r
+0000c6c0: 616e 6464 6d60 2066 756e 6374 696f 6e2c  anddm` function,
+0000c6d0: 2073 6f20 7468 6174 2060 7261 6e64 646d   so that `randdm
+0000c6e0: 6020 776f 756c 6420 6661 696c 2077 6865  ` would fail whe
+0000c6f0: 6e20 6174 7465 6d70 7469 6e67 2074 6f20  n attempting to 
+0000c700: 6372 6561 7465 206c 6172 6765 2064 6973  create large dis
+0000c710: 7461 6e63 6520 6d61 7472 6963 6573 2e20  tance matrices. 
+0000c720: 285b 2339 3433 5d28 6874 7470 733a 2f2f  ([#943](https://
+0000c730: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
+0000c740: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
+0000c750: 7375 6573 2f39 3433 2929 0a0a 2323 2320  sues/943))..### 
+0000c760: 4465 7072 6563 6174 6564 2066 756e 6374  Deprecated funct
+0000c770: 696f 6e61 6c69 7479 0a2a 2044 6570 7265  ionality.* Depre
+0000c780: 6361 7465 6420 6073 6b62 696f 2e75 7469  cated `skbio.uti
+0000c790: 6c2e 666c 6174 7465 6e60 2e20 5468 6973  l.flatten`. This
+0000c7a0: 2066 756e 6374 696f 6e20 7769 6c6c 2062   function will b
+0000c7b0: 6520 7265 6d6f 7665 6420 696e 2073 6369  e removed in sci
+0000c7c0: 6b69 742d 6269 6f20 302e 332e 312e 2050  kit-bio 0.3.1. P
+0000c7d0: 6c65 6173 6520 7573 6520 7374 616e 6461  lease use standa
+0000c7e0: 7264 2070 7974 686f 6e20 6c69 6272 6172  rd python librar
+0000c7f0: 7920 6675 6e63 7469 6f6e 616c 6974 790a  y functionality.
+0000c800: 6465 7363 7269 6265 6420 6865 7265 205b  described here [
+0000c810: 4d61 6b69 6e67 2061 2066 6c61 7420 6c69  Making a flat li
+0000c820: 7374 206f 7574 206f 6620 6c69 7374 7320  st out of lists 
+0000c830: 6f66 206c 6973 7473 5d28 6874 7470 3a2f  of lists](http:/
+0000c840: 2f73 7461 636b 6f76 6572 666c 6f77 2e63  /stackoverflow.c
+0000c850: 6f6d 2f61 2f39 3532 3935 322f 3336 3339  om/a/952952/3639
+0000c860: 3032 3329 2c20 5b46 6c61 7474 656e 696e  023), [Flattenin
+0000c870: 6720 6120 7368 616c 6c6f 7720 6c69 7374  g a shallow list
+0000c880: 5d28 6874 7470 3a2f 2f73 7461 636b 6f76  ](http://stackov
+0000c890: 6572 666c 6f77 2e63 6f6d 2f61 2f34 3036  erflow.com/a/406
+0000c8a0: 3139 392f 3336 3339 3032 3329 2028 5b23  199/3639023) ([#
+0000c8b0: 3833 335d 2868 7474 7073 3a2f 2f67 6974  833](https://git
+0000c8c0: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+0000c8d0: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
+0000c8e0: 732f 3833 3329 290a 2a20 4465 7072 6563  s/833)).* Deprec
+0000c8f0: 6174 6564 2060 736b 6269 6f2e 7374 6174  ated `skbio.stat
+0000c900: 732e 706f 7765 722e 626f 6f74 7374 7261  s.power.bootstra
+0000c910: 705f 706f 7765 725f 6375 7276 6560 2077  p_power_curve` w
+0000c920: 696c 6c20 6265 2072 656d 6f76 6564 2069  ill be removed i
+0000c930: 6e20 7363 696b 6974 2d62 696f 2030 2e34  n scikit-bio 0.4
+0000c940: 2e31 2e20 4974 2069 7320 6465 7072 6563  .1. It is deprec
+0000c950: 6174 6564 2069 6e20 6661 766f 7220 6f66  ated in favor of
+0000c960: 2075 7369 6e67 2060 6073 7562 7361 6d70   using ``subsamp
+0000c970: 6c65 5f70 6f77 6572 6060 206f 7220 6060  le_power`` or ``
+0000c980: 7361 6d70 6c65 5f70 6169 7265 645f 706f  sample_paired_po
+0000c990: 7765 7260 6020 746f 2063 616c 6375 6c61  wer`` to calcula
+0000c9a0: 7465 2061 2070 6f77 6572 206d 6174 7269  te a power matri
+0000c9b0: 782c 2061 6e64 2074 6865 6e20 7468 6520  x, and then the 
+0000c9c0: 7573 6520 6f66 2060 6063 6f6e 6669 6465  use of ``confide
+0000c9d0: 6e63 655f 626f 756e 6473 6060 2074 6f20  nce_bounds`` to 
+0000c9e0: 6361 6c63 756c 6174 6520 7468 6520 6176  calculate the av
+0000c9f0: 6572 6167 6520 616e 6420 636f 6e66 6964  erage and confid
+0000ca00: 656e 6365 2069 6e74 6572 7661 6c73 2e0a  ence intervals..
+0000ca10: 0a23 2323 2042 6163 6b77 6172 642d 696e  .### Backward-in
+0000ca20: 636f 6d70 6174 6962 6c65 2063 6861 6e67  compatible chang
+0000ca30: 6573 0a2a 2052 656d 6f76 6564 2074 6865  es.* Removed the
+0000ca40: 2066 6f6c 6c6f 7769 6e67 2064 6570 7265   following depre
+0000ca50: 6361 7465 6420 6675 6e63 7469 6f6e 616c  cated functional
+0000ca60: 6974 793a 0a20 2020 202d 2060 736b 6269  ity:.    - `skbi
+0000ca70: 6f2e 7061 7273 6560 2073 7562 7061 636b  o.parse` subpack
+0000ca80: 6167 652c 2069 6e63 6c75 6469 6e67 2060  age, including `
+0000ca90: 5365 7175 656e 6365 4974 6572 6174 6f72  SequenceIterator
+0000caa0: 602c 2060 4661 7374 6149 7465 7261 746f  `, `FastaIterato
+0000cab0: 7260 2c20 6046 6173 7471 4974 6572 6174  r`, `FastqIterat
+0000cac0: 6f72 602c 2060 6c6f 6164 602c 2060 7061  or`, `load`, `pa
+0000cad0: 7273 655f 6661 7374 6160 2c20 6070 6172  rse_fasta`, `par
+0000cae0: 7365 5f66 6173 7471 602c 2060 7061 7273  se_fastq`, `pars
+0000caf0: 655f 7175 616c 602c 2060 7772 6974 655f  e_qual`, `write_
+0000cb00: 636c 7573 7461 6c60 2c20 6070 6172 7365  clustal`, `parse
+0000cb10: 5f63 6c75 7374 616c 602c 2061 6e64 2060  _clustal`, and `
+0000cb20: 4661 7374 7150 6172 7365 4572 726f 7260  FastqParseError`
+0000cb30: 3b20 706c 6561 7365 2075 7365 2060 736b  ; please use `sk
+0000cb40: 6269 6f2e 696f 6020 696e 7374 6561 642e  bio.io` instead.
+0000cb50: 0a20 2020 202d 2060 736b 6269 6f2e 666f  .    - `skbio.fo
+0000cb60: 726d 6174 6020 7375 6270 6163 6b61 6765  rmat` subpackage
+0000cb70: 2c20 696e 636c 7564 696e 6720 6066 6173  , including `fas
+0000cb80: 7461 5f66 726f 6d5f 7365 7175 656e 6365  ta_from_sequence
+0000cb90: 602c 2060 6661 7374 615f 6672 6f6d 5f61  `, `fasta_from_a
+0000cba0: 6c69 676e 6d65 6e74 602c 2061 6e64 2060  lignment`, and `
+0000cbb0: 666f 726d 6174 5f66 6173 7471 5f72 6563  format_fastq_rec
+0000cbc0: 6f72 6460 3b20 706c 6561 7365 2075 7365  ord`; please use
+0000cbd0: 2060 736b 6269 6f2e 696f 6020 696e 7374   `skbio.io` inst
+0000cbe0: 6561 642e 0a20 2020 202d 2060 736b 6269  ead..    - `skbi
+0000cbf0: 6f2e 616c 6967 6e6d 656e 742e 5365 7175  o.alignment.Sequ
+0000cc00: 656e 6365 436f 6c6c 6563 7469 6f6e 2e69  enceCollection.i
+0000cc10: 6e74 5f6d 6170 603b 2070 6c65 6173 6520  nt_map`; please 
+0000cc20: 7573 6520 6053 6571 7565 6e63 6543 6f6c  use `SequenceCol
+0000cc30: 6c65 6374 696f 6e2e 7570 6461 7465 5f69  lection.update_i
+0000cc40: 6473 6020 696e 7374 6561 642e 0a20 2020  ds` instead..   
+0000cc50: 202d 2060 736b 6269 6f2e 616c 6967 6e6d   - `skbio.alignm
+0000cc60: 656e 742e 5365 7175 656e 6365 436f 6c6c  ent.SequenceColl
+0000cc70: 6563 7469 6f6e 6020 6d65 7468 6f64 7320  ection` methods 
+0000cc80: 6074 6f5f 6661 7374 6160 2061 6e64 2060  `to_fasta` and `
+0000cc90: 746f 4661 7374 6160 3b20 706c 6561 7365  toFasta`; please
+0000cca0: 2075 7365 2060 5365 7175 656e 6365 436f   use `SequenceCo
+0000ccb0: 6c6c 6563 7469 6f6e 2e77 7269 7465 6020  llection.write` 
+0000ccc0: 696e 7374 6561 642e 0a20 2020 202d 2060  instead..    - `
+0000ccd0: 636f 6e73 7472 7563 746f 7260 2070 6172  constructor` par
+0000cce0: 616d 6574 6572 2069 6e20 6073 6b62 696f  ameter in `skbio
+0000ccf0: 2e61 6c69 676e 6d65 6e74 2e41 6c69 676e  .alignment.Align
+0000cd00: 6d65 6e74 2e6d 616a 6f72 6974 795f 636f  ment.majority_co
+0000cd10: 6e73 656e 7375 7360 3b20 706c 6561 7365  nsensus`; please
+0000cd20: 2063 6f6e 7665 7274 2072 6574 7572 6e65   convert returne
+0000cd30: 6420 6269 6f6c 6f67 6963 616c 2073 6571  d biological seq
+0000cd40: 7565 6e63 6520 6f62 6a65 6374 206d 616e  uence object man
+0000cd50: 7561 6c6c 7920 6173 2064 6573 6972 6564  ually as desired
+0000cd60: 2028 652e 672e 2c20 6073 7472 2873 6571   (e.g., `str(seq
+0000cd70: 2960 292e 0a20 2020 202d 2060 736b 6269  )`)..    - `skbi
+0000cd80: 6f2e 616c 6967 6e6d 656e 742e 416c 6967  o.alignment.Alig
+0000cd90: 6e6d 656e 742e 746f 5f70 6879 6c69 7060  nment.to_phylip`
+0000cda0: 3b20 706c 6561 7365 2075 7365 2060 416c  ; please use `Al
+0000cdb0: 6967 6e6d 656e 742e 7772 6974 6560 2069  ignment.write` i
+0000cdc0: 6e73 7465 6164 2e0a 2020 2020 2d20 6073  nstead..    - `s
+0000cdd0: 6b62 696f 2e73 6571 7565 6e63 652e 4269  kbio.sequence.Bi
+0000cde0: 6f6c 6f67 6963 616c 5365 7175 656e 6365  ologicalSequence
+0000cdf0: 2e74 6f5f 6661 7374 6160 3b20 706c 6561  .to_fasta`; plea
+0000ce00: 7365 2075 7365 2060 4269 6f6c 6f67 6963  se use `Biologic
+0000ce10: 616c 5365 7175 656e 6365 2e77 7269 7465  alSequence.write
+0000ce20: 6020 696e 7374 6561 642e 0a20 2020 202d  ` instead..    -
+0000ce30: 2060 736b 6269 6f2e 7472 6565 2e54 7265   `skbio.tree.Tre
+0000ce40: 654e 6f64 6560 206d 6574 686f 6473 2060  eNode` methods `
+0000ce50: 6672 6f6d 5f6e 6577 6963 6b60 2c20 6066  from_newick`, `f
+0000ce60: 726f 6d5f 6669 6c65 602c 2061 6e64 2060  rom_file`, and `
+0000ce70: 746f 5f6e 6577 6963 6b60 3b20 706c 6561  to_newick`; plea
+0000ce80: 7365 2075 7365 2060 5472 6565 4e6f 6465  se use `TreeNode
+0000ce90: 2e72 6561 6460 2061 6e64 2060 5472 6565  .read` and `Tree
+0000cea0: 4e6f 6465 2e77 7269 7465 6020 696e 7374  Node.write` inst
+0000ceb0: 6561 642e 0a20 2020 202d 2060 736b 6269  ead..    - `skbi
+0000cec0: 6f2e 7374 6174 732e 6469 7374 616e 6365  o.stats.distance
+0000ced0: 2e44 6973 7369 6d69 6c61 7269 7479 4d61  .DissimilarityMa
+0000cee0: 7472 6978 6020 6d65 7468 6f64 7320 6066  trix` methods `f
+0000cef0: 726f 6d5f 6669 6c65 6020 616e 6420 6074  rom_file` and `t
+0000cf00: 6f5f 6669 6c65 603b 2070 6c65 6173 6520  o_file`; please 
+0000cf10: 7573 6520 6044 6973 7369 6d69 6c61 7269  use `Dissimilari
+0000cf20: 7479 4d61 7472 6978 2e72 6561 6460 2061  tyMatrix.read` a
+0000cf30: 6e64 2060 4469 7373 696d 696c 6172 6974  nd `Dissimilarit
+0000cf40: 794d 6174 7269 782e 7772 6974 6560 2069  yMatrix.write` i
+0000cf50: 6e73 7465 6164 2e0a 2020 2020 2d20 6073  nstead..    - `s
+0000cf60: 6b62 696f 2e73 7461 7473 2e6f 7264 696e  kbio.stats.ordin
+0000cf70: 6174 696f 6e2e 4f72 6469 6e61 7469 6f6e  ation.Ordination
+0000cf80: 5265 7375 6c74 7360 206d 6574 686f 6473  Results` methods
+0000cf90: 2060 6672 6f6d 5f66 696c 6560 2061 6e64   `from_file` and
+0000cfa0: 2060 746f 5f66 696c 6560 3b20 706c 6561   `to_file`; plea
+0000cfb0: 7365 2075 7365 2060 4f72 6469 6e61 7469  se use `Ordinati
+0000cfc0: 6f6e 5265 7375 6c74 732e 7265 6164 6020  onResults.read` 
+0000cfd0: 616e 6420 604f 7264 696e 6174 696f 6e52  and `OrdinationR
+0000cfe0: 6573 756c 7473 2e77 7269 7465 6020 696e  esults.write` in
+0000cff0: 7374 6561 642e 0a20 2020 202d 2060 736b  stead..    - `sk
+0000d000: 6269 6f2e 7374 6174 732e 705f 7661 6c75  bio.stats.p_valu
+0000d010: 655f 746f 5f73 7472 603b 2074 6865 7265  e_to_str`; there
+0000d020: 2069 7320 6e6f 2072 6570 6c61 6365 6d65   is no replaceme
+0000d030: 6e74 2e0a 2020 2020 2d20 6073 6b62 696f  nt..    - `skbio
+0000d040: 2e73 7461 7473 2e73 7562 7361 6d70 6c65  .stats.subsample
+0000d050: 603b 2070 6c65 6173 6520 7573 6520 6073  `; please use `s
+0000d060: 6b62 696f 2e73 7461 7473 2e73 7562 7361  kbio.stats.subsa
+0000d070: 6d70 6c65 5f63 6f75 6e74 7360 2069 6e73  mple_counts` ins
+0000d080: 7465 6164 2e0a 2020 2020 2d20 6073 6b62  tead..    - `skb
+0000d090: 696f 2e73 7461 7473 2e64 6973 7461 6e63  io.stats.distanc
+0000d0a0: 652e 414e 4f53 494d 603b 2070 6c65 6173  e.ANOSIM`; pleas
+0000d0b0: 6520 7573 6520 6073 6b62 696f 2e73 7461  e use `skbio.sta
+0000d0c0: 7473 2e64 6973 7461 6e63 652e 616e 6f73  ts.distance.anos
+0000d0d0: 696d 6020 696e 7374 6561 642e 0a20 2020  im` instead..   
+0000d0e0: 202d 2060 736b 6269 6f2e 7374 6174 732e   - `skbio.stats.
+0000d0f0: 6469 7374 616e 6365 2e50 4552 4d41 4e4f  distance.PERMANO
+0000d100: 5641 603b 2070 6c65 6173 6520 7573 6520  VA`; please use 
+0000d110: 6073 6b62 696f 2e73 7461 7473 2e64 6973  `skbio.stats.dis
+0000d120: 7461 6e63 652e 7065 726d 616e 6f76 6160  tance.permanova`
+0000d130: 2069 6e73 7465 6164 2e0a 2020 2020 2d20   instead..    - 
+0000d140: 6073 6b62 696f 2e73 7461 7473 2e64 6973  `skbio.stats.dis
+0000d150: 7461 6e63 652e 4361 7465 676f 7269 6361  tance.Categorica
+0000d160: 6c53 7461 7473 5265 7375 6c74 7360 3b20  lStatsResults`; 
+0000d170: 7468 6572 6520 6973 206e 6f20 7265 706c  there is no repl
+0000d180: 6163 656d 656e 742c 2070 6c65 6173 6520  acement, please 
+0000d190: 7573 6520 6073 6b62 696f 2e73 7461 7473  use `skbio.stats
+0000d1a0: 2e64 6973 7461 6e63 652e 616e 6f73 696d  .distance.anosim
+0000d1b0: 6020 6f72 2060 736b 6269 6f2e 7374 6174  ` or `skbio.stat
+0000d1c0: 732e 6469 7374 616e 6365 2e70 6572 6d61  s.distance.perma
+0000d1d0: 6e6f 7661 602c 2077 6869 6368 2077 696c  nova`, which wil
+0000d1e0: 6c20 7265 7475 726e 2061 2060 7061 6e64  l return a `pand
+0000d1f0: 6173 2e53 6572 6965 7360 206f 626a 6563  as.Series` objec
+0000d200: 742e 0a2a 2060 736b 6269 6f2e 616c 6967  t..* `skbio.alig
+0000d210: 6e6d 656e 742e 416c 6967 6e6d 656e 742e  nment.Alignment.
+0000d220: 6d61 6a6f 7269 7479 5f63 6f6e 7365 6e73  majority_consens
+0000d230: 7573 6020 6e6f 7720 7265 7475 726e 7320  us` now returns 
+0000d240: 6042 696f 6c6f 6769 6361 6c53 6571 7565  `BiologicalSeque
+0000d250: 6e63 6528 2727 2960 2069 6620 7468 6520  nce('')` if the 
+0000d260: 616c 6967 6e6d 656e 7420 6973 2065 6d70  alignment is emp
+0000d270: 7479 2e20 5072 6576 696f 7573 6c79 2c20  ty. Previously, 
+0000d280: 6027 2760 2077 6173 2072 6574 7572 6e65  `''` was returne
+0000d290: 642e 0a2a 2060 6d69 6e5f 6f62 7365 7276  d..* `min_observ
+0000d2a0: 6174 696f 6e73 6020 7761 7320 7265 6d6f  ations` was remo
+0000d2b0: 7665 6420 6672 6f6d 2060 736b 6269 6f2e  ved from `skbio.
+0000d2c0: 7374 6174 732e 706f 7765 722e 7375 6273  stats.power.subs
+0000d2d0: 616d 706c 655f 706f 7765 7260 2061 6e64  ample_power` and
+0000d2e0: 2060 736b 6269 6f2e 7374 6174 732e 706f   `skbio.stats.po
+0000d2f0: 7765 722e 7375 6273 616d 706c 655f 7061  wer.subsample_pa
+0000d300: 6972 6564 5f70 6f77 6572 602e 2054 6865  ired_power`. The
+0000d310: 206d 696e 696d 756d 206e 756d 6265 7220   minimum number 
+0000d320: 6f66 2073 616d 706c 6573 2066 6f72 2073  of samples for s
+0000d330: 7562 7361 6d70 6c69 6e67 2064 6570 656e  ubsampling depen
+0000d340: 6473 206f 6e20 7468 6520 6461 7461 2073  ds on the data s
+0000d350: 6574 2061 6e64 2073 7461 7469 7374 6963  et and statistic
+0000d360: 616c 2074 6573 7473 2e20 4861 7669 6e67  al tests. Having
+0000d370: 2061 2064 6566 6175 6c74 2070 6172 616d   a default param
+0000d380: 6574 6572 2074 6f20 7365 7420 756e 6e65  eter to set unne
+0000d390: 6365 7373 6172 7920 6c69 6d69 7461 7469  cessary limitati
+0000d3a0: 6f6e 7320 6f6e 2074 6865 2074 6563 686e  ons on the techn
+0000d3b0: 6971 7565 2e0a 0a23 2323 204d 6973 6365  ique...### Misce
+0000d3c0: 6c6c 616e 656f 7573 0a2a 2043 6861 6e67  llaneous.* Chang
+0000d3d0: 6564 2074 6573 7469 6e67 2070 726f 6365  ed testing proce
+0000d3e0: 6475 7265 730a 2020 2020 2d20 4465 7665  dures.    - Deve
+0000d3f0: 6c6f 7065 7273 2073 686f 756c 6420 6e6f  lopers should no
+0000d400: 7720 7573 6520 606d 616b 6520 7465 7374  w use `make test
+0000d410: 600a 2020 2020 2d20 5573 6572 7320 6361  `.    - Users ca
+0000d420: 6e20 7573 6520 6070 7974 686f 6e20 2d6d  n use `python -m
+0000d430: 2073 6b62 696f 2e74 6573 7460 0a20 2020   skbio.test`.   
+0000d440: 202d 2041 6464 6564 2060 736b 6269 6f2e   - Added `skbio.
+0000d450: 7574 696c 2e5f 7465 7374 696e 672e 5465  util._testing.Te
+0000d460: 7374 5275 6e6e 6572 6020 2861 7661 696c  stRunner` (avail
+0000d470: 6162 6c65 2074 6872 6f75 6768 2060 736b  able through `sk
+0000d480: 6269 6f2e 7574 696c 2e54 6573 7452 756e  bio.util.TestRun
+0000d490: 6e65 7260 292e 2055 7365 6420 746f 2070  ner`). Used to p
+0000d4a0: 726f 7669 6465 2061 2060 7465 7374 6020  rovide a `test` 
+0000d4b0: 6d65 7468 6f64 2066 6f72 2065 6163 6820  method for each 
+0000d4c0: 6d6f 6475 6c65 2069 6e69 7420 6669 6c65  module init file
+0000d4d0: 2e20 5468 6973 2063 6c61 7373 2072 6570  . This class rep
+0000d4e0: 7265 7365 6e74 7320 6120 756e 6966 6965  resents a unifie
+0000d4f0: 6420 7465 7374 696e 6720 7061 7468 2077  d testing path w
+0000d500: 6869 6368 2077 7261 7073 2061 6c6c 2060  hich wraps all `
+0000d510: 736b 6269 6f60 2074 6573 7469 6e67 2066  skbio` testing f
+0000d520: 756e 6374 696f 6e61 6c69 7479 2e0a 2020  unctionality..  
+0000d530: 2020 2d20 4175 746f 6465 7465 6374 2050    - Autodetect P
+0000d540: 7974 686f 6e20 7665 7273 696f 6e20 616e  ython version an
+0000d550: 6420 6469 7361 626c 6520 646f 6374 6573  d disable doctes
+0000d560: 7473 2066 6f72 2050 7974 686f 6e20 332e  ts for Python 3.
+0000d570: 0a2a 2060 6e75 6d70 7960 2069 7320 6e6f  .* `numpy` is no
+0000d580: 206c 6f6e 6765 7220 7265 7175 6972 6564   longer required
+0000d590: 2074 6f20 6265 2069 6e73 7461 6c6c 6564   to be installed
+0000d5a0: 2062 6566 6f72 6520 696e 7374 616c 6c69   before installi
+0000d5b0: 6e67 2073 6369 6b69 742d 6269 6f21 0a2a  ng scikit-bio!.*
+0000d5c0: 2055 7067 7261 6465 6420 6368 6563 6b6c   Upgraded checkl
+0000d5d0: 6973 742e 7079 2074 6f20 6368 6563 6b20  ist.py to check 
+0000d5e0: 736f 7572 6365 2066 696c 6573 206e 6f6e  source files non
+0000d5f0: 2d63 6f6e 666f 726d 696e 6720 746f 205b  -conforming to [
+0000d600: 6e65 7720 6865 6164 6572 2073 7479 6c65  new header style
+0000d610: 5d28 6874 7470 3a2f 2f73 6369 6b69 742d  ](http://scikit-
+0000d620: 6269 6f2e 6f72 672f 646f 6373 2f6c 6174  bio.org/docs/lat
+0000d630: 6573 742f 6465 7665 6c6f 706d 656e 742f  est/development/
+0000d640: 6e65 775f 6d6f 6475 6c65 2e68 746d 6c29  new_module.html)
+0000d650: 2e20 285b 2338 3535 5d28 6874 7470 733a  . ([#855](https:
+0000d660: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
+0000d670: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
+0000d680: 6973 7375 6573 2f38 3535 2929 0a2a 2055  issues/855)).* U
+0000d690: 7064 6174 6564 2074 6f20 7573 6520 606e  pdated to use `n
+0000d6a0: 6174 736f 7274 6020 3e3d 2034 2e30 2e30  atsort` >= 4.0.0
+0000d6b0: 2e0a 2a20 5468 6520 6d65 7468 6f64 206f  ..* The method o
+0000d6c0: 6620 7375 6273 616d 706c 696e 6720 7761  f subsampling wa
+0000d6d0: 7320 6368 616e 6765 6420 666f 7220 6060  s changed for ``
+0000d6e0: 736b 6269 6f2e 7374 6174 732e 706f 7765  skbio.stats.powe
+0000d6f0: 722e 7375 6273 616d 706c 655f 7061 6972  r.subsample_pair
+0000d700: 6564 5f70 6f77 6572 6060 2e20 5261 7468  ed_power``. Rath
+0000d710: 6572 2074 6861 6e20 6472 6177 696e 6720  er than drawing 
+0000d720: 6120 7061 6972 6564 2073 616d 706c 6520  a paired sample 
+0000d730: 666f 7220 7468 6520 7275 6e20 616e 6420  for the run and 
+0000d740: 7468 656e 2073 7562 7361 6d70 6c69 6e67  then subsampling
+0000d750: 2066 6f72 2065 6163 6820 636f 756e 742c   for each count,
+0000d760: 2074 6865 2073 7562 7361 6d70 6c65 2069   the subsample i
+0000d770: 7320 6e6f 7720 6472 6177 6e20 666f 7220  s now drawn for 
+0000d780: 6561 6368 2073 616d 706c 6520 616e 6420  each sample and 
+0000d790: 6561 6368 2072 756e 2e20 496e 2074 6573  each run. In tes
+0000d7a0: 7420 6461 7461 2c20 7468 6973 2064 6964  t data, this did
+0000d7b0: 206e 6f74 2073 6967 6e69 6669 6361 6e74   not significant
+0000d7c0: 6c79 2061 6c74 6572 2074 6865 2070 6f77  ly alter the pow
+0000d7d0: 6572 2072 6573 756c 7473 2e0a 2a20 6368  er results..* ch
+0000d7e0: 6563 6b6c 6973 742e 7079 206e 6f77 2065  ecklist.py now e
+0000d7f0: 6e66 6f72 6365 7320 605f 5f66 7574 7572  nforces `__futur
+0000d800: 655f 5f60 2069 6d70 6f72 7473 2069 6e20  e__` imports in 
+0000d810: 2e70 7920 6669 6c65 732e 0a0a 2323 2056  .py files...## V
+0000d820: 6572 7369 6f6e 2030 2e32 2e33 2028 3230  ersion 0.2.3 (20
+0000d830: 3135 2d30 322d 3133 290a 0a23 2323 2046  15-02-13)..### F
+0000d840: 6561 7475 7265 730a 2a20 4d6f 6469 6669  eatures.* Modifi
+0000d850: 6564 2060 6073 6b62 696f 2e73 7461 7473  ed ``skbio.stats
+0000d860: 2e64 6973 7461 6e63 652e 7077 6d61 6e74  .distance.pwmant
+0000d870: 656c 6060 2074 6f20 6163 6365 7074 2061  el`` to accept a
+0000d880: 206c 6973 7420 6f66 2066 696c 6570 6174   list of filepat
+0000d890: 6873 2e20 5468 6973 2069 7320 7573 6566  hs. This is usef
+0000d8a0: 756c 2061 7320 6974 2061 6c6c 6f77 7320  ul as it allows 
+0000d8b0: 666f 7220 6120 736d 616c 6c65 7220 616d  for a smaller am
+0000d8c0: 6f75 6e74 206f 6620 6d65 6d6f 7279 2063  ount of memory c
+0000d8d0: 6f6e 7375 6d70 7469 6f6e 2061 7320 6974  onsumption as it
+0000d8e0: 206f 6e6c 7920 6c6f 6164 7320 7477 6f20   only loads two 
+0000d8f0: 6d61 7472 6963 6573 2061 7420 6120 7469  matrices at a ti
+0000d900: 6d65 2061 7320 6f70 706f 7365 6420 746f  me as opposed to
+0000d910: 2072 6571 7569 7269 6e67 2074 6861 7420   requiring that 
+0000d920: 616c 6c20 6469 7374 616e 6365 206d 6174  all distance mat
+0000d930: 7269 6365 7320 6172 6520 6c6f 6164 6564  rices are loaded
+0000d940: 2069 6e74 6f20 6d65 6d6f 7279 2e0a 2a20   into memory..* 
+0000d950: 4164 6465 6420 6060 736b 6269 6f2e 7574  Added ``skbio.ut
+0000d960: 696c 2e66 696e 645f 6475 706c 6963 6174  il.find_duplicat
+0000d970: 6573 6060 2066 6f72 2066 696e 6469 6e67  es`` for finding
+0000d980: 2064 7570 6c69 6361 7465 2065 6c65 6d65   duplicate eleme
+0000d990: 6e74 7320 696e 2061 6e20 6974 6572 6162  nts in an iterab
+0000d9a0: 6c65 2e0a 0a23 2323 2042 7567 2066 6978  le...### Bug fix
+0000d9b0: 6573 0a2a 2046 6978 6564 2066 6c6f 6174  es.* Fixed float
+0000d9c0: 696e 6720 706f 696e 7420 7072 6563 6973  ing point precis
+0000d9d0: 696f 6e20 6275 6773 2069 6e20 6060 416c  ion bugs in ``Al
+0000d9e0: 6967 6e6d 656e 742e 706f 7369 7469 6f6e  ignment.position
+0000d9f0: 5f66 7265 7175 656e 6369 6573 6060 2c20  _frequencies``, 
+0000da00: 6060 416c 6967 6e6d 656e 742e 706f 7369  ``Alignment.posi
+0000da10: 7469 6f6e 5f65 6e74 726f 7069 6573 6060  tion_entropies``
+0000da20: 2c20 6060 416c 6967 6e6d 656e 742e 6f6d  , ``Alignment.om
+0000da30: 6974 5f67 6170 5f70 6f73 6974 696f 6e73  it_gap_positions
+0000da40: 6060 2c20 6060 416c 6967 6e6d 656e 742e  ``, ``Alignment.
+0000da50: 6f6d 6974 5f67 6170 5f73 6571 7565 6e63  omit_gap_sequenc
+0000da60: 6573 6060 2c20 6060 4269 6f6c 6f67 6963  es``, ``Biologic
+0000da70: 616c 5365 7175 656e 6365 2e6b 5f77 6f72  alSequence.k_wor
+0000da80: 645f 6672 6571 7565 6e63 6965 7360 602c  d_frequencies``,
+0000da90: 2061 6e64 2060 6053 6571 7565 6e63 6543   and ``SequenceC
+0000daa0: 6f6c 6c65 6374 696f 6e2e 6b5f 776f 7264  ollection.k_word
+0000dab0: 5f66 7265 7175 656e 6369 6573 6060 2028  _frequencies`` (
+0000dac0: 5b23 3830 315d 2868 7474 7073 3a2f 2f67  [#801](https://g
+0000dad0: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
+0000dae0: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
+0000daf0: 7565 732f 3830 3129 292e 0a0a 2323 2320  ues/801))...### 
+0000db00: 4261 636b 7761 7264 2d69 6e63 6f6d 7061  Backward-incompa
+0000db10: 7469 626c 6520 6368 616e 6765 730a 2a20  tible changes.* 
+0000db20: 5265 6d6f 7665 6420 6060 6665 6174 7572  Removed ``featur
+0000db30: 655f 7479 7065 7360 6020 6174 7472 6962  e_types`` attrib
+0000db40: 7574 6520 6672 6f6d 2060 6042 696f 6c6f  ute from ``Biolo
+0000db50: 6769 6361 6c53 6571 7565 6e63 6560 6020  gicalSequence`` 
+0000db60: 616e 6420 616c 6c20 7375 6263 6c61 7373  and all subclass
+0000db70: 6573 2028 5b23 3739 375d 2868 7474 7073  es ([#797](https
+0000db80: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
+0000db90: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
+0000dba0: 2f70 756c 6c2f 3739 3729 292e 0a2a 2052  /pull/797))..* R
+0000dbb0: 656d 6f76 6564 2060 6066 696e 645f 6665  emoved ``find_fe
+0000dbc0: 6174 7572 6573 6060 206d 6574 686f 6420  atures`` method 
+0000dbd0: 6672 6f6d 2060 6042 696f 6c6f 6769 6361  from ``Biologica
+0000dbe0: 6c53 6571 7565 6e63 6560 6020 616e 6420  lSequence`` and 
+0000dbf0: 6060 5072 6f74 6569 6e53 6571 7565 6e63  ``ProteinSequenc
+0000dc00: 6560 6020 285b 2337 3937 5d28 6874 7470  e`` ([#797](http
+0000dc10: 733a 2f2f 6769 7468 7562 2e63 6f6d 2f62  s://github.com/b
+0000dc20: 696f 636f 7265 2f73 6369 6b69 742d 6269  iocore/scikit-bi
+0000dc30: 6f2f 7075 6c6c 2f37 3937 2929 2e0a 2a20  o/pull/797))..* 
+0000dc40: 6060 4269 6f6c 6f67 6963 616c 5365 7175  ``BiologicalSequ
+0000dc50: 656e 6365 2e6b 5f77 6f72 645f 6672 6571  ence.k_word_freq
+0000dc60: 7565 6e63 6965 7360 6020 6e6f 7720 7265  uencies`` now re
+0000dc70: 7475 726e 7320 6120 6060 636f 6c6c 6563  turns a ``collec
+0000dc80: 7469 6f6e 732e 6465 6661 756c 7464 6963  tions.defaultdic
+0000dc90: 7460 6020 6f66 2074 7970 6520 6060 666c  t`` of type ``fl
+0000dca0: 6f61 7460 6020 696e 7374 6561 6420 6f66  oat`` instead of
+0000dcb0: 2074 7970 6520 6060 696e 7460 602e 2054   type ``int``. T
+0000dcc0: 6869 7320 6f6e 6c79 2061 6666 6563 7473  his only affects
+0000dcd0: 2074 6865 2022 6465 6661 756c 7422 2063   the "default" c
+0000dce0: 6173 652c 2077 6865 6e20 6120 6b65 7920  ase, when a key 
+0000dcf0: 6973 6e27 7420 7072 6573 656e 7420 696e  isn't present in
+0000dd00: 2074 6865 2064 6963 7469 6f6e 6172 792e   the dictionary.
+0000dd10: 2050 7265 7669 6f75 7320 6265 6861 7669   Previous behavi
+0000dd20: 6f72 2077 6f75 6c64 2072 6574 7572 6e20  or would return 
+0000dd30: 6060 3060 6020 6173 2061 6e20 6060 696e  ``0`` as an ``in
+0000dd40: 7460 602c 2077 6869 6c65 2074 6865 206e  t``, while the n
+0000dd50: 6577 2062 6568 6176 696f 7220 6973 2074  ew behavior is t
+0000dd60: 6f20 7265 7475 726e 2060 6030 2e30 6060  o return ``0.0``
+0000dd70: 2061 7320 6120 6060 666c 6f61 7460 602e   as a ``float``.
+0000dd80: 2054 6869 7320 6368 616e 6765 2061 6c73   This change als
+0000dd90: 6f20 6166 6665 6374 7320 7468 6520 6060  o affects the ``
+0000dda0: 6465 6661 756c 7464 6963 7460 6073 2074  defaultdict``s t
+0000ddb0: 6861 7420 6172 6520 7265 7475 726e 6564  hat are returned
+0000ddc0: 2062 7920 6060 5365 7175 656e 6365 436f   by ``SequenceCo
+0000ddd0: 6c6c 6563 7469 6f6e 2e6b 5f77 6f72 645f  llection.k_word_
+0000dde0: 6672 6571 7565 6e63 6965 7360 602e 0a0a  frequencies``...
+0000ddf0: 2323 2320 4d69 7363 656c 6c61 6e65 6f75  ### Miscellaneou
+0000de00: 730a 2a20 6060 4469 7373 696d 696c 6172  s.* ``Dissimilar
+0000de10: 6974 794d 6174 7269 7860 6020 616e 6420  ityMatrix`` and 
+0000de20: 6060 4469 7374 616e 6365 4d61 7472 6978  ``DistanceMatrix
+0000de30: 6060 206e 6f77 2072 6570 6f72 7420 6475  `` now report du
+0000de40: 706c 6963 6174 6520 4944 7320 696e 2074  plicate IDs in t
+0000de50: 6865 2060 6044 6973 7369 6d69 6c61 7269  he ``Dissimilari
+0000de60: 7479 4d61 7472 6978 4572 726f 7260 6020  tyMatrixError`` 
+0000de70: 6d65 7373 6167 6520 7468 6174 2063 616e  message that can
+0000de80: 2062 6520 7261 6973 6564 2064 7572 696e   be raised durin
+0000de90: 6720 7661 6c69 6461 7469 6f6e 2e0a 0a23  g validation...#
+0000dea0: 2320 5665 7273 696f 6e20 302e 322e 3220  # Version 0.2.2 
+0000deb0: 2832 3031 342d 3132 2d30 3429 0a0a 2323  (2014-12-04)..##
+0000dec0: 2320 4665 6174 7572 6573 0a2a 2041 6464  # Features.* Add
+0000ded0: 6564 2060 6070 6c6f 7460 6020 6d65 7468  ed ``plot`` meth
+0000dee0: 6f64 2074 6f20 6060 736b 6269 6f2e 7374  od to ``skbio.st
+0000def0: 6174 732e 6469 7374 616e 6365 2e44 6973  ats.distance.Dis
+0000df00: 7369 6d69 6c61 7269 7479 4d61 7472 6978  similarityMatrix
+0000df10: 6060 2066 6f72 2063 7265 6174 696e 6720  `` for creating 
+0000df20: 6261 7369 6320 6865 6174 6d61 7073 206f  basic heatmaps o
+0000df30: 6620 6120 6469 7373 696d 696c 6172 6974  f a dissimilarit
+0000df40: 792f 6469 7374 616e 6365 206d 6174 7269  y/distance matri
+0000df50: 7820 2873 6565 205b 2336 3834 5d28 6874  x (see [#684](ht
+0000df60: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+0000df70: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
+0000df80: 6269 6f2f 6973 7375 6573 2f36 3834 2929  bio/issues/684))
+0000df90: 2e20 416c 736f 2061 6464 6564 2020 6060  . Also added  ``
+0000dfa0: 5f72 6570 725f 706e 675f 6060 2061 6e64  _repr_png_`` and
+0000dfb0: 2060 605f 7265 7072 5f73 7667 5f60 6020   ``_repr_svg_`` 
+0000dfc0: 6d65 7468 6f64 7320 666f 7220 6175 746f  methods for auto
+0000dfd0: 6d61 7469 6320 6469 7370 6c61 7920 696e  matic display in
+0000dfe0: 2074 6865 2049 5079 7468 6f6e 204e 6f74   the IPython Not
+0000dff0: 6562 6f6f 6b2c 2077 6974 6820 6060 706e  ebook, with ``pn
+0000e000: 6760 6020 616e 6420 6060 7376 6760 6020  g`` and ``svg`` 
+0000e010: 7072 6f70 6572 7469 6573 2066 6f72 2064  properties for d
+0000e020: 6972 6563 7420 6163 6365 7373 2e0a 2a20  irect access..* 
+0000e030: 4164 6465 6420 605f 5f73 7472 5f5f 6020  Added `__str__` 
+0000e040: 6d65 7468 6f64 2074 6f20 6073 6b62 696f  method to `skbio
+0000e050: 2e73 7461 7473 2e6f 7264 696e 6174 696f  .stats.ordinatio
+0000e060: 6e2e 4f72 6469 6e61 7469 6f6e 5265 7375  n.OrdinationResu
+0000e070: 6c74 7360 2e0a 2a20 4164 6465 6420 6060  lts`..* Added ``
+0000e080: 736b 6269 6f2e 7374 6174 732e 6469 7374  skbio.stats.dist
+0000e090: 616e 6365 2e61 6e6f 7369 6d60 6020 616e  ance.anosim`` an
+0000e0a0: 6420 6060 736b 6269 6f2e 7374 6174 732e  d ``skbio.stats.
+0000e0b0: 6469 7374 616e 6365 2e70 6572 6d61 6e6f  distance.permano
+0000e0c0: 7661 6060 2066 756e 6374 696f 6e73 2c20  va`` functions, 
+0000e0d0: 7768 6963 6820 7265 706c 6163 6520 7468  which replace th
+0000e0e0: 6520 6060 736b 6269 6f2e 7374 6174 732e  e ``skbio.stats.
+0000e0f0: 6469 7374 616e 6365 2e41 4e4f 5349 4d60  distance.ANOSIM`
+0000e100: 6020 616e 6420 6060 736b 6269 6f2e 7374  ` and ``skbio.st
+0000e110: 6174 732e 6469 7374 616e 6365 2e50 4552  ats.distance.PER
+0000e120: 4d41 4e4f 5641 6060 2063 6c61 7373 6573  MANOVA`` classes
+0000e130: 2e20 5468 6573 6520 6e65 7720 6675 6e63  . These new func
+0000e140: 7469 6f6e 7320 7072 6f76 6964 6520 7369  tions provide si
+0000e150: 6d70 6c65 7220 7072 6f63 6564 7572 616c  mpler procedural
+0000e160: 2069 6e74 6572 6661 6365 7320 746f 2072   interfaces to r
+0000e170: 756e 6e69 6e67 2074 6865 7365 2073 7461  unning these sta
+0000e180: 7469 7374 6963 616c 206d 6574 686f 6473  tistical methods
+0000e190: 2e20 5468 6579 2061 6c73 6f20 7072 6f76  . They also prov
+0000e1a0: 6964 6520 6d6f 7265 2063 6f6e 7665 6e69  ide more conveni
+0000e1b0: 656e 7420 6163 6365 7373 2074 6f20 7265  ent access to re
+0000e1c0: 7375 6c74 7320 6279 2072 6574 7572 6e69  sults by returni
+0000e1d0: 6e67 2061 2060 6070 616e 6461 732e 5365  ng a ``pandas.Se
+0000e1e0: 7269 6573 6060 2069 6e73 7465 6164 206f  ries`` instead o
+0000e1f0: 6620 6120 6060 4361 7465 676f 7269 6361  f a ``Categorica
+0000e200: 6c53 7461 7473 5265 7375 6c74 7360 6020  lStatsResults`` 
+0000e210: 6f62 6a65 6374 2e20 5468 6573 6520 6675  object. These fu
+0000e220: 6e63 7469 6f6e 7320 6861 7665 206d 6f72  nctions have mor
+0000e230: 6520 6578 7465 6e73 6976 6520 646f 6375  e extensive docu
+0000e240: 6d65 6e74 6174 696f 6e20 7468 616e 2074  mentation than t
+0000e250: 6865 6972 2070 7265 7669 6f75 7320 7665  heir previous ve
+0000e260: 7273 696f 6e73 2e20 4966 2073 6967 6e69  rsions. If signi
+0000e270: 6669 6361 6e63 6520 7465 7374 7320 6172  ficance tests ar
+0000e280: 6520 7375 7070 7265 7373 6564 2c20 702d  e suppressed, p-
+0000e290: 7661 6c75 6573 2061 7265 2072 6574 7572  values are retur
+0000e2a0: 6e65 6420 6173 2060 606e 702e 6e61 6e60  ned as ``np.nan`
+0000e2b0: 6020 696e 7374 6561 6420 6f66 2060 604e  ` instead of ``N
+0000e2c0: 6f6e 6560 6020 666f 7220 636f 6e73 6973  one`` for consis
+0000e2d0: 7465 6e63 7920 7769 7468 206f 7468 6572  tency with other
+0000e2e0: 2073 7461 7469 7374 6963 616c 206d 6574   statistical met
+0000e2f0: 686f 6473 2069 6e20 7363 696b 6974 2d62  hods in scikit-b
+0000e300: 696f 2e20 5b23 3735 345d 2868 7474 7073  io. [#754](https
+0000e310: 3a2f 2f67 6974 6875 622e 636f 6d2f 6269  ://github.com/bi
+0000e320: 6f63 6f72 652f 7363 696b 6974 2d62 696f  ocore/scikit-bio
+0000e330: 2f69 7373 7565 732f 3735 3429 0a2a 2041  /issues/754).* A
+0000e340: 6464 6564 2060 736b 6269 6f2e 7374 6174  dded `skbio.stat
+0000e350: 732e 706f 7765 7260 2066 6f72 2070 6572  s.power` for per
+0000e360: 666f 726d 696e 6720 656d 7069 7269 6361  forming empirica
+0000e370: 6c20 706f 7765 7220 616e 616c 7973 6973  l power analysis
+0000e380: 2e20 5468 6520 6d6f 6475 6c65 2075 7365  . The module use
+0000e390: 7320 6578 6973 7469 6e67 2064 6174 6173  s existing datas
+0000e3a0: 6574 7320 616e 6420 6974 6572 6174 6976  ets and iterativ
+0000e3b0: 656c 7920 6472 6177 7320 7361 6d70 6c65  ely draws sample
+0000e3c0: 7320 746f 2065 7374 696d 6174 6520 7468  s to estimate th
+0000e3d0: 6520 6e75 6d62 6572 206f 6620 7361 6d70  e number of samp
+0000e3e0: 6c65 7320 6e65 6564 6564 2074 6f20 7365  les needed to se
+0000e3f0: 6520 6120 7369 676e 6966 6963 616e 7420  e a significant 
+0000e400: 6469 6666 6572 656e 6365 2066 6f72 2061  difference for a
+0000e410: 2067 6976 656e 2063 7269 7469 6361 6c20   given critical 
+0000e420: 7661 6c75 652e 0a2a 2041 6464 6564 2060  value..* Added `
+0000e430: 736b 6269 6f2e 7374 6174 732e 6973 7562  skbio.stats.isub
+0000e440: 7361 6d70 6c65 6020 666f 7220 7375 6273  sample` for subs
+0000e450: 616d 706c 696e 6720 6672 6f6d 2061 6e20  ampling from an 
+0000e460: 756e 6b6e 6f77 6e20 6e75 6d62 6572 206f  unknown number o
+0000e470: 6620 7661 6c75 6573 2e20 5468 6973 206d  f values. This m
+0000e480: 6574 686f 6420 7375 7070 6f72 7473 2073  ethod supports s
+0000e490: 7562 7361 6d70 6c69 6e67 2066 726f 6d20  ubsampling from 
+0000e4a0: 6d75 6c74 6970 6c65 2070 6172 7469 7469  multiple partiti
+0000e4b0: 6f6e 7320 616e 6420 646f 6573 206e 6f74  ons and does not
+0000e4c0: 2072 6571 7569 7265 2074 6861 7420 616c   require that al
+0000e4d0: 6c20 6974 656d 7320 6265 2073 746f 7265  l items be store
+0000e4e0: 6420 696e 206d 656d 6f72 792c 2072 6571  d in memory, req
+0000e4f0: 7569 7269 6e67 2061 7070 726f 7869 6d61  uiring approxima
+0000e500: 7465 6c79 2060 4f28 4e2a 4d29 6060 2073  tely `O(N*M)`` s
+0000e510: 7061 6365 2077 6865 7265 2060 4e60 2069  pace where `N` i
+0000e520: 7320 7468 6520 6e75 6d62 6572 206f 6620  s the number of 
+0000e530: 7061 7274 6974 696f 6e73 2061 6e64 2060  partitions and `
+0000e540: 4d60 2069 7320 7468 6520 6d61 7869 6d75  M` is the maximu
+0000e550: 6d20 7375 6273 616d 706c 6520 7369 7a65  m subsample size
+0000e560: 2e0a 2a20 4164 6465 6420 6060 736b 6269  ..* Added ``skbi
+0000e570: 6f2e 7374 6174 732e 7375 6273 616d 706c  o.stats.subsampl
+0000e580: 655f 636f 756e 7473 6060 2c20 7768 6963  e_counts``, whic
+0000e590: 6820 7265 706c 6163 6573 2060 6073 6b62  h replaces ``skb
+0000e5a0: 696f 2e73 7461 7473 2e73 7562 7361 6d70  io.stats.subsamp
+0000e5b0: 6c65 6060 2e20 5365 6520 6465 7072 6563  le``. See deprec
+0000e5c0: 6174 696f 6e20 7365 6374 696f 6e20 6265  ation section be
+0000e5d0: 6c6f 7720 666f 7220 6d6f 7265 2064 6574  low for more det
+0000e5e0: 6169 6c73 2028 5b23 3737 305d 2868 7474  ails ([#770](htt
+0000e5f0: 7073 3a2f 2f67 6974 6875 622e 636f 6d2f  ps://github.com/
+0000e600: 6269 6f63 6f72 652f 7363 696b 6974 2d62  biocore/scikit-b
+0000e610: 696f 2f69 7373 7565 732f 3737 3029 292e  io/issues/770)).
+0000e620: 0a0a 2323 2320 4275 6720 6669 7865 730a  ..### Bug fixes.
+0000e630: 2a20 4669 7865 6420 6973 7375 6520 7768  * Fixed issue wh
+0000e640: 6572 6520 5353 5720 776f 756c 646e 2774  ere SSW wouldn't
+0000e650: 2063 6f6d 7069 6c65 206f 6e20 6936 3836   compile on i686
+0000e660: 2061 7263 6869 7465 6374 7572 6573 2028   architectures (
+0000e670: 5b23 3430 395d 2868 7474 7073 3a2f 2f67  [#409](https://g
+0000e680: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
+0000e690: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
+0000e6a0: 7565 732f 3430 3929 292e 0a0a 2323 2320  ues/409))...### 
+0000e6b0: 4465 7072 6563 6174 6564 2066 756e 6374  Deprecated funct
+0000e6c0: 696f 6e61 6c69 7479 0a2a 2044 6570 7265  ionality.* Depre
+0000e6d0: 6361 7465 6420 6060 736b 6269 6f2e 7374  cated ``skbio.st
+0000e6e0: 6174 732e 705f 7661 6c75 655f 746f 5f73  ats.p_value_to_s
+0000e6f0: 7472 6060 2e20 5468 6973 2066 756e 6374  tr``. This funct
+0000e700: 696f 6e20 7769 6c6c 2062 6520 7265 6d6f  ion will be remo
+0000e710: 7665 6420 696e 2073 6369 6b69 742d 6269  ved in scikit-bi
+0000e720: 6f20 302e 332e 302e 2050 6572 6d75 7461  o 0.3.0. Permuta
+0000e730: 7469 6f6e 2d62 6173 6564 2070 2d76 616c  tion-based p-val
+0000e740: 7565 7320 696e 2073 6369 6b69 742d 6269  ues in scikit-bi
+0000e750: 6f20 6172 6520 6361 6c63 756c 6174 6564  o are calculated
+0000e760: 2061 7320 6060 286e 756d 5f65 7874 7265   as ``(num_extre
+0000e770: 6d65 202b 2031 2920 2f20 286e 756d 5f70  me + 1) / (num_p
+0000e780: 6572 6d75 7461 7469 6f6e 7320 2b20 3129  ermutations + 1)
+0000e790: 6060 2c20 736f 2069 7420 6973 2069 6d70  ``, so it is imp
+0000e7a0: 6f73 7369 626c 6520 746f 206f 6274 6169  ossible to obtai
+0000e7b0: 6e20 6120 702d 7661 6c75 6520 6f66 207a  n a p-value of z
+0000e7c0: 6572 6f2e 2054 6869 7320 6675 6e63 7469  ero. This functi
+0000e7d0: 6f6e 2068 6973 746f 7269 6361 6c6c 7920  on historically 
+0000e7e0: 6578 6973 7465 6420 666f 7220 636f 7272  existed for corr
+0000e7f0: 6563 7469 6e67 2074 6865 206e 756d 6265  ecting the numbe
+0000e800: 7220 6f66 2064 6967 6974 7320 6469 7370  r of digits disp
+0000e810: 6c61 7965 6420 7768 656e 206f 6274 6169  layed when obtai
+0000e820: 6e69 6e67 2061 2070 2d76 616c 7565 206f  ning a p-value o
+0000e830: 6620 7a65 726f 2e20 5369 6e63 6520 7468  f zero. Since th
+0000e840: 6973 2069 7320 6e6f 206c 6f6e 6765 7220  is is no longer 
+0000e850: 706f 7373 6962 6c65 2c20 7468 6973 2066  possible, this f
+0000e860: 756e 6374 696f 6e61 6c69 7479 2077 696c  unctionality wil
+0000e870: 6c20 6265 2072 656d 6f76 6564 2e0a 2a20  l be removed..* 
+0000e880: 4465 7072 6563 6174 6564 2060 6073 6b62  Deprecated ``skb
+0000e890: 696f 2e73 7461 7473 2e64 6973 7461 6e63  io.stats.distanc
+0000e8a0: 652e 414e 4f53 494d 6060 2061 6e64 2060  e.ANOSIM`` and `
+0000e8b0: 6073 6b62 696f 2e73 7461 7473 2e64 6973  `skbio.stats.dis
+0000e8c0: 7461 6e63 652e 5045 524d 414e 4f56 4160  tance.PERMANOVA`
+0000e8d0: 6020 696e 2066 6176 6f72 206f 6620 6060  ` in favor of ``
+0000e8e0: 736b 6269 6f2e 7374 6174 732e 6469 7374  skbio.stats.dist
+0000e8f0: 616e 6365 2e61 6e6f 7369 6d60 6020 616e  ance.anosim`` an
+0000e900: 6420 6060 736b 6269 6f2e 7374 6174 732e  d ``skbio.stats.
+0000e910: 6469 7374 616e 6365 2e70 6572 6d61 6e6f  distance.permano
+0000e920: 7661 6060 2c20 7265 7370 6563 7469 7665  va``, respective
+0000e930: 6c79 2e0a 2a20 4465 7072 6563 6174 6564  ly..* Deprecated
+0000e940: 2060 6073 6b62 696f 2e73 7461 7473 2e64   ``skbio.stats.d
+0000e950: 6973 7461 6e63 652e 4361 7465 676f 7269  istance.Categori
+0000e960: 6361 6c53 7461 7473 5265 7375 6c74 7360  calStatsResults`
+0000e970: 6020 696e 2066 6176 6f72 206f 6620 7573  ` in favor of us
+0000e980: 696e 6720 6060 7061 6e64 6173 2e53 6572  ing ``pandas.Ser
+0000e990: 6965 7360 6020 746f 2073 746f 7265 2073  ies`` to store s
+0000e9a0: 7461 7469 7374 6963 616c 206d 6574 686f  tatistical metho
+0000e9b0: 6420 7265 7375 6c74 732e 2060 6061 6e6f  d results. ``ano
+0000e9c0: 7369 6d60 6020 616e 6420 6060 7065 726d  sim`` and ``perm
+0000e9d0: 616e 6f76 6160 6020 7265 7475 726e 2060  anova`` return `
+0000e9e0: 6070 616e 6461 732e 5365 7269 6573 6060  `pandas.Series``
+0000e9f0: 2069 6e73 7465 6164 206f 6620 6060 4361   instead of ``Ca
+0000ea00: 7465 676f 7269 6361 6c53 7461 7473 5265  tegoricalStatsRe
+0000ea10: 7375 6c74 7360 602e 0a2a 2044 6570 7265  sults``..* Depre
+0000ea20: 6361 7465 6420 6060 736b 6269 6f2e 7374  cated ``skbio.st
+0000ea30: 6174 732e 7375 6273 616d 706c 6560 6020  ats.subsample`` 
+0000ea40: 696e 2066 6176 6f72 206f 6620 6060 736b  in favor of ``sk
+0000ea50: 6269 6f2e 7374 6174 732e 7375 6273 616d  bio.stats.subsam
+0000ea60: 706c 655f 636f 756e 7473 6060 2c20 7768  ple_counts``, wh
+0000ea70: 6963 6820 7072 6f76 6964 6573 2061 6e20  ich provides an 
+0000ea80: 6964 656e 7469 6361 6c20 696e 7465 7266  identical interf
+0000ea90: 6163 653b 206f 6e6c 7920 7468 6520 6675  ace; only the fu
+0000eaa0: 6e63 7469 6f6e 206e 616d 6520 6861 7320  nction name has 
+0000eab0: 6368 616e 6765 642e 2060 6073 6b62 696f  changed. ``skbio
+0000eac0: 2e73 7461 7473 2e73 7562 7361 6d70 6c65  .stats.subsample
+0000ead0: 6060 2077 696c 6c20 6265 2072 656d 6f76  `` will be remov
+0000eae0: 6564 2069 6e20 7363 696b 6974 2d62 696f  ed in scikit-bio
+0000eaf0: 2030 2e33 2e30 2e0a 0a23 2323 2042 6163   0.3.0...### Bac
+0000eb00: 6b77 6172 642d 696e 636f 6d70 6174 6962  kward-incompatib
+0000eb10: 6c65 2063 6861 6e67 6573 0a2a 2044 6570  le changes.* Dep
+0000eb20: 7265 6361 7469 6f6e 2077 6172 6e69 6e67  recation warning
+0000eb30: 7320 6172 6520 6e6f 7720 7261 6973 6564  s are now raised
+0000eb40: 2075 7369 6e67 2060 6044 6570 7265 6361   using ``Depreca
+0000eb50: 7469 6f6e 5761 726e 696e 6760 6020 696e  tionWarning`` in
+0000eb60: 7374 6561 6420 6f66 2060 6055 7365 7257  stead of ``UserW
+0000eb70: 6172 6e69 6e67 6060 2028 5b23 3737 345d  arning`` ([#774]
+0000eb80: 2868 7474 7073 3a2f 2f67 6974 6875 622e  (https://github.
+0000eb90: 636f 6d2f 6269 6f63 6f72 652f 7363 696b  com/biocore/scik
+0000eba0: 6974 2d62 696f 2f69 7373 7565 732f 3737  it-bio/issues/77
+0000ebb0: 3429 292e 0a0a 2323 2320 4d69 7363 656c  4))...### Miscel
+0000ebc0: 6c61 6e65 6f75 730a 2a20 5468 6520 6060  laneous.* The ``
+0000ebd0: 7061 6e64 6173 2e44 6174 6146 7261 6d65  pandas.DataFrame
+0000ebe0: 6060 2072 6574 7572 6e65 6420 6279 2060  `` returned by `
+0000ebf0: 6073 6b62 696f 2e73 7461 7473 2e64 6973  `skbio.stats.dis
+0000ec00: 7461 6e63 652e 7077 6d61 6e74 656c 6060  tance.pwmantel``
+0000ec10: 206e 6f77 2073 746f 7265 7320 702d 7661   now stores p-va
+0000ec20: 6c75 6573 2061 7320 666c 6f61 7473 2061  lues as floats a
+0000ec30: 6e64 2064 6f65 7320 6e6f 7420 636f 6e76  nd does not conv
+0000ec40: 6572 7420 7468 656d 2074 6f20 7374 7269  ert them to stri
+0000ec50: 6e67 7320 7769 7468 2061 2073 7065 6369  ngs with a speci
+0000ec60: 6669 6320 6e75 6d62 6572 206f 6620 6469  fic number of di
+0000ec70: 6769 7473 2e20 702d 7661 6c75 6573 2074  gits. p-values t
+0000ec80: 6861 7420 7765 7265 2070 7265 7669 6f75  hat were previou
+0000ec90: 736c 7920 7374 6f72 6564 2061 7320 224e  sly stored as "N
+0000eca0: 2f41 2220 6172 6520 6e6f 7720 7374 6f72  /A" are now stor
+0000ecb0: 6564 2061 7320 6060 6e70 2e6e 616e 6060  ed as ``np.nan``
+0000ecc0: 2066 6f72 2063 6f6e 7369 7374 656e 6379   for consistency
+0000ecd0: 2077 6974 6820 6f74 6865 7220 7374 6174   with other stat
+0000ece0: 6973 7469 6361 6c20 6d65 7468 6f64 7320  istical methods 
+0000ecf0: 696e 2073 6369 6b69 742d 6269 6f2e 2053  in scikit-bio. S
+0000ed00: 6565 206e 6f74 6520 696e 2022 4465 7072  ee note in "Depr
+0000ed10: 6563 6174 6564 2066 756e 6374 696f 6e61  ecated functiona
+0000ed20: 6c69 7479 2220 6162 6f76 6520 7265 6761  lity" above rega
+0000ed30: 7264 696e 6720 6060 705f 7661 6c75 655f  rding ``p_value_
+0000ed40: 746f 5f73 7472 6060 2066 6f72 2064 6574  to_str`` for det
+0000ed50: 6169 6c73 2e0a 2a20 7363 696b 6974 2d62  ails..* scikit-b
+0000ed60: 696f 206e 6f77 2073 7570 706f 7274 7320  io now supports 
+0000ed70: 7665 7273 696f 6e73 206f 6620 4950 7974  versions of IPyt
+0000ed80: 686f 6e20 3c20 322e 302e 3020 285b 2337  hon < 2.0.0 ([#7
+0000ed90: 3637 5d28 6874 7470 733a 2f2f 6769 7468  67](https://gith
+0000eda0: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+0000edb0: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
+0000edc0: 2f37 3637 2929 2e0a 0a23 2320 5665 7273  /767))...## Vers
+0000edd0: 696f 6e20 302e 322e 3120 2832 3031 342d  ion 0.2.1 (2014-
+0000ede0: 3130 2d32 3729 0a0a 5468 6973 2069 7320  10-27)..This is 
+0000edf0: 616e 2061 6c70 6861 2072 656c 6561 7365  an alpha release
+0000ee00: 206f 6620 7363 696b 6974 2d62 696f 2e20   of scikit-bio. 
+0000ee10: 4174 2074 6869 7320 7374 6167 652c 206d  At this stage, m
+0000ee20: 616a 6f72 2062 6163 6b77 6172 6473 2d69  ajor backwards-i
+0000ee30: 6e63 6f6d 7061 7469 626c 6520 4150 4920  ncompatible API 
+0000ee40: 6368 616e 6765 7320 6361 6e20 616e 6420  changes can and 
+0000ee50: 7769 6c6c 2068 6170 7065 6e2e 2055 6e69  will happen. Uni
+0000ee60: 6669 6564 2049 2f4f 2077 6974 6820 7468  fied I/O with th
+0000ee70: 6520 7363 696b 6974 2d62 696f 2049 2f4f  e scikit-bio I/O
+0000ee80: 2072 6567 6973 7472 7920 7761 7320 7468   registry was th
+0000ee90: 6520 666f 6375 7320 6f66 2074 6869 7320  e focus of this 
+0000eea0: 7265 6c65 6173 652e 0a0a 2323 2320 4665  release...### Fe
+0000eeb0: 6174 7572 6573 0a2a 2041 6464 6564 2060  atures.* Added `
+0000eec0: 6073 7472 6963 7460 6020 616e 6420 6060  `strict`` and ``
+0000eed0: 6c6f 6f6b 7570 6060 206f 7074 696f 6e61  lookup`` optiona
+0000eee0: 6c20 7061 7261 6d65 7465 7273 2074 6f20  l parameters to 
+0000eef0: 6060 736b 6269 6f2e 7374 6174 732e 6469  ``skbio.stats.di
+0000ef00: 7374 616e 6365 2e6d 616e 7465 6c60 6020  stance.mantel`` 
+0000ef10: 666f 7220 6861 6e64 6c69 6e67 2072 656f  for handling reo
+0000ef20: 7264 6572 696e 6720 616e 6420 6d61 7463  rdering and matc
+0000ef30: 6869 6e67 206f 6620 4944 7320 7768 656e  hing of IDs when
+0000ef40: 2070 726f 7669 6465 6420 6060 4469 7374   provided ``Dist
+0000ef50: 616e 6365 4d61 7472 6978 6060 2069 6e73  anceMatrix`` ins
+0000ef60: 7461 6e63 6573 2061 7320 696e 7075 7420  tances as input 
+0000ef70: 2874 6865 7365 2070 6172 616d 6574 6572  (these parameter
+0000ef80: 7320 7765 7265 2070 7265 7669 6f75 736c  s were previousl
+0000ef90: 7920 6f6e 6c79 2061 7661 696c 6162 6c65  y only available
+0000efa0: 2069 6e20 6060 736b 6269 6f2e 7374 6174   in ``skbio.stat
+0000efb0: 732e 6469 7374 616e 6365 2e70 776d 616e  s.distance.pwman
+0000efc0: 7465 6c60 6029 2e0a 2a20 6060 736b 6269  tel``)..* ``skbi
+0000efd0: 6f2e 7374 6174 732e 6469 7374 616e 6365  o.stats.distance
+0000efe0: 2e70 776d 616e 7465 6c60 6020 6e6f 7720  .pwmantel`` now 
+0000eff0: 6163 6365 7074 7320 616e 2069 7465 7261  accepts an itera
+0000f000: 626c 6520 6f66 2060 6061 7272 6179 5f6c  ble of ``array_l
+0000f010: 696b 6560 6020 6f62 6a65 6374 732e 2050  ike`` objects. P
+0000f020: 7265 7669 6f75 736c 792c 206f 6e6c 7920  reviously, only 
+0000f030: 6060 4469 7374 616e 6365 4d61 7472 6978  ``DistanceMatrix
+0000f040: 6060 2069 6e73 7461 6e63 6573 2077 6572  `` instances wer
+0000f050: 6520 616c 6c6f 7765 642e 0a2a 2041 6464  e allowed..* Add
+0000f060: 6564 2060 6070 6c6f 7460 6020 6d65 7468  ed ``plot`` meth
+0000f070: 6f64 2074 6f20 6060 736b 6269 6f2e 7374  od to ``skbio.st
+0000f080: 6174 732e 6f72 6469 6e61 7469 6f6e 2e4f  ats.ordination.O
+0000f090: 7264 696e 6174 696f 6e52 6573 756c 7473  rdinationResults
+0000f0a0: 6060 2066 6f72 2063 7265 6174 696e 6720  `` for creating 
+0000f0b0: 6261 7369 6320 332d 4420 6d61 7470 6c6f  basic 3-D matplo
+0000f0c0: 746c 6962 2073 6361 7474 6572 706c 6f74  tlib scatterplot
+0000f0d0: 7320 6f66 206f 7264 696e 6174 696f 6e20  s of ordination 
+0000f0e0: 7265 7375 6c74 732c 206f 7074 696f 6e61  results, optiona
+0000f0f0: 6c6c 7920 636f 6c6f 7265 6420 6279 206d  lly colored by m
+0000f100: 6574 6164 6174 6120 696e 2061 2060 6070  etadata in a ``p
+0000f110: 616e 6461 732e 4461 7461 4672 616d 6560  andas.DataFrame`
+0000f120: 6020 2873 6565 205b 2335 3138 5d28 6874  ` (see [#518](ht
+0000f130: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+0000f140: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
+0000f150: 6269 6f2f 6973 7375 6573 2f35 3138 2929  bio/issues/518))
+0000f160: 2e20 416c 736f 2061 6464 6564 2020 6060  . Also added  ``
+0000f170: 5f72 6570 725f 706e 675f 6060 2061 6e64  _repr_png_`` and
+0000f180: 2060 605f 7265 7072 5f73 7667 5f60 6020   ``_repr_svg_`` 
+0000f190: 6d65 7468 6f64 7320 666f 7220 6175 746f  methods for auto
+0000f1a0: 6d61 7469 6320 6469 7370 6c61 7920 696e  matic display in
+0000f1b0: 2074 6865 2049 5079 7468 6f6e 204e 6f74   the IPython Not
+0000f1c0: 6562 6f6f 6b2c 2077 6974 6820 6060 706e  ebook, with ``pn
+0000f1d0: 6760 6020 616e 6420 6060 7376 6760 6020  g`` and ``svg`` 
+0000f1e0: 7072 6f70 6572 7469 6573 2066 6f72 2064  properties for d
+0000f1f0: 6972 6563 7420 6163 6365 7373 2e0a 2a20  irect access..* 
+0000f200: 4164 6465 6420 6060 736b 6269 6f2e 7374  Added ``skbio.st
+0000f210: 6174 732e 6f72 6469 6e61 7469 6f6e 2e61  ats.ordination.a
+0000f220: 7373 6572 745f 6f72 6469 6e61 7469 6f6e  ssert_ordination
+0000f230: 5f72 6573 756c 7473 5f65 7175 616c 6060  _results_equal``
+0000f240: 2066 6f72 2063 6f6d 7061 7269 6e67 2060   for comparing `
+0000f250: 604f 7264 696e 6174 696f 6e52 6573 756c  `OrdinationResul
+0000f260: 7473 6060 206f 626a 6563 7473 2066 6f72  ts`` objects for
+0000f270: 2065 7175 616c 6974 7920 696e 2075 6e69   equality in uni
+0000f280: 7420 7465 7374 732e 0a2a 2060 6042 696f  t tests..* ``Bio
+0000f290: 6c6f 6769 6361 6c53 6571 7565 6e63 6560  logicalSequence`
+0000f2a0: 6020 2861 6e64 2069 7473 2073 7562 636c  ` (and its subcl
+0000f2b0: 6173 7365 7329 206e 6f77 206f 7074 696f  asses) now optio
+0000f2c0: 6e61 6c6c 7920 7374 6f72 6520 5068 7265  nally store Phre
+0000f2d0: 6420 7175 616c 6974 7920 7363 6f72 6573  d quality scores
+0000f2e0: 2e20 4120 6269 6f6c 6f67 6963 616c 2073  . A biological s
+0000f2f0: 6571 7565 6e63 6527 7320 7175 616c 6974  equence's qualit
+0000f300: 7920 7363 6f72 6573 2061 7265 2073 746f  y scores are sto
+0000f310: 7265 6420 6173 2061 2031 2d44 2060 606e  red as a 1-D ``n
+0000f320: 756d 7079 2e6e 6461 7272 6179 6060 206f  umpy.ndarray`` o
+0000f330: 6620 6e6f 6e6e 6567 6174 6976 6520 696e  f nonnegative in
+0000f340: 7465 6765 7273 2074 6861 7420 6973 2074  tegers that is t
+0000f350: 6865 2073 616d 6520 6c65 6e67 7468 2061  he same length a
+0000f360: 7320 7468 6520 6269 6f6c 6f67 6963 616c  s the biological
+0000f370: 2073 6571 7565 6e63 652e 2051 7561 6c69   sequence. Quali
+0000f380: 7479 2073 636f 7265 7320 6361 6e20 6265  ty scores can be
+0000f390: 2070 726f 7669 6465 6420 7570 6f6e 206f   provided upon o
+0000f3a0: 626a 6563 7420 696e 7374 616e 7469 6174  bject instantiat
+0000f3b0: 696f 6e20 7669 6120 7468 6520 6b65 7977  ion via the keyw
+0000f3c0: 6f72 6420 6172 6775 6d65 6e74 2060 6071  ord argument ``q
+0000f3d0: 7561 6c69 7479 6060 2c20 616e 6420 6361  uality``, and ca
+0000f3e0: 6e20 6265 2072 6574 7269 6576 6564 2076  n be retrieved v
+0000f3f0: 6961 2074 6865 2060 6042 696f 6c6f 6769  ia the ``Biologi
+0000f400: 6361 6c53 6571 7565 6e63 652e 7175 616c  calSequence.qual
+0000f410: 6974 7960 6020 7072 6f70 6572 7479 2e20  ity`` property. 
+0000f420: 6060 4269 6f6c 6f67 6963 616c 5365 7175  ``BiologicalSequ
+0000f430: 656e 6365 2e68 6173 5f71 7561 6c69 7479  ence.has_quality
+0000f440: 6060 2069 7320 616c 736f 2070 726f 7669  `` is also provi
+0000f450: 6465 6420 666f 7220 6465 7465 726d 696e  ded for determin
+0000f460: 696e 6720 7768 6574 6865 7220 6120 6269  ing whether a bi
+0000f470: 6f6c 6f67 6963 616c 2073 6571 7565 6e63  ological sequenc
+0000f480: 6520 6861 7320 7175 616c 6974 7920 7363  e has quality sc
+0000f490: 6f72 6573 206f 7220 6e6f 742e 2053 6565  ores or not. See
+0000f4a0: 205b 2336 3136 5d28 6874 7470 733a 2f2f   [#616](https://
+0000f4b0: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
+0000f4c0: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
+0000f4d0: 7375 6573 2f36 3136 2920 666f 7220 6d6f  sues/616) for mo
+0000f4e0: 7265 2064 6574 6169 6c73 2e0a 2a20 4164  re details..* Ad
+0000f4f0: 6465 6420 6060 4269 6f6c 6f67 6963 616c  ded ``Biological
+0000f500: 5365 7175 656e 6365 2e73 6571 7565 6e63  Sequence.sequenc
+0000f510: 6560 6020 7072 6f70 6572 7479 2066 6f72  e`` property for
+0000f520: 2072 6574 7269 6576 696e 6720 7468 6520   retrieving the 
+0000f530: 756e 6465 726c 7969 6e67 2073 7472 696e  underlying strin
+0000f540: 6720 7265 7072 6573 656e 7469 6e67 2074  g representing t
+0000f550: 6865 2073 6571 7565 6e63 6520 6368 6172  he sequence char
+0000f560: 6163 7465 7273 2e20 5468 6973 2077 6173  acters. This was
+0000f570: 2070 7265 7669 6f75 736c 7920 2861 6e64   previously (and
+0000f580: 2073 7469 6c6c 2069 7329 2061 6363 6573   still is) acces
+0000f590: 7369 626c 6520 7669 6120 6060 4269 6f6c  sible via ``Biol
+0000f5a0: 6f67 6963 616c 5365 7175 656e 6365 2e5f  ogicalSequence._
+0000f5b0: 5f73 7472 5f5f 6060 2e20 4974 2069 7320  _str__``. It is 
+0000f5c0: 7072 6f76 6964 6564 2076 6961 2061 2070  provided via a p
+0000f5d0: 726f 7065 7274 7920 666f 7220 636f 6e76  roperty for conv
+0000f5e0: 656e 6965 6e63 6520 616e 6420 6578 706c  enience and expl
+0000f5f0: 6963 6974 6e65 7373 2e0a 2a20 4164 6465  icitness..* Adde
+0000f600: 6420 6060 4269 6f6c 6f67 6963 616c 5365  d ``BiologicalSe
+0000f610: 7175 656e 6365 2e65 7175 616c 7360 6020  quence.equals`` 
+0000f620: 666f 7220 6675 6c6c 2063 6f6e 7472 6f6c  for full control
+0000f630: 206f 7665 7220 6571 7561 6c69 7479 2074   over equality t
+0000f640: 6573 7469 6e67 206f 6620 6269 6f6c 6f67  esting of biolog
+0000f650: 6963 616c 2073 6571 7565 6e63 6573 2e20  ical sequences. 
+0000f660: 4279 2064 6566 6175 6c74 2c20 6269 6f6c  By default, biol
+0000f670: 6f67 6963 616c 2073 6571 7565 6e63 6573  ogical sequences
+0000f680: 206d 7573 7420 6861 7665 2074 6865 2073   must have the s
+0000f690: 616d 6520 7479 7065 2c20 756e 6465 726c  ame type, underl
+0000f6a0: 7969 6e67 2073 6571 7565 6e63 6520 6f66  ying sequence of
+0000f6b0: 2063 6861 7261 6374 6572 732c 2069 6465   characters, ide
+0000f6c0: 6e74 6966 6965 722c 2064 6573 6372 6970  ntifier, descrip
+0000f6d0: 7469 6f6e 2c20 616e 6420 7175 616c 6974  tion, and qualit
+0000f6e0: 7920 7363 6f72 6573 2074 6f20 636f 6d70  y scores to comp
+0000f6f0: 6172 6520 6571 7561 6c2e 2054 6865 7365  are equal. These
+0000f700: 2070 726f 7065 7274 6965 7320 6361 6e20   properties can 
+0000f710: 6265 2069 676e 6f72 6564 2076 6961 2074  be ignored via t
+0000f720: 6865 206b 6579 776f 7264 2061 7267 756d  he keyword argum
+0000f730: 656e 7420 6060 6967 6e6f 7265 6060 2e20  ent ``ignore``. 
+0000f740: 5468 6520 6265 6861 7669 6f72 206f 6620  The behavior of 
+0000f750: 6060 4269 6f6c 6f67 6963 616c 5365 7175  ``BiologicalSequ
+0000f760: 656e 6365 2e5f 5f65 715f 5f60 602f 6060  ence.__eq__``/``
+0000f770: 5f5f 6e65 5f5f 6060 2072 656d 6169 6e73  __ne__`` remains
+0000f780: 2075 6e63 6861 6e67 6564 2028 6f6e 6c79   unchanged (only
+0000f790: 2074 7970 6520 616e 6420 756e 6465 726c   type and underl
+0000f7a0: 7969 6e67 2073 6571 7565 6e63 6520 6f66  ying sequence of
+0000f7b0: 2063 6861 7261 6374 6572 7320 6172 6520   characters are 
+0000f7c0: 636f 6d70 6172 6564 292e 0a2a 2041 6464  compared)..* Add
+0000f7d0: 6564 2060 6042 696f 6c6f 6769 6361 6c53  ed ``BiologicalS
+0000f7e0: 6571 7565 6e63 652e 636f 7079 6060 2066  equence.copy`` f
+0000f7f0: 6f72 2063 7265 6174 696e 6720 6120 636f  or creating a co
+0000f800: 7079 206f 6620 6120 6269 6f6c 6f67 6963  py of a biologic
+0000f810: 616c 2073 6571 7565 6e63 652c 206f 7074  al sequence, opt
+0000f820: 696f 6e61 6c6c 7920 7769 7468 206f 6e65  ionally with one
+0000f830: 206f 7220 6d6f 7265 2061 7474 7269 6275   or more attribu
+0000f840: 7465 7320 7570 6461 7465 642e 0a2a 2060  tes updated..* `
+0000f850: 6042 696f 6c6f 6769 6361 6c53 6571 7565  `BiologicalSeque
+0000f860: 6e63 652e 5f5f 6765 7469 7465 6d5f 5f60  nce.__getitem__`
+0000f870: 6020 6e6f 7720 7375 7070 6f72 7473 2073  ` now supports s
+0000f880: 7065 6369 6679 696e 6720 6120 7365 7175  pecifying a sequ
+0000f890: 656e 6365 206f 6620 696e 6469 6365 7320  ence of indices 
+0000f8a0: 746f 2074 616b 6520 6672 6f6d 2074 6865  to take from the
+0000f8b0: 2062 696f 6c6f 6769 6361 6c20 7365 7175   biological sequ
+0000f8c0: 656e 6365 2e0a 2a20 4d65 7468 6f64 7320  ence..* Methods 
+0000f8d0: 746f 2072 6561 6420 616e 6420 7772 6974  to read and writ
+0000f8e0: 6520 7461 786f 6e6f 6d69 6573 2061 7265  e taxonomies are
+0000f8f0: 206e 6f77 2061 7661 696c 6162 6c65 2075   now available u
+0000f900: 6e64 6572 2060 6073 6b62 696f 2e74 7265  nder ``skbio.tre
+0000f910: 652e 5472 6565 4e6f 6465 2e66 726f 6d5f  e.TreeNode.from_
+0000f920: 7461 786f 6e6f 6d79 6060 2061 6e64 2060  taxonomy`` and `
+0000f930: 6073 6b62 696f 2e74 7265 652e 5472 6565  `skbio.tree.Tree
+0000f940: 4e6f 6465 2e74 6f5f 7461 786f 6e6f 6d79  Node.to_taxonomy
+0000f950: 6060 2072 6573 7065 6374 6976 656c 792e  `` respectively.
+0000f960: 0a2a 2041 6464 6564 2060 6053 6571 7565  .* Added ``Seque
+0000f970: 6e63 6543 6f6c 6c65 6374 696f 6e2e 7570  nceCollection.up
+0000f980: 6461 7465 5f69 6473 6060 2c20 7768 6963  date_ids``, whic
+0000f990: 6820 7072 6f76 6964 6573 2061 2066 6c65  h provides a fle
+0000f9a0: 7869 626c 6520 7761 7920 6f66 2075 7064  xible way of upd
+0000f9b0: 6174 696e 6720 7365 7175 656e 6365 2049  ating sequence I
+0000f9c0: 4473 206f 6e20 6120 6060 5365 7175 656e  Ds on a ``Sequen
+0000f9d0: 6365 436f 6c6c 6563 7469 6f6e 6060 206f  ceCollection`` o
+0000f9e0: 7220 6060 416c 6967 6e6d 656e 7460 6020  r ``Alignment`` 
+0000f9f0: 286e 6f74 6520 7468 6174 2061 206e 6577  (note that a new
+0000fa00: 206f 626a 6563 7420 6973 2072 6574 7572   object is retur
+0000fa10: 6e65 642c 2073 696e 6365 2069 6e73 7461  ned, since insta
+0000fa20: 6e63 6573 206f 6620 7468 6573 6520 636c  nces of these cl
+0000fa30: 6173 7365 7320 6172 6520 696d 6d75 7461  asses are immuta
+0000fa40: 626c 6529 2e20 4465 7072 6563 6174 6564  ble). Deprecated
+0000fa50: 2060 6053 6571 7565 6e63 6543 6f6c 6c65   ``SequenceColle
+0000fa60: 6374 696f 6e2e 696e 745f 6d61 7060 6020  ction.int_map`` 
+0000fa70: 696e 2066 6176 6f72 206f 6620 7468 6973  in favor of this
+0000fa80: 206e 6577 206d 6574 686f 643b 2069 7420   new method; it 
+0000fa90: 7769 6c6c 2062 6520 7265 6d6f 7665 6420  will be removed 
+0000faa0: 696e 2073 6369 6b69 742d 6269 6f20 302e  in scikit-bio 0.
+0000fab0: 332e 302e 0a2a 2041 6464 6564 2060 6073  3.0..* Added ``s
+0000fac0: 6b62 696f 2e75 7469 6c2e 6361 7264 696e  kbio.util.cardin
+0000fad0: 616c 5f74 6f5f 6f72 6469 6e61 6c60 6020  al_to_ordinal`` 
+0000fae0: 666f 7220 636f 6e76 6572 7469 6e67 2061  for converting a
+0000faf0: 2063 6172 6469 6e61 6c20 6e75 6d62 6572   cardinal number
+0000fb00: 2074 6f20 6f72 6469 6e61 6c20 7374 7269   to ordinal stri
+0000fb10: 6e67 2028 652e 672e 2c20 7573 6566 756c  ng (e.g., useful
+0000fb20: 2066 6f72 2065 7272 6f72 206d 6573 7361   for error messa
+0000fb30: 6765 7329 2e0a 2a20 4e65 7720 492f 4f20  ges)..* New I/O 
+0000fb40: 5265 6769 7374 7279 3a20 7375 7070 6f72  Registry: suppor
+0000fb50: 7473 206d 756c 7469 706c 6520 6669 6c65  ts multiple file
+0000fb60: 2066 6f72 6d61 7473 2c20 6175 746f 6d61   formats, automa
+0000fb70: 7469 6320 6669 6c65 2066 6f72 6d61 7420  tic file format 
+0000fb80: 6465 7465 6374 696f 6e20 7768 656e 2072  detection when r
+0000fb90: 6561 6469 6e67 2c20 756e 6966 6965 6420  eading, unified 
+0000fba0: 7072 6f63 6564 7572 616c 2060 6073 6b62  procedural ``skb
+0000fbb0: 696f 2e69 6f2e 7265 6164 6060 2061 6e64  io.io.read`` and
+0000fbc0: 2060 6073 6b62 696f 2e69 6f2e 7772 6974   ``skbio.io.writ
+0000fbd0: 6560 6020 696e 2061 6464 6974 696f 6e20  e`` in addition 
+0000fbe0: 746f 204f 4f50 2069 6e74 6572 6661 6365  to OOP interface
+0000fbf0: 7320 2860 6072 6561 642f 7772 6974 6560  s (``read/write`
+0000fc00: 6020 6d65 7468 6f64 7329 206f 6e20 7468  ` methods) on th
+0000fc10: 6520 6265 6c6f 7720 6f62 6a65 6374 732e  e below objects.
+0000fc20: 2053 6565 2060 6073 6b62 696f 2e69 6f60   See ``skbio.io`
+0000fc30: 6020 666f 7220 6d6f 7265 2064 6574 6169  ` for more detai
+0000fc40: 6c73 2e0a 2020 2020 2d20 4164 6465 6420  ls..    - Added 
+0000fc50: 2263 6c75 7374 616c 2220 666f 726d 6174  "clustal" format
+0000fc60: 2073 7570 706f 7274 3a0a 2020 2020 2020   support:.      
+0000fc70: 2020 2a20 4861 7320 736e 6966 6665 720a    * Has sniffer.
+0000fc80: 2020 2020 2020 2020 2a20 5265 6164 6572          * Reader
+0000fc90: 733a 2060 6041 6c69 676e 6d65 6e74 6060  s: ``Alignment``
+0000fca0: 0a20 2020 2020 2020 202a 2057 7269 7465  .        * Write
+0000fcb0: 7273 3a20 6060 416c 6967 6e6d 656e 7460  rs: ``Alignment`
+0000fcc0: 600a 2020 2020 2d20 4164 6465 6420 226c  `.    - Added "l
+0000fcd0: 736d 6174 2220 666f 726d 6174 2073 7570  smat" format sup
+0000fce0: 706f 7274 3a0a 2020 2020 2020 2020 2a20  port:.        * 
+0000fcf0: 4861 7320 736e 6966 6665 720a 2020 2020  Has sniffer.    
+0000fd00: 2020 2020 2a20 5265 6164 6572 733a 2060      * Readers: `
+0000fd10: 6044 6973 7369 6d69 6c61 7269 7479 4d61  `DissimilarityMa
+0000fd20: 7472 6978 6060 2c20 6060 4469 7374 616e  trix``, ``Distan
+0000fd30: 6365 4d61 7472 6978 6060 0a20 2020 2020  ceMatrix``.     
+0000fd40: 2020 202a 2057 7269 7465 7273 3a20 6060     * Writers: ``
+0000fd50: 4469 7373 696d 696c 6172 6974 794d 6174  DissimilarityMat
+0000fd60: 7269 7860 602c 2060 6044 6973 7461 6e63  rix``, ``Distanc
+0000fd70: 654d 6174 7269 7860 600a 2020 2020 2d20  eMatrix``.    - 
+0000fd80: 4164 6465 6420 226f 7264 696e 6174 696f  Added "ordinatio
+0000fd90: 6e22 2066 6f72 6d61 7420 7375 7070 6f72  n" format suppor
+0000fda0: 743a 0a20 2020 2020 2020 202a 2048 6173  t:.        * Has
+0000fdb0: 2073 6e69 6666 6572 0a20 2020 2020 2020   sniffer.       
+0000fdc0: 202a 2052 6561 6465 7273 3a20 6060 4f72   * Readers: ``Or
+0000fdd0: 6469 6e61 7469 6f6e 5265 7375 6c74 7360  dinationResults`
+0000fde0: 600a 2020 2020 2020 2020 2a20 5772 6974  `.        * Writ
+0000fdf0: 6572 733a 2060 604f 7264 696e 6174 696f  ers: ``Ordinatio
+0000fe00: 6e52 6573 756c 7473 6060 0a20 2020 202d  nResults``.    -
+0000fe10: 2041 6464 6564 2022 6e65 7769 636b 2220   Added "newick" 
+0000fe20: 666f 726d 6174 2073 7570 706f 7274 3a0a  format support:.
+0000fe30: 2020 2020 2020 2020 2a20 4861 7320 736e          * Has sn
+0000fe40: 6966 6665 720a 2020 2020 2020 2020 2a20  iffer.        * 
+0000fe50: 5265 6164 6572 733a 2060 6054 7265 654e  Readers: ``TreeN
+0000fe60: 6f64 6560 600a 2020 2020 2020 2020 2a20  ode``.        * 
+0000fe70: 5772 6974 6572 733a 2060 6054 7265 654e  Writers: ``TreeN
+0000fe80: 6f64 6560 600a 2020 2020 2d20 4164 6465  ode``.    - Adde
+0000fe90: 6420 2270 6879 6c69 7022 2066 6f72 6d61  d "phylip" forma
+0000fea0: 7420 7375 7070 6f72 743a 0a20 2020 2020  t support:.     
+0000feb0: 2020 202a 204e 6f20 736e 6966 6665 720a     * No sniffer.
+0000fec0: 2020 2020 2020 2020 2a20 5265 6164 6572          * Reader
+0000fed0: 733a 204e 6f6e 650a 2020 2020 2020 2020  s: None.        
+0000fee0: 2a20 5772 6974 6572 733a 2060 6041 6c69  * Writers: ``Ali
+0000fef0: 676e 6d65 6e74 6060 0a20 2020 202d 2041  gnment``.    - A
+0000ff00: 6464 6564 2022 7173 6571 2220 666f 726d  dded "qseq" form
+0000ff10: 6174 2073 7570 706f 7274 3a0a 2020 2020  at support:.    
+0000ff20: 2020 2020 2a20 4861 7320 736e 6966 6665      * Has sniffe
+0000ff30: 720a 2020 2020 2020 2020 2a20 5265 6164  r.        * Read
+0000ff40: 6572 733a 2067 656e 6572 6174 6f72 206f  ers: generator o
+0000ff50: 6620 6060 4269 6f6c 6f67 6963 616c 5365  f ``BiologicalSe
+0000ff60: 7175 656e 6365 6060 206f 7220 6974 7320  quence`` or its 
+0000ff70: 7375 6263 6c61 7373 6573 2c20 6060 5365  subclasses, ``Se
+0000ff80: 7175 656e 6365 436f 6c6c 6563 7469 6f6e  quenceCollection
+0000ff90: 6060 2c20 6060 4269 6f6c 6f67 6963 616c  ``, ``Biological
+0000ffa0: 5365 7175 656e 6365 6060 2c20 6060 4e75  Sequence``, ``Nu
+0000ffb0: 636c 656f 7469 6465 5365 7175 656e 6365  cleotideSequence
+0000ffc0: 6060 2c20 6060 444e 4153 6571 7565 6e63  ``, ``DNASequenc
+0000ffd0: 6560 602c 2060 6052 4e41 5365 7175 656e  e``, ``RNASequen
+0000ffe0: 6365 6060 2c20 6060 5072 6f74 6569 6e53  ce``, ``ProteinS
+0000fff0: 6571 7565 6e63 6560 600a 2020 2020 2020  equence``.      
+00010000: 2020 2a20 5772 6974 6572 733a 204e 6f6e    * Writers: Non
+00010010: 650a 2020 2020 2d20 4164 6465 6420 2266  e.    - Added "f
+00010020: 6173 7461 222f 5155 414c 2066 6f72 6d61  asta"/QUAL forma
+00010030: 7420 7375 7070 6f72 743a 0a20 2020 2020  t support:.     
+00010040: 2020 202a 2048 6173 2073 6e69 6666 6572     * Has sniffer
+00010050: 0a20 2020 2020 2020 202a 2052 6561 6465  .        * Reade
+00010060: 7273 3a20 6765 6e65 7261 746f 7220 6f66  rs: generator of
+00010070: 2060 6042 696f 6c6f 6769 6361 6c53 6571   ``BiologicalSeq
+00010080: 7565 6e63 6560 6020 6f72 2069 7473 2073  uence`` or its s
+00010090: 7562 636c 6173 7365 732c 2060 6053 6571  ubclasses, ``Seq
+000100a0: 7565 6e63 6543 6f6c 6c65 6374 696f 6e60  uenceCollection`
+000100b0: 602c 2060 6041 6c69 676e 6d65 6e74 6060  `, ``Alignment``
+000100c0: 2c20 6060 4269 6f6c 6f67 6963 616c 5365  , ``BiologicalSe
+000100d0: 7175 656e 6365 6060 2c20 6060 4e75 636c  quence``, ``Nucl
+000100e0: 656f 7469 6465 5365 7175 656e 6365 6060  eotideSequence``
+000100f0: 2c20 6060 444e 4153 6571 7565 6e63 6560  , ``DNASequence`
+00010100: 602c 2060 6052 4e41 5365 7175 656e 6365  `, ``RNASequence
+00010110: 6060 2c20 6060 5072 6f74 6569 6e53 6571  ``, ``ProteinSeq
+00010120: 7565 6e63 6560 600a 2020 2020 2020 2020  uence``.        
+00010130: 2a20 5772 6974 6572 733a 2073 616d 6520  * Writers: same 
+00010140: 6173 2072 6561 6465 7273 0a20 2020 202d  as readers.    -
+00010150: 2041 6464 6564 2022 6661 7374 7122 2066   Added "fastq" f
+00010160: 6f72 6d61 7420 7375 7070 6f72 743a 0a20  ormat support:. 
+00010170: 2020 2020 2020 202a 2048 6173 2073 6e69         * Has sni
+00010180: 6666 6572 0a20 2020 2020 2020 202a 2052  ffer.        * R
+00010190: 6561 6465 7273 3a20 6765 6e65 7261 746f  eaders: generato
+000101a0: 7220 6f66 2060 6042 696f 6c6f 6769 6361  r of ``Biologica
+000101b0: 6c53 6571 7565 6e63 6560 6020 6f72 2069  lSequence`` or i
+000101c0: 7473 2073 7562 636c 6173 7365 732c 2060  ts subclasses, `
+000101d0: 6053 6571 7565 6e63 6543 6f6c 6c65 6374  `SequenceCollect
+000101e0: 696f 6e60 602c 2060 6041 6c69 676e 6d65  ion``, ``Alignme
+000101f0: 6e74 6060 2c20 6060 4269 6f6c 6f67 6963  nt``, ``Biologic
+00010200: 616c 5365 7175 656e 6365 6060 2c20 6060  alSequence``, ``
+00010210: 4e75 636c 656f 7469 6465 5365 7175 656e  NucleotideSequen
+00010220: 6365 6060 2c20 6060 444e 4153 6571 7565  ce``, ``DNASeque
+00010230: 6e63 6560 602c 2060 6052 4e41 5365 7175  nce``, ``RNASequ
+00010240: 656e 6365 6060 2c20 6060 5072 6f74 6569  ence``, ``Protei
+00010250: 6e53 6571 7565 6e63 6560 600a 2020 2020  nSequence``.    
+00010260: 2020 2020 2a20 5772 6974 6572 733a 2073      * Writers: s
+00010270: 616d 6520 6173 2072 6561 6465 7273 0a0a  ame as readers..
+00010280: 2323 2320 4275 6720 6669 7865 730a 0a2a  ### Bug fixes..*
+00010290: 2052 656d 6f76 6564 2060 6063 6f6e 7374   Removed ``const
+000102a0: 7275 6374 6f72 6060 2070 6172 616d 6574  ructor`` paramet
+000102b0: 6572 2066 726f 6d20 6060 416c 6967 6e6d  er from ``Alignm
+000102c0: 656e 742e 6b5f 776f 7264 5f66 7265 7175  ent.k_word_frequ
+000102d0: 656e 6369 6573 6060 2c20 6060 4269 6f6c  encies``, ``Biol
+000102e0: 6f67 6963 616c 5365 7175 656e 6365 2e6b  ogicalSequence.k
+000102f0: 5f77 6f72 6473 6060 2c20 6060 4269 6f6c  _words``, ``Biol
+00010300: 6f67 6963 616c 5365 7175 656e 6365 2e6b  ogicalSequence.k
+00010310: 5f77 6f72 645f 636f 756e 7473 6060 2c20  _word_counts``, 
+00010320: 616e 6420 6060 4269 6f6c 6f67 6963 616c  and ``Biological
+00010330: 5365 7175 656e 6365 2e6b 5f77 6f72 645f  Sequence.k_word_
+00010340: 6672 6571 7565 6e63 6965 7360 6020 6173  frequencies`` as
+00010350: 2069 7420 6861 6420 6e6f 2065 6666 6563   it had no effec
+00010360: 7420 2869 7420 7761 7320 6e65 7665 7220  t (it was never 
+00010370: 686f 6f6b 6564 2075 7020 696e 2074 6865  hooked up in the
+00010380: 2075 6e64 6572 6c79 696e 6720 636f 6465   underlying code
+00010390: 292e 2060 6042 696f 6c6f 6769 6361 6c53  ). ``BiologicalS
+000103a0: 6571 7565 6e63 652e 6b5f 776f 7264 7360  equence.k_words`
+000103b0: 6020 6e6f 7720 7265 7475 726e 7320 6120  ` now returns a 
+000103c0: 6765 6e65 7261 746f 7220 6f66 2060 6042  generator of ``B
+000103d0: 696f 6c6f 6769 6361 6c53 6571 7565 6e63  iologicalSequenc
+000103e0: 6560 6020 6f62 6a65 6374 7320 696e 7374  e`` objects inst
+000103f0: 6561 6420 6f66 2073 7472 696e 6773 2e0a  ead of strings..
+00010400: 2a20 4d6f 6469 6669 6564 2074 6865 2060  * Modified the `
+00010410: 6041 6c69 676e 6d65 6e74 6060 2063 6f6e  `Alignment`` con
+00010420: 7374 7275 6374 6f72 2074 6f20 7665 7269  structor to veri
+00010430: 6679 2074 6861 7420 616c 6c20 7365 7175  fy that all sequ
+00010440: 656e 6365 7320 6861 7665 2074 6865 2073  ences have the s
+00010450: 616d 6520 6c65 6e67 7468 2c20 6966 206e  ame length, if n
+00010460: 6f74 2c20 7261 6973 6520 616e 2060 6041  ot, raise an ``A
+00010470: 6c69 676e 6d65 6e74 4572 726f 7260 6020  lignmentError`` 
+00010480: 6578 6365 7074 696f 6e2e 2020 5570 6461  exception.  Upda
+00010490: 7465 6420 7468 6520 6d65 7468 6f64 2060  ted the method `
+000104a0: 6041 6c69 676e 6d65 6e74 2e73 7562 616c  `Alignment.subal
+000104b0: 6967 6e6d 656e 7460 6020 746f 2063 616c  ignment`` to cal
+000104c0: 6375 6c61 7465 2074 6865 2069 6e64 6963  culate the indic
+000104d0: 6573 206f 6e6c 7920 6f6e 6365 206e 6f77  es only once now
+000104e0: 2074 6861 7420 6964 656e 7469 6361 6c20   that identical 
+000104f0: 7365 7175 656e 6365 206c 656e 6774 6820  sequence length 
+00010500: 6973 2067 7561 7261 6e74 6565 642e 0a0a  is guaranteed...
+00010510: 2323 2320 4465 7072 6563 6174 6564 2066  ### Deprecated f
+00010520: 756e 6374 696f 6e61 6c69 7479 0a2a 2044  unctionality.* D
+00010530: 6570 7265 6361 7465 6420 6060 636f 6e73  eprecated ``cons
+00010540: 7472 7563 746f 7260 6020 7061 7261 6d65  tructor`` parame
+00010550: 7465 7220 696e 2060 6041 6c69 676e 6d65  ter in ``Alignme
+00010560: 6e74 2e6d 616a 6f72 6974 795f 636f 6e73  nt.majority_cons
+00010570: 656e 7375 7360 6020 696e 2066 6176 6f72  ensus`` in favor
+00010580: 206f 6620 6861 7669 6e67 2075 7365 7273   of having users
+00010590: 2063 616c 6c20 6060 7374 7260 6020 6f6e   call ``str`` on
+000105a0: 2074 6865 2072 6574 7572 6e65 6420 6060   the returned ``
+000105b0: 4269 6f6c 6f67 6963 616c 5365 7175 656e  BiologicalSequen
+000105c0: 6365 6060 2e20 5468 6973 2070 6172 616d  ce``. This param
+000105d0: 6574 6572 2077 696c 6c20 6265 2072 656d  eter will be rem
+000105e0: 6f76 6564 2069 6e20 7363 696b 6974 2d62  oved in scikit-b
+000105f0: 696f 2030 2e33 2e30 2e0a 0a2a 2045 7869  io 0.3.0...* Exi
+00010600: 7374 696e 6720 492f 4f20 6675 6e63 7469  sting I/O functi
+00010610: 6f6e 616c 6974 7920 6465 7072 6563 6174  onality deprecat
+00010620: 6564 2069 6e20 6661 766f 7220 6f66 2049  ed in favor of I
+00010630: 2f4f 2072 6567 6973 7472 792c 206f 6c64  /O registry, old
+00010640: 2066 756e 6374 696f 6e61 6c69 7479 2077   functionality w
+00010650: 696c 6c20 6265 2072 656d 6f76 6564 2069  ill be removed i
+00010660: 6e20 7363 696b 6974 2d62 696f 2030 2e33  n scikit-bio 0.3
+00010670: 2e30 2e20 416c 6c20 6675 6e63 7469 6f6e  .0. All function
+00010680: 616c 6974 7920 6361 6e20 6265 2066 6f75  ality can be fou
+00010690: 6e64 2061 7420 6060 736b 6269 6f2e 696f  nd at ``skbio.io
+000106a0: 2e72 6561 6460 602c 2060 6073 6b62 696f  .read``, ``skbio
+000106b0: 2e69 6f2e 7772 6974 6560 602c 2061 6e64  .io.write``, and
+000106c0: 2074 6865 206d 6574 686f 6473 206c 6973   the methods lis
+000106d0: 7465 6420 6265 6c6f 773a 0a20 2020 202a  ted below:.    *
+000106e0: 2044 6570 7265 6361 7465 6420 7468 6520   Deprecated the 
+000106f0: 666f 6c6c 6f77 696e 6720 2263 6c75 7374  following "clust
+00010700: 616c 2220 7265 6164 6572 732f 7772 6974  al" readers/writ
+00010710: 6572 733a 0a20 2020 2020 2020 202d 2060  ers:.        - `
+00010720: 6077 7269 7465 5f63 6c75 7374 616c 6060  `write_clustal``
+00010730: 202d 3e20 6060 416c 6967 6e6d 656e 742e   -> ``Alignment.
+00010740: 7772 6974 6560 600a 2020 2020 2020 2020  write``.        
+00010750: 2d20 6060 7061 7273 655f 636c 7573 7461  - ``parse_clusta
+00010760: 6c60 6020 2d3e 2060 6041 6c69 676e 6d65  l`` -> ``Alignme
+00010770: 6e74 2e72 6561 6460 600a 0a20 2020 202a  nt.read``..    *
+00010780: 2044 6570 7265 6361 7465 6420 7468 6520   Deprecated the 
+00010790: 666f 6c6c 6f77 696e 6720 6469 7374 616e  following distan
+000107a0: 6365 206d 6174 7269 7820 666f 726d 6174  ce matrix format
+000107b0: 2028 226c 736d 6174 2229 2072 6561 6465   ("lsmat") reade
+000107c0: 7273 2f77 7269 7465 7273 3a0a 2020 2020  rs/writers:.    
+000107d0: 2020 2020 2d20 6060 4469 7373 696d 696c      - ``Dissimil
+000107e0: 6172 6974 794d 6174 7269 782e 6672 6f6d  arityMatrix.from
+000107f0: 5f66 696c 6560 6020 2d3e 2060 6044 6973  _file`` -> ``Dis
+00010800: 7369 6d69 6c61 7269 7479 4d61 7472 6978  similarityMatrix
+00010810: 2e72 6561 6460 600a 2020 2020 2020 2020  .read``.        
+00010820: 2d20 6060 4469 7373 696d 696c 6172 6974  - ``Dissimilarit
+00010830: 794d 6174 7269 782e 746f 5f66 696c 6560  yMatrix.to_file`
+00010840: 6020 2d3e 2060 6044 6973 7369 6d69 6c61  ` -> ``Dissimila
+00010850: 7269 7479 4d61 7472 6978 2e77 7269 7465  rityMatrix.write
+00010860: 6060 0a20 2020 2020 2020 202d 2060 6044  ``.        - ``D
+00010870: 6973 7461 6e63 654d 6174 7269 782e 6672  istanceMatrix.fr
+00010880: 6f6d 5f66 696c 6560 6020 2d3e 2060 6044  om_file`` -> ``D
+00010890: 6973 7461 6e63 654d 6174 7269 782e 7265  istanceMatrix.re
+000108a0: 6164 6060 0a20 2020 2020 2020 202d 2060  ad``.        - `
+000108b0: 6044 6973 7461 6e63 654d 6174 7269 782e  `DistanceMatrix.
+000108c0: 746f 5f66 696c 6560 6020 2d3e 2060 6044  to_file`` -> ``D
+000108d0: 6973 7461 6e63 654d 6174 7269 782e 7772  istanceMatrix.wr
+000108e0: 6974 6560 600a 0a20 2020 202a 2044 6570  ite``..    * Dep
+000108f0: 7265 6361 7465 6420 7468 6520 666f 6c6c  recated the foll
+00010900: 6f77 696e 6720 6f72 6469 6e61 7469 6f6e  owing ordination
+00010910: 2066 6f72 6d61 7420 2822 6f72 6469 6e61   format ("ordina
+00010920: 7469 6f6e 2229 2072 6561 6465 7273 2f77  tion") readers/w
+00010930: 7269 7465 7273 3a0a 2020 2020 2020 2020  riters:.        
+00010940: 2d20 6060 4f72 6469 6e61 7469 6f6e 5265  - ``OrdinationRe
+00010950: 7375 6c74 732e 6672 6f6d 5f66 696c 6560  sults.from_file`
+00010960: 6020 2d3e 2060 604f 7264 696e 6174 696f  ` -> ``Ordinatio
+00010970: 6e52 6573 756c 7473 2e72 6561 6460 600a  nResults.read``.
+00010980: 2020 2020 2020 2020 2d20 6060 4f72 6469          - ``Ordi
+00010990: 6e61 7469 6f6e 5265 7375 6c74 732e 746f  nationResults.to
+000109a0: 5f66 696c 6560 6020 2d3e 2060 604f 7264  _file`` -> ``Ord
+000109b0: 696e 6174 696f 6e52 6573 756c 7473 2e77  inationResults.w
+000109c0: 7269 7465 6060 0a0a 2020 2020 2a20 4465  rite``..    * De
+000109d0: 7072 6563 6174 6564 2074 6865 2066 6f6c  precated the fol
+000109e0: 6c6f 7769 6e67 2022 6e65 7769 636b 2220  lowing "newick" 
+000109f0: 7265 6164 6572 732f 7772 6974 6572 733a  readers/writers:
+00010a00: 0a20 2020 2020 2020 202d 2060 6054 7265  .        - ``Tre
+00010a10: 654e 6f64 652e 6672 6f6d 5f66 696c 6560  eNode.from_file`
+00010a20: 6020 2d3e 2060 6054 7265 654e 6f64 652e  ` -> ``TreeNode.
+00010a30: 7265 6164 6060 0a20 2020 2020 2020 202d  read``.        -
+00010a40: 2060 6054 7265 654e 6f64 652e 6672 6f6d   ``TreeNode.from
+00010a50: 5f6e 6577 6963 6b60 6020 2d3e 2060 6054  _newick`` -> ``T
+00010a60: 7265 654e 6f64 652e 7265 6164 6060 0a20  reeNode.read``. 
+00010a70: 2020 2020 2020 202d 2060 6054 7265 654e         - ``TreeN
+00010a80: 6f64 652e 746f 5f6e 6577 6963 6b60 6020  ode.to_newick`` 
+00010a90: 2d3e 2060 6054 7265 654e 6f64 652e 7772  -> ``TreeNode.wr
+00010aa0: 6974 6560 600a 0a20 2020 202a 2044 6570  ite``..    * Dep
+00010ab0: 7265 6361 7465 6420 7468 6520 666f 6c6c  recated the foll
+00010ac0: 6f77 696e 6720 2270 6879 6c69 7022 2077  owing "phylip" w
+00010ad0: 7269 7465 7273 3a0a 2020 2020 2020 2020  riters:.        
+00010ae0: 2d20 6060 416c 6967 6e6d 656e 742e 746f  - ``Alignment.to
+00010af0: 5f70 6879 6c69 7060 6020 2d3e 2060 6041  _phylip`` -> ``A
+00010b00: 6c69 676e 6d65 6e74 2e77 7269 7465 6060  lignment.write``
+00010b10: 0a0a 2020 2020 2a20 4465 7072 6563 6174  ..    * Deprecat
+00010b20: 6564 2074 6865 2066 6f6c 6c6f 7769 6e67  ed the following
+00010b30: 2022 6661 7374 6122 2f51 5541 4c20 7265   "fasta"/QUAL re
+00010b40: 6164 6572 732f 7772 6974 6572 733a 0a20  aders/writers:. 
+00010b50: 2020 2020 2020 202d 2060 6053 6571 7565         - ``Seque
+00010b60: 6e63 6543 6f6c 6c65 6374 696f 6e2e 6672  nceCollection.fr
+00010b70: 6f6d 5f66 6173 7461 5f72 6563 6f72 6473  om_fasta_records
+00010b80: 6060 202d 3e20 6060 5365 7175 656e 6365  `` -> ``Sequence
+00010b90: 436f 6c6c 6563 7469 6f6e 2e72 6561 6460  Collection.read`
+00010ba0: 600a 2020 2020 2020 2020 2d20 6060 5365  `.        - ``Se
+00010bb0: 7175 656e 6365 436f 6c6c 6563 7469 6f6e  quenceCollection
+00010bc0: 2e74 6f5f 6661 7374 6160 6020 2d3e 2060  .to_fasta`` -> `
+00010bd0: 6053 6571 7565 6e63 6543 6f6c 6c65 6374  `SequenceCollect
+00010be0: 696f 6e2e 7772 6974 6560 600a 2020 2020  ion.write``.    
+00010bf0: 2020 2020 2d20 6060 6661 7374 615f 6672      - ``fasta_fr
+00010c00: 6f6d 5f73 6571 7565 6e63 6573 6060 202d  om_sequences`` -
+00010c10: 3e20 6060 736b 6269 6f2e 696f 2e77 7269  > ``skbio.io.wri
+00010c20: 7465 286f 626a 2c20 696e 746f 3d3c 6669  te(obj, into=<fi
+00010c30: 6c65 3e2c 2066 6f72 6d61 743d 2766 6173  le>, format='fas
+00010c40: 7461 2729 6060 0a20 2020 2020 2020 202d  ta')``.        -
+00010c50: 2060 6066 6173 7461 5f66 726f 6d5f 616c   ``fasta_from_al
+00010c60: 6967 6e6d 656e 7460 6020 2d3e 2060 6041  ignment`` -> ``A
+00010c70: 6c69 676e 6d65 6e74 2e77 7269 7465 6060  lignment.write``
+00010c80: 0a20 2020 2020 2020 202d 2060 6070 6172  .        - ``par
+00010c90: 7365 5f66 6173 7461 6060 202d 3e20 6060  se_fasta`` -> ``
+00010ca0: 736b 6269 6f2e 696f 2e72 6561 6428 3c66  skbio.io.read(<f
+00010cb0: 6173 7461 3e2c 2066 6f72 6d61 743d 2766  asta>, format='f
+00010cc0: 6173 7461 2729 6060 0a20 2020 2020 2020  asta')``.       
+00010cd0: 202d 2060 6070 6172 7365 5f71 7561 6c60   - ``parse_qual`
+00010ce0: 6020 2d3e 2060 6073 6b62 696f 2e69 6f2e  ` -> ``skbio.io.
+00010cf0: 7265 6164 283c 6661 7374 613e 2c20 666f  read(<fasta>, fo
+00010d00: 726d 6174 3d27 6661 7374 6127 2c20 7175  rmat='fasta', qu
+00010d10: 616c 3d3c 6669 6c65 3e29 6060 0a20 2020  al=<file>)``.   
+00010d20: 2020 2020 202d 2060 6042 696f 6c6f 6769       - ``Biologi
+00010d30: 6361 6c53 6571 7565 6e63 652e 746f 5f66  calSequence.to_f
+00010d40: 6173 7461 6060 202d 3e20 6060 4269 6f6c  asta`` -> ``Biol
+00010d50: 6f67 6963 616c 5365 7175 656e 6365 2e77  ogicalSequence.w
+00010d60: 7269 7465 6060 0a0a 2020 2020 2a20 4465  rite``..    * De
+00010d70: 7072 6563 6174 6564 2074 6865 2066 6f6c  precated the fol
+00010d80: 6c6f 7769 6e67 2022 6661 7374 7122 2072  lowing "fastq" r
+00010d90: 6561 6465 7273 2f77 7269 7465 7273 3a0a  eaders/writers:.
+00010da0: 2020 2020 2020 2020 2d20 6060 7061 7273          - ``pars
+00010db0: 655f 6661 7374 7160 6020 2d3e 2060 6073  e_fastq`` -> ``s
+00010dc0: 6b62 696f 2e69 6f2e 7265 6164 283c 6661  kbio.io.read(<fa
+00010dd0: 7374 713e 2c20 666f 726d 6174 3d27 6661  stq>, format='fa
+00010de0: 7374 7127 2960 600a 2020 2020 2020 2020  stq')``.        
+00010df0: 2d20 6060 666f 726d 6174 5f66 6173 7471  - ``format_fastq
+00010e00: 5f72 6563 6f72 6460 6020 2d3e 2060 6073  _record`` -> ``s
+00010e10: 6b62 696f 2e69 6f2e 7772 6974 6528 3c66  kbio.io.write(<f
+00010e20: 6173 7471 3e2c 2066 6f72 6d61 743d 2766  astq>, format='f
+00010e30: 6173 7471 2729 6060 0a0a 2323 2320 4261  astq')``..### Ba
+00010e40: 636b 7761 7264 2d69 6e63 6f6d 7061 7469  ckward-incompati
+00010e50: 626c 6520 6368 616e 6765 730a 0a2a 2060  ble changes..* `
+00010e60: 6073 6b62 696f 2e73 7461 7473 2e64 6973  `skbio.stats.dis
+00010e70: 7461 6e63 652e 6d61 6e74 656c 6060 206e  tance.mantel`` n
+00010e80: 6f77 2072 6574 7572 6e73 2061 2033 2d65  ow returns a 3-e
+00010e90: 6c65 6d65 6e74 2074 7570 6c65 2063 6f6e  lement tuple con
+00010ea0: 7461 696e 696e 6720 636f 7272 656c 6174  taining correlat
+00010eb0: 696f 6e20 636f 6566 6669 6369 656e 742c  ion coefficient,
+00010ec0: 2070 2d76 616c 7565 2c20 616e 6420 7468   p-value, and th
+00010ed0: 6520 6e75 6d62 6572 206f 6620 6d61 7463  e number of matc
+00010ee0: 6869 6e67 2072 6f77 732f 636f 6c73 2069  hing rows/cols i
+00010ef0: 6e20 7468 6520 6469 7374 616e 6365 206d  n the distance m
+00010f00: 6174 7269 6365 7320 2860 606e 6060 292e  atrices (``n``).
+00010f10: 2054 6865 2072 6574 7572 6e20 7661 6c75   The return valu
+00010f20: 6520 7761 7320 7072 6576 696f 7573 6c79  e was previously
+00010f30: 2061 2032 2d65 6c65 6d65 6e74 2074 7570   a 2-element tup
+00010f40: 6c65 2063 6f6e 7461 696e 696e 6720 6f6e  le containing on
+00010f50: 6c79 2074 6865 2063 6f72 7265 6c61 7469  ly the correlati
+00010f60: 6f6e 2063 6f65 6666 6963 6965 6e74 2061  on coefficient a
+00010f70: 6e64 2070 2d76 616c 7565 2e0a 2a20 6060  nd p-value..* ``
+00010f80: 736b 6269 6f2e 7374 6174 732e 6469 7374  skbio.stats.dist
+00010f90: 616e 6365 2e6d 616e 7465 6c60 6020 7265  ance.mantel`` re
+00010fa0: 6f72 6465 7273 2069 6e70 7574 2060 6044  orders input ``D
+00010fb0: 6973 7461 6e63 654d 6174 7269 7860 6020  istanceMatrix`` 
+00010fc0: 696e 7374 616e 6365 7320 6261 7365 6420  instances based 
+00010fd0: 6f6e 206d 6174 6368 696e 6720 4944 7320  on matching IDs 
+00010fe0: 2873 6565 206f 7074 696f 6e61 6c20 7061  (see optional pa
+00010ff0: 7261 6d65 7465 7273 2060 6073 7472 6963  rameters ``stric
+00011000: 7460 6020 616e 6420 6060 6c6f 6f6b 7570  t`` and ``lookup
+00011010: 6060 2066 6f72 2063 6f6e 7472 6f6c 6c69  `` for controlli
+00011020: 6e67 2074 6869 7320 6265 6861 7669 6f72  ng this behavior
+00011030: 292e 2049 6e20 7468 6520 7061 7374 2c20  ). In the past, 
+00011040: 6060 4469 7374 616e 6365 4d61 7472 6978  ``DistanceMatrix
+00011050: 6060 2069 6e73 7461 6e63 6573 2077 6572  `` instances wer
+00011060: 6520 7472 6561 7465 6420 7468 6520 7361  e treated the sa
+00011070: 6d65 2061 7320 6060 6172 7261 795f 6c69  me as ``array_li
+00011080: 6b65 6060 2069 6e70 7574 2061 6e64 206e  ke`` input and n
+00011090: 6f20 7265 6f72 6465 7269 6e67 2074 6f6f  o reordering too
+000110a0: 6b20 706c 6163 652c 2072 6567 6172 646c  k place, regardl
+000110b0: 6573 7320 6f66 2049 4420 286d 6973 296d  ess of ID (mis)m
+000110c0: 6174 6368 6573 2e20 6060 6172 7261 795f  atches. ``array_
+000110d0: 6c69 6b65 6060 2069 6e70 7574 2062 6568  like`` input beh
+000110e0: 6176 696f 7220 7265 6d61 696e 7320 7468  avior remains th
+000110f0: 6520 7361 6d65 2e0a 2a20 4966 206d 6973  e same..* If mis
+00011100: 6d61 7463 6865 6420 7479 7065 7320 6172  matched types ar
+00011110: 6520 7072 6f76 6964 6564 2074 6f20 6060  e provided to ``
+00011120: 736b 6269 6f2e 7374 6174 732e 6469 7374  skbio.stats.dist
+00011130: 616e 6365 2e6d 616e 7465 6c60 6020 2865  ance.mantel`` (e
+00011140: 2e67 2e2c 2061 2060 6044 6973 7461 6e63  .g., a ``Distanc
+00011150: 654d 6174 7269 7860 6020 616e 6420 6060  eMatrix`` and ``
+00011160: 6172 7261 795f 6c69 6b65 6060 292c 2061  array_like``), a
+00011170: 2060 6054 7970 6545 7272 6f72 6060 2077   ``TypeError`` w
+00011180: 696c 6c20 6265 2072 6169 7365 642e 0a0a  ill be raised...
+00011190: 2323 2320 4d69 7363 656c 6c61 6e65 6f75  ### Miscellaneou
+000111a0: 730a 0a2a 2041 6464 6564 2067 6974 2074  s..* Added git t
+000111b0: 696d 6573 7461 6d70 2063 6865 636b 696e  imestamp checkin
+000111c0: 6720 746f 2063 6865 636b 6c69 7374 2e70  g to checklist.p
+000111d0: 792c 2065 6e73 7572 696e 6720 7468 6174  y, ensuring that
+000111e0: 2077 6865 6e20 6368 616e 6765 7320 6172   when changes ar
+000111f0: 6520 6d61 6465 2074 6f20 4379 7468 6f6e  e made to Cython
+00011200: 2028 2e70 7978 2920 6669 6c65 732c 2074   (.pyx) files, t
+00011210: 6865 6972 2063 6f72 7265 7370 6f6e 6469  heir correspondi
+00011220: 6e67 2067 656e 6572 6174 6564 2043 2066  ng generated C f
+00011230: 696c 6573 2061 7265 2061 6c73 6f20 7570  iles are also up
+00011240: 6461 7465 642e 0a2a 2046 6978 6564 2070  dated..* Fixed p
+00011250: 6572 666f 726d 616e 6365 2062 7567 2077  erformance bug w
+00011260: 6865 6e20 696e 7374 616e 7469 6174 696e  hen instantiatin
+00011270: 6720 6060 4269 6f6c 6f67 6963 616c 5365  g ``BiologicalSe
+00011280: 7175 656e 6365 6060 206f 626a 6563 7473  quence`` objects
+00011290: 2e20 5468 6520 7072 6576 696f 7573 2072  . The previous r
+000112a0: 756e 7469 6d65 2073 6361 6c65 6420 6c69  untime scaled li
+000112b0: 6e65 6172 6c79 2077 6974 6820 7365 7175  nearly with sequ
+000112c0: 656e 6365 206c 656e 6774 683b 2069 7420  ence length; it 
+000112d0: 6973 206e 6f77 2063 6f6e 7374 616e 7420  is now constant 
+000112e0: 7469 6d65 2077 6865 6e20 7468 6520 7365  time when the se
+000112f0: 7175 656e 6365 2069 7320 616c 7265 6164  quence is alread
+00011300: 7920 6120 7374 7269 6e67 2e20 5365 6520  y a string. See 
+00011310: 5b23 3632 335d 2868 7474 7073 3a2f 2f67  [#623](https://g
+00011320: 6974 6875 622e 636f 6d2f 6269 6f63 6f72  ithub.com/biocor
+00011330: 652f 7363 696b 6974 2d62 696f 2f69 7373  e/scikit-bio/iss
+00011340: 7565 732f 3632 3329 2066 6f72 2064 6574  ues/623) for det
+00011350: 6169 6c73 2e0a 2a20 4950 7974 686f 6e20  ails..* IPython 
+00011360: 616e 6420 7369 7820 6172 6520 6e6f 7720  and six are now 
+00011370: 7265 7175 6972 6564 2064 6570 656e 6465  required depende
+00011380: 6e63 6965 732e 0a0a 2323 2056 6572 7369  ncies...## Versi
+00011390: 6f6e 2030 2e32 2e30 2028 3230 3134 2d30  on 0.2.0 (2014-0
+000113a0: 382d 3037 290a 0a54 6869 7320 6973 2061  8-07)..This is a
+000113b0: 6e20 696e 6974 6961 6c20 616c 7068 6120  n initial alpha 
+000113c0: 7265 6c65 6173 6520 6f66 2073 6369 6b69  release of sciki
+000113d0: 742d 6269 6f2e 2041 7420 7468 6973 2073  t-bio. At this s
+000113e0: 7461 6765 2c20 6d61 6a6f 7220 6261 636b  tage, major back
+000113f0: 7761 7264 732d 696e 636f 6d70 6174 6962  wards-incompatib
+00011400: 6c65 2041 5049 2063 6861 6e67 6573 2063  le API changes c
+00011410: 616e 2061 6e64 2077 696c 6c20 6861 7070  an and will happ
+00011420: 656e 2e20 4d61 6e79 2062 6163 6b77 6172  en. Many backwar
+00011430: 6473 2d69 6e63 6f6d 7061 7469 626c 6520  ds-incompatible 
+00011440: 4150 4920 6368 616e 6765 7320 7765 7265  API changes were
+00011450: 206d 6164 6520 7369 6e63 6520 7468 6520   made since the 
+00011460: 7072 6576 696f 7573 2072 656c 6561 7365  previous release
+00011470: 2e0a 0a23 2323 2046 6561 7475 7265 730a  ...### Features.
+00011480: 0a2a 2041 6464 6564 2061 6269 6c69 7479  .* Added ability
+00011490: 2074 6f20 636f 6d70 7574 6520 6469 7374   to compute dist
+000114a0: 616e 6365 7320 6265 7477 6565 6e20 7365  ances between se
+000114b0: 7175 656e 6365 7320 696e 2061 2060 6053  quences in a ``S
+000114c0: 6571 7565 6e63 6543 6f6c 6c65 6374 696f  equenceCollectio
+000114d0: 6e60 6020 6f62 6a65 6374 2028 5b23 3530  n`` object ([#50
+000114e0: 395d 2868 7474 7073 3a2f 2f67 6974 6875  9](https://githu
+000114f0: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
+00011500: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
+00011510: 3530 3929 292c 2061 6e64 2065 7870 616e  509)), and expan
+00011520: 6465 6420 6060 416c 6967 6e6d 656e 742e  ded ``Alignment.
+00011530: 6469 7374 616e 6365 6060 2074 6f20 616c  distance`` to al
+00011540: 6c6f 7720 7468 6520 7573 6572 2074 6f20  low the user to 
+00011550: 7061 7373 2061 2066 756e 6374 696f 6e20  pass a function 
+00011560: 666f 7220 636f 6d70 7574 696e 6720 6469  for computing di
+00011570: 7374 616e 6365 7320 2874 6865 2064 6566  stances (the def
+00011580: 6175 6c74 2064 6973 7461 6e63 6520 6d65  ault distance me
+00011590: 7472 6963 2069 7320 7374 696c 6c20 6060  tric is still ``
+000115a0: 7363 6970 792e 7370 6174 6961 6c2e 6469  scipy.spatial.di
+000115b0: 7374 616e 6365 2e68 616d 6d69 6e67 6060  stance.hamming``
+000115c0: 2920 285b 2331 3934 5d28 6874 7470 733a  ) ([#194](https:
+000115d0: 2f2f 6769 7468 7562 2e63 6f6d 2f62 696f  //github.com/bio
+000115e0: 636f 7265 2f73 6369 6b69 742d 6269 6f2f  core/scikit-bio/
+000115f0: 6973 7375 6573 2f31 3934 2929 2e0a 2a20  issues/194))..* 
+00011600: 4164 6465 6420 6675 6e63 7469 6f6e 616c  Added functional
+00011610: 6974 7920 746f 206e 6f74 2070 656e 616c  ity to not penal
+00011620: 697a 6520 7465 726d 696e 616c 2067 6170  ize terminal gap
+00011630: 7320 696e 2067 6c6f 6261 6c20 616c 6967  s in global alig
+00011640: 6e6d 656e 742e 2054 6869 7320 6675 6e63  nment. This func
+00011650: 7469 6f6e 616c 6974 7920 7265 7375 6c74  tionality result
+00011660: 7320 696e 206d 6f72 6520 6269 6f6c 6f67  s in more biolog
+00011670: 6963 616c 6c79 2072 656c 6576 616e 7420  ically relevant 
+00011680: 676c 6f62 616c 2061 6c69 676e 6d65 6e74  global alignment
+00011690: 7320 2873 6565 205b 2335 3337 5d28 6874  s (see [#537](ht
+000116a0: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+000116b0: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
+000116c0: 6269 6f2f 6973 7375 6573 2f35 3337 2920  bio/issues/537) 
+000116d0: 666f 7220 6469 7363 7573 7369 6f6e 206f  for discussion o
+000116e0: 6620 7468 6520 6973 7375 6529 2061 6e64  f the issue) and
+000116f0: 2069 7320 6e6f 7720 7468 6520 6465 6661   is now the defa
+00011700: 756c 7420 6265 6861 7669 6f72 2066 6f72  ult behavior for
+00011710: 2067 6c6f 6261 6c20 616c 6967 6e6d 656e   global alignmen
+00011720: 742e 0a2a 2054 6865 2070 7974 686f 6e20  t..* The python 
+00011730: 676c 6f62 616c 2061 6c69 676e 6572 7320  global aligners 
+00011740: 2860 6067 6c6f 6261 6c5f 7061 6972 7769  (``global_pairwi
+00011750: 7365 5f61 6c69 676e 6060 2c20 6060 676c  se_align``, ``gl
+00011760: 6f62 616c 5f70 6169 7277 6973 655f 616c  obal_pairwise_al
+00011770: 6967 6e5f 6e75 636c 656f 7469 6465 6060  ign_nucleotide``
+00011780: 2c20 616e 6420 6060 676c 6f62 616c 5f70  , and ``global_p
+00011790: 6169 7277 6973 655f 616c 6967 6e5f 7072  airwise_align_pr
+000117a0: 6f74 6569 6e60 6029 206e 6f77 2073 7570  otein``) now sup
+000117b0: 706f 7274 2061 6c69 676e 696e 6720 7061  port aligning pa
+000117c0: 6972 7320 6f66 2073 6571 7565 6e63 6573  irs of sequences
+000117d0: 2c20 7061 6972 7320 6f66 2061 6c69 676e  , pairs of align
+000117e0: 6d65 6e74 732c 2061 6e64 2061 2073 6571  ments, and a seq
+000117f0: 7565 6e63 6520 616e 6420 616e 2061 6c69  uence and an ali
+00011800: 676e 6d65 6e74 2028 7365 6520 5b23 3535  gnment (see [#55
+00011810: 305d 2868 7474 7073 3a2f 2f67 6974 6875  0](https://githu
+00011820: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
+00011830: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
+00011840: 3535 3029 292e 2054 6869 7320 6675 6e63  550)). This func
+00011850: 7469 6f6e 616c 6974 7920 7375 7070 6f72  tionality suppor
+00011860: 7473 2070 726f 6772 6573 7369 7665 206d  ts progressive m
+00011870: 756c 7469 706c 6520 7365 7175 656e 6365  ultiple sequence
+00011880: 2061 6c69 676e 6d65 6e74 2c20 616d 6f6e   alignment, amon
+00011890: 6720 6f74 6865 7220 7468 696e 6773 2073  g other things s
+000118a0: 7563 6820 6173 2061 6464 696e 6720 6120  uch as adding a 
+000118b0: 7365 7175 656e 6365 2074 6f20 616e 2065  sequence to an e
+000118c0: 7869 7374 696e 6720 616c 6967 6e6d 656e  xisting alignmen
+000118d0: 742e 0a2a 2041 6464 6564 2060 6053 746f  t..* Added ``Sto
+000118e0: 636b 686f 6c6d 416c 6967 6e6d 656e 742e  ckholmAlignment.
+000118f0: 746f 5f66 696c 6560 6020 666f 7220 7772  to_file`` for wr
+00011900: 6974 696e 6720 5374 6f63 6b68 6f6c 6d2d  iting Stockholm-
+00011910: 666f 726d 6174 7465 6420 6669 6c65 732e  formatted files.
+00011920: 0a2a 2041 6464 6564 2060 6073 7472 6963  .* Added ``stric
+00011930: 743d 5472 7565 6060 206f 7074 696f 6e61  t=True`` optiona
+00011940: 6c20 7061 7261 6d65 7465 7220 746f 2060  l parameter to `
+00011950: 6044 6973 7369 6d69 6c61 7269 7479 4d61  `DissimilarityMa
+00011960: 7472 6978 2e66 696c 7465 7260 602e 0a2a  trix.filter``..*
+00011970: 2041 6464 6564 2060 6054 7265 654e 6f64   Added ``TreeNod
+00011980: 652e 6669 6e64 5f61 6c6c 6060 2066 6f72  e.find_all`` for
+00011990: 2066 696e 6469 6e67 2061 6c6c 2074 7265   finding all tre
+000119a0: 6520 6e6f 6465 7320 7468 6174 206d 6174  e nodes that mat
+000119b0: 6368 2061 2067 6976 656e 206e 616d 652e  ch a given name.
+000119c0: 0a0a 0a23 2323 2042 7567 2066 6978 6573  ...### Bug fixes
+000119d0: 0a0a 2a20 4669 7865 6420 6275 6720 7468  ..* Fixed bug th
+000119e0: 6174 2072 6573 756c 7465 6420 696e 2061  at resulted in a
+000119f0: 2060 6056 616c 7565 4572 726f 7260 6020   ``ValueError`` 
+00011a00: 6672 6f6d 2060 606c 6f63 616c 5f61 6c69  from ``local_ali
+00011a10: 676e 5f70 6169 7277 6973 655f 6e75 636c  gn_pairwise_nucl
+00011a20: 656f 7469 6465 6060 2028 7365 6520 5b23  eotide`` (see [#
+00011a30: 3530 345d 2868 7474 7073 3a2f 2f67 6974  504](https://git
+00011a40: 6875 622e 636f 6d2f 6269 6f63 6f72 652f  hub.com/biocore/
+00011a50: 7363 696b 6974 2d62 696f 2f69 7373 7565  scikit-bio/issue
+00011a60: 732f 3530 3429 2920 756e 6465 7220 6d61  s/504)) under ma
+00011a70: 6e79 2063 6972 6375 6d73 7461 6e63 6573  ny circumstances
+00011a80: 2e20 5468 6973 2077 6f75 6c64 206e 6f74  . This would not
+00011a90: 2067 656e 6572 6174 6520 696e 636f 7272   generate incorr
+00011aa0: 6563 7420 7265 7375 6c74 732c 2062 7574  ect results, but
+00011ab0: 2077 6f75 6c64 2063 6175 7365 2074 6865   would cause the
+00011ac0: 2063 6f64 6520 746f 2066 6169 6c2e 0a0a   code to fail...
+00011ad0: 2323 2320 4261 636b 7761 7264 2d69 6e63  ### Backward-inc
+00011ae0: 6f6d 7061 7469 626c 6520 6368 616e 6765  ompatible change
+00011af0: 730a 0a2a 2052 656d 6f76 6564 2060 6073  s..* Removed ``s
+00011b00: 6b62 696f 2e6d 6174 6860 602c 206c 6561  kbio.math``, lea
+00011b10: 7669 6e67 2060 6073 7461 7473 6060 2061  ving ``stats`` a
+00011b20: 6e64 2060 6064 6976 6572 7369 7479 6060  nd ``diversity``
+00011b30: 2074 6f20 6265 636f 6d65 2074 6f70 206c   to become top l
+00011b40: 6576 656c 2070 6163 6b61 6765 732e 2046  evel packages. F
+00011b50: 6f72 2065 7861 6d70 6c65 2c20 696e 7374  or example, inst
+00011b60: 6561 6420 6f66 2060 6066 726f 6d20 736b  ead of ``from sk
+00011b70: 6269 6f2e 6d61 7468 2e73 7461 7473 2e6f  bio.math.stats.o
+00011b80: 7264 696e 6174 696f 6e20 696d 706f 7274  rdination import
+00011b90: 2050 436f 4160 6020 796f 7520 776f 756c   PCoA`` you woul
+00011ba0: 6420 6e6f 7720 696d 706f 7274 2060 6066  d now import ``f
+00011bb0: 726f 6d20 736b 6269 6f2e 7374 6174 732e  rom skbio.stats.
+00011bc0: 6f72 6469 6e61 7469 6f6e 2069 6d70 6f72  ordination impor
+00011bd0: 7420 5043 6f41 6060 2e0a 2a20 5468 6520  t PCoA``..* The 
+00011be0: 6d6f 6475 6c65 2060 6073 6b62 696f 2e6d  module ``skbio.m
+00011bf0: 6174 682e 6772 6164 6965 6e74 6060 2061  ath.gradient`` a
+00011c00: 7320 7765 6c6c 2061 7320 7468 6520 636f  s well as the co
+00011c10: 6e74 656e 7473 206f 6620 6060 736b 6269  ntents of ``skbi
+00011c20: 6f2e 6d61 7468 2e73 7562 7361 6d70 6c65  o.math.subsample
+00011c30: 6060 2061 6e64 2060 6073 6b62 696f 2e6d  `` and ``skbio.m
+00011c40: 6174 682e 7374 6174 732e 6d69 7363 6060  ath.stats.misc``
+00011c50: 2061 7265 206e 6f77 2066 6f75 6e64 2069   are now found i
+00011c60: 6e20 6060 736b 6269 6f2e 7374 6174 7360  n ``skbio.stats`
+00011c70: 602e 2041 7320 616e 2065 7861 6d70 6c65  `. As an example
+00011c80: 2c20 746f 2069 6d70 6f72 7420 7375 6273  , to import subs
+00011c90: 616d 706c 653a 2060 6066 726f 6d20 736b  ample: ``from sk
+00011ca0: 6269 6f2e 7374 6174 7320 696d 706f 7274  bio.stats import
+00011cb0: 2073 7562 7361 6d70 6c65 6060 3b20 746f   subsample``; to
+00011cc0: 2069 6d70 6f72 7420 6576 6572 7974 6869   import everythi
+00011cd0: 6e67 2066 726f 6d20 6772 6164 6965 6e74  ng from gradient
+00011ce0: 3a20 6060 6672 6f6d 2073 6b62 696f 2e73  : ``from skbio.s
+00011cf0: 7461 7473 2e67 7261 6469 656e 7420 696d  tats.gradient im
+00011d00: 706f 7274 202a 6060 2e0a 2a20 5468 6520  port *``..* The 
+00011d10: 636f 6e74 656e 7473 206f 6620 6060 736b  contents of ``sk
+00011d20: 6269 6f2e 6d61 7468 2e73 7461 7473 2e6f  bio.math.stats.o
+00011d30: 7264 696e 6174 696f 6e2e 7574 696c 7360  rdination.utils`
+00011d40: 6020 6172 6520 6e6f 7720 696e 2060 6073  ` are now in ``s
+00011d50: 6b62 696f 2e73 7461 7473 2e6f 7264 696e  kbio.stats.ordin
+00011d60: 6174 696f 6e60 602e 0a2a 2052 656d 6f76  ation``..* Remov
+00011d70: 6564 2060 6073 6b62 696f 2e61 7070 6060  ed ``skbio.app``
+00011d80: 2073 7562 7061 636b 6167 6520 2869 2e65   subpackage (i.e
+00011d90: 2e2c 2074 6865 202a 6170 706c 6963 6174  ., the *applicat
+00011da0: 696f 6e20 636f 6e74 726f 6c6c 6572 2066  ion controller f
+00011db0: 7261 6d65 776f 726b 2a29 2061 7320 7468  ramework*) as th
+00011dc0: 6973 2063 6f64 6520 6861 7320 6265 656e  is code has been
+00011dd0: 2070 6f72 7465 6420 746f 2074 6865 2073   ported to the s
+00011de0: 7461 6e64 616c 6f6e 6520 5b62 7572 7269  tandalone [burri
+00011df0: 746f 5d28 6874 7470 733a 2f2f 6769 7468  to](https://gith
+00011e00: 7562 2e63 6f6d 2f62 696f 636f 7265 2f62  ub.com/biocore/b
+00011e10: 7572 7269 746f 2920 5079 7468 6f6e 2070  urrito) Python p
+00011e20: 6163 6b61 6765 2e20 5468 6973 2063 6f64  ackage. This cod
+00011e30: 6520 7761 7320 6e6f 7420 7370 6563 6966  e was not specif
+00011e40: 6963 2074 6f20 6269 6f69 6e66 6f72 6d61  ic to bioinforma
+00011e50: 7469 6373 2061 6e64 2069 7320 7573 6566  tics and is usef
+00011e60: 756c 2066 6f72 2077 7261 7070 696e 6720  ul for wrapping 
+00011e70: 636f 6d6d 616e 642d 6c69 6e65 2061 7070  command-line app
+00011e80: 6c69 6361 7469 6f6e 7320 696e 2067 656e  lications in gen
+00011e90: 6572 616c 2e0a 2a20 5265 6d6f 7665 6420  eral..* Removed 
+00011ea0: 6060 736b 6269 6f2e 636f 7265 6060 2c20  ``skbio.core``, 
+00011eb0: 6c65 6176 696e 6720 6060 616c 6967 6e6d  leaving ``alignm
+00011ec0: 656e 7460 602c 2060 6067 656e 6574 6963  ent``, ``genetic
+00011ed0: 5f63 6f64 6560 602c 2060 6073 6571 7565  _code``, ``seque
+00011ee0: 6e63 6560 602c 2060 6074 7265 6560 602c  nce``, ``tree``,
+00011ef0: 2061 6e64 2060 6077 6f72 6b66 6c6f 7760   and ``workflow`
+00011f00: 6020 746f 2062 6563 6f6d 6520 746f 7020  ` to become top 
+00011f10: 6c65 7665 6c20 7061 636b 6167 6573 2e20  level packages. 
+00011f20: 466f 7220 6578 616d 706c 652c 2069 6e73  For example, ins
+00011f30: 7465 6164 206f 6620 6060 6672 6f6d 2073  tead of ``from s
+00011f40: 6b62 696f 2e63 6f72 652e 7365 7175 656e  kbio.core.sequen
+00011f50: 6365 2069 6d70 6f72 7420 444e 4160 6020  ce import DNA`` 
+00011f60: 796f 7520 776f 756c 6420 6e6f 7720 696d  you would now im
+00011f70: 706f 7274 2060 6066 726f 6d20 736b 6269  port ``from skbi
+00011f80: 6f2e 7365 7175 656e 6365 2069 6d70 6f72  o.sequence impor
+00011f90: 7420 444e 4160 602e 0a2a 2052 656d 6f76  t DNA``..* Remov
+00011fa0: 6564 2060 6073 6b62 696f 2e75 7469 6c2e  ed ``skbio.util.
+00011fb0: 6578 6365 7074 696f 6e60 6020 616e 6420  exception`` and 
+00011fc0: 6060 736b 6269 6f2e 7574 696c 2e77 6172  ``skbio.util.war
+00011fd0: 6e69 6e67 6060 2028 7365 6520 5b23 3537  ning`` (see [#57
+00011fe0: 375d 2868 7474 7073 3a2f 2f67 6974 6875  7](https://githu
+00011ff0: 622e 636f 6d2f 6269 6f63 6f72 652f 7363  b.com/biocore/sc
+00012000: 696b 6974 2d62 696f 2f69 7373 7565 732f  ikit-bio/issues/
+00012010: 3537 3729 2066 6f72 2074 6865 2072 6561  577) for the rea
+00012020: 736f 6e69 6e67 2062 6568 696e 6420 7468  soning behind th
+00012030: 6973 2063 6861 6e67 6529 2e20 5468 6520  is change). The 
+00012040: 6578 6365 7074 696f 6e73 2f77 6172 6e69  exceptions/warni
+00012050: 6e67 7320 7765 7265 206d 6f76 6564 2074  ngs were moved t
+00012060: 6f20 7468 6520 666f 6c6c 6f77 696e 6720  o the following 
+00012070: 6c6f 6361 7469 6f6e 733a 0a20 2d20 6060  locations:. - ``
+00012080: 4669 6c65 466f 726d 6174 4572 726f 7260  FileFormatError`
+00012090: 602c 2060 6052 6563 6f72 6445 7272 6f72  `, ``RecordError
+000120a0: 6060 2c20 6060 4669 656c 6445 7272 6f72  ``, ``FieldError
+000120b0: 6060 2c20 616e 6420 6060 4566 6669 6369  ``, and ``Effici
+000120c0: 656e 6379 5761 726e 696e 6760 6020 6861  encyWarning`` ha
+000120d0: 7665 2062 6565 6e20 6d6f 7665 6420 746f  ve been moved to
+000120e0: 2060 6073 6b62 696f 2e75 7469 6c60 600a   ``skbio.util``.
+000120f0: 202d 2060 6042 696f 6c6f 6769 6361 6c53   - ``BiologicalS
+00012100: 6571 7565 6e63 6545 7272 6f72 6060 2068  equenceError`` h
+00012110: 6173 2062 6565 6e20 6d6f 7665 6420 746f  as been moved to
+00012120: 2060 6073 6b62 696f 2e73 6571 7565 6e63   ``skbio.sequenc
+00012130: 6560 600a 202d 2060 6053 6571 7565 6e63  e``. - ``Sequenc
+00012140: 6543 6f6c 6c65 6374 696f 6e45 7272 6f72  eCollectionError
+00012150: 6060 2061 6e64 2060 6053 746f 636b 686f  `` and ``Stockho
+00012160: 6c6d 5061 7273 6545 7272 6f72 6060 2068  lmParseError`` h
+00012170: 6176 6520 6265 656e 206d 6f76 6564 2074  ave been moved t
+00012180: 6f20 6060 736b 6269 6f2e 616c 6967 6e6d  o ``skbio.alignm
+00012190: 656e 7460 600a 202d 2060 6044 6973 7369  ent``. - ``Dissi
+000121a0: 6d69 6c61 7269 7479 4d61 7472 6978 4572  milarityMatrixEr
+000121b0: 726f 7260 602c 2060 6044 6973 7461 6e63  ror``, ``Distanc
+000121c0: 654d 6174 7269 7845 7272 6f72 6060 2c20  eMatrixError``, 
+000121d0: 6060 4469 7373 696d 696c 6172 6974 794d  ``DissimilarityM
+000121e0: 6174 7269 7846 6f72 6d61 7445 7272 6f72  atrixFormatError
+000121f0: 6060 2c20 616e 6420 6060 4d69 7373 696e  ``, and ``Missin
+00012200: 6749 4445 7272 6f72 6060 2068 6176 6520  gIDError`` have 
+00012210: 6265 656e 206d 6f76 6564 2074 6f20 6060  been moved to ``
+00012220: 736b 6269 6f2e 7374 6174 732e 6469 7374  skbio.stats.dist
+00012230: 616e 6365 6060 0a20 2d20 6060 5472 6565  ance``. - ``Tree
+00012240: 4572 726f 7260 602c 2060 604e 6f4c 656e  Error``, ``NoLen
+00012250: 6774 6845 7272 6f72 6060 2c20 6060 4475  gthError``, ``Du
+00012260: 706c 6963 6174 654e 6f64 6545 7272 6f72  plicateNodeError
+00012270: 6060 2c20 6060 4d69 7373 696e 674e 6f64  ``, ``MissingNod
+00012280: 6545 7272 6f72 6060 2c20 616e 6420 6060  eError``, and ``
+00012290: 4e6f 5061 7265 6e74 4572 726f 7260 6020  NoParentError`` 
+000122a0: 6861 7665 2062 6565 6e20 6d6f 7665 6420  have been moved 
+000122b0: 746f 2060 6073 6b62 696f 2e74 7265 6560  to ``skbio.tree`
+000122c0: 600a 202d 2060 6046 6173 7471 5061 7273  `. - ``FastqPars
+000122d0: 6545 7272 6f72 6060 2068 6173 2062 6565  eError`` has bee
+000122e0: 6e20 6d6f 7665 6420 746f 2060 6073 6b62  n moved to ``skb
+000122f0: 696f 2e70 6172 7365 2e73 6571 7565 6e63  io.parse.sequenc
+00012300: 6573 6060 0a20 2d20 6060 4765 6e65 7469  es``. - ``Geneti
+00012310: 6343 6f64 6545 7272 6f72 6060 2c20 6060  cCodeError``, ``
+00012320: 4765 6e65 7469 6343 6f64 6549 6e69 7445  GeneticCodeInitE
+00012330: 7272 6f72 6060 2c20 616e 6420 6060 496e  rror``, and ``In
+00012340: 7661 6c69 6443 6f64 6f6e 4572 726f 7260  validCodonError`
+00012350: 6020 6861 7665 2062 6565 6e20 6d6f 7665  ` have been move
+00012360: 6420 746f 2060 6073 6b62 696f 2e67 656e  d to ``skbio.gen
+00012370: 6574 6963 5f63 6f64 6560 600a 2a20 5468  etic_code``.* Th
+00012380: 6520 636f 6e74 656e 7473 206f 6620 6060  e contents of ``
+00012390: 736b 6269 6f2e 6765 6e65 7469 635f 636f  skbio.genetic_co
+000123a0: 6465 6060 2066 6f72 6d65 726c 7920 6060  de`` formerly ``
+000123b0: 736b 6269 6f2e 636f 7265 2e67 656e 6574  skbio.core.genet
+000123c0: 6963 5f63 6f64 6560 6020 6172 6520 6e6f  ic_code`` are no
+000123d0: 7720 696e 2060 6073 6b62 696f 2e73 6571  w in ``skbio.seq
+000123e0: 7565 6e63 6560 602e 2054 6865 2060 6047  uence``. The ``G
+000123f0: 656e 6574 6963 436f 6465 7360 6020 6469  eneticCodes`` di
+00012400: 6374 696f 6e61 7279 2069 7320 6e6f 7720  ctionary is now 
+00012410: 6120 6675 6e63 7469 6f6e 2060 6067 656e  a function ``gen
+00012420: 6574 6963 5f63 6f64 6560 602e 2054 6865  etic_code``. The
+00012430: 2066 756e 6374 696f 6e61 6c69 7479 2069   functionality i
+00012440: 7320 7468 6520 7361 6d65 2c20 6578 6365  s the same, exce
+00012450: 7074 2074 6861 7420 6265 6361 7573 6520  pt that because 
+00012460: 7468 6973 2069 7320 6e6f 7720 6120 6675  this is now a fu
+00012470: 6e63 7469 6f6e 2072 6174 6865 7220 7468  nction rather th
+00012480: 616e 2061 2064 6963 742c 2072 6574 7269  an a dict, retri
+00012490: 6576 696e 6720 6120 6765 6e65 7469 6320  eving a genetic 
+000124a0: 636f 6465 2069 7320 646f 6e65 2075 7369  code is done usi
+000124b0: 6e67 2061 2066 756e 6374 696f 6e20 6361  ng a function ca
+000124c0: 6c6c 2072 6174 6865 7220 7468 616e 2061  ll rather than a
+000124d0: 206c 6f6f 6b75 7020 2873 6f2c 2066 6f72   lookup (so, for
+000124e0: 2065 7861 6d70 6c65 2c20 6060 4765 6e65   example, ``Gene
+000124f0: 7469 6343 6f64 6573 5b32 5d60 6020 6265  ticCodes[2]`` be
+00012500: 636f 6d65 7320 6060 6765 6e65 7469 635f  comes ``genetic_
+00012510: 636f 6465 2832 2960 602e 0a2a 204d 616e  code(2)``..* Man
+00012520: 7920 7375 626d 6f64 756c 6573 2068 6176  y submodules hav
+00012530: 6520 6265 656e 206d 6164 6520 7072 6976  e been made priv
+00012540: 6174 6520 7769 7468 2074 6865 2069 6e74  ate with the int
+00012550: 656e 7469 6f6e 206f 6620 7369 6d70 6c69  ention of simpli
+00012560: 6679 696e 6720 696d 706f 7274 7320 666f  fying imports fo
+00012570: 7220 7573 6572 732e 2053 6565 205b 2335  r users. See [#5
+00012580: 3632 5d28 6874 7470 733a 2f2f 6769 7468  62](https://gith
+00012590: 7562 2e63 6f6d 2f62 696f 636f 7265 2f73  ub.com/biocore/s
+000125a0: 6369 6b69 742d 6269 6f2f 6973 7375 6573  cikit-bio/issues
+000125b0: 2f35 3632 2920 666f 7220 6469 7363 7573  /562) for discus
+000125c0: 7369 6f6e 206f 6620 7468 6973 2063 6861  sion of this cha
+000125d0: 6e67 652e 2054 6865 2066 6f6c 6c6f 7769  nge. The followi
+000125e0: 6e67 206c 6973 7420 636f 6e74 6169 6e73  ng list contains
+000125f0: 2074 6865 2070 7265 7669 6f75 7320 6d6f   the previous mo
+00012600: 6475 6c65 206e 616d 6520 616e 6420 7768  dule name and wh
+00012610: 6572 6520 696d 706f 7274 7320 6672 6f6d  ere imports from
+00012620: 2074 6861 7420 6d6f 6475 6c65 2073 686f   that module sho
+00012630: 756c 6420 6e6f 7720 636f 6d65 2066 726f  uld now come fro
+00012640: 6d2e 0a20 2d20 6060 736b 6269 6f2e 616c  m.. - ``skbio.al
+00012650: 6967 6e6d 656e 742e 7373 7760 6020 746f  ignment.ssw`` to
+00012660: 2060 6073 6b62 696f 2e61 6c69 676e 6d65   ``skbio.alignme
+00012670: 6e74 6060 0a20 2d20 6060 736b 6269 6f2e  nt``. - ``skbio.
+00012680: 616c 6967 6e6d 656e 742e 616c 6967 6e6d  alignment.alignm
+00012690: 656e 7460 6020 746f 2060 6073 6b62 696f  ent`` to ``skbio
+000126a0: 2e61 6c69 676e 6d65 6e74 6060 0a20 2d20  .alignment``. - 
+000126b0: 6060 736b 6269 6f2e 616c 6967 6e6d 656e  ``skbio.alignmen
+000126c0: 742e 7061 6972 7769 7365 6060 2074 6f20  t.pairwise`` to 
+000126d0: 6060 736b 6269 6f2e 616c 6967 6e6d 656e  ``skbio.alignmen
+000126e0: 7460 600a 202d 2060 6073 6b62 696f 2e64  t``. - ``skbio.d
+000126f0: 6976 6572 7369 7479 2e61 6c70 6861 2e62  iversity.alpha.b
+00012700: 6173 6560 6020 746f 2060 6073 6b62 696f  ase`` to ``skbio
+00012710: 2e64 6976 6572 7369 7479 2e61 6c70 6861  .diversity.alpha
+00012720: 6060 0a20 2d20 6060 736b 6269 6f2e 6469  ``. - ``skbio.di
+00012730: 7665 7273 6974 792e 616c 7068 612e 6769  versity.alpha.gi
+00012740: 6e69 6060 2074 6f20 6060 736b 6269 6f2e  ni`` to ``skbio.
+00012750: 6469 7665 7273 6974 792e 616c 7068 6160  diversity.alpha`
+00012760: 600a 202d 2060 6073 6b62 696f 2e64 6976  `. - ``skbio.div
+00012770: 6572 7369 7479 2e61 6c70 6861 2e6c 6c61  ersity.alpha.lla
+00012780: 6473 6572 6060 2074 6f20 6060 736b 6269  dser`` to ``skbi
+00012790: 6f2e 6469 7665 7273 6974 792e 616c 7068  o.diversity.alph
+000127a0: 6160 600a 202d 2060 6073 6b62 696f 2e64  a``. - ``skbio.d
+000127b0: 6976 6572 7369 7479 2e62 6574 612e 6261  iversity.beta.ba
+000127c0: 7365 6060 2074 6f20 6060 736b 6269 6f2e  se`` to ``skbio.
+000127d0: 6469 7665 7273 6974 792e 6265 7461 6060  diversity.beta``
+000127e0: 0a20 2d20 6060 736b 6269 6f2e 6472 6177  . - ``skbio.draw
+000127f0: 2e64 6973 7472 6962 7574 696f 6e73 6060  .distributions``
+00012800: 2074 6f20 6060 736b 6269 6f2e 6472 6177   to ``skbio.draw
+00012810: 6060 0a20 2d20 6060 736b 6269 6f2e 7374  ``. - ``skbio.st
+00012820: 6174 732e 6469 7374 616e 6365 2e61 6e6f  ats.distance.ano
+00012830: 7369 6d60 6020 746f 2060 6073 6b62 696f  sim`` to ``skbio
+00012840: 2e73 7461 7473 2e64 6973 7461 6e63 6560  .stats.distance`
+00012850: 600a 202d 2060 6073 6b62 696f 2e73 7461  `. - ``skbio.sta
+00012860: 7473 2e64 6973 7461 6e63 652e 6261 7365  ts.distance.base
+00012870: 6060 2074 6f20 6060 736b 6269 6f2e 7374  `` to ``skbio.st
+00012880: 6174 732e 6469 7374 616e 6365 6060 0a20  ats.distance``. 
+00012890: 2d20 6060 736b 6269 6f2e 7374 6174 732e  - ``skbio.stats.
+000128a0: 6469 7374 616e 6365 2e70 6572 6d61 6e6f  distance.permano
+000128b0: 7661 6060 2074 6f20 6060 736b 6269 6f2e  va`` to ``skbio.
+000128c0: 7374 6174 732e 6469 7374 616e 6365 6060  stats.distance``
+000128d0: 0a20 2d20 6060 736b 6269 6f2e 6469 7374  . - ``skbio.dist
+000128e0: 616e 6365 6060 2074 6f20 6060 736b 6269  ance`` to ``skbi
+000128f0: 6f2e 7374 6174 732e 6469 7374 616e 6365  o.stats.distance
+00012900: 6060 0a20 2d20 6060 736b 6269 6f2e 7374  ``. - ``skbio.st
+00012910: 6174 732e 6f72 6469 6e61 7469 6f6e 2e62  ats.ordination.b
+00012920: 6173 6560 6020 746f 2060 6073 6b62 696f  ase`` to ``skbio
+00012930: 2e73 7461 7473 2e6f 7264 696e 6174 696f  .stats.ordinatio
+00012940: 6e60 600a 202d 2060 6073 6b62 696f 2e73  n``. - ``skbio.s
+00012950: 7461 7473 2e6f 7264 696e 6174 696f 6e2e  tats.ordination.
+00012960: 6361 6e6f 6e69 6361 6c5f 636f 7272 6573  canonical_corres
+00012970: 706f 6e64 656e 6365 5f61 6e61 6c79 7369  pondence_analysi
+00012980: 7360 6020 746f 2060 6073 6b62 696f 2e73  s`` to ``skbio.s
+00012990: 7461 7473 2e6f 7264 696e 6174 696f 6e60  tats.ordination`
+000129a0: 600a 202d 2060 6073 6b62 696f 2e73 7461  `. - ``skbio.sta
+000129b0: 7473 2e6f 7264 696e 6174 696f 6e2e 636f  ts.ordination.co
+000129c0: 7272 6573 706f 6e64 656e 6365 5f61 6e61  rrespondence_ana
+000129d0: 6c79 7369 7360 6020 746f 2060 6073 6b62  lysis`` to ``skb
+000129e0: 696f 2e73 7461 7473 2e6f 7264 696e 6174  io.stats.ordinat
+000129f0: 696f 6e60 600a 202d 2060 6073 6b62 696f  ion``. - ``skbio
+00012a00: 2e73 7461 7473 2e6f 7264 696e 6174 696f  .stats.ordinatio
+00012a10: 6e2e 7072 696e 6369 7061 6c5f 636f 6f72  n.principal_coor
+00012a20: 6469 6e61 7465 5f61 6e61 6c79 7369 7360  dinate_analysis`
+00012a30: 6020 746f 2060 6073 6b62 696f 2e73 7461  ` to ``skbio.sta
+00012a40: 7473 2e6f 7264 696e 6174 696f 6e60 600a  ts.ordination``.
+00012a50: 202d 2060 6073 6b62 696f 2e73 7461 7473   - ``skbio.stats
+00012a60: 2e6f 7264 696e 6174 696f 6e2e 7265 6475  .ordination.redu
+00012a70: 6e64 616e 6379 5f61 6e61 6c79 7369 7360  ndancy_analysis`
+00012a80: 6020 746f 2060 6073 6b62 696f 2e73 7461  ` to ``skbio.sta
+00012a90: 7473 2e6f 7264 696e 6174 696f 6e60 600a  ts.ordination``.
+00012aa0: 202d 2060 6073 6b62 696f 2e74 7265 652e   - ``skbio.tree.
+00012ab0: 7472 6565 6060 2074 6f20 6060 736b 6269  tree`` to ``skbi
+00012ac0: 6f2e 7472 6565 6060 0a20 2d20 6060 736b  o.tree``. - ``sk
+00012ad0: 6269 6f2e 7472 6565 2e74 7269 6560 6020  bio.tree.trie`` 
+00012ae0: 746f 2060 6073 6b62 696f 2e74 7265 6560  to ``skbio.tree`
+00012af0: 600a 202d 2060 6073 6b62 696f 2e75 7469  `. - ``skbio.uti
+00012b00: 6c2e 6d69 7363 6060 2074 6f20 6060 736b  l.misc`` to ``sk
+00012b10: 6269 6f2e 7574 696c 6060 0a20 2d20 6060  bio.util``. - ``
+00012b20: 736b 6269 6f2e 7574 696c 2e74 6573 7469  skbio.util.testi
+00012b30: 6e67 6060 2074 6f20 6060 736b 6269 6f2e  ng`` to ``skbio.
+00012b40: 7574 696c 6060 0a20 2d20 6060 736b 6269  util``. - ``skbi
+00012b50: 6f2e 7574 696c 2e65 7863 6570 7469 6f6e  o.util.exception
+00012b60: 6060 2074 6f20 6060 736b 6269 6f2e 7574  `` to ``skbio.ut
+00012b70: 696c 6060 0a20 2d20 6060 736b 6269 6f2e  il``. - ``skbio.
+00012b80: 7574 696c 2e77 6172 6e69 6e67 6060 2074  util.warning`` t
+00012b90: 6f20 6060 736b 6269 6f2e 7574 696c 6060  o ``skbio.util``
+00012ba0: 0a2a 204d 6f76 6564 2060 6073 6b62 696f  .* Moved ``skbio
+00012bb0: 2e64 6973 7461 6e63 6560 6020 636f 6e74  .distance`` cont
+00012bc0: 656e 7473 2069 6e74 6f20 6060 736b 6269  ents into ``skbi
+00012bd0: 6f2e 7374 6174 732e 6469 7374 616e 6365  o.stats.distance
+00012be0: 6060 2e0a 0a23 2323 204d 6973 6365 6c6c  ``...### Miscell
+00012bf0: 616e 656f 7573 0a0a 2a20 5265 6c61 7865  aneous..* Relaxe
+00012c00: 6420 7265 7175 6972 656d 656e 7420 696e  d requirement in
+00012c10: 2060 6042 696f 6c6f 6769 6361 6c53 6571   ``BiologicalSeq
+00012c20: 7565 6e63 652e 6469 7374 616e 6365 6060  uence.distance``
+00012c30: 2074 6861 7420 7365 7175 656e 6365 7320   that sequences 
+00012c40: 6265 696e 6720 636f 6d70 6172 6564 2061  being compared a
+00012c50: 7265 206f 6620 6571 7561 6c20 6c65 6e67  re of equal leng
+00012c60: 7468 2e20 5468 6973 2069 7320 7265 6c65  th. This is rele
+00012c70: 7661 6e74 2066 6f72 2048 616d 6d69 6e67  vant for Hamming
+00012c80: 2064 6973 7461 6e63 652c 2073 6f20 7468   distance, so th
+00012c90: 6520 6368 6563 6b20 6973 2073 7469 6c6c  e check is still
+00012ca0: 2070 6572 666f 726d 6564 2069 6e20 7468   performed in th
+00012cb0: 6174 2063 6173 652c 2062 7574 206f 7468  at case, but oth
+00012cc0: 6572 2064 6973 7461 6e63 6520 6d65 7472  er distance metr
+00012cd0: 6963 7320 6d61 7920 6e6f 7420 6861 7665  ics may not have
+00012ce0: 2074 6861 7420 7265 7175 6972 656d 656e   that requiremen
+00012cf0: 742e 2053 6565 205b 2335 3034 5d28 6874  t. See [#504](ht
+00012d00: 7470 733a 2f2f 6769 7468 7562 2e63 6f6d  tps://github.com
+00012d10: 2f62 696f 636f 7265 2f73 6369 6b69 742d  /biocore/scikit-
+00012d20: 6269 6f2f 6973 7375 6573 2f35 3037 2929  bio/issues/507))
+00012d30: 2e0a 2a20 5265 6e61 6d65 6420 6060 706f  ..* Renamed ``po
+00012d40: 7765 7274 7269 702e 7079 6060 2072 6570  wertrip.py`` rep
+00012d50: 6f2d 6368 6563 6b69 6e67 2073 6372 6970  o-checking scrip
+00012d60: 7420 746f 2060 6063 6865 636b 6c69 7374  t to ``checklist
+00012d70: 2e70 7960 6020 666f 7220 636c 6172 6974  .py`` for clarit
+00012d80: 792e 0a2a 2060 6063 6865 636b 6c69 7374  y..* ``checklist
+00012d90: 2e70 7960 6020 6e6f 7720 656e 7375 7265  .py`` now ensure
+00012da0: 7320 7468 6174 2061 6c6c 2075 6e69 7420  s that all unit 
+00012db0: 7465 7374 7320 696d 706f 7274 2066 726f  tests import fro
+00012dc0: 6d20 6120 6d69 6e69 6d61 6c6c 7920 6465  m a minimally de
+00012dd0: 6570 2041 5049 2e20 466f 7220 6578 616d  ep API. For exam
+00012de0: 706c 652c 2069 7420 7769 6c6c 2070 726f  ple, it will pro
+00012df0: 6475 6365 2061 6e20 6572 726f 7220 6966  duce an error if
+00012e00: 2060 6073 6b62 696f 2e63 6f72 652e 6469   ``skbio.core.di
+00012e10: 7374 616e 6365 2e44 6973 7461 6e63 654d  stance.DistanceM
+00012e20: 6174 7269 7860 6020 6973 2075 7365 6420  atrix`` is used 
+00012e30: 6f76 6572 2060 6073 6b62 696f 2e44 6973  over ``skbio.Dis
+00012e40: 7461 6e63 654d 6174 7269 7860 602e 0a2a  tanceMatrix``..*
+00012e50: 2045 7874 7261 2064 696d 656e 7369 6f6e   Extra dimension
+00012e60: 2069 7320 6e6f 206c 6f6e 6765 7220 6361   is no longer ca
+00012e70: 6c63 756c 6174 6564 2069 6e20 6060 736b  lculated in ``sk
+00012e80: 6269 6f2e 7374 6174 732e 7370 6174 6961  bio.stats.spatia
+00012e90: 6c2e 7072 6f63 7275 7374 6573 6060 2e0a  l.procrustes``..
+00012ea0: 2a20 4578 7061 6e64 6564 2064 6f63 756d  * Expanded docum
+00012eb0: 656e 7461 7469 6f6e 2069 6e20 7661 7269  entation in vari
+00012ec0: 6f75 7320 7375 6270 6163 6b61 6765 732e  ous subpackages.
+00012ed0: 0a2a 2041 6464 6564 206e 6577 2073 6369  .* Added new sci
+00012ee0: 6b69 742d 6269 6f20 6c6f 676f 2e20 5468  kit-bio logo. Th
+00012ef0: 616e 6b73 205b 416c 696e 6120 5072 6173  anks [Alina Pras
+00012f00: 7361 735d 2868 7474 703a 2f2f 6361 7267  sas](http://carg
+00012f10: 6f63 6f6c 6c65 6374 6976 652e 636f 6d2f  ocollective.com/
+00012f20: 616c 696e 6170 7261 7373 6173 2921 0a0a  alinaprassas)!..
+00012f30: 2323 2056 6572 7369 6f6e 2030 2e31 2e34  ## Version 0.1.4
+00012f40: 2028 3230 3134 2d30 362d 3235 290a 0a54   (2014-06-25)..T
+00012f50: 6869 7320 6973 2061 2070 7265 2d61 6c70  his is a pre-alp
+00012f60: 6861 2072 656c 6561 7365 2e20 4174 2074  ha release. At t
+00012f70: 6869 7320 7374 6167 652c 206d 616a 6f72  his stage, major
+00012f80: 2062 6163 6b77 6172 6473 2d69 6e63 6f6d   backwards-incom
+00012f90: 7061 7469 626c 6520 4150 4920 6368 616e  patible API chan
+00012fa0: 6765 7320 6361 6e20 616e 6420 7769 6c6c  ges can and will
+00012fb0: 2068 6170 7065 6e2e 0a0a 2323 2320 4665   happen...### Fe
+00012fc0: 6174 7572 6573 0a0a 2a20 4164 6465 6420  atures..* Added 
+00012fd0: 5079 7468 6f6e 2069 6d70 6c65 6d65 6e74  Python implement
+00012fe0: 6174 696f 6e73 206f 6620 536d 6974 682d  ations of Smith-
+00012ff0: 5761 7465 726d 616e 2061 6e64 204e 6565  Waterman and Nee
+00013000: 646c 656d 616e 2d57 756e 7363 6820 616c  dleman-Wunsch al
+00013010: 6967 6e6d 656e 7420 6173 2060 6073 6b62  ignment as ``skb
+00013020: 696f 2e63 6f72 652e 616c 6967 6e6d 656e  io.core.alignmen
+00013030: 742e 7061 6972 7769 7365 2e6c 6f63 616c  t.pairwise.local
+00013040: 5f70 6169 7277 6973 655f 616c 6967 6e60  _pairwise_align`
+00013050: 6020 616e 6420 6060 736b 6269 6f2e 636f  ` and ``skbio.co
+00013060: 7265 2e61 6c69 676e 6d65 6e74 2e70 6169  re.alignment.pai
+00013070: 7277 6973 652e 676c 6f62 616c 5f70 6169  rwise.global_pai
+00013080: 7277 6973 655f 616c 6967 6e60 602e 2054  rwise_align``. T
+00013090: 6865 7365 2061 7265 206d 7563 6820 736c  hese are much sl
+000130a0: 6f77 6572 2074 6861 6e20 6e61 7469 7665  ower than native
+000130b0: 2043 2069 6d70 6c65 6d65 6e74 6174 696f   C implementatio
+000130c0: 6e73 2028 652e 672e 2c20 6060 736b 6269  ns (e.g., ``skbi
+000130d0: 6f2e 636f 7265 2e61 6c69 676e 6d65 6e74  o.core.alignment
+000130e0: 2e6c 6f63 616c 5f70 6169 7277 6973 655f  .local_pairwise_
+000130f0: 616c 6967 6e5f 7373 7760 6029 2061 6e64  align_ssw``) and
+00013100: 2061 7320 6120 7265 7375 6c74 2072 6169   as a result rai
+00013110: 7365 2061 6e20 6060 4566 6669 6365 6e63  se an ``Efficenc
+00013120: 7957 6172 6e69 6e67 6060 2077 6865 6e20  yWarning`` when 
+00013130: 6361 6c6c 6564 2c20 6275 7420 6172 6520  called, but are 
+00013140: 696e 636c 7564 6564 2061 7320 7468 6579  included as they
+00013150: 2073 6572 7665 2061 7320 7573 6566 756c   serve as useful
+00013160: 2065 6475 6361 7469 6f6e 616c 2065 7861   educational exa
+00013170: 6d70 6c65 7320 6173 2074 6865 79e2 8099  mples as they...
+00013180: 7265 2073 696d 706c 6520 746f 2065 7870  re simple to exp
+00013190: 6572 696d 656e 7420 7769 7468 2e0a 2a20  eriment with..* 
+000131a0: 4164 6465 6420 6060 736b 6269 6f2e 636f  Added ``skbio.co
+000131b0: 7265 2e64 6976 6572 7369 7479 2e62 6574  re.diversity.bet
+000131c0: 612e 7077 5f64 6973 7461 6e63 6573 6060  a.pw_distances``
+000131d0: 2061 6e64 2060 6073 6b62 696f 2e63 6f72   and ``skbio.cor
+000131e0: 652e 6469 7665 7273 6974 792e 6265 7461  e.diversity.beta
+000131f0: 2e70 775f 6469 7374 616e 6365 735f 6672  .pw_distances_fr
+00013200: 6f6d 5f74 6162 6c65 6060 2e20 5468 6573  om_table``. Thes
+00013210: 6520 7072 6f76 6964 6520 636f 6e76 656e  e provide conven
+00013220: 6965 6e74 2061 6363 6573 7320 746f 2074  ient access to t
+00013230: 6865 2060 6073 6369 7079 2e73 7061 7469  he ``scipy.spati
+00013240: 616c 2e64 6973 7461 6e63 652e 7064 6973  al.distance.pdis
+00013250: 7460 6020 2a62 6574 6120 6469 7665 7273  t`` *beta divers
+00013260: 6974 792a 206d 6574 7269 6373 2066 726f  ity* metrics fro
+00013270: 6d20 7769 7468 696e 2073 6369 6b69 742d  m within scikit-
+00013280: 6269 6f2e 2054 6865 2060 6073 6b62 696f  bio. The ``skbio
+00013290: 2e63 6f72 652e 6469 7665 7273 6974 792e  .core.diversity.
+000132a0: 6265 7461 2e70 775f 6469 7374 616e 6365  beta.pw_distance
+000132b0: 735f 6672 6f6d 5f74 6162 6c65 6060 2066  s_from_table`` f
+000132c0: 756e 6374 696f 6e20 7769 6c6c 206f 6e6c  unction will onl
+000132d0: 7920 6265 2061 7661 696c 6162 6c65 2074  y be available t
+000132e0: 656d 706f 7261 7269 6c79 2c20 756e 7469  emporarily, unti
+000132f0: 6c20 7468 6520 6060 6269 6f6d 2e74 6162  l the ``biom.tab
+00013300: 6c65 2e54 6162 6c65 6060 206f 626a 6563  le.Table`` objec
+00013310: 7420 6973 206d 6572 6765 6420 696e 746f  t is merged into
+00013320: 2073 6369 6b69 742d 6269 6f20 2873 6565   scikit-bio (see
+00013330: 205b 2334 3839 5d28 6874 7470 733a 2f2f   [#489](https://
+00013340: 6769 7468 7562 2e63 6f6d 2f62 696f 636f  github.com/bioco
+00013350: 7265 2f73 6369 6b69 742d 6269 6f2f 6973  re/scikit-bio/is
+00013360: 7375 6573 2f34 3839 2929 2c20 6174 2077  sues/489)), at w
+00013370: 6869 6368 2070 6f69 6e74 2060 6073 6b62  hich point ``skb
+00013380: 696f 2e63 6f72 652e 6469 7665 7273 6974  io.core.diversit
+00013390: 792e 6265 7461 2e70 775f 6469 7374 616e  y.beta.pw_distan
+000133a0: 6365 7360 6020 7769 6c6c 2062 6520 7570  ces`` will be up
+000133b0: 6461 7465 6420 746f 2075 7365 2074 6861  dated to use tha
+000133c0: 742e 0a2a 2041 6464 6564 2060 6073 6b62  t..* Added ``skb
+000133d0: 696f 2e63 6f72 652e 616c 6967 6e6d 656e  io.core.alignmen
+000133e0: 742e 5374 6f63 6b68 6f6c 6d41 6c69 676e  t.StockholmAlign
+000133f0: 6d65 6e74 6060 2c20 7768 6963 6820 7072  ment``, which pr
+00013400: 6f76 6964 6573 2073 7570 706f 7274 2066  ovides support f
+00013410: 6f72 2070 6172 7369 6e67 205b 5374 6f63  or parsing [Stoc
+00013420: 6b68 6f6c 6d2d 666f 726d 6174 7465 6420  kholm-formatted 
+00013430: 616c 6967 6e6d 656e 7420 6669 6c65 735d  alignment files]
+00013440: 2868 7474 703a 2f2f 736f 6e6e 6861 6d6d  (http://sonnhamm
+00013450: 6572 2e73 6263 2e73 752e 7365 2f53 746f  er.sbc.su.se/Sto
+00013460: 636b 686f 6c6d 2e68 746d 6c29 2061 6e64  ckholm.html) and
+00013470: 2077 6f72 6b69 6e67 2077 6974 6820 7468   working with th
+00013480: 6f73 6520 616c 6967 6e6d 656e 7473 2069  ose alignments i
+00013490: 6e20 7468 6520 636f 6e74 6578 7420 524e  n the context RN
+000134a0: 4120 7365 636f 6e64 6172 7920 7374 7275  A secondary stru
+000134b0: 6374 7572 616c 2069 6e66 6f72 6d61 7469  ctural informati
+000134c0: 6f6e 2e0a 2a20 4164 6465 6420 6060 736b  on..* Added ``sk
+000134d0: 6269 6f2e 636f 7265 2e74 7265 652e 6d61  bio.core.tree.ma
+000134e0: 6a6f 7269 7479 5f72 756c 6560 6020 6675  jority_rule`` fu
+000134f0: 6e63 7469 6f6e 2066 6f72 2063 6f6d 7075  nction for compu
+00013500: 7469 6e67 2063 6f6e 7365 6e73 7573 2074  ting consensus t
+00013510: 7265 6573 2066 726f 6d20 6120 6c69 7374  rees from a list
+00013520: 206f 6620 7472 6565 732e 0a0a 2323 2320   of trees...### 
+00013530: 4261 636b 7761 7264 2d69 6e63 6f6d 7061  Backward-incompa
+00013540: 7469 626c 6520 6368 616e 6765 730a 0a2a  tible changes..*
+00013550: 2046 756e 6374 696f 6e20 6060 736b 6269   Function ``skbi
+00013560: 6f2e 636f 7265 2e61 6c69 676e 6d65 6e74  o.core.alignment
+00013570: 2e61 6c69 676e 5f73 7472 6970 6564 5f73  .align_striped_s
+00013580: 6d69 7468 5f77 6174 6572 6d61 6e60 6020  mith_waterman`` 
+00013590: 7265 6e61 6d65 6420 746f 2060 606c 6f63  renamed to ``loc
+000135a0: 616c 5f70 6169 7277 6973 655f 616c 6967  al_pairwise_alig
+000135b0: 6e5f 7373 7760 6020 616e 6420 6e6f 7720  n_ssw`` and now 
+000135c0: 7265 7475 726e 7320 616e 2060 6041 6c69  returns an ``Ali
+000135d0: 676e 6d65 6e74 6060 206f 626a 6563 7420  gnment`` object 
+000135e0: 696e 7374 6561 6420 6f66 2061 6e20 6060  instead of an ``
+000135f0: 416c 6967 6e6d 656e 7453 7472 7563 7475  AlignmentStructu
+00013600: 7265 6060 0a2a 2054 6865 2066 6f6c 6c6f  re``.* The follo
+00013610: 7769 6e67 206b 6579 776f 7264 2d61 7267  wing keyword-arg
+00013620: 756d 656e 7473 2066 6f72 2060 6053 7472  uments for ``Str
+00013630: 6970 6564 536d 6974 6857 6174 6572 6d61  ipedSmithWaterma
+00013640: 6e60 6020 616e 6420 6060 6c6f 6361 6c5f  n`` and ``local_
+00013650: 7061 6972 7769 7365 5f61 6c69 676e 5f73  pairwise_align_s
+00013660: 7377 6060 2068 6176 6520 6265 656e 2072  sw`` have been r
+00013670: 656e 616d 6564 3a0a 2020 2020 2a20 6060  enamed:.    * ``
+00013680: 6761 705f 6f70 656e 6060 202d 3e20 6060  gap_open`` -> ``
+00013690: 6761 705f 6f70 656e 5f70 656e 616c 7479  gap_open_penalty
+000136a0: 6060 0a20 2020 202a 2060 6067 6170 5f65  ``.    * ``gap_e
+000136b0: 7874 656e 6460 6020 2d3e 2060 6067 6170  xtend`` -> ``gap
+000136c0: 5f65 7874 656e 645f 7065 6e61 6c74 7960  _extend_penalty`
+000136d0: 600a 2020 2020 2a20 6060 6d61 7463 6860  `.    * ``match`
+000136e0: 6020 2d3e 2060 606d 6174 6368 5f73 636f  ` -> ``match_sco
+000136f0: 7265 6060 0a20 2020 202a 2060 606d 6973  re``.    * ``mis
+00013700: 6d61 7463 6860 6020 2d3e 2060 606d 6973  match`` -> ``mis
+00013710: 6d61 7463 685f 7363 6f72 6560 600a 2a20  match_score``.* 
+00013720: 5265 6d6f 7665 6420 6060 736b 6269 6f2e  Removed ``skbio.
+00013730: 7574 696c 2e73 6f72 7460 6020 6d6f 6475  util.sort`` modu
+00013740: 6c65 2069 6e20 6661 766f 7220 6f66 205b  le in favor of [
+00013750: 6e61 7473 6f72 745d 2868 7474 7073 3a2f  natsort](https:/
+00013760: 2f70 7970 692e 7079 7468 6f6e 2e6f 7267  /pypi.python.org
+00013770: 2f70 7970 692f 6e61 7473 6f72 7429 2070  /pypi/natsort) p
+00013780: 6163 6b61 6765 2e0a 0a23 2323 204d 6973  ackage...### Mis
+00013790: 6365 6c6c 616e 656f 7573 0a0a 2a20 4164  cellaneous..* Ad
+000137a0: 6465 6420 706f 7765 7274 7269 702e 7079  ded powertrip.py
+000137b0: 2073 6372 6970 7420 746f 2070 6572 666f   script to perfo
+000137c0: 726d 2062 6173 6963 2073 616e 6974 792d  rm basic sanity-
+000137d0: 6368 6563 6b69 6e67 206f 6620 7468 6520  checking of the 
+000137e0: 7265 706f 2062 6173 6564 206f 6e20 7265  repo based on re
+000137f0: 6375 7272 696e 6720 6973 7375 6573 2074  curring issues t
+00013800: 6861 7420 7765 7265 6e27 7420 6265 696e  hat weren't bein
+00013810: 6720 6361 7567 6874 2075 6e74 696c 2072  g caught until r
+00013820: 656c 6561 7365 2074 696d 653b 2061 6464  elease time; add
+00013830: 6564 2074 6f20 5472 6176 6973 2062 7569  ed to Travis bui
+00013840: 6c64 2e0a 2a20 4164 6465 6420 5245 4c45  ld..* Added RELE
+00013850: 4153 452e 6d64 2077 6974 6820 7265 6c65  ASE.md with rele
+00013860: 6173 6520 696e 7374 7275 6374 696f 6e73  ase instructions
+00013870: 2e0a 2a20 4164 6465 6420 696e 7465 7273  ..* Added inters
+00013880: 7068 696e 7820 6d61 7070 696e 6773 2074  phinx mappings t
+00013890: 6f20 646f 6373 2073 6f20 7468 6174 2022  o docs so that "
+000138a0: 5365 6520 416c 736f 2220 7265 6665 7265  See Also" refere
+000138b0: 6e63 6573 2074 6f20 6e75 6d70 792c 2073  nces to numpy, s
+000138c0: 6369 7079 2c20 6d61 7470 6c6f 746c 6962  cipy, matplotlib
+000138d0: 2c20 616e 6420 7061 6e64 6173 2061 7265  , and pandas are
+000138e0: 2068 7970 6572 6c69 6e6b 732e 0a2a 2054   hyperlinks..* T
+000138f0: 6865 2066 6f6c 6c6f 7769 6e67 2063 6c61  he following cla
+00013900: 7373 6573 2061 7265 206e 6f20 6c6f 6e67  sses are no long
+00013910: 6572 2060 606e 616d 6564 7475 706c 6560  er ``namedtuple`
+00013920: 6020 7375 6263 6c61 7373 6573 2028 7365  ` subclasses (se
+00013930: 6520 5b23 3335 395d 2868 7474 7073 3a2f  e [#359](https:/
+00013940: 2f67 6974 6875 622e 636f 6d2f 6269 6f63  /github.com/bioc
+00013950: 6f72 652f 7363 696b 6974 2d62 696f 2f69  ore/scikit-bio/i
+00013960: 7373 7565 732f 3335 3929 2066 6f72 2074  ssues/359) for t
+00013970: 6865 2072 6174 696f 6e61 6c65 293a 0a20  he rationale):. 
+00013980: 2020 202a 2060 6073 6b62 696f 2e6d 6174     * ``skbio.mat
+00013990: 682e 7374 6174 732e 6f72 6469 6e61 7469  h.stats.ordinati
+000139a0: 6f6e 2e4f 7264 696e 6174 696f 6e52 6573  on.OrdinationRes
+000139b0: 756c 7473 6060 0a20 2020 202a 2060 6073  ults``.    * ``s
+000139c0: 6b62 696f 2e6d 6174 682e 6772 6164 6965  kbio.math.gradie
+000139d0: 6e74 2e47 726f 7570 5265 7375 6c74 7360  nt.GroupResults`
+000139e0: 600a 2020 2020 2a20 6060 736b 6269 6f2e  `.    * ``skbio.
+000139f0: 6d61 7468 2e67 7261 6469 656e 742e 4361  math.gradient.Ca
+00013a00: 7465 676f 7279 5265 7375 6c74 7360 600a  tegoryResults``.
+00013a10: 2020 2020 2a20 6060 736b 6269 6f2e 6d61      * ``skbio.ma
+00013a20: 7468 2e67 7261 6469 656e 742e 4772 6164  th.gradient.Grad
+00013a30: 6965 6e74 414e 4f56 4152 6573 756c 7473  ientANOVAResults
+00013a40: 6060 0a2a 2041 6464 6564 2063 6f64 696e  ``.* Added codin
+00013a50: 6720 6775 6964 656c 696e 6573 2064 7261  g guidelines dra
+00013a60: 6674 2e0a 2a20 4164 6465 6420 6e65 7720  ft..* Added new 
+00013a70: 616c 7068 6120 6469 7665 7273 6974 7920  alpha diversity 
+00013a80: 666f 726d 756c 6173 2074 6f20 7468 6520  formulas to the 
+00013a90: 6060 736b 6269 6f2e 6d61 7468 2e64 6976  ``skbio.math.div
+00013aa0: 6572 7369 7479 2e61 6c70 6861 6060 2064  ersity.alpha`` d
+00013ab0: 6f63 756d 656e 7461 7469 6f6e 2e0a 0a23  ocumentation...#
+00013ac0: 2320 5665 7273 696f 6e20 302e 312e 3320  # Version 0.1.3 
+00013ad0: 2832 3031 342d 3036 2d31 3229 0a0a 5468  (2014-06-12)..Th
+00013ae0: 6973 2069 7320 6120 7072 652d 616c 7068  is is a pre-alph
+00013af0: 6120 7265 6c65 6173 652e 2041 7420 7468  a release. At th
+00013b00: 6973 2073 7461 6765 2c20 6d61 6a6f 7220  is stage, major 
+00013b10: 6261 636b 7761 7264 732d 696e 636f 6d70  backwards-incomp
+00013b20: 6174 6962 6c65 2041 5049 2063 6861 6e67  atible API chang
+00013b30: 6573 2063 616e 2061 6e64 2077 696c 6c20  es can and will 
+00013b40: 6861 7070 656e 2e0a 0a23 2323 2046 6561  happen...### Fea
+00013b50: 7475 7265 730a 0a2a 2041 6464 6564 2060  tures..* Added `
+00013b60: 6065 6e66 6f72 6365 5f71 7561 6c5f 7261  `enforce_qual_ra
+00013b70: 6e67 6560 6020 7061 7261 6d65 7465 7220  nge`` parameter 
+00013b80: 746f 2060 6070 6172 7365 5f66 6173 7471  to ``parse_fastq
+00013b90: 6060 2028 6f6e 2062 7920 6465 6661 756c  `` (on by defaul
+00013ba0: 742c 206d 6169 6e74 6169 6e69 6e67 2062  t, maintaining b
+00013bb0: 6163 6b77 6172 6420 636f 6d70 6174 6962  ackward compatib
+00013bc0: 696c 6974 7929 2e20 5468 6973 2061 6c6c  ility). This all
+00013bd0: 6f77 7320 6469 7361 626c 696e 6720 6f66  ows disabling of
+00013be0: 2074 6865 2071 7561 6c69 7479 2073 636f   the quality sco
+00013bf0: 7265 2072 616e 6765 2d63 6865 636b 696e  re range-checkin
+00013c00: 672e 0a2a 2041 6464 6564 2060 6073 6b62  g..* Added ``skb
+00013c10: 696f 2e63 6f72 652e 7472 6565 2e6e 6a60  io.core.tree.nj`
+00013c20: 602c 2077 6869 6368 2061 7070 6c69 6573  `, which applies
+00013c30: 206e 6569 6768 626f 722d 6a6f 696e 696e   neighbor-joinin
+00013c40: 6720 666f 7220 7068 796c 6f67 656e 6574  g for phylogenet
+00013c50: 6963 2072 6563 6f6e 7374 7275 6374 696f  ic reconstructio
+00013c60: 6e2e 0a2a 2041 6464 6564 2060 6062 696f  n..* Added ``bio
+00013c70: 656e 7660 602c 2060 606d 616e 7465 6c60  env``, ``mantel`
+00013c80: 602c 2061 6e64 2060 6070 776d 616e 7465  `, and ``pwmante
+00013c90: 6c60 6020 6469 7374 616e 6365 2d62 6173  l`` distance-bas
+00013ca0: 6564 2073 7461 7469 7374 6963 7320 746f  ed statistics to
+00013cb0: 2060 6073 6b62 696f 2e6d 6174 682e 7374   ``skbio.math.st
+00013cc0: 6174 732e 6469 7374 616e 6365 6060 2073  ats.distance`` s
+00013cd0: 7562 7061 636b 6167 652e 0a2a 2041 6464  ubpackage..* Add
+00013ce0: 6564 2060 6073 6b62 696f 2e6d 6174 682e  ed ``skbio.math.
+00013cf0: 7374 6174 732e 6d69 7363 6060 206d 6f64  stats.misc`` mod
+00013d00: 756c 6520 666f 7220 6d69 7363 656c 6c61  ule for miscella
+00013d10: 6e65 6f75 7320 7374 6174 7320 7574 696c  neous stats util
+00013d20: 6974 7920 6675 6e63 7469 6f6e 732e 0a2a  ity functions..*
+00013d30: 2049 4473 2061 7265 206e 6f77 206f 7074   IDs are now opt
+00013d40: 696f 6e61 6c20 7768 656e 2063 6f6e 7374  ional when const
+00013d50: 7275 6374 696e 6720 6120 6060 4469 7373  ructing a ``Diss
+00013d60: 696d 696c 6172 6974 794d 6174 7269 7860  imilarityMatrix`
+00013d70: 6020 6f72 2060 6044 6973 7461 6e63 654d  ` or ``DistanceM
+00013d80: 6174 7269 7860 6020 286d 6f6e 6f74 6f6e  atrix`` (monoton
+00013d90: 6963 616c 6c79 2d69 6e63 7265 6173 696e  ically-increasin
+00013da0: 6720 696e 7465 6765 7273 2063 6173 7420  g integers cast 
+00013db0: 6173 2073 7472 696e 6773 2061 7265 2061  as strings are a
+00013dc0: 7574 6f6d 6174 6963 616c 6c79 2075 7365  utomatically use
+00013dd0: 6429 2e0a 2a20 4164 6465 6420 6060 4469  d)..* Added ``Di
+00013de0: 7374 616e 6365 4d61 7472 6978 2e70 6572  stanceMatrix.per
+00013df0: 6d75 7465 6060 206d 6574 686f 6420 666f  mute`` method fo
+00013e00: 7220 7261 6e64 6f6d 6c79 2070 6572 6d75  r randomly permu
+00013e10: 7469 6e67 2072 6f77 7320 616e 6420 636f  ting rows and co
+00013e20: 6c75 6d6e 7320 6f66 2061 2064 6973 7461  lumns of a dista
+00013e30: 6e63 6520 6d61 7472 6978 2e0a 2a20 4164  nce matrix..* Ad
+00013e40: 6465 6420 7468 6520 666f 6c6c 6f77 696e  ded the followin
+00013e50: 6720 6d65 7468 6f64 7320 746f 2060 6044  g methods to ``D
+00013e60: 6973 7369 6d69 6c61 7269 7479 4d61 7472  issimilarityMatr
+00013e70: 6978 6060 3a20 6060 6669 6c74 6572 6060  ix``: ``filter``
+00013e80: 2c20 6060 696e 6465 7860 602c 2061 6e64  , ``index``, and
+00013e90: 2060 605f 5f63 6f6e 7461 696e 735f 5f60   ``__contains__`
+00013ea0: 6020 666f 7220 4944 2d62 6173 6564 2066  ` for ID-based f
+00013eb0: 696c 7465 7269 6e67 2c20 696e 6465 7820  iltering, index 
+00013ec0: 6c6f 6f6b 7570 2c20 616e 6420 6d65 6d62  lookup, and memb
+00013ed0: 6572 7368 6970 2074 6573 7469 6e67 2c20  ership testing, 
+00013ee0: 7265 7370 6563 7469 7665 6c79 2e0a 2a20  respectively..* 
+00013ef0: 4164 6465 6420 6060 6967 6e6f 7265 5f63  Added ``ignore_c
+00013f00: 6f6d 6d65 6e74 6060 2070 6172 616d 6574  omment`` paramet
+00013f10: 6572 2074 6f20 6060 7061 7273 655f 6661  er to ``parse_fa
+00013f20: 7374 6160 6020 286f 6666 2062 7920 6465  sta`` (off by de
+00013f30: 6661 756c 742c 206d 6169 6e74 6169 6e69  fault, maintaini
+00013f40: 6e67 2062 6163 6b77 6172 6420 636f 6d70  ng backward comp
+00013f50: 6174 6962 696c 6974 7929 2e20 5468 6973  atibility). This
+00013f60: 2068 616e 646c 6573 2073 7472 6970 7069   handles strippi
+00013f70: 6e67 2074 6865 2063 6f6d 6d65 6e74 2066  ng the comment f
+00013f80: 6965 6c64 2066 726f 6d20 7468 6520 6865  ield from the he
+00013f90: 6164 6572 206c 696e 6520 2869 2e65 2e2c  ader line (i.e.,
+00013fa0: 2061 6c6c 2063 6861 7261 6374 6572 7320   all characters 
+00013fb0: 6265 6769 6e6e 696e 6720 7769 7468 2074  beginning with t
+00013fc0: 6865 2066 6972 7374 2073 7061 6365 2920  he first space) 
+00013fd0: 6265 666f 7265 2072 6574 7572 6e69 6e67  before returning
+00013fe0: 2074 6865 206c 6162 656c 2e0a 2a20 4164   the label..* Ad
+00013ff0: 6465 6420 696d 706f 7274 7320 6f66 2060  ded imports of `
+00014000: 6042 696f 6c6f 6769 6361 6c53 6571 7565  `BiologicalSeque
+00014010: 6e63 6560 602c 2060 604e 7563 6c65 6f74  nce``, ``Nucleot
+00014020: 6964 6553 6571 7565 6e63 6560 602c 2060  ideSequence``, `
+00014030: 6044 4e41 6060 2c20 6060 444e 4153 6571  `DNA``, ``DNASeq
+00014040: 7565 6e63 6560 602c 2060 6052 4e41 6060  uence``, ``RNA``
+00014050: 2c20 6060 524e 4153 6571 7565 6e63 6560  , ``RNASequence`
+00014060: 602c 2060 6050 726f 7465 696e 6060 2c20  `, ``Protein``, 
+00014070: 6060 5072 6f74 6569 6e53 6571 7565 6e63  ``ProteinSequenc
+00014080: 6560 602c 2060 6044 6973 7461 6e63 654d  e``, ``DistanceM
+00014090: 6174 7269 7860 602c 2060 6061 6c69 676e  atrix``, ``align
+000140a0: 5f73 7472 6970 6564 5f73 6d69 7468 5f77  _striped_smith_w
+000140b0: 6174 6572 6d61 6e60 602c 2060 6020 5365  aterman``, `` Se
+000140c0: 7175 656e 6365 436f 6c6c 6563 7469 6f6e  quenceCollection
+000140d0: 6060 2c20 6060 416c 6967 6e6d 656e 7460  ``, ``Alignment`
+000140e0: 602c 2060 6054 7265 654e 6f64 6560 602c  `, ``TreeNode``,
+000140f0: 2060 606e 6a60 602c 2060 6070 6172 7365   ``nj``, ``parse
+00014100: 5f66 6173 7461 6060 2c20 6060 7061 7273  _fasta``, ``pars
+00014110: 655f 6661 7374 7160 602c 2060 6070 6172  e_fastq``, ``par
+00014120: 7365 5f71 7561 6c60 602c 2060 6046 6173  se_qual``, ``Fas
+00014130: 7461 4974 6572 6174 6f72 6060 2c20 6060  taIterator``, ``
+00014140: 4661 7374 7149 7465 7261 746f 7260 602c  FastqIterator``,
+00014150: 2060 6053 6571 7565 6e63 6549 7465 7261   ``SequenceItera
+00014160: 746f 7260 6020 696e 2060 6073 6b62 696f  tor`` in ``skbio
+00014170: 2f5f 5f69 6e69 745f 5f2e 7079 6060 2066  /__init__.py`` f
+00014180: 6f72 2063 6f6e 7665 6e69 656e 7420 696d  or convenient im
+00014190: 706f 7274 696e 672e 2046 6f72 2065 7861  porting. For exa
+000141a0: 6d70 6c65 2c20 6974 2773 206e 6f77 2070  mple, it's now p
+000141b0: 6f73 7369 626c 6520 746f 2060 6066 726f  ossible to ``fro
+000141c0: 6d20 736b 6269 6f20 696d 706f 7274 2041  m skbio import A
+000141d0: 6c69 676e 6d65 6e74 6060 2c20 7261 7468  lignment``, rath
+000141e0: 6572 2074 6861 6e20 6060 6672 6f6d 2073  er than ``from s
+000141f0: 6b62 696f 2e63 6f72 652e 616c 6967 6e6d  kbio.core.alignm
+00014200: 656e 7420 696d 706f 7274 2041 6c69 676e  ent import Align
+00014210: 6d65 6e74 6060 2e0a 0a23 2323 2042 7567  ment``...### Bug
+00014220: 2066 6978 6573 0a0a 2a20 4669 7865 6420   fixes..* Fixed 
+00014230: 6120 636f 7570 6c65 206f 6620 756e 6974  a couple of unit
+00014240: 2074 6573 7473 2074 6861 7420 636f 756c   tests that coul
+00014250: 6420 6661 696c 2073 746f 6368 6173 7469  d fail stochasti
+00014260: 6361 6c6c 792e 0a2a 2041 6464 6564 206d  cally..* Added m
+00014270: 6973 7369 6e67 2060 605f 5f69 6e69 745f  issing ``__init_
+00014280: 5f2e 7079 6060 2066 696c 6573 2074 6f20  _.py`` files to 
+00014290: 6120 636f 7570 6c65 206f 6620 7465 7374  a couple of test
+000142a0: 2064 6972 6563 746f 7269 6573 2073 6f20   directories so 
+000142b0: 7468 6174 2074 6865 7365 2074 6573 7473  that these tests
+000142c0: 2077 6f6e 2774 2062 6520 736b 6970 7065   won't be skippe
+000142d0: 642e 0a2a 2060 6070 6172 7365 5f66 6173  d..* ``parse_fas
+000142e0: 7471 6060 206e 6f77 2072 6169 7365 7320  tq`` now raises 
+000142f0: 616e 2065 7272 6f72 206f 6e20 6461 6e67  an error on dang
+00014300: 6c69 6e67 2072 6563 6f72 6473 2e0a 2a20  ling records..* 
+00014310: 4669 7865 6420 7365 7665 7261 6c20 7761  Fixed several wa
+00014320: 726e 696e 6773 2074 6861 7420 7765 7265  rnings that were
+00014330: 2072 6169 7365 6420 7768 696c 6520 7275   raised while ru
+00014340: 6e6e 696e 6720 7468 6520 7465 7374 2073  nning the test s
+00014350: 7569 7465 2077 6974 6820 5079 7468 6f6e  uite with Python
+00014360: 2033 2e34 2e0a 0a23 2323 2042 6163 6b77   3.4...### Backw
+00014370: 6172 642d 696e 636f 6d70 6174 6962 6c65  ard-incompatible
+00014380: 2063 6861 6e67 6573 0a0a 2a20 4675 6e63   changes..* Func
+00014390: 7469 6f6e 616c 6974 7920 696d 706f 7274  tionality import
+000143a0: 6564 2066 726f 6d20 6060 736b 6269 6f2e  ed from ``skbio.
+000143b0: 636f 7265 2e73 7377 6060 206d 7573 7420  core.ssw`` must 
+000143c0: 6e6f 7720 6265 2069 6d70 6f72 7465 6420  now be imported 
+000143d0: 6672 6f6d 2060 6073 6b62 696f 2e63 6f72  from ``skbio.cor
+000143e0: 652e 616c 6967 6e6d 656e 7460 6020 696e  e.alignment`` in
+000143f0: 7374 6561 642e 0a0a 2323 2320 4d69 7363  stead...### Misc
+00014400: 656c 6c61 6e65 6f75 730a 0a2a 2043 6f64  ellaneous..* Cod
+00014410: 6520 6973 206e 6f77 2066 6c61 6b65 382d  e is now flake8-
+00014420: 636f 6d70 6c69 616e 743b 2061 6464 6564  compliant; added
+00014430: 2066 6c61 6b65 3820 6368 6563 6b69 6e67   flake8 checking
+00014440: 2074 6f20 5472 6176 6973 2062 7569 6c64   to Travis build
+00014450: 2e0a 2a20 5661 7269 6f75 7320 6164 6469  ..* Various addi
+00014460: 7469 6f6e 7320 616e 6420 696d 7072 6f76  tions and improv
+00014470: 656d 656e 7473 2074 6f20 646f 6375 6d65  ements to docume
+00014480: 6e74 6174 696f 6e20 2841 5049 2c20 696e  ntation (API, in
+00014490: 7374 616c 6c61 7469 6f6e 2069 6e73 7472  stallation instr
+000144a0: 7563 7469 6f6e 732c 2064 6576 656c 6f70  uctions, develop
+000144b0: 6572 2069 6e73 7472 7563 7469 6f6e 732c  er instructions,
+000144c0: 2065 7463 2e29 2e0a 2a20 6060 5f5f 6675   etc.)..* ``__fu
+000144d0: 7475 7265 5f5f 6060 2069 6d70 6f72 7473  ture__`` imports
+000144e0: 2061 7265 206e 6f77 2073 7461 6e64 6172   are now standar
+000144f0: 6469 7a65 6420 6163 726f 7373 2074 6865  dized across the
+00014500: 2063 6f64 6562 6173 652e 0a2a 204e 6577   codebase..* New
+00014510: 2077 6562 7369 7465 2066 726f 6e74 2070   website front p
+00014520: 6167 6520 616e 6420 7374 796c 696e 6720  age and styling 
+00014530: 6368 616e 6765 7320 7468 726f 7567 686f  changes througho
+00014540: 7574 2e20 4d6f 7665 6420 646f 6373 2073  ut. Moved docs s
+00014550: 6974 6520 746f 2069 7473 206f 776e 2076  ite to its own v
+00014560: 6572 7369 6f6e 6564 2073 7562 6469 7265  ersioned subdire
+00014570: 6374 6f72 6965 732e 0a2a 2052 656f 7267  ctories..* Reorg
+00014580: 616e 697a 6564 2061 6c69 676e 6d65 6e74  anized alignment
+00014590: 2064 6174 6120 7374 7275 6374 7572 6573   data structures
+000145a0: 2061 6e64 2061 6c67 6f72 6974 686d 7320   and algorithms 
+000145b0: 2865 2e67 2e2c 2053 5357 2063 6f64 652c  (e.g., SSW code,
+000145c0: 2060 6041 6c69 676e 6d65 6e74 6060 2063   ``Alignment`` c
+000145d0: 6c61 7373 2c20 6574 632e 2920 696e 746f  lass, etc.) into
+000145e0: 2061 6e20 6060 736b 6269 6f2e 636f 7265   an ``skbio.core
+000145f0: 2e61 6c69 676e 6d65 6e74 6060 2073 7562  .alignment`` sub
+00014600: 7061 636b 6167 652e 0a0a 2323 2056 6572  package...## Ver
+00014610: 7369 6f6e 2030 2e31 2e31 2028 3230 3134  sion 0.1.1 (2014
+00014620: 2d30 352d 3136 290a 0a46 6978 6573 2074  -05-16)..Fixes t
+00014630: 6f20 7365 7475 702e 7079 2e20 5468 6973  o setup.py. This
+00014640: 2069 7320 6120 7072 652d 616c 7068 6120   is a pre-alpha 
+00014650: 7265 6c65 6173 652e 2041 7420 7468 6973  release. At this
+00014660: 2073 7461 6765 2c20 6d61 6a6f 7220 6261   stage, major ba
+00014670: 636b 7761 7264 732d 696e 636f 6d70 6174  ckwards-incompat
+00014680: 6962 6c65 2041 5049 2063 6861 6e67 6573  ible API changes
+00014690: 2063 616e 2061 6e64 2077 696c 6c20 6861   can and will ha
+000146a0: 7070 656e 2e0a 0a23 2320 5665 7273 696f  ppen...## Versio
+000146b0: 6e20 302e 312e 3020 2832 3031 342d 3035  n 0.1.0 (2014-05
+000146c0: 2d31 3529 0a0a 496e 6974 6961 6c20 7072  -15)..Initial pr
+000146d0: 652d 616c 7068 6120 7265 6c65 6173 652e  e-alpha release.
+000146e0: 2041 7420 7468 6973 2073 7461 6765 2c20   At this stage, 
+000146f0: 6d61 6a6f 7220 6261 636b 7761 7264 732d  major backwards-
+00014700: 696e 636f 6d70 6174 6962 6c65 2041 5049  incompatible API
+00014710: 2063 6861 6e67 6573 2063 616e 2061 6e64   changes can and
+00014720: 2077 696c 6c20 6861 7070 656e 2e0a        will happen..
```

### Comparing `scikit-bio-0.5.8/LICENSE.txt` & `scikit-bio-0.5.9/LICENSE.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/Makefile` & `scikit-bio-0.5.9/Makefile`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/PKG-INFO` & `scikit-bio-0.5.9/PKG-INFO`

 * *Files 1% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 1.2
 Name: scikit-bio
-Version: 0.5.8
+Version: 0.5.9
 Summary: Data structures, algorithms and educational resources for bioinformatics.
 Home-page: http://scikit-bio.org
 Author: scikit-bio development team
 Author-email: gregcaporaso@gmail.com
 Maintainer: scikit-bio development team
 Maintainer-email: gregcaporaso@gmail.com
 License: BSD-3-Clause
@@ -55,15 +55,15 @@
         To get help with scikit-bio, you should use the `skbio <http://stackoverflow.com/questions/tagged/skbio>`_ tag on StackOverflow (SO). Before posting a question, check out SO's guide on how to `ask a question <http://stackoverflow.com/questions/how-to-ask>`_. The scikit-bio developers regularly monitor the ``skbio`` SO tag.
         
         Projects using scikit-bio
         -------------------------
         
         Some of the projects that we know of that are using scikit-bio are:
         
-        - `QIIME <http://qiime.org/>`__
+        - `QIIME 2 <http://qiime2.org/>`__
         - `Emperor <http://biocore.github.io/emperor/>`__
         - `An Introduction to Applied
           Bioinformatics <http://readIAB.org>`__
         - `tax2tree <https://github.com/biocore/tax2tree>`__
         - `Qiita <http://qiita.microbio.me>`__
         - `ghost-tree <https://github.com/JTFouquier/ghost-tree>`__
         - `Platypus-Conquistador <https://github.com/biocore/Platypus-Conquistador>`__
```

### Comparing `scikit-bio-0.5.8/README.rst` & `scikit-bio-0.5.9/README.rst`

 * *Files 1% similar despite different names*

```diff
@@ -45,15 +45,15 @@
 To get help with scikit-bio, you should use the `skbio <http://stackoverflow.com/questions/tagged/skbio>`_ tag on StackOverflow (SO). Before posting a question, check out SO's guide on how to `ask a question <http://stackoverflow.com/questions/how-to-ask>`_. The scikit-bio developers regularly monitor the ``skbio`` SO tag.
 
 Projects using scikit-bio
 -------------------------
 
 Some of the projects that we know of that are using scikit-bio are:
 
-- `QIIME <http://qiime.org/>`__
+- `QIIME 2 <http://qiime2.org/>`__
 - `Emperor <http://biocore.github.io/emperor/>`__
 - `An Introduction to Applied
   Bioinformatics <http://readIAB.org>`__
 - `tax2tree <https://github.com/biocore/tax2tree>`__
 - `Qiita <http://qiita.microbio.me>`__
 - `ghost-tree <https://github.com/JTFouquier/ghost-tree>`__
 - `Platypus-Conquistador <https://github.com/biocore/Platypus-Conquistador>`__
```

### Comparing `scikit-bio-0.5.8/assets/.no.gif` & `scikit-bio-0.5.9/assets/.no.gif`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/assets/horizontal_powered_by.png` & `scikit-bio-0.5.9/assets/horizontal_powered_by.png`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/assets/horizontal_powered_by.svg` & `scikit-bio-0.5.9/assets/horizontal_powered_by.svg`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/assets/logo-inverted.svg` & `scikit-bio-0.5.9/assets/logo-inverted.svg`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/assets/logo.png` & `scikit-bio-0.5.9/assets/logo.png`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/assets/logo.svg` & `scikit-bio-0.5.9/assets/logo.svg`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/assets/logo_and_powered.pdf` & `scikit-bio-0.5.9/assets/logo_and_powered.pdf`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/assets/vertical_powered_by.png` & `scikit-bio-0.5.9/assets/vertical_powered_by.png`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/assets/vertical_powered_by.svg` & `scikit-bio-0.5.9/assets/vertical_powered_by.svg`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/Makefile` & `scikit-bio-0.5.9/doc/Makefile`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/README.md` & `scikit-bio-0.5.9/doc/README.md`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/source/_static/copybutton.js` & `scikit-bio-0.5.9/doc/source/_static/copybutton.js`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/source/_static/style.css` & `scikit-bio-0.5.9/doc/source/_static/style.css`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/source/_templates/autosummary/class.rst` & `scikit-bio-0.5.9/doc/source/_templates/autosummary/class.rst`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/source/_templates/layout.html` & `scikit-bio-0.5.9/doc/source/_templates/layout.html`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/source/conf.py` & `scikit-bio-0.5.9/doc/source/conf.py`

 * *Files 4% similar despite different names*

```diff
@@ -419,20 +419,20 @@
 
 matplotlib.rcParams.update(plot_rcparams)
 
 # -----------------------------------------------------------------------------
 # Intersphinx configuration
 # -----------------------------------------------------------------------------
 intersphinx_mapping = {
-        'http://docs.python.org/dev': None,
-        'http://docs.scipy.org/doc/numpy': None,
-        'http://docs.scipy.org/doc/scipy/reference': None,
-        'http://matplotlib.org': None,
-        'http://pandas.pydata.org/pandas-docs/stable': None,
-        'http://www.biom-format.org':None
+        'Python': ('http://docs.python.org/dev', None),
+        'NumPy': ('http://docs.scipy.org/doc/numpy', None),
+        'SciPy': ('http://docs.scipy.org/doc/scipy/reference', None),
+        'Matplotlib': ('http://matplotlib.org', None),
+        'Pandas': ('http://pandas.pydata.org/pandas-docs/stable', None),
+        'BIOM-Format': ('http://www.biom-format.org', None)
 }
 
 # -----------------------------------------------------------------------------
 # Source code links
 # -----------------------------------------------------------------------------
 
 import inspect
```

### Comparing `scikit-bio-0.5.8/doc/source/development/coding_guidelines.rst` & `scikit-bio-0.5.9/doc/source/development/coding_guidelines.rst`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/source/development/new_module.rst` & `scikit-bio-0.5.9/doc/source/development/new_module.rst`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/source/index.rst` & `scikit-bio-0.5.9/doc/source/index.rst`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/source/user/api_stability.rst` & `scikit-bio-0.5.9/doc/source/user/api_stability.rst`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/doc/source/user/assets/api-lifecycle.png` & `scikit-bio-0.5.9/doc/source/user/assets/api-lifecycle.png`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/bx_python.txt` & `scikit-bio-0.5.9/licenses/bx_python.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/fastq-example-files-readme.txt` & `scikit-bio-0.5.9/licenses/fastq-example-files-readme.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/flask.txt` & `scikit-bio-0.5.9/licenses/flask.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/ipython.txt` & `scikit-bio-0.5.9/licenses/ipython.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/nb-slideshow-template.txt` & `scikit-bio-0.5.9/licenses/nb-slideshow-template.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/qiita.txt` & `scikit-bio-0.5.9/licenses/qiita.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/scikit-learn.txt` & `scikit-bio-0.5.9/licenses/scikit-learn.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/scipy.txt` & `scikit-bio-0.5.9/licenses/scipy.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/simde.txt` & `scikit-bio-0.5.9/licenses/simde.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/sphinx-bootstrap-theme.txt` & `scikit-bio-0.5.9/licenses/sphinx-bootstrap-theme.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/ssw.txt` & `scikit-bio-0.5.9/licenses/ssw.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/licenses/verman.txt` & `scikit-bio-0.5.9/licenses/verman.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/scikit_bio.egg-info/PKG-INFO` & `scikit-bio-0.5.9/scikit_bio.egg-info/PKG-INFO`

 * *Files 1% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 1.2
 Name: scikit-bio
-Version: 0.5.8
+Version: 0.5.9
 Summary: Data structures, algorithms and educational resources for bioinformatics.
 Home-page: http://scikit-bio.org
 Author: scikit-bio development team
 Author-email: gregcaporaso@gmail.com
 Maintainer: scikit-bio development team
 Maintainer-email: gregcaporaso@gmail.com
 License: BSD-3-Clause
@@ -55,15 +55,15 @@
         To get help with scikit-bio, you should use the `skbio <http://stackoverflow.com/questions/tagged/skbio>`_ tag on StackOverflow (SO). Before posting a question, check out SO's guide on how to `ask a question <http://stackoverflow.com/questions/how-to-ask>`_. The scikit-bio developers regularly monitor the ``skbio`` SO tag.
         
         Projects using scikit-bio
         -------------------------
         
         Some of the projects that we know of that are using scikit-bio are:
         
-        - `QIIME <http://qiime.org/>`__
+        - `QIIME 2 <http://qiime2.org/>`__
         - `Emperor <http://biocore.github.io/emperor/>`__
         - `An Introduction to Applied
           Bioinformatics <http://readIAB.org>`__
         - `tax2tree <https://github.com/biocore/tax2tree>`__
         - `Qiita <http://qiita.microbio.me>`__
         - `ghost-tree <https://github.com/JTFouquier/ghost-tree>`__
         - `Platypus-Conquistador <https://github.com/biocore/Platypus-Conquistador>`__
```

### Comparing `scikit-bio-0.5.8/scikit_bio.egg-info/SOURCES.txt` & `scikit-bio-0.5.9/scikit_bio.egg-info/SOURCES.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/setup.py` & `scikit-bio-0.5.9/setup.py`

 * *Files 5% similar despite different names*

```diff
@@ -146,23 +146,22 @@
       maintainer_email="gregcaporaso@gmail.com",
       url='http://scikit-bio.org',
       packages=find_packages(),
       ext_modules=extensions,
       include_dirs=[np.get_include()],
       tests_require=['pytest', 'coverage'],
       install_requires=[
-          'lockfile >= 0.10.2',  # req'd for our usage of CacheControl
-          'CacheControl >= 0.11.5',
+          'requests >= 2.20.0',
           'decorator >= 3.4.2',
           'IPython >= 3.2.0',
           'matplotlib >= 1.4.3',
           'natsort >= 4.0.3',
           'numpy >= 1.9.2',
           'pandas >= 1.5.0',
-          'scipy >= 1.9.0',
+          'scipy <= 1.10.1',
           'h5py >= 3.6.0',
           'hdmedians >= 0.14.1',
       ],
       classifiers=classifiers,
       package_data={
           'skbio.diversity.alpha.tests': ['data/qiime-191-tt/*'],
           'skbio.diversity.beta.tests': ['data/qiime-191-tt/*'],
```

### Comparing `scikit-bio-0.5.8/simde-sse2.h` & `scikit-bio-0.5.9/simde-sse2.h`

 * *Files 1% similar despite different names*

```diff
@@ -1,9 +1,9 @@
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin x86/sse2.h :: */
 /* SPDX-License-Identifier: MIT
  *
  * Permission is hereby granted, free of charge, to any person
  * obtaining a copy of this software and associated documentation
  * files (the "Software"), to deal in the Software without
  * restriction, including without limitation the rights to use, copy,
@@ -32,15 +32,15 @@
  *   2018      Jeff Daily <jeff.daily@amd.com>
  */
 
 #if !defined(SIMDE_X86_SSE2_H)
 #define SIMDE_X86_SSE2_H
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin x86/sse.h :: */
 /* SPDX-License-Identifier: MIT
  *
  * Permission is hereby granted, free of charge, to any person
  * obtaining a copy of this software and associated documentation
  * files (the "Software"), to deal in the Software without
  * restriction, including without limitation the rights to use, copy,
@@ -67,15 +67,15 @@
  *   2015      Ken Fast <kfast@gdeb.com>
  */
 
 #if !defined(SIMDE_X86_SSE_H)
 #define SIMDE_X86_SSE_H
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin x86/mmx.h :: */
 /* SPDX-License-Identifier: MIT
  *
  * Permission is hereby granted, free of charge, to any person
  * obtaining a copy of this software and associated documentation
  * files (the "Software"), to deal in the Software without
  * restriction, including without limitation the rights to use, copy,
@@ -99,15 +99,15 @@
  *   2017-2020 Evan Nemerson <evan@nemerson.com>
  */
 
 #if !defined(SIMDE_X86_MMX_H)
 #define SIMDE_X86_MMX_H
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin simde-common.h :: */
 /* SPDX-License-Identifier: MIT
  *
  * Permission is hereby granted, free of charge, to any person
  * obtaining a copy of this software and associated documentation
  * files (the "Software"), to deal in the Software without
  * restriction, including without limitation the rights to use, copy,
@@ -131,15 +131,15 @@
  *   2017-2020 Evan Nemerson <evan@nemerson.com>
  */
 
 #if !defined(SIMDE_COMMON_H)
 #define SIMDE_COMMON_H
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin hedley.h :: */
 /* Hedley - https://nemequ.github.io/hedley
  * Created by Evan Nemerson <evan@nemerson.com>
  *
  * To the extent possible under law, the author(s) have dedicated all
  * copyright and related and neighboring rights to this software to
  * the public domain worldwide. This software is distributed without
@@ -2181,23 +2181,23 @@
 #define HEDLEY_CLANG_HAS_WARNING(warning) HEDLEY_HAS_WARNING(warning)
 
 #endif /* !defined(HEDLEY_VERSION) || (HEDLEY_VERSION < X) */
 /* :: End hedley.h :: */
 
 #define SIMDE_VERSION_MAJOR 0
 #define SIMDE_VERSION_MINOR 7
-#define SIMDE_VERSION_MICRO 3
+#define SIMDE_VERSION_MICRO 6
 #define SIMDE_VERSION HEDLEY_VERSION_ENCODE(SIMDE_VERSION_MAJOR, SIMDE_VERSION_MINOR, SIMDE_VERSION_MICRO)
 // Also update meson.build in the root directory of the repository
 
 #include <stddef.h>
 #include <stdint.h>
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin simde-detect-clang.h :: */
 /* Detect Clang Version
  * Created by Evan Nemerson <evan@nemerson.com>
  *
  * To the extent possible under law, the author(s) have dedicated all
  * copyright and related and neighboring rights to this software to
  * the public domain worldwide. This software is distributed without
@@ -2247,28 +2247,44 @@
 #define SIMDE_DETECT_CLANG_H 1
 
 /* Attempt to detect the upstream clang version number.  I usually only
  * worry about major version numbers (at least for 4.0+), but if you
  * need more resolution I'm happy to accept patches that are able to
  * detect minor versions as well.  That said, you'll probably have a
  * hard time with detection since AFAIK most minor releases don't add
- * anything we can detect. */
+ * anything we can detect. Updated based on
+ * https://github.com/google/highway/blob/438c705a295176b96a50336527bb3e7ea365ffac/hwy/detect_compiler_arch.h#L73
+ * - would welcome patches/updates there as well.
+ */
 
 #if defined(__clang__) && !defined(SIMDE_DETECT_CLANG_VERSION)
-#  if __has_warning("-Wformat-insufficient-args")
+#  if __has_attribute(nouwtable)  // no new warnings in 16.0
+#    define SIMDE_DETECT_CLANG_VERSION 160000
+#  elif __has_warning("-Warray-parameter")
+#    define SIMDE_DETECT_CLANG_VERSION 150000
+#  elif __has_warning("-Wbitwise-instead-of-logical")
+#    define SIMDE_DETECT_CLANG_VERSION 140000
+#  elif __has_warning("-Wwaix-compat")
+#    define SIMDE_DETECT_CLANG_VERSION 130000
+#  elif __has_warning("-Wformat-insufficient-args")
 #    define SIMDE_DETECT_CLANG_VERSION 120000
 #  elif __has_warning("-Wimplicit-const-int-float-conversion")
 #    define SIMDE_DETECT_CLANG_VERSION 110000
 #  elif __has_warning("-Wmisleading-indentation")
 #    define SIMDE_DETECT_CLANG_VERSION 100000
 #  elif defined(__FILE_NAME__)
 #    define SIMDE_DETECT_CLANG_VERSION 90000
 #  elif __has_warning("-Wextra-semi-stmt") || __has_builtin(__builtin_rotateleft32)
 #    define SIMDE_DETECT_CLANG_VERSION 80000
-#  elif __has_warning("-Wc++98-compat-extra-semi")
+// For reasons unknown, XCode 10.3 (Apple LLVM version 10.0.1) is apparently
+// based on Clang 7, but does not support the warning we test.
+// See https://en.wikipedia.org/wiki/Xcode#Toolchain_versions and
+// https://trac.macports.org/wiki/XcodeVersionInfo.
+#  elif __has_warning("-Wc++98-compat-extra-semi") || \
+      (defined(__apple_build_version__) && __apple_build_version__ >= 10010000)
 #    define SIMDE_DETECT_CLANG_VERSION 70000
 #  elif __has_warning("-Wpragma-pack")
 #    define SIMDE_DETECT_CLANG_VERSION 60000
 #  elif __has_warning("-Wbitfield-enum-conversion")
 #    define SIMDE_DETECT_CLANG_VERSION 50000
 #  elif __has_attribute(diagnose_if)
 #    define SIMDE_DETECT_CLANG_VERSION 40000
@@ -2302,15 +2318,15 @@
 #  define SIMDE_DETECT_CLANG_VERSION_CHECK(major, minor, revision) (0)
 #  define SIMDE_DETECT_CLANG_VERSION_NOT(major, minor, revision) (0)
 #endif
 
 #endif /* !defined(SIMDE_DETECT_CLANG_H) */
 /* :: End simde-detect-clang.h :: */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin simde-arch.h :: */
 /* Architecture detection
  * Created by Evan Nemerson <evan@nemerson.com>
  *
  *   To the extent possible under law, the authors have waived all
  *   copyright and related or neighboring rights to this code.  For
  *   details, see the Creative Commons Zero 1.0 Universal license at
@@ -2348,14 +2364,17 @@
  * your project, even if you change the prefix, please keep the links
  * to SIMDe intact so others know where to report issues, submit
  * enhancements, and find the latest version. */
 
 #if !defined(SIMDE_ARCH_H)
 #define SIMDE_ARCH_H
 
+/* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
+
 /* Alpha
    <https://en.wikipedia.org/wiki/DEC_Alpha> */
 #if defined(__alpha__) || defined(__alpha) || defined(_M_ALPHA)
 #  if defined(__alpha_ev6__)
 #    define SIMDE_ARCH_ALPHA 6
 #  elif defined(__alpha_ev5__)
 #    define SIMDE_ARCH_ALPHA 5
@@ -2579,14 +2598,17 @@
 #    endif
 #    if !defined(SIMDE_ARCH_X86_SSE4_1)
 #      define SIMDE_ARCH_X86_SSE4_2 1
 #    endif
 #  endif
 #  if defined(__AVX2__)
 #    define SIMDE_ARCH_X86_AVX2 1
+#    if defined(_MSC_VER)
+#      define SIMDE_ARCH_X86_FMA 1
+#    endif
 #  endif
 #  if defined(__FMA__)
 #    define SIMDE_ARCH_X86_FMA 1
 #    if !defined(SIMDE_ARCH_X86_AVX)
 #      define SIMDE_ARCH_X86_AVX 1
 #    endif
 #  endif
@@ -2634,15 +2656,15 @@
 #  endif
 #  if defined(__PCLMUL__)
 #    define SIMDE_ARCH_X86_PCLMUL 1
 #  endif
 #  if defined(__VPCLMULQDQ__)
 #    define SIMDE_ARCH_X86_VPCLMULQDQ 1
 #  endif
-#  if defined(__F16C__)
+#  if defined(__F16C__) || (defined(HEDLEY_MSVC_VERSION) && HEDLEY_MSVC_VERSION_CHECK(19,30,0) && defined(SIMDE_ARCH_X86_AVX2) )
 #    define SIMDE_ARCH_X86_F16C 1
 #  endif
 #endif
 
 /* Itanium
    <https://en.wikipedia.org/wiki/Itanium> */
 #if defined(__ia64__) || defined(_IA64) || defined(__IA64__) || defined(__ia64) || defined(_M_IA64) || defined(__itanium__)
@@ -2869,18 +2891,41 @@
 
 /* Xtensa
    <https://en.wikipedia.org/wiki/> */
 #if defined(__xtensa__) || defined(__XTENSA__)
 #  define SIMDE_ARCH_XTENSA 1
 #endif
 
+/* Availability of 16-bit floating-point arithmetic intrinsics */
+#if defined(__ARM_FEATURE_FP16_VECTOR_ARITHMETIC)
+#  define SIMDE_ARCH_ARM_NEON_FP16
+#endif
+
+/* LoongArch
+   <https://en.wikipedia.org/wiki/Loongson#LoongArch> */
+#if defined(__loongarch32)
+#  define SIMDE_ARCH_LOONGARCH 1
+#elif defined(__loongarch64)
+#  define SIMDE_ARCH_LOONGARCH 2
+#endif
+
+/* LSX: LoongArch 128-bits SIMD extension */
+#if defined(__loongarch_sx)
+#  define SIMDE_ARCH_LOONGARCH_LSX 1
+#endif
+
+/* LASX: LoongArch 256-bits SIMD extension */
+#if defined(__loongarch_asx)
+#  define SIMDE_ARCH_LOONGARCH_LASX 2
+#endif
+
 #endif /* !defined(SIMDE_ARCH_H) */
 /* :: End simde-arch.h :: */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin simde-features.h :: */
 /* SPDX-License-Identifier: MIT
  *
  * Permission is hereby granted, free of charge, to any person
  * obtaining a copy of this software and associated documentation
  * files (the "Software"), to deal in the Software without
  * restriction, including without limitation the rights to use, copy,
@@ -2908,17 +2953,17 @@
    to the compiler.  However, we want to make it possible to forcibly enable
    or disable APIs */
 
 #if !defined(SIMDE_FEATURES_H)
 #define SIMDE_FEATURES_H
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin simde-diagnostic.h :: */
 /* SPDX-License-Identifier: MIT
  *
  * Permission is hereby granted, free of charge, to any person
  * obtaining a copy of this software and associated documentation
  * files (the "Software"), to deal in the Software without
  * restriction, including without limitation the rights to use, copy,
@@ -2962,19 +3007,19 @@
  * SIMDe is not deprived of these warnings.
  */
 
 #if !defined(SIMDE_DIAGNOSTIC_H)
 #define SIMDE_DIAGNOSTIC_H
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 
 /* This is only to help us implement functions like _mm_undefined_ps. */
 #if defined(SIMDE_DIAGNOSTIC_DISABLE_UNINITIALIZED_)
   #undef SIMDE_DIAGNOSTIC_DISABLE_UNINITIALIZED_
 #endif
 #if HEDLEY_HAS_WARNING("-Wuninitialized")
   #define SIMDE_DIAGNOSTIC_DISABLE_UNINITIALIZED_ _Pragma("clang diagnostic ignored \"-Wuninitialized\"")
@@ -3198,14 +3243,22 @@
  * the compiler and version first, so we know it's not a problem. */
 #if HEDLEY_HAS_WARNING("-Wc99-extensions")
   #define SIMDE_DIAGNOSTIC_DISABLE_C99_EXTENSIONS_ _Pragma("clang diagnostic ignored \"-Wc99-extensions\"")
 #else
   #define SIMDE_DIAGNOSTIC_DISABLE_C99_EXTENSIONS_
 #endif
 
+/* Similar problm as above; we rely on some basic C99 support, but clang
+ * has started warning obut this even in C17 mode with -Weverything. */
+#if HEDLEY_HAS_WARNING("-Wdeclaration-after-statement")
+  #define SIMDE_DIAGNOSTIC_DISABLE_DECLARATION_AFTER_STATEMENT_ _Pragma("clang diagnostic ignored \"-Wdeclaration-after-statement\"")
+#else
+  #define SIMDE_DIAGNOSTIC_DISABLE_DECLARATION_AFTER_STATEMENT_
+#endif
+
 /* https://github.com/simd-everywhere/simde/issues/277 */
 #if defined(HEDLEY_GCC_VERSION) && HEDLEY_GCC_VERSION_CHECK(4,6,0) && !HEDLEY_GCC_VERSION_CHECK(6,4,0) && defined(__cplusplus)
   #define SIMDE_DIAGNOSTIC_DISABLE_BUGGY_UNUSED_BUT_SET_VARIBALE_ _Pragma("GCC diagnostic ignored \"-Wunused-but-set-variable\"")
 #else
   #define SIMDE_DIAGNOSTIC_DISABLE_BUGGY_UNUSED_BUT_SET_VARIBALE_
 #endif
 
@@ -3309,14 +3362,16 @@
 /* MSVC doesn't like (__assume(0), code) and will warn about code being
  * unreachable, but we want it there because not all compilers
  * understand the unreachable macro and will complain if it is missing.
  * I'm planning on adding a new macro to Hedley to handle this a bit
  * more elegantly, but until then... */
 #if defined(HEDLEY_MSVC_VERSION)
   #define SIMDE_DIAGNOSTIC_DISABLE_UNREACHABLE_ __pragma(warning(disable:4702))
+#elif defined(__clang__)
+  #define SIMDE_DIAGNOSTIC_DISABLE_UNREACHABLE_ HEDLEY_PRAGMA(clang diagnostic ignored "-Wunreachable-code")
 #else
   #define SIMDE_DIAGNOSTIC_DISABLE_UNREACHABLE_
 #endif
 
 /* This is a false positive from GCC in a few places. */
 #if HEDLEY_GCC_VERSION_CHECK(4,7,0)
   #define SIMDE_DIAGNOSTIC_DISABLE_MAYBE_UNINITIAZILED_ _Pragma("GCC diagnostic ignored \"-Wmaybe-uninitialized\"")
@@ -3346,14 +3401,15 @@
 #define SIMDE_DISABLE_UNWANTED_DIAGNOSTICS \
   HEDLEY_DIAGNOSTIC_DISABLE_UNUSED_FUNCTION \
   SIMDE_DISABLE_UNWANTED_DIAGNOSTICS_NATIVE_ALIASES_ \
   SIMDE_DIAGNOSTIC_DISABLE_PSABI_ \
   SIMDE_DIAGNOSTIC_DISABLE_NO_EMMS_INSTRUCTION_ \
   SIMDE_DIAGNOSTIC_DISABLE_SIMD_PRAGMA_DEPRECATED_ \
   SIMDE_DIAGNOSTIC_DISABLE_CONDITIONAL_UNINITIALIZED_ \
+  SIMDE_DIAGNOSTIC_DISABLE_DECLARATION_AFTER_STATEMENT_ \
   SIMDE_DIAGNOSTIC_DISABLE_FLOAT_EQUAL_ \
   SIMDE_DIAGNOSTIC_DISABLE_NON_CONSTANT_AGGREGATE_INITIALIZER_ \
   SIMDE_DIAGNOSTIC_DISABLE_EXTRA_SEMI_ \
   SIMDE_DIAGNOSTIC_DISABLE_VLA_ \
   SIMDE_DIAGNOSTIC_DISABLE_USED_BUT_MARKED_UNUSED_ \
   SIMDE_DIAGNOSTIC_DISABLE_PASS_FAILED_ \
   SIMDE_DIAGNOSTIC_DISABLE_CPP98_COMPAT_PEDANTIC_ \
@@ -3511,15 +3567,15 @@
 #endif
 
 #if !defined(SIMDE_X86_AVX_NATIVE) && !defined(SIMDE_X86_AVX_NO_NATIVE) && !defined(SIMDE_NO_NATIVE)
   #if defined(SIMDE_ARCH_X86_AVX)
     #define SIMDE_X86_AVX_NATIVE
   #endif
 #endif
-#if defined(SIMDE_X86_AVX_NATIVE) && !defined(SIMDE_X86_SSE4_1_NATIVE)
+#if defined(SIMDE_X86_AVX_NATIVE) && !defined(SIMDE_X86_SSE4_2_NATIVE)
   #define SIMDE_X86_SSE4_2_NATIVE
 #endif
 
 #if !defined(SIMDE_X86_XOP_NATIVE) && !defined(SIMDE_X86_XOP_NO_NATIVE) && !defined(SIMDE_NO_NATIVE)
   #if defined(SIMDE_ARCH_X86_XOP)
     #define SIMDE_X86_XOP_NATIVE
   #endif
@@ -3821,38 +3877,75 @@
 #if defined(SIMDE_MIPS_MSA_NATIVE)
   #include <msa.h>
 #endif
 
 /* This is used to determine whether or not to fall back on a vector
  * function in an earlier ISA extensions, as well as whether
  * we expected any attempts at vectorization to be fruitful or if we
- * expect to always be running serial code. */
+ * expect to always be running serial code.
+ *
+ * Note that, for some architectures (okay, *one* architecture) there
+ * can be a split where some types are supported for one vector length
+ * but others only for a shorter length.  Therefore, it is possible to
+ * provide separate values for float/int/double types. */
 
 #if !defined(SIMDE_NATURAL_VECTOR_SIZE)
   #if defined(SIMDE_X86_AVX512F_NATIVE)
     #define SIMDE_NATURAL_VECTOR_SIZE (512)
-  #elif defined(SIMDE_X86_AVX_NATIVE)
+  #elif defined(SIMDE_X86_AVX2_NATIVE)
     #define SIMDE_NATURAL_VECTOR_SIZE (256)
+  #elif defined(SIMDE_X86_AVX_NATIVE)
+    #define SIMDE_NATURAL_FLOAT_VECTOR_SIZE (256)
+    #define SIMDE_NATURAL_INT_VECTOR_SIZE (128)
+    #define SIMDE_NATURAL_DOUBLE_VECTOR_SIZE (128)
   #elif \
-      defined(SIMDE_X86_SSE_NATIVE) || \
+      defined(SIMDE_X86_SSE2_NATIVE) || \
       defined(SIMDE_ARM_NEON_A32V7_NATIVE) || \
       defined(SIMDE_WASM_SIMD128_NATIVE) || \
       defined(SIMDE_POWER_ALTIVEC_P5_NATIVE) || \
       defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE) || \
       defined(SIMDE_MIPS_MSA_NATIVE)
     #define SIMDE_NATURAL_VECTOR_SIZE (128)
+  #elif defined(SIMDE_X86_SSE_NATIVE)
+    #define SIMDE_NATURAL_FLOAT_VECTOR_SIZE (128)
+    #define SIMDE_NATURAL_INT_VECTOR_SIZE (64)
+    #define SIMDE_NATURAL_DOUBLE_VECTOR_SIZE (0)
   #endif
 
   #if !defined(SIMDE_NATURAL_VECTOR_SIZE)
-    #define SIMDE_NATURAL_VECTOR_SIZE (0)
+    #if defined(SIMDE_NATURAL_FLOAT_VECTOR_SIZE)
+      #define SIMDE_NATURAL_VECTOR_SIZE SIMDE_NATURAL_FLOAT_VECTOR_SIZE
+    #elif defined(SIMDE_NATURAL_INT_VECTOR_SIZE)
+      #define SIMDE_NATURAL_VECTOR_SIZE SIMDE_NATURAL_INT_VECTOR_SIZE
+    #elif defined(SIMDE_NATURAL_DOUBLE_VECTOR_SIZE)
+      #define SIMDE_NATURAL_VECTOR_SIZE SIMDE_NATURAL_DOUBLE_VECTOR_SIZE
+    #else
+      #define SIMDE_NATURAL_VECTOR_SIZE (0)
+    #endif
+  #endif
+
+  #if !defined(SIMDE_NATURAL_FLOAT_VECTOR_SIZE)
+    #define SIMDE_NATURAL_FLOAT_VECTOR_SIZE SIMDE_NATURAL_VECTOR_SIZE
+  #endif
+  #if !defined(SIMDE_NATURAL_INT_VECTOR_SIZE)
+    #define SIMDE_NATURAL_INT_VECTOR_SIZE SIMDE_NATURAL_VECTOR_SIZE
+  #endif
+  #if !defined(SIMDE_NATURAL_DOUBLE_VECTOR_SIZE)
+    #define SIMDE_NATURAL_DOUBLE_VECTOR_SIZE SIMDE_NATURAL_VECTOR_SIZE
   #endif
 #endif
 
 #define SIMDE_NATURAL_VECTOR_SIZE_LE(x) ((SIMDE_NATURAL_VECTOR_SIZE > 0) && (SIMDE_NATURAL_VECTOR_SIZE <= (x)))
 #define SIMDE_NATURAL_VECTOR_SIZE_GE(x) ((SIMDE_NATURAL_VECTOR_SIZE > 0) && (SIMDE_NATURAL_VECTOR_SIZE >= (x)))
+#define SIMDE_NATURAL_FLOAT_VECTOR_SIZE_LE(x) ((SIMDE_NATURAL_FLOAT_VECTOR_SIZE > 0) && (SIMDE_NATURAL_FLOAT_VECTOR_SIZE <= (x)))
+#define SIMDE_NATURAL_FLOAT_VECTOR_SIZE_GE(x) ((SIMDE_NATURAL_FLOAT_VECTOR_SIZE > 0) && (SIMDE_NATURAL_FLOAT_VECTOR_SIZE >= (x)))
+#define SIMDE_NATURAL_INT_VECTOR_SIZE_LE(x) ((SIMDE_NATURAL_INT_VECTOR_SIZE > 0) && (SIMDE_NATURAL_INT_VECTOR_SIZE <= (x)))
+#define SIMDE_NATURAL_INT_VECTOR_SIZE_GE(x) ((SIMDE_NATURAL_INT_VECTOR_SIZE > 0) && (SIMDE_NATURAL_INT_VECTOR_SIZE >= (x)))
+#define SIMDE_NATURAL_DOUBLE_VECTOR_SIZE_LE(x) ((SIMDE_NATURAL_DOUBLE_VECTOR_SIZE > 0) && (SIMDE_NATURAL_DOUBLE_VECTOR_SIZE <= (x)))
+#define SIMDE_NATURAL_DOUBLE_VECTOR_SIZE_GE(x) ((SIMDE_NATURAL_DOUBLE_VECTOR_SIZE > 0) && (SIMDE_NATURAL_DOUBLE_VECTOR_SIZE >= (x)))
 
 /* Native aliases */
 #if defined(SIMDE_ENABLE_NATIVE_ALIASES)
   #if !defined(SIMDE_X86_MMX_NATIVE)
     #define SIMDE_X86_MMX_ENABLE_NATIVE_ALIASES
   #endif
   #if !defined(SIMDE_X86_SSE_NATIVE)
@@ -3969,20 +4062,43 @@
  * this, if that test fails sowewhere please let us know and we'll add
  * an exception for that platform.  Meanwhile, you can define
  * SIMDE_NO_IEEE754_STORAGE. */
 #if !defined(SIMDE_IEEE754_STORAGE) && !defined(SIMDE_NO_IEE754_STORAGE)
   #define SIMDE_IEEE754_STORAGE
 #endif
 
+#if defined(SIMDE_ARCH_ARM_NEON_FP16)
+  #define SIMDE_ARM_NEON_FP16
+#endif
+
+#if !defined(SIMDE_LOONGARCH_LASX_NATIVE) && !defined(SIMDE_LOONGARCH_LASX_NO_NATIVE) && !defined(SIMDE_NO_NATIVE)
+  #if defined(SIMDE_ARCH_LOONGARCH_LASX)
+    #define SIMDE_LOONGARCH_LASX_NATIVE
+  #endif
+#endif
+
+#if !defined(SIMDE_LOONGARCH_LSX_NATIVE) && !defined(SIMDE_LOONGARCH_LSX_NO_NATIVE) && !defined(SIMDE_NO_NATIVE)
+  #if defined(SIMDE_ARCH_LOONGARCH_LSX)
+    #define SIMDE_LOONGARCH_LSX_NATIVE
+  #endif
+#endif
+
+#if defined(SIMDE_LOONGARCH_LASX_NATIVE)
+  #include <lasxintrin.h>
+#endif
+#if defined(SIMDE_LOONGARCH_LSX_NATIVE)
+  #include <lsxintrin.h>
+#endif
+
 #endif /* !defined(SIMDE_FEATURES_H) */
 /* :: End simde-features.h :: */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin simde-math.h :: */
 /* SPDX-License-Identifier: MIT
  *
  * Permission is hereby granted, free of charge, to any person
  * obtaining a copy of this software and associated documentation
  * files (the "Software"), to deal in the Software without
  * restriction, including without limitation the rights to use, copy,
@@ -4011,17 +4127,17 @@
  * specific headers.  In some cases, especially those not built in to
  * libm, we may need to define our own implementations. */
 
 #if !defined(SIMDE_MATH_H)
 #define SIMDE_MATH_H 1
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 
 #include <stdint.h>
 #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
   #include <arm_neon.h>
 #endif
 
 HEDLEY_DIAGNOSTIC_PUSH
@@ -4686,14 +4802,28 @@
   #elif defined(SIMDE_MATH_HAVE_CMATH)
     #define simde_math_copysignf(x, y) std::copysignf(x, y)
   #elif defined(SIMDE_MATH_HAVE_MATH_H)
     #define simde_math_copysignf(x, y) copysignf(x, y)
   #endif
 #endif
 
+#if !defined(simde_math_signbit)
+  #if SIMDE_MATH_BUILTIN_LIBM(signbit)
+    #if (!defined(__clang__) || SIMDE_DETECT_CLANG_VERSION_CHECK(7,0,0))
+      #define simde_math_signbit(x) __builtin_signbit(x)
+    #else
+      #define simde_math_signbit(x) __builtin_signbit(HEDLEY_STATIC_CAST(double, (x)))
+    #endif
+  #elif defined(SIMDE_MATH_HAVE_CMATH)
+    #define simde_math_signbit(x) std::signbit(x)
+  #elif defined(SIMDE_MATH_HAVE_MATH_H)
+    #define simde_math_signbit(x) signbit(x)
+  #endif
+#endif
+
 #if !defined(simde_math_cos)
   #if SIMDE_MATH_BUILTIN_LIBM(cos)
     #define simde_math_cos(v) __builtin_cos(v)
   #elif defined(SIMDE_MATH_HAVE_CMATH)
     #define simde_math_cos(v) std::cos(v)
   #elif defined(SIMDE_MATH_HAVE_MATH_H)
     #define simde_math_cos(v) cos(v)
@@ -5146,15 +5276,15 @@
   #elif defined(SIMDE_MATH_HAVE_MATH_H)
     #define simde_math_roundf(v) roundf(v)
   #endif
 #endif
 
 #if !defined(simde_math_roundeven)
   #if \
-      HEDLEY_HAS_BUILTIN(__builtin_roundeven) || \
+     (!defined(HEDLEY_EMSCRIPTEN_VERSION) && HEDLEY_HAS_BUILTIN(__builtin_roundeven)) || \
       HEDLEY_GCC_VERSION_CHECK(10,0,0)
     #define simde_math_roundeven(v) __builtin_roundeven(v)
   #elif defined(simde_math_round) && defined(simde_math_fabs)
     static HEDLEY_INLINE
     double
     simde_math_roundeven(double v) {
       double rounded = simde_math_round(v);
@@ -5166,15 +5296,15 @@
     }
     #define simde_math_roundeven simde_math_roundeven
   #endif
 #endif
 
 #if !defined(simde_math_roundevenf)
   #if \
-      HEDLEY_HAS_BUILTIN(__builtin_roundevenf) || \
+     (!defined(HEDLEY_EMSCRIPTEN_VERSION) && HEDLEY_HAS_BUILTIN(__builtin_roundevenf)) || \
       HEDLEY_GCC_VERSION_CHECK(10,0,0)
     #define simde_math_roundevenf(v) __builtin_roundevenf(v)
   #elif defined(simde_math_roundf) && defined(simde_math_fabsf)
     static HEDLEY_INLINE
     float
     simde_math_roundevenf(float v) {
       float rounded = simde_math_roundf(v);
@@ -5379,17 +5509,14 @@
     float y = 1.0f - (((((a5 * t + a4) * t) + a3) * t + a2) * t + a1) * t * simde_math_expf(-x * x);
 
     return 0.5f * (1.0f + (sign ? -y : y));
   }
   #define simde_math_cdfnormf simde_math_cdfnormf
 #endif
 
-HEDLEY_DIAGNOSTIC_PUSH
-SIMDE_DIAGNOSTIC_DISABLE_FLOAT_EQUAL_
-
 #if !defined(simde_math_cdfnorminv) && defined(simde_math_log) && defined(simde_math_sqrt)
   /*https://web.archive.org/web/20150910081113/http://home.online.no/~pjacklam/notes/invnorm/impl/sprouse/ltqnorm.c*/
   static HEDLEY_INLINE
   double
   simde_math_cdfnorminv(double p) {
     static const double a[] = {
       -3.969683028665376e+01,
@@ -5564,39 +5691,39 @@
 #if !defined(simde_math_erfcinv) && defined(simde_math_erfinv) && defined(simde_math_log) && defined(simde_math_sqrt)
   static HEDLEY_INLINE
   double
   simde_math_erfcinv(double x) {
     if(x >= 0.0625 && x < 2.0) {
       return simde_math_erfinv(1.0 - x);
     } else if (x < 0.0625 && x >= 1.0e-100) {
-      double p[6] = {
+      static const double p[6] = {
         0.1550470003116,
         1.382719649631,
         0.690969348887,
         -1.128081391617,
         0.680544246825,
         -0.16444156791
       };
-      double q[3] = {
+      static const double q[3] = {
         0.155024849822,
         1.385228141995,
         1.000000000000
       };
 
       const double t = 1.0 / simde_math_sqrt(-simde_math_log(x));
       return (p[0] / t + p[1] + t * (p[2] + t * (p[3] + t * (p[4] + t * p[5])))) /
             (q[0] + t * (q[1] + t * (q[2])));
     } else if (x < 1.0e-100 && x >= SIMDE_MATH_DBL_MIN) {
-      double p[4] = {
+      static const double p[4] = {
         0.00980456202915,
         0.363667889171,
         0.97302949837,
         -0.5374947401
       };
-      double q[3] = {
+      static const double q[3] = {
         0.00980451277802,
         0.363699971544,
         1.000000000000
       };
 
       const double t = 1.0 / simde_math_sqrt(-simde_math_log(x));
       return (p[0] / t + p[1] + t * (p[2] + t * p[3])) /
@@ -5655,16 +5782,14 @@
       return simde_math_isnormalf(x) ? -SIMDE_MATH_INFINITYF : SIMDE_MATH_INFINITYF;
     }
   }
 
   #define simde_math_erfcinvf simde_math_erfcinvf
 #endif
 
-HEDLEY_DIAGNOSTIC_POP
-
 static HEDLEY_INLINE
 double
 simde_math_rad2deg(double radians) {
  return radians * SIMDE_MATH_180_OVER_PI;
 }
 
 static HEDLEY_INLINE
@@ -5940,15 +6065,15 @@
 }
 
 HEDLEY_DIAGNOSTIC_POP
 
 #endif /* !defined(SIMDE_MATH_H) */
 /* :: End simde-math.h :: */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin simde-constify.h :: */
 /* SPDX-License-Identifier: MIT
  *
  * Permission is hereby granted, free of charge, to any person
  * obtaining a copy of this software and associated documentation
  * files (the "Software"), to deal in the Software without
  * restriction, including without limitation the rights to use, copy,
@@ -5995,15 +6120,15 @@
  * to use non-ICE parameters
  */
 
 #if !defined(SIMDE_CONSTIFY_H)
 #define SIMDE_CONSTIFY_H
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 
 HEDLEY_DIAGNOSTIC_PUSH
 SIMDE_DIAGNOSTIC_DISABLE_VARIADIC_MACROS_
 SIMDE_DIAGNOSTIC_DISABLE_CPP98_COMPAT_PEDANTIC_
 
 #define SIMDE_CONSTIFY_2_(func_name, result, default_case, imm, ...) \
   do { \
@@ -6342,15 +6467,15 @@
   } while (0)
 
 HEDLEY_DIAGNOSTIC_POP
 
 #endif
 /* :: End simde-constify.h :: */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin simde-align.h :: */
 /* Alignment
  * Created by Evan Nemerson <evan@nemerson.com>
  *
  *   To the extent possible under law, the authors have waived all
  *   copyright and related or neighboring rights to this code.  For
  *   details, see the Creative Commons Zero 1.0 Universal license at
@@ -6399,15 +6524,15 @@
  *    everything should basically work as well as in C11/C++11.
  */
 
 #if !defined(SIMDE_ALIGN_H)
 #define SIMDE_ALIGN_H
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 
 /* I know this seems a little silly, but some non-hosted compilers
  * don't have stddef.h, so we try to accomodate them. */
 #if !defined(SIMDE_ALIGN_SIZE_T_)
   #if defined(__SIZE_TYPE__)
     #define SIMDE_ALIGN_SIZE_T_ __SIZE_TYPE__
   #elif defined(__SIZE_T_TYPE__)
@@ -6928,19 +7053,32 @@
   !defined(__cplusplus) && ( \
       (defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L)) || \
       HEDLEY_HAS_FEATURE(c_static_assert) || \
       HEDLEY_GCC_VERSION_CHECK(6,0,0) || \
       HEDLEY_INTEL_VERSION_CHECK(13,0,0) || \
       defined(_Static_assert) \
     )
-#  define SIMDE_STATIC_ASSERT(expr, message) _Static_assert(expr, message)
+  /* Sometimes _Static_assert is defined (in cdefs.h) using a symbol which
+   * starts with a double-underscore. This is a system header so we have no
+   * control over it, but since it's a macro it will emit a diagnostic which
+   * prevents compilation with -Werror. */
+  #if HEDLEY_HAS_WARNING("-Wreserved-identifier")
+    #define SIMDE_STATIC_ASSERT(expr, message) (__extension__({ \
+      HEDLEY_DIAGNOSTIC_PUSH \
+      _Pragma("clang diagnostic ignored \"-Wreserved-identifier\"") \
+      _Static_assert(expr, message); \
+      HEDLEY_DIAGNOSTIC_POP \
+    }))
+  #else
+    #define SIMDE_STATIC_ASSERT(expr, message) _Static_assert(expr, message)
+  #endif
 #elif \
   (defined(__cplusplus) && (__cplusplus >= 201103L)) || \
   HEDLEY_MSVC_VERSION_CHECK(16,0,0)
-#  define SIMDE_STATIC_ASSERT(expr, message) HEDLEY_DIAGNOSTIC_DISABLE_CPP98_COMPAT_WRAP_(static_assert(expr, message))
+  #define SIMDE_STATIC_ASSERT(expr, message) HEDLEY_DIAGNOSTIC_DISABLE_CPP98_COMPAT_WRAP_(static_assert(expr, message))
 #endif
 
 /* Statement exprs */
 #if \
     HEDLEY_GNUC_VERSION_CHECK(2,95,0) || \
     HEDLEY_TINYC_VERSION_CHECK(0,9,26) || \
     HEDLEY_INTEL_VERSION_CHECK(9,0,0) || \
@@ -7450,14 +7588,44 @@
         return 0;
       }
     #define simde_memcmp(s1, s2, n) simde_memcmp_(s1, s2, n)
     #endif
   #endif
 #endif
 
+/*** Functions that quiet a signaling NaN ***/
+
+static HEDLEY_INLINE
+double
+simde_math_quiet(double x) {
+  uint64_t tmp, mask;
+  if (!simde_math_isnan(x)) {
+    return x;
+  }
+  simde_memcpy(&tmp, &x, 8);
+  mask = 0x7ff80000;
+  mask <<= 32;
+  tmp |= mask;
+  simde_memcpy(&x, &tmp, 8);
+  return x;
+}
+
+static HEDLEY_INLINE
+float
+simde_math_quietf(float x) {
+  uint32_t tmp;
+  if (!simde_math_isnanf(x)) {
+    return x;
+  }
+  simde_memcpy(&tmp, &x, 4);
+  tmp |= 0x7fc00000lu;
+  simde_memcpy(&x, &tmp, 4);
+  return x;
+}
+
 #if defined(FE_ALL_EXCEPT)
   #define SIMDE_HAVE_FENV_H
 #elif defined(__has_include)
   #if __has_include(<fenv.h>)
     #include <fenv.h>
     #define SIMDE_HAVE_FENV_H
   #endif
@@ -7503,18 +7671,18 @@
 
 SIMDE_DEFINE_CONVERSION_FUNCTION_(simde_float32_as_uint32,      uint32_t, simde_float32)
 SIMDE_DEFINE_CONVERSION_FUNCTION_(simde_uint32_as_float32, simde_float32, uint32_t)
 SIMDE_DEFINE_CONVERSION_FUNCTION_(simde_float64_as_uint64,      uint64_t, simde_float64)
 SIMDE_DEFINE_CONVERSION_FUNCTION_(simde_uint64_as_float64, simde_float64, uint64_t)
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin check.h :: */
 /* Check (assertions)
- * Portable Snippets - https://gitub.com/nemequ/portable-snippets
+ * Portable Snippets - https://github.com/nemequ/portable-snippets
  * Created by Evan Nemerson <evan@nemerson.com>
  *
  *   To the extent possible under law, the authors have waived all
  *   copyright and related or neighboring rights to this code.  For
  *   details, see the Creative Commons Zero 1.0 Universal license at
  *   https://creativecommons.org/publicdomain/zero/1.0/
  *
@@ -7525,17 +7693,17 @@
 #define SIMDE_CHECK_H
 
 #if !defined(SIMDE_NDEBUG) && !defined(SIMDE_DEBUG)
 #  define SIMDE_NDEBUG 1
 #endif
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 #include <stdint.h>
 
 #if !defined(_WIN32)
 #  define SIMDE_SIZE_MODIFIER "z"
 #  define SIMDE_CHAR_MODIFIER "hh"
 #  define SIMDE_SHORT_MODIFIER "h"
 #else
@@ -7568,18 +7736,18 @@
 #  elif defined(__STDC_HOSTED__)
 #    if __STDC_HOSTETD__ == 1
 #      include <stdio.h>
 #    endif
 #  endif
 
 /* AUTOMATICALLY GENERATED FILE, DO NOT MODIFY */
-/* 3f186a0f35bb73f01ffda73fa9bf060a444bb46b */
+/* 8025b06b07f4789aad472563d363f86671d9e372 */
 /* :: Begin debug-trap.h :: */
 /* Debugging assertions and traps
- * Portable Snippets - https://gitub.com/nemequ/portable-snippets
+ * Portable Snippets - https://github.com/nemequ/portable-snippets
  * Created by Evan Nemerson <evan@nemerson.com>
  *
  *   To the extent possible under law, the authors have waived all
  *   copyright and related or neighboring rights to this code.  For
  *   details, see the Creative Commons Zero 1.0 Universal license at
  *   https://creativecommons.org/publicdomain/zero/1.0/
  *
@@ -7933,14 +8101,17 @@
   #define SIMDE_BUILTIN_SUFFIX_64_ l
   #define SIMDE_BUILTIN_TYPE_64_ long
 #elif (INT64_MAX == LLONG_MAX) && (INT64_MIN == LLONG_MIN)
   #define SIMDE_BUILTIN_SUFFIX_64_ ll
   #define SIMDE_BUILTIN_TYPE_64_ long long
 #endif
 
+/* SIMDE_DIAGNOSTIC_DISABLE_CPP98_COMPAT_PEDANTIC_ */
+HEDLEY_DIAGNOSTIC_POP
+
 #if defined(SIMDE_BUILTIN_SUFFIX_8_)
   #define SIMDE_BUILTIN_8_(name) HEDLEY_CONCAT3(__builtin_, name, SIMDE_BUILTIN_SUFFIX_8_)
   #define SIMDE_BUILTIN_HAS_8_(name) HEDLEY_HAS_BUILTIN(HEDLEY_CONCAT3(__builtin_, name, SIMDE_BUILTIN_SUFFIX_8_))
 #else
   #define SIMDE_BUILTIN_HAS_8_(name) 0
 #endif
 #if defined(SIMDE_BUILTIN_SUFFIX_16_)
@@ -8030,15 +8201,15 @@
 #    if (defined(SIMDE_ARCH_X86) && !defined(SIMDE_ARCH_AMD64)) || defined(SIMDE_ARCH_ZARCH)
 #      define SIMDE_BUG_GCC_53784
 #    endif
 #    if defined(SIMDE_ARCH_X86) || defined(SIMDE_ARCH_AMD64)
 #      if HEDLEY_GCC_VERSION_CHECK(4,3,0) /* -Wsign-conversion */
 #        define SIMDE_BUG_GCC_95144
 #      endif
-#      if !HEDLEY_GCC_VERSION_CHECK(11,0,0)
+#      if !HEDLEY_GCC_VERSION_CHECK(11,2,0)
 #        define SIMDE_BUG_GCC_95483
 #      endif
 #      if defined(__OPTIMIZE__)
 #        define SIMDE_BUG_GCC_100927
 #      endif
 #      define SIMDE_BUG_GCC_98521
 #    endif
@@ -8050,39 +8221,46 @@
 #    endif
 #    if defined(SIMDE_ARCH_ARM)
 #      define SIMDE_BUG_GCC_95399
 #      define SIMDE_BUG_GCC_95471
 #    elif defined(SIMDE_ARCH_POWER)
 #      define SIMDE_BUG_GCC_95227
 #      define SIMDE_BUG_GCC_95782
-#      define SIMDE_BUG_VEC_CPSGN_REVERSED_ARGS
+#      if !HEDLEY_GCC_VERSION_CHECK(12,0,0)
+#        define SIMDE_BUG_VEC_CPSGN_REVERSED_ARGS
+#      endif
 #    elif defined(SIMDE_ARCH_X86) || defined(SIMDE_ARCH_AMD64)
 #      if !HEDLEY_GCC_VERSION_CHECK(10,2,0) && !defined(__OPTIMIZE__)
 #        define SIMDE_BUG_GCC_96174
 #      endif
 #    elif defined(SIMDE_ARCH_ZARCH)
 #      define SIMDE_BUG_GCC_95782
 #      if HEDLEY_GCC_VERSION_CHECK(10,0,0)
 #        define SIMDE_BUG_GCC_101614
 #      endif
 #    endif
 #    if defined(SIMDE_ARCH_MIPS_MSA)
 #      define SIMDE_BUG_GCC_97248
-#      define SIMDE_BUG_GCC_100760
-#      define SIMDE_BUG_GCC_100761
-#      define SIMDE_BUG_GCC_100762
+#      if !HEDLEY_GCC_VERSION_CHECK(12,1,0)
+#        define SIMDE_BUG_GCC_100760
+#        define SIMDE_BUG_GCC_100761
+#        define SIMDE_BUG_GCC_100762
+#      endif
 #    endif
 #    define SIMDE_BUG_GCC_95399
+#    if !defined(__OPTIMIZE__)
+#      define SIMDE_BUG_GCC_105339
+#    endif
 #  elif defined(__clang__)
 #    if defined(SIMDE_ARCH_AARCH64)
 #      define SIMDE_BUG_CLANG_45541
-#      define SIMDE_BUG_CLANG_46844
 #      define SIMDE_BUG_CLANG_48257
 #      if !SIMDE_DETECT_CLANG_VERSION_CHECK(12,0,0)
 #        define SIMDE_BUG_CLANG_46840
+#        define SIMDE_BUG_CLANG_46844
 #      endif
 #      if SIMDE_DETECT_CLANG_VERSION_CHECK(10,0,0) && SIMDE_DETECT_CLANG_VERSION_NOT(11,0,0)
 #        define SIMDE_BUG_CLANG_BAD_VI64_OPS
 #      endif
 #      if SIMDE_DETECT_CLANG_VERSION_NOT(9,0,0)
 #        define SIMDE_BUG_CLANG_GIT_4EC445B8
 #        define SIMDE_BUG_CLANG_REV_365298 /* 0464e07c8f6e3310c28eb210a4513bc2243c2a7e */
@@ -8093,16 +8271,18 @@
 #        define SIMDE_BUG_CLANG_BAD_VGET_SET_LANE_TYPES
 #      endif
 #    endif
 #    if defined(SIMDE_ARCH_POWER) && !SIMDE_DETECT_CLANG_VERSION_CHECK(12,0,0)
 #      define SIMDE_BUG_CLANG_46770
 #    endif
 #    if defined(SIMDE_ARCH_POWER) && (SIMDE_ARCH_POWER == 700) && (SIMDE_DETECT_CLANG_VERSION_CHECK(11,0,0))
-#      define SIMDE_BUG_CLANG_50893
-#      define SIMDE_BUG_CLANG_50901
+#      if !SIMDE_DETECT_CLANG_VERSION_CHECK(13,0,0)
+#        define SIMDE_BUG_CLANG_50893
+#        define SIMDE_BUG_CLANG_50901
+#      endif
 #    endif
 #    if defined(_ARCH_PWR9) && !SIMDE_DETECT_CLANG_VERSION_CHECK(12,0,0) && !defined(__OPTIMIZE__)
 #      define SIMDE_BUG_CLANG_POWER9_16x4_BAD_SHIFT
 #    endif
 #    if defined(SIMDE_ARCH_POWER)
 #      define SIMDE_BUG_CLANG_50932
 #      if !SIMDE_DETECT_CLANG_VERSION_CHECK(12,0,0)
@@ -8130,14 +8310,17 @@
 #      endif
 #      if HEDLEY_HAS_WARNING("-Wvector-conversion") && SIMDE_DETECT_CLANG_VERSION_NOT(11,0,0)
 #        define SIMDE_BUG_CLANG_44589
 #      endif
 #      define SIMDE_BUG_CLANG_48673
 #    endif
 #    define SIMDE_BUG_CLANG_45959
+#    if defined(SIMDE_ARCH_WASM_SIMD128)
+#      define SIMDE_BUG_CLANG_60655
+#    endif
 #  elif defined(HEDLEY_MSVC_VERSION)
 #    if defined(SIMDE_ARCH_X86)
 #      define SIMDE_BUG_MSVC_ROUND_EXTRACT
 #    endif
 #  elif defined(HEDLEY_INTEL_VERSION)
 #    define SIMDE_BUG_INTEL_857088
 #  elif defined(HEDLEY_MCST_LCC_VERSION)
@@ -8157,18 +8340,17 @@
  * This is just an easy way to work around it.
  */
 #if \
     (HEDLEY_HAS_WARNING("-Wsign-conversion") && SIMDE_DETECT_CLANG_VERSION_NOT(11,0,0)) || \
     HEDLEY_GCC_VERSION_CHECK(4,3,0)
 #  define SIMDE_BUG_IGNORE_SIGN_CONVERSION(expr) (__extension__ ({ \
        HEDLEY_DIAGNOSTIC_PUSH  \
-       HEDLEY_DIAGNOSTIC_POP  \
        _Pragma("GCC diagnostic ignored \"-Wsign-conversion\"") \
        __typeof__(expr) simde_bug_ignore_sign_conversion_v_= (expr); \
-       HEDLEY_DIAGNOSTIC_PUSH  \
+       HEDLEY_DIAGNOSTIC_POP  \
        simde_bug_ignore_sign_conversion_v_; \
      }))
 #else
 #  define SIMDE_BUG_IGNORE_SIGN_CONVERSION(expr) (expr)
 #endif
 
 /* Usually the shift count is signed (for example, NEON or SSE).
@@ -9621,26 +9803,25 @@
 simde_mm_slli_pi16 (simde__m64 a, int count) {
   #if defined(SIMDE_X86_MMX_NATIVE) && !defined(__PGI)
     return _mm_slli_pi16(a, count);
   #else
     simde__m64_private r_;
     simde__m64_private a_ = simde__m64_to_private(a);
 
-    #if defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR) && defined(SIMDE_BUG_CLANG_POWER9_16x4_BAD_SHIFT)
+    #if defined(SIMDE_MIPS_LOONGSON_MMI_NATIVE)
+      r_.mmi_i16 = psllh_s(a_.mmi_i16, count);
+    #elif defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR) && defined(SIMDE_BUG_CLANG_POWER9_16x4_BAD_SHIFT)
       if (HEDLEY_UNLIKELY(count > 15))
         return simde_mm_setzero_si64();
 
       r_.i16 = a_.i16 << HEDLEY_STATIC_CAST(int16_t, count);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
       r_.i16 = a_.i16 << count;
-    #elif defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i16 = vshl_s16(a_.neon_i16, vmov_n_s16((int16_t) count));
-    #elif defined(SIMDE_MIPS_LOONGSON_MMI_NATIVE)
-      r_.mmi_i16 = psllh_s(a_.mmi_i16, b_.mmi_i16);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.u16) / sizeof(r_.u16[0])) ; i++) {
         r_.u16[i] = HEDLEY_STATIC_CAST(uint16_t, a_.u16[i] << count);
       }
     #endif
 
@@ -10311,18 +10492,18 @@
   #else
     simde__m64_private r_;
     simde__m64_private a_ = simde__m64_to_private(a);
     simde__m64_private b_ = simde__m64_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_i8 = vzip2_s8(a_.neon_i8, b_.neon_i8);
-    #elif defined(SIMDE_SHUFFLE_VECTOR_)
-      r_.i8 = SIMDE_SHUFFLE_VECTOR_(8, 8, a_.i8, b_.i8, 4, 12, 5, 13, 6, 14, 7, 15);
     #elif defined(SIMDE_MIPS_LOONGSON_MMI_NATIVE)
       r_.mmi_i8 = punpckhbh_s(a_.mmi_i8, b_.mmi_i8);
+    #elif defined(SIMDE_SHUFFLE_VECTOR_)
+      r_.i8 = SIMDE_SHUFFLE_VECTOR_(8, 8, a_.i8, b_.i8, 4, 12, 5, 13, 6, 14, 7, 15);
     #else
       r_.i8[0] = a_.i8[4];
       r_.i8[1] = b_.i8[4];
       r_.i8[2] = a_.i8[5];
       r_.i8[3] = b_.i8[5];
       r_.i8[4] = a_.i8[6];
       r_.i8[5] = b_.i8[6];
@@ -10553,15 +10734,16 @@
 SIMDE_END_DECLS_
 
 HEDLEY_DIAGNOSTIC_POP
 
 #endif /* !defined(SIMDE_X86_MMX_H) */
 /* :: End x86/mmx.h :: */
 
-#if defined(_WIN32)
+#if defined(_WIN32) && !defined(SIMDE_X86_SSE_NATIVE) && defined(_MSC_VER)
+  #define NOMINMAX
   #include <windows.h>
 #endif
 
 #if defined(__ARM_ACLE)
   #include <arm_acle.h>
 #endif
 
@@ -10642,25 +10824,38 @@
     SIMDE_ALIGN_TO_16 SIMDE_POWER_ALTIVEC_VECTOR(signed int)         altivec_i32;
     SIMDE_ALIGN_TO_16 SIMDE_POWER_ALTIVEC_VECTOR(float)              altivec_f32;
     #if defined(SIMDE_POWER_ALTIVEC_P7_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE)
       SIMDE_ALIGN_TO_16 SIMDE_POWER_ALTIVEC_VECTOR(unsigned long long) altivec_u64;
       SIMDE_ALIGN_TO_16 SIMDE_POWER_ALTIVEC_VECTOR(signed long long)   altivec_i64;
       SIMDE_ALIGN_TO_16 SIMDE_POWER_ALTIVEC_VECTOR(double)             altivec_f64;
     #endif
+  #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+    v16i8 lsx_i8;
+    v8i16 lsx_i16;
+    v4i32 lsx_i32;
+    v2i64 lsx_i64;
+    v16u8 lsx_u8;
+    v8u16 lsx_u16;
+    v4u32 lsx_u32;
+    v2u64 lsx_u64;
+    v4f32 lsx_f32;
+    v2f64 lsx_f64;
   #endif
 } simde__m128_private;
 
 #if defined(SIMDE_X86_SSE_NATIVE)
   typedef __m128 simde__m128;
 #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
    typedef float32x4_t simde__m128;
 #elif defined(SIMDE_WASM_SIMD128_NATIVE)
    typedef v128_t simde__m128;
 #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE)
    typedef SIMDE_POWER_ALTIVEC_VECTOR(float) simde__m128;
+#elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+  typedef v4f32 simde__m128;
 #elif defined(SIMDE_VECTOR_SUBSCRIPT)
   typedef simde_float32 simde__m128 SIMDE_ALIGN_TO_16 SIMDE_VECTOR(16) SIMDE_MAY_ALIAS;
 #else
   typedef simde__m128_private simde__m128;
 #endif
 
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
@@ -10736,14 +10931,27 @@
     SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, SIMDE_POWER_ALTIVEC_VECTOR(signed long long), altivec, i64)
     SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, SIMDE_POWER_ALTIVEC_VECTOR(unsigned long long), altivec, u64)
   #endif
 #elif defined(SIMDE_WASM_SIMD128_NATIVE)
   SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v128_t, wasm, v128);
 #endif /* defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) */
 
+#if defined(SIMDE_LOONGARCH_LSX_NATIVE)
+  SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v16i8, lsx, i8)
+  SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v8i16, lsx, i16)
+  SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v4i32, lsx, i32)
+  SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v2i64, lsx, i64)
+  SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v16u8, lsx, u8)
+  SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v8u16, lsx, u16)
+  SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v4u32, lsx, u32)
+  SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v2u64, lsx, u64)
+  SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v4f32, lsx, f32)
+  SIMDE_X86_GENERATE_CONVERSION_FUNCTION(m128, v2f64, lsx, f64)
+#endif /* defined(SIMDE_LOONGARCH_LSX_NATIVE) */
+
 enum {
   #if defined(SIMDE_X86_SSE_NATIVE)
     SIMDE_MM_ROUND_NEAREST     = _MM_ROUND_NEAREST,
     SIMDE_MM_ROUND_DOWN        = _MM_ROUND_DOWN,
     SIMDE_MM_ROUND_UP          = _MM_ROUND_UP,
     SIMDE_MM_ROUND_TOWARD_ZERO = _MM_ROUND_TOWARD_ZERO
   #else
@@ -11090,14 +11298,16 @@
 
   switch (rounding & ~SIMDE_MM_FROUND_NO_EXC) {
     case SIMDE_MM_FROUND_CUR_DIRECTION:
       #if defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
         r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_round(a_.altivec_f32));
       #elif defined(SIMDE_ARM_NEON_A32V8_NATIVE) && !defined(SIMDE_BUG_GCC_95399)
         r_.neon_f32 = vrndiq_f32(a_.neon_f32);
+      #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+        r_.wasm_v128 = wasm_f32x4_nearest(a_.wasm_v128);
       #elif defined(simde_math_nearbyintf)
         SIMDE_VECTORIZE
         for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
           r_.f32[i] = simde_math_nearbyintf(a_.f32[i]);
         }
       #else
         HEDLEY_UNREACHABLE_RETURN(simde_mm_undefined_pd());
@@ -11105,14 +11315,18 @@
       break;
 
     case SIMDE_MM_FROUND_TO_NEAREST_INT:
       #if defined(SIMDE_POWER_ALTIVEC_P7_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
         r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_rint(a_.altivec_f32));
       #elif defined(SIMDE_ARM_NEON_A32V8_NATIVE)
         r_.neon_f32 = vrndnq_f32(a_.neon_f32);
+      #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+        r_.lsx_i64 = __lsx_vfrintrne_s(a_.lsx_f32);
+      #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+        r_.wasm_v128 = wasm_f32x4_nearest(a_.wasm_v128);
       #elif defined(simde_math_roundevenf)
         SIMDE_VECTORIZE
         for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
           r_.f32[i] = simde_math_roundevenf(a_.f32[i]);
         }
       #else
         HEDLEY_UNREACHABLE_RETURN(simde_mm_undefined_pd());
@@ -11120,14 +11334,18 @@
       break;
 
     case SIMDE_MM_FROUND_TO_NEG_INF:
       #if defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
         r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_floor(a_.altivec_f32));
       #elif defined(SIMDE_ARM_NEON_A32V8_NATIVE)
         r_.neon_f32 = vrndmq_f32(a_.neon_f32);
+      #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+        r_.lsx_i64 = __lsx_vfrintrm_s(a_.lsx_f32);
+      #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+        r_.wasm_v128 = wasm_f32x4_floor(a_.wasm_v128);
       #elif defined(simde_math_floorf)
         SIMDE_VECTORIZE
         for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
           r_.f32[i] = simde_math_floorf(a_.f32[i]);
         }
       #else
         HEDLEY_UNREACHABLE_RETURN(simde_mm_undefined_pd());
@@ -11135,14 +11353,18 @@
       break;
 
     case SIMDE_MM_FROUND_TO_POS_INF:
       #if defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
         r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_ceil(a_.altivec_f32));
       #elif defined(SIMDE_ARM_NEON_A32V8_NATIVE)
         r_.neon_f32 = vrndpq_f32(a_.neon_f32);
+      #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+        r_.lsx_i64 = __lsx_vfrintrp_s(a_.lsx_f32);
+      #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+        r_.wasm_v128 = wasm_f32x4_ceil(a_.wasm_v128);
       #elif defined(simde_math_ceilf)
         SIMDE_VECTORIZE
         for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
           r_.f32[i] = simde_math_ceilf(a_.f32[i]);
         }
       #else
         HEDLEY_UNREACHABLE_RETURN(simde_mm_undefined_pd());
@@ -11150,14 +11372,18 @@
       break;
 
     case SIMDE_MM_FROUND_TO_ZERO:
       #if defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
         r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_trunc(a_.altivec_f32));
       #elif defined(SIMDE_ARM_NEON_A32V8_NATIVE)
         r_.neon_f32 = vrndq_f32(a_.neon_f32);
+      #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+        r_.lsx_i64 = __lsx_vfrintrz_s(a_.lsx_f32);
+      #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+        r_.wasm_v128 = wasm_f32x4_trunc(a_.wasm_v128);
       #elif defined(simde_math_truncf)
         SIMDE_VECTORIZE
         for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
           r_.f32[i] = simde_math_truncf(a_.f32[i]);
         }
       #else
         HEDLEY_UNREACHABLE_RETURN(simde_mm_undefined_pd());
@@ -11212,14 +11438,18 @@
   #if defined(SIMDE_X86_SSE_NATIVE)
     return _mm_set_ps1(a);
   #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
     return vdupq_n_f32(a);
   #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
     (void) a;
     return vec_splats(a);
+  #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+    return (simde__m128)__lsx_vldrepl_w(&a, 0);
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    return wasm_f32x4_splat(a);
   #else
     return simde_mm_set_ps(a, a, a, a);
   #endif
 }
 #define simde_mm_set1_ps(a) simde_mm_set_ps1(a)
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_set_ps1(a) simde_mm_set_ps1(a)
@@ -11233,23 +11463,25 @@
     return _mm_move_ss(a, b);
   #else
     simde__m128_private
       r_,
       a_ = simde__m128_to_private(a),
       b_ = simde__m128_to_private(b);
 
-    #if defined(SIMDE_SHUFFLE_VECTOR_)
-      r_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, b_.f32, 4, 1, 2, 3);
-    #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
+    #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_f32 = vsetq_lane_f32(vgetq_lane_f32(b_.neon_f32, 0), a_.neon_f32, 0);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       static const SIMDE_POWER_ALTIVEC_VECTOR(unsigned int) m = { ~0U, 0U, 0U, 0U };
       r_.altivec_f32 = vec_sel(a_.altivec_f32, b_.altivec_f32, m);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_i8x16_shuffle(b_.wasm_v128, a_.wasm_v128, 0, 1, 2, 3, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vextrins_w(a_.lsx_i64, b_.lsx_i64, 0);
+    #elif defined(SIMDE_SHUFFLE_VECTOR_)
+      r_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, b_.f32, 4, 1, 2, 3);
     #else
       r_.f32[0] = b_.f32[0];
       r_.f32[1] = a_.f32[1];
       r_.f32[2] = a_.f32[2];
       r_.f32[3] = a_.f32[3];
     #endif
 
@@ -11274,14 +11506,18 @@
       r_,
       a_ = simde__m128_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f32 = vdupq_laneq_f32(a_.neon_f32, 0);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = vec_splat(a_.altivec_f32, 0);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vreplvei_w(a_.lsx_i64, 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_f32x4_splat(a_.f32[0]);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, a_.f32, 0, 0, 0, 0);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.f32[i] = a_.f32[0];
       }
@@ -11304,14 +11540,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_f32 = vaddq_f32(a_.neon_f32, b_.neon_f32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_add(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = vec_add(a_.altivec_f32, b_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_f32 = __lsx_vfadd_s(a_.lsx_f32, b_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.f32 = a_.f32 + b_.f32;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.f32[i] = a_.f32[i] + b_.f32[i];
       }
@@ -11369,14 +11607,16 @@
       a_ = simde__m128_to_private(a),
       b_ = simde__m128_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vandq_s32(a_.neon_i32, b_.neon_i32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_v128_and(a_.wasm_v128, b_.wasm_v128);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vand_v(a_.lsx_i64, b_.lsx_i64);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = a_.i32 & b_.i32;
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = vec_and(a_.altivec_f32, b_.altivec_f32);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i32) / sizeof(r_.i32[0])) ; i++) {
@@ -11404,14 +11644,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vbicq_s32(b_.neon_i32, a_.neon_i32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_v128_andnot(b_.wasm_v128, a_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
       r_.altivec_f32 = vec_andc(b_.altivec_f32, a_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vandn_v(a_.lsx_i64, b_.lsx_i64);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = ~a_.i32 & b_.i32;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i32) / sizeof(r_.i32[0])) ; i++) {
         r_.i32[i] = ~(a_.i32[i]) & b_.i32[i];
       }
@@ -11437,14 +11679,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = veorq_s32(a_.neon_i32, b_.neon_i32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_v128_xor(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_i32 = vec_xor(a_.altivec_i32, b_.altivec_i32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vxor_v(a_.lsx_i64, b_.lsx_i64);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32f = a_.i32f ^ b_.i32f;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.u32) / sizeof(r_.u32[0])) ; i++) {
         r_.u32[i] = a_.u32[i] ^ b_.u32[i];
       }
@@ -11470,14 +11714,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vorrq_s32(a_.neon_i32, b_.neon_i32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_v128_or(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_i32 = vec_or(a_.altivec_i32, b_.altivec_i32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vor_v(a_.lsx_i64, b_.lsx_i64);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32f = a_.i32f | b_.i32f;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.u32) / sizeof(r_.u32[0])) ; i++) {
         r_.u32[i] = a_.u32[i] | b_.u32[i];
       }
@@ -11508,14 +11754,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vmvnq_s32(a_.neon_i32);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_i32 = vec_nor(a_.altivec_i32, a_.altivec_i32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_v128_not(a_.wasm_v128);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vnor_v(a_.lsx_i64, a_.lsx_i64);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = ~a_.i32;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i32) / sizeof(r_.i32[0])) ; i++) {
         r_.i32[i] = ~(a_.i32[i]);
       }
@@ -11547,14 +11795,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vbslq_s32(mask_.neon_u32, b_.neon_i32, a_.neon_i32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_v128_bitselect(b_.wasm_v128, a_.wasm_v128, mask_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE)
       r_.altivec_i32 = vec_sel(a_.altivec_i32, b_.altivec_i32, mask_.altivec_u32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vbitsel_v(a_.lsx_i64, b_.lsx_i64, mask_.lsx_i64);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = a_.i32 ^ ((a_.i32 ^ b_.i32) & mask_.i32);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i32) / sizeof(r_.i32[0])) ; i++) {
         r_.i32[i] = a_.i32[i] ^ ((a_.i32[i] ^ b_.i32[i]) & mask_.i32[i]);
       }
@@ -11681,14 +11931,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_u32 = vceqq_f32(a_.neon_f32, b_.neon_f32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_eq(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_cmpeq(a_.altivec_f32, b_.altivec_f32));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vfcmp_ceq_s(a_.lsx_f32, b_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = HEDLEY_REINTERPRET_CAST(__typeof__(r_.i32), a_.f32 == b_.f32);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.u32[i] = (a_.f32[i] == b_.f32[i]) ? ~UINT32_C(0) : UINT32_C(0);
       }
@@ -11742,14 +11994,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_u32 = vcgeq_f32(a_.neon_f32, b_.neon_f32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_ge(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_cmpge(a_.altivec_f32, b_.altivec_f32));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vfcmp_cle_s(b_.lsx_f32, a_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = HEDLEY_REINTERPRET_CAST(__typeof__(r_.i32), (a_.f32 >= b_.f32));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.u32[i] = (a_.f32[i] >= b_.f32[i]) ? ~UINT32_C(0) : UINT32_C(0);
       }
@@ -11803,14 +12057,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_u32 = vcgtq_f32(a_.neon_f32, b_.neon_f32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_gt(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_cmpgt(a_.altivec_f32, b_.altivec_f32));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vfcmp_clt_s(b_.lsx_f32, a_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = HEDLEY_REINTERPRET_CAST(__typeof__(r_.i32), (a_.f32 > b_.f32));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.u32[i] = (a_.f32[i] > b_.f32[i]) ? ~UINT32_C(0) : UINT32_C(0);
       }
@@ -11864,14 +12120,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_u32 = vcleq_f32(a_.neon_f32, b_.neon_f32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_le(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_cmple(a_.altivec_f32, b_.altivec_f32));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vfcmp_cle_s(a_.lsx_f32, b_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = HEDLEY_REINTERPRET_CAST(__typeof__(r_.i32), (a_.f32 <= b_.f32));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.u32[i] = (a_.f32[i] <= b_.f32[i]) ? ~UINT32_C(0) : UINT32_C(0);
       }
@@ -11925,14 +12183,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_u32 = vcltq_f32(a_.neon_f32, b_.neon_f32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_lt(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_cmplt(a_.altivec_f32, b_.altivec_f32));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vfcmp_clt_s(a_.lsx_f32, b_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = HEDLEY_REINTERPRET_CAST(__typeof__(r_.i32), (a_.f32 < b_.f32));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.u32[i] = (a_.f32[i] < b_.f32[i]) ? ~UINT32_C(0) : UINT32_C(0);
       }
@@ -11987,14 +12247,16 @@
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_u32 = vmvnq_u32(vceqq_f32(a_.neon_f32, b_.neon_f32));
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_ne(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P7_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_cmpeq(a_.altivec_f32, b_.altivec_f32));
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float), vec_nor(r_.altivec_f32, r_.altivec_f32));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vfcmp_cune_s(a_.lsx_f32, b_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = HEDLEY_REINTERPRET_CAST(__typeof__(r_.i32), (a_.f32 != b_.f32));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.u32[i] = (a_.f32[i] != b_.f32[i]) ? ~UINT32_C(0) : UINT32_C(0);
       }
@@ -12128,14 +12390,17 @@
       uint32x4_t ceqbb = vceqq_f32(b_.neon_f32, b_.neon_f32);
       r_.neon_u32 = vandq_u32(ceqaa, ceqbb);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_v128_and(wasm_f32x4_eq(a_.wasm_v128, a_.wasm_v128), wasm_f32x4_eq(b_.wasm_v128, b_.wasm_v128));
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float),
           vec_and(vec_cmpeq(a_.altivec_f32, a_.altivec_f32), vec_cmpeq(b_.altivec_f32, b_.altivec_f32)));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vfcmp_cun_s(a_.lsx_f32, b_.lsx_f32);
+      r_.lsx_i64 = __lsx_vnor_v(r_.lsx_i64, r_.lsx_i64);
     #elif defined(simde_math_isnanf)
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.u32[i] = (simde_math_isnanf(a_.f32[i]) || simde_math_isnanf(b_.f32[i])) ? UINT32_C(0) : ~UINT32_C(0);
       }
     #else
       HEDLEY_UNREACHABLE();
@@ -12170,14 +12435,16 @@
     #elif defined(SIMDE_POWER_ALTIVEC_P8_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float),
           vec_nand(vec_cmpeq(a_.altivec_f32, a_.altivec_f32), vec_cmpeq(b_.altivec_f32, b_.altivec_f32)));
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float),
           vec_and(vec_cmpeq(a_.altivec_f32, a_.altivec_f32), vec_cmpeq(b_.altivec_f32, b_.altivec_f32)));
       r_.altivec_f32 = vec_nor(r_.altivec_f32, r_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vfcmp_cun_s(a_.lsx_f32, b_.lsx_f32);
     #elif defined(simde_math_isnanf)
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.u32[i] = (simde_math_isnanf(a_.f32[i]) || simde_math_isnanf(b_.f32[i])) ? ~UINT32_C(0) : UINT32_C(0);
       }
     #else
       HEDLEY_UNREACHABLE();
@@ -12234,14 +12501,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_or_b_nan = vmvnq_u32(vandq_u32(a_not_nan, b_not_nan));
       uint32x4_t a_eq_b = vceqq_f32(a_.neon_f32, b_.neon_f32);
       return !!(vgetq_lane_u32(vorrq_u32(a_or_b_nan, a_eq_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      return wasm_f32x4_extract_lane(a_.wasm_v128, 0) == wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #else
       return a_.f32[0] == b_.f32[0];
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_comieq_ss(a, b) simde_mm_comieq_ss((a), (b))
@@ -12259,14 +12528,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_and_b_not_nan = vandq_u32(a_not_nan, b_not_nan);
       uint32x4_t a_ge_b = vcgeq_f32(a_.neon_f32, b_.neon_f32);
       return !!(vgetq_lane_u32(vandq_u32(a_and_b_not_nan, a_ge_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      return wasm_f32x4_extract_lane(a_.wasm_v128, 0) >= wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #else
       return a_.f32[0] >= b_.f32[0];
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_comige_ss(a, b) simde_mm_comige_ss((a), (b))
@@ -12284,14 +12555,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_and_b_not_nan = vandq_u32(a_not_nan, b_not_nan);
       uint32x4_t a_gt_b = vcgtq_f32(a_.neon_f32, b_.neon_f32);
       return !!(vgetq_lane_u32(vandq_u32(a_and_b_not_nan, a_gt_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      return wasm_f32x4_extract_lane(a_.wasm_v128, 0) > wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #else
       return a_.f32[0] > b_.f32[0];
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_comigt_ss(a, b) simde_mm_comigt_ss((a), (b))
@@ -12309,14 +12582,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_or_b_nan = vmvnq_u32(vandq_u32(a_not_nan, b_not_nan));
       uint32x4_t a_le_b = vcleq_f32(a_.neon_f32, b_.neon_f32);
       return !!(vgetq_lane_u32(vorrq_u32(a_or_b_nan, a_le_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      return wasm_f32x4_extract_lane(a_.wasm_v128, 0) <= wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #else
       return a_.f32[0] <= b_.f32[0];
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_comile_ss(a, b) simde_mm_comile_ss((a), (b))
@@ -12334,14 +12609,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_or_b_nan = vmvnq_u32(vandq_u32(a_not_nan, b_not_nan));
       uint32x4_t a_lt_b = vcltq_f32(a_.neon_f32, b_.neon_f32);
       return !!(vgetq_lane_u32(vorrq_u32(a_or_b_nan, a_lt_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      return wasm_f32x4_extract_lane(a_.wasm_v128, 0) < wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #else
       return a_.f32[0] < b_.f32[0];
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_comilt_ss(a, b) simde_mm_comilt_ss((a), (b))
@@ -12359,14 +12636,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_and_b_not_nan = vandq_u32(a_not_nan, b_not_nan);
       uint32x4_t a_neq_b = vmvnq_u32(vceqq_f32(a_.neon_f32, b_.neon_f32));
       return !!(vgetq_lane_u32(vandq_u32(a_and_b_not_nan, a_neq_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      return wasm_f32x4_extract_lane(a_.wasm_v128, 0) != wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #else
       return a_.f32[0] != b_.f32[0];
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_comineq_ss(a, b) simde_mm_comineq_ss((a), (b))
@@ -12391,14 +12670,17 @@
       r_.altivec_f32 = vec_cpsgn(dest_.altivec_f32, src_.altivec_f32);
     #else
       r_.altivec_f32 = vec_cpsgn(src_.altivec_f32, dest_.altivec_f32);
     #endif
   #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
     const SIMDE_POWER_ALTIVEC_VECTOR(unsigned int) sign_pos = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(unsigned int), vec_splats(-0.0f));
     r_.altivec_f32 = vec_sel(dest_.altivec_f32, src_.altivec_f32, sign_pos);
+  #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+    const v4f32 sign_pos = {-0.0f, -0.0f, -0.0f, -0.0f};
+    r_.lsx_i64 = __lsx_vbitsel_v(dest_.lsx_i64, src_.lsx_i64, (v2i64)sign_pos);
   #elif defined(SIMDE_IEEE754_STORAGE)
     (void) src_;
     (void) dest_;
     simde__m128 sign_pos = simde_mm_set1_ps(-0.0f);
     r_ = simde__m128_to_private(simde_mm_xor_ps(dest, simde_mm_and_ps(simde_mm_xor_ps(dest, src), sign_pos)));
   #else
     SIMDE_VECTORIZE
@@ -13013,14 +13295,16 @@
       float32x4_t recip0 = vrecpeq_f32(b_.neon_f32);
       float32x4_t recip1 = vmulq_f32(recip0, vrecpsq_f32(recip0, b_.neon_f32));
       r_.neon_f32 = vmulq_f32(a_.neon_f32, recip1);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 =  wasm_f32x4_div(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P7_NATIVE)
       r_.altivec_f32 = vec_div(a_.altivec_f32, b_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LASX_NATIVE)
+      r_.lsx_f32 = __lsx_vfdiv_s(a_.lsx_f32, b_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.f32 = a_.f32 / b_.f32;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.f32[i] = a_.f32[i] / b_.f32[i];
       }
@@ -13117,14 +13401,18 @@
 
   #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
     r_.neon_f32 = vld1q_f32(mem_addr);
   #elif defined(SIMDE_POWER_ALTIVEC_P7_NATIVE)
     r_.altivec_f32 = vec_vsx_ld(0, mem_addr);
   #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
     r_.altivec_f32 = vec_ld(0, mem_addr);
+  #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+    r_.lsx_i64 = __lsx_vld(mem_addr, 0);
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    r_.wasm_v128 = wasm_v128_load(mem_addr);
   #else
     simde_memcpy(&r_, SIMDE_ALIGN_ASSUME_LIKE(mem_addr, simde__m128), sizeof(r_));
   #endif
 
   return simde__m128_from_private(r_);
 #endif
 }
@@ -13138,14 +13426,18 @@
   #if defined(SIMDE_X86_SSE_NATIVE)
     return _mm_load_ps1(mem_addr);
   #else
     simde__m128_private r_;
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_f32 = vld1q_dup_f32(mem_addr);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vldrepl_w(mem_addr, 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_load32_splat(mem_addr);
     #else
       r_ = simde__m128_to_private(simde_mm_set1_ps(*mem_addr));
     #endif
 
     return simde__m128_from_private(r_);
   #endif
 }
@@ -13161,14 +13453,16 @@
   #if defined(SIMDE_X86_SSE_NATIVE)
     return _mm_load_ss(mem_addr);
   #else
     simde__m128_private r_;
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_f32 = vsetq_lane_f32(*mem_addr, vdupq_n_f32(0), 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_load32_zero(mem_addr);
     #else
       r_.f32[0] = *mem_addr;
       r_.i32[1] = 0;
       r_.i32[2] = 0;
       r_.i32[3] = 0;
     #endif
 
@@ -13178,23 +13472,25 @@
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_load_ss(mem_addr) simde_mm_load_ss(mem_addr)
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128
 simde_mm_loadh_pi (simde__m128 a, simde__m64 const* mem_addr) {
-  #if defined(SIMDE_X86_SSE_NATIVE) && defined(SIMDE_X86_MMX_NATIVE)
+  #if defined(SIMDE_X86_SSE_NATIVE)
     return _mm_loadh_pi(a, HEDLEY_REINTERPRET_CAST(__m64 const*, mem_addr));
   #else
     simde__m128_private
       r_,
       a_ = simde__m128_to_private(a);
 
   #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
     r_.neon_f32 = vcombine_f32(vget_low_f32(a_.neon_f32), vld1_f32(HEDLEY_REINTERPRET_CAST(const float32_t*, mem_addr)));
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    r_.wasm_v128 = wasm_v128_load64_lane(mem_addr, a_.wasm_v128, 1);
   #else
     simde__m64_private b_ = *HEDLEY_REINTERPRET_CAST(simde__m64_private const*, mem_addr);
     r_.f32[0] = a_.f32[0];
     r_.f32[1] = a_.f32[1];
     r_.f32[2] = b_.f32[0];
     r_.f32[3] = b_.f32[1];
   #endif
@@ -13228,14 +13524,16 @@
     simde__m128_private
       r_,
       a_ = simde__m128_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_f32 = vcombine_f32(vld1_f32(
         HEDLEY_REINTERPRET_CAST(const float32_t*, mem_addr)), vget_high_f32(a_.neon_f32));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_load64_lane(mem_addr, a_.wasm_v128, 0);
     #else
       simde__m64_private b_;
       simde_memcpy(&b_, mem_addr, sizeof(b_));
       r_.i32[0] = b_.i32[0];
       r_.i32[1] = b_.i32[1];
       r_.i32[2] = a_.i32[2];
       r_.i32[3] = a_.i32[3];
@@ -13263,14 +13561,16 @@
       v_ = simde__m128_to_private(simde_mm_load_ps(mem_addr));
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_f32 = vrev64q_f32(v_.neon_f32);
       r_.neon_f32 = vextq_f32(r_.neon_f32, r_.neon_f32, 2);
     #elif defined(SIMDE_POWER_ALTIVEC_P7_NATIVE) && defined(__PPC64__)
       r_.altivec_f32 = vec_reve(v_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vshuf4i_w(v_.lsx_i64, 0x1b);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, v_.f32, v_.f32, 3, 2, 1, 0);
     #else
       r_.f32[0] = v_.f32[3];
       r_.f32[1] = v_.f32[2];
       r_.f32[2] = v_.f32[1];
       r_.f32[3] = v_.f32[0];
@@ -13293,14 +13593,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_f32 = vld1q_f32(HEDLEY_REINTERPRET_CAST(const float32_t*, mem_addr));
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_v128_load(mem_addr);
     #elif defined(SIMDE_POWER_ALTIVEC_P7_NATIVE) && defined(__PPC64__)
       r_.altivec_f32 = vec_vsx_ld(0, mem_addr);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vld(mem_addr, 0);
     #else
       simde_memcpy(&r_, mem_addr, sizeof(r_));
     #endif
 
     return simde__m128_from_private(r_);
   #endif
 }
@@ -13378,14 +13680,16 @@
       r_.wasm_v128 = wasm_f32x4_max(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_v128_bitselect(a_.wasm_v128, b_.wasm_v128, wasm_f32x4_gt(a_.wasm_v128, b_.wasm_v128));
     #elif (defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)) && defined(SIMDE_FAST_NANS)
       r_.altivec_f32 = vec_max(a_.altivec_f32, b_.altivec_f32);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
       r_.altivec_f32 = vec_sel(b_.altivec_f32, a_.altivec_f32, vec_cmpgt(a_.altivec_f32, b_.altivec_f32));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE) && defined(SIMDE_FAST_NANS)
+      r_.lsx_f32 = __lsx_vfmax_s(a_.lsx_f32, b_.lsx_f32);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.f32[i] = (a_.f32[i] > b_.f32[i]) ? a_.f32[i] : b_.f32[i];
       }
     #endif
 
@@ -13503,14 +13807,16 @@
       r_.wasm_v128 = wasm_f32x4_pmin(b_.wasm_v128, a_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
       #if defined(SIMDE_FAST_NANS)
         r_.altivec_f32 = vec_min(a_.altivec_f32, b_.altivec_f32);
       #else
         r_.altivec_f32 = vec_sel(b_.altivec_f32, a_.altivec_f32, vec_cmpgt(b_.altivec_f32, a_.altivec_f32));
       #endif
+    #elif defined(SIMDE_FAST_NANS) && defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_f32 = __lsx_vfmin_s(a_.lsx_f32, b_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       uint32_t SIMDE_VECTOR(16) m = HEDLEY_REINTERPRET_CAST(__typeof__(m), a_.f32 < b_.f32);
       r_.f32 =
         HEDLEY_REINTERPRET_CAST(
           __typeof__(r_.f32),
           ( (HEDLEY_REINTERPRET_CAST(__typeof__(m), a_.f32) &  m) |
             (HEDLEY_REINTERPRET_CAST(__typeof__(m), b_.f32) & ~m)
@@ -13605,14 +13911,16 @@
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       float32x2_t a32 = vget_high_f32(a_.neon_f32);
       float32x2_t b32 = vget_high_f32(b_.neon_f32);
       r_.neon_f32 = vcombine_f32(b32, a32);
     #elif defined(SIMDE_POWER_ALTIVEC_P8_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float),
           vec_mergel(b_.altivec_i64, a_.altivec_i64));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vilvh_d(a_.lsx_i64, b_.lsx_i64);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, b_.f32, 6, 7, 2, 3);
     #else
       r_.f32[0] = b_.f32[2];
       r_.f32[1] = b_.f32[3];
       r_.f32[2] = a_.f32[2];
       r_.f32[3] = a_.f32[3];
@@ -13632,23 +13940,25 @@
     return _mm_movelh_ps(a, b);
   #else
     simde__m128_private
       r_,
       a_ = simde__m128_to_private(a),
       b_ = simde__m128_to_private(b);
 
-    #if defined(SIMDE_SHUFFLE_VECTOR_)
-      r_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, b_.f32, 0, 1, 4, 5);
-    #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
+    #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       float32x2_t a10 = vget_low_f32(a_.neon_f32);
       float32x2_t b10 = vget_low_f32(b_.neon_f32);
       r_.neon_f32 = vcombine_f32(a10, b10);
     #elif defined(SIMDE_POWER_ALTIVEC_P8_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE)
       r_.altivec_f32 = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(float),
           vec_mergeh(a_.altivec_i64, b_.altivec_i64));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vilvl_d(b_.lsx_i64, a_.lsx_i64);
+    #elif defined(SIMDE_SHUFFLE_VECTOR_)
+      r_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, b_.f32, 0, 1, 4, 5);
     #else
       r_.f32[0] = a_.f32[0];
       r_.f32[1] = a_.f32[1];
       r_.f32[2] = b_.f32[0];
       r_.f32[3] = b_.f32[1];
     #endif
 
@@ -13692,15 +14002,15 @@
 #  define _mm_movemask_pi8(a) simde_mm_movemask_pi8(a)
 #  define _m_pmovmskb(a) simde_mm_movemask_pi8(a)
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 int
 simde_mm_movemask_ps (simde__m128 a) {
-  #if defined(SIMDE_X86_SSE_NATIVE) && defined(SIMDE_X86_MMX_NATIVE)
+  #if defined(SIMDE_X86_SSE_NATIVE)
     return _mm_movemask_ps(a);
   #else
     int r = 0;
     simde__m128_private a_ = simde__m128_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       // Shift out everything but the sign bits with a 32-bit unsigned shift right.
@@ -13728,14 +14038,19 @@
       SIMDE_POWER_ALTIVEC_VECTOR(unsigned char) idx = { 96, 64, 32, 0, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128 };
       SIMDE_POWER_ALTIVEC_VECTOR(unsigned char) res = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(unsigned char), vec_bperm(HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(unsigned __int128), a_.altivec_u64), idx));
       return HEDLEY_STATIC_CAST(int32_t, vec_extract(HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(signed int), res), 2));
     #elif defined(SIMDE_POWER_ALTIVEC_P8_NATIVE)
       SIMDE_POWER_ALTIVEC_VECTOR(unsigned char) idx = { 96, 64, 32, 0, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128 };
       SIMDE_POWER_ALTIVEC_VECTOR(unsigned char) res = vec_bperm(a_.altivec_u8, idx);
       return HEDLEY_STATIC_CAST(int32_t, vec_extract(HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(signed int), res), 2));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      v2i64 t64 = __lsx_vmskltz_w(a_.lsx_i64);
+      r = __lsx_vpickve2gr_wu(t64, 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      return HEDLEY_STATIC_CAST(int32_t, wasm_i32x4_bitmask(a_.wasm_v128));
     #else
       SIMDE_VECTORIZE_REDUCTION(|:r)
       for (size_t i = 0 ; i < sizeof(a_.u32) / sizeof(a_.u32[0]) ; i++) {
         r |= (a_.u32[i] >> ((sizeof(a_.u32[i]) * CHAR_BIT) - 1)) << i;
       }
     #endif
 
@@ -13761,14 +14076,16 @@
       r_.neon_f32 = vmulq_f32(a_.neon_f32, b_.neon_f32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_mul(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.f32 = a_.f32 * b_.f32;
     #elif defined(SIMDE_POWER_ALTIVEC_P7_NATIVE)
       r_.altivec_f32 = vec_mul(a_.altivec_f32, b_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_f32 = __lsx_vfmul_s(a_.lsx_f32, b_.lsx_f32);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.f32[i] = a_.f32[i] * b_.f32[i];
       }
     #endif
 
@@ -13992,14 +14309,17 @@
       case SIMDE_MM_HINT_ET1:
         __prefetch_by_load(p, 1, 2);
         break;
       case SIMDE_MM_HINT_ET2:
         __prefetch_by_load(p, 0, 1);
         break;
     }
+  #elif HEDLEY_MSVC_VERSION
+    (void) i;
+    (void) p;
   #endif
 }
 #if defined(SIMDE_X86_SSE_NATIVE)
   #if defined(__clang__) && !SIMDE_DETECT_CLANG_VERSION_CHECK(10,0,0) /* https://reviews.llvm.org/D71718 */
     #define simde_mm_prefetch(p, i) \
       (__extension__({ \
         HEDLEY_DIAGNOSTIC_PUSH \
@@ -14027,14 +14347,17 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_f32 = vnegq_f32(a_.neon_f32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_neg(a_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P8_NATIVE)
       r_.altivec_f32 = vec_neg(a_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      const v4f32 f32 = {0.0f, 0.0f, 0.0f, 0.0f};
+      r_.lsx_f32 = __lsx_vfsub_s(f32, a_.lsx_f32);
     #elif defined(SIMDE_VECTOR_NEGATE)
       r_.f32 = -a_.f32;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.f32[i] = -a_.f32[i];
       }
@@ -14064,14 +14387,16 @@
       #endif
 
       r_.neon_f32 = recip;
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_div(simde_mm_set1_ps(1.0f), a_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = vec_re(a_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_f32 = __lsx_vfrecip_s(a_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_SCALAR)
       r_.f32 = 1.0f / a_.f32;
     #elif defined(SIMDE_IEEE754_STORAGE)
       /* https://stackoverflow.com/questions/12227126/division-as-multiply-and-lut-fast-float-division-reciprocal/12228234#12228234 */
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         int32_t ix;
@@ -14132,14 +14457,18 @@
       r_,
       a_ = simde__m128_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_f32 = vrsqrteq_f32(a_.neon_f32);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = vec_rsqrte(a_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_f32 = __lsx_vfrsqrt_s(a_.lsx_f32);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_f32x4_div(simde_mm_set1_ps(1.0f), wasm_f32x4_sqrt(a_.wasm_v128));
     #elif defined(SIMDE_IEEE754_STORAGE)
       /* https://basesandframes.files.wordpress.com/2020/04/even_faster_math_functions_green_2020.pdf
         Pages 100 - 103 */
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         #if SIMDE_ACCURACY_PREFERENCE <= 0
           r_.i32[i] = INT32_C(0x5F37624F) - (a_.i32[i] >> 1);
@@ -14314,14 +14643,16 @@
 simde_mm_setzero_ps (void) {
   #if defined(SIMDE_X86_SSE_NATIVE)
     return _mm_setzero_ps();
   #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
     return vdupq_n_f32(SIMDE_FLOAT32_C(0.0));
   #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
     return vec_splats(SIMDE_FLOAT32_C(0.0));
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    return wasm_f32x4_const(0.f, 0.f, 0.f, 0.f);
   #else
     simde__m128 r;
     simde_memset(&r, 0, sizeof(r));
     return r;
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
@@ -14395,19 +14726,19 @@
 #  define _MM_SHUFFLE(z, y, x, w) SIMDE_MM_SHUFFLE(z, y, x, w)
 #endif
 
 #if defined(SIMDE_X86_SSE_NATIVE) && defined(SIMDE_X86_MMX_NATIVE) && !defined(__PGI)
 #  define simde_mm_shuffle_pi16(a, imm8) _mm_shuffle_pi16(a, imm8)
 #elif defined(SIMDE_SHUFFLE_VECTOR_)
 #  define simde_mm_shuffle_pi16(a, imm8) (__extension__ ({ \
-      const simde__m64_private simde__tmp_a_ = simde__m64_to_private(a); \
+      const simde__m64_private simde_tmp_a_ = simde__m64_to_private(a); \
       simde__m64_from_private((simde__m64_private) { .i16 = \
         SIMDE_SHUFFLE_VECTOR_(16, 8, \
-          (simde__tmp_a_).i16, \
-          (simde__tmp_a_).i16, \
+          (simde_tmp_a_).i16, \
+          (simde_tmp_a_).i16, \
           (((imm8)     ) & 3), \
           (((imm8) >> 2) & 3), \
           (((imm8) >> 4) & 3), \
           (((imm8) >> 6) & 3)) }); }))
 #else
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m64
@@ -14452,36 +14783,46 @@
   r_.f32[2] = b_.f32[(imm8 >> 4) & 3];
   r_.f32[3] = b_.f32[(imm8 >> 6) & 3];
 
   return simde__m128_from_private(r_);
 }
 #if defined(SIMDE_X86_SSE_NATIVE) && !defined(__PGI)
 #  define simde_mm_shuffle_ps(a, b, imm8) _mm_shuffle_ps(a, b, imm8)
-#elif defined(SIMDE_SHUFFLE_VECTOR_)
+#elif defined(SIMDE_WASM_SIMD128_NATIVE)
   #define simde_mm_shuffle_ps(a, b, imm8) (__extension__ ({ \
-      simde__m128_from_private((simde__m128_private) { .f32 = \
-        SIMDE_SHUFFLE_VECTOR_(32, 16, \
-          simde__m128_to_private(a).f32, \
-          simde__m128_to_private(b).f32, \
-          (((imm8)     ) & 3), \
-          (((imm8) >> 2) & 3), \
-          (((imm8) >> 4) & 3) + 4, \
-          (((imm8) >> 6) & 3) + 4) }); }))
+    simde__m128_from_private((simde__m128_private) { .wasm_v128 = \
+      wasm_i32x4_shuffle( \
+        simde__m128_to_private(a).wasm_v128, \
+        simde__m128_to_private(b).wasm_v128, \
+        (((imm8)     ) & 3), \
+        (((imm8) >> 2) & 3), \
+        (((imm8) >> 4) & 3) + 4, \
+        (((imm8) >> 6) & 3) + 4) }); }))
 #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE) && defined(SIMDE_STATEMENT_EXPR_)
   #define simde_mm_shuffle_ps(a, b, imm8) \
     (__extension__({ \
-      float32x4_t simde_mm_shuffle_ps_a_ = simde__m128i_to_neon_f32(a); \
-      float32x4_t simde_mm_shuffle_ps_b_ = simde__m128i_to_neon_f32(b); \
+      float32x4_t simde_mm_shuffle_ps_a_ = simde__m128_to_neon_f32(a); \
+      float32x4_t simde_mm_shuffle_ps_b_ = simde__m128_to_neon_f32(b); \
       float32x4_t simde_mm_shuffle_ps_r_; \
       \
       simde_mm_shuffle_ps_r_ = vmovq_n_f32(vgetq_lane_f32(simde_mm_shuffle_ps_a_, (imm8) & (0x3))); \
       simde_mm_shuffle_ps_r_ = vsetq_lane_f32(vgetq_lane_f32(simde_mm_shuffle_ps_a_, ((imm8) >> 2) & 0x3), simde_mm_shuffle_ps_r_, 1); \
       simde_mm_shuffle_ps_r_ = vsetq_lane_f32(vgetq_lane_f32(simde_mm_shuffle_ps_b_, ((imm8) >> 4) & 0x3), simde_mm_shuffle_ps_r_, 2); \
                                vsetq_lane_f32(vgetq_lane_f32(simde_mm_shuffle_ps_b_, ((imm8) >> 6) & 0x3), simde_mm_shuffle_ps_r_, 3); \
     }))
+#elif defined(SIMDE_SHUFFLE_VECTOR_)
+  #define simde_mm_shuffle_ps(a, b, imm8) (__extension__ ({ \
+      simde__m128_from_private((simde__m128_private) { .f32 = \
+        SIMDE_SHUFFLE_VECTOR_(32, 16, \
+          simde__m128_to_private(a).f32, \
+          simde__m128_to_private(b).f32, \
+          (((imm8)     ) & 3), \
+          (((imm8) >> 2) & 3), \
+          (((imm8) >> 4) & 3) + 4, \
+          (((imm8) >> 6) & 3) + 4) }); }))
 #endif
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_shuffle_ps(a, b, imm8) simde_mm_shuffle_ps((a), (b), imm8)
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128
@@ -14501,14 +14842,16 @@
         est = vmulq_f32(vrsqrtsq_f32(vmulq_f32(a_.neon_f32, est), est), est);
       }
       r_.neon_f32 = vmulq_f32(a_.neon_f32, est);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_sqrt(a_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P7_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_14_NATIVE)
       r_.altivec_f32 = vec_sqrt(a_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_f32 = __lsx_vfsqrt_s(a_.lsx_f32);
     #elif defined(simde_math_sqrt)
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < sizeof(r_.f32) / sizeof(r_.f32[0]) ; i++) {
         r_.f32[i] = simde_math_sqrtf(a_.f32[i]);
       }
     #else
       HEDLEY_UNREACHABLE();
@@ -14565,14 +14908,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       vst1q_f32(mem_addr, a_.neon_f32);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       vec_st(a_.altivec_f32, 0, mem_addr);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       wasm_v128_store(mem_addr, a_.wasm_v128);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      __lsx_vst(a_.lsx_f32, mem_addr, 0);
     #else
       simde_memcpy(mem_addr, &a_, sizeof(a));
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_store_ps(mem_addr, a) simde_mm_store_ps(SIMDE_CHECKED_REINTERPRET_CAST(float*, simde_float32*, mem_addr), (a))
@@ -14590,14 +14935,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       vst1q_f32(mem_addr_, vdupq_lane_f32(vget_low_f32(a_.neon_f32), 0));
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       wasm_v128_store(mem_addr_, wasm_i32x4_shuffle(a_.wasm_v128, a_.wasm_v128, 0, 0, 0, 0));
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       vec_st(vec_splat(a_.altivec_f32, 0), 0, mem_addr_);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      __lsx_vst(__lsx_vreplvei_w(a_.lsx_f32, 0), mem_addr_, 0);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       simde__m128_private tmp_;
       tmp_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, a_.f32, 0, 0, 0, 0);
       simde_mm_store_ps(mem_addr_, tmp_.f32);
     #else
       SIMDE_VECTORIZE_ALIGNED(mem_addr_:16)
       for (size_t i = 0 ; i < sizeof(a_.f32) / sizeof(a_.f32[0]) ; i++) {
@@ -14618,14 +14965,18 @@
   #if defined(SIMDE_X86_SSE_NATIVE)
     _mm_store_ss(mem_addr, a);
   #else
     simde__m128_private a_ = simde__m128_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       vst1q_lane_f32(mem_addr, a_.neon_f32, 0);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      __lsx_vstelm_w(a_.lsx_f32, mem_addr, 0, 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      wasm_v128_store32_lane(HEDLEY_REINTERPRET_CAST(void*, mem_addr), a_.wasm_v128, 0);
     #else
       *mem_addr = a_.f32[0];
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_store_ss(mem_addr, a) simde_mm_store_ss(SIMDE_CHECKED_REINTERPRET_CAST(float*, simde_float32*, mem_addr), (a))
@@ -14637,28 +14988,33 @@
   #if defined(SIMDE_X86_SSE_NATIVE)
     _mm_storeh_pi(HEDLEY_REINTERPRET_CAST(__m64*, mem_addr), a);
   #else
     simde__m128_private a_ = simde__m128_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       vst1_f32(HEDLEY_REINTERPRET_CAST(float32_t*, mem_addr), vget_high_f32(a_.neon_f32));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      wasm_v128_store64_lane(HEDLEY_REINTERPRET_CAST(void*, mem_addr), a_.wasm_v128, 1);
     #else
       simde_memcpy(mem_addr, &(a_.m64[1]), sizeof(a_.m64[1]));
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_storeh_pi(mem_addr, a) simde_mm_storeh_pi(mem_addr, (a))
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 void
 simde_mm_storel_pi (simde__m64* mem_addr, simde__m128 a) {
   #if defined(SIMDE_X86_SSE_NATIVE)
     _mm_storel_pi(HEDLEY_REINTERPRET_CAST(__m64*, mem_addr), a);
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    simde__m128_private a_ = simde__m128_to_private(a);
+    wasm_v128_store64_lane(HEDLEY_REINTERPRET_CAST(void*, mem_addr), a_.wasm_v128, 0);
   #else
     simde__m64_private* dest_ = HEDLEY_REINTERPRET_CAST(simde__m64_private*, mem_addr);
     simde__m128_private a_ = simde__m128_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       dest_->neon_f32 = vget_low_f32(a_.neon_f32);
     #else
@@ -14680,14 +15036,16 @@
     simde__m128_private a_ = simde__m128_to_private(a);
 
     #if defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       vec_st(vec_reve(a_.altivec_f32), 0, mem_addr);
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       float32x4_t tmp = vrev64q_f32(a_.neon_f32);
       vst1q_f32(mem_addr, vextq_f32(tmp, tmp, 2));
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      __lsx_vst(__lsx_vshuf4i_w(a_.lsx_f32, 0x1b), mem_addr, 0);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       a_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, a_.f32, 3, 2, 1, 0);
       simde_mm_store_ps(mem_addr, simde__m128_from_private(a_));
     #else
       SIMDE_VECTORIZE_ALIGNED(mem_addr:16)
       for (size_t i = 0 ; i < sizeof(a_.f32) / sizeof(a_.f32[0]) ; i++) {
         mem_addr[i] = a_.f32[((sizeof(a_.f32) / sizeof(a_.f32[0])) - 1) - i];
@@ -14707,14 +15065,18 @@
   #else
     simde__m128_private a_ = simde__m128_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       vst1q_f32(mem_addr, a_.neon_f32);
     #elif defined(SIMDE_POWER_ALTIVEC_P7_NATIVE)
       vec_vsx_st(a_.altivec_f32, 0, mem_addr);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      __lsx_vst(a_.lsx_f32, mem_addr, 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      wasm_v128_store(mem_addr, a_.wasm_v128);
     #else
       simde_memcpy(mem_addr, &a_, sizeof(a_));
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE_ENABLE_NATIVE_ALIASES)
 #  define _mm_storeu_ps(mem_addr, a) simde_mm_storeu_ps(SIMDE_CHECKED_REINTERPRET_CAST(float*, simde_float32*, mem_addr), (a))
@@ -14733,14 +15095,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_f32 = vsubq_f32(a_.neon_f32, b_.neon_f32);
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
       r_.wasm_v128 = wasm_f32x4_sub(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = vec_sub(a_.altivec_f32, b_.altivec_f32);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_f32 = __lsx_vfsub_s(a_.lsx_f32, b_.lsx_f32);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.f32 = a_.f32 - b_.f32;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f32) / sizeof(r_.f32[0])) ; i++) {
         r_.f32[i] = a_.f32[i] - b_.f32[i];
       }
@@ -14794,14 +15158,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_or_b_nan = vmvnq_u32(vandq_u32(a_not_nan, b_not_nan));
       uint32x4_t a_eq_b = vceqq_f32(a_.neon_f32, b_.neon_f32);
       r = !!(vgetq_lane_u32(vorrq_u32(a_or_b_nan, a_eq_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r = wasm_f32x4_extract_lane(a_.wasm_v128, 0) == wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #elif defined(SIMDE_HAVE_FENV_H)
       fenv_t envp;
       int x = feholdexcept(&envp);
       r = a_.f32[0] == b_.f32[0];
       if (HEDLEY_LIKELY(x == 0))
         fesetenv(&envp);
     #else
@@ -14828,14 +15194,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_and_b_not_nan = vandq_u32(a_not_nan, b_not_nan);
       uint32x4_t a_ge_b = vcgeq_f32(a_.neon_f32, b_.neon_f32);
       r = !!(vgetq_lane_u32(vandq_u32(a_and_b_not_nan, a_ge_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r = wasm_f32x4_extract_lane(a_.wasm_v128, 0) >= wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #elif defined(SIMDE_HAVE_FENV_H)
       fenv_t envp;
       int x = feholdexcept(&envp);
       r = a_.f32[0] >= b_.f32[0];
       if (HEDLEY_LIKELY(x == 0))
         fesetenv(&envp);
     #else
@@ -14862,14 +15230,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_and_b_not_nan = vandq_u32(a_not_nan, b_not_nan);
       uint32x4_t a_gt_b = vcgtq_f32(a_.neon_f32, b_.neon_f32);
       r = !!(vgetq_lane_u32(vandq_u32(a_and_b_not_nan, a_gt_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r = wasm_f32x4_extract_lane(a_.wasm_v128, 0) > wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #elif defined(SIMDE_HAVE_FENV_H)
       fenv_t envp;
       int x = feholdexcept(&envp);
       r = a_.f32[0] > b_.f32[0];
       if (HEDLEY_LIKELY(x == 0))
         fesetenv(&envp);
     #else
@@ -14896,14 +15266,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_or_b_nan = vmvnq_u32(vandq_u32(a_not_nan, b_not_nan));
       uint32x4_t a_le_b = vcleq_f32(a_.neon_f32, b_.neon_f32);
       r = !!(vgetq_lane_u32(vorrq_u32(a_or_b_nan, a_le_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r = wasm_f32x4_extract_lane(a_.wasm_v128, 0) <= wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #elif defined(SIMDE_HAVE_FENV_H)
       fenv_t envp;
       int x = feholdexcept(&envp);
       r = a_.f32[0] <= b_.f32[0];
       if (HEDLEY_LIKELY(x == 0))
         fesetenv(&envp);
     #else
@@ -14930,14 +15302,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_or_b_nan = vmvnq_u32(vandq_u32(a_not_nan, b_not_nan));
       uint32x4_t a_lt_b = vcltq_f32(a_.neon_f32, b_.neon_f32);
       r = !!(vgetq_lane_u32(vorrq_u32(a_or_b_nan, a_lt_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r = wasm_f32x4_extract_lane(a_.wasm_v128, 0) < wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #elif defined(SIMDE_HAVE_FENV_H)
       fenv_t envp;
       int x = feholdexcept(&envp);
       r = a_.f32[0] < b_.f32[0];
       if (HEDLEY_LIKELY(x == 0))
         fesetenv(&envp);
     #else
@@ -14964,14 +15338,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x4_t a_not_nan = vceqq_f32(a_.neon_f32, a_.neon_f32);
       uint32x4_t b_not_nan = vceqq_f32(b_.neon_f32, b_.neon_f32);
       uint32x4_t a_and_b_not_nan = vandq_u32(a_not_nan, b_not_nan);
       uint32x4_t a_neq_b = vmvnq_u32(vceqq_f32(a_.neon_f32, b_.neon_f32));
       r = !!(vgetq_lane_u32(vandq_u32(a_and_b_not_nan, a_neq_b), 0) != 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r = wasm_f32x4_extract_lane(a_.wasm_v128, 0) != wasm_f32x4_extract_lane(b_.wasm_v128, 0);
     #elif defined(SIMDE_HAVE_FENV_H)
       fenv_t envp;
       int x = feholdexcept(&envp);
       r = a_.f32[0] != b_.f32[0];
       if (HEDLEY_LIKELY(x == 0))
         fesetenv(&envp);
     #else
@@ -14991,19 +15367,14 @@
 #      define SIMDE_HAVE_UNDEFINED128
 #    endif
 #  elif !defined(__PGI) && !defined(SIMDE_BUG_GCC_REV_208793) && !defined(_MSC_VER)
 #    define SIMDE_HAVE_UNDEFINED128
 #  endif
 #endif
 
-#if defined(SIMDE_DIAGNOSTIC_DISABLE_UNINITIALIZED_)
-  HEDLEY_DIAGNOSTIC_PUSH
-  SIMDE_DIAGNOSTIC_DISABLE_UNINITIALIZED_
-#endif
-
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128
 simde_mm_unpackhi_ps (simde__m128 a, simde__m128 b) {
   #if defined(SIMDE_X86_SSE_NATIVE)
     return _mm_unpackhi_ps(a, b);
   #else
     simde__m128_private
@@ -15014,14 +15385,18 @@
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f32 = vzip2q_f32(a_.neon_f32, b_.neon_f32);
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       float32x2_t a1 = vget_high_f32(a_.neon_f32);
       float32x2_t b1 = vget_high_f32(b_.neon_f32);
       float32x2x2_t result = vzip_f32(a1, b1);
       r_.neon_f32 = vcombine_f32(result.val[0], result.val[1]);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vilvh_w(b_.lsx_i64, a_.lsx_i64);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i32x4_shuffle(a_.wasm_v128, b_.wasm_v128, 2, 6, 3, 7);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, b_.f32, 2, 6, 3, 7);
     #else
       r_.f32[0] = a_.f32[2];
       r_.f32[1] = b_.f32[2];
       r_.f32[2] = a_.f32[3];
       r_.f32[3] = b_.f32[3];
@@ -15045,21 +15420,25 @@
       a_ = simde__m128_to_private(a),
       b_ = simde__m128_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f32 = vzip1q_f32(a_.neon_f32, b_.neon_f32);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_f32 = vec_mergeh(a_.altivec_f32, b_.altivec_f32);
-    #elif defined(SIMDE_SHUFFLE_VECTOR_)
-      r_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, b_.f32, 0, 4, 1, 5);
+    #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
+      r_.lsx_i64 = __lsx_vilvl_w(b_.lsx_i64, a_.lsx_i64);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i32x4_shuffle(a_.wasm_v128, b_.wasm_v128, 0, 4, 1, 5);
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       float32x2_t a1 = vget_low_f32(a_.neon_f32);
       float32x2_t b1 = vget_low_f32(b_.neon_f32);
       float32x2x2_t result = vzip_f32(a1, b1);
       r_.neon_f32 = vcombine_f32(result.val[0], result.val[1]);
+    #elif defined(SIMDE_SHUFFLE_VECTOR_)
+      r_.f32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.f32, b_.f32, 0, 4, 1, 5);
     #else
       r_.f32[0] = a_.f32[0];
       r_.f32[1] = b_.f32[0];
       r_.f32[2] = a_.f32[1];
       r_.f32[3] = b_.f32[1];
     #endif
 
@@ -15550,14 +15929,16 @@
       r_,
       a_ = simde__m128d_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f64 = vabsq_f64(a_.neon_f64);
     #elif defined(SIMDE_POWER_ALTIVEC_P7_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE)
       r_.altivec_f64 = vec_abs(a_.altivec_f64);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_f64x2_abs(a_.wasm_v128);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f64) / sizeof(r_.f64[0])) ; i++) {
         r_.f64[i] = simde_math_fabs(a_.f64[i]);
       }
     #endif
 
@@ -15848,14 +16229,16 @@
       r_,
       a_ = simde__m128d_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f64 = vdupq_laneq_f64(a_.neon_f64, 0);
     #elif defined(SIMDE_POWER_ALTIVEC_P7_NATIVE)
       r_.altivec_f64 = vec_splat(a_.altivec_f64, 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_f64x2_splat(a_.f64[0]);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.f64 = SIMDE_SHUFFLE_VECTOR_(64, 16, a_.f64, a_.f64, 0, 0);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f64) / sizeof(r_.f64[0])) ; i++) {
         r_.f64[i] = a_.f64[0];
       }
@@ -16082,14 +16465,16 @@
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vandq_s32(b_.neon_i32, a_.neon_i32);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_u32f = vec_and(a_.altivec_u32f, b_.altivec_u32f);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_and(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32f = a_.i32f & b_.i32f;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i32f) / sizeof(r_.i32f[0])) ; i++) {
         r_.i32f[i] = a_.i32f[i] & b_.i32f[i];
       }
@@ -16148,14 +16533,16 @@
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vbicq_s32(b_.neon_i32, a_.neon_i32);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) || defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE)
       r_.altivec_i32 = vec_andc(b_.altivec_i32, a_.altivec_i32);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_andnot(b_.wasm_v128, a_.wasm_v128);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32f = ~a_.i32f & b_.i32f;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i32f) / sizeof(r_.i32f[0])) ; i++) {
         r_.i32f[i] = ~(a_.i32f[i]) & b_.i32f[i];
       }
@@ -16341,26 +16728,47 @@
   return simde__m128i_from_private(r_);
 }
 #if defined(SIMDE_X86_SSE2_NATIVE) && !defined(__PGI)
   #define simde_mm_bslli_si128(a, imm8) _mm_slli_si128(a, imm8)
 #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE) && !defined(__clang__)
   #define simde_mm_bslli_si128(a, imm8) \
   simde__m128i_from_neon_i8(((imm8) <= 0) ? simde__m128i_to_neon_i8(a) : (((imm8) > 15) ? (vdupq_n_s8(0)) : (vextq_s8(vdupq_n_s8(0), simde__m128i_to_neon_i8(a), 16 - (imm8)))))
+#elif defined(SIMDE_WASM_SIMD128_NATIVE)
+  #define simde_mm_bslli_si128(a, imm8) __extension__ ({        \
+    simde__m128i_from_wasm_v128(                                \
+      wasm_i8x16_shuffle(wasm_i32x4_splat(INT32_C(0)),          \
+                         simde__m128i_to_wasm_v128((a)),        \
+                         ((imm8)&0xF0) ? 0 : 16 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 17 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 18 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 19 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 20 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 21 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 22 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 23 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 24 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 25 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 26 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 27 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 28 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 29 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 30 - ((imm8)&0xF), \
+                         ((imm8)&0xF0) ? 0 : 31 - ((imm8)&0xF))); })
 #elif defined(SIMDE_SHUFFLE_VECTOR_) && !defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) && !defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE)
   #define simde_mm_bslli_si128(a, imm8) (__extension__ ({ \
-    const simde__m128i_private simde__tmp_a_ = simde__m128i_to_private(a); \
-    const simde__m128i_private simde__tmp_z_ = simde__m128i_to_private(simde_mm_setzero_si128()); \
-    simde__m128i_private simde__tmp_r_; \
+    const simde__m128i_private simde_tmp_a_ = simde__m128i_to_private(a); \
+    const simde__m128i_private simde_tmp_z_ = simde__m128i_to_private(simde_mm_setzero_si128()); \
+    simde__m128i_private simde_tmp_r_; \
     if (HEDLEY_UNLIKELY(imm8 > 15)) { \
-      simde__tmp_r_ = simde__m128i_to_private(simde_mm_setzero_si128()); \
+      simde_tmp_r_ = simde__m128i_to_private(simde_mm_setzero_si128()); \
     } else { \
-      simde__tmp_r_.i8 = \
+      simde_tmp_r_.i8 = \
         SIMDE_SHUFFLE_VECTOR_(8, 16, \
-          simde__tmp_z_.i8, \
-          (simde__tmp_a_).i8, \
+          simde_tmp_z_.i8, \
+          (simde_tmp_a_).i8, \
           HEDLEY_STATIC_CAST(int8_t, (16 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (17 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (18 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (19 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (20 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (21 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (22 - imm8) & 31), \
@@ -16370,15 +16778,15 @@
           HEDLEY_STATIC_CAST(int8_t, (26 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (27 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (28 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (29 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (30 - imm8) & 31), \
           HEDLEY_STATIC_CAST(int8_t, (31 - imm8) & 31)); \
     } \
-    simde__m128i_from_private(simde__tmp_r_); }))
+    simde__m128i_from_private(simde_tmp_r_); }))
 #endif
 #define simde_mm_slli_si128(a, imm8) simde_mm_bslli_si128(a, imm8)
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_bslli_si128(a, imm8) simde_mm_bslli_si128(a, imm8)
   #define _mm_slli_si128(a, imm8) simde_mm_bslli_si128(a, imm8)
 #endif
 
@@ -16415,26 +16823,56 @@
   return simde__m128i_from_private(r_);
 }
 #if defined(SIMDE_X86_SSE2_NATIVE) && !defined(__PGI)
   #define simde_mm_bsrli_si128(a, imm8) _mm_srli_si128(a, imm8)
 #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE) && !defined(__clang__)
   #define simde_mm_bsrli_si128(a, imm8) \
   simde__m128i_from_neon_i8(((imm8 < 0) || (imm8 > 15)) ? vdupq_n_s8(0) : (vextq_s8(simde__m128i_to_private(a).neon_i8, vdupq_n_s8(0), ((imm8 & 15) != 0) ? imm8 : (imm8 & 15))))
+#elif defined(SIMDE_WASM_SIMD128_NATIVE)
+  #define simde_mm_bsrli_si128(a, imm8) (__extension__ ({ \
+    const simde__m128i_private simde_tmp_a_ = simde__m128i_to_private(a); \
+    const simde__m128i_private simde_tmp_z_ = simde__m128i_to_private(simde_mm_setzero_si128()); \
+    simde__m128i_private simde_tmp_r_ = simde__m128i_to_private(a); \
+    if (HEDLEY_UNLIKELY(imm8 > 15)) { \
+      simde_tmp_r_ = simde__m128i_to_private(simde_mm_setzero_si128()); \
+    } else { \
+      simde_tmp_r_.wasm_v128 = \
+      wasm_i8x16_shuffle( \
+        simde_tmp_z_.wasm_v128, \
+        simde_tmp_a_.wasm_v128, \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 16) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 17) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 18) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 19) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 20) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 21) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 22) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 23) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 24) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 25) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 26) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 27) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 28) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 29) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 30) & 31), \
+        HEDLEY_STATIC_CAST(int8_t, (imm8 + 31) & 31)); \
+    } \
+    simde__m128i_from_private(simde_tmp_r_); }))
 #elif defined(SIMDE_SHUFFLE_VECTOR_) && !defined(SIMDE_POWER_ALTIVEC_P6_NATIVE) && !defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE)
   #define simde_mm_bsrli_si128(a, imm8) (__extension__ ({ \
-    const simde__m128i_private simde__tmp_a_ = simde__m128i_to_private(a); \
-    const simde__m128i_private simde__tmp_z_ = simde__m128i_to_private(simde_mm_setzero_si128()); \
-    simde__m128i_private simde__tmp_r_ = simde__m128i_to_private(a); \
+    const simde__m128i_private simde_tmp_a_ = simde__m128i_to_private(a); \
+    const simde__m128i_private simde_tmp_z_ = simde__m128i_to_private(simde_mm_setzero_si128()); \
+    simde__m128i_private simde_tmp_r_ = simde__m128i_to_private(a); \
     if (HEDLEY_UNLIKELY(imm8 > 15)) { \
-      simde__tmp_r_ = simde__m128i_to_private(simde_mm_setzero_si128()); \
+      simde_tmp_r_ = simde__m128i_to_private(simde_mm_setzero_si128()); \
     } else { \
-      simde__tmp_r_.i8 = \
+      simde_tmp_r_.i8 = \
       SIMDE_SHUFFLE_VECTOR_(8, 16, \
-        simde__tmp_z_.i8, \
-        (simde__tmp_a_).i8, \
+        simde_tmp_z_.i8, \
+        (simde_tmp_a_).i8, \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 16) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 17) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 18) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 19) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 20) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 21) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 22) & 31), \
@@ -16444,15 +16882,15 @@
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 26) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 27) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 28) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 29) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 30) & 31), \
         HEDLEY_STATIC_CAST(int8_t, (imm8 + 31) & 31)); \
     } \
-    simde__m128i_from_private(simde__tmp_r_); }))
+    simde__m128i_from_private(simde_tmp_r_); }))
 #endif
 #define simde_mm_srli_si128(a, imm8) simde_mm_bsrli_si128((a), (imm8))
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_bsrli_si128(a, imm8) simde_mm_bsrli_si128((a), (imm8))
   #define _mm_srli_si128(a, imm8) simde_mm_bsrli_si128((a), (imm8))
 #endif
 
@@ -16462,15 +16900,15 @@
   #if defined(SIMDE_X86_SSE2_NATIVE)
     _mm_clflush(p);
   #else
     (void) p;
   #endif
 }
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
-  #define _mm_clflush(a, b) simde_mm_clflush()
+  #define _mm_clflush(p) simde_mm_clflush(p)
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 int
 simde_mm_comieq_sd (simde__m128d a, simde__m128d b) {
   #if defined(SIMDE_X86_SSE2_NATIVE)
     return _mm_comieq_sd(a, b);
@@ -17512,14 +17950,17 @@
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       /* Note: NEON does not have ordered compare builtin
         Need to compare a eq a and b eq b to check for NaN
         Do AND of results to get final */
       uint64x2_t ceqaa = vceqq_f64(a_.neon_f64, a_.neon_f64);
       uint64x2_t ceqbb = vceqq_f64(b_.neon_f64, b_.neon_f64);
       r_.neon_u64 = vandq_u64(ceqaa, ceqbb);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_and(wasm_f64x2_eq(a_.wasm_v128, a_.wasm_v128),
+                                   wasm_f64x2_eq(b_.wasm_v128, b_.wasm_v128));
     #elif defined(simde_math_isnan)
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f64) / sizeof(r_.f64[0])) ; i++) {
         r_.u64[i] = (!simde_math_isnan(a_.f64[i]) && !simde_math_isnan(b_.f64[i])) ? ~UINT64_C(0) : UINT64_C(0);
       }
     #else
       HEDLEY_UNREACHABLE();
@@ -17537,14 +17978,16 @@
 simde_mm_cvtsd_f64 (simde__m128d a) {
   #if defined(SIMDE_X86_SSE2_NATIVE) && !defined(__PGI)
     return _mm_cvtsd_f64(a);
   #else
     simde__m128d_private a_ = simde__m128d_to_private(a);
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       return HEDLEY_STATIC_CAST(simde_float64, vgetq_lane_f64(a_.neon_f64, 0));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      return HEDLEY_STATIC_CAST(simde_float64, wasm_f64x2_extract_lane(a_.wasm_v128, 0));
     #else
       return a_.f64[0];
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_cvtsd_f64(a) simde_mm_cvtsd_f64(a)
@@ -17590,14 +18033,17 @@
       a_ = simde__m128d_to_private(a),
       b_ = simde__m128d_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       uint64x2_t ceqaa = vceqq_f64(a_.neon_f64, a_.neon_f64);
       uint64x2_t ceqbb = vceqq_f64(b_.neon_f64, b_.neon_f64);
       r_.neon_u64 = vreinterpretq_u64_u32(vmvnq_u32(vreinterpretq_u32_u64(vandq_u64(ceqaa, ceqbb))));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_or(wasm_f64x2_ne(a_.wasm_v128, a_.wasm_v128),
+                                  wasm_f64x2_ne(b_.wasm_v128, b_.wasm_v128));
     #elif defined(simde_math_isnan)
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f64) / sizeof(r_.f64[0])) ; i++) {
         r_.u64[i] = (simde_math_isnan(a_.f64[i]) || simde_math_isnan(b_.f64[i])) ? ~UINT64_C(0) : UINT64_C(0);
       }
     #else
       HEDLEY_UNREACHABLE();
@@ -17644,15 +18090,17 @@
 simde_mm_cvtepi32_pd (simde__m128i a) {
   #if defined(SIMDE_X86_SSE2_NATIVE)
     return _mm_cvtepi32_pd(a);
   #else
     simde__m128d_private r_;
     simde__m128i_private a_ = simde__m128i_to_private(a);
 
-    #if defined(SIMDE_CONVERT_VECTOR_)
+    #if defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_f64x2_convert_low_i32x4(a_.wasm_v128);
+    #elif defined(SIMDE_CONVERT_VECTOR_)
       SIMDE_CONVERT_VECTOR_(r_.f64, a_.m64_private[0].i32);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f64) / sizeof(r_.f64[0])) ; i++) {
         r_.f64[i] = (simde_float64) a_.i32[i];
       }
     #endif
@@ -17854,15 +18302,17 @@
 simde_mm_cvtps_pd (simde__m128 a) {
   #if defined(SIMDE_X86_SSE2_NATIVE)
     return _mm_cvtps_pd(a);
   #else
     simde__m128d_private r_;
     simde__m128_private a_ = simde__m128_to_private(a);
 
-    #if defined(SIMDE_CONVERT_VECTOR_)
+    #if defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_f64x2_promote_low_f32x4(a_.wasm_v128);
+    #elif defined(SIMDE_CONVERT_VECTOR_)
       SIMDE_CONVERT_VECTOR_(r_.f64, a_.m64_private[0].f32);
     #elif defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f64 = vcvt_f64_f32(vget_low_f32(a_.neon_f32));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.f64) / sizeof(r_.f64[0])) ; i++) {
         r_.f64[i] = a_.f32[i];
@@ -18426,14 +18876,16 @@
 
   return  HEDLEY_STATIC_CAST(int32_t, r);
 }
 #if defined(SIMDE_X86_SSE2_NATIVE) && (!defined(HEDLEY_GCC_VERSION) || HEDLEY_GCC_VERSION_CHECK(4,6,0))
   #define simde_mm_extract_epi16(a, imm8) _mm_extract_epi16(a, imm8)
 #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
   #define simde_mm_extract_epi16(a, imm8) (HEDLEY_STATIC_CAST(int32_t, vgetq_lane_s16(simde__m128i_to_private(a).neon_i16, (imm8))) & (INT32_C(0x0000ffff)))
+#elif defined(SIMDE_WASM_SIMD128_NATIVE)
+  #define simde_mm_extract_epi16(a, imm8) HEDLEY_STATIC_CAST(int32_t, wasm_u16x8_extract_lane(simde__m128i_to_wasm_v128((a)), (imm8) & 7))
 #endif
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_extract_epi16(a, imm8) simde_mm_extract_epi16(a, imm8)
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
@@ -18443,14 +18895,16 @@
   a_.i16[imm8 & 7] = i;
   return simde__m128i_from_private(a_);
 }
 #if defined(SIMDE_X86_SSE2_NATIVE) && !defined(__PGI)
   #define simde_mm_insert_epi16(a, i, imm8) _mm_insert_epi16((a), (i), (imm8))
 #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
   #define simde_mm_insert_epi16(a, i, imm8) simde__m128i_from_neon_i16(vsetq_lane_s16((i), simde__m128i_to_neon_i16(a), (imm8)))
+#elif defined(SIMDE_WASM_SIMD128_NATIVE)
+  #define simde_mm_insert_epi16(a, i, imm8) wasm_i16x8_replace_lane(simde__m128i_to_wasm_v128((a)), (imm8) & 7, (i))
 #endif
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_insert_epi16(a, i, imm8) simde_mm_insert_epi16(a, i, imm8)
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128d
@@ -18460,14 +18914,16 @@
   #else
     simde__m128d_private r_;
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f64 = vld1q_f64(mem_addr);
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_u32 = vld1q_u32(HEDLEY_REINTERPRET_CAST(uint32_t const*, mem_addr));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_load(mem_addr);
     #else
       simde_memcpy(&r_, SIMDE_ALIGN_ASSUME_LIKE(mem_addr, simde__m128d), sizeof(r_));
     #endif
 
     return simde__m128d_from_private(r_);
   #endif
 }
@@ -18500,14 +18956,16 @@
   #if defined(SIMDE_X86_SSE2_NATIVE)
     return _mm_load_sd(mem_addr);
   #else
     simde__m128d_private r_;
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f64 = vsetq_lane_f64(*mem_addr, vdupq_n_f64(0), 0);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_load64_zero(HEDLEY_REINTERPRET_CAST(const void*, mem_addr));
     #else
       r_.f64[0] = *mem_addr;
       r_.u64[1] = UINT64_C(0);
     #endif
 
     return simde__m128d_from_private(r_);
   #endif
@@ -18547,14 +19005,16 @@
   #else
     simde__m128d_private
       r_,
       a_ = simde__m128d_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f64 = vcombine_f64(vget_low_f64(a_.neon_f64), vld1_f64(HEDLEY_REINTERPRET_CAST(const float64_t*, mem_addr)));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_load64_lane(HEDLEY_REINTERPRET_CAST(const void*, mem_addr), a_.wasm_v128, 1);
     #else
       simde_float64 t;
 
       simde_memcpy(&t, mem_addr, sizeof(t));
       r_.f64[0] = a_.f64[0];
       r_.f64[1] = t;
     #endif
@@ -18600,14 +19060,16 @@
     simde__m128d_private
       r_,
       a_ = simde__m128d_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f64 = vcombine_f64(vld1_f64(
         HEDLEY_REINTERPRET_CAST(const float64_t*, mem_addr)), vget_high_f64(a_.neon_f64));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_load64_lane(HEDLEY_REINTERPRET_CAST(const void*, mem_addr), a_.wasm_v128, 0);
     #else
       r_.f64[0] = *mem_addr;
       r_.u64[1] = a_.u64[1];
     #endif
 
     return simde__m128d_from_private(r_);
   #endif
@@ -18661,108 +19123,123 @@
     return simde__m128d_from_private(r_);
   #endif
 }
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_loadu_pd(mem_addr) simde_mm_loadu_pd(mem_addr)
 #endif
 
+#if defined(SIMDE_X86_AVX512VL_NATIVE) && defined(SIMDE_X86_AVX512BW_NATIVE) \
+    && !defined(SIMDE_BUG_GCC_95483) && !defined(SIMDE_BUG_CLANG_REV_344862) \
+    && (!defined(HEDLEY_MSVC_VERSION) || HEDLEY_MSVC_VERSION_CHECK(19,20,0))
+  #define simde_mm_loadu_epi8(mem_addr) _mm_loadu_epi8(mem_addr)
+#else
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
 simde_mm_loadu_epi8(void const * mem_addr) {
-  #if defined(SIMDE_X86_AVX512VL_NATIVE) && defined(SIMDE_X86_AVX512BW_NATIVE) && !defined(SIMDE_BUG_GCC_95483) && !defined(SIMDE_BUG_CLANG_REV_344862)
-    return _mm_loadu_epi8(mem_addr);
-  #elif defined(SIMDE_X86_SSE2_NATIVE)
+  #if defined(SIMDE_X86_SSE2_NATIVE)
     return _mm_loadu_si128(SIMDE_ALIGN_CAST(__m128i const *, mem_addr));
   #else
     simde__m128i_private r_;
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i8 = vld1q_s8(HEDLEY_REINTERPRET_CAST(int8_t const*, mem_addr));
     #else
       simde_memcpy(&r_, mem_addr, sizeof(r_));
     #endif
 
     return simde__m128i_from_private(r_);
   #endif
 }
+#endif
 #define simde_x_mm_loadu_epi8(mem_addr) simde_mm_loadu_epi8(mem_addr)
 #if defined(SIMDE_X86_AVX512VL_ENABLE_NATIVE_ALIASES) || defined(SIMDE_X86_AVX512BW_ENABLE_NATIVE_ALIASES) || (defined(SIMDE_ENABLE_NATIVE_ALIASES) && (defined(SIMDE_BUG_GCC_95483) || defined(SIMDE_BUG_CLANG_REV_344862)))
   #undef _mm_loadu_epi8
   #define _mm_loadu_epi8(a) simde_mm_loadu_epi8(a)
 #endif
 
+#if defined(SIMDE_X86_AVX512VL_NATIVE) && defined(SIMDE_X86_AVX512BW_NATIVE) \
+    && !defined(SIMDE_BUG_GCC_95483) && !defined(SIMDE_BUG_CLANG_REV_344862) \
+    && (!defined(HEDLEY_MSVC_VERSION) || HEDLEY_MSVC_VERSION_CHECK(19,20,0))
+  #define simde_mm_loadu_epi16(mem_addr) _mm_loadu_epi16(mem_addr)
+#else
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
 simde_mm_loadu_epi16(void const * mem_addr) {
-  #if defined(SIMDE_X86_AVX512VL_NATIVE) && defined(SIMDE_X86_AVX512BW_NATIVE) && !defined(SIMDE_BUG_GCC_95483) && !defined(SIMDE_BUG_CLANG_REV_344862)
-    return _mm_loadu_epi16(mem_addr);
-  #elif defined(SIMDE_X86_SSE2_NATIVE)
+  #if defined(SIMDE_X86_SSE2_NATIVE)
     return _mm_loadu_si128(SIMDE_ALIGN_CAST(__m128i const *, mem_addr));
   #else
     simde__m128i_private r_;
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i16 = vreinterpretq_s16_s8(vld1q_s8(HEDLEY_REINTERPRET_CAST(int8_t const*, mem_addr)));
     #else
       simde_memcpy(&r_, mem_addr, sizeof(r_));
     #endif
 
     return simde__m128i_from_private(r_);
   #endif
 }
+#endif
 #define simde_x_mm_loadu_epi16(mem_addr) simde_mm_loadu_epi16(mem_addr)
 #if defined(SIMDE_X86_AVX512VL_ENABLE_NATIVE_ALIASES) || defined(SIMDE_X86_AVX512BW_ENABLE_NATIVE_ALIASES) || (defined(SIMDE_ENABLE_NATIVE_ALIASES) && (defined(SIMDE_BUG_GCC_95483) || defined(SIMDE_BUG_CLANG_REV_344862)))
   #undef _mm_loadu_epi16
   #define _mm_loadu_epi16(a) simde_mm_loadu_epi16(a)
 #endif
 
+#if defined(SIMDE_X86_AVX512VL_NATIVE) && !defined(SIMDE_BUG_GCC_95483) \
+    && !defined(SIMDE_BUG_CLANG_REV_344862) && (!defined(HEDLEY_MSVC_VERSION) || HEDLEY_MSVC_VERSION_CHECK(19,20,0))
+  #define simde_mm_loadu_epi32(mem_addr) _mm_loadu_epi32(mem_addr)
+#else
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
 simde_mm_loadu_epi32(void const * mem_addr) {
-  #if defined(SIMDE_X86_AVX512VL_NATIVE) && !defined(SIMDE_BUG_GCC_95483) && !defined(SIMDE_BUG_CLANG_REV_344862)
-    return _mm_loadu_epi32(mem_addr);
-  #elif defined(SIMDE_X86_SSE2_NATIVE)
+  #if defined(SIMDE_X86_SSE2_NATIVE)
     return _mm_loadu_si128(SIMDE_ALIGN_CAST(__m128i const *, mem_addr));
   #else
     simde__m128i_private r_;
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vreinterpretq_s32_s8(vld1q_s8(HEDLEY_REINTERPRET_CAST(int8_t const*, mem_addr)));
     #else
       simde_memcpy(&r_, mem_addr, sizeof(r_));
     #endif
 
     return simde__m128i_from_private(r_);
   #endif
 }
+#endif
 #define simde_x_mm_loadu_epi32(mem_addr) simde_mm_loadu_epi32(mem_addr)
 #if defined(SIMDE_X86_AVX512VL_ENABLE_NATIVE_ALIASES) || (defined(SIMDE_ENABLE_NATIVE_ALIASES) && (defined(SIMDE_BUG_GCC_95483) || defined(SIMDE_BUG_CLANG_REV_344862)))
   #undef _mm_loadu_epi32
   #define _mm_loadu_epi32(a) simde_mm_loadu_epi32(a)
 #endif
 
+#if defined(SIMDE_X86_AVX512VL_NATIVE) && !defined(SIMDE_BUG_GCC_95483) \
+    && !defined(SIMDE_BUG_CLANG_REV_344862) \
+    && (!defined(HEDLEY_MSVC_VERSION) || HEDLEY_MSVC_VERSION_CHECK(19,20,0))
+  #define simde_mm_loadu_epi64(mem_addr) _mm_loadu_epi64(mem_addr)
+#else
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
 simde_mm_loadu_epi64(void const * mem_addr) {
-  #if defined(SIMDE_X86_AVX512VL_NATIVE) && !defined(SIMDE_BUG_GCC_95483) && !defined(SIMDE_BUG_CLANG_REV_344862)
-    return _mm_loadu_epi64(mem_addr);
-  #elif defined(SIMDE_X86_SSE2_NATIVE)
+  #if defined(SIMDE_X86_SSE2_NATIVE)
     return _mm_loadu_si128(SIMDE_ALIGN_CAST(__m128i const *, mem_addr));
   #else
     simde__m128i_private r_;
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i64 = vreinterpretq_s64_s8(vld1q_s8(HEDLEY_REINTERPRET_CAST(int8_t const*, mem_addr)));
     #else
       simde_memcpy(&r_, mem_addr, sizeof(r_));
     #endif
 
     return simde__m128i_from_private(r_);
   #endif
 }
+#endif
 #define simde_x_mm_loadu_epi64(mem_addr) simde_mm_loadu_epi64(mem_addr)
 #if defined(SIMDE_X86_AVX512VL_ENABLE_NATIVE_ALIASES) || (defined(SIMDE_ENABLE_NATIVE_ALIASES) && (defined(SIMDE_BUG_GCC_95483) || defined(SIMDE_BUG_CLANG_REV_344862)))
   #undef _mm_loadu_epi64
   #define _mm_loadu_epi64(a) simde_mm_loadu_epi64(a)
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
@@ -18815,14 +19292,16 @@
       int32x2_t rl = vpadd_s32(vget_low_s32(pl), vget_high_s32(pl));
       int32x2_t rh = vpadd_s32(vget_low_s32(ph), vget_high_s32(ph));
       r_.neon_i32 = vcombine_s32(rl, rh);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_i32 = vec_msum(a_.altivec_i16, b_.altivec_i16, vec_splats(0));
     #elif defined(SIMDE_ZARCH_ZVECTOR_13_NATIVE)
       r_.altivec_i32 = vec_mule(a_.altivec_i16, b_.altivec_i16) + vec_mulo(a_.altivec_i16, b_.altivec_i16);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i32x4_dot_i16x8(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS) && defined(SIMDE_CONVERT_VECTOR_) && HEDLEY_HAS_BUILTIN(__builtin_shufflevector)
       int32_t SIMDE_VECTOR(32) a32, b32, p32;
       SIMDE_CONVERT_VECTOR_(a32, a_.i16);
       SIMDE_CONVERT_VECTOR_(b32, b_.i16);
       p32 = a32 * b32;
       r_.i32 =
         __builtin_shufflevector(p32, p32, 0, 2, 4, 6) +
@@ -18898,14 +19377,16 @@
       #endif
     #elif defined(SIMDE_POWER_ALTIVEC_P8_NATIVE) && !defined(HEDLEY_IBM_VERSION) && (SIMDE_ENDIAN_ORDER == SIMDE_ENDIAN_LITTLE)
       static const SIMDE_POWER_ALTIVEC_VECTOR(unsigned char) perm = { 120, 112, 104, 96, 88, 80, 72, 64, 56, 48, 40, 32, 24, 16, 8, 0 };
       r = HEDLEY_STATIC_CAST(int32_t, vec_extract(vec_vbpermq(a_.altivec_u8, perm), 1));
     #elif defined(SIMDE_POWER_ALTIVEC_P8_NATIVE) && !defined(HEDLEY_IBM_VERSION) && (SIMDE_ENDIAN_ORDER == SIMDE_ENDIAN_BIG)
       static const SIMDE_POWER_ALTIVEC_VECTOR(unsigned char) perm = { 120, 112, 104, 96, 88, 80, 72, 64, 56, 48, 40, 32, 24, 16, 8, 0 };
       r = HEDLEY_STATIC_CAST(int32_t, vec_extract(vec_vbpermq(a_.altivec_u8, perm), 14));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r = HEDLEY_STATIC_CAST(int32_t, wasm_i8x16_bitmask(a_.wasm_v128));
     #else
       SIMDE_VECTORIZE_REDUCTION(|:r)
       for (size_t i = 0 ; i < (sizeof(a_.u8) / sizeof(a_.u8[0])) ; i++) {
         r |= (a_.u8[15 - i] >> 7) << (15 - i);
       }
     #endif
 
@@ -18937,14 +19418,16 @@
       SIMDE_POWER_ALTIVEC_VECTOR(unsigned char) idx = { 64, 0, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128 };
       SIMDE_POWER_ALTIVEC_VECTOR(unsigned char) res = HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(unsigned char), vec_bperm(HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(unsigned __int128), a_.altivec_u64), idx));
       r = HEDLEY_STATIC_CAST(int32_t, vec_extract(HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(signed int), res), 2));
     #elif defined(SIMDE_POWER_ALTIVEC_P8_NATIVE)
       SIMDE_POWER_ALTIVEC_VECTOR(unsigned char) idx = { 64, 0, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128 };
       SIMDE_POWER_ALTIVEC_VECTOR(unsigned char) res = vec_bperm(a_.altivec_u8, idx);
       r = HEDLEY_STATIC_CAST(int32_t, vec_extract(HEDLEY_REINTERPRET_CAST(SIMDE_POWER_ALTIVEC_VECTOR(signed int), res), 2));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r = HEDLEY_STATIC_CAST(int32_t, wasm_i64x2_bitmask(a_.wasm_v128));
     #else
       SIMDE_VECTORIZE_REDUCTION(|:r)
       for (size_t i = 0 ; i < (sizeof(a_.u64) / sizeof(a_.u64[0])) ; i++) {
         r |= (a_.u64[i] >> 63) << i;
       }
     #endif
 
@@ -19254,14 +19737,16 @@
   #else
     simde__m128i_private
       r_,
       a_ = simde__m128i_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i64 = vsetq_lane_s64(0, a_.neon_i64, 1);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i64x2_shuffle(a_.wasm_v128, wasm_i64x2_const(0, 0), 0, 2);
     #else
       r_.i64[0] = a_.i64[0];
       r_.i64[1] = 0;
     #endif
 
     return simde__m128i_from_private(r_);
   #endif
@@ -19281,14 +19766,18 @@
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       uint32x2_t a_lo = vmovn_u64(a_.neon_u64);
       uint32x2_t b_lo = vmovn_u64(b_.neon_u64);
       r_.neon_u64 = vmull_u32(a_lo, b_lo);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_u64x2_extmul_low_u32x4(
+        wasm_i32x4_shuffle(a_.wasm_v128, a_.wasm_v128, 0, 2, 0, 2),
+        wasm_i32x4_shuffle(b_.wasm_v128, b_.wasm_v128, 0, 2, 0, 2));
     #elif defined(SIMDE_SHUFFLE_VECTOR_) && (SIMDE_ENDIAN_ORDER == SIMDE_ENDIAN_LITTLE)
       __typeof__(a_.u32) z = { 0, };
       a_.u32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.u32, z, 0, 4, 2, 6);
       b_.u32 = SIMDE_SHUFFLE_VECTOR_(32, 16, b_.u32, z, 0, 4, 2, 6);
       r_.u64 = HEDLEY_REINTERPRET_CAST(__typeof__(r_.u64), a_.u32) *
                HEDLEY_REINTERPRET_CAST(__typeof__(r_.u64), b_.u32);
     #else
@@ -19309,15 +19798,17 @@
 simde__m128i
 simde_x_mm_mul_epi64 (simde__m128i a, simde__m128i b) {
   simde__m128i_private
     r_,
     a_ = simde__m128i_to_private(a),
     b_ = simde__m128i_to_private(b);
 
-  #if defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
+  #if defined(SIMDE_WASM_SIMD128_NATIVE)
+    r_.wasm_v128 = wasm_i64x2_mul(a_.wasm_v128, b_.wasm_v128);
+  #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
     r_.i64 = a_.i64 * b_.i64;
   #else
     SIMDE_VECTORIZE
     for (size_t i = 0 ; i < (sizeof(r_.i64) / sizeof(r_.i64[0])) ; i++) {
       r_.i64[i] = a_.i64[i] * b_.i64[i];
     }
   #endif
@@ -19451,14 +19942,18 @@
       #else
         int16x4_t a7654 = vget_high_s16(a_.neon_i16);
         int16x4_t b7654 = vget_high_s16(b_.neon_i16);
         int32x4_t ab7654 = vmull_s16(a7654, b7654); /* 7777666655554444 */
         uint16x8x2_t rv = vuzpq_u16(vreinterpretq_u16_s32(ab3210), vreinterpretq_u16_s32(ab7654));
         r_.neon_u16 = rv.val[1];
       #endif
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      const v128_t lo = wasm_i32x4_extmul_low_i16x8(a_.wasm_v128, b_.wasm_v128);
+      const v128_t hi = wasm_i32x4_extmul_high_i16x8(a_.wasm_v128, b_.wasm_v128);
+      r_.wasm_v128 = wasm_i16x8_shuffle(lo, hi, 1, 3, 5, 7, 9, 11, 13, 15);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i16) / sizeof(r_.i16[0])) ; i++) {
         r_.u16[i] = HEDLEY_STATIC_CAST(uint16_t, (HEDLEY_STATIC_CAST(uint32_t, HEDLEY_STATIC_CAST(int32_t, a_.i16[i]) * HEDLEY_STATIC_CAST(int32_t, b_.i16[i])) >> 16));
       }
     #endif
 
@@ -19490,14 +19985,18 @@
       #else
         uint16x4_t a7654 = vget_high_u16(a_.neon_u16);
         uint16x4_t b7654 = vget_high_u16(b_.neon_u16);
         uint32x4_t ab7654 = vmull_u16(a7654, b7654); /* 7777666655554444 */
         uint16x8x2_t neon_r = vuzpq_u16(vreinterpretq_u16_u32(ab3210), vreinterpretq_u16_u32(ab7654));
         r_.neon_u16 = neon_r.val[1];
       #endif
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      const v128_t lo = wasm_u32x4_extmul_low_u16x8(a_.wasm_v128, b_.wasm_v128);
+      const v128_t hi = wasm_u32x4_extmul_high_u16x8(a_.wasm_v128, b_.wasm_v128);
+      r_.wasm_v128 = wasm_i16x8_shuffle(lo, hi, 1, 3, 5, 7, 9, 11, 13, 15);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.u16) / sizeof(r_.u16[0])) ; i++) {
         r_.u16[i] = HEDLEY_STATIC_CAST(uint16_t, HEDLEY_STATIC_CAST(uint32_t, a_.u16[i]) * HEDLEY_STATIC_CAST(uint32_t, b_.u16[i]) >> 16);
       }
     #endif
 
@@ -19521,14 +20020,16 @@
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i16 = vmulq_s16(a_.neon_i16, b_.neon_i16);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       (void) a_;
       (void) b_;
       r_.altivec_i16 = vec_mul(a_.altivec_i16, b_.altivec_i16);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i16x8_mul(a_.wasm_v128, b_.wasm_v128);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i16) / sizeof(r_.i16[0])) ; i++) {
         r_.u16[i] = HEDLEY_STATIC_CAST(uint16_t, HEDLEY_STATIC_CAST(uint32_t, a_.u16[i]) * HEDLEY_STATIC_CAST(uint32_t, b_.u16[i]));
       }
     #endif
 
@@ -19581,14 +20082,16 @@
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vorrq_s32(a_.neon_i32, b_.neon_i32);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_i32 = vec_or(a_.altivec_i32, b_.altivec_i32);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_or(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32f = a_.i32f | b_.i32f;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i32f) / sizeof(r_.i32f[0])) ; i++) {
         r_.i32f[i] = a_.i32f[i] | b_.i32f[i];
       }
@@ -19876,15 +20379,16 @@
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
 simde_mm_loadu_si16 (void const* mem_addr) {
   #if defined(SIMDE_X86_SSE2_NATIVE) && ( \
       SIMDE_DETECT_CLANG_VERSION_CHECK(8,0,0) || \
-      HEDLEY_INTEL_VERSION_CHECK(20,21,1))
+      HEDLEY_INTEL_VERSION_CHECK(20,21,1) || \
+      HEDLEY_GCC_VERSION_CHECK(12,1,0))
     return _mm_loadu_si16(mem_addr);
   #else
     int16_t val;
     simde_memcpy(&val, mem_addr, sizeof(val));
     return simde_x_mm_cvtsi16_si128(val);
   #endif
 }
@@ -19920,16 +20424,19 @@
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
 simde_mm_loadu_si32 (void const* mem_addr) {
   #if defined(SIMDE_X86_SSE2_NATIVE) && ( \
       SIMDE_DETECT_CLANG_VERSION_CHECK(8,0,0) || \
-      HEDLEY_INTEL_VERSION_CHECK(20,21,1))
+      HEDLEY_INTEL_VERSION_CHECK(20,21,1) || \
+      HEDLEY_GCC_VERSION_CHECK(12,1,0))
     return _mm_loadu_si32(mem_addr);
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    return simde__m128i_from_wasm_v128(wasm_v128_load32_zero(mem_addr));
   #else
     int32_t val;
     simde_memcpy(&val, mem_addr, sizeof(val));
     return simde_mm_cvtsi32_si128(val);
   #endif
 }
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
@@ -20019,14 +20526,16 @@
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       SIMDE_ALIGN_LIKE_16(uint8x16_t) uint8_t data[16] = {
         e0,  e1,  e2,  e3,
         e4,  e5,  e6,  e7,
         e8,  e9,  e10, e11,
         e12, e13, e14, e15};
       r_.neon_u8 = vld1q_u8(data);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_u8x16_make(e0, e1, e2, e3, e4, e5, e6, e7, e8, e9, e10, e11, e12, e13, e14, e15);
     #else
       r_.u8[ 0] =  e0; r_.u8[ 1] =  e1; r_.u8[ 2] =  e2; r_.u8[ 3] =  e3;
       r_.u8[ 4] =  e4; r_.u8[ 5] =  e5; r_.u8[ 6] =  e6; r_.u8[ 7] =  e7;
       r_.u8[ 8] =  e8; r_.u8[ 9] =  e9; r_.u8[10] = e10; r_.u8[11] = e11;
       r_.u8[12] = e12; r_.u8[13] = e13; r_.u8[14] = e14; r_.u8[15] = e15;
     #endif
 
@@ -20044,14 +20553,16 @@
       HEDLEY_STATIC_CAST(short,  e3), HEDLEY_STATIC_CAST(short,  e2), HEDLEY_STATIC_CAST(short,  e1), HEDLEY_STATIC_CAST(short,  e0));
   #else
     simde__m128i_private r_;
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       SIMDE_ALIGN_LIKE_16(uint16x8_t) uint16_t data[8] = { e0, e1, e2, e3, e4, e5, e6, e7 };
       r_.neon_u16 = vld1q_u16(data);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_u16x8_make(e0, e1, e2, e3, e4, e5, e6, e7);
     #else
       r_.u16[0] = e0; r_.u16[1] = e1; r_.u16[2] = e2; r_.u16[3] = e3;
       r_.u16[4] = e4; r_.u16[5] = e5; r_.u16[6] = e6; r_.u16[7] = e7;
     #endif
 
     return simde__m128i_from_private(r_);
   #endif
@@ -20065,14 +20576,16 @@
       HEDLEY_STATIC_CAST(int,  e3), HEDLEY_STATIC_CAST(int,  e2), HEDLEY_STATIC_CAST(int,  e1), HEDLEY_STATIC_CAST(int,  e0));
   #else
     simde__m128i_private r_;
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       SIMDE_ALIGN_LIKE_16(uint32x4_t) uint32_t data[4] = { e0, e1, e2, e3 };
       r_.neon_u32 = vld1q_u32(data);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_u32x4_make(e0, e1, e2, e3);
     #else
       r_.u32[0] = e0;
       r_.u32[1] = e1;
       r_.u32[2] = e2;
       r_.u32[3] = e3;
     #endif
 
@@ -20087,14 +20600,16 @@
     return _mm_set_epi64x(HEDLEY_STATIC_CAST(int64_t,  e1), HEDLEY_STATIC_CAST(int64_t,  e0));
   #else
     simde__m128i_private r_;
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       SIMDE_ALIGN_LIKE_16(uint64x2_t) uint64_t data[2] = {e0, e1};
       r_.neon_u64 = vld1q_u64(data);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_u64x2_make(e0, e1);
     #else
       r_.u64[0] = e0;
       r_.u64[1] = e1;
     #endif
 
     return simde__m128i_from_private(r_);
   #endif
@@ -20103,14 +20618,16 @@
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128d
 simde_mm_set_sd (simde_float64 a) {
   #if defined(SIMDE_X86_SSE2_NATIVE)
     return _mm_set_sd(a);
   #elif defined(SIMDE_ARM_NEON_A64V8_NATIVE)
     return vsetq_lane_f64(a, vdupq_n_f64(SIMDE_FLOAT64_C(0.0)), 0);
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    return simde__m128d_from_wasm_v128(wasm_f64x2_make(a, 0));
   #else
     return simde_mm_set_pd(SIMDE_FLOAT64_C(0.0), a);
   #endif
 }
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_set_sd(a) simde_mm_set_sd(a)
 #endif
@@ -20242,44 +20759,52 @@
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
 simde_x_mm_set1_epu8 (uint8_t value) {
   #if defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
     return simde__m128i_from_altivec_u8(vec_splats(HEDLEY_STATIC_CAST(unsigned char, value)));
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    return simde__m128i_from_wasm_v128(wasm_u8x16_splat(value));
   #else
     return simde_mm_set1_epi8(HEDLEY_STATIC_CAST(int8_t, value));
   #endif
 }
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
 simde_x_mm_set1_epu16 (uint16_t value) {
   #if defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
     return simde__m128i_from_altivec_u16(vec_splats(HEDLEY_STATIC_CAST(unsigned short, value)));
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    return simde__m128i_from_wasm_v128(wasm_u16x8_splat(value));
   #else
     return simde_mm_set1_epi16(HEDLEY_STATIC_CAST(int16_t, value));
   #endif
 }
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
 simde_x_mm_set1_epu32 (uint32_t value) {
   #if defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
     return simde__m128i_from_altivec_u32(vec_splats(HEDLEY_STATIC_CAST(unsigned int, value)));
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    return simde__m128i_from_wasm_v128(wasm_u32x4_splat(value));
   #else
     return simde_mm_set1_epi32(HEDLEY_STATIC_CAST(int32_t, value));
   #endif
 }
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
 simde_x_mm_set1_epu64 (uint64_t value) {
   #if defined(SIMDE_POWER_ALTIVEC_P7_NATIVE)
     return simde__m128i_from_altivec_u64(vec_splats(HEDLEY_STATIC_CAST(unsigned long long, value)));
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    return simde__m128i_from_wasm_v128(wasm_u64x2_splat(value));
   #else
     return simde_mm_set1_epi64x(HEDLEY_STATIC_CAST(int64_t, value));
   #endif
 }
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
@@ -20355,14 +20880,16 @@
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128d
 simde_mm_setzero_pd (void) {
   #if defined(SIMDE_X86_SSE2_NATIVE)
     return _mm_setzero_pd();
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    return simde__m128d_from_wasm_v128(wasm_f64x2_const(0.0, 0.0));
   #else
     return simde_mm_castsi128_pd(simde_mm_setzero_si128());
   #endif
 }
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_setzero_pd() simde_mm_setzero_pd()
 #endif
@@ -20434,36 +20961,47 @@
     r_.i32[i] = a_.i32[(imm8 >> (i * 2)) & 3];
   }
 
   return simde__m128i_from_private(r_);
 }
 #if defined(SIMDE_X86_SSE2_NATIVE)
   #define simde_mm_shuffle_epi32(a, imm8) _mm_shuffle_epi32((a), (imm8))
-#elif defined(SIMDE_SHUFFLE_VECTOR_)
+#elif defined(SIMDE_WASM_SIMD128_NATIVE)
   #define simde_mm_shuffle_epi32(a, imm8) (__extension__ ({ \
-      const simde__m128i_private simde__tmp_a_ = simde__m128i_to_private(a); \
-      simde__m128i_from_private((simde__m128i_private) { .i32 = \
-        SIMDE_SHUFFLE_VECTOR_(32, 16, \
-          (simde__tmp_a_).i32, \
-          (simde__tmp_a_).i32, \
+      const simde__m128i_private simde_tmp_a_ = simde__m128i_to_private(a); \
+      simde__m128i_from_wasm_v128( \
+        wasm_i32x4_shuffle( \
+          (simde_tmp_a_).wasm_v128, \
+          (simde_tmp_a_).wasm_v128, \
           ((imm8)     ) & 3, \
           ((imm8) >> 2) & 3, \
           ((imm8) >> 4) & 3, \
-          ((imm8) >> 6) & 3) }); }))
+          ((imm8) >> 6) & 3)); }))
 #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE) && defined(SIMDE_STATEMENT_EXPR_)
   #define simde_mm_shuffle_epi32(a, imm8) \
     (__extension__ ({ \
       const int32x4_t simde_mm_shuffle_epi32_a_ = simde__m128i_to_neon_i32(a); \
       int32x4_t simde_mm_shuffle_epi32_r_; \
       simde_mm_shuffle_epi32_r_ = vmovq_n_s32(vgetq_lane_s32(simde_mm_shuffle_epi32_a_, (imm8) & (0x3))); \
       simde_mm_shuffle_epi32_r_ = vsetq_lane_s32(vgetq_lane_s32(simde_mm_shuffle_epi32_a_, ((imm8) >> 2) & 0x3), simde_mm_shuffle_epi32_r_, 1); \
       simde_mm_shuffle_epi32_r_ = vsetq_lane_s32(vgetq_lane_s32(simde_mm_shuffle_epi32_a_, ((imm8) >> 4) & 0x3), simde_mm_shuffle_epi32_r_, 2); \
       simde_mm_shuffle_epi32_r_ = vsetq_lane_s32(vgetq_lane_s32(simde_mm_shuffle_epi32_a_, ((imm8) >> 6) & 0x3), simde_mm_shuffle_epi32_r_, 3); \
       vreinterpretq_s64_s32(simde_mm_shuffle_epi32_r_); \
     }))
+#elif defined(SIMDE_SHUFFLE_VECTOR_)
+  #define simde_mm_shuffle_epi32(a, imm8) (__extension__ ({ \
+      const simde__m128i_private simde_tmp_a_ = simde__m128i_to_private(a); \
+      simde__m128i_from_private((simde__m128i_private) { .i32 = \
+        SIMDE_SHUFFLE_VECTOR_(32, 16, \
+          (simde_tmp_a_).i32, \
+          (simde_tmp_a_).i32, \
+          ((imm8)     ) & 3, \
+          ((imm8) >> 2) & 3, \
+          ((imm8) >> 4) & 3, \
+          ((imm8) >> 6) & 3) }); }))
 #endif
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_shuffle_epi32(a, imm8) simde_mm_shuffle_epi32(a, imm8)
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128d
@@ -20510,37 +21048,49 @@
     r_.i16[i] = a_.i16[((imm8 >> ((i - 4) * 2)) & 3) + 4];
   }
 
   return simde__m128i_from_private(r_);
 }
 #if defined(SIMDE_X86_SSE2_NATIVE)
   #define simde_mm_shufflehi_epi16(a, imm8) _mm_shufflehi_epi16((a), (imm8))
-#elif defined(SIMDE_SHUFFLE_VECTOR_)
-  #define simde_mm_shufflehi_epi16(a, imm8) (__extension__ ({ \
-      const simde__m128i_private simde__tmp_a_ = simde__m128i_to_private(a); \
-      simde__m128i_from_private((simde__m128i_private) { .i16 = \
-        SIMDE_SHUFFLE_VECTOR_(16, 16, \
-          (simde__tmp_a_).i16, \
-          (simde__tmp_a_).i16, \
-          0, 1, 2, 3, \
-          (((imm8)     ) & 3) + 4, \
-          (((imm8) >> 2) & 3) + 4, \
-          (((imm8) >> 4) & 3) + 4, \
-          (((imm8) >> 6) & 3) + 4) }); }))
 #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE) && defined(SIMDE_STATEMENT_EXPR_)
   #define simde_mm_shufflehi_epi16(a, imm8) \
     (__extension__ ({ \
       int16x8_t simde_mm_shufflehi_epi16_a_ = simde__m128i_to_neon_i16(a); \
       int16x8_t simde_mm_shufflehi_epi16_r_ = simde_mm_shufflehi_epi16_a_; \
       simde_mm_shufflehi_epi16_r_ = vsetq_lane_s16(vgetq_lane_s16(simde_mm_shufflehi_epi16_a_, (((imm8)     ) & 0x3) + 4), simde_mm_shufflehi_epi16_r_, 4); \
       simde_mm_shufflehi_epi16_r_ = vsetq_lane_s16(vgetq_lane_s16(simde_mm_shufflehi_epi16_a_, (((imm8) >> 2) & 0x3) + 4), simde_mm_shufflehi_epi16_r_, 5); \
       simde_mm_shufflehi_epi16_r_ = vsetq_lane_s16(vgetq_lane_s16(simde_mm_shufflehi_epi16_a_, (((imm8) >> 4) & 0x3) + 4), simde_mm_shufflehi_epi16_r_, 6); \
       simde_mm_shufflehi_epi16_r_ = vsetq_lane_s16(vgetq_lane_s16(simde_mm_shufflehi_epi16_a_, (((imm8) >> 6) & 0x3) + 4), simde_mm_shufflehi_epi16_r_, 7); \
       simde__m128i_from_neon_i16(simde_mm_shufflehi_epi16_r_); \
     }))
+#elif defined(SIMDE_WASM_SIMD128_NATIVE)
+  #define simde_mm_shufflehi_epi16(a, imm8) (__extension__ ({ \
+      const simde__m128i_private simde_tmp_a_ = simde__m128i_to_private(a); \
+      simde__m128i_from_private((simde__m128i_private) { .wasm_v128 = \
+        wasm_i16x8_shuffle( \
+          (simde_tmp_a_).wasm_v128, \
+          (simde_tmp_a_).wasm_v128, \
+          0, 1, 2, 3, \
+          (((imm8)     ) & 3) + 4, \
+          (((imm8) >> 2) & 3) + 4, \
+          (((imm8) >> 4) & 3) + 4, \
+          (((imm8) >> 6) & 3) + 4) }); }))
+#elif defined(SIMDE_SHUFFLE_VECTOR_)
+  #define simde_mm_shufflehi_epi16(a, imm8) (__extension__ ({ \
+      const simde__m128i_private simde_tmp_a_ = simde__m128i_to_private(a); \
+      simde__m128i_from_private((simde__m128i_private) { .i16 = \
+        SIMDE_SHUFFLE_VECTOR_(16, 16, \
+          (simde_tmp_a_).i16, \
+          (simde_tmp_a_).i16, \
+          0, 1, 2, 3, \
+          (((imm8)     ) & 3) + 4, \
+          (((imm8) >> 2) & 3) + 4, \
+          (((imm8) >> 4) & 3) + 4, \
+          (((imm8) >> 6) & 3) + 4) }); }))
 #endif
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_shufflehi_epi16(a, imm8) simde_mm_shufflehi_epi16(a, imm8)
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
@@ -20558,37 +21108,48 @@
     r_.i16[i] = a_.i16[i];
   }
 
   return simde__m128i_from_private(r_);
 }
 #if defined(SIMDE_X86_SSE2_NATIVE)
   #define simde_mm_shufflelo_epi16(a, imm8) _mm_shufflelo_epi16((a), (imm8))
-#elif defined(SIMDE_SHUFFLE_VECTOR_)
-  #define simde_mm_shufflelo_epi16(a, imm8) (__extension__ ({ \
-      const simde__m128i_private simde__tmp_a_ = simde__m128i_to_private(a); \
-      simde__m128i_from_private((simde__m128i_private) { .i16 = \
-        SIMDE_SHUFFLE_VECTOR_(16, 16, \
-          (simde__tmp_a_).i16, \
-          (simde__tmp_a_).i16, \
-          (((imm8)     ) & 3), \
-          (((imm8) >> 2) & 3), \
-          (((imm8) >> 4) & 3), \
-          (((imm8) >> 6) & 3), \
-          4, 5, 6, 7) }); }))
+#elif defined(SIMDE_WASM_SIMD128_NATIVE)
+  #define simde_mm_shufflelo_epi16(a, imm8) \
+    simde__m128i_from_wasm_v128(            \
+      wasm_i16x8_shuffle(                   \
+        simde__m128i_to_wasm_v128((a)),     \
+        wasm_i16x8_splat(0),                \
+        (((imm8) & 0x03)     ),             \
+        (((imm8) & 0x0c) >> 2),             \
+        (((imm8) & 0x30) >> 4),             \
+        (((imm8) & 0xc0) >> 6),             \
+        4, 5, 6, 7))
 #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE) && defined(SIMDE_STATEMENT_EXPR_)
   #define simde_mm_shufflelo_epi16(a, imm8) \
     (__extension__({ \
       int16x8_t simde_mm_shufflelo_epi16_a_ = simde__m128i_to_neon_i16(a); \
       int16x8_t simde_mm_shufflelo_epi16_r_ = simde_mm_shufflelo_epi16_a_; \
       simde_mm_shufflelo_epi16_r_ = vsetq_lane_s16(vgetq_lane_s16(simde_mm_shufflelo_epi16_a_, (((imm8)     ) & 0x3)), simde_mm_shufflelo_epi16_r_, 0); \
       simde_mm_shufflelo_epi16_r_ = vsetq_lane_s16(vgetq_lane_s16(simde_mm_shufflelo_epi16_a_, (((imm8) >> 2) & 0x3)), simde_mm_shufflelo_epi16_r_, 1); \
       simde_mm_shufflelo_epi16_r_ = vsetq_lane_s16(vgetq_lane_s16(simde_mm_shufflelo_epi16_a_, (((imm8) >> 4) & 0x3)), simde_mm_shufflelo_epi16_r_, 2); \
       simde_mm_shufflelo_epi16_r_ = vsetq_lane_s16(vgetq_lane_s16(simde_mm_shufflelo_epi16_a_, (((imm8) >> 6) & 0x3)), simde_mm_shufflelo_epi16_r_, 3); \
       simde__m128i_from_neon_i16(simde_mm_shufflelo_epi16_r_); \
     }))
+#elif defined(SIMDE_SHUFFLE_VECTOR_)
+  #define simde_mm_shufflelo_epi16(a, imm8) (__extension__ ({ \
+      const simde__m128i_private simde_tmp_a_ = simde__m128i_to_private(a); \
+      simde__m128i_from_private((simde__m128i_private) { .i16 = \
+        SIMDE_SHUFFLE_VECTOR_(16, 16, \
+          (simde_tmp_a_).i16, \
+          (simde_tmp_a_).i16, \
+          (((imm8)     ) & 3), \
+          (((imm8) >> 2) & 3), \
+          (((imm8) >> 4) & 3), \
+          (((imm8) >> 6) & 3), \
+          4, 5, 6, 7) }); }))
 #endif
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_shufflelo_epi16(a, imm8) simde_mm_shufflelo_epi16(a, imm8)
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 simde__m128i
@@ -20672,15 +21233,15 @@
     if (count_.u64[0] > 63)
       return simde_mm_setzero_si128();
 
     const int_fast16_t s = HEDLEY_STATIC_CAST(int_fast16_t, count_.u64[0]);
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_u64 = vshlq_u64(a_.neon_u64, vdupq_n_s64(HEDLEY_STATIC_CAST(int64_t, s)));
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
-      r_.wasm_v128 = (s < 64) ? wasm_i64x2_shl(a_.wasm_v128, s) : wasm_i64x2_const(0,0);
+      r_.wasm_v128 = (s < 64) ? wasm_i64x2_shl(a_.wasm_v128, HEDLEY_STATIC_CAST(uint32_t, s)) : wasm_i64x2_const(0,0);
     #else
       #if !defined(SIMDE_BUG_GCC_94488)
         SIMDE_VECTORIZE
       #endif
       for (size_t i = 0 ; i < (sizeof(r_.u64) / sizeof(r_.u64[0])) ; i++) {
         r_.u64[i] = a_.u64[i] << s;
       }
@@ -20795,15 +21356,15 @@
       count_ = simde__m128i_to_private(count);
 
     const int cnt = HEDLEY_STATIC_CAST(int, (count_.i64[0] > 32 ? 32 : count_.i64[0]));
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_u32 = vshlq_u32(a_.neon_u32, vdupq_n_s32(HEDLEY_STATIC_CAST(int32_t, -cnt)));
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
-      r_.wasm_v128 = wasm_u32x4_shr(a_.wasm_v128, cnt);
+      r_.wasm_v128 = wasm_u32x4_shr(a_.wasm_v128, HEDLEY_STATIC_CAST(uint32_t, cnt));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.u32) / sizeof(r_.u32[0])) ; i++) {
         r_.u32[i] = a_.u32[i] >> cnt;
       }
     #endif
 
@@ -20826,15 +21387,15 @@
       count_ = simde__m128i_to_private(count);
 
     const int cnt = HEDLEY_STATIC_CAST(int, (count_.i64[0] > 64 ? 64 : count_.i64[0]));
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_u64 = vshlq_u64(a_.neon_u64, vdupq_n_s64(HEDLEY_STATIC_CAST(int64_t, -cnt)));
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
-      r_.wasm_v128 = wasm_u64x2_shr(a_.wasm_v128, cnt);
+      r_.wasm_v128 = wasm_u64x2_shr(a_.wasm_v128, HEDLEY_STATIC_CAST(uint32_t, cnt));
     #else
       #if !defined(SIMDE_BUG_GCC_94488)
         SIMDE_VECTORIZE
       #endif
       for (size_t i = 0 ; i < (sizeof(r_.u64) / sizeof(r_.u64[0])) ; i++) {
         r_.u64[i] = a_.u64[i] >> cnt;
       }
@@ -20857,15 +21418,15 @@
     a_ = simde__m128i_to_private(a);
 
   const int cnt = (imm8 & ~15) ? 15 : imm8;
 
   #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
     r_.neon_i16 = vshlq_s16(a_.neon_i16, vdupq_n_s16(HEDLEY_STATIC_CAST(int16_t, -cnt)));
   #elif defined(SIMDE_WASM_SIMD128_NATIVE)
-    r_.wasm_v128 = wasm_i16x8_shr(a_.wasm_v128, cnt);
+    r_.wasm_v128 = wasm_i16x8_shr(a_.wasm_v128, HEDLEY_STATIC_CAST(uint32_t, cnt));
   #else
     SIMDE_VECTORIZE
     for (size_t i = 0 ; i < (sizeof(r_) / sizeof(r_.i16[0])) ; i++) {
       r_.i16[i] = a_.i16[i] >> cnt;
     }
   #endif
 
@@ -20888,15 +21449,15 @@
     a_ = simde__m128i_to_private(a);
 
   const int cnt = (imm8 & ~31) ? 31 : imm8;
 
   #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
     r_.neon_i32 = vshlq_s32(a_.neon_i32, vdupq_n_s32(-cnt));
   #elif defined(SIMDE_WASM_SIMD128_NATIVE)
-    r_.wasm_v128 = wasm_i32x4_shr(a_.wasm_v128, cnt);
+    r_.wasm_v128 = wasm_i32x4_shr(a_.wasm_v128, HEDLEY_STATIC_CAST(uint32_t, cnt));
   #else
     SIMDE_VECTORIZE
     for (size_t i = 0 ; i < (sizeof(r_) / sizeof(r_.i32[0])) ; i++) {
       r_.i32[i] = a_.i32[i] >> cnt;
     }
   #endif
 
@@ -20921,15 +21482,15 @@
       count_ = simde__m128i_to_private(count);
 
     const int cnt = HEDLEY_STATIC_CAST(int, (count_.i64[0] > 15 ? 15 : count_.i64[0]));
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i16 = vshlq_s16(a_.neon_i16, vdupq_n_s16(HEDLEY_STATIC_CAST(int16_t, -cnt)));
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
-      r_.wasm_v128 = wasm_i16x8_shr(a_.wasm_v128, cnt);
+      r_.wasm_v128 = wasm_i16x8_shr(a_.wasm_v128, HEDLEY_STATIC_CAST(uint32_t, cnt));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i16) / sizeof(r_.i16[0])) ; i++) {
         r_.i16[i] = a_.i16[i] >> cnt;
       }
     #endif
 
@@ -20952,15 +21513,15 @@
       count_ = simde__m128i_to_private(count);
 
     const int cnt = count_.u64[0] > 31 ? 31 : HEDLEY_STATIC_CAST(int, count_.u64[0]);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vshlq_s32(a_.neon_i32, vdupq_n_s32(HEDLEY_STATIC_CAST(int32_t, -cnt)));
     #elif defined(SIMDE_WASM_SIMD128_NATIVE)
-      r_.wasm_v128 = wasm_i32x4_shr(a_.wasm_v128, cnt);
+      r_.wasm_v128 = wasm_i32x4_shr(a_.wasm_v128, HEDLEY_STATIC_CAST(uint32_t, cnt));
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i32) / sizeof(r_.i32[0])) ; i++) {
         r_.i32[i] = a_.i32[i] >> cnt;
       }
     #endif
 
@@ -21302,14 +21863,16 @@
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       const simde_float64 v = vgetq_lane_f64(a_.neon_f64, 0);
       simde_memcpy(mem_addr, &v, sizeof(v));
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       const int64_t v = vgetq_lane_s64(a_.neon_i64, 0);
       simde_memcpy(HEDLEY_REINTERPRET_CAST(int64_t*, mem_addr), &v, sizeof(v));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      wasm_v128_store64_lane(HEDLEY_REINTERPRET_CAST(void*, mem_addr), a_.wasm_v128, 0);
     #else
       simde_float64 v = a_.f64[0];
       simde_memcpy(mem_addr, &v, sizeof(simde_float64));
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
@@ -21341,14 +21904,16 @@
   #if defined(SIMDE_X86_SSE2_NATIVE)
     _mm_storeh_pd(mem_addr, a);
   #else
     simde__m128d_private a_ = simde__m128d_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       *mem_addr = vgetq_lane_f64(a_.neon_f64, 1);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+       wasm_v128_store64_lane(HEDLEY_REINTERPRET_CAST(void*, mem_addr), a_.wasm_v128, 1);
     #else
       *mem_addr = a_.f64[1];
     #endif
   #endif
 }
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_storeh_pd(mem_addr, a) simde_mm_storeh_pd(HEDLEY_REINTERPRET_CAST(double*, mem_addr), a)
@@ -21385,14 +21950,16 @@
 #endif
 
 SIMDE_FUNCTION_ATTRIBUTES
 void
 simde_mm_storel_pd (simde_float64* mem_addr, simde__m128d a) {
   #if defined(SIMDE_X86_SSE2_NATIVE)
     _mm_storel_pd(mem_addr, a);
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    wasm_v128_store64_lane(HEDLEY_REINTERPRET_CAST(void*, mem_addr), simde__m128d_to_wasm_v128(a), 0);
   #else
     simde__m128d_private a_ = simde__m128d_to_private(a);
 
     simde_float64 tmp;
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       tmp = vgetq_lane_f64(a_.neon_f64, 0);
     #else
@@ -21411,14 +21978,17 @@
   #if defined(SIMDE_X86_SSE2_NATIVE)
     _mm_storer_pd(mem_addr, a);
   #else
     simde__m128d_private a_ = simde__m128d_to_private(a);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       vst1q_s64(HEDLEY_REINTERPRET_CAST(int64_t*, mem_addr), vextq_s64(a_.neon_i64, a_.neon_i64, 1));
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      a_.wasm_v128 = wasm_i64x2_shuffle(a_.wasm_v128, a_.wasm_v128, 1, 0);
+      simde_mm_store_pd(mem_addr, simde__m128d_from_private(a_));
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       a_.f64 = SIMDE_SHUFFLE_VECTOR_(64, 16, a_.f64, a_.f64, 1, 0);
       simde_mm_store_pd(mem_addr, simde__m128d_from_private(a_));
     #else
       mem_addr[0] = a_.f64[1];
       mem_addr[1] = a_.f64[0];
     #endif
@@ -21477,14 +22047,16 @@
 void
 simde_mm_storeu_si32 (void* mem_addr, simde__m128i a) {
   #if defined(SIMDE_X86_SSE2_NATIVE) && ( \
       SIMDE_DETECT_CLANG_VERSION_CHECK(8,0,0) || \
       HEDLEY_GCC_VERSION_CHECK(11,0,0) || \
       HEDLEY_INTEL_VERSION_CHECK(20,21,1))
     _mm_storeu_si32(mem_addr, a);
+  #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+    wasm_v128_store32_lane(mem_addr, simde__m128i_to_wasm_v128(a), 0);
   #else
     int32_t val = simde_mm_cvtsi128_si32(a);
     simde_memcpy(mem_addr, &val, sizeof(val));
   #endif
 }
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_storeu_si32(mem_addr, a) simde_mm_storeu_si32(mem_addr, a)
@@ -21570,14 +22142,16 @@
     simde__m128i_private
       r_,
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i8 = vsubq_s8(a_.neon_i8, b_.neon_i8);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i8x16_sub(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i8 = a_.i8 - b_.i8;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i8) / sizeof(r_.i8[0])) ; i++) {
         r_.i8[i] = a_.i8[i] - b_.i8[i];
       }
@@ -21599,14 +22173,16 @@
     simde__m128i_private
       r_,
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i16 = vsubq_s16(a_.neon_i16, b_.neon_i16);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i16x8_sub(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i16 = a_.i16 - b_.i16;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i16) / sizeof(r_.i16[0])) ; i++) {
         r_.i16[i] = a_.i16[i] - b_.i16[i];
       }
@@ -21628,14 +22204,16 @@
     simde__m128i_private
       r_,
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = vsubq_s32(a_.neon_i32, b_.neon_i32);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i32x4_sub(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32 = a_.i32 - b_.i32;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i32) / sizeof(r_.i32[0])) ; i++) {
         r_.i32[i] = a_.i32[i] - b_.i32[i];
       }
@@ -21657,14 +22235,16 @@
     simde__m128i_private
       r_,
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i64 = vsubq_s64(a_.neon_i64, b_.neon_i64);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i64x2_sub(a_.wasm_v128, b_.wasm_v128);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i64 = a_.i64 - b_.i64;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i64) / sizeof(r_.i64[0])) ; i++) {
         r_.i64[i] = a_.i64[i] - b_.i64[i];
       }
@@ -22113,23 +22693,14 @@
     return r;
   #endif
 }
 #if defined(SIMDE_X86_SSE2_ENABLE_NATIVE_ALIASES)
   #define _mm_ucomineq_sd(a, b) simde_mm_ucomineq_sd(a, b)
 #endif
 
-#if defined(SIMDE_DIAGNOSTIC_DISABLE_UNINITIALIZED_)
-  HEDLEY_DIAGNOSTIC_PUSH
-  SIMDE_DIAGNOSTIC_DISABLE_UNINITIALIZED_
-#endif
-
-#if defined(SIMDE_DIAGNOSTIC_DISABLE_UNINITIALIZED_)
-  HEDLEY_DIAGNOSTIC_POP
-#endif
-
 SIMDE_FUNCTION_ATTRIBUTES
 void
 simde_mm_lfence (void) {
   #if defined(SIMDE_X86_SSE2_NATIVE)
     _mm_lfence();
   #else
     simde_mm_sfence();
@@ -22166,14 +22737,16 @@
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_i8 = vzip2q_s8(a_.neon_i8, b_.neon_i8);
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       int8x8_t a1 = vreinterpret_s8_s16(vget_high_s16(a_.neon_i16));
       int8x8_t b1 = vreinterpret_s8_s16(vget_high_s16(b_.neon_i16));
       int8x8x2_t result = vzip_s8(a1, b1);
       r_.neon_i8 = vcombine_s8(result.val[0], result.val[1]);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i8x16_shuffle(a_.wasm_v128, b_.wasm_v128, 8, 24, 9, 25, 10, 26, 11, 27, 12, 28, 13, 29, 14, 30, 15, 31);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.i8 = SIMDE_SHUFFLE_VECTOR_(8, 16, a_.i8, b_.i8, 8, 24, 9, 25, 10, 26, 11, 27, 12, 28, 13, 29, 14, 30, 15, 31);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < ((sizeof(r_) / sizeof(r_.i8[0])) / 2) ; i++) {
         r_.i8[(i * 2)]     = a_.i8[i + ((sizeof(r_) / sizeof(r_.i8[0])) / 2)];
         r_.i8[(i * 2) + 1] = b_.i8[i + ((sizeof(r_) / sizeof(r_.i8[0])) / 2)];
@@ -22201,14 +22774,16 @@
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_i16 = vzip2q_s16(a_.neon_i16, b_.neon_i16);
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       int16x4_t a1 = vget_high_s16(a_.neon_i16);
       int16x4_t b1 = vget_high_s16(b_.neon_i16);
       int16x4x2_t result = vzip_s16(a1, b1);
       r_.neon_i16 = vcombine_s16(result.val[0], result.val[1]);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i16x8_shuffle(a_.wasm_v128, b_.wasm_v128, 4, 12, 5, 13, 6, 14, 7, 15);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.i16 = SIMDE_SHUFFLE_VECTOR_(16, 16, a_.i16, b_.i16, 4, 12, 5, 13, 6, 14, 7, 15);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < ((sizeof(r_) / sizeof(r_.i16[0])) / 2) ; i++) {
         r_.i16[(i * 2)]     = a_.i16[i + ((sizeof(r_) / sizeof(r_.i16[0])) / 2)];
         r_.i16[(i * 2) + 1] = b_.i16[i + ((sizeof(r_) / sizeof(r_.i16[0])) / 2)];
@@ -22236,14 +22811,16 @@
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_i32 = vzip2q_s32(a_.neon_i32, b_.neon_i32);
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       int32x2_t a1 = vget_high_s32(a_.neon_i32);
       int32x2_t b1 = vget_high_s32(b_.neon_i32);
       int32x2x2_t result = vzip_s32(a1, b1);
       r_.neon_i32 = vcombine_s32(result.val[0], result.val[1]);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i32x4_shuffle(a_.wasm_v128, b_.wasm_v128, 2, 6, 3, 7);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.i32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.i32, b_.i32, 2, 6, 3, 7);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < ((sizeof(r_) / sizeof(r_.i32[0])) / 2) ; i++) {
         r_.i32[(i * 2)]     = a_.i32[i + ((sizeof(r_) / sizeof(r_.i32[0])) / 2)];
         r_.i32[(i * 2) + 1] = b_.i32[i + ((sizeof(r_) / sizeof(r_.i32[0])) / 2)];
@@ -22268,14 +22845,16 @@
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       int64x1_t a_h = vget_high_s64(a_.neon_i64);
       int64x1_t b_h = vget_high_s64(b_.neon_i64);
       r_.neon_i64 = vcombine_s64(a_h, b_h);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i64x2_shuffle(a_.wasm_v128, b_.wasm_v128, 1, 3);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.i64 = SIMDE_SHUFFLE_VECTOR_(64, 16, a_.i64, b_.i64, 1, 3);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < ((sizeof(r_) / sizeof(r_.i64[0])) / 2) ; i++) {
         r_.i64[(i * 2)]     = a_.i64[i + ((sizeof(r_) / sizeof(r_.i64[0])) / 2)];
         r_.i64[(i * 2) + 1] = b_.i64[i + ((sizeof(r_) / sizeof(r_.i64[0])) / 2)];
@@ -22335,14 +22914,16 @@
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_i8 = vzip1q_s8(a_.neon_i8, b_.neon_i8);
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       int8x8_t a1 = vreinterpret_s8_s16(vget_low_s16(a_.neon_i16));
       int8x8_t b1 = vreinterpret_s8_s16(vget_low_s16(b_.neon_i16));
       int8x8x2_t result = vzip_s8(a1, b1);
       r_.neon_i8 = vcombine_s8(result.val[0], result.val[1]);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i8x16_shuffle(a_.wasm_v128, b_.wasm_v128, 0, 16, 1, 17, 2, 18, 3, 19, 4, 20, 5, 21, 6, 22, 7, 23);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.i8 = SIMDE_SHUFFLE_VECTOR_(8, 16, a_.i8, b_.i8, 0, 16, 1, 17, 2, 18, 3, 19, 4, 20, 5, 21, 6, 22, 7, 23);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < ((sizeof(r_) / sizeof(r_.i8[0])) / 2) ; i++) {
         r_.i8[(i * 2)]     = a_.i8[i];
         r_.i8[(i * 2) + 1] = b_.i8[i];
@@ -22370,14 +22951,16 @@
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_i16 = vzip1q_s16(a_.neon_i16, b_.neon_i16);
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       int16x4_t a1 = vget_low_s16(a_.neon_i16);
       int16x4_t b1 = vget_low_s16(b_.neon_i16);
       int16x4x2_t result = vzip_s16(a1, b1);
       r_.neon_i16 = vcombine_s16(result.val[0], result.val[1]);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i16x8_shuffle(a_.wasm_v128, b_.wasm_v128, 0, 8, 1, 9, 2, 10, 3, 11);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.i16 = SIMDE_SHUFFLE_VECTOR_(16, 16, a_.i16, b_.i16, 0, 8, 1, 9, 2, 10, 3, 11);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < ((sizeof(r_) / sizeof(r_.i16[0])) / 2) ; i++) {
         r_.i16[(i * 2)]     = a_.i16[i];
         r_.i16[(i * 2) + 1] = b_.i16[i];
@@ -22405,14 +22988,16 @@
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_i32 = vzip1q_s32(a_.neon_i32, b_.neon_i32);
     #elif defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       int32x2_t a1 = vget_low_s32(a_.neon_i32);
       int32x2_t b1 = vget_low_s32(b_.neon_i32);
       int32x2x2_t result = vzip_s32(a1, b1);
       r_.neon_i32 = vcombine_s32(result.val[0], result.val[1]);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i32x4_shuffle(a_.wasm_v128, b_.wasm_v128, 0, 4, 1, 5);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.i32 = SIMDE_SHUFFLE_VECTOR_(32, 16, a_.i32, b_.i32, 0, 4, 1, 5);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < ((sizeof(r_) / sizeof(r_.i32[0])) / 2) ; i++) {
         r_.i32[(i * 2)]     = a_.i32[i];
         r_.i32[(i * 2) + 1] = b_.i32[i];
@@ -22437,14 +23022,16 @@
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       int64x1_t a_l = vget_low_s64(a_.neon_i64);
       int64x1_t b_l = vget_low_s64(b_.neon_i64);
       r_.neon_i64 = vcombine_s64(a_l, b_l);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i64x2_shuffle(a_.wasm_v128, b_.wasm_v128, 0, 2);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.i64 = SIMDE_SHUFFLE_VECTOR_(64, 16, a_.i64, b_.i64, 0, 2);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < ((sizeof(r_) / sizeof(r_.i64[0])) / 2) ; i++) {
         r_.i64[(i * 2)]     = a_.i64[i];
         r_.i64[(i * 2) + 1] = b_.i64[i];
@@ -22467,14 +23054,16 @@
     simde__m128d_private
       r_,
       a_ = simde__m128d_to_private(a),
       b_ = simde__m128d_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A64V8_NATIVE)
       r_.neon_f64 = vzip1q_f64(a_.neon_f64, b_.neon_f64);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_i64x2_shuffle(a_.wasm_v128, b_.wasm_v128, 0, 2);
     #elif defined(SIMDE_SHUFFLE_VECTOR_)
       r_.f64 = SIMDE_SHUFFLE_VECTOR_(64, 16, a_.f64, b_.f64, 0, 2);
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < ((sizeof(r_) / sizeof(r_.f64[0])) / 2) ; i++) {
         r_.f64[(i * 2)]     = a_.f64[i];
         r_.f64[(i * 2) + 1] = b_.f64[i];
@@ -22529,14 +23118,16 @@
       a_ = simde__m128i_to_private(a),
       b_ = simde__m128i_to_private(b);
 
     #if defined(SIMDE_ARM_NEON_A32V7_NATIVE)
       r_.neon_i32 = veorq_s32(a_.neon_i32, b_.neon_i32);
     #elif defined(SIMDE_POWER_ALTIVEC_P6_NATIVE)
       r_.altivec_i32 = vec_xor(a_.altivec_i32, b_.altivec_i32);
+    #elif defined(SIMDE_WASM_SIMD128_NATIVE)
+      r_.wasm_v128 = wasm_v128_xor(b_.wasm_v128, a_.wasm_v128);
     #elif defined(SIMDE_VECTOR_SUBSCRIPT_OPS)
       r_.i32f = a_.i32f ^ b_.i32f;
     #else
       SIMDE_VECTORIZE
       for (size_t i = 0 ; i < (sizeof(r_.i32f) / sizeof(r_.i32f[0])) ; i++) {
         r_.i32f[i] = a_.i32f[i] ^ b_.i32f[i];
       }
```

### Comparing `scikit-bio-0.5.8/skbio/__init__.py` & `scikit-bio-0.5.9/skbio/__init__.py`

 * *Files 1% similar despite different names*

```diff
@@ -20,15 +20,16 @@
 import skbio.stats.evolve  # noqa
 
 __all__ = ['Sequence', 'DNA', 'RNA', 'Protein', 'GeneticCode',
            'DistanceMatrix', 'local_pairwise_align_ssw', 'TabularMSA',
            'TreeNode', 'nj', 'read', 'write', 'OrdinationResults']
 
 __credits__ = "https://github.com/biocore/scikit-bio/graphs/contributors"
-__version__ = "0.5.8"
+__version__ = "0.5.9"
+
 
 mottos = [
     # 03/15/2014
     "It's gonna get weird, bro.",
     # 05/14/2014
     "no cog yay",
     # 03/18/2015
```

### Comparing `scikit-bio-0.5.8/skbio/_base.py` & `scikit-bio-0.5.9/skbio/_base.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/alignment/__init__.py` & `scikit-bio-0.5.9/skbio/alignment/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/alignment/_indexing.py` & `scikit-bio-0.5.9/skbio/alignment/_indexing.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/alignment/_lib/ssw.c` & `scikit-bio-0.5.9/skbio/alignment/_lib/ssw.c`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/alignment/_lib/ssw.h` & `scikit-bio-0.5.9/skbio/alignment/_lib/ssw.h`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/alignment/_pairwise.py` & `scikit-bio-0.5.9/skbio/alignment/_pairwise.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/alignment/_repr.py` & `scikit-bio-0.5.9/skbio/alignment/_repr.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/alignment/_ssw_wrapper.c` & `scikit-bio-0.5.9/skbio/alignment/_ssw_wrapper.c`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/alignment/_ssw_wrapper.pyx` & `scikit-bio-0.5.9/skbio/alignment/_ssw_wrapper.pyx`

 * *Files 0% similar despite different names*

```diff
@@ -577,15 +577,15 @@
                                 " protein sequences")
             matrix = self._build_match_matrix(match_score, mismatch_score)
         else:
             matrix = self._convert_dict2d_to_matrix(substitution_matrix)
         # Set up our mask_length
         # Mask is recommended to be max(query_sequence/2, 15)
         if mask_auto:
-            self.mask_length = len(query_sequence) / 2
+            self.mask_length = len(query_sequence) // 2
             if self.mask_length < mask_length:
                 self.mask_length = mask_length
         else:
             self.mask_length = mask_length
 
         cdef cnp.ndarray[cnp.int8_t, ndim = 1, mode = "c"] read_seq
         read_seq = self._seq_converter(query_sequence)
```

### Comparing `scikit-bio-0.5.8/skbio/alignment/_tabular_msa.py` & `scikit-bio-0.5.9/skbio/alignment/_tabular_msa.py`

 * *Files 1% similar despite different names*

```diff
@@ -1979,16 +1979,17 @@
             # adjacent RangeIndexes in the future (pandas 0.18.0 creates an
             # Int64Index).
             index = pd.RangeIndex(start=len(self),
                                   stop=len(self) + len(sequences),
                                   step=1)
 
         if len(self):
-            self._seqs = self._seqs.append(pd.Series(sequences, index=index,
-                                                     dtype=object))
+            self._seqs = pd.concat([self._seqs,
+                                    pd.Series(sequences, index=index,
+                                              dtype=object)])
         else:
             # Not using Series.append to avoid turning a RangeIndex supplied
             # via `index` parameter into an Int64Index (this happens in pandas
             # 0.18.0).
             self._seqs = pd.Series(sequences, index=index, dtype=object)
 
             # When extending a TabularMSA without sequences, the number of
@@ -2411,18 +2412,16 @@
         >>> msa.index
         Index(['a', 'b', 'c'], dtype='object')
 
         Note that since the sort is in-place, the ``TabularMSA`` object is
         modified (a new object is *not* returned).
 
         """
-        series = self._seqs.sort_index(ascending=ascending, level=level)
-        self._seqs = series
-        positional_metadata = self.positional_metadata.sort_index(axis=1)
-        self.positional_metadata = positional_metadata
+        self._seqs.sort_index(ascending=ascending, level=level, inplace=True)
+        self.positional_metadata.sort_index(axis=1, inplace=True)
 
     @experimental(as_of='0.4.1')
     def to_dict(self):
         """Create a ``dict`` from this ``TabularMSA``.
 
         Returns
         -------
```

### Comparing `scikit-bio-0.5.8/skbio/alignment/tests/test_pairwise.py` & `scikit-bio-0.5.9/skbio/alignment/tests/test_pairwise.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/alignment/tests/test_ssw.py` & `scikit-bio-0.5.9/skbio/alignment/tests/test_ssw.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/alignment/tests/test_tabular_msa.py` & `scikit-bio-0.5.9/skbio/alignment/tests/test_tabular_msa.py`

 * *Files 0% similar despite different names*

```diff
@@ -135,15 +135,15 @@
     def test_constructor_invalid_index_scalar(self):
         with self.assertRaises(TypeError):
             TabularMSA([DNA('ACGT'), DNA('TGCA')], index=42)
 
     def test_constructor_non_unique_labels(self):
         msa = TabularMSA([DNA('ACGT'), DNA('ACGT')], index=[1, 1])
 
-        assert_index_equal(msa.index, pd.Int64Index([1, 1]))
+        assert_index_equal(msa.index, pd.Index([1, 1], dtype=np.int64))
 
     def test_constructor_empty_no_index(self):
         # sequence empty
         msa = TabularMSA([])
         self.assertIsNone(msa.dtype)
         self.assertEqual(msa.shape, (0, 0))
         assert_index_equal(msa.index, pd.RangeIndex(0))
@@ -545,15 +545,15 @@
     def test_eq_default_index_and_equivalent_provided_index(self):
         msa1 = TabularMSA([DNA('ACGT'), DNA('----'), DNA('....')])
         msa2 = TabularMSA([DNA('ACGT'), DNA('----'), DNA('....')],
                           index=[0, 1, 2])
 
         self.assertReallyEqual(msa1, msa2)
         assert_index_equal(msa1.index, pd.RangeIndex(3))
-        assert_index_equal(msa2.index, pd.Int64Index([0, 1, 2]))
+        assert_index_equal(msa2.index, pd.Index([0, 1, 2], dtype=np.int64))
 
     def test_reassign_index_empty(self):
         # sequence empty
         msa = TabularMSA([])
         msa.reassign_index()
         self.assertEqual(msa, TabularMSA([]))
         assert_index_equal(msa.index, pd.RangeIndex(0))
@@ -747,23 +747,22 @@
             DNA('TCCG', metadata={'id': 10}),
             DNA('TAGG', metadata={'id': 10}),
             DNA('GGGG', metadata={'id': 8}),
             DNA('TGGG', metadata={'id': 10}),
             DNA('ACGT', metadata={'id': 0}),
             DNA('TAGA', metadata={'id': 10})], minter='id')
         msa.sort()
-        self.assertEqual(
-            msa,
-            TabularMSA([
-                DNA('ACGT', metadata={'id': 0}),
-                DNA('GGGG', metadata={'id': 8}),
-                DNA('TCCG', metadata={'id': 10}),
-                DNA('TAGG', metadata={'id': 10}),
-                DNA('TGGG', metadata={'id': 10}),
-                DNA('TAGA', metadata={'id': 10})], minter='id'))
+        self.assertEqual(msa._seqs.index.to_list(), [0, 8, 10, 10, 10, 10])
+        vals = list(msa._seqs.values)
+        self.assertEqual(vals[0], DNA('ACGT', metadata={'id': 0}))
+        self.assertEqual(vals[1], DNA('GGGG', metadata={'id': 8}))
+        self.assertIn(DNA('TCCG', metadata={'id': 10}), vals)
+        self.assertIn(DNA('TAGG', metadata={'id': 10}), vals[2:])
+        self.assertIn(DNA('TGGG', metadata={'id': 10}), vals[2:])
+        self.assertIn(DNA('TAGA', metadata={'id': 10}), vals[2:])
 
     def test_sort_on_key_with_all_repeats(self):
         msa = TabularMSA([
             DNA('TTT', metadata={'id': 'a'}),
             DNA('TTT', metadata={'id': 'b'}),
             DNA('TTT', metadata={'id': 'c'})], minter=str)
         msa.sort()
@@ -1356,49 +1355,14 @@
 
         self.assertEqual(self.get(msa, (('a', 'x'), Ellipsis)),
                          TabularMSA([a, b], metadata={'x': 'y'},
                                     positional_metadata={'c': ['a', 'b', 'c',
                                                                'd']},
                                     index=[0, 1]))
 
-    def test_multiindex_complicated_axis(self):
-        a = RNA("UUAG", metadata={0: 0}, positional_metadata={0: [1, 2, 3, 4]})
-        b = RNA("UAAG", metadata={1: 0}, positional_metadata={1: [1, 2, 3, 4]})
-        c = RNA("UAA-", metadata={2: 0}, positional_metadata={2: [1, 2, 3, 4]})
-        d = RNA("UA-G", metadata={3: 0}, positional_metadata={3: [1, 2, 3, 4]})
-        msa = TabularMSA([a, b, c, d], metadata={'x': 'y'},
-                         positional_metadata={'c': ['a', 'b', 'c', 'd']},
-                         index=[('a', 'x', 0), ('a', 'x', 1), ('a', 'y', 2),
-                                ('b', 'x', 0)])
-
-        self.assertEqual(self.get(msa, (([False, True, False, True],
-                                         'x', 0), Ellipsis)),
-                         TabularMSA([d], metadata={'x': 'y'},
-                                    positional_metadata={'c': ['a', 'b', 'c',
-                                                               'd']},
-                                    index=[('b', 'x', 0)]))
-
-    @unittest.skipIf(tuple(map(int, pd.__version__.split('.'))) < (1, 2, 5),
-                     "Old pandas will return empty frame")
-    def test_multiindex_complicated_axis_empty_selection(self):
-        a = RNA("UUAG", metadata={0: 0}, positional_metadata={0: [1, 2, 3, 4]})
-        b = RNA("UAAG", metadata={1: 0}, positional_metadata={1: [1, 2, 3, 4]})
-        c = RNA("UAA-", metadata={2: 0}, positional_metadata={2: [1, 2, 3, 4]})
-        d = RNA("UA-G", metadata={3: 0}, positional_metadata={3: [1, 2, 3, 4]})
-        msa = TabularMSA([a, b, c, d], metadata={'x': 'y'},
-                         positional_metadata={'c': ['a', 'b', 'c', 'd']},
-                         index=[('a', 'x', 0), ('a', 'x', 1), ('a', 'y', 2),
-                                ('b', 'x', 0)])
-        # Pandas will KeyError when the intersection is empty
-        # change appears to have happened in:
-        # https://github.com/pandas-dev/pandas/pull/42245
-        # but this was not bisected to confirm
-        with self.assertRaises(KeyError):
-            self.get(msa, (([False, True, False, True], 'x', 2), Ellipsis))
-
     def test_bool_index_scalar_bool_label(self):
         a = DNA("ACGA", metadata={0: 0}, positional_metadata={0: [1, 2, 3, 4]})
         b = DNA("A-GA", metadata={1: 1}, positional_metadata={1: [1, 2, 3, 4]})
         c = DNA("AAGA", metadata={2: 2}, positional_metadata={2: [1, 2, 3, 4]})
         d = DNA("ACCA", metadata={3: 3}, positional_metadata={3: [1, 2, 3, 4]})
 
         msa = TabularMSA([a, b, c, d], metadata={'x': 'y'},
```

### Comparing `scikit-bio-0.5.8/skbio/diversity/__init__.py` & `scikit-bio-0.5.9/skbio/diversity/__init__.py`

 * *Files 1% similar despite different names*

```diff
@@ -380,15 +380,15 @@
    >>> fig = sample_md.boxplot(column='Faith PD', by='body_site')
 
 We can also compute Spearman correlations between all pairs of columns in
 this ``DataFrame``. Since our alpha diversity metrics are the only two
 numeric columns (and thus the only columns for which Spearman correlation
 is relevant), this will give us a symmetric 2x2 correlation matrix.
 
->>> sample_md.corr(method="spearman")
+>>> sample_md.corr(method="spearman", numeric_only=True)
                Observed OTUs  Faith PD
 Observed OTUs       1.000000  0.939336
 Faith PD            0.939336  1.000000
 
 """
 
 # ----------------------------------------------------------------------------
```

### Comparing `scikit-bio-0.5.8/skbio/diversity/_block.py` & `scikit-bio-0.5.9/skbio/diversity/_block.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/_driver.py` & `scikit-bio-0.5.9/skbio/diversity/_driver.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/_phylogenetic.c` & `scikit-bio-0.5.9/skbio/diversity/_phylogenetic.c`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/_phylogenetic.pyx` & `scikit-bio-0.5.9/skbio/diversity/_phylogenetic.pyx`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/_util.py` & `scikit-bio-0.5.9/skbio/diversity/_util.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/__init__.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/_ace.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/_ace.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/_base.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/_base.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/_chao1.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/_chao1.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/_faith_pd.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/_faith_pd.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/_gini.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/_gini.py`

 * *Files 3% similar despite different names*

```diff
@@ -74,15 +74,15 @@
        E, Salvemini, T). Rome: Libreria Eredi Virgilio Veschi (1955).
 
     """
     # Suppress cast to int because this method supports ints and floats.
     data = _validate_counts_vector(data, suppress_cast=True)
     lorenz_points = _lorenz_curve(data)
     B = _lorenz_curve_integrator(lorenz_points, method)
-    return 1 - 2 * B
+    return max(0.0, 1 - 2 * B)
 
 
 def _lorenz_curve(data):
     """Calculate the Lorenz curve for input data.
 
     Notes
     -----
```

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/_lladser.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/_lladser.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_ace.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_ace.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_base.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_base.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_chao1.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_chao1.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_faith_pd.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_faith_pd.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_gini.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_gini.py`

 * *Files 12% similar despite different names*

```diff
@@ -22,14 +22,19 @@
         self.lorenz_curve_points = (
             np.array([0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1.0]),
             np.array([0.047287899860917935, 0.095966620305980521,
                       0.15855354659248957, 0.2517385257301808,
                       0.34492350486787204, 0.541029207232267,
                       0.74965229485396379, 1.0]))
 
+    def test_gini_index_bug_1844(self):
+        exp = 0.0
+        obs = gini_index([2, 2, 2, 2, 2])
+        self.assertAlmostEqual(obs, exp)
+
     def test_gini_index(self):
         exp = 0.32771210013908214
         obs = gini_index(self.data, 'trapezoids')
         self.assertAlmostEqual(obs, exp)
 
         exp = 0.20271210013908214
         obs = gini_index(self.data, 'rectangles')
```

### Comparing `scikit-bio-0.5.8/skbio/diversity/alpha/tests/test_lladser.py` & `scikit-bio-0.5.9/skbio/diversity/alpha/tests/test_lladser.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/beta/__init__.py` & `scikit-bio-0.5.9/skbio/diversity/beta/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/beta/_unifrac.py` & `scikit-bio-0.5.9/skbio/diversity/beta/_unifrac.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/README.md` & `scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/README.md`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/unweighted_unifrac_dm.txt` & `scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/unweighted_unifrac_dm.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/weighted_normalized_unifrac_dm.txt` & `scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/weighted_normalized_unifrac_dm.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/beta/tests/data/qiime-191-tt/weighted_unifrac_dm.txt` & `scikit-bio-0.5.9/skbio/diversity/beta/tests/data/qiime-191-tt/weighted_unifrac_dm.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/beta/tests/test_unifrac.py` & `scikit-bio-0.5.9/skbio/diversity/beta/tests/test_unifrac.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/tests/test_block.py` & `scikit-bio-0.5.9/skbio/diversity/tests/test_block.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/tests/test_driver.py` & `scikit-bio-0.5.9/skbio/diversity/tests/test_driver.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/diversity/tests/test_util.py` & `scikit-bio-0.5.9/skbio/diversity/tests/test_util.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/__init__.py` & `scikit-bio-0.5.9/skbio/io/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/_exception.py` & `scikit-bio-0.5.9/skbio/io/_exception.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/_fileobject.py` & `scikit-bio-0.5.9/skbio/io/_fileobject.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/_iosources.py` & `scikit-bio-0.5.9/skbio/io/_iosources.py`

 * *Files 5% similar despite different names*

```diff
@@ -9,16 +9,14 @@
 import io
 import gzip
 import bz2
 import tempfile
 import itertools
 
 import requests
-from cachecontrol import CacheControl
-from cachecontrol.caches import FileCache
 
 from skbio.io import IOSourceError
 from ._fileobject import (IterableStringWriterIO, IterableStringReaderIO,
                           WrappedBufferedRandom)
 
 
 # NamedTemporaryFile isn't an actual file class, it is a function which
@@ -100,17 +98,15 @@
 class HTTPSource(IOSource):
     def can_read(self):
         return (
             isinstance(self.file, str) and
             requests.compat.urlparse(self.file).scheme in {'http', 'https'})
 
     def get_reader(self):
-        sess = CacheControl(requests.Session(),
-                            cache=FileCache(tempfile.gettempdir()))
-        req = sess.get(self.file)
+        req = requests.get(self.file)
 
         # if the response is not 200, an exception will be raised
         req.raise_for_status()
 
         return io.BufferedReader(io.BytesIO(req.content))
```

### Comparing `scikit-bio-0.5.8/skbio/io/_warning.py` & `scikit-bio-0.5.9/skbio/io/_warning.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/_base.py` & `scikit-bio-0.5.9/skbio/io/format/_base.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/_blast.py` & `scikit-bio-0.5.9/skbio/io/format/_blast.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/_sequence_feature_vocabulary.py` & `scikit-bio-0.5.9/skbio/io/format/_sequence_feature_vocabulary.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/binary_dm.py` & `scikit-bio-0.5.9/skbio/io/format/binary_dm.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/blast6.py` & `scikit-bio-0.5.9/skbio/io/format/blast6.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/blast7.py` & `scikit-bio-0.5.9/skbio/io/format/blast7.py`

 * *Files 4% similar despite different names*

```diff
@@ -232,24 +232,19 @@
 1\\t0.0\\t1233.5'
 ... ])
 >>> fh = StringIO(fs)
 
 Read the file into a ``pd.DataFrame``:
 
 >>> df = skbio.io.read(fh, into=pd.DataFrame)
->>> df # doctest: +NORMALIZE_WHITESPACE
-     qseqid          sseqid  pident  length  mismatch  gapopen  qstart  qend \\
-0  AF178033  EMORG:AF178033  100.00   811.0       0.0      0.0     1.0  811.0
-1  AF178033  EMORG:AF178032   94.57   811.0      44.0      0.0     1.0  811.0
-2  AF178033  EMORG:AF178031   94.82   811.0      42.0      0.0     1.0  811.0
-<BLANKLINE>
-   sstart   send  evalue  bitscore
-0     1.0  811.0     0.0    1566.6
-1     1.0  811.0     0.0    1217.7
-2     1.0  811.0     0.0    1233.5
+>>> df[['qseqid', 'sseqid', 'pident']] # doctest: +NORMALIZE_WHITESPACE
+     qseqid          sseqid  pident
+0  AF178033  EMORG:AF178033  100.00
+1  AF178033  EMORG:AF178032   94.57
+2  AF178033  EMORG:AF178031   94.82
 
 References
 ==========
 .. [1] Altschul, S.F., Gish, W., Miller, W., Myers, E.W. & Lipman, D.J. (1990)
    "Basic local alignment search tool." J. Mol. Biol. 215:403-410.
 .. [2] http://www.ncbi.nlm.nih.gov/books/NBK279682/
 """
```

### Comparing `scikit-bio-0.5.8/skbio/io/format/clustal.py` & `scikit-bio-0.5.9/skbio/io/format/clustal.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/embl.py` & `scikit-bio-0.5.9/skbio/io/format/embl.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/emptyfile.py` & `scikit-bio-0.5.9/skbio/io/format/emptyfile.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/fasta.py` & `scikit-bio-0.5.9/skbio/io/format/fasta.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/fastq.py` & `scikit-bio-0.5.9/skbio/io/format/fastq.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/genbank.py` & `scikit-bio-0.5.9/skbio/io/format/genbank.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/gff3.py` & `scikit-bio-0.5.9/skbio/io/format/gff3.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/lsmat.py` & `scikit-bio-0.5.9/skbio/io/format/lsmat.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/newick.py` & `scikit-bio-0.5.9/skbio/io/format/newick.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/ordination.py` & `scikit-bio-0.5.9/skbio/io/format/ordination.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/phylip.py` & `scikit-bio-0.5.9/skbio/io/format/phylip.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/qseq.py` & `scikit-bio-0.5.9/skbio/io/format/qseq.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/stockholm.py` & `scikit-bio-0.5.9/skbio/io/format/stockholm.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/taxdump.py` & `scikit-bio-0.5.9/skbio/io/format/taxdump.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/embl_constructed` & `scikit-bio-0.5.9/skbio/io/format/tests/data/embl_constructed`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/embl_feature_level_record` & `scikit-bio-0.5.9/skbio/io/format/tests/data/embl_feature_level_record`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/embl_feature_level_record_no_FT` & `scikit-bio-0.5.9/skbio/io/format/tests/data/embl_feature_level_record_no_FT`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/embl_multi_records` & `scikit-bio-0.5.9/skbio/io/format/tests/data/embl_multi_records`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/embl_single_record` & `scikit-bio-0.5.9/skbio/io/format/tests/data/embl_single_record`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/embl_single_record_simple` & `scikit-bio-0.5.9/skbio/io/format/tests/data/embl_single_record_simple`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/embl_uniprot_record` & `scikit-bio-0.5.9/skbio/io/format/tests/data/embl_uniprot_record`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_diff_ids.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_diff_ids.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_double_qual.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_double_qual.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_double_seq.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_double_seq.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_long_qual.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_long_qual.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_del.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_del.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_escape.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_escape.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_null.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_null.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_space.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_space.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_tab.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_tab.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_unit_sep.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_unit_sep.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_qual_vtab.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_qual_vtab.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_short_qual.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_short_qual.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_spaces.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_spaces.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_tabs.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_tabs.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_at_plus.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_at_plus.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_at_qual.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_at_qual.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_at_seq.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_at_seq.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_in_plus.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_in_plus.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_in_qual.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_in_qual.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_in_seq.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_in_seq.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/error_trunc_in_title.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/error_trunc_in_title.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_wrapping_as_illumina_no_description` & `scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_wrapping_as_illumina_no_description`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_wrapping_as_sanger_no_description` & `scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_wrapping_as_sanger_no_description`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/fastq_wrapping_original_sanger_no_description` & `scikit-bio-0.5.9/skbio/io/format/tests/data/fastq_wrapping_original_sanger_no_description`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/genbank_multi_records` & `scikit-bio-0.5.9/skbio/io/format/tests/data/genbank_multi_records`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/genbank_single_record` & `scikit-bio-0.5.9/skbio/io/format/tests/data/genbank_single_record`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/gff3_multi_record` & `scikit-bio-0.5.9/skbio/io/format/tests/data/gff3_multi_record`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/longreads_as_illumina.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/longreads_as_illumina.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/longreads_as_sanger.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/longreads_as_sanger.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/longreads_original_sanger.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/longreads_original_sanger.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/misc_dna_as_illumina.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/misc_dna_as_illumina.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/misc_dna_as_sanger.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/misc_dna_as_sanger.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/misc_dna_original_sanger.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/misc_dna_original_sanger.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/misc_rna_as_illumina.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/misc_rna_as_illumina.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/misc_rna_as_sanger.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/misc_rna_as_sanger.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/misc_rna_original_sanger.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/misc_rna_original_sanger.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_PCoA_sample_data_3_scores` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_PCoA_sample_data_3_scores`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error1` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error1`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error11` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error11`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error12` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error12`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error13` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error13`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error14` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error14`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error15` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error15`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error16` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error16`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error17` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error17`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error18` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error18`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error2` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error2`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error3` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error3`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error4` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error4`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error5` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error5`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error6` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error6`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_error8` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_error8`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_example2_scores` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_example2_scores`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_example3_scores` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_example3_scores`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_site` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_site`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_site_constraints` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_site_constraints`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_species` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_CCA_species`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_PCoA_site` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_PCoA_site`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_site` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_site`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_site_constraints` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_site_constraints`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_species` & `scikit-bio-0.5.9/skbio/io/format/tests/data/ordination_exp_Ordination_RDA_species`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/sanger_full_range_as_illumina.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/sanger_full_range_as_illumina.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/sanger_full_range_as_sanger.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/sanger_full_range_as_sanger.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/sanger_full_range_original_sanger.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/sanger_full_range_original_sanger.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_extensive` & `scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_extensive`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/stockholm_extensive_mixed` & `scikit-bio-0.5.9/skbio/io/format/tests/data/stockholm_extensive_mixed`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/taxdump_names.dmp` & `scikit-bio-0.5.9/skbio/io/format/tests/data/taxdump_names.dmp`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/taxdump_nodes.dmp` & `scikit-bio-0.5.9/skbio/io/format/tests/data/taxdump_nodes.dmp`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/wrapping_as_illumina.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/wrapping_as_illumina.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/wrapping_as_sanger.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/wrapping_as_sanger.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/data/wrapping_original_sanger.fastq` & `scikit-bio-0.5.9/skbio/io/format/tests/data/wrapping_original_sanger.fastq`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_base.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_base.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_binary_dm.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_binary_dm.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_blast6.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_blast6.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_blast7.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_blast7.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_clustal.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_clustal.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_embl.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_embl.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_emptyfile.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_emptyfile.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_fasta.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_fasta.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_fastq.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_fastq.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_genbank.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_genbank.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_gff3.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_gff3.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_lsmat.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_lsmat.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_newick.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_newick.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_ordination.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_ordination.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_phylip.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_phylip.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_qseq.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_qseq.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_sequence_feature_vocabulary.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_sequence_feature_vocabulary.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_stockholm.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_stockholm.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/format/tests/test_taxdump.py` & `scikit-bio-0.5.9/skbio/io/format/tests/test_taxdump.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/registry.py` & `scikit-bio-0.5.9/skbio/io/registry.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/tests/test_iosources.py` & `scikit-bio-0.5.9/skbio/io/tests/test_iosources.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/tests/test_registry.py` & `scikit-bio-0.5.9/skbio/io/tests/test_registry.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/tests/test_util.py` & `scikit-bio-0.5.9/skbio/io/tests/test_util.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/io/util.py` & `scikit-bio-0.5.9/skbio/io/util.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/metadata/__init__.py` & `scikit-bio-0.5.9/skbio/metadata/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/metadata/_intersection.c` & `scikit-bio-0.5.9/skbio/metadata/_intersection.c`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/metadata/_intersection.pyx` & `scikit-bio-0.5.9/skbio/metadata/_intersection.pyx`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/metadata/_interval.py` & `scikit-bio-0.5.9/skbio/metadata/_interval.py`

 * *Files 0% similar despite different names*

```diff
@@ -824,14 +824,17 @@
             meet this requirement.
 
         Yields
         ------
         Interval
             ``Interval`` object satisfying the search criteria.
         """
+        if bounds is None and metadata is None:
+            metadata = {}
+
         if bounds is None:
             for intvl in self._query_attribute(metadata):
                 yield intvl
         else:
             for loc in bounds:
                 intvls = self._query_interval(loc)
                 if metadata is None:
```

### Comparing `scikit-bio-0.5.8/skbio/metadata/_mixin.py` & `scikit-bio-0.5.9/skbio/metadata/_mixin.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/metadata/_repr.py` & `scikit-bio-0.5.9/skbio/metadata/_repr.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/metadata/_testing.py` & `scikit-bio-0.5.9/skbio/metadata/_testing.py`

 * *Files 1% similar despite different names*

```diff
@@ -387,15 +387,15 @@
                           'bar': np.arange(5)[::-1]}, index=range(5)))
 
     def test_constructor_with_positional_metadata_int64_index(self):
         # Test that memory-inefficient index is converted to memory-efficient
         # index.
         df = pd.DataFrame({'foo': np.arange(5), 'bar': np.arange(5)[::-1]},
                           index=np.arange(5))
-        self.assertIsInstance(df.index, pd.Int64Index)
+        self.assertEqual(df.index.dtype, np.int64)
 
         obj = self._positional_metadata_constructor_(
             5, positional_metadata=df)
 
         assert_data_frame_almost_equal(
             obj.positional_metadata,
             pd.DataFrame({'foo': np.arange(5),
@@ -773,15 +773,15 @@
     def test_positional_metadata_setter_int64_index(self):
         # Test that memory-inefficient index is converted to memory-efficient
         # index.
         obj = self._positional_metadata_constructor_(5)
 
         df = pd.DataFrame({'foo': np.arange(5), 'bar': np.arange(5)[::-1]},
                           index=np.arange(5))
-        self.assertIsInstance(df.index, pd.Int64Index)
+        self.assertEqual(df.index.dtype, np.int64)
 
         obj.positional_metadata = df
 
         assert_data_frame_almost_equal(
             obj.positional_metadata,
             pd.DataFrame({'foo': np.arange(5),
                           'bar': np.arange(5)[::-1]}, index=range(5)))
```

### Comparing `scikit-bio-0.5.8/skbio/metadata/tests/test_intersection.py` & `scikit-bio-0.5.9/skbio/metadata/tests/test_intersection.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/metadata/tests/test_interval.py` & `scikit-bio-0.5.9/skbio/metadata/tests/test_interval.py`

 * *Files 0% similar despite different names*

```diff
@@ -661,15 +661,15 @@
         intervals = list(self.im_2.query(bounds=[(1, 5)],
                                          metadata={'gene': 'sagA'}))
         self.assertEqual(len(intervals), 1)
         self.assertEqual(intervals[0], self.im_2_1)
 
     def test_query_empty(self):
         intervals = list(self.im_1.query())
-        self.assertEqual(len(intervals), 0)
+        self.assertEqual(len(intervals), 1)
 
     def test_query_no_hits(self):
         intervals = list(self.im_2.query(bounds=[(self.upper_bound, 200)]))
         self.assertEqual(len(intervals), 0)
 
         intervals = list(self.im_2.query(metadata={'gene': 'sagC'}))
         self.assertEqual(len(intervals), 0)
```

### Comparing `scikit-bio-0.5.8/skbio/metadata/tests/test_mixin.py` & `scikit-bio-0.5.9/skbio/metadata/tests/test_mixin.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/__init__.py` & `scikit-bio-0.5.9/skbio/sequence/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/_dna.py` & `scikit-bio-0.5.9/skbio/sequence/_dna.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/_genetic_code.py` & `scikit-bio-0.5.9/skbio/sequence/_genetic_code.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/_grammared_sequence.py` & `scikit-bio-0.5.9/skbio/sequence/_grammared_sequence.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/_nucleotide_mixin.py` & `scikit-bio-0.5.9/skbio/sequence/_nucleotide_mixin.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/_protein.py` & `scikit-bio-0.5.9/skbio/sequence/_protein.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/_repr.py` & `scikit-bio-0.5.9/skbio/sequence/_repr.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/_rna.py` & `scikit-bio-0.5.9/skbio/sequence/_rna.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/_sequence.py` & `scikit-bio-0.5.9/skbio/sequence/_sequence.py`

 * *Files 1% similar despite different names*

```diff
@@ -2148,26 +2148,26 @@
             if np.any(normalized != sliceable):
                 raise ValueError("Index regions are out of order.")
 
         return normalized
 
     def _munge_to_self_type(self, other, method):
         if isinstance(other, Sequence):
-            if type(other) != type(self):
+            if type(other) is not type(self):
                 raise TypeError("Cannot use %s and %s together with `%s`" %
                                 (self.__class__.__name__,
                                  other.__class__.__name__, method))
             else:
                 return other
 
         return self.__class__(other)
 
     def _munge_to_sequence(self, other, method):
         if isinstance(other, Sequence):
-            if type(other) != type(self):
+            if type(other) is not type(self):
                 raise TypeError("Cannot use %s and %s together with `%s`" %
                                 (self.__class__.__name__,
                                  other.__class__.__name__, method))
             else:
                 return other
 
         # We don't use self.__class__ or self._constructor here because we want
```

### Comparing `scikit-bio-0.5.8/skbio/sequence/distance.py` & `scikit-bio-0.5.9/skbio/sequence/distance.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/tests/test_distance.py` & `scikit-bio-0.5.9/skbio/sequence/tests/test_distance.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/tests/test_dna.py` & `scikit-bio-0.5.9/skbio/sequence/tests/test_dna.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/tests/test_genetic_code.py` & `scikit-bio-0.5.9/skbio/sequence/tests/test_genetic_code.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/tests/test_grammared_sequence.py` & `scikit-bio-0.5.9/skbio/sequence/tests/test_grammared_sequence.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/tests/test_nucleotide_sequences.py` & `scikit-bio-0.5.9/skbio/sequence/tests/test_nucleotide_sequences.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/tests/test_protein.py` & `scikit-bio-0.5.9/skbio/sequence/tests/test_protein.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/tests/test_rna.py` & `scikit-bio-0.5.9/skbio/sequence/tests/test_rna.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/sequence/tests/test_sequence.py` & `scikit-bio-0.5.9/skbio/sequence/tests/test_sequence.py`

 * *Files 0% similar despite different names*

```diff
@@ -454,15 +454,15 @@
             # int64
             Sequence(np.array([1, 2, 3]))
         with self.assertRaises(TypeError):
             # |S21
             Sequence(np.array([1, "23", 3]))
         with self.assertRaises(TypeError):
             # object
-            Sequence(np.array([1, {}, ()]))
+            Sequence(np.array([1, {}, ()], dtype=object))
 
         # invalid input type (non-numpy.ndarray input)
         with self.assertRaisesRegex(TypeError, r'tuple'):
             Sequence(('a', 'b', 'c'))
         with self.assertRaisesRegex(TypeError, r'list'):
             Sequence(['a', 'b', 'c'])
         with self.assertRaisesRegex(TypeError, r'set'):
@@ -2199,32 +2199,20 @@
 
         def slices_only():
             return (slice(i, i+1) for i in range(0, 10, 2))
 
         def mixed():
             return (slice(i, i+1) if i % 2 == 0 else i for i in range(10))
 
-        def unthinkable():
-            for i in range(10):
-                if i % 3 == 0:
-                    yield slice(i, i+1)
-                elif i % 3 == 1:
-                    yield i
-                else:
-                    yield np.array([i], dtype=int)
         for c in (lambda x: x, list, tuple, lambda x: np.array(tuple(x)),
                   lambda x: pd.Series(tuple(x))):
             exp = np.arange(10, dtype=int)
             obs = s._munge_to_index_array(c(mixed()))
             npt.assert_equal(obs, exp)
 
-            exp = np.arange(10, dtype=int)
-            obs = s._munge_to_index_array(c(unthinkable()))
-            npt.assert_equal(obs, exp)
-
             exp = np.arange(10, step=2, dtype=int)
             obs = s._munge_to_index_array(c(slices_only()))
             npt.assert_equal(obs, exp)
 
     def test_munge_to_index_array_invalid_iterable(self):
         s = Sequence('')
```

### Comparing `scikit-bio-0.5.8/skbio/stats/__init__.py` & `scikit-bio-0.5.9/skbio/stats/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/__subsample.c` & `scikit-bio-0.5.9/skbio/stats/__subsample.c`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/__subsample.pyx` & `scikit-bio-0.5.9/skbio/stats/__subsample.pyx`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/_misc.py` & `scikit-bio-0.5.9/skbio/stats/_misc.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/_subsample.py` & `scikit-bio-0.5.9/skbio/stats/_subsample.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/composition.py` & `scikit-bio-0.5.9/skbio/stats/composition.py`

 * *Files 16% similar despite different names*

```diff
@@ -103,14 +103,16 @@
 # ----------------------------------------------------------------------------
 
 import numpy as np
 import pandas as pd
 import scipy.stats
 import skbio.util
 from skbio.util._decorator import experimental
+from skbio.stats.distance import DistanceMatrix
+from scipy.sparse import coo_matrix
 
 
 @experimental(as_of="0.4.0")
 def closure(mat):
     """
     Performs closure to ensure that all elements add up to 1.
 
@@ -225,16 +227,15 @@
                          'proportions. Consider using a smaller `delta`.')
     mat = np.where(z_mat, delta, zcnts * mat)
     return mat.squeeze()
 
 
 @experimental(as_of="0.4.0")
 def perturb(x, y):
-    r"""
-    Performs the perturbation operation.
+    r""" Performs the perturbation operation.
 
     This operation is defined as
 
     .. math::
         x \oplus y = C[x_1 y_1, \ldots, x_D y_D]
 
     :math:`C[x]` is the closure operation defined as
@@ -275,16 +276,15 @@
     """
     x, y = closure(x), closure(y)
     return closure(x * y)
 
 
 @experimental(as_of="0.4.0")
 def perturb_inv(x, y):
-    r"""
-    Performs the inverse perturbation operation.
+    r""" Performs the inverse perturbation operation.
 
     This operation is defined as
 
     .. math::
         x \ominus y = C[x_1 y_1^{-1}, \ldots, x_D y_D^{-1}]
 
     :math:`C[x]` is the closure operation defined as
@@ -325,16 +325,15 @@
     """
     x, y = closure(x), closure(y)
     return closure(x / y)
 
 
 @experimental(as_of="0.4.0")
 def power(x, a):
-    r"""
-    Performs the power operation.
+    r""" Performs the power operation.
 
     This operation is defined as follows
 
     .. math::
         `x \odot a = C[x_1^a, \ldots, x_D^a]
 
     :math:`C[x]` is the closure operation defined as
@@ -372,16 +371,15 @@
     """
     x = closure(x)
     return closure(x**a).squeeze()
 
 
 @experimental(as_of="0.4.0")
 def inner(x, y):
-    r"""
-    Calculates the Aitchson inner product.
+    r""" Calculates the Aitchson inner product.
 
     This inner product is defined as follows
 
     .. math::
         \langle x, y \rangle_a =
         \frac{1}{2D} \sum\limits_{i=1}^{D} \sum\limits_{j=1}^{D}
         \ln\left(\frac{x_i}{x_j}\right) \ln\left(\frac{y_i}{y_j}\right)
@@ -415,16 +413,15 @@
     y = closure(y)
     a, b = clr(x), clr(y)
     return a.dot(b.T)
 
 
 @experimental(as_of="0.4.0")
 def clr(mat):
-    r"""
-    Performs centre log ratio transformation.
+    r""" Performs centre log ratio transformation.
 
     This function transforms compositions from Aitchison geometry to
     the real space. The :math:`clr` transform is both an isometry and an
     isomorphism defined on the following spaces
 
     :math:`clr: S^D \rightarrow U`
 
@@ -464,16 +461,15 @@
     lmat = np.log(mat)
     gm = lmat.mean(axis=-1, keepdims=True)
     return (lmat - gm).squeeze()
 
 
 @experimental(as_of="0.4.0")
 def clr_inv(mat):
-    r"""
-    Performs inverse centre log ratio transformation.
+    r""" Performs inverse centre log ratio transformation.
 
     This function transforms compositions from the real space to
     Aitchison geometry. The :math:`clr^{-1}` transform is both an isometry,
     and an isomorphism defined on the following spaces
 
     :math:`clr^{-1}: U \rightarrow S^D`
 
@@ -500,23 +496,24 @@
     Examples
     --------
     >>> import numpy as np
     >>> from skbio.stats.composition import clr_inv
     >>> x = np.array([.1, .3, .4, .2])
     >>> clr_inv(x)
     array([ 0.21383822,  0.26118259,  0.28865141,  0.23632778])
-
     """
-    return closure(np.exp(mat))
+    # for numerical stability (aka softmax trick)
+    mat = np.atleast_2d(mat)
+    emat = np.exp(mat - mat.max(axis=-1, keepdims=True))
+    return closure(emat)
 
 
 @experimental(as_of="0.4.0")
 def ilr(mat, basis=None, check=True):
-    r"""
-    Performs isometric log ratio transformation.
+    r""" Performs isometric log ratio transformation.
 
     This function transforms compositions from Aitchison simplex to
     the real space. The :math: ilr` transform is both an isometry,
     and an isomorphism defined on the following spaces
 
     :math:`ilr: S^D \rightarrow \mathbb{R}^{D-1}`
 
@@ -535,15 +532,15 @@
     Parameters
     ----------
     mat: numpy.ndarray
        a matrix of proportions where
        rows = compositions and
        columns = components
 
-    basis: numpy.ndarray, float, optional
+    basis: numpy.ndarray or scipy.sparse.matrix, optional
         orthonormal basis for Aitchison simplex
         defaults to J.J.Egozcue orthonormal basis.
 
     check: bool
         Specifies if the basis is orthonormal.
 
     Examples
@@ -559,30 +556,30 @@
     If the `basis` parameter is specified, it is expected to be a basis in the
     Aitchison simplex.  If there are `D-1` elements specified in `mat`, then
     the dimensions of the basis needs be `D-1 x D`, where rows represent
     basis vectors, and the columns represent proportions.
     """
     mat = closure(mat)
     if basis is None:
-        basis = clr_inv(_gram_schmidt_basis(mat.shape[-1]))
+        d = mat.shape[-1]
+        basis = _gram_schmidt_basis(d)  # dimension (d-1) x d
     else:
         if len(basis.shape) != 2:
             raise ValueError("Basis needs to be a 2D matrix, "
                              "not a %dD matrix." %
                              (len(basis.shape)))
         if check:
             _check_orthogonality(basis)
 
-    return inner(mat, basis)
+    return clr(mat) @ basis.T
 
 
 @experimental(as_of="0.4.0")
 def ilr_inv(mat, basis=None, check=True):
-    r"""
-    Performs inverse isometric log ratio transform.
+    r""" Performs inverse isometric log ratio transform.
 
     This function transforms compositions from the real space to
     Aitchison geometry. The :math:`ilr^{-1}` transform is both an isometry,
     and an isomorphism defined on the following spaces
 
     :math:`ilr^{-1}: \mathbb{R}^{D-1} \rightarrow S^D`
 
@@ -624,35 +621,35 @@
     Notes
     -----
     If the `basis` parameter is specified, it is expected to be a basis in the
     Aitchison simplex.  If there are `D-1` elements specified in `mat`, then
     the dimensions of the basis needs be `D-1 x D`, where rows represent
     basis vectors, and the columns represent proportions.
     """
-
+    mat = np.atleast_2d(mat)
     if basis is None:
+        # dimension d-1 x d basis
         basis = _gram_schmidt_basis(mat.shape[-1] + 1)
     else:
         if len(basis.shape) != 2:
             raise ValueError("Basis needs to be a 2D matrix, "
                              "not a %dD matrix." %
                              (len(basis.shape)))
         if check:
             _check_orthogonality(basis)
         # this is necessary, since the clr function
         # performs np.squeeze()
-        basis = np.atleast_2d(clr(basis))
+        basis = np.atleast_2d(basis)
 
-    return clr_inv(np.dot(mat, basis))
+    return clr_inv(mat @ basis)
 
 
 @experimental(as_of="0.5.5")
 def alr(mat, denominator_idx=0):
-    r"""
-    Performs additive log ratio transformation.
+    r""" Performs additive log ratio transformation.
 
     This function transforms compositions from a D-part Aitchison simplex to
     a non-isometric real space of D-1 dimensions. The argument
     `denominator_col` defines the index of the column used as the common
     denominator. The :math: `alr` transformed data are amenable to multivariate
     analysis as long as statistics don't involve distances.
 
@@ -705,16 +702,15 @@
     else:
         raise ValueError("mat must be either 1D or 2D")
     return lr
 
 
 @experimental(as_of="0.5.5")
 def alr_inv(mat, denominator_idx=0):
-    r"""
-    Performs inverse additive log ratio transform.
+    r""" Performs inverse additive log ratio transform.
 
     This function transforms compositions from the non-isometric real space of
     alrs to Aitchison geometry.
 
     :math:`alr^{-1}: \mathbb{R}^{D-1} \rightarrow S^D`
 
     The inverse alr transformation is defined as follows
@@ -803,14 +799,391 @@
 
     """
     mat = closure(mat)
     cen = scipy.stats.gmean(mat, axis=0)
     return perturb_inv(mat, cen)
 
 
+@experimental(as_of="0.5.7")
+def _vlr(x: np.array, y: np.array, ddof: int):
+    r""" Calculates variance log ratio
+
+    Parameters
+    ----------
+    x : array_like, float
+       a 1-dimensional vector of proportions
+    y : array_like, float
+       a 1-dimensional vector of proportions
+    ddof: int
+        degrees of freedom
+
+    Returns
+    -------
+    float
+         variance log ratio value
+
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S,
+           Bhler J (2015) Proportionality: A Valid Alternative to
+           Correlation for Relative Data. PLoS Comput Biol 11(3): e1004075.
+           https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C.
+           How should we measure proportionality on relative gene
+           expression data?. Theory Biosci. 135, 2136 (2016).
+           https://doi.org/10.1007/s12064-015-0220-8
+    """
+
+    # Log transformation
+    x = np.log(x)
+    y = np.log(y)
+
+    # Variance log ratio
+    return np.var(x - y, ddof=ddof)
+
+
+@experimental(as_of="0.5.7")
+def _robust_vlr(x: np.ndarray, y: np.ndarray, ddof: int):
+    r""" Calculates variance log ratio while masking zeros
+
+    Parameters
+    ----------
+    x : array_like, float
+       a 1-dimensional vector of proportions
+    y : array_like, float
+       a 1-dimensional vector of proportions
+    ddof: int
+        degrees of freedom
+
+    Returns
+    -------
+    float
+         variance log ratio value
+
+
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S,
+           Bhler J (2015) Proportionality: A Valid Alternative to
+           Correlation for Relative Data. PLoS Comput Biol 11(3): e1004075.
+           https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C.
+           How should we measure proportionality on relative gene
+           expression data?. Theory Biosci. 135, 2136 (2016).
+           https://doi.org/10.1007/s12064-015-0220-8
+    """
+
+    # Mask zeros
+    x = np.ma.masked_array(x, mask=x == 0)
+    y = np.ma.masked_array(y, mask=y == 0)
+
+    # Log transformation
+    x = np.ma.log(x)
+    y = np.ma.log(y)
+
+    # Variance log ratio
+    return np.ma.var(x - y, ddof=ddof)
+
+
+@experimental(as_of="0.5.7")
+def vlr(x: np.ndarray, y: np.ndarray, ddof: int = 1, robust: bool = False):
+    r""" Calculates variance log ratio
+
+    Parameters
+    ----------
+    x : array_like, float
+       a 1-dimensional vector of proportions
+    y : array_like, float
+       a 1-dimensional vector of proportions
+
+    ddof: int
+        degrees of freedom
+
+    robust: bool
+        mask zeros at the cost of performance
+
+    Returns
+    -------
+    float
+         variance log ratio value
+
+    Examples
+    --------
+    No zeros
+    >>> import numpy as np
+    >>> from skbio.stats.composition import vlr
+    >>> x = np.exp([1,2,3])
+    >>> y = np.exp([2,3,4])
+    >>> vlr(x,y)
+    0.0
+
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S,
+           Bhler J (2015) Proportionality: A Valid Alternative to
+           Correlation for Relative Data. PLoS Comput Biol 11(3): e1004075.
+           https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C.
+           How should we measure proportionality on relative gene
+           expression data?. Theory Biosci. 135, 2136 (2016).
+           https://doi.org/10.1007/s12064-015-0220-8
+    """
+    # Convert array_like to numpy array
+    x = closure(x)
+    y = closure(y)
+
+    # Set up input and parameters
+    kwargs = {
+        "x": x,
+        "y": y,
+        "ddof": ddof,
+    }
+
+    # Run backend function
+    if robust:
+        return _robust_vlr(**kwargs)
+    else:
+        return _vlr(**kwargs)
+
+
+@experimental(as_of="0.5.7")
+def _pairwise_vlr(mat: np.ndarray, ddof: int):
+    r""" Performs pairwise variance log ratio transformation
+
+    Parameters
+    ----------
+    mat : array_like, float
+       a matrix of proportions where
+       rows = compositions and
+       columns = components
+    ids : array_like, str
+        component names
+    ddof : int
+        degrees of freedom
+
+    Returns
+    -------
+    skbio.DistanceMatrix
+         distance matrix of variance log ratio values
+
+
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S,
+           Bhler J (2015) Proportionality: A Valid Alternative to
+           Correlation for Relative Data. PLoS Comput Biol 11(3): e1004075.
+           https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C.
+           How should we measure proportionality on relative gene
+           expression data?. Theory Biosci. 135, 2136 (2016).
+           https://doi.org/10.1007/s12064-015-0220-8
+    """
+
+    # Log Transform
+    X_log = np.log(mat)
+
+    # Variance Log Ratio
+    covariance = np.cov(X_log.T, ddof=ddof)
+    diagonal = np.diagonal(covariance)
+    vlr_data = -2*covariance + diagonal[:, np.newaxis] + diagonal
+    return vlr_data
+
+
+@experimental(as_of="0.5.7")
+def pairwise_vlr(mat,
+                 ids=None,
+                 ddof: int = 1,
+                 robust: bool = False,
+                 validate: bool = True):
+
+    r""" Performs pairwise variance log ratio transformation
+
+    Parameters
+    ----------
+    mat: array_like, float
+       a matrix of proportions where
+       rows = compositions and
+       columns = components
+    ids: array_like, str
+        Component names
+    ddof: int
+        Degrees of freedom
+    robust: bool
+        Mask zeros at the cost of performance
+    validate: bool
+        Whether to validate the distance matrix after construction.
+
+    Returns
+    -------
+    skbio.DistanceMatrix if validate=True
+         distance matrix of variance log ratio values
+    skbio.DissimilarityMatrix if validate=False
+         dissimilarity matrix of variance log ratio values
+
+    Examples
+    --------
+    import numpy as np
+    from skbio.stats.composition import pairwise_vlr
+    >>> mat = np.array([np.exp([1,2,2]),
+    ...                 np.exp([2,3,6]),
+    ...                 np.exp([2,3,12])]).T
+    >>> dism = pairwise_vlr(mat)
+    >>> dism.redundant_form()
+    array([[  0.,   3.,  27.],
+           [  3.,   0.,  12.],
+           [ 27.,  12.,   0.]])
+
+
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S,
+           Bhler J (2015) Proportionality: A Valid Alternative to
+           Correlation for Relative Data. PLoS Comput Biol 11(3): e1004075.
+           https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C.
+           How should we measure proportionality on relative gene
+           expression data?. Theory Biosci. 135, 2136 (2016).
+           https://doi.org/10.1007/s12064-015-0220-8
+    """
+
+    # Mask zeros
+    mat = closure(mat.astype(np.float64))
+
+    # Set up input and parameters
+    kwargs = {
+        "mat": mat,
+        "ddof": ddof,
+    }
+
+    # Variance Log Ratio
+    if robust:
+        raise NotImplementedError(
+            'Pairwise version of robust VLR not implemented.')
+    else:
+        vlr_data = _pairwise_vlr(**kwargs)
+
+    # Return distance matrix
+    if validate:
+        vlr_data = 0.5 * (vlr_data + vlr_data.T)
+        return DistanceMatrix(vlr_data, ids=ids)
+
+    # Return dissimilarity matrix
+    else:
+        return DistanceMatrix(vlr_data, ids=ids, validate=False)
+
+
+@experimental(as_of="0.5.8")
+def tree_basis(tree):
+    r""" Calculates sparse representation of an ilr basis from a tree.
+
+    This computes an orthonormal basis specified from a bifurcating tree.
+
+    Parameters
+    ----------
+    tree : skbio.TreeNode
+        Input bifurcating tree.  Must be strictly bifurcating
+        (i.e. every internal node needs to have exactly 2 children).
+        This is used to specify the ilr basis.
+
+    Returns
+    -------
+    scipy.sparse.coo_matrix
+       The ilr basis required to perform the ilr_inv transform.
+       This is also known as the sequential binary partition.
+       Note that this matrix is represented in clr coordinates.
+    nodes : list, str
+        List of tree nodes indicating the ordering in the basis.
+
+    Raises
+    ------
+    ValueError
+        The tree doesn't contain two branches.
+
+    >>> from skbio import TreeNode
+    >>> tree = u"((b,c)a, d)root;"
+    >>> t = TreeNode.read([tree])
+    >>> basis, nodes = tree_basis(t)
+    >>> basis.toarray()
+    array([[-0.40824829, -0.40824829,  0.81649658],
+           [-0.70710678,  0.70710678,  0.        ]])
+
+    """
+    # Specifies which child is numerator and denominator
+    # within any given node in a tree.
+    NUMERATOR = 1
+    DENOMINATOR = 0
+    # this is inspired by @wasade in
+    # https://github.com/biocore/gneiss/pull/8
+    t = tree.copy()
+    D = len(list(tree.tips()))
+    # calculate number of tips under each node
+    for n in t.postorder(include_self=True):
+        if n.is_tip():
+            n._tip_count = 1
+        else:
+            if len(n.children) == 2:
+                left, right = n.children[NUMERATOR], n.children[DENOMINATOR],
+            else:
+                raise ValueError("Not a strictly bifurcating tree.")
+            n._tip_count = left._tip_count + right._tip_count
+
+    # calculate k, r, s, t coordinate for each node
+    left, right = t.children[NUMERATOR], t.children[DENOMINATOR],
+    t._k, t._r, t._s, t._t = 0, left._tip_count, right._tip_count, 0
+    for n in t.preorder(include_self=False):
+        if n.is_tip():
+            n._k, n._r, n._s, n._t = 0, 0, 0, 0
+
+        elif n == n.parent.children[NUMERATOR]:
+            n._k = n.parent._k
+            n._r = n.children[NUMERATOR]._tip_count
+            n._s = n.children[DENOMINATOR]._tip_count
+            n._t = n.parent._s + n.parent._t
+        elif n == n.parent.children[DENOMINATOR]:
+            n._k = n.parent._r + n.parent._k
+            n._r = n.children[NUMERATOR]._tip_count
+            n._s = n.children[DENOMINATOR]._tip_count
+            n._t = n.parent._t
+        else:
+            raise ValueError("Tree topology is not correct.")
+
+    # navigate through tree to build the basis in a sparse matrix form
+    value = []
+    row, col = [], []
+    nodes = []
+    i = 0
+
+    for n in t.levelorder(include_self=True):
+
+        if n.is_tip():
+            continue
+
+        for j in range(n._k, n._k + n._r):
+            row.append(i)
+            # consider tips in reverse order. May want to rethink
+            # this orientation in the future.
+            col.append(D - 1 - j)
+            A = np.sqrt(n._s / (n._r * (n._s + n._r)))
+
+            value.append(A)
+
+        for j in range(n._k + n._r, n._k + n._r + n._s):
+            row.append(i)
+            col.append(D - 1 - j)
+            B = -np.sqrt(n._r / (n._s * (n._s + n._r)))
+
+            value.append(B)
+        i += 1
+        nodes.append(n.name)
+
+    basis = coo_matrix((value, (row, col)), shape=(D - 1, D))
+
+    return basis, nodes
+
+
 @experimental(as_of="0.4.1")
 def ancom(table, grouping,
           alpha=0.05,
           tau=0.02,
           theta=0.1,
           multiple_comparisons_correction='holm-bonferroni',
           significance_test=None,
@@ -1229,14 +1602,19 @@
     Builds clr transformed basis derived from
     gram schmidt orthogonalization
 
     Parameters
     ----------
     n : int
         Dimension of the Aitchison simplex
+
+    Returns
+    -------
+    basis : np.array
+        Dimension (n-1) x n basis matrix
     """
     basis = np.zeros((n, n-1))
     for j in range(n-1):
         i = j + 1
         e = np.array([(1/i)]*i + [-1] +
                      [0]*(n-i-1))*np.sqrt(i/(i+1))
         basis[:, j] = e
@@ -1284,18 +1662,18 @@
     >>> import numpy as np
     >>> sbp = np.array([[1, 1,-1,-1,-1],
     ...                 [1,-1, 0, 0, 0],
     ...                 [0, 0, 1,-1,-1],
     ...                 [0, 0, 0, 1,-1]])
     ...
     >>> sbp_basis(sbp)
-    array([[ 0.31209907,  0.31209907,  0.12526729,  0.12526729,  0.12526729],
-           [ 0.36733337,  0.08930489,  0.18112058,  0.18112058,  0.18112058],
-           [ 0.17882092,  0.17882092,  0.40459293,  0.11888261,  0.11888261],
-           [ 0.18112058,  0.18112058,  0.18112058,  0.36733337,  0.08930489]])
+    array([[ 0.54772256,  0.54772256, -0.36514837, -0.36514837, -0.36514837],
+           [ 0.70710678, -0.70710678,  0.        ,  0.        ,  0.        ],
+           [ 0.        ,  0.        ,  0.81649658, -0.40824829, -0.40824829],
+           [ 0.        ,  0.        ,  0.        ,  0.70710678, -0.70710678]])
 
     References
     ----------
     .. [1] Parent, S.., Parent, L.E., Egozcue, J.J., Rozane, D.E.,
        Hernandes, A., Lapointe, L., Hbert-Gentile, V., Naess, K.,
        Marchand, S., Lafond, J., Mattos, D., Barlow, P., Natale, W., 2013.
        The plant ionome revisited by the nutrient balance concept.
@@ -1307,24 +1685,24 @@
 
     n_pos = (sbp == 1).sum(axis=1)
     n_neg = (sbp == -1).sum(axis=1)
     psi = np.zeros(sbp.shape)
     for i in range(0, sbp.shape[0]):
         psi[i, :] = sbp[i, :] * np.sqrt((n_neg[i] / n_pos[i])**sbp[i, :] /
                                         np.sum(np.abs(sbp[i, :])))
-    return clr_inv(psi)
+    return psi
 
 
 def _check_orthogonality(basis):
     """
     Checks to see if basis is truly orthonormal in the
     Aitchison simplex
 
     Parameters
     ----------
     basis: numpy.ndarray
-        basis in the Aitchison simplex
+        basis in the Aitchison simplex of dimension d-1 x d
     """
     basis = np.atleast_2d(basis)
-    if not np.allclose(inner(basis, basis), np.identity(len(basis)),
+    if not np.allclose(basis @ basis.T, np.identity(len(basis)),
                        rtol=1e-4, atol=1e-6):
-        raise ValueError("Aitchison basis is not orthonormal")
+        raise ValueError("Basis is not orthonormal")
```

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/__init__.py` & `scikit-bio-0.5.9/skbio/stats/distance/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/_anosim.py` & `scikit-bio-0.5.9/skbio/stats/distance/_anosim.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/_base.py` & `scikit-bio-0.5.9/skbio/stats/distance/_base.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/_bioenv.py` & `scikit-bio-0.5.9/skbio/stats/distance/_bioenv.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/_cutils.c` & `scikit-bio-0.5.9/skbio/stats/distance/_cutils.c`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/_cutils.pyx` & `scikit-bio-0.5.9/skbio/stats/distance/_cutils.pyx`

 * *Files 0% similar despite different names*

```diff
@@ -61,15 +61,15 @@
                    is_hollow &= (mat[col,col]==0)
 
     return [(is_sym==True), (is_hollow==True)]
 
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def distmat_reorder_cy(TReal[:, ::1] in_mat, long[::1] reorder_vec, 
+def distmat_reorder_cy(TReal[:, ::1] in_mat, long[::1] reorder_vec,
                        TReal[:, ::1] out_mat):
     """
     Reorder the rows and columns of a distance matrix
     given a reorder vector.
     Not all of the columns need to be used.
 
     For example:
@@ -156,15 +156,15 @@
 
     cdef Py_ssize_t row,col
     cdef Py_ssize_t vrow
     cdef Py_ssize_t idx
 
     for row in prange(out_n-1, nogil=True):
         vrow = reorder_vec[row]
-        idx = row*(out_n-1) - ((row-1)*row)/2
+        idx = row*(out_n-1) - ((row-1)*row)//2
         for col in range(out_n-row-1):
            out_mat_condensed[idx+col] = in_mat[vrow, reorder_vec[col+row+1]]
 
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
 def mantel_perm_pearsonr_cy(TReal[:, ::1] x_data, long[:, ::1] perm_order,
@@ -226,15 +226,15 @@
     cdef TReal yval
     cdef TReal xval
 
     for p in prange(perms_n, nogil=True):
         my_ps = 0.0
         for row in range(out_n-1):
             vrow = perm_order[p, row]
-            idx = row*(out_n-1) - ((row-1)*row)/2
+            idx = row*(out_n-1) - ((row-1)*row)//2
             for icol in range(out_n-row-1):
                col = icol+row+1
                yval = ym_normalized[idx+icol]
                xval = x_data[vrow, perm_order[p, col]]*mul + add
                # do not use += to avoid having prange consider it for reduction
                my_ps = yval*xval + my_ps
```

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/_mantel.py` & `scikit-bio-0.5.9/skbio/stats/distance/_mantel.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/_permanova.py` & `scikit-bio-0.5.9/skbio/stats/distance/_permanova.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/_permdisp.py` & `scikit-bio-0.5.9/skbio/stats/distance/_permdisp.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/_utils.py` & `scikit-bio-0.5.9/skbio/stats/distance/_utils.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/bioenv_df_vegan.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/bioenv_df_vegan.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/bioenv_dm_vegan.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/bioenv_dm_vegan.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/df.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/df.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/df_extra_column.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/df_extra_column.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/dm.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/dm.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/dm2.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/dm2.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/dm3.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/dm3.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/dm4.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/dm4.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/dm_reordered.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/dm_reordered.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/exp_results.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/exp_results.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/exp_results_different_column_order.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/exp_results_different_column_order.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/mantel_env_dm_vegan.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/mantel_env_dm_vegan.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/mantel_veg_dm_vegan.txt` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/mantel_veg_dm_vegan.txt`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/moving_pictures_dm.tsv` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/moving_pictures_dm.tsv`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/data/moving_pictures_mf.tsv` & `scikit-bio-0.5.9/skbio/stats/distance/tests/data/moving_pictures_mf.tsv`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/test_anosim.py` & `scikit-bio-0.5.9/skbio/stats/distance/tests/test_anosim.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/test_base.py` & `scikit-bio-0.5.9/skbio/stats/distance/tests/test_base.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/test_bioenv.py` & `scikit-bio-0.5.9/skbio/stats/distance/tests/test_bioenv.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/test_mantel.py` & `scikit-bio-0.5.9/skbio/stats/distance/tests/test_mantel.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/test_permanova.py` & `scikit-bio-0.5.9/skbio/stats/distance/tests/test_permanova.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/distance/tests/test_permdisp.py` & `scikit-bio-0.5.9/skbio/stats/distance/tests/test_permdisp.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/evolve/__init__.py` & `scikit-bio-0.5.9/skbio/stats/evolve/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/evolve/_hommola.py` & `scikit-bio-0.5.9/skbio/stats/evolve/_hommola.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/evolve/tests/test_hommola.py` & `scikit-bio-0.5.9/skbio/stats/evolve/tests/test_hommola.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/gradient.py` & `scikit-bio-0.5.9/skbio/stats/gradient.py`

 * *Files 0% similar despite different names*

```diff
@@ -497,14 +497,17 @@
         sample_ids = coords_sample_ids.intersection(mm_sample_ids)
 
         # Check if they actually have sample ids in common
         if not sample_ids:
             raise ValueError("Coordinates and metadata map had no samples "
                              "in common")
 
+        # pandas no longer allows use of set with .loc
+        sample_ids = list(sample_ids)
+
         # Need to take a subset of coords
         if coords_sample_ids != sample_ids:
             self._coords = self._coords.loc[sample_ids]
         # Need to take a subset of metadata_map
         if mm_sample_ids != sample_ids:
             self._metadata_map = self._metadata_map.loc[sample_ids]
```

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/__init__.py` & `scikit-bio-0.5.9/skbio/stats/ordination/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/_canonical_correspondence_analysis.py` & `scikit-bio-0.5.9/skbio/stats/ordination/_canonical_correspondence_analysis.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/_correspondence_analysis.py` & `scikit-bio-0.5.9/skbio/stats/ordination/_correspondence_analysis.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/_cutils.c` & `scikit-bio-0.5.9/skbio/stats/ordination/_cutils.c`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/_cutils.pyx` & `scikit-bio-0.5.9/skbio/stats/ordination/_cutils.pyx`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/_ordination_results.py` & `scikit-bio-0.5.9/skbio/stats/ordination/_ordination_results.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/_principal_coordinate_analysis.py` & `scikit-bio-0.5.9/skbio/stats/ordination/_principal_coordinate_analysis.py`

 * *Files 0% similar despite different names*

```diff
@@ -369,15 +369,15 @@
         raise ValueError('This biplot computation can only be performed in a '
                          'PCoA matrix.')
 
     if set(y.index) != set(ordination.samples.index):
         raise ValueError('The eigenvectors and the descriptors must describe '
                          'the same samples.')
 
-    eigvals = ordination.eigvals
+    eigvals = ordination.eigvals.values
     coordinates = ordination.samples
     N = coordinates.shape[0]
 
     # align the descriptors and eigenvectors in a sample-wise fashion
     y = y.reindex(coordinates.index)
 
     # S_pc from equation 9.44
```

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/_redundancy_analysis.py` & `scikit-bio-0.5.9/skbio/stats/ordination/_redundancy_analysis.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/_utils.py` & `scikit-bio-0.5.9/skbio/stats/ordination/_utils.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_biplot_descriptors` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_biplot_descriptors`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_biplot_projected_descriptors` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_biplot_projected_descriptors`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data_12dim` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data_12dim`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data_2` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data_2`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data_3` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data_3`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data_3_eigh_ref_3dim` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data_3_eigh_ref_3dim`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_sample_data_3_fsvd_ref_3dim` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_sample_data_3_fsvd_ref_3dim`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/PCoA_skbio` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/PCoA_skbio`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_X` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_X`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_sample_constraints_scaling1` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_sample_constraints_scaling1`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_sample_constraints_scaling2` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_sample_constraints_scaling2`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_site_scaling1_from_vegan` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_site_scaling1_from_vegan`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example2_site_scaling2_from_vegan` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example2_site_scaling2_from_vegan`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_X` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_X`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_sample_constraints_scaling1` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_sample_constraints_scaling1`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_sample_constraints_scaling2` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_sample_constraints_scaling2`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_site_scaling1_from_vegan` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_site_scaling1_from_vegan`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_site_scaling2_from_vegan` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_site_scaling2_from_vegan`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_species_scaling1_from_vegan` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_species_scaling1_from_vegan`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/example3_species_scaling2_from_vegan` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/example3_species_scaling2_from_vegan`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/exp_PCoAEigenResults_site` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/exp_PCoAEigenResults_site`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/exp_PCoAzeros_site` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/exp_PCoAzeros_site`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_biplot_from_vegan.csv` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_biplot_from_vegan.csv`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_constraints_from_vegan.csv` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_constraints_from_vegan.csv`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_eigvals_from_vegan.csv` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_eigvals_from_vegan.csv`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_sites_from_vegan.csv` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_sites_from_vegan.csv`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/vare_rda_species_from_vegan.csv` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/vare_rda_species_from_vegan.csv`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/varechem.csv` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/varechem.csv`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/data/varespec.csv` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/data/varespec.csv`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/test_canonical_correspondence_analysis.py` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/test_canonical_correspondence_analysis.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/test_correspondence_analysis.py` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/test_correspondence_analysis.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/test_ordination_results.py` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/test_ordination_results.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/test_principal_coordinate_analysis.py` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/test_principal_coordinate_analysis.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/test_redundancy_analysis.py` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/test_redundancy_analysis.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/ordination/tests/test_util.py` & `scikit-bio-0.5.9/skbio/stats/ordination/tests/test_util.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/power.py` & `scikit-bio-0.5.9/skbio/stats/power.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/tests/data/vr_real_out` & `scikit-bio-0.5.9/skbio/stats/tests/data/vr_real_out`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/tests/test_composition.py` & `scikit-bio-0.5.9/skbio/stats/tests/test_composition.py`

 * *Files 4% similar despite different names*

```diff
@@ -11,20 +11,37 @@
 import numpy as np
 import numpy.testing as npt
 import pandas.testing as pdt
 from numpy.random import normal
 import pandas as pd
 import scipy
 import copy
+from skbio import TreeNode
 from skbio.util import assert_data_frame_almost_equal
+from skbio.stats.distance import DistanceMatrixError
 from skbio.stats.composition import (closure, multiplicative_replacement,
                                      perturb, perturb_inv, power, inner,
                                      clr, clr_inv, ilr, ilr_inv, alr, alr_inv,
                                      sbp_basis, _gram_schmidt_basis,
-                                     centralize, _holm_bonferroni, ancom)
+                                     centralize, _holm_bonferroni, ancom,
+                                     vlr, pairwise_vlr, tree_basis)
+
+from scipy.sparse import coo_matrix
+
+
+def assert_coo_allclose(res, exp, rtol=1e-7, atol=1e-7):
+    res_data = np.vstack((res.row, res.col, res.data)).T
+    exp_data = np.vstack((exp.row, exp.col, exp.data)).T
+
+    # sort by row and col
+    res_data = res_data[res_data[:, 1].argsort()]
+    res_data = res_data[res_data[:, 0].argsort()]
+    exp_data = exp_data[exp_data[:, 1].argsort()]
+    exp_data = exp_data[exp_data[:, 0].argsort()]
+    npt.assert_allclose(res_data, exp_data, rtol=rtol, atol=atol)
 
 
 class CompositionTests(TestCase):
 
     def setUp(self):
         # Compositional data
         self.cdata1 = np.array([[2, 2, 6],
@@ -309,21 +326,21 @@
 
     def test_ilr_basis(self):
         table = np.array([[1., 10.],
                           [1.14141414, 9.90909091],
                           [1.28282828, 9.81818182],
                           [1.42424242, 9.72727273],
                           [1.56565657, 9.63636364]])
-        basis = np.array([[0.80442968, 0.19557032]])
+        basis = np.atleast_2d(clr([[0.80442968, 0.19557032]]))
         res = ilr(table, basis=basis)
-        exp = np.array([np.log(1/10)*np.sqrt(1/2),
-                        np.log(1.14141414 / 9.90909091)*np.sqrt(1/2),
-                        np.log(1.28282828 / 9.81818182)*np.sqrt(1/2),
-                        np.log(1.42424242 / 9.72727273)*np.sqrt(1/2),
-                        np.log(1.56565657 / 9.63636364)*np.sqrt(1/2)])
+        exp = np.array([[np.log(1/10)*np.sqrt(1/2)],
+                        [np.log(1.14141414 / 9.90909091)*np.sqrt(1/2)],
+                        [np.log(1.28282828 / 9.81818182)*np.sqrt(1/2)],
+                        [np.log(1.42424242 / 9.72727273)*np.sqrt(1/2)],
+                        [np.log(1.56565657 / 9.63636364)*np.sqrt(1/2)]])
 
         npt.assert_allclose(res, exp)
 
     def test_ilr_basis_one_dimension_error(self):
         table = np.array([[1., 10.],
                           [1.14141414, 9.90909091],
                           [1.28282828, 9.81818182],
@@ -348,49 +365,51 @@
         npt.assert_allclose(self.cdata1,
                             np.array([[2, 2, 6],
                                       [4, 4, 2]]))
 
     def test_ilr_basis_isomorphism(self):
         # tests to make sure that the isomorphism holds
         # with the introduction of the basis.
-        basis = np.array([[0.80442968, 0.19557032]])
+        basis = np.atleast_2d(clr([[0.80442968, 0.19557032]]))
         table = np.array([[np.log(1/10)*np.sqrt(1/2),
                            np.log(1.14141414 / 9.90909091)*np.sqrt(1/2),
                            np.log(1.28282828 / 9.81818182)*np.sqrt(1/2),
                            np.log(1.42424242 / 9.72727273)*np.sqrt(1/2),
                            np.log(1.56565657 / 9.63636364)*np.sqrt(1/2)]]).T
-        res = ilr(ilr_inv(table, basis=basis), basis=basis)
-        npt.assert_allclose(res, table.squeeze())
+        lr = ilr_inv(table, basis=basis)
+        res = ilr(lr, basis=basis)
+        npt.assert_allclose(res, table)
 
         table = np.array([[1., 10.],
                           [1.14141414, 9.90909091],
                           [1.28282828, 9.81818182],
                           [1.42424242, 9.72727273],
                           [1.56565657, 9.63636364]])
 
-        res = ilr_inv(np.atleast_2d(ilr(table, basis=basis)).T, basis=basis)
+        res = ilr_inv(ilr(table, basis=basis), basis=basis)
         npt.assert_allclose(res, closure(table.squeeze()))
 
     def test_ilr_inv_basis(self):
         exp = closure(np.array([[1., 10.],
                                 [1.14141414, 9.90909091],
                                 [1.28282828, 9.81818182],
                                 [1.42424242, 9.72727273],
                                 [1.56565657, 9.63636364]]))
-        basis = np.array([[0.80442968, 0.19557032]])
+        basis = np.atleast_2d(clr([[0.80442968, 0.19557032]]))
         table = np.array([[np.log(1/10)*np.sqrt(1/2),
                            np.log(1.14141414 / 9.90909091)*np.sqrt(1/2),
                            np.log(1.28282828 / 9.81818182)*np.sqrt(1/2),
                            np.log(1.42424242 / 9.72727273)*np.sqrt(1/2),
                            np.log(1.56565657 / 9.63636364)*np.sqrt(1/2)]]).T
+
         res = ilr_inv(table, basis=basis)
         npt.assert_allclose(res, exp)
 
     def test_ilr_inv_basis_one_dimension_error(self):
-        basis = clr(np.array([[0.80442968, 0.19557032]]))
+        basis = clr([0.80442968, 0.19557032])
         table = np.array([[np.log(1/10)*np.sqrt(1/2),
                            np.log(1.14141414 / 9.90909091)*np.sqrt(1/2),
                            np.log(1.28282828 / 9.81818182)*np.sqrt(1/2),
                            np.log(1.42424242 / 9.72727273)*np.sqrt(1/2),
                            np.log(1.56565657 / 9.63636364)*np.sqrt(1/2)]]).T
         with self.assertRaises(ValueError):
             ilr_inv(table, basis=basis)
@@ -453,15 +472,15 @@
                                       [0.28867513, 0.28867513,
                                        0.28867513, -0.8660254]]))
 
         with self.assertRaises(ValueError):
             alr_inv(self.bad2)
 
     def test_sbp_basis_gram_schmidt(self):
-        gsbasis = clr_inv(_gram_schmidt_basis(5))
+        gsbasis = _gram_schmidt_basis(5)
         sbp = np.array([[1, -1, 0, 0, 0],
                         [1, 1, -1, 0, 0],
                         [1, 1, 1, -1, 0],
                         [1, 1, 1, 1, -1]])
         sbpbasis = sbp_basis(sbp)
         npt.assert_allclose(gsbasis, sbpbasis)
 
@@ -486,16 +505,67 @@
         psi = np.zeros(sbp.shape)
         for i in range(0, sbp.shape[0]):
             for j in range(0, sbp.shape[1]):
                 if sbp[i, j] == 1:
                     psi[i, j] = np.sqrt(s[i]/(r[i]*(r[i]+s[i])))
                 elif sbp[i, j] == -1:
                     psi[i, j] = -np.sqrt(r[i]/(s[i]*(r[i]+s[i])))
-        basis_byhand = clr_inv(psi)
-        npt.assert_allclose(basis_byhand, sbpbasis)
+        npt.assert_allclose(psi, sbpbasis)
+
+
+class TestTreeBasis(TestCase):
+
+    def test_tree_basis_base_case(self):
+        tree = u"(a,b);"
+        t = TreeNode.read([tree])
+
+        exp_basis = coo_matrix(
+            np.array([[-np.sqrt(1. / 2),
+                       np.sqrt(1. / 2)]]))
+        exp_keys = [t.name]
+        res_basis, res_keys = tree_basis(t)
+
+        assert_coo_allclose(exp_basis, res_basis)
+        self.assertListEqual(exp_keys, res_keys)
+
+    def test_tree_basis_invalid(self):
+        with self.assertRaises(ValueError):
+            tree = u"(a,b,c);"
+            t = TreeNode.read([tree])
+            tree_basis(t)
+
+    def test_tree_basis_unbalanced(self):
+        tree = u"((a,b)c, d);"
+        t = TreeNode.read([tree])
+        exp_basis = coo_matrix(np.array(
+            [[-np.sqrt(1. / 6), -np.sqrt(1. / 6), np.sqrt(2. / 3)],
+             [-np.sqrt(1. / 2), np.sqrt(1. / 2), 0]]
+        ))
+        exp_keys = [t.name, t[0].name]
+        res_basis, res_keys = tree_basis(t)
+
+        assert_coo_allclose(exp_basis, res_basis)
+        self.assertListEqual(exp_keys, res_keys)
+
+    def test_tree_basis_unbalanced2(self):
+        tree = u"(d, (a,b)c);"
+
+        t = TreeNode.read([tree])
+
+        exp_basis = coo_matrix(np.array(
+            [
+                [-np.sqrt(2. / 3), np.sqrt(1. / 6), np.sqrt(1. / 6)],
+                [0, -np.sqrt(1. / 2), np.sqrt(1. / 2)]
+            ]
+        ))
+
+        exp_keys = [t.name, t[1].name]
+        res_basis, res_keys = tree_basis(t)
+        assert_coo_allclose(exp_basis, res_basis, atol=1e-7, rtol=1e-7)
+        self.assertListEqual(exp_keys, res_keys)
 
 
 class AncomTests(TestCase):
     def setUp(self):
         # Basic count data with 2 groupings
         self.table1 = pd.DataFrame([
             [10, 10, 10, 20, 20, 20],
@@ -1160,9 +1230,80 @@
         p = [0.005, 0.011, 0.02, 0.04, 0.13]
         corrected_p = p * np.arange(1, 6)[::-1]
         guessed_p = _holm_bonferroni(p)
         for a, b in zip(corrected_p, guessed_p):
             self.assertAlmostEqual(a, b)
 
 
+class TestVLR(TestCase):
+    def setUp(self):
+        self.mat = np.array([[1, 1, 2], [3, 5, 8], [13, 21, 55]])
+        self.mat_neg = np.array([[-1, 1, 2], [3, -5, 8], [13, 21, -55]])
+        self.mat_with_zero = np.array([[0, 1, 2], [3, 5, 8], [13, 21, 55]])
+
+    def test_vlr(self):
+        # No zeros
+        output = vlr(
+            x=self.mat[0],
+            y=self.mat[1],
+            ddof=1,
+            robust=False,
+        )
+        self.assertAlmostEqual(output, 0.0655828061998637)
+
+        # With zeros
+        output = vlr(
+            x=self.mat_with_zero[0],
+            y=self.mat_with_zero[1],
+            ddof=1,
+            robust=False,
+        )
+        assert np.isnan(output)
+
+        # assert raises error
+        with self.assertRaises(ValueError):
+            vlr(
+                x=self.mat_neg[0],
+                y=self.mat_neg[1],
+                ddof=1,
+                robust=False,
+            )
+
+    def test_robust_vlr(self):
+        # No zeros
+        output = vlr(
+            x=self.mat[0],
+            y=self.mat[1],
+            ddof=1,
+            robust=True,
+        )
+        self.assertAlmostEqual(output, 0.0655828061998637)
+
+        # With zeros
+        output = vlr(
+            x=self.mat_with_zero[0],
+            y=self.mat_with_zero[1],
+            ddof=1,
+            robust=True,
+        )
+        self.assertAlmostEqual(output, 0.024896522246558722)
+
+    def test_pairwise_vlr(self):
+
+        # No zeros
+        dism = pairwise_vlr(self.mat, ids=None, ddof=1, robust=False)
+        output = dism.condensed_form().sum()
+        self.assertAlmostEqual(output, 0.2857382286903922)
+
+        # With zeros
+        with self.assertRaises(DistanceMatrixError):
+            pairwise_vlr(self.mat_with_zero, ids=None, ddof=1, robust=False)
+
+        # no validation
+        dism = pairwise_vlr(self.mat, ids=None, ddof=1, robust=False,
+                            validate=False)
+        output = dism.data.sum() / 2
+        self.assertAlmostEqual(output, 0.2857382286903922)
+
+
 if __name__ == "__main__":
     main()
```

### Comparing `scikit-bio-0.5.8/skbio/stats/tests/test_gradient.py` & `scikit-bio-0.5.9/skbio/stats/tests/test_gradient.py`

 * *Files 2% similar despite different names*

```diff
@@ -476,35 +476,40 @@
         """Correctly initializes the class attributes"""
         # Note self._groups is tested on test_make_groups
         # so we are not testing it here
 
         # Test with weighted = False
         bv = GradientANOVA(self.coords, self.prop_expl, self.metadata_map)
 
-        assert_data_frame_almost_equal(bv._coords, self.coords_3axes)
+        assert_data_frame_almost_equal(bv._coords.loc[self.coords_3axes.index],
+                                       self.coords_3axes)
         exp_prop_expl = np.array([25.6216900347, 15.7715955926,
                                   14.1215046787])
         npt.assert_equal(bv._prop_expl, exp_prop_expl)
-        assert_data_frame_almost_equal(bv._metadata_map, self.metadata_map)
+        assert_data_frame_almost_equal(bv._metadata_map.loc[self.metadata_map.index],  # noqa
+                                       self.metadata_map)
         self.assertTrue(bv._weighting_vector is None)
         self.assertFalse(bv._weighted)
 
         # Test with weighted = True
         bv = GradientANOVA(self.coords, self.prop_expl, self.metadata_map,
                            sort_category='Weight', weighted=True)
 
-        assert_data_frame_almost_equal(bv._coords, self.coords_3axes)
+        assert_data_frame_almost_equal(bv._coords.loc[self.coords_3axes.index],
+                                       self.coords_3axes)
         npt.assert_equal(bv._prop_expl, exp_prop_expl)
-        assert_data_frame_almost_equal(bv._metadata_map, self.metadata_map)
+        assert_data_frame_almost_equal(bv._metadata_map.loc[self.metadata_map.index],  # noqa
+                                       self.metadata_map)
         exp_weighting_vector = pd.Series(
             np.array([60, 55, 50, 52, 57, 65, 68, 70, 72]),
             ['PC.354', 'PC.355', 'PC.356', 'PC.481', 'PC.593', 'PC.607',
              'PC.634', 'PC.635', 'PC.636'], name='Weight'
             ).astype(np.float64)
-        pdt.assert_series_equal(bv._weighting_vector, exp_weighting_vector)
+        pdt.assert_series_equal(bv._weighting_vector.loc[exp_weighting_vector.index],  # noqa
+                                exp_weighting_vector)
         self.assertTrue(bv._weighted)
 
     def test_init_error(self):
         """Raises an error with erroneous inputs"""
         # Raises ValueError if any category in trajectory_categories is not
         # present in metadata_map
         with self.assertRaises(ValueError):
```

### Comparing `scikit-bio-0.5.8/skbio/stats/tests/test_misc.py` & `scikit-bio-0.5.9/skbio/stats/tests/test_misc.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/tests/test_power.py` & `scikit-bio-0.5.9/skbio/stats/tests/test_power.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/stats/tests/test_subsample.py` & `scikit-bio-0.5.9/skbio/stats/tests/test_subsample.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/tests/test_base.py` & `scikit-bio-0.5.9/skbio/tests/test_base.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/tests/test_workflow.py` & `scikit-bio-0.5.9/skbio/tests/test_workflow.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/tree/__init__.py` & `scikit-bio-0.5.9/skbio/tree/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/tree/_exception.py` & `scikit-bio-0.5.9/skbio/tree/_exception.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/tree/_majority_rule.py` & `scikit-bio-0.5.9/skbio/tree/_majority_rule.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/tree/_nj.py` & `scikit-bio-0.5.9/skbio/tree/_nj.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/tree/_tree.py` & `scikit-bio-0.5.9/skbio/tree/_tree.py`

 * *Files 0% similar despite different names*

```diff
@@ -11,15 +11,15 @@
 from copy import deepcopy
 from itertools import combinations
 from functools import reduce
 from collections import defaultdict
 
 import numpy as np
 import pandas as pd
-from scipy.stats import pearsonr
+from scipy.spatial.distance import correlation
 
 from skbio._base import SkbioObject
 from skbio.stats.distance import DistanceMatrix
 from ._exception import (NoLengthError, DuplicateNodeError, NoParentError,
                          MissingNodeError, TreeError)
 from skbio.util import RepresentationWarning
 from skbio.util._decorator import experimental, classonlymethod
@@ -35,17 +35,16 @@
     m2 : DistanceMatrix
         a distance matrix to compare
 
     Returns
     -------
     float
         The distance between m1 and m2
-
     """
-    return (1-pearsonr(m1.data.flat, m2.data.flat)[0])/2
+    return correlation(m1.data.flat, m2.data.flat) / 2
 
 
 class TreeNode(SkbioObject):
     r"""Representation of a node within a tree
 
     A `TreeNode` instance stores links to its parent and optional children
     nodes. In addition, the `TreeNode` can represent a `length` (e.g., a
@@ -2932,15 +2931,15 @@
                 while curr is not lca:
                     ancestors[curr.id] = curr.length if curr.length is not \
                         None else 0.0
                     curr = curr.parent
             return sum(ancestors.values())
 
         else:
-            return sum(n.length for n in self.postorder(include_self=True) if
+            return sum(n.length for n in self.postorder(include_self=False) if
                        n.length is not None)
 
     @experimental(as_of="0.4.0")
     def cache_attr(self, func, cache_attrname, cache_type=list):
         """Cache attributes on internal nodes of the tree
 
         Parameters
```

### Comparing `scikit-bio-0.5.8/skbio/tree/tests/test_majority_rule.py` & `scikit-bio-0.5.9/skbio/tree/tests/test_majority_rule.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/tree/tests/test_nj.py` & `scikit-bio-0.5.9/skbio/tree/tests/test_nj.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/tree/tests/test_tree.py` & `scikit-bio-0.5.9/skbio/tree/tests/test_tree.py`

 * *Files 0% similar despite different names*

```diff
@@ -1095,14 +1095,21 @@
         self.assertEqual(str(obs), exp)
 
         t_ids = {id(n) for n in t.traverse()}
         obs_ids = {id(n) for n in obs.traverse()}
 
         self.assertEqual(t_ids.intersection(obs_ids), set())
 
+    def test_descending_branch_length_bug_1847(self):
+        tr = TreeNode.read(io.StringIO(
+            "(((A:.1,B:1.2)C:.6,(D:.9,E:.6)F:.9)G:2.4,(H:.4,I:.5)J:1.3)K;"))
+        tr.length = 1
+        tdbl = tr.descending_branch_length()
+        npt.assert_almost_equal(tdbl, 8.9)
+
     def test_descending_branch_length(self):
         """Calculate descending branch_length"""
         tr = TreeNode.read(io.StringIO(
             "(((A:.1,B:1.2)C:.6,(D:.9,E:.6)F:.9)G:2.4,(H:.4,I:.5)J:1.3)K;"))
         tdbl = tr.descending_branch_length()
         sdbl = tr.descending_branch_length(['A', 'E'])
         npt.assert_almost_equal(tdbl, 8.9)
```

### Comparing `scikit-bio-0.5.8/skbio/util/__init__.py` & `scikit-bio-0.5.9/skbio/util/__init__.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/util/_decorator.py` & `scikit-bio-0.5.9/skbio/util/_decorator.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/util/_exception.py` & `scikit-bio-0.5.9/skbio/util/_exception.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/util/_misc.py` & `scikit-bio-0.5.9/skbio/util/_misc.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/util/_testing.py` & `scikit-bio-0.5.9/skbio/util/_testing.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/util/_warning.py` & `scikit-bio-0.5.9/skbio/util/_warning.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/util/tests/test_decorator.py` & `scikit-bio-0.5.9/skbio/util/tests/test_decorator.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/util/tests/test_misc.py` & `scikit-bio-0.5.9/skbio/util/tests/test_misc.py`

 * *Files 1% similar despite different names*

```diff
@@ -127,16 +127,14 @@
             return str(x)
 
         self.assertEqual(resolve_key(1, func), "1")
         self.assertEqual(resolve_key(4, func), "4")
 
     def test_index(self):
         class MetadataHaver(dict):
-            metadata = {}
-
             @property
             def metadata(self):
                 return self
 
         obj = MetadataHaver({'foo': 123})
         self.assertEqual(resolve_key(obj, 'foo'), 123)
```

### Comparing `scikit-bio-0.5.8/skbio/util/tests/test_testing.py` & `scikit-bio-0.5.9/skbio/util/tests/test_testing.py`

 * *Files identical despite different names*

### Comparing `scikit-bio-0.5.8/skbio/workflow.py` & `scikit-bio-0.5.9/skbio/workflow.py`

 * *Files identical despite different names*

